<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="make word cloud or correlation network from PubMed abstracts using easyPubMed"><title>wcAbst — wcAbst • wcGeneSummary</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="wcAbst — wcAbst"><meta property="og:description" content="make word cloud or correlation network from PubMed abstracts using easyPubMed"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">wcGeneSummary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/basic_usage_of_wcGeneSummary.html">wcGeneSummary</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>wcAbst</h1>
      
      <div class="d-none name"><code>wcAbst.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>make word cloud or correlation network from PubMed abstracts using easyPubMed</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">wcAbst</span><span class="op">(</span></span>
<span>  <span class="va">queries</span>,</span>
<span>  redo <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  madeUpper <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dna"</span>, <span class="st">"rna"</span><span class="op">)</span>,</span>
<span>  target <span class="op">=</span> <span class="st">"abstract"</span>,</span>
<span>  usefil <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  filnum <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  geneUpper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  apiKey <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pal <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>  numWords <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  scaleRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span>,</span>
<span>  showLegend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plotType <span class="op">=</span> <span class="st">"wc"</span>,</span>
<span>  colorText <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  corThresh <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  layout <span class="op">=</span> <span class="st">"nicely"</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  edgeLabel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  edgeLink <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ngram <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  genePlot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  deleteZeroDeg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  additionalRemove <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>queries</dt>
<dd><p>gene symbols (max: 5)</p></dd>


<dt>redo</dt>
<dd><p>if plot in other parameters, input the previous list</p></dd>


<dt>madeUpper</dt>
<dd><p>make the words uppercase in resulting plot</p></dd>


<dt>target</dt>
<dd><p>"abstract" or "title"</p></dd>


<dt>usefil</dt>
<dd><p>filter based on "gstfidf" or "bsdbtfidf"</p></dd>


<dt>filnum</dt>
<dd><p>specify filter tfidf</p></dd>


<dt>geneUpper</dt>
<dd><p>make queries uppercase</p></dd>


<dt>apiKey</dt>
<dd><p>api key for eutilities</p></dd>


<dt>pal</dt>
<dd><p>palette for color gradient in correlation network</p></dd>


<dt>numWords</dt>
<dd><p>the number of words to be shown</p></dd>


<dt>scaleRange</dt>
<dd><p>scale for label and node size in correlation network</p></dd>


<dt>showLegend</dt>
<dd><p>whether to show legend in correlation network</p></dd>


<dt>plotType</dt>
<dd><p>"wc" or "network"</p></dd>


<dt>colorText</dt>
<dd><p>color text label based on frequency in correlation network</p></dd>


<dt>corThresh</dt>
<dd><p>the correlation threshold</p></dd>


<dt>layout</dt>
<dd><p>the layout for correlation network, defaul to "nicely"</p></dd>


<dt>tag</dt>
<dd><p>cluster the words based on text</p></dd>


<dt>edgeLabel</dt>
<dd><p>if TRUE, plot the edge label (default: FALSE)</p></dd>


<dt>edgeLink</dt>
<dd><p>if FALSE, use geom_edge_diagonal</p></dd>


<dt>ngram</dt>
<dd><p>default to NA (1)</p></dd>


<dt>genePlot</dt>
<dd><p>plot associated genes (default: FALSE)</p></dd>


<dt>deleteZeroDeg</dt>
<dd><p>delete zero degree node from plot in correlation network</p></dd>


<dt>additionalRemove</dt>
<dd><p>specific words to be excluded</p></dd>


<dt>...</dt>
<dd><p>parameters to pass to wordcloud()</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>list of data frame and ggplot2 object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">wcAbst</span><span class="op">(</span><span class="st">"DDX41"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> querying pubmed for DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> converting to a data frame ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rawdf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         pmid                                      doi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   36253799               10.1186/s40164-022-00328-4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   36237573                10.3389/fmolb.2022.954087</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   36229594               10.1038/s41375-022-01708-9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   36185231                 10.3389/fonc.2022.997530</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   36165492                 10.1152/ajpgi.00104.2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   36119490                 10.3389/fonc.2022.992340</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7   36037623            10.1016/j.leukres.2022.106931</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   36036093                  10.1002/1873-3468.14487</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9   36029113                        10.1111/bjh.18433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10  36006669                     10.1099/jgv.0.001766</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11  36001442         10.1182/bloodadvances.2022008172</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  35987913               10.1038/s41409-022-01776-6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  35980681                 10.1182/blood.2022016598</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  35969835                 10.1182/blood.2022015790</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  35918249            10.1016/j.urolonc.2022.07.001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  35914158                  10.1073/pnas.2119514119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17  35885562              10.3390/diagnostics12071658</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18  35844724                         10.1002/jha2.256</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19  35781188               10.1007/s11899-022-00667-3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20  35712074               10.1016/j.isci.2022.104404</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21  35671390                 10.1182/blood.2021015135</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22  35613581             10.1016/j.celrep.2022.110856</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23  35612271                        10.1111/bjh.18273</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24  35458396                        10.3390/v14040666</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25  35443031                 10.1182/blood.2021015328</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26  35356204                 10.3389/fonc.2022.813149</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27  35338116               10.1038/s41419-022-04664-5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28  35303436             10.1016/j.stemcr.2022.02.010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29  35246110               10.1186/s12920-022-01191-2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30  35022662             10.1093/rheumatology/keac016</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31  34929216           10.1016/j.virusres.2021.198658</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32  34916496               10.1038/s41467-021-27530-y</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33  34908463                10.1128/spectrum.02254-21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34  34876838                     10.2147/IJGM.S341175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35  34793200                     10.1200/JCO.21.02286</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36  34784413                10.5858/arpa.2021-0124-OA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37  34742679                10.1016/j.cca.2021.11.003</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38  34680866                    10.3390/genes12101471</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39  34671111               10.1038/s41375-021-01404-0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40  34644397         10.1182/bloodadvances.2021005738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41  34641967               10.1186/s13053-021-00200-2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42  34482403                 10.1182/blood.2021011354</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43  34473945               10.1016/j.stem.2021.08.004</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44  34422009                10.3389/fgene.2021.707305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45  34390506                        10.1002/ajh.26321</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46  34349893                10.1177/20406207211032433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47  33967153                 10.11406/rinketsu.62.278</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48  33929502                     10.1093/ajcp/aqab027</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49  33850299               10.1038/s41375-021-01246-w</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50  33836623            10.1080/10428194.2021.1910689</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51  33763998             10.3324/haematol.2020.257246</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52  33711218                    10.1515/hsz-2020-0367</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53  33692849                    10.3892/ol.2021.12578</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54  33689767             10.1016/j.devcel.2021.02.014</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55  33686252               10.1038/s41409-021-01214-z</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56  33651979             10.1016/j.devcel.2021.02.006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57  33650667                     10.3892/or.2021.7951</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58  33626862             10.3324/haematol.2020.270553</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59  33585199                 10.3389/fonc.2020.582213</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60  33533142                  10.1002/1878-0261.12921</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61  33518151                10.1016/j.psj.2020.11.057</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62  33329609                10.3389/fimmu.2020.613039</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63  33328584               10.1038/s41568-020-00315-z</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64  33190445 10.3760/cma.j.issn.0253-2727.2020.10.012</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65  33170089                    10.1089/vim.2020.0096</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66  33038986               10.1016/j.beha.2020.101197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67  33035329         10.1182/bloodadvances.2020001721</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68  33023222                     10.3390/ijms21197289</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69  33006365                      10.1042/BSR20202603</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70  32965163            10.1080/15476286.2020.1822636</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71  32868804               10.1038/s41598-020-71386-z</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72  32781207                10.1016/j.fsi.2020.08.005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73  32619037                        10.1111/cas.14552</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74  32571604             10.1016/j.mayocp.2019.12.013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75  32307695                        10.1111/bjh.16668</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76  32247047                10.1016/j.fsi.2020.03.064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77  32081808                10.1016/j.fsi.2020.02.031</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78  32060018              10.1534/genetics.119.302973</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79  32047491                 10.3389/fimmu.2019.03022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80  32027562               10.1152/physiol.00022.2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81  31949493                        10.7150/jca.34188</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82  31755557                        10.1111/imm.13157</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83  31713024               10.1007/s12185-019-02770-3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84  31698430                 10.1182/blood.2019002982</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85  31655218            10.1016/j.micpath.2019.103816</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86  31604929               10.1038/s41467-019-12632-5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87  31552028                 10.3389/fimmu.2019.02069</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88  31484648                 10.1182/blood.2019000909</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89  31443275                        10.3390/v11080756</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90  31400013                        10.1002/ajh.25610</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91  31391370                 10.11406/rinketsu.60.800</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92  31309983                      10.1093/ajcp/aqz067</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93  31281151                 10.11406/rinketsu.60.600</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94  31203998               10.1016/j.beha.2019.05.001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95  31093808               10.1007/s10147-019-01462-6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96  31082519                10.1016/j.fsi.2019.05.019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97  30963592                        10.1002/ajh.25486</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98  30926971                10.1038/s41588-019-0375-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99  30466750               10.1016/j.beha.2018.09.014</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100 30407884            10.1080/10428194.2018.1522443</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 101 30253386                        10.1159/000492679</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 102 30144193                        10.1002/gcc.22680</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 103 30025984                10.1016/j.dci.2018.07.015</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 104 29942316                 10.3389/fimmu.2018.01327</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 105 29930723                       10.7150/thno.24699</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 106 29871919                    10.1128/mBio.00923-18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 107 29365323         10.1182/bloodadvances.2017013037</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 29329019               10.1016/j.etap.2018.01.001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 109 29248385                10.1016/j.dci.2017.12.013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 110 28978821                10.11406/rinketsu.58.1828</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 111 28684273                10.1016/j.dci.2017.07.001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 112 28637623       10.1053/j.seminhematol.2017.04.007</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 113 28631176                10.1007/s12185-017-2284-3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 114 28602976             10.1016/j.canlet.2017.05.026</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 115 28547672                10.1007/s12185-017-2260-y</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 116 28426938                 10.1177/2472555217705952</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 117 28357685                10.1007/s12253-017-0216-4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 118 28348086                  10.1074/jbc.M116.772327</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 119 28275134                 10.4049/jimmunol.1601420</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 120 28194039                      10.1038/leu.2017.44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 121 28126619                10.1016/j.fsi.2017.01.031</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 122 28084564                10.1007/s13238-016-0361-7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123 28031539                10.18632/oncotarget.14130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 124 27928732                10.1007/s13238-016-0351-9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 125 27913495          10.1182/asheducation-2016.1.302</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126 27819178            10.1080/17474086.2016.1257936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127 27795557                     10.1038/leu.2016.310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 128 27721487                        10.1038/srep34756</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 129 27502187                10.1007/s13238-016-0303-4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 130 27384852                 10.11406/rinketsu.57.730</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 131 27384851                 10.11406/rinketsu.57.723</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 132 27248996                     10.3390/ijms17060838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 133 27210295               10.1016/j.clml.2016.04.001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 134 27174803             10.1016/j.exphem.2016.04.017</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 135 27133828                     10.1038/leu.2016.124</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 136 26944477             10.3324/haematol.2015.139790</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 137 26917736             10.1182/blood-2016-01-690909</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 138 26876264                   10.1556/650.2016.30375</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 139 26712909             10.1182/blood-2015-10-676098</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 140 26693794            10.3109/10428194.2015.1115041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 141 26382053                  10.1111/1348-0421.12322</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 142 25965566              10.1016/j.ccell.2015.04.013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 143 25920683              10.1016/j.ccell.2015.03.017</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 144 25907950                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145 25816774               10.1016/j.chom.2015.02.021</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 146 25776036                10.1016/j.fsi.2015.03.008</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 147 25704810             10.1016/j.celrep.2015.01.039</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 148 25625930             10.4161/21645515.2014.979640</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 149 25609843                 10.4049/jimmunol.1402066</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 150 25086295                10.1016/j.dci.2014.07.020</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 151 24727060            10.1016/j.jconrel.2014.04.004</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 152 24349538             10.1371/journal.pone.0083639</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 153 24198409                     10.1128/JVI.02702-13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 154 23846113                          10.1038/nri3479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 155 23567047             10.1016/j.bbagrm.2013.03.012</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 156 23222971                          10.1038/ni.2492</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 157 23160208                          10.1038/ni.2469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 158 23142775                          10.1038/ni.2460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 159 23015711                     10.1128/JVI.02296-12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 160 22730532                 10.4049/jimmunol.1102985</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 161 22345436                     10.1128/JVI.07071-11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 162 22237025                      10.1038/nature10738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 163 21934672                          10.1038/ni.2118</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164 21892174                          10.1038/ni.2091</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165 20941364             10.1371/journal.pone.0012791</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 166 15885226        10.1016/j.clinbiochem.2005.01.010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 167 15703824                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 168 15690390                        10.1002/jcp.20285</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                         title</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                                                                                  IPSS-M has greater survival predictive accuracy compared with IPSS-R in persons ≥ 60 years with myelodysplastic syndromes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                                                         Combining affinity purification and mass spectrometry to define the network of the nuclear proteins interacting with the N-terminal region of FMRP.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                                                                                     DDX41 coordinates RNA splicing and transcriptional elongation to prevent DNA replication stress in hematopoietic cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                                                                                                                         Adult-onset hereditary myeloid malignancy and allogeneic stem cell transplantation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                                               DNA sensor associated type I Interferon signalling is increased in ulcerative colitis and induces JAK-dependent inflammatory cell death in colonic organoids.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                                                                                                                    Unique role of DDX41, a DEAD-box type RNA helicase, in hematopoiesis and leukemogenesis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7                                                                                                                                               Comparison between GATA2 and DDX41-mutated myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8                                                                                                                   Clinical and mechanistic insights into the roles of DDX41 in haematological malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9                                                Successive relapses from donor and host cells in a patient with DEAD-box helicase 41 (DDX41)-associated myelodysplastic syndrome: The lessons to be learned.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10                                                                                                           DExH/D-box helicases at the frontline of intrinsic and innate immunity against viral infections.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11                                                                                                      Allogeneic hematopoietic stem cell transplant outcomes in adults with inherited myeloid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12                                                                                      Outcomes of allogeneic transplant in patients with DDX41 mutated myelodysplastic syndrome and acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13                                                                                                                                                                  DDX41: the poster child for familial AML.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14                                                                                                                   Germline predisposition variants occur in myelodysplastic syndrome patients of all ages.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15                                                           DDX41 expression is associated with tumor necrosis in clear cell renal cell carcinoma and in cooperation with VHL loss leads to worse prognosis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16                                                                                                                              CFI-402257, a TTK inhibitor, effectively suppresses hepatocellular carcinoma.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17                                                           Cytogenetic and Genetic Abnormalities with Diagnostic Value in Myelodysplastic Syndromes (MDS): Focus on the Pre-Messenger RNA Splicing Process.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18                                                                             Next-generation sequencing reveals the presence of &lt;i&gt;DDX41&lt;/i&gt; mutations in acute lymphoblastic leukemia and aplastic anemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19                                                                                                                                 Germline and Somatic Defects in DDX41 and its Impact on Myeloid Neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20                                                                                   cGAS and DDX41-STING mediated intrinsic immunity spreads intercellularly to promote neuroinflammation in SOD1 ALS model.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21                                                                                                                        The genetic landscape of germline DDX41 variants predisposing to myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22                                                                                                                                   DDX41 is required for cGAS-STING activation against DNA virus infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23                                                               Clinical implications and genetic features of clonal cytopenia of undetermined significance compared to lower-risk myelodysplastic syndrome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24                                                                                                                                                        The Interplay between Viruses and Host DNA Sensors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25                                                                                  Prognostic impact of DDX41 germline mutations in intensively treated acute myeloid leukemia patients: an ALFA-FILO study.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26                                                                                                             Lessons From Pediatric MDS: Approaches to Germline Predisposition to Hematologic Malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27                                                                                                       STING agonist diABZI induces PANoptosis and DNA mediated acute respiratory distress syndrome (ARDS).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28                                                                                                                    DDX41 is needed for pre- and postnatal hematopoietic stem cell differentiation in mice.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29                                                                                                                         A novel bi-alleleic DDX41 mutations in B-cell lymphoblastic leukemia: case report.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30                                                                                                               Hyperresponsive cytosolic DNA-sensing pathway in monocytes from primary Sjögren's syndrome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31                                                                                                                                     DEAD/H-box helicases:Anti-viral and pro-viral roles during infections.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32                                                                                                    R-loop proximity proteomics identifies a role of DDX41 in transcription-associated genomic instability.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33                                                         Genome-Wide Characterization of Zebrafish Endogenous Retroviruses Reveals Unexpected Diversity in Genetic Organizations and Functional Potentials.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34                                                                                               Identification of an Innate Immune-Related Prognostic Signature in Early-Stage Lung Squamous Cell Carcinoma.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35                                                                                                                                   Donor Clonal Hematopoiesis and Recipient Outcomes After Transplantation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36                                                                                                                                                                             Designing Myeloid Gene Panels.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37                                                                          The role of cyclic GMP-AMP synthase and Interferon-I-inducible protein 16 as candidatebiomarkers of systemic lupus erythematosus.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38                                                                                                                                                               DEAD-Box RNA Helicases and Genome Stability.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39                                                                        AML with germline DDX41 variants is a clinicopathologically distinct entity with an indolent clinical course and favorable outcome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40                                                                                            Genetic features and clinical outcomes of patients with isolated and comutated DDX41-mutated myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41                                                                                                                                      Germline mutations among Polish patients with acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42                                                                                    Identification and prioritization of myeloid malignancy germline variants in a large cohort of adult patients with AML.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43                                                                                                                               Germline DDX41 mutations cause ineffective hematopoiesis and myelodysplasia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44                                          Systematic Analysis Identifies a Specific RNA-Binding Protein-Related Gene Model for Prognostication and Risk-Adjustment in HBV-Related Hepatocellular Carcinoma.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45                                                                  Spectrum of hematological malignancies, clonal evolution and outcomes in 144 Mayo Clinic patients with germline predisposition syndromes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46                                                                                             Clinical features of &lt;i&gt;DDX41&lt;/i&gt; mutation-related diseases: a systematic review with individual patient data.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47                                                                                                                                             [Predisposition and progression of myelodysplastic syndromes].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48                                                                                                                                Clinical and Pathologic Spectrum of DDX41-Mutated Hematolymphoid Neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49                            Targeted gene panels identify a high frequency of pathogenic germline variants in patients diagnosed with a hematological malignancy and at least one other independent cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50                                                                                                                New dead/H-box helicase gene (ddx41) mutation in an Italian family with recurrent leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51                                                                                                              Ddx41 inhibition of DNA damage signaling permits erythroid progenitor expansion in zebrafish.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52                                                                                                               DDX41: a multifunctional DEAD-box protein involved in pre-mRNA splicing and innate immunity.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53                                                                                                                           New progress in the study of germline susceptibility genes of myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54                                                                                                                                           Ddx41 loss R-loops in cGAS to fuel inflammatory HSPC production.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55                                                                                                                Genetics of donor cell leukemia in acute myelogenous leukemia and myelodysplastic syndrome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56                                                                                                                    Excessive R-loops trigger an inflammatory cascade leading to increased HSPC production.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57                                                                                            DDX41 regulates the expression and alternative splicing of genes involved in tumorigenesis and immune response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58                                  Unique ethnic features of &lt;i&gt;DDX41&lt;/i&gt; mutations in patients with idiopathic cytopenia of undetermined significance, myelodysplastic syndrome, or acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59                                                                                      Next-Generation Sequencing of &lt;i&gt;DDX41&lt;/i&gt; in Myeloid Neoplasms Leads to Increased Detection of Germline Alterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60                                                                               Clinical utility of targeted next-generation sequencing for the diagnosis of myeloid neoplasms with germline predisposition.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61                                                            Fasting and overfeeding affect the expression of the immunity- or inflammation-related genes in the liver of poultry via endogenous retrovirus.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62                                                                                                                                Molecular and Structural Basis of DNA Sensors in Antiviral Innate Immunity.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63                                                                                                                             Advances in germline predisposition to acute leukaemias and myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64                                                                         [Lenalidomide for myelodysplastic syndrome with excess blasts with germline DDX41 mutation: a case report and literatures review].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65          Hepatitis B Virus Might Be Sensed by STING-Dependent DNA Sensors and Attenuates the Response of STING-Dependent DNA Sensing Pathway in Humans with Acute and Chronic Hepatitis B Virus Infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66                                                                         Germline predisposition in myeloid neoplasms: Unique genetic and clinical features of GATA2 deficiency and SAMD9/SAMD9L syndromes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67                                                                                    Telomere biology disorder prevalence and phenotypes in adults with familial hematologic and/or pulmonary presentations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68                                                                                                                                Cytosolic Sensors for Pathogenic Viral and Bacterial Nucleic Acids in Fish.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69                                                                             Identification of prognostic biomarkers and correlations with immune infiltrates among cGAS-STING in hepatocellular carcinoma.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70                                                                                                                                Identification of spliceosome components pivotal to breast cancer survival.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71                                                                                    Prevalence and clinical implications of germline predisposition gene mutations in patients with acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72                                                                                                         Ctenopharyngodon idellus DDX41 initiates IFN I and ISG15 expression in response to GCRV infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73                                                                                  Clinical utility of target capture-based panel sequencing in hematological malignancies: A multicenter feasibility study.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74                                                                                                            Hereditary Predisposition to Hematopoietic Neoplasms: When Bloodline Matters for Blood Cancers.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75                                                                                                                         Molecular and clinical features of myeloid neoplasms with somatic DDX41 mutations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76                                                                                                                                     Immune sensing of DNA and strategies for fish DNA vaccine development.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77                                                                                                                              Molecular and functional characterization of tilapia DDX41 in IFN regulation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78                                                       Insights into the Involvement of Spliceosomal Mutations in Myelodysplastic Disorders from Analysis of SACY-1/DDX41 in &lt;i&gt;Caenorhabditis elegans&lt;/i&gt;.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79                                                                                              Identification of Novel Genetic Variants in CVID Patients With Autoimmunity, Autoinflammation, or Malignancy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80                                                                                                                                                                 DNA Sensing in the Innate Immune Response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81                                                    DNA sensors, crucial receptors to resist pathogens, are deregulated in colorectal cancer and associated with initiation and progression of the disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82                                                                                                                         TRIM21 controls Toll-like receptor 2 responses in bone-marrow-derived macrophages.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83                                                                                                                                              Novel DDX41 variants in Thai patients with myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84                                                                                                                                                              Lifting the veil on germline DDX41 mutations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85                                                                        Duckling short beak and dwarfism syndrome virus infection activates host innate immune response involving both DNA and RNA sensors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86                                                                                                               Influenza A virus M2 protein triggers mitochondrial DNA-mediated antiviral immune responses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87  A Cytosolic Sensor, &lt;i&gt;Pm&lt;/i&gt;DDX41, Binds Double Stranded-DNA and Triggers the Activation of an Innate Antiviral Response in the Shrimp &lt;i&gt;Penaeus monodon&lt;/i&gt; via the STING-Dependent Signaling Pathway.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88                                                                                                                        Germline DDX41 mutations define a significant entity within adult MDS/AML patients.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89                                                                    Herpes Simplex Virus Type 1-Encoded miR-H2-3p Manipulates Cytosolic DNA-Stimulated Antiviral Innate Immune Response by Targeting DDX41.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90                                                                                                      Successful lenalidomide treatment in high risk myelodysplastic syndrome with germline DDX41 mutation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91                                                                                                                                            [Genetic defects of chromosome 5q and 7q in myeloid neoplasms].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92                                                                                                                                                       Germline Predisposition to Hematolymphoid Neoplasia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93                                                                                                                                  [Genomic aberrations in myelodysplastic syndromes and related disorders].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94                                                                                                                                                                           Hereditary myeloid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95                                                                                                                                                          Genetic abnormalities and pathophysiology of MDS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96                                                                                                                   Grouper DDX41 exerts antiviral activity against fish iridovirus and nodavirus infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97                                                                                                DDX41 mutations in myeloid neoplasms are associated with male gender, TP53 mutations and high-risk disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98                                                                                                                                      Genomic subtyping and therapeutic targeting of acute erythroleukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99                                                                                                                                Identifying patients with genetic predisposition to acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100                                                                             Novel germline missense DDX41 variant in a patient with an adult-onset myeloid neoplasm with excess blasts without dysplasia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 101                                                                                                              Cyclic Dimeric Guanosine Monophosphate: Activation and Inhibition of Innate Immune Response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 102                                                                                 Putative new childhood leukemia cancer predisposition syndrome caused by germline bi-allelic missense mutations in DDX41.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 103                                                                                                                                          Molecular cloning and functional characterization of duck DDX41.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 104                                                   Identification of DEAD-Box RNA Helicase DDX41 as a Trafficking Protein That Involves in Multiple Innate Immune Signaling Pathways in a Zebrafish Model.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 105                                                                       DNA damage predicts prognosis and treatment response in colorectal liver metastases superior to immunogenic cell death and T cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 106                                                                    DDX41 Recognizes RNA/DNA Retroviral Reverse Transcripts and Is Critical for &lt;i&gt;In Vivo&lt;/i&gt; Control of Murine Leukemia Virus Infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 107                                                                          Prognostic tumor sequencing panels frequently identify germ line variants associated with hereditary hematopoietic malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108                                                                                             Salidroside influences the cellular cross-talk of human fetal lung diploid fibroblasts: A proteomic approach.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 109                                                                                                    A cytosolic sensor, PmDDX41, mediates antiviral immune response in black tiger shrimp Penaeus monodon.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 110                                                                                                                                         Sequential acquisition of mutations in myelodysplastic syndromes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 111                                                                                                                     Chicken DNA virus sensor DDX41 activates IFN-β signaling pathway dependent on STING.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 112                                                                                                                                                                          DDX41-related myeloid neoplasia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 113                                                                                   Familial predisposition of myeloid malignancies: biological and clinical significance of recurrent germ line mutations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 114                                                                                                                                     STING signaling in tumorigenesis and cancer therapy: A friend or foe?</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 115                                                                                                                                                           Myeloid neoplasms with germline DDX41 mutation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 116                                                                                                                              High-Throughput Screening to Identify Inhibitors of DEAD Box Helicase DDX41.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 117                                                                                                                                            Familial Acute Myeloid Leukemia and Myelodysplasia in Hungary.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 118                                                                                                         The DEAD-box RNA helicase DDX41 is a novel repressor of p21&lt;sup&gt;WAF1/CIP1&lt;/sup&gt; mRNA translation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 119      The Complement Anaphylatoxins C5a and C3a Suppress IFN-β Production in Response to &lt;i&gt;Listeria monocytogenes&lt;/i&gt; by Inhibition of the Cyclic Dinucleotide-Activated Cytosolic Surveillance Pathway.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 120                                                         Donor cell leukemia arising from preleukemic clones with a novel germline DDX41 mutation after allogenic hematopoietic stem cell transplantation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 121                                                                Enhancement of glycoprotein-based DNA vaccine for viral hemorrhagic septicemia virus (VHSV) via addition of the molecular adjuvant, DDX41.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 122                                                                                                                                Erratum to: Structural and functional analyses of human DDX41 DEAD domain.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123                                                                                                                                          Recurrent genetic defects on chromosome 5q in myeloid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 124                                                                                                                                            Structural and functional analyses of human DDX41 DEAD domain.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 125                                                                                                                                                            Germ line mutations associated with leukemias.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126                                                                                             Myelodysplastic syndromes and acute leukemia with genetic predispositions: a new challenge for hematologists.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127                                                                    Re-emergence of acute myeloid leukemia in donor cells following allogeneic transplantation in a family with a germline DDX41 mutation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 128                                                                                                Structural and Functional Analysis of DDX41: a bispecific immune receptor for DNA and cyclic dinucleotide.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 129                                                                                                                                         The emerging roles of the DDX41 protein in immunity and diseases.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 130                                                                                                                                                 Genetic predisposition to pediatric myeloid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 131                                                                                                                                                            Clonal evolution in myelodysplastic syndromes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 132                                                                                                                                                   Hereditary Predispositions to Myelodysplastic Syndrome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 133                                              Evaluation of Patients and Families With Concern for Predispositions to Hematologic Malignancies Within the Hereditary Hematologic Malignancy Clinic (HHMC).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 134                                                                                                                      Biological implications of somatic DDX41 p.R525H mutation in acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 135                                                                                                   Germline heterozygous DDX41 variants in a subset of familial myelodysplasia and acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 136                                                                                                      Two novel germline DDX41 mutations in a family with inherited myelodysplasia/acute myeloid leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 137                                                                                                                                                         Inherited DDX41 mutations: 11 genes and counting.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 138                                                                                                                  [Clinical and genetic background of familial myelodysplasia and acute myeloid leukemia].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 139                                                                                              Novel germ line DDX41 mutations define families with a lower age of MDS/AML onset and lymphoid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 140                                                                                               Genetic predisposition to myelodysplastic syndrome and acute myeloid leukemia in children and young adults.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 141                                                                                                                     Evolution of the DEAD box helicase family in chicken: chickens have no DHX9 ortholog.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 142                                                                                                               Functionally relevant RNA helicase mutations in familial and sporadic myeloid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 143                                                                                                                                              Inherited and Somatic Defects in DDX41 in Myeloid Neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 144                                                                                                                                    [Expression of helicase DDX41 in human dental pulp tissues and cells].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145                                                                                                                            Nucleic acid recognition orchestrates the anti-viral response to retroviruses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 146              The cytosolic sensor, DDX41, activates antiviral and inflammatory immunity in response to stimulation with double-stranded DNA adherent cells of the olive flounder, Paralichthys olivaceus.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 147                                                                        Bruton's tyrosine kinase phosphorylates DDX41 and activates its binding of dsDNA and STING to initiate type 1 interferon response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 148                                                                                                                  Mechanisms and pathways of innate immune activation and regulation in health and cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 149                                                                                                               Positive feedback regulation of type I IFN production by the IFN-inducible DNA sensor cGAS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 150                                                                                           Molecular cloning and functional characterization of porcine DEAD (Asp-Glu-Ala-Asp) box polypeptide 41 (DDX41).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 151                                                                                                                   A new adjuvant delivery system 'cyclic di-GMP/YSK05 liposome' for cancer immunotherapy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 152                                                                                                         Cell type-specific subcellular localization of phospho-TBK1 in response to cytoplasmic viral DNA.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 153                                                                                                                                          Adenovirus detection by the cGAS/STING/TBK1 DNA sensing cascade.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 154                                                                                                                    Newly described pattern recognition receptors team up against intracellular pathogens.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 155                                                                                     DExD/H-box RNA helicases as mediators of anti-viral innate immunity and essential host factors for viral replication.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 156                                                                                      The E3 ubiquitin ligase TRIM21 negatively regulates the innate immune response to intracellular double-stranded DNA.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 157                                                                                                                                   Innate sensing of bacterial cyclic dinucleotides: more than just STING.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 158                                                         The helicase DDX41 recognizes the bacterial secondary messengers cyclic di-GMP and cyclic di-AMP to activate a type I interferon immune response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 159                                                                                               Cell-specific regulation of nucleic acid sensor cascades: a controlling interest in the antiviral response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 160                                                          IFN-β expression is directly activated in human neutrophils transfected with plasmid DNA and is further increased via TLR-4-mediated signaling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 161                                                                                                            Sensing adenovirus infection: activation of interferon regulatory factor 3 in RAW 264.7 cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 162                                                                                                                 Clonal evolution in relapsed acute myeloid leukaemia revealed by whole-genome sequencing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 163                                                                                                                                                                                STING-dependent signaling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164                                                                                                             The helicase DDX41 senses intracellular DNA mediated by the adaptor STING in dendritic cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165                                                                                                                                          Comparative structural analysis of human DEAD-box RNA helicases.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 166                                                                                                                                                             RNA helicases: regulators of differentiation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 167                                                                                                                         Stimulation of transforming activity of DJ-1 by Abstrakt, a DJ-1-binding protein.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 168                                                                                                                                  Abstrakt interacts with and regulates the expression of sorting nexin-2.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      abstract</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   There are considerable new data on mutation topography in persons with myelodysplastic syndromes (MDS). These data have been used to update conventional risk models such as the Revised International Prognostic Scoring System (IPSS-R). Whether the molecular IPSS (IPSS-M) which includes these data improves survival prediction accuracy is untested. To answer this question, we compared survival prediction accuracies of the IPSS-R and IPSS-M in 852 consecutive subjects with de novo MDS. Concordance statistics (C-statistics) of the IPSS-R and IPSS-M in the entire cohort were similar, 0.67 (95% Confidence Interval [CI] 0.64, 0.71) and 0.68 (0.64, 0.71). Average numbers of mutations and of IPSS-M related mutations were greater in persons ≥ 60 years (2.0 [Interquartile Range [IQR], 1, 3] vs. 1.6 [0, 2], P = 0.003; 1.6 [0, 2] vs. 1.3 [0, 2], P = 0.006). Subjects ≥ 60 years had a higher incidence of mutations in RUNX1, TP53, TET2, SRSF2, DNMT3A, STAG2, EZH2 and DDX41. In contrast, mutations in U2AF1 were more common in persons &amp;lt; 60 years. Next we tested survival prediction accuracy based on age &amp;lt; or ≥ 60 years. C-statistics of the IPSS-R and IPSS-M in subjects ≥ 60 years were 0.66 (0.61, 0.71) and 0.69 (0.64, 0.73) whereas in subjects &amp;lt; 60 years they were 0.67 (0.61, 0.72) and 0.65 (0.59, 0.71). These data indicate an advantage for the IPSS-M over the IPSS-R in subjects ≥ 60 years but not in those &amp;lt; 60 years probably because of a great frequency of mutations correlated with survival in those ≥ 60 years.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Fragile X-Syndrome (FXS) represents the most common inherited form of intellectual disability and the leading monogenic cause of Autism Spectrum Disorders. In most cases, this disease results from the absence of expression of the protein FMRP encoded by the &lt;i&gt;FMR1&lt;/i&gt; gene (Fragile X messenger ribonucleoprotein 1). FMRP is mainly defined as a cytoplasmic RNA-binding protein regulating the local translation of thousands of target mRNAs. Interestingly, FMRP is also able to shuttle between the nucleus and the cytoplasm. However, to date, its roles in the nucleus of mammalian neurons are just emerging. To broaden our insight into the contribution of nuclear FMRP in mammalian neuronal physiology, we identified here a nuclear interactome of the protein by combining subcellular fractionation of rat forebrains with pull- down affinity purification and mass spectrometry analysis. By this approach, we listed 55 candidate nuclear partners. This interactome includes known nuclear FMRP-binding proteins as Adar or Rbm14 as well as several novel candidates, notably Ddx41, Poldip3, or Hnrnpa3 that we further validated by target-specific approaches. Through our approach, we identified factors involved in different steps of mRNA biogenesis, as transcription, splicing, editing or nuclear export, revealing a potential central regulatory function of FMRP in the biogenesis of its target mRNAs. Therefore, our work considerably enlarges the nuclear proteins interaction network of FMRP in mammalian neurons and lays the basis for exciting future mechanistic studies deepening the roles of nuclear FMRP in neuronal physiology and the etiology of the FXS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Myeloid malignancies with DDX41 mutations are often associated with bone marrow failure and cytopenia before overt disease manifestation. However, the mechanisms underlying these specific conditions remain elusive. Here, we demonstrate that loss of DDX41 function impairs efficient RNA splicing, resulting in DNA replication stress with excess R-loop formation. Mechanistically, DDX41 binds to the 5' splice site (5'SS) of coding RNA and coordinates RNA splicing and transcriptional elongation; loss of DDX41 prevents splicing-coupled transient pausing of RNA polymerase II at 5'SS, causing aberrant R-loop formation and transcription-replication collisions. Although the degree of DNA replication stress acquired in S phase is small, cells undergo mitosis with under-replicated DNA being remained, resulting in micronuclei formation and significant DNA damage, thus leading to impaired cell proliferation and genomic instability. These processes may be responsible for disease phenotypes associated with DDX41 mutations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Hereditary myeloid malignancies, especially in adults or elderly persons, had been considered quite rare before the next-generation sequencing era; however, increased usage of clinical sequencing has revealed much higher prevalence of inherited myeloid malignancies. &lt;i&gt;DDX41&lt;/i&gt; and various pathogenic germline mutations have newly been recognized as the cause of adult-onset familial leukemia and myeloid malignancies. Although germline predisposition to myeloid neoplasms had been categorized as a provisional entity in the World Health Organization classification of hematopoietic neoplasms in 2016, methodology for the identification of hereditary myeloid malignancies has not been fully established yet. In addition, many unresolved problems, such as epidemiology, the exact pathogenic mechanisms, and ideal treatment strategy, including indications of allogeneic hematopoietic stem cell transplantation, still remain. Related donor selection for stem cell transplant is a particularly sensitive issue due to the possibility of germline mutation of the candidate relatives and the risk of donor cell leukemia after transplantation. Here, we reviewed the current evidence regarding epidemiology, diagnosis, mechanisms of progression, and transplantation strategy for hereditary myeloid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                DNA sensor pathways can initiate inflammasome, cell death and type I interferon (IFN) signalling in immune-mediated inflammatory diseases (IMIDs); including type I interferonopathies. We investigated the involvement of these pathways in the pathogenesis of ulcerative colitis (UC); by analysing expression of DNA sensor, inflammasome, and type I IFN biomarker genes in colonic mucosal biopsy tissue from control (n=31), inactive UC (n=31), active UC (n=33) and a UC single cell RNA-Seq dataset. The effects of type I IFN (IFN-β), IFN-γ and TNF-α on gene expression, cytokine production and cell death were investigated in human colonic organoids. In organoids treated with cytokines alone, or in combination with NLRP3, caspase or JAK inhibitors, cell death was measured, and supernatants were assayed for IL-1β/IL-18/CXCL10. The expression of DNA sensor pathway genes - PYHIN family members (AIM2, IFI16, MNDA, PYHIN1), as well as ZBP1, cGAS and DDX41 were increased in active UC and expressed in a cell type restricted pattern. Inflammasome genes (CASP1, IL1B, IL18), type I IFN inducers (STING, TBK1, IRF3), IFNB1 and type I IFN biomarker genes (OAS2, IFIT2, MX2) were also increased in active UC. Co-treatment of organoids with IFN-β or IFN-γ and TNFα increased expression of IFI16, ZBP1, CASP1, cGAS and STING, induced cell death and IL-1β/IL-18 secretion. This inflammatory cell death was blocked by the JAK inhibitor tofacitinib but not by inflammasome or caspase inhibitors. Increased type I IFN activity may drive elevated expression of DNA sensor genes and JAK-dependent but inflammasome-independent inflammatory cell death of colonic epithelial cells in UC.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        In myeloid malignancies including acute myeloid leukemia (AML) and myelodysplastic syndromes (MDS), patient selection and therapeutic strategies are increasingly based on tumor-specific genetic mutations. Among these, mutations in &lt;i&gt;DDX41&lt;/i&gt;, which encodes a DEAD-box type RNA helicase, are present in approximately 2-5% of AML and MDS patients; this disease subtype exhibits a distinctive disease phenotype characterized by late age of onset, tendency toward cytopenia in the peripheral blood and bone marrow, a relatively favorable prognosis, and a high frequency of normal karyotypes. Typically, individuals with a loss-of-function germline &lt;i&gt;DDX41&lt;/i&gt; variant in one allele later acquire the p.R525H mutation in the other allele before overt disease manifestation, suggesting that the progressive decrease in DDX41 expression and/or function is involved in myeloid leukemogenesis.RNA helicases play roles in many processes involving RNA metabolism by altering RNA structure and RNA-protein interactions through ATP-dependent helicase activity. A single RNA helicase can play multiple cellular roles, making it difficult to elucidate the mechanisms by which mutations in &lt;i&gt;DDX41&lt;/i&gt; are involved in leukemogenesis. Nevertheless, multiple DDX41 functions have been associated with disease development. The enzyme has been implicated in the regulation of RNA splicing, nucleic acid sensing in the cytoplasm, R-loop resolution, and snoRNA processing.Most of the mutated RNA splicing-related factors in MDS are involved in the recognition and determination of 3' splice sites (SS), although their individual roles are distinct. On the other hand, DDX41 is likely incorporated into the C complex of the spliceosome, which may define a distinctive disease phenotype. This review summarizes the current understanding of how DDX41 is involved in this unique myeloid malignancy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            DEAD-box Helicase 41 (DDX41) is a member of the DExD/H-box helicase family that has a variety of cellular functions. Of note, germline and somatic mutations in the DDX41 gene are prevalently found in myeloid malignancies. Here, we present a comprehensive and analytic review covering relevant clinical, translational and basic science findings on DDX41. We first describe the initial characterisation of DDX41 mutations in patients affected by myelodysplastic syndromes, their associated clinical characteristics, and current treatment modalities. We then cover the known cellular functions of DDX41, spanning from its discovery in Drosophila as a neuroregulator through its more recently described roles in inflammatory signalling, R-loop metabolism and snoRNA processing. We end with a summary of the identified basic functions of DDX41 that when perturbed may contribute to the underlying pathology of haematologic neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              DExH/D-box helicases are essential nucleic acid and ribonucleoprotein remodelers involved in all aspects of nucleic acid metabolism including replication, gene expression and post-transcriptional modifications. In parallel to their importance in basic cellular functions, DExH/D-box helicases play multiple roles in viral life cycles, with some of them highjacked by viruses or negatively regulating innate immune activation. However, other DExH/D-box helicases have recurrently been highlighted as direct antiviral effectors or as positive regulators of innate immune activation. Innate immunity relies on the ability of Pathogen Recognition Receptors to recognize viral signatures and trigger the production of interferons (IFNs) and pro-inflammatory cytokines. Secreted IFNs interact with their receptors to establish antiviral cellular reprogramming via expression regulation of the interferon-stimulated genes (ISGs). Several DExH/D-box helicases have been reported to act as viral sensors (DDX3, DDX41, DHX9, DDX1/DDX21/DHX36 complex), and others to play roles in innate immune activation (DDX60, DDX60L, DDX23). In contrast, the DDX39A, DDX46, DDX5 and DDX24 helicases act as negative regulators and impede IFN production upon viral infection. Beyond their role in viral sensing, the ISGs DDX60 and DDX60L act as viral inhibitors. Interestingly, the constitutively expressed DEAD-box helicases DDX56, DDX17, DDX42 intrinsically restrict viral replication. Hence, DExH/D-box helicases appear to form a multilayer network of primary and secondary factors involved in both intrinsic and innate antiviral immunity. In this review, we highlight recent findings on the extent of antiviral defences played by helicases and emphasize the need to better understand their immune functions as well as their complex interplay.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    There is increasing recognition that pathogenic germline variants drive the development of hematopoietic cancers in many individuals. Currently, patients with hereditary hematologic malignancies (HHMs) receive similar standard therapies and hematopoietic stem cell transplant (HSCT) approaches as those with sporadic disease. We hypothesize that patients with myeloid malignancies and deleterious germline predisposition variants have different post-transplant outcomes compared to those without such alleles. We studied 472 patients with myeloid neoplasms, of whom 26% had deleterious germline variants (DGVs) and 34% underwent HSCT. DGVs in CHEK2 and DDX41 were most commonly seen in American and Australian cohorts, respectively. Patients with deleterious germline DDX41 variants had a higher incidence of severe (stage 3-4) acute graft versus host disease (GVHD) (38%) compared to recipients with deleterious CHEK2 variants (0%), other HHM variants (12%), or patients without such germline variants (9%) (p= 0.002). Importantly, the use of post-transplant cyclophosphamide reduced the risk of severe acute GVHD in patients receiving HSCT for deleterious germline DDX41-associated myeloid neoplasms (0% vs 53%, p=0.03). Based on these results, we advocate the use of post-transplant cyclophosphamide when individuals with deleterious germline DDX41 variants undergo allogeneic HSCT for myeloid malignancies, even when transplantation has been performed using wild-type donors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The frequency of pathogenic/likely pathogenic (P/LP) germline variants in myelodysplastic syndrome (MDS) patients diagnosed at or younger than 40 years old is 15 to 20%. However, there are no comprehensive studies assessing the frequency of such variants across the age spectrum. We performed augmented whole exome sequencing from peripheral blood of 404 MDS patients and their related donors prior to allogeneic hematopoietic stem cell transplantation. Single nucleotide and copy number variants in 233 genes were analyzed and interpreted. Germline status was established by the presence of a variant in the patient and related donor or for those seen previously only as germline alleles. We identified P/LP germline variants in 28 out of 404 MDS patients (7%) within all age deciles. Patients with P/LP variants were more likely to develop higher-grade MDS than those without (43% vs. 25%, p=0.04). There was no statistical difference in outcome parameters between patients with and without a germline variant, but the analysis was underpowered. P/LP variants in bone marrow failure syndrome genes were found in five patients under age 40, whereas variants in DDX41 (n=4), telomere biology disorder genes (n=2), and tumor predisposition genes (n=17) were found in patients over 40. If presumed germline variants were included, the yield of P/LP variants would increase to 11% and by adding suspicious variants of unknown significance, it would rise further to 12%. The high frequency of P/LP germline variants in our study supports comprehensive germline genetic testing for all MDS patients regardless of their age at diagnosis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Histologic tumor necrosis (TN) is a well-established independent prognostic indicator in patients treated surgically for clear cell renal cell carcinoma (ccRCC). However, the precise mechanisms by which TN alters disease progression remain unknown. The DEAD-box protein DDX41, a member of a large family of helicases, has been characterized as a pattern recognition receptor against an array of double-stranded (ds)DNA produced from bacteria, dsDNA viruses, and nearby cells that have released dsDNA fragments through necrosis. We hypothesized that DDX41 expression may be upregulated in ccRCC with TN, leading to worse prognosis. Relationship between the presence of TN and DDX41 expression were examined using The Cancer Genome Atlas data sets or using ccRCC samples in our institution. Further, the molecular functions of DDX41 were investigated with human ccRCC cells. The presence of TN was significantly associated with the upregulation of mRNA and protein expression of DDX41 in the 2different patient cohorts with ccRCC. In addition, the mRNA and protein expression levels of DDX41 revealed a worse prognosis. In vitro analyses with ccRCC cells revealed that DDX41 expression promotes tumor-promoting activity. Furthermore, VHL loss, 1of the most common features in ccRCC, was shown to play an extremely important role in increasing the expression of the CXCL family in DDX41-expressing ccRCC, leading to the acquisition of a worse malignant phenotype. DDX41 expression is associated with TN in ccRCC and leads to a worse prognosis in cooperation with VHL loss.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Deregulation of cell cycle is a typical feature of cancer cells. Normal cells rely on the strictly coordinated spindle assembly checkpoint (SAC) to maintain the genome integrity and survive. However, cancer cells could bypass this checkpoint mechanism. In this study, we showed the clinical relevance of threonine tyrosine kinase (TTK) protein kinase, a central regulator of the SAC, in hepatocellular carcinoma (HCC) and its potential as therapeutic target. Here, we reported that a newly developed, orally active small molecule inhibitor targeting TTK (CFI-402257) effectively suppressed HCC growth and induced highly aneuploid HCC cells, DNA damage, and micronuclei formation. We identified that CFI-402257 also induced cytosolic DNA, senescence-like response, and activated DDX41-STING cytosolic DNA sensing pathway to produce senescence-associated secretory phenotypes (SASPs) in HCC cells. These SASPs subsequently led to recruitment of different subsets of immune cells (natural killer cells, CD4&lt;sup&gt;+&lt;/sup&gt; T cells, and CD8&lt;sup&gt;+&lt;/sup&gt; T cells) for tumor clearance. Our mass cytometry data illustrated the dynamic changes in the tumor-infiltrating immune populations after treatment with CFI-402257. Further, CFI-402257 improved survival in HCC-bearing mice treated with anti-PD-1, suggesting the possibility of combination treatment with immune checkpoint inhibitors in HCC patients. In summary, our study characterized CFI-402257 as a potential therapeutic for HCC, both used as a single agent and in combination therapy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Myelodysplastic syndromes (MDS) are considered to be diseases associated with splicing defects. A large number of genes involved in the pre-messenger RNA splicing process are mutated in MDS. Deletion of 5q and 7q are of diagnostic value, and those chromosome regions bear the numbers of splicing genes potentially deleted in del(5q) and del(7q)/-7 MDS. In this review, we present the splicing genes already known or suspected to be implicated in MDS pathogenesis. First, we focus on the splicing genes located on chromosome 5 (&lt;i&gt;HNRNPA0&lt;/i&gt;, &lt;i&gt;RBM27&lt;/i&gt;, &lt;i&gt;RBM22&lt;/i&gt;, &lt;i&gt;SLU7&lt;/i&gt;, &lt;i&gt;DDX41&lt;/i&gt;), chromosome 7 (&lt;i&gt;LUC7L2&lt;/i&gt;), and on the &lt;i&gt;SF3B1&lt;/i&gt; gene since both chromosome aberrations and the &lt;i&gt;SF3B1&lt;/i&gt; mutation are the only genetic abnormalities in splicing genes with clear diagnostic values. Then, we present and discuss other splicing genes that are showing a prognostic interest (&lt;i&gt;SRSF2&lt;/i&gt;, &lt;i&gt;U2AF1&lt;/i&gt;, &lt;i&gt;ZRSR2&lt;/i&gt;, &lt;i&gt;U2AF2&lt;/i&gt;, and &lt;i&gt;PRPF8&lt;/i&gt;). Finally, we discuss the haploinsufficiency of splicing genes, especially from chromosomes 5 and 7, the important amplifier process of splicing defects, and the cumulative and synergistic effect of splicing genes defects in the MDS pathogenesis. At the time, when many authors suggest including the sequencing of some splicing genes to improve the diagnosis and the prognosis of MDS, a better understanding of these cooperative defects is needed.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Limited studies have been described DEAD-box helicase 41 (&lt;i&gt;DDX41)&lt;/i&gt; mutations in hematological diseases other than myeloid neoplasms. In this study, &lt;i&gt;DDX41&lt;/i&gt; mutations were identified in 0.8% of myeloid neoplasms, 0.9% of acute lymphoblastic leukemia (ALL), and 1.0% of aplastic anemia (AA). A total of 15 causal &lt;i&gt;DDX41&lt;/i&gt; variants in 14 patients were detected; seven of which have not been reported previously. In myeloid neoplasms, the median age of patients with germline missense was lower than that of germline nonsense mutations. In ALL, the characteristics of &lt;i&gt;DDX41&lt;/i&gt; mutation were distinct. This study first reported &lt;i&gt;DDX41&lt;/i&gt; mutations in ALL and AA, expanding its mutation and phenotypic spectrum.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            While DDX41 mutation (m) is one of the most prevalent predisposition genes in adult myelodysplastic syndrome (MDS)/acute myeloid leukemia (AML), most patients do not always present with a family history of MDS/AML. In this review, we will be highlighting epidemiological data on DDX41m, roles of DDX41 in oncogenesis, mechanisms of clonal evolution with somatic DDX41m, and clinical phenotypes and management of MDS/AML in patients harboring DDX41m. DDX41 encodes a DEAD-box helicase protein that is considered essential for cell growth and viability. High incidence of myeloid malignancies and other cancers in patients bearing DDX41m suggests that defects in DDX41 lead to loss of a tumor suppressor function, likely related to activities in RNA splicing and processing pathways. Seventy percent of cancer cases with DDX41m are associated with MDS/AML alone. More than 65% of familial cases harbor heterozygous germline frameshift mutations, of which p.D140Gfs*2 is the most common. A somatic DDX41m of the second allele is acquired in 70% of cases, leading to hematological malignancy. Myeloid neoplasms with DDX41m are typically characterized by long latency, high-risk disease at presentation with normal cytogenetics and without any additional molecular markers. Recent reports suggests that a subgroup of these patients have an indolent clinical course and have a better long-term survival compared to favorable or intermediate risk AML. Distinct clinical/pathologic features and favorable outcomes in MDS/AML highlight the need for standardized classification and gene specific guidelines that could assist in management decisions in patients with DDX41m.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Neuroinflammation exacerbates the progression of SOD1-driven amyotrophic lateral sclerosis (ALS), although the underlying mechanisms remain largely unknown. Herein, we demonstrate that misfolded SOD1 (SOD1&lt;sup&gt;Mut&lt;/sup&gt;)-causing ALS results in mitochondrial damage, thus triggering the release of mtDNA and an RNA:DNA hybrid into the cytosol in an mPTP-independent manner to activate IRF3- and IFNAR-dependent type I interferon (IFN-I) and interferon-stimulating genes. The neuronal hyper-IFN-I and pro-inflammatory responses triggered in ALS-SOD1&lt;sup&gt;Mut&lt;/sup&gt; were sufficiently robust to cause a strong physiological outcome &lt;i&gt;in vitro&lt;/i&gt; and &lt;i&gt;in vivo&lt;/i&gt;. cGAS/DDX41-STING-signaling is amplified in bystander cells through inter-neuronal gap junctions. Our results highlight the importance of a common DNA-sensing pathway between SOD1 and TDP-43 in influencing the progression of ALS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Germline DDX41 variants are the most common mutations predisposing to acute myeloid leukemia (AML)/myelodysplastic syndrome (MDS) in adults, but the causal variant (CV) landscape and clinical spectrum of hematologic malignancies (HMs) remain unexplored. Here, we analyzed the genomic profiles of 176 patients with HM carrying 82 distinct presumably germline DDX41 variants among a group of 9821 unrelated patients. Using our proposed DDX41-specific variant classification, we identified features distinguishing 116 patients with HM with CV from 60 patients with HM with variant of uncertain significance (VUS): an older age (median 69 years), male predominance (74% in CV vs 60% in VUS, P = .03), frequent concurrent somatic DDX41 variants (79% in CV vs 5% in VUS, P &amp;lt; .0001), a lower somatic mutation burden (1.4 ± 0.1 in CV vs 2.9 ± 0.04 in VUS, P = .012), near exclusion of canonical recurrent genetic abnormalities including mutations in NPM1, CEBPA, and FLT3 in AML, and favorable overall survival (OS) in patients with AML/MDS. This superior OS was determined independent of blast count, abnormal karyotypes, and concurrent variants, including TP53 in patients with AML/MDS, regardless of patient's sex, age, or specific germline CV, suggesting that germline DDX41 variants define a distinct clinical entity. Furthermore, unrelated patients with myeloproliferative neoplasm and B-cell lymphoma were linked by DDX41 CV, thus expanding the known disease spectrum. This study outlines the CV landscape, expands the phenotypic spectrum in unrelated DDX41-mutated patients, and underscores the urgent need for gene-specific diagnostic and clinical management guidelines.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Upon binding double-stranded DNA (dsDNA), cyclic GMP-AMP synthase (cGAS) is activated and initiates the cGAS-stimulator of IFN genes (STING)-type I interferon pathway. DEAD-box helicase 41 (DDX41) is a DEAD-box helicase, and mutations in DDX41 cause myelodysplastic syndromes (MDSs) and acute myeloid leukemia (AML). Here, we show that DDX41-knockout (KO) cells have reduced type I interferon production after DNA virus infection. Unexpectedly, activations of cGAS and STING are affected in DDX41 KO cells, suggesting that DDX41 functions upstream of cGAS. The recombinant DDX41 protein exhibits ATP-dependent DNA-unwinding activity and ATP-independent strand-annealing activity. The MDS/AML-derived mutant R525H has reduced unwinding activity but retains normal strand-annealing activity and stimulates greater cGAS dinucleotide-synthesis activity than wild-type DDX41. Overexpression of R525H in either DDX41-deficient or -proficient cells results in higher type I interferon production. Our results have led to the hypothesis that DDX41 utilizes its unwinding and annealing activities to regulate the homeostasis of dsDNA and single-stranded DNA (ssDNA), which, in turn, regulates cGAS-STING activation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Clonal cytopenia of undetermined significance (CCUS) is characterized by persistent cytopenias with genetic aberrations, which do not meet the diagnostic criteria for myelodysplastic syndrome (MDS). We aimed to compare the clinical and genetic characteristics of CCUS with lower-risk MDS and identify patients with CCUS with a high risk of progression. We performed targeted sequencing of bone marrow (BM) samples from patients with idiopathic cytopenia of undetermined significance (ICUS) (n = 139) and MDS (n = 226). Overall survival (OS) of patients with CCUS (n = 78) was worse than non-clonal ICUS (n = 61) and superior to lower-risk MDS (n = 99). Patients with CCUS showed similar characteristics to those with lower-risk MDS, except for higher haemoglobin, lower BM cellularity, and less frequent SF3B1 mutations. Lower haemoglobin, DDX41 (biallelic germline and somatic), ETV6, and RUNX1 mutations were independent prognostic factors for worse OS. Lower haemoglobin and DDX41 mutations were also associated with lower progression-free survival. Patients with CCUS with high-risk features showed similar or worse OS than patients with lower-risk MDS. Our findings suggest that patients with CCUS having certain clinical or genetic features should be regarded and treated as lower-risk MDS despite lacking significant dysplasia or MDS-associated chromosomal abnormalities.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                DNA virus infections are often lifelong and can cause serious diseases in their hosts. Their recognition by the sensors of the innate immune system represents the front line of host defence. Understanding the molecular mechanisms of innate immunity responses is an important prerequisite for the design of effective antivirotics. This review focuses on the present state of knowledge surrounding the mechanisms of viral DNA genome sensing and the main induced pathways of innate immunity responses. The studies that have been performed to date indicate that herpesviruses, adenoviruses, and polyomaviruses are sensed by various DNA sensors. In non-immune cells, STING pathways have been shown to be activated by cGAS, IFI16, DDX41, or DNA-PK. The activation of TLR9 has mainly been described in pDCs and in other immune cells. Importantly, studies on herpesviruses have unveiled novel participants (BRCA1, H2B, or DNA-PK) in the IFI16 sensing pathway. Polyomavirus studies have revealed that, in addition to viral DNA, micronuclei are released into the cytosol due to genotoxic stress. Papillomaviruses, HBV, and HIV have been shown to evade DNA sensing by sophisticated intracellular trafficking, unique cell tropism, and viral or cellular protein actions that prevent or block DNA sensing. Further research is required to fully understand the interplay between viruses and DNA sensors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       DDX41 germline mutations (DDX41MutGL) are the most common genetic predisposition to myelodysplastic syndrome and acute myeloid leukemia (AML). Recent reports suggest that DDX41MutGL myeloid malignancies could be considered as a distinct entity, even if their specific presentation and outcome remain to be defined. We describe here the clinical and biological features of 191 patients with DDX41MutGL AML. Baseline characteristics and outcome of 86 of these patients, treated with intensive chemotherapy in 5 prospective Acute Leukemia French Association/French Innovative Leukemia Organization trials, were compared with those of 1604 patients with DDX41 wild-type (DDX41WT) AML, representing a prevalence of 5%. Patients with DDX41MutGL AML were mostly male (75%), in their seventh decade, and with low leukocyte count (median, 2 × 109/L), low bone marrow blast infiltration (median, 33%), normal cytogenetics (75%), and few additional somatic mutations (median, 2). A second somatic DDX41 mutation (DDX41MutSom) was found in 82% of patients, and clonal architecture inference suggested that it could be the main driver for AML progression. DDX41MutGL patients displayed higher complete remission rates (94% vs 69%; P &amp;lt; .0001) and longer restricted mean overall survival censored at hematopoietic stem cell transplantation (HSCT) than 2017 European LeukemiaNet intermediate/adverse (Int/Adv) DDX41WT patients (5-year difference in restricted mean survival times, 13.6 months; P &amp;lt; .001). Relapse rates censored at HSCT were lower at 1 year in DDX41MutGL patients (15% vs 44%) but later increased to be similar to Int/Adv DDX41WT patients at 3 years (82% vs 75%). HSCT in first complete remission was associated with prolonged relapse-free survival (hazard ratio, 0.43; 95% confidence interval, 0.21-0.88; P = .02) but not with longer overall survival (hazard ratio, 0.77; 95% confidence interval, 0.35-1.68; P = .5).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Pediatric myelodysplastic syndromes (MDS) often raise concern for an underlying germline predisposition to hematologic malignancies, referred to as germline predisposition herein. With the availability of genetic testing, it is now clear that syndromic features may be lacking in patients with germline predisposition. Many genetic lesions underlying germline predisposition may also be mutated somatically in &lt;i&gt;de novo&lt;/i&gt; MDS and leukemias, making it critical to distinguish their germline origin. The verification of a suspected germline predisposition informs therapeutic considerations, guides monitoring pre- and post-treatment, and allows for family counseling. Presentation of MDS due to germline predisposition is not limited to children and spans a wide age range. In fact, the risk of MDS may increase with age in many germline predisposition conditions and can present in adults who lack classical stigmata in their childhood. Furthermore, germline predisposition associated with &lt;i&gt;DDX41&lt;/i&gt; mutations presents with older adult-onset MDS. Although a higher proportion of pediatric patients with MDS will have a germline predisposition, the greater number of MDS diagnoses in adult patients may result in a larger overall number of those with an underlying germline predisposition. In this review, we present a framework for the evaluation of germline predisposition to MDS across all ages. We discuss characteristics of personal and family history, clinical exam and laboratory findings, and integration of genetic sequencing results to assist in the diagnostic evaluation. We address the implications of a diagnosis of germline predisposition for the individual, for their care after MDS therapy, and for family members. Studies on MDS with germline predisposition have provided unique insights into the pathogenesis of hematologic malignancies and mechanisms of somatic genetic rescue &lt;i&gt;vs.&lt;/i&gt; disease progression. Increasing recognition in adult patients will inform medical management and may provide potential opportunities for the prevention or interception of malignancy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27  Stimulator of interferon genes (STING) contributes to immune responses against tumors and may control viral infection including SARS-CoV-2 infection. However, activation of the STING pathway by airway silica or smoke exposure leads to cell death, self-dsDNA release, and STING/type I IFN dependent acute lung inflammation/ARDS. The inflammatory response induced by a synthetic non-nucleotide-based diABZI STING agonist, in comparison to the natural cyclic dinucleotide cGAMP, is unknown. A low dose of diABZI (1 µg by endotracheal route for 3 consecutive days) triggered an acute neutrophilic inflammation, disruption of the respiratory barrier, DNA release with NET formation, PANoptosis cell death, and inflammatory cytokines with type I IFN dependent acute lung inflammation. Downstream upregulation of DNA sensors including cGAS, DDX41, IFI204, as well as NLRP3 and AIM2 inflammasomes, suggested a secondary inflammatory response to dsDNA as a danger signal. DNase I treatment, inhibition of NET formation together with an investigation in gene-deficient mice highlighted extracellular DNA and TLR9, but not cGAS, as central to diABZI-induced neutrophilic response. Therefore, activation of acute cell death with DNA release may lead to ARDS which may be modeled by diABZI. These results show that airway targeting by STING activator as a therapeutic strategy for infection may enhance lung inflammation with severe ARDS. STING agonist diABZI induces neutrophilic lung inflammation and PANoptosis A, Airway STING priming induce a neutrophilic lung inflammation with epithelial barrier damage, double-stranded DNA release in the bronchoalvelolar space, cell death, NETosis and type I interferon release. B, 1. The diamidobenzimidazole (diABZI), a STING agonist is internalized into the cytoplasm through unknown receptor and induce the activation and dimerization of STING followed by TBK1/IRF3 phosporylation leading to type I IFN response. STING activation also leads to NF-kB activation and the production of pro-inflammatory cytokines TNFα and IL-6. 2. The activation of TNFR1 and IFNAR1 signaling pathway results in ZBP1 and RIPK3/ASC/CASP8 activation leading to MLKL phosphorylation and necroptosis induction. 3. This can also leads to Caspase-3 cleavage and apoptosis induction. 4. Self-dsDNA or mtDNA sensing by NLRP3 or AIM2 induces inflammsome formation leading to Gasdermin D cleavage enabling Gasdermin D pore formation and the release mature IL-1β and pyroptosis. NLRP3 inflammasome formation can be enhanced by the ZBP1/RIPK3/CASP8 complex. 5. A second signal of STING activation with diABZI induces cell death and the release of self-DNA which is sensed by cGAS and form 2'3'-cGAMP leading to STING hyper activation, the amplification of TBK1/IRF3 and NF-kB pathway and the subsequent production of IFN-I and inflammatory TNFα and IL-6. This also leads to IFI204 and DDX41 upregulation thus, amplifying the inflammatory loop. The upregulation of apoptosis, pyroptosis and necroptosis is indicative of STING-dependent PANoptosis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               DDX41 is a tumor suppressor frequently mutated in human myeloid neoplasms, but whether it affects hematopoiesis is unknown. Using a knockout mouse, we demonstrate that DDX41 is required for mouse hematopoietic stem and progenitor cell (HSPC) survival and differentiation, particularly of myeloid lineage cells. Transplantation of Ddx41 knockout fetal liver and adult bone marrow (BM) cells was unable to rescue mice from lethal irradiation, and knockout stem cells were also defective in colony formation assays. RNA-seq analysis of Lin&lt;sup&gt;-&lt;/sup&gt;/cKit&lt;sup&gt;+&lt;/sup&gt;/Sca1&lt;sup&gt;+&lt;/sup&gt;Ddx41 knockout cells from fetal liver demonstrated that the expression of many genes associated with hematopoietic differentiation were altered. Furthermore, differential splicing of genes involved in key biological processes was observed. Our data reveal a critical role for DDX41 in HSPC differentiation and myeloid progenitor development, likely through regulating gene expression programs and splicing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        The germline mutations of DDX41, also known as DEAD box RNA helicase 41, have been found in about 1.5% of myeloid neoplasms (MNs). Development of MDS/AML is relatively common in germline DDX41 mutations. However, a variety of hematological malignancies (HMs) have been reported. We report a novel case of bi-alleleic DDX41 mutations in B-cell lymphoblastic leukemia (B-ALL), with unusual location of DDX41 mutations. The gene expression profile (GEP) of Ph + B-ALL with bi-alleleic DDX41 mutations showed heterogeneously transitional GEP and altered gene expression levels of genes involved in the process essential for red blood cells and myeloid cell differentiation were noted. We report that DDX41 mutations are unusual but can be an underlying event in Ph + B-ALL and screening DDX41 mutations can be also informative for patients awaiting for haploidentical stem cell transplantation and choosing the therapy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Cytosolic DNA-sensing pathway stimulation prompts type I IFN (IFN-I) production, but its role in systemic IFN-I pathway activation in primary SS (pSS) is poorly studied. Here we investigate the responsiveness of pSS monocytes and plasmacytoid dendritic cells (pDCs) to stimulator of interferon genes (STING) activation in relation to systemic IFN-I pathway activation and compare this with SLE. Expression of DNA-sensing receptors cGAS, IFI16, ZBP-1 and DDX41, signalling molecules STING, TBK1 and IRF3, positive and negative STING regulators, and IFN-I-stimulated genes MxA, IFI44, IFI44L, IFIT1 and IFIT3 was analysed in whole blood, CD14+ monocytes, pDCs, and salivary glands by RT-PCR, monocyte RNA sequencing data, flow cytometry and immunohistochemical staining. Peripheral blood mononuclear cells (PBMCs) from pSS, SLE and healthy controls (HCs) were stimulated with STING agonist 2'3'-cGAMP. STING phosphorylation (pSTING) and intracellular IFNα were evaluated using flow cytometry. STING activation induced a significantly higher proportion of IFNα-producing monocytes, but not pDCs, in both IFN-low and IFN-high pSS compared with HC PBMCs. Additionally, a trend towards more pSTING+ monocytes was observed in pSS and SLE, most pronounced in IFN-high patients. Positive STING regulators TRIM38, TRIM56, USP18 and SENP7 were significantly higher expression in pSS than HC monocytes, while the dual-function STING regulator RNF26 was downregulated in pSS monocytes. STING was expressed in mononuclear infiltrates and ductal epithelium in pSS salivary glands. STING stimulation induced pSTING and IFNα in pSS and SLE pDCs. pSS monocytes and pDCs are hyperresponsive to stimulation of the STING pathway, which was not restricted to patients with IFN-I pathway activation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  DEAD/H-box RNA helicases make the prominent family of helicases super family-2 which take part in almost all RNA-related processes, from initiation of transcription to RNA decay pathways. In addition to these RNA-related activities, in recent years a certain number of these helicases are reported to play important roles in anti-viral immunity through various ways. Along with RLHs, endosomal TLRs, and cytosolic DNA receptors, many RNA helicases including DDX3, DHX9, DDX6, DDX41, DHX33, DDX60, DHX36 and DDX1-DDX21-DHX36 complex act as viral nucleic acid sensors or co-sensors. These helicases mostly follow RLHs-MAVS and STING mediated signaling cascades to trigger induction of type-I interferons and pro-inflammatory cytokines. Many of them also function as downstream adaptor molecules (DDX3), segments of stress and processing bodies (DDX3 and DDX6) or negative regulators (DDX19, DDX24, DDX25, DDX39A and DDX46). On the contrary, many studies indicated that several DEAD/H-box helicases such as DDX1, DDX3, DDX6, DDX24, and DHX9 could be exploited by viruses to evade innate immune responses, suggesting that these helicases seem to have a dual function as anti-viral innate immune mediators and viral replication cofactors. In this review, we summarized the current knowledge on several representative DEAD/H-box helicases, with an emphasis on their functions in innate immunity responses, involved in their anti-viral and pro-viral roles.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Transcription poses a threat to genomic stability through the formation of R-loops that can obstruct progression of replication forks. R-loops are three-stranded nucleic acid structures formed by an RNA-DNA hybrid with a displaced non-template DNA strand. We developed RNA-DNA Proximity Proteomics to map the R-loop proximal proteome of human cells using quantitative mass spectrometry. We implicate different cellular proteins in R-loop regulation and identify a role of the tumor suppressor DDX41 in opposing R-loop and double strand DNA break accumulation in promoters. DDX41 is enriched in promoter regions in vivo, and can unwind RNA-DNA hybrids in vitro. R-loop accumulation upon loss of DDX41 is accompanied with replication stress, an increase in the formation of double strand DNA breaks and transcriptome changes associated with the inflammatory response. Germline loss-of-function mutations in DDX41 lead to predisposition to acute myeloid leukemia in adulthood. We propose that R-loop accumulation and genomic instability-associated inflammatory response may contribute to the development of familial AML with mutated DDX41.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33                                                                                                                                                                                                                       Endogenous retroviruses (ERVs) occupy a substantial fraction of mammalian genomes. However, whether ERVs extensively exist in ancient vertebrates remains unexplored. Here, we performed a genome-wide characterization of ERVs in a zebrafish (Danio rerio) model. Approximately 3,315 ERV-like elements (&lt;i&gt;Dr&lt;/i&gt;ERVs) were identified as Gypsy, Copia, Bel, and class I-III groups. &lt;i&gt;Dr&lt;/i&gt;ERVs accounted for approximately 2.3% of zebrafish genome and were distributed in all 25 chromosomes, with a remarkable bias on chromosome 4. Gypsy and class I are the two most abundant groups with earlier insertion times. The vast majority of the &lt;i&gt;Dr&lt;/i&gt;ERVs have varied structural defects. A total of 509 &lt;i&gt;gag&lt;/i&gt; and 71 &lt;i&gt;env&lt;/i&gt; genes with coding potentials were detected. The &lt;i&gt;env&lt;/i&gt;-coding elements were well-characterized and classified into four subgroups. A ERV-E4.8.43-DanRer element shows high similarity with HERV9NC-int in humans and analogous sequences were detected in species spanning from fish to mammals. RNA-seq data showed that hundreds of &lt;i&gt;Dr&lt;/i&gt;ERVs were expressed in embryos and tissues under physiological conditions, and most of them exhibited stage and tissue specificity. Additionally, 421 &lt;i&gt;Dr&lt;/i&gt;ERVs showed strong responsiveness to virus infection. A unique group of &lt;i&gt;Dr&lt;/i&gt;ERVs with immune-relevant genes, such as &lt;i&gt;fga&lt;/i&gt;, &lt;i&gt;ddx41, ftr35, igl1c3&lt;/i&gt;, and &lt;i&gt;tbk1&lt;/i&gt;, instead of intrinsic viral genes were identified. These &lt;i&gt;Dr&lt;/i&gt;ERVs are regulated by transcriptional factors binding at the long terminal repeats. This study provided a survey of the composition, phylogeny, and potential functions of ERVs in a fish model, which benefits the understanding of the evolutionary history of ERVs from fish to mammals. &lt;b&gt;IMPORTANCE&lt;/b&gt; Endogenous retroviruses (ERVs) are relics of past infection that constitute up to 8% of the human genome. Understanding the genetic evolution of the ERV family and the interplay of ERVs and encoded RNAs and proteins with host function has become a new frontier in biology. Fish, as the most primitive vertebrate host for retroviruses, is an indispensable integral part for such investigations. In the present study, we report the genome-wide characterization of ERVs in zebrafish, an attractive model organism of ancient vertebrates from multiple perspectives, including composition, genomic organization, chromosome distribution, classification, phylogeny, insertion time, characterization of &lt;i&gt;gag&lt;/i&gt; and &lt;i&gt;env&lt;/i&gt; genes, and expression profiles in embryos and tissues. The result helps uncover the evolutionarily conserved and fish-specific ERVs, as well as the immune-relevant ERVs in response to virus infection. This study demonstrates the previously unrecognized abundance, diversification, and extensive activity of ERVs at the early stage of ERV evolution.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Early-stage lung squamous cell carcinoma (LUSC) progression is accompanied by changes in immune microenvironments and the expression of immune-related genes (IRGs). Identifying innate IRGs associated with prognosis may improve treatment and reveal new immunotherapeutic targets. Gene expression profiles and clinical data of early-stage LUSC patients were obtained from the Gene Expression Omnibus and The Cancer Genome Atlas databases and IRGs from the InnateDB database. Univariate and multivariate Cox regression and LASSO regression analyses were performed to identify an innate IRG signature model prognostic in patients with early-stage LUSC. The predictive ability of this model was assessed by time-dependent receiver operator characteristic curve analysis, with the independence of the model-determined risk score assessed by univariate and multivariate Cox regression analyses. Overall survival (OS) in early-stage LUSC patients was assessed using a nomogram and decision curve analysis (DCA). Functional and biological pathways were determined by gene set enrichment analysis, and differences in biological functions and immune microenvironments between the high- and low-risk groups were assessed by ESTIMATE and the CIBERSORT algorithm. A signature involving six IRGs (SREBF2, GP2, BMX, NR1H4, DDX41, and GOPC) was prognostic of OS. Samples were divided into high- and low-risk groups based on median risk scores. OS was significantly shorter in the high-risk than in the low-risk group in the training (P &amp;lt; 0.001), GEO validation (P = 0.00021) and TCGA validation (P = 0.034) cohorts. Multivariate Cox regression analysis showed that risk score was an independent risk factor for OS, with the combination of risk score and T stage being optimally predictive of clinical benefit. GSEA, ESTIMATE, and the CIBERSORT algorithm showed that immune cell infiltration was higher and immune-related pathways were more strongly expressed in the low-risk group. A signature that includes these six innate IRGs may predict prognosis in patients with early-stage LUSC.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Clonal hematopoiesis (CH) can be transmitted from a donor to a recipient during allogeneic hematopoietic cell transplantation. Exclusion of candidate donors with CH is controversial since its impact on recipient outcomes and graft alloimmune function is uncertain. We performed targeted error-corrected sequencing on samples from 1,727 donors age 40 years or older and assessed the effect of donor CH on recipient clinical outcomes. We measured long-term engraftment of 102 donor clones and cytokine levels in 256 recipients at 3 and 12 months after transplant. CH was present in 22.5% of donors, with &lt;i&gt;DNMT3A&lt;/i&gt; (14.6%) and &lt;i&gt;TET2&lt;/i&gt; (5.2%) mutations being most common; 85% of donor clones showed long-term engraftment in recipients after transplantation, including clones with a variant allele fraction &amp;lt; 0.01. &lt;i&gt;DNMT3A-&lt;/i&gt;CH with a variant allele fraction ≥ 0.01, but not smaller clones, was associated with improved recipient overall (hazard ratio [HR], 0.79; &lt;i&gt;P&lt;/i&gt; = .042) and progression-free survival (HR, 0.72; &lt;i&gt;P&lt;/i&gt; = .003) after adjustment for significant clinical variables. In patients who received calcineurin-based graft-versus-host disease prophylaxis, donor &lt;i&gt;DNMT3A-&lt;/i&gt;CH was associated with reduced relapse (subdistribution HR, 0.59; &lt;i&gt;P&lt;/i&gt; = .014), increased chronic graft-versus-host disease (subdistribution HR, 1.36; &lt;i&gt;P&lt;/i&gt; = .042), and higher interleukin-12p70 levels in recipients. No recipient of sole &lt;i&gt;DNMT3A&lt;/i&gt; or &lt;i&gt;TET2&lt;/i&gt;-CH developed donor cell leukemia (DCL). In seven of eight cases, DCL evolved from donor CH with rare &lt;i&gt;TP53&lt;/i&gt; or splicing factor mutations or from donors carrying germline &lt;i&gt;DDX41&lt;/i&gt; mutations. Donor CH is closely associated with clinical outcomes in transplant recipients, with differential impact on graft alloimmune function and potential for leukemic transformation related to mutated gene and somatic clonal abundance. Donor &lt;i&gt;DNMT3A&lt;/i&gt;-CH is associated with improved recipient survival because of reduced relapse risk and with an augmented network of inflammatory cytokines in recipients. Risk of DCL in allogeneic hematopoietic cell transplantation is driven by somatic myelodysplastic syndrome-associated mutations or germline predisposition in donors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Next-generation sequencing studies are increasingly used in the evaluation of suspected chronic myeloid neoplasms (CMNs), but there is wide variability among laboratories in the genes analyzed for this purpose. Recently, the Association for Molecular Pathology CMN working group recommended a core 34-gene set as a minimum target list for evaluation of CMNs. This list was recommended based on literature review, and its diagnostic yield in clinical practice is unknown. To determine the diagnostic yield of the core 34 genes and assess the potential impact of including selected additional genes. We retrospectively reviewed 185 patients with known or suspected CMNs tested using a 62-gene next-generation sequencing panel that included all 34 core genes. The Association for Molecular Pathology's core 34 genes had a diagnostic yield of 158 of 185 (85.4%) to detect at least 1 variant with strong/potential clinical significance and 107 of 185 (57.8%) to detect at least 2 such variants. The 62-gene panel had a diagnostic yield of 160 of 185 (86.5%) and 112 of 185 (60.5%), respectively. Variants of unknown significance were identified in 49 of 185 (26.5%) using the core 34 genes versus 76 of 185 (41.1%) using the 62-gene panel. This study demonstrates that the Association for Molecular Pathology-recommended core 34-gene set has a high diagnostic yield in CMNs. Inclusion of selected additional genes slightly increases the rate of abnormal results, while also increasing the detection of variants of unknown significance. We recommend inclusion of CUX1, DDX41, ETNK1, RIT1, and SUZ12 in addition to the Association for Molecular Pathology's 34-gene core set for routine evaluation of CMNs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Diverse clinical and serological manifestations of systemic lupus erythematosus (SLE) compromise its diagnosis and treatment. A more reliable biomarker for SLE, which can play a critical role in either diagnosis, monitoring the disease progress or evaluating the response to treatment for individualized therapeutic, is necessary. DNA sensor is an important mediator of inflammation in systemic autoimmune diseases. However, the potential role for DNA sensor as disease activity biomarkers for SLE remained obscure. We detected the aberrant activation of DNA sensors and the corresponding IFN-β response in SLE patients, and to evaluate their potential role as disease biomarkers for SLE. We quantified the expressions of IFN-I and DNA sensor, such as cGAS, IFI16, DDX41, DAI and their down-stream adaptor STING in PBMC derived from patients with SLE (n = 100), healthy controls (HCs) (n = 62) by real-time PCR. The relationships between the expression of cGAS or IFI16 and clinical features in SLE patients were investigated. ROC curve analysis was performed to examine the predictive value of cGAS and IFI16 in SLE diagnosis, disease activity monitoring, specific organ manifestation and therapeutic response. RNA interference-mediated depletion of IFI16 or cGAS was conducted to evaluate their impact on IFN-I response. The expressions of cGAS and IFI16 were significantly higher in PBMC from SLE patients, closely correlated with the SLEDAI scores and high anti-dsDNA antibody titers. While the AUC for cGAS (0.767) was less than that of IFI16 and IFN-β, the AUC for IFI16 (0.856) and IFN-β (0.856) were similar. Expression of cGAS and IFI16 combine with IFN-β in PBMC showed high sensitivity (89.2%) and specificity (89.1%) for discrimination between mild and moderate/severe disease activity in SLE. Higher expression of IFI16 was association with ocular disorder in SLE patients. Neither IFI16 nor cGAS was a reliable indicator of therapeutic response. RNA interference-mediated depletion of IFI16 or cGAS prevented active SLE serum-induced upregulating in both IFN-α and IFN-β. High expression levels of cGAS and IFI16 in PBMC from SLE patients correlated strongly with disease activity. Both cGAS and IFI16 mediated signaling pathway were account for the robust production of IFN-β. Expression of cGAS and IFI16 combined with IFN-β in PBMC might serve as potential biomarkers for early diagnosis and monitoring disease activity in SLE.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    DEAD-box RNA helicases are important regulators of RNA metabolism and have been implicated in the development of cancer. Interestingly, these helicases constitute a major recurring family of RNA-binding proteins important for protecting the genome. Current studies have provided insight into the connection between genomic stability and several DEAD-box RNA helicase family proteins including DDX1, DDX3X, DDX5, DDX19, DDX21, DDX39B, and DDX41. For each helicase, we have reviewed evidence supporting their role in protecting the genome and their suggested mechanisms. Such helicases regulate the expression of factors promoting genomic stability, prevent DNA damage, and can participate directly in the response and repair of DNA damage. Finally, we summarized the pathological and therapeutic relationship between DEAD-box RNA helicases and cancer with respect to their novel role in genome stability.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Germline DDX41 variants in myeloid neoplasms (MNs) are not uncommon, and we explored the prevalence and characterized the clinical and pathologic features in a cohort of 3132 unrelated adult MN patients. By targeted next-generation sequencing, we identified 28 patients (20 men and 8 women) with pathogenic germline DDX41 variants who developed acute myeloid leukemia (AML), in which only 3 (11%) had a family history (FH) of MNs. A subacute clinical course of cytopenia (mean duration of 11.2 months, range 0-72 months) prior to the initial AML diagnosis was accompanied by a low blast count (median at 30%, range 20-70%) in hypocellular marrows (93% of all patients), in vast contrast to the typical proliferative subtypes of AML in the elderly. Most patients had a normal karyotype (75%) and acquired a second DDX41 variant (69%). A favorable overall survival (OS) was observed in comparison to that of common subtypes of AML with wild-type DDX41 in age-matched patients. Our study demonstrated that the frequent germline pathogenic DDX41 variants characterized a clinically distinct AML entity. Features characteristic of DDX41-mutated AML include male predominance, often lack of FH, indolent course, low proliferative potential, frequent somatic DDX41 variants, and a favorable OS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       DDX41 mutations (germline and somatic) are associated with late onset myelodysplastic syndromes/acute myeloid leukemia (MDS/AML). Myeloid neoplasms (MN) with germline predisposition was identified as a distinct category in the 2016 WHO classification revision, including MN with germline DDX41 mutation. We retrospectively analyzed the molecular findings and clinical characteristics of thirty-three DDX41-mutated (mDDX41) patients at our institution. We identified 14 distinct pathogenic DDX41 variants in 32 patients and 8 DDX41 variants of unknown significance (VUS) in 9 patients. Five (16%) patients had a second DDX41 somatic mutation p.R525H and 13 (40%) had at least one additional oncogenic co-mutation in other genes. The median age at the time of diagnosis was 66 years, with male predominance (72%) and the majority of patients had normal cytogenetics (91%). Two-year overall survival (OS) was 86% and 6 (21%) MDS/AML patients with relatively preserved hematopoietic function were observed without further intervention. In comparison to AML patients with prognostically more favorable subtypes [t(8;21), n=27 and inv(16), n=40], mDDX41 patients in our cohort showed similarly favorable OS. Our study highlights that mDDX41-MN patients often have an indolent course and mDDX41-AML has comparable OS to favorable-risk AML.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  A small but important proportion of patients (4-10 %) with AML have germline mutations. They can cause the development of AML at an earlier age, confer a higher risk of relapse or predispose to secondary leukemias, including therapy-related leukemias. The analysis of germline mutations in a patient and his/her family is also critical for the selection of suitable family donors if the patient is a candidate for hematopoietic stem cell transplantation (HSCT). 103 unrelated consecutive patients with de novo AML were enrolled in the study. Control group consisted of 103 persons from the general population. We performed NGS sequencing of bone marrow cells and buccal swabs DNA of six genes: CEBPA, DDX41, ETV6, TERT, GATA2, and IDH2 to detect germline pathogenic mutations. In the investigated group, 49 variants were detected in six genes. 26 of them were somatic and 23 germline. Germline variants were detected in all six tested genes. Eight pathogenic germline mutations were detected in 7 AML patients, in three genes: CEBPA, ETV6, and IDH2. One patient had two pathogenic germinal mutations, one in ETV6 and one in CEBPA gene. We identified one novel pathogenic germline mutation in CEBPA gene. The difference in frequency of all pathogenic germline mutations between the tested (7.77 %) and control groups (0.97 %) was statistically significant (p = 0.046). In the tested group, the median age at AML diagnosis was 11 years lower in patients with pathogenic germline mutations than in patients without them (p = 0.028). We showed higher frequency of CEBPA, ETV6, and IDH2 germline mutations in AML patients than in control group, which confirms the role of these mutations in the development of AML. We also showed that the median age at the onset of AML in patients with pathogenic germline mutations is significantly lower than in patients without them.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Inherited predisposition to myeloid malignancies is more common than previously appreciated. We analyzed the whole-exome sequencing data of paired leukemia and skin biopsy samples from 391 adult patients from the Beat AML 1.0 consortium. Using the 2015 American College of Medical Genetics and Genomics (ACMG) guidelines for variant interpretation, we curated 1547 unique variants from 228 genes. The pathogenic/likely pathogenic (P/LP) germline variants were identified in 53 acute myeloid leukemia (AML) patients (13.6%) in 34 genes, including 6.39% (25/391) of patients harboring P/LP variants in genes considered clinically actionable (tier 1). 41.5% of the 53 patients with P/LP variants were in genes associated with the DNA damage response. The most frequently mutated genes were CHEK2 (8 patients) and DDX41 (7 patients). Pathogenic germline variants were also found in new candidate genes (DNAH5, DNAH9, DNMT3A, and SUZ12). No strong correlation was found between the germline mutational rate and age of AML onset. Among 49 patients who have a reported history of at least one family member affected with hematological malignancies, 6 patients harbored known P/LP germline variants and the remaining patients had at least one variant of uncertain significance, suggesting a need for further functional validation studies. Using CHEK2 as an example, we show that three-dimensional protein modeling can be one of the effective methodologies to prioritize variants of unknown significance for functional studies. Further, we evaluated an in silico approach that applies ACMG curation in an automated manner using the tool for assessment and (TAPES) prioritization in exome studies, which can minimize manual curation time for variants. Overall, our findings suggest a need to comprehensively understand the predisposition potential of many germline variants in order to enable closer monitoring for disease management and treatment interventions for affected patients and families.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        DDX41 mutations are the most common germline alterations in adult myelodysplastic syndromes (MDSs). The majority of affected individuals harbor germline monoallelic frameshift DDX41 mutations and subsequently acquire somatic mutations in their other DDX41 allele, typically missense R525H. Hematopoietic progenitor cells (HPCs) with biallelic frameshift and R525H mutations undergo cell cycle arrest and apoptosis, causing bone marrow failure in mice. Mechanistically, DDX41 is essential for small nucleolar RNA (snoRNA) processing, ribosome assembly, and protein synthesis. Although monoallelic DDX41 mutations do not affect hematopoiesis in young mice, a subset of aged mice develops features of MDS. Biallelic mutations in DDX41 are observed at a low frequency in non-dominant hematopoietic stem cell clones in bone marrow (BM) from individuals with MDS. Mice chimeric for monoallelic DDX41 mutant BM cells and a minor population of biallelic mutant BM cells develop hematopoietic defects at a younger age, suggesting that biallelic DDX41 mutant cells are disease modifying in the context of monoallelic DDX41 mutant BM.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Increasing evidence shows that dysregulated RNA binding proteins (RBPs) modulate the progression of several malignancies. Nevertheless, their clinical implications of RBPs in HBV-related hepatocellular carcinoma (HCC) remain largely undefined. Here, this study systematically analyzed the associations of RBPs with HBV-related HCC prognosis. Based on differentially expressed RBPs between HBV-related HCC and control specimens, prognosis-related RBPs were screened by univariate analyses. A LASSO model was then created. Kaplan-Meier curves, ROCs, multivariate analyses, subgroup analyses and external verification were separately applied to assess the efficacy of this model in predicting prognosis and recurrence of patients. A nomogram was created by incorporating the model and clinical indicators, which was verified by ROCs, calibration curves and decision curve analyses. By CIBERSORT algorithm, the association between the risk score and immune cell infiltrations was evaluated. Totally, 54 RBPs were distinctly correlated to prognosis of HBV-related HCC. An 11-RBP model was created, containing POLR2L, MRPS12, DYNLL1, ZFP36, PPIH, RARS, SRP14, DDX41, EIF2B4, and NOL12. This risk score sensitively and accurately predicted one-, three- and five-year overall survival, disease-free survival, and progression-free interval. Compared to other clinical parameters, this risk score had the best predictive efficacy. Also, the clinical generalizability of the model was externally verified in the GSE14520 dataset. The nomogram may predict patients' survival probabilities. Also, the risk score was related to the components in the immune microenvironment. Collectively, RBPs may act as critical elements in the malignant progression of HBV-related HCC and possess potential implications on prognostication and therapy decision.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Germline predisposition syndromes (GPS) result from constitutional aberrations in tumor suppressive and homeostatic genes, increasing risk for neoplasia in affected kindred. In this study, we present clinical and genomic data on 144 Mayo Clinic patients with GPS; 59 evaluated prospectively using an algorithm-based diagnostic approach in the setting of a dedicated GPS/ inherited bone marrow failure syndrome (IBMFS) clinic. Seventy-two (50%) patients had IBMFS (telomere biology disorders-32,Fanconi anemia-18, Diamond Blackfan Anemia - 11, congenital neutropenia-5, Schwachman-Diamond Syndrome-5 and Bloom Syndrome-1), 27 (19%) had GPS with antecedent thrombocytopenia (RUNX1-FPD-15, ANKRD26-6, ETV6-2, GATA1-1, MPL-3), 28 (19%) had GPS without antecedent thrombocytopenia (GATA2 haploinsufficiency-16, DDX41-10, CBL-1 and CEBPA-1) and 17 (12%) had general cancer predisposition syndromes (ataxia telangiectasia-7, heterozygous ATM variants-3, CHEK2-2, TP53-2, CDK2NA-1, NF1-1 and Nijmegen Breakage Syndrome-1). Homozygous and heterozygous ATM pathogenic variants were exclusively associated with lymphoproliferative disorders (LPD), while DDX41 GPS was associated with LPD and myeloid neoplasms. The use of somatic NGS-testing identified clonal evolution in GPS patients, with ASXL1, RAS pathway genes, SRSF2 and TET2 being most frequently mutated. Fifty-two (91%) of 59 prospectively identified GPS patients had a change in their management approach, including additional GPS-related screening in 42 (71%), referral for allogenic HSCT workup and screening of related donors in 16 (27%), medication initiation and selection of specific conditioning regimens in 14 (24%), and genetic counseling with specific intent of fertility preservation and preconceptual counseling in 10 (17%) patients; highlighting the importance of dedicated GPS screening, detection and management programs for patients with hematological neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      DDX41 serves as a DNA sensor in innate immunity and mutated &lt;i&gt;DDX41&lt;/i&gt; is pathogenic, mainly for myeloid neoplasms. In this study, "&lt;i&gt;DDX41&lt;/i&gt;" was searched in &lt;i&gt;PubMed&lt;/i&gt; and &lt;i&gt;Web of Science&lt;/i&gt; between 1 January 2015 and 29 April 2021 with individual-patient data seeking. A meta-analysis was not valid here due to the absence of a large dataset. Thirty articles were finally included in the qualitative analysis and 277 patients from 20 studies without overlap were involved in the quantitative summary. Pooled incidence was 3.3% (95% confidence interval 2.4-4.2%) of unselected myeloid neoplasms. Patients with hematologic disorders harboring mutated &lt;i&gt;DDX41&lt;/i&gt; were featured as 80% males, median 66 (20-88) years old at diagnosis, 75% acute myeloid leukemia (AML) or myelodysplastic syndromes (MDS), 64% with normal karyotype. Eighty-five percent of patients had germline variants which were nationally diverse and more of frameshift type, whereas 64% of patients had somatic &lt;i&gt;DDX41&lt;/i&gt; variants where p.R525H and missense dominated. &lt;i&gt;ASXL1&lt;/i&gt; and &lt;i&gt;TP53&lt;/i&gt; were the top frequent concomitant somatic mutations. Therapeutically, 70% overall response rate was obtained of hypomethylating agents in MDS, 96% complete remission of chemotherapy in AML, and 8% of relapse in hematopoietic stem cell transplant. Neither overall survival nor progression-free survival could be summed. Several significant clinical differences were observed in different diagnosis groups, familial and sporadic cases, and p.R525H compared with other somatic variants. In conclusion, myeloid neoplasms carrying &lt;i&gt;DDX41&lt;/i&gt; mutations were mainly older, male, MDS, and AML patients who had promising responses to treatment. Both germline and somatic &lt;i&gt;DDX41&lt;/i&gt; variants possessed unique characteristics and groups of interest presented certain differences worth further research. (CRD42021228886).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Recent advances in sequencing technologies have increased the detection rate for identifying germline mutations that predispose an individual to various myeloid neoplasms and somatic mutations acquired during progression from myelodysplastic syndromes (MDS) to acute myeloid leukemia (AML). In addition to pediatric subjects, adult patients were analyzed in order to obtain a complete spectrum of driver mutations in germline cells and/or somatic tumor samples. As shown in several recent studies, such driver mutations are acquired in a gene-specific fashion. DDX41 mutations are observed in germline cells long before MDS presentation. SAMD9/SAMD9L germline mutations associated with defective hematopoiesis account for recurrent and familial -7/del (7q) lesions, which result in the removal of the disadvantageous allele. Additionally, MDS cases in younger population display compound heterozygous germline mutations in the Shwachman-Diamond syndrome-associated SBDS gene. In peripheral blood samples from healthy elderly individuals, DNMT3A, TET2, and ASXL1 somatic mutations are usually detected due to age-related clonal hematopoiesis and are considered to be a risk factor for hematological neoplasms. In MDS, mutations of genes, such as NRAS and FLT3, designated as type-1 genes, are significantly associated with leukemic evolution. On the other hand, mutations in type-2 genes, including RUNX1 and GATA2, are related to progression from low risk MDS to high risk MDS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   This study seeks to further characterize the clinicopathologic spectrum of DDX41-mutated hematolymphoid malignancies. We identified DDX41 mutations from a cohort of known or suspected hematologic disorders and reviewed the corresponding clinical, genetic, phenotypic, and morphologic findings. DDX41 mutations were identified in 20 (1.4%) of 1,371 cases, including 8 cases of acute myeloid leukemia (AML), 5 cases of myelodysplastic syndrome (MDS), 2 cases of therapy-related MDS/AML, 1 case of primary myelofibrosis, 1 case of chronic myeloid leukemia, 1 case of clonal cytopenia of uncertain significance (CCUS), 1 case of T-cell large granular lymphocytic leukemia (T-LGL), and 1 case of multiple myeloma. DDX41-mutated neoplasms were morphologically heterogeneous with a median cellularity of 20% (range, 10%-100%). Megakaryocyte dysplasia occurred in 7 (35%) of 20 cases and trilineage dysplasia in 1 (5%). Frequently comutated genes include a second, somatic DDX41 mutation (8/19, 42%) followed by mutations in TET2 (20%), DNMT3A (20%), ASXL1 (20%), and CUX1 (20%). Karyotypes were noncomplex in 17 (89%) of 19. This report extends the spectrum of DDX41-mutated disorders to include CCUS, T-LGL, and plasma cell disorders. The morphologic features are heterogeneous and nonspecific, highlighting the importance of DDX41 testing during routine workup of hematolymphoid neoplasms.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The majority of studies assessing the contribution of pathogenic germline variants (PGVs) to cancer predisposition have focused on patients with single cancers. We analyzed 45 known cancer predisposition genes (CPGs) in germline samples of 202 patients with hematological malignancies (HMs) plus one or more other independent cancer managed at major tertiary medical centers on two different continents. This included 120 patients with therapy-related myeloid neoplasms (t-MNs), where the HM occurred after cytotoxic treatment for a first malignancy, and 82 patients with multiple cancers in which the HM was not preceded by cytotoxic therapy (MC-HM). Using American College of Medical Genetics/Association for Molecular Pathology variant classification guidelines, 13% of patients had PGVs, most frequently identified in CHEK2 (17% of PGVs), BRCA1 (13%), DDX41 (13%), and TP53 (7%). The frequency of PGVs in MC-HM was higher than in t-MN, although not statistically significant (18 vs. 9%; p = 0.085). The frequency of PGVs in lymphoid and myeloid HM patients was similar (19 vs. 17.5%; p &amp;gt; 0.9). Critically, patients with PGVs in BRCA1, BRCA2 or TP53 did not satisfy current clinical phenotypic criteria for germline testing. Our data suggest that a personal history of multiple cancers, one being a HM, should trigger screening for PGVs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 DEAD-box Helicase 41 (DDX41) is a recently identified factor mutated in hematologic malignancies whose function in hematopoiesis is unknown. Using an in vivo model of Ddx41 deficiency, we unveiled a critical role for this helicase in regulating erythropoiesis. We demonstrated that loss of ddx41 leads to anemia caused by diminished proliferation and defective differentiation of erythroid progenitors. Mis-expression and alternative splicing of cell cycle genes is rampant in ddx41 mutant erythroid progenitors. We delineated that the DNA damage response is activated in mutant cells resulting in an Ataxiatelangiectasia mutated (ATM) and Ataxia-telangiectasia and Rad3-related (ATR)-triggered cell cycle arrest. Inhibition of these kinases partially suppressed ddx41 mutant anemia. These findings establish a critical function for Ddx41 in promoting healthy erythropoiesis via protection from genomic stress and delineate a mechanistic framework to explore a role for ATM and ATR signaling in DDX41-mutant hematopoietic pathologies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        DEAD-box helicases participate in nearly all steps of an RNA's life. In recent years, increasing evidence has shown that several family members are multitasking enzymes. They are often involved in different processes, which may be typical for RNA helicases, such as RNA export and translation, or atypical, e.g., acting as nucleic acid sensors that activate downstream innate immune signaling. This review focuses on the DEAD-box protein DDX41 and summarizes our current understanding of its roles as an innate immunity sensor in the cytosol and in pre-mRNA splicing in the nucleus and discusses DDX41's involvement in disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 In 2016, the World Health Organization incorporated 'myeloid neoplasms with germline predisposition' into its classification of tumors of hematopoietic and lymphoid tissues, revealing the important role of germline mutations in certain myeloid neoplasms, particularly myelodysplastic syndrome and acute myeloid leukemia. The awareness of germline susceptibility has increased, and some patients with myeloid neoplasms present with a preexisting disorder or organ dysfunction. In such cases, mutations in genes including CCAAT enhancer binding protein α (CEBPA), DEAD (Asp-Glu-Ala-Asp) box polypeptide 41 (DDX41), RUNX family transcription factor 1 (RUNX1), GATA binding protein 2 (GATA2), Janus kinase 2 (JAK2) and ETS variant transcription factor 6 (ETV6) have been recognized. Moreover, with the application of advanced technologies and reports of more cases, additional germline mutations associated with myeloid neoplasms have been identified and provide insights into the formation, prognosis and therapy of myeloid neoplasms. The present review discusses the well-known CEBPA, DDX41, RUNX1, GATA2, JAK2 and ETV6 germline mutations, and other mutations including those of lymphocyte adapter protein/SH2B adapter protein 3 and duplications of autophagy related 2B, GSK3B interacting protein αnd RB binding protein 6, ubiquitin ligase, that remain to be confirmed or explored. Recommendations for the management of diseases associated with germline mutations are also provided.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            In this issue of Developmental Cell, Weinreb et al. reveal that loss of the DEAD-box helicase Ddx41 unexpectedly triggers an R-loop-mediated sterile inflammatory cascade which drives HSPC production during embryonic development. Human studies suggest mechanistic conservation for inflammation in DDX41-associated hematologic disease, uncovering a potential route for future therapeutic intervention.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Allogeneic hematopoietic stem cell transplantation (HSCT) is an important therapeutic modality for patients with acute myelogenous leukemia (AML) with poor risk features. Nonetheless, roughly 30% of such patients have leukemia recurrence and up to 2% of these are donor-derived leukemias, in which malignancy develops in the donor's transplanted cells, despite extremely low rates of leukemia in the donors themselves. Notably, over 20% of these malignancies carry chromosome 7 abnormalities nearly all of which are monosomies. Recent advances in whole exome and genome sequencing have allowed for detection of candidate genes that likely contribute to the development of AML in donor cells (donor leukemia, DCL). These genes include CEBPA, GATA2, JAK2, RUNX1, DDX41, EZH2, IDH1/2, DNMT3A, ASXL1, XPD, XRCC3, and CHEK1. The potential roles of variants in these genes are evaluated based on familial clustering of MDS/AML and corresponding animal studies demonstrating their leukemogenic nature. This review describes the spectrum of genetic aberrations detected in DCL cases in the literature with regard to the character of the individual cases, existing family cohorts that carry individual genes, and functional studies that support etiologic roles in AML development. DCL presents a unique opportunity to examine genetic variants in the donors and recipients with regards to progression to malignancy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Hematopoietic stem and progenitor cells (HSPCs) arise during embryonic development and are essential for sustaining the blood and immune systems throughout life. Tight regulation of HSPC numbers is critical for hematopoietic homeostasis. Here, we identified DEAD-box helicase 41 (Ddx41) as a gatekeeper of HSPC production. Using zebrafish ddx41 mutants, we unveiled a critical role for this helicase in regulating HSPC production at the endothelial-to-hematopoietic transition. We determined that Ddx41 suppresses the accumulation of R-loops, nucleic acid structures consisting of RNA:DNA hybrids and ssDNAs whose equilibrium is essential for cellular fitness. Excess R-loop levels in ddx41 mutants triggered the cGAS-STING inflammatory pathway leading to increased numbers of hemogenic endothelium and HSPCs. Elevated R-loop accumulation and inflammatory signaling were observed in human cells with decreased DDX41, suggesting possible conservation of mechanism. These findings delineate that precise regulation of R-loop levels during development is critical for limiting cGAS-STING activity and HSPC numbers.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          DEAD‑box helicase 41 (DDX41) is an RNA helicase and accumulating evidence has suggested that DDX41 is involved in pre‑mRNA splicing during tumor development. However, the role of DDX41 in tumorigenesis remains unclear. In order to determine the function of DDX41, the human DDX41 gene was cloned and overexpressed in HeLa cells. The present study demonstrated that DDX41 overexpression inhibited proliferation and promoted apoptosis in HeLa cells. RNA‑sequencing analysis of the transcriptomes in overexpressed and normal control samples. DDX41 regulated 959 differentially expressed genes compared with control cells. Expression levels of certain oncogenes were also regulated by DDX41. DDX41 selectively regulated the alternative splicing of genes in cancer‑associated pathways including the EGFR and FGFR signaling pathways. DDX41 selectively upregulated the expression levels of five antigen processing and presentation genes (HSPA1A, HSPA1B, HSPA6, HLA‑DMB and HLA‑G) and downregulated other immune‑response genes in HeLa cells. Additionally, DDX41‑regulated oncogenes and antigen processing and presentation genes were associated with patient survival rates. Moreover, DDX41 expression was associated with immune infiltration in cervical and endocervical squamous cancer. The present findings showed that DDX41 regulated the cancer cell transcriptome at both the transcriptional and alternative splicing levels. The DDX41 regulatory network predicted the biological function of DDX41 in suppressing tumor cell growth and regulating cancer immunity, which may be important for developing anticancer therapeutics.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            DDX41 mutations are associated with hematologic malignancies including myelodysplastic syndrome (MDS) and acute myeloid leukemia (AML), but the incidence in idiopathic cytopenia of undetermined significance (ICUS) is unknown. We investigated the incidence, genetic characteristics, and clinical features of DDX41 mutations in Korean patients with ICUS, MDS, or AML. We performed targeted deep sequencing of 61 genes including DDX41 in 457 patients with ICUS (n=75), MDS (n=210), or AML (n=172). Germline DDX41 mutations with causality were identified in 28 (6.1%) patients, of whom 27 (96.4%) had somatic mutations in the other position of DDX41. Germline origins of the DDX41 mutations were confirmed in all of the 11 patients in whom germline-based testing was performed. Of the germline DDX41 mutations, p.V152G (n=10) was most common, followed by p.Y259C (n=8), p.A500fs (n=6), and p.E7* (n=3). Compared with non-mutated patients, patients with a DDX41 mutation were more frequently male, older, had a normal karyotype, low leukocyte count, and hypocellular marrow at diagnosis. Three of the four ICUS patients with germline DDX41 mutations progressed to MDS. The incidence of DDX41 mutations in Korean patients was high and there was a distinct mutation pattern, in that p.V152G was a unique germline variant. ICUS harboring germline DDX41 mutations may be regarded as a hereditary myeloid neoplasm. Germline DDX41 mutations are not uncommon and should be explored when treating patients with myeloid malignancies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Previously considered rare, inherited hematologic malignancies are increasingly identified. Germline mutations in the RNA helicase &lt;i&gt;DDX41&lt;/i&gt; predispose to increased lifetime risks of myeloid neoplasms with disease often occurring later in life which presents challenges for germline recognition. To improve identification of germline &lt;i&gt;DDX41&lt;/i&gt;, individuals presenting with ≥1 &lt;i&gt;DDX41&lt;/i&gt; alteration on an institutional MDS/AML next-generation sequencing based panel with at least one at &amp;gt;40% variant allele frequency were flagged for review and genetic counseling referral. Of 5,801 individuals, 90 (1.5%) had ≥1 &lt;i&gt;DDX41&lt;/i&gt; mutation(s) identified. Thirty-eight (42%) patients with a median age of 66 years were referred for genetic counseling; thirty-one were male (81.5%). Thirty-five (92%) referred patients elected to pursue germline evaluation and in 33/35 (94%) a germline &lt;i&gt;DDX41&lt;/i&gt; variant was confirmed. Twenty-two patients (66%) with germline variants reported antecedent cytopenias, seven (21%) had a prior history of malignancy, and twenty-seven (82%) reported a family history of cancer. Predictive genetic testing for healthy family members under consideration as stem cell transplant donors was successfully performed in 11 family members, taking an average of 15 days. Near-heterozygous &lt;i&gt;DDX41&lt;/i&gt; mutations identified on next-generation sequencing, particularly nonsense/frameshift variants or those at recurrent germline "hot spots" are highly suggestive of a germline mutation. Next-generation sequencing screening is a feasible tool to screen unselected myeloid neoplasms for germline &lt;i&gt;DDX41&lt;/i&gt; mutations, enabling timely and appropriate care.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Myeloid neoplasms (MN) with germline predisposition (MNGP) are likely to be more common than currently appreciated. Many of the genes involved in MNGP are also recurrently mutated in sporadic MN. Therefore, routine analysis of gene panels by next-generation sequencing provides an effective approach to detect germline variants with clinical significance in patients with hematological malignancies. Gene panel sequencing was performed in 88 consecutive and five nonconsecutive patients with MN diagnosis. Disease-causing germline mutations in CEBPα, ASXL1, TP53, MPL, GATA2, DDX41, and ETV6 genes were identified in nine patients. Six out of the nine patients with germline variants had a strong family history. These patients presented great heterogeneity in the age of diagnosis and phenotypic characteristics. In our study, there were families in which all the affected members presented the same subtype of disease, whereas members of other families presented various disease phenotypes. This intrafamiliar heterogeneity suggests that the acquisition of particular somatic variants may drive the evolution of the disease. This approach enabled high-throughput detection of MNGP in patients with MN diagnosis, which is of great relevance for both the patients themselves and the asymptomatic mutation carriers within the family. It is crucial to make a proper diagnosis of these patients to provide them with the most suitable treatment, follow-up, and genetic counseling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          It is known that nutrition and immunity are connected, but the mechanism is not very clear. Endogenous retroviruses (ERV) account for 8 to 10% of the human and mouse genomes and play an important role in some biological processes of animals. Recent studies indicate that the activation of ERV can affect the expression of the immunity- or inflammation-related genes, and the activities of ERV are subjected to regulation of many factors including nutritional factors. Therefore, we hypothesize that nutritional status can affect the expression of the immunity- or inflammation-related genes via ERV. To verify this hypothesis, the nutritional status of animals was altered by fasting or overfeeding, and the expression of intact ERV (ERVK18P, ERVK25P) and immunity- or inflammation-related genes (DDX41, IFIH1, IFNG, IRF7, STAT3) in the liver was determined by quantitative PCR, followed by overexpressing ERVK25P in goose primary hepatocytes and determining the expression of the immunity- or inflammation-related genes. The data showed that compared with the control group (no fasting), the expression of ERV and the immunity- or inflammation-related genes was increased in the liver of the fasted chickens but decreased in the liver of the fasted geese. Moreover, compared with the control group (routinely fed), the expression of ERV and the immunity- or inflammation-related genes was increased in the liver of the overfed geese. In addition, overexpression of ERVK25P in goose primary hepatocytes can induce the expression of the immunity- or inflammation-related genes. In conclusion, these findings suggest that ERV mediate the effects of fasting and overfeeding on the expression of the immunity- or inflammation-related genes, the mediation varied with poultry species, and ERV and the immunity- or inflammation-related genes may be involved in the development of goose fatty liver. This study provides a potential mechanism for the connection between nutrition and immunity.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    DNA viruses are a source of great morbidity and mortality throughout the world by causing many diseases; thus, we need substantial knowledge regarding viral pathogenesis and the host's antiviral immune responses to devise better preventive and therapeutic strategies. The innate immune system utilizes numerous germ-line encoded receptors called pattern-recognition receptors (PRRs) to detect various pathogen-associated molecular patterns (PAMPs) such as viral nucleic acids, ultimately resulting in antiviral immune responses in the form of proinflammatory cytokines and type I interferons. The immune-stimulatory role of DNA is known for a long time; however, DNA sensing ability of the innate immune system was unraveled only recently. At present, multiple DNA sensors have been proposed, and most of them use STING as a key adaptor protein to exert antiviral immune responses. In this review, we aim to provide molecular and structural underpinnings on endosomal DNA sensor Toll-like receptor 9 (TLR9) and multiple cytosolic DNA sensors including cyclic GMP-AMP synthase (cGAS), interferon-gamma inducible 16 (IFI16), absent in melanoma 2 (AIM2), and DNA-dependent activator of IRFs (DAI) to provide new insights on their signaling mechanisms and physiological relevance. We have also addressed less well-understood DNA sensors such as DEAD-box helicase DDX41, RNA polymerase III (RNA pol III), DNA-dependent protein kinase (DNA-PK), and meiotic recombination 11 homolog A (MRE11). By comprehensive understanding of molecular and structural aspects of DNA-sensing antiviral innate immune signaling pathways, potential new targets for viral and autoimmune diseases can be identified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Although much work has focused on the elucidation of somatic alterations that drive the development of acute leukaemias and other haematopoietic diseases, it has become increasingly recognized that germline mutations are common in many of these neoplasms. In this Review, we highlight the different genetic pathways impacted by germline mutations that can ultimately lead to the development of familial and sporadic haematological malignancies, including acute lymphoblastic leukaemia, acute myeloid leukaemia (AML) and myelodysplastic syndrome (MDS). Many of the genes disrupted by somatic mutations in these diseases (for example, TP53, RUNX1, IKZF1 and ETV6) are the same as those that harbour germline mutations in children and adolescents who develop these malignancies. Moreover, the presumption that familial leukaemias only present in childhood is no longer true, in large part due to the numerous studies demonstrating germline DDX41 mutations in adults with MDS and AML. Lastly, we highlight how different cooperating events can influence the ultimate phenotype in these different familial leukaemia syndromes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  DNA-dependent activator of interferon regulatory factors (DAIs), interferon gamma inducible protein 16 (IFI16), DEAD-box polypeptide 41 (DDX41), DNA-dependent protein kinase (DNA-PK), meiotic recombination 11 homolog A (MRE11), and cyclic GMP-AMP synthase (cGAS) have been identified as intracellular STING-dependent DNA sensors in recent years. Studies have shown that the DNA sensor-STING-interferon (IFN)-&lt;i&gt;β&lt;/i&gt; pathway plays an important role in the defense against intracellular invasion of many DNA viruses. However, the intracellular recognition of hepatitis B virus (HBV) DNA by DNA sensors is still largely unclear. In this study, we aimed to determine whether the DNA sensor-STING pathway in peripheral blood mononuclear cells (PBMCs) can be activated by acute and chronic HBV infections in humans. We first evaluated the expression of these DNA sensors in PBMCs of acute and chronic HBV-infected patients by quantitative real-time polymerase chain reaction. We next compared the expression of the upregulated DNA sensor between monocytes and nonmonocytes to find its cellular source. Finally, by &lt;i&gt;in vitro&lt;/i&gt; stimulation, we analyzed the IFN-&lt;i&gt;β&lt;/i&gt; response of the DNA sensor-STING pathway in PBMCs and monocytes from chronic HBV-infected patients. The results showed that IFI16, DDX41, MRE11, and the adaptor STING were upregulated in chronic HBV-infected patients, whereas only IFI16 was upregulated in acute HBV-infected patients. However, IFN-&lt;i&gt;β&lt;/i&gt; expression was not changed in PBMCs from acute and chronic HBV-infected patients. We next found IFI16 was mainly expressed in monocytes of acute and chronic hepatitis B patients. Finally, by stimulation of monocytes with VACV ds 70mer, a ligand for IFI16, we confirmed the attenuated response of the IFI16-STING pathway. Taken together, our results suggest that HBV might be sensed by DNA sensors in PBMCs of acute and chronic HBV-infected patients, and meanwhile HBV infection attenuates the response of the DNA sensor-STING pathway in PBMCs and monocytes, which may facilitate the persistence of HBV infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Increasing awareness about germline predisposition and the widespread application of unbiased whole exome sequencing contributed to the discovery of new clinical entities with high risk for the development of haematopoietic malignancies. The revised 2016 WHO classification introduced a novel category of "myeloid neoplasms with germline predisposition" with GATA2, CEBPA, DDX41, RUNX1, ANKRD26 and ETV6 genes expanding the spectrum of hereditary myeloid neoplasms (MN). Since then, more germline causes of MN were identified, including SAMD9, SAMD9L, and ERCC6L2. This review describes the genetic and clinical spectrum of predisposition to MN. The main focus lies in delineation of phenotypes, genetics and management of GATA2 deficiency and the novel SAMD9/SAMD9L-related disorders. Combined, GATA2 and SAMD9/SAMD9L (SAMD9/9L) syndromes are recognized as most frequent causes of primary paediatric myelodysplastic syndromes, particularly in setting of monosomy 7. To date, ~550 cases with germline GATA2 mutations, and ~130 patients with SAMD9/9L mutations had been reported in literature. GATA2 deficiency is a highly penetrant disorder with a progressive course that often rapidly necessitates bone marrow transplantation. In contrast, SAMD9/9L disorders show incomplete penetrance with various clinical outcomes ranging from spontaneous haematological remission observed in young children to malignant progression.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Telomere biology disorders (TBDs) present heterogeneously, ranging from infantile bone marrow failure associated with very short telomeres to adult-onset interstitial lung disease (ILD) with normal telomere length. Yield of genetic testing and phenotypic spectra for TBDs caused by the expanding list of telomere genes in adults remain understudied. Thus, we screened adults aged ≥18 years with a personal and/or family history clustering hematologic disorders and/or ILD enrolled on The University of Chicago Inherited Hematologic Disorders Registry for causative variants in 13 TBD genes. Sixteen (10%) of 153 probands carried causative variants distributed among TERT (n = 6), TERC (n = 4), PARN (n = 5), or RTEL1 (n = 1), of which 19% were copy number variants. The highest yield (9 of 22 [41%]) was in families with mixed hematologic and ILD presentations, suggesting that ILD in hematology populations and hematologic abnormalities in ILD populations warrant TBD genetic testing. Four (3%) of 117 familial hematologic disorder families without ILD carried TBD variants, making TBD second to only DDX41 in frequency for genetic diagnoses in this population. Phenotypes of 17 carriers with heterozygous PARN variants included 4 (24%) with hematologic abnormalities, 67% with lymphocyte telomere lengths measured by flow cytometry and fluorescence in situ hybridization at or above the 10th percentile, and a high penetrance for ILD. Alternative etiologies for cytopenias and/or ILD such as autoimmune features were noted in multiple TBD families, emphasizing the need to maintain clinical suspicion for a TBD despite the presence of alternative explanations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Recognition of the non-self signature of invading pathogens is a crucial step for the initiation of the innate immune mechanisms of the host. The host response to viral and bacterial infection involves sets of pattern recognition receptors (PRRs), which bind evolutionarily conserved pathogen structures, known as pathogen-associated molecular patterns (PAMPs). Recent advances in the identification of different types of PRRs in teleost fish revealed a number of cytosolic sensors for recognition of viral and bacterial nucleic acids. These are DExD/H-box RNA helicases including a group of well-characterized retinoic acid inducible gene I (RIG-I)-like receptors (RLRs) and non-RLR DExD/H-box RNA helicases (e.g., DDX1, DDX3, DHX9, DDX21, DHX36 and DDX41) both involved in recognition of viral RNAs. Another group of PRRs includes cytosolic DNA sensors (CDSs), such as cGAS and LSm14A involved in recognition of viral and intracellular bacterial dsDNAs. Moreover, dsRNA-sensing protein kinase R (PKR), which has a role in antiviral immune responses in higher vertebrates, has been identified in fish. Additionally, fish possess a novel PKR-like protein kinase containing Z-DNA binding domain, known as PKZ. Here, we review the current knowledge concerning cytosolic sensors for recognition of viral and bacterial nucleic acids in teleosts.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The cyclic GMP-AMP synthase-stimulator of interferon genes (cGAS-STING) pathway induces innate immunity by activating the production of inflammatory cytokines and type I interferons. Recently, studies revealed that self-DNA from by-products of chromosome instability and tumors could activate the cGAS-STING pathway, and subsequently promote or inhibit tumor development. However, the prognostic value and correlations with immune infiltrates of the cGAS-STING pathway in hepatocellular carcinoma (HCC) have not been clarified. In the present study, we used the Molecular Signatures Database, Oncomine, UALCAN, Human Protein Atlas, Kaplan-Meier plotter, LinkedOmics, and Tumor Immune Estimation Resource databases. Overexpression of XRCC5, IRF3, TRIM21, STAT6, DDX41, TBK1, XRCC6, TREX1, PRKDC, and TMEM173 was markedly correlated with clinical stages and pathological grades in HCC. Moreover, higher mRNA expression of XRCC5, XRCC6, and PRKDC was significantly related with shorter overall survival. However, higher mRNA expression of IFI16, STAT6, NLRC3, and TMEM173 was associated with favorable overall survival. Our results suggested that the kinase targets of the cGAS-STING pathway included the SRC family of tyrosine kinases (LCK and LYN), phosphoinositide 3-kinase-related protein kinase (PIKK) family kinases (ATM and ATR), and mitogen-activated protein kinase 1 (MAPK1). Furthermore, we identified significant correlations among the expression of cGAS-STING pathway and infiltration of B cells, CD4+T cells, CD8+ T cells, macrophages, neutrophils, and dendritic cells in HCC. The expression of the cGAS-STING pathway also exhibited strong relationships with diverse immune marker sets in HCC. These findings suggest that cGAS-STING pathway members may be used as prognostic biomarkers and immunotherapeutic targets HCC patients.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Cancer cells employ alternative splicing (AS) to acquire splicing isoforms favouring their survival. However, the causes of aberrant AS in breast cancer are poorly understood. In this study, the METABRIC (Molecular Taxonomy of Breast Cancer International Consortium) data were analysed with univariate feature selection. Of 122 analysed spliceosome components, &lt;i&gt;U2SURP, PUF60, DDX41, HNRNPAB, EIF4A3&lt;/i&gt;, and &lt;i&gt;PPIL3&lt;/i&gt; were significantly associated with breast cancer survival. The top 4 four genes, &lt;i&gt;U2SURP, PUF60, DDX41&lt;/i&gt;, and &lt;i&gt;HNRNPAB&lt;/i&gt;, were chosen for further analyses. Their expression was significantly associated with cancer molecular subtype, tumour stage, tumour grade, overall survival (OS), and cancer-specific survival in the METABRIC data. These results were verifiable using other cohorts. The Cancer Genome Atlas data unveiled the elevated expression of &lt;i&gt;PUF60, DDX41&lt;/i&gt;, and &lt;i&gt;HNRNPAB&lt;/i&gt; in tumours compared with the normal tissue and confirmed the differential expression of the four genes among cancer molecular subtypes, as well as the associations of &lt;i&gt;U2SURP, PUF60&lt;/i&gt;, and &lt;i&gt;DDX41&lt;/i&gt; expression with tumour stage. A meta-analysis data verified the associations of &lt;i&gt;U2SURP, PUF60&lt;/i&gt;, and &lt;i&gt;HNRNPAB&lt;/i&gt; expression with tumour grade, the associations of &lt;i&gt;PUF60, DDX41&lt;/i&gt;, and &lt;i&gt;HNRNPAB&lt;/i&gt; expression with OS and distant metastasis-free survival, and the associations of &lt;i&gt;U2SURP&lt;/i&gt; and &lt;i&gt;HNRNPAB&lt;/i&gt; expression with relapse-free survival. Experimentally, we demonstrated that inhibiting the expression of the four genes separately suppressed cell colony formation and slowed down cell growth considerably in breast cancer cells, but not in immortal breast epithelial cells. In conclusion, we have identified &lt;i&gt;U2SURP, PUF60, DDX41&lt;/i&gt;, and &lt;i&gt;HNRNPAB&lt;/i&gt; are spliceosome-related genes pivotal for breast cancer survival.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Acute myeloid leukemia (AML) is one of the most common types of leukemia. With the recent advances in sequencing technology and the growing body of knowledge on the genetics of AML, there is increasing concern about cancer predisposing germline mutations as well as somatic mutations. As familial cases sharing germline mutations are constantly reported, germline predisposition gene mutations in patients with AML are gaining attention. We performed genomic sequencing of Korean patients diagnosed with AML to identify the prevalence and characteristics of germline predisposition mutations. Among 180 patients, germline predisposition mutations were identified in 13 patients (13/180, 7.2%, eight adults and five children). Germline mutations of BLM, BRCA1, BRCA2, CTC1, DDX41, ERCC4, ERCC6, FANCI, FANCM, PALB2, and SBDS were identified. Most of the mutations are in genes involved in DNA repair and genomic stability maintenance. Patients harboring germline mutations tended to have earlier onset of AML (p = 0.005), however, the presence of germline mutations did not showed significant association with other clinical characteristics or treatment outcome. Since each mutation was rare, further study with a larger number of cases would be needed to establish the effect of the mutations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         As a member of DExD/H-box helicase family, DDX41 (DEAD box polypeptide 41) acts as an intracellular DNA sensor that induces type I IFN expression in mammals. Fish DDX41 shares some similar properties with the mammalian counterparts. In this study, a DDX41 orthologous gene from grass carp (Ctenopharyngodon idellus) (CiDDX41) was cloned and characterized. The ORF of CiDDX41 encodes a polypeptide of 614 amino acids. Multiple alignments showed that DDX41 is highly conserved among different species. Phylogenetic tree analysis revealed that CiDDX41 shares a high degree of homology with Sinocyclocheilus rhinocerous DDX41. CiDDX41 is highly expressed in kidney, intestines, liver and spleen. Their expressions are up-regulated more obviously after the treatment with GCRV. Over-expression of CiDDX41 in CIK cells increases the transcription level of grass carp IFN I and ISG15. On the contrary, knockdown of CiDDX41 inhibits the IFN I and ISG15 transcription. Moreover, a part of CiDDX41 translocates from the nuclear to cytoplasm to interact with grass carp STING (CiSTING). In CIK cells, overexpression of CiDDX41 and CiSTING can promote the phosphorylation and nuclear-cytoplasm translocation of grass carp IRF7 (CiIRF7) and then acutely up-regulate the IFN I and ISG15 expression. However, the knockdown of CiDDX41 inhibits the phosphorylation IRF7. Taken together, all these results above suggested that CiDDX41 performs as an activator for innate immune through STING-IRF7 mediated signaling pathway.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Although next-generation sequencing-based panel testing is well practiced in the field of cancer medicine for the identification of target molecules in solid tumors, the clinical utility and clinical issues surrounding panel testing in hematological malignancies have yet to be fully evaluated. We conducted a multicenter prospective clinical sequencing study to verify the feasibility of a panel test for hematological tumors, including acute myeloid leukemia, acute lymphoblastic leukemia, multiple myeloma, and diffuse large B-cell lymphoma. Out of 96 eligible patients, 79 patients (82%) showed potentially actionable findings, based on the clinical sequencing assays. We identified that genetic alterations with a strong clinical significance were found at a higher frequency in terms of diagnosis (n = 60; 63%) and prognosis (n = 61; 64%) than in terms of therapy (n = 8; 8%). Three patients who harbored a germline mutation in either DDX41 (n = 2) or BRCA2 (n = 1) were provided with genetic counseling. At 6 mo after sequencing, clinical actions based on the diagnostic (n = 5) or prognostic (n = 3) findings were reported, but no patients were enrolled in a clinical trial or received targeted therapies based on the sequencing results. These results suggest that panel testing for hematological malignancies would be feasible given the availability of useful diagnostic and prognostic information. This study is registered with the UMIN Clinical Trial Registry (UMIN000029879, multiple myeloma; UMIN000031343, adult acute myeloid leukemia; UMIN000033144, diffuse large B-cell lymphoma; and UMIN000034243, childhood leukemia).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   With the advent of precision genomics, hereditary predisposition to hematopoietic neoplasms- collectively known as hereditary predisposition syndromes (HPS)-are being increasingly recognized in clinical practice. Familial clustering was first observed in patients with leukemia, which led to the identification of several germline variants, such as RUNX1, CEBPA, GATA2, ANKRD26, DDX41, and ETV6, among others, now established as HPS, with tendency to develop myeloid neoplasms. However, evidence for hereditary predisposition is also apparent in lymphoid and plasma--cell neoplasms, with recent discoveries of germline variants in genes such as IKZF1, SH2B3, PAX5 (familial acute lymphoblastic leukemia), and KDM1A/LSD1 (familial multiple myeloma). Specific inherited bone marrow failure syndromes-such as GATA2 haploinsufficiency syndromes, short telomere syndromes, Shwachman-Diamond syndrome, Diamond-Blackfan anemia, severe congenital neutropenia, and familial thrombocytopenias-also have an increased predisposition to develop myeloid neoplasms, whereas inherited immune deficiency syndromes, such as ataxia-telangiectasia, Bloom syndrome, Wiskott Aldrich syndrome, and Bruton agammaglobulinemia, are associated with an increased risk for lymphoid neoplasms. Timely recognition of HPS is critical to ensure safe choice of donors and/or conditioning-regimen intensity for allogeneic hematopoietic stem-cell transplantation and to enable direction of appropriate genomics-driven personalized therapies. The purpose of this review is to provide a comprehensive overview of HPS and serve as a useful reference for clinicians to recognize relevant signs and symptoms among patients to enable timely screening and referrals to pursue germline assessment. In addition, we also discuss our institutional approach toward identification of HPS and offer a stepwise diagnostic and management algorithm.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           We screened 47 subjects with DDX41 variants among 1529 subjects with myeloid neoplasms. The most common germline variants included Splice c.935 + 4A&amp;gt;T, p.T360Ifs*33, p.V152G, p.S217Ifs*4, p.R311* and p.R369*. Except for the p.R369*, no other variants have been previously reported. Clinical covariates of subjects with simple DDX41 somatic variants and germline/somatic biallelic variants are similar. The two-year overall survival (OS) of subjects with DDX41 variants was 85%. Overall response rate to demethylation therapy in subjects with DDX41 variants was 69%. The response did not correlate with the presence of a germline variant.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Studies of DNA vaccines have shown that understanding the mechanism of DNA vaccine-mediated action is the key for vaccine development. Current knowledge has shown the presence of antigen presenting cells (APCs) involving in B and T cells at the muscle injection site and the upregulation of type I interferon (IFN-I) that initiates antiviral response and benefits adaptive immunity in fish DNA vaccines. IFN-I may be triggered by expressed antigen such as the rhabdovirus G protein encoded DNA vaccine or by plasmid DNA itself through cytosolic DNA sensing. The investigating of Toll-like receptor 9, and 21 are the CpG-motif sensors in many fish species, and the cytosolic DNA receptors DDX41 and downstream STING signaling revealed the mechanisms for IFN-I production. This review article describes the recent finding of receptors for cytosolic DNA, the STING-TBK1-IRF signaling, and the possibility of turning these findings into strategies for the future development of DNA vaccines.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            DEAD-box helicase 41 (DDX41) is a key cytosolic DNA sensor playing critical roles in the regulation of type I IFN responses, and their functions have been well-characterized in mammals. However, little information is available regarding the function of fish DDX41. In this study, a DDX41 gene, named On-DDX41, was identified in Nile tilapia, Oreochromis niloticus. The predicted protein of On-DDX41 contains several structural features known in DDX41, including conserved DEADc and HELICc domains, and a conserved sequence "Asp-Glu-Ala-Asp (D-E-A-D)". On-DDX41 gene was constitutively expressed in all tissues examined, with the highest expression level observed in liver and muscle, and was inducible after poly(I:C) stimulation. Moreover, the overexpression of On-DDX41 can elicit a strong activation of both zebrafish IFN1 and IFN3 promoter in fish cells treated with poly(dA:dT). The present study thus contributes to a better understanding of the functional properties of DDX41 in fish.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Mutations affecting spliceosomal proteins are frequently found in hematological malignancies, including myelodysplastic syndromes and acute myeloid leukemia (AML). DDX41/Abstrakt is a metazoan-specific spliceosomal DEAD-box RNA helicase that is recurrently mutated in inherited myelodysplastic syndromes and in relapsing cases of AML. The genetic properties and genomic impacts of disease-causing missense mutations in DDX41 and other spliceosomal proteins have been uncertain. Here, we conduct a comprehensive analysis of the &lt;i&gt;Caenorhabditis elegans&lt;/i&gt; DDX41 ortholog, SACY-1 Biochemical analyses defined SACY-1 as a component of the &lt;i&gt;C. elegans&lt;/i&gt; spliceosome, and genetic analyses revealed synthetic lethal interactions with spliceosomal components. We used the auxin-inducible degradation system to analyze the consequence of SACY-1 depletion on the transcriptome using RNA sequencing. SACY-1 depletion impacts the transcriptome through splicing-dependent and splicing-independent mechanisms. Altered 3' splice site usage represents the predominant splicing defect observed upon SACY-1 depletion, consistent with a role for SACY-1 in the second step of splicing. Missplicing events appear more prevalent in the soma than the germline, suggesting that surveillance mechanisms protect the germline from aberrant splicing. The transcriptome changes observed after SACY-1 depletion suggest that disruption of the spliceosome induces a stress response, which could contribute to the cellular phenotypes conferred by &lt;i&gt;sacy-1&lt;/i&gt; mutant alleles. Multiple &lt;i&gt;sacy-1&lt;/i&gt; &lt;i&gt;/ddx41&lt;/i&gt; missense mutations, including the R525H human oncogenic variant, confer antimorphic activity, suggesting that their incorporation into the spliceosome is detrimental. Antagonistic variants that perturb the function of the spliceosome may be relevant to the disease-causing mutations, including DDX41, affecting highly conserved components of the spliceosome in humans.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Common variable immunodeficiency (CVID) is a primary immunodeficiency characterized by recurrent bacterial infections and defined by reduced levels of IgG, IgA, and/or IgM, insufficient response to polysaccharide vaccination, and an abnormal B-cell immunophenotype with a significantly reduced fraction of isotype-switched memory B cells. In addition to this infectious phenotype, at least one third of the patients experience autoimmune, autoinflammatory, granulomatous, and/or malignant complications. The very heterogeneous presentation strongly suggests a collection of different disease entities with somewhat different pathogeneses and most likely diverse genetic etiologies. Major progress has been made during recent years with the advent and introduction of next-generation sequencing, initially for research purposes, but more recently in clinical practice. In the present study, we performed whole exome sequencing on 20 CVID patients with autoimmunity, autoinflammation, and/or malignancy from the Danish CVID cohort with the aim to identify gene variants with a certain, possible, or potential disease-causing role in CVID. Through bioinformatics analyses, we identified variants with possible/probable disease-causing potential in nine of the patients. Of these, three patients had four variants in three different genes classified as likely pathogenic (&lt;i&gt;NFKB1, TNFAIP3&lt;/i&gt;, and &lt;i&gt;TTC37&lt;/i&gt;), whereas in six patients, we identified seven variants of possible pathogenic potential classified as variants of unknown significance (&lt;i&gt;STAT3, IL17F, IRAK4, DDX41, NLRC3, TNFRSF1A&lt;/i&gt;, and &lt;i&gt;PLCG2&lt;/i&gt;). In the remaining 11 patients, we did not identify possible genetic causes. Genetic findings were correlated to clinical disease presentation, clinical immunological phenotype, and disease complications. We suggest that the variants identified in the present work should lay the ground for future studies to functionally validate their disease-causing potential and to investigate at the mechanistic and molecular level their precise role in CVID pathogenesis. Overall, we believe that the present work contributes important new insights into the genetic basis of CVID and particular in the subset of CVID patients with a complex phenotype involving not only infection, but also autoimmunity, autoinflammation, and malignancy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    The innate immune system recognizes conserved pathogen-associated molecular patterns and produces inflammatory cytokines that direct downstream immune responses. The inappropriate localization of DNA within the cell cytosol or endosomal compartments indicates that a cell may either be infected by a DNA virus or bacterium, or has problems with its own nuclear integrity. This DNA is sensed by certain receptors that mediate cytokine production and, in some cases, initiate an inflammatory and lytic form of cell death called pyroptosis. Dysregulation of these DNA-sensing pathways is thought to contribute to autoimmune diseases and the development of cancer. In this review, we will discuss the DNA sensors Toll-like receptor 9 (TLR9), cyclic GMP-AMP synthase (cGAS), stimulator of interferon genes (STING), absent in melanoma 2 (AIM2), and interferon gamma-inducible 16 (IFI16), their ligands, and their physiological significance. We will also examine the less-well-understood DEAH- and DEAD-box helicases DHX9, DHX36, DDX41, and RNA polymerase III, each of which may play an important role in DNA-mediated innate immunity.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 &lt;b&gt;Background:&lt;/b&gt; DNA sensors are innate immune receptors that detect intracellular endogenous or exogenous DNA. They are critical to trigger immune response against DNA viral and intracellular bacterial infection, and are involved in inflammatory diseases and tumorigenesis. Recent accumulating evidences indicated that DNA sensors are also crucial for controlling the development of colorectal cancer (CRC). However, a systematic study on the expression profile of DNA sensors in CRC and their clinical significance are still lacking. &lt;b&gt;Methods:&lt;/b&gt; We investigated the expression profile of DNA sensors in CRC and their clinical significance by taking advantage of clinical CRC samples, mouse AOM/DSS treatment model, and Oncomine ® bioinformatics platform. &lt;b&gt;Results:&lt;/b&gt; Our study identified that the expression of DNA sensors, including &lt;i&gt;AIM2, DAI,&lt;/i&gt; as well as inflammasome molecules &lt;i&gt;ASC/IL-18&lt;/i&gt;, &lt;i&gt;TLR9&lt;/i&gt; and adaptor &lt;i&gt;MyD88&lt;/i&gt;, and &lt;i&gt;DDX60&lt;/i&gt; decreased in human CRC, whereas the expression of &lt;i&gt;DHX9&lt;/i&gt;, &lt;i&gt;DHX36&lt;/i&gt;, and &lt;i&gt;DDX41&lt;/i&gt; significantly increased. Among them, the expression of &lt;i&gt;AIM2/ASC/IL-18&lt;/i&gt;, &lt;i&gt;MyD88&lt;/i&gt;, &lt;i&gt;DAI&lt;/i&gt;, &lt;i&gt;DHX36&lt;/i&gt;, and &lt;i&gt;DDX60&lt;/i&gt; were associated with cancer stages. In addition, we also performed correlation analysis between DNA sensors and their main signaling molecules to explore the possible mechanisms. The results showed that there were positive correlations between &lt;i&gt;AIM2&lt;/i&gt; and &lt;i&gt;ASC/IL-18&lt;/i&gt;, &lt;i&gt;DHX9&lt;/i&gt; and &lt;i&gt;MAVS&lt;/i&gt;, and &lt;i&gt;TLR9&lt;/i&gt; and &lt;i&gt;MyD88&lt;/i&gt; expression. In addition, the gene expression patterns of some DNA sensors were confirmed by Western-blot analysis. &lt;b&gt;Conclusions:&lt;/b&gt; Our study revealed that the expression of multiple DNA sensors was deregulated in CRC and might be involved in tumor development. More importantly, the study identified that, among all these DNA sensors, AIM2, DAI, and DDX60 could be potentially critical for diagnosis, prognosis, and therapy of CRC and deserve further investigation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             TRIM21 is an interferon-stimulated E3 ligase that controls the activity of pattern-recognition signaling via ubiquitination of interferon regulatory factors and DDX41. Previous studies on the role of TRIM21 in innate immune responses have yielded contradictory results, suggesting that the role of TRIM21 is cell specific. Here, we report that bone-marrow-derived macrophages (BMDMs) generated from Trim21&lt;sup&gt;-/-&lt;/sup&gt; mice have reduced expression of mature macrophage markers. Reflecting their reduced differentiation in response to macrophage colony-stimulating factor (M-CSF), Trim21&lt;sup&gt;-/-&lt;/sup&gt; BMDMs had decreased expression of M-CSF signature genes. Although Trim21&lt;sup&gt;-/-&lt;/sup&gt; BMDMs responded normally to Toll-like receptor 9 (TLR9) activation, they produced lower levels of pro-inflammatory cytokines in response to the TLR2 agonist PAM3CSK4. In line with this, the response to infection with the Bacillus Calmette-Guérin strain of Mycobacterium bovis was also diminished in Trim21&lt;sup&gt;-/-&lt;/sup&gt; BMDMs. Our results indicate that TRIM21 controls responses to TLR2 agonists.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Germline DDX41 mutations were recently reported to cause MDS/AML and donor-derived leukemia after transplantation. While previously described in Western countries, DDX41 variants have not been reported in a Southeast Asian population. We performed targeted sequencing of blood or bone marrow samples from 109 Thai patients with myeloid malignancies. Among the 109 patients (75 MDS, 8 MPN, 11 MDS/MPN and 15 AML), the most frequent mutations were in ASXL1 (17.4%), TET2 (16.5%) and SRSF2 (12.8%), respectively. DDX41 variants were detectable in six (5.5%) cases. Four patients exhibited three presumable germline DDX41 mutations: p.S21fs (n = 2), p.F235fs (n = 1), and p.R339H (n = 1). While p.S21fs was previously reported in myeloid neoplasm, the latter two variants have not been described. Two of these cases harbored concomitant probable germline/somatic DDX41 mutations (p.S21fs/p.R525H and p.R339H/p.K494T), while the other two patients carried only somatic mutations (p.R525H and p.F438L). The p.K494T and p.F438L variants have not been previously reported. In patients with DDX41 alterations, the diagnoses were MDS with excess blasts (4), secondary AML (1) and low-risk MDS (1). In conclusion, we identified DDX41 variants in Thai patients with myeloid malignancies in which these variants could be used to assess predisposition to MDS in Southeast Asia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Duckling short beak and dwarfism syndrome virus (SBDSV), a newly identified goose parvovirus, causes devastating disease in domestic waterfowl and considerable economic losses to Chinese waterfowl industry. The molecular pathogenesis of SBDSV infection, nature and dynamics of host immune responses against SBDSV infection remained elusive. In this study, we systematically explored the relative mRNA expression profiles of major innate immune-related genes in SBDSV infected duck embryo fibroblasts. We found that SBDSV infection effectively activated host innate immune responses and resulted in significant up-regulation of IFN-β and several vital IFN-stimulated genes (ISGs). These up-regulation responses were mainly attributed to viral genomic DNA and dsRNA replication intermediates. Importantly, the expression of cGAS was significantly induced, whereas the expression of other DNA receptors including DDX41, STING, ZBP1, LSM14A and LRRFIP1 have no significant change. Furthermore, SBDSV infection also activates the up-regulation of TLR3 and inhibited the expression of TLR2 and TLR4; however, no effect was observed on the expression of TLR1, TLR5, TLR7, TLR15 and TLR21. Intriguingly, SBDSV infection significantly up-regulated the expression of RNA sensors such as MDA5 and LGP2, and resulted in a delayed but significant up-regulation of RIG-I gene. Taken together, these data indicate that host multiple sensors including DNA sensor (cGAS) and RNA sensors (TLR3, MDA5 and LGP2) are involved in recognizing a variety of different pathogen associated molecular patterns (PAMPs) including viral genomic ssDNA and dsRNA replication intermediates, which trigger an effective antiviral innate immune response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Cytosolic mitochondrial DNA (mtDNA) activates cGAS-mediated antiviral immune responses, but the mechanism by which RNA viruses stimulate mtDNA release remains unknown. Here we show that viroporin activity of influenza virus M2 or encephalomyocarditis virus (EMCV) 2B protein triggers translocation of mtDNA into the cytosol in a MAVS-dependent manner. Although influenza virus-induced cytosolic mtDNA stimulates cGAS- and DDX41-dependent innate immune responses, the nonstructural protein 1 (NS1) of influenza virus associates with mtDNA to evade the STING-dependent antiviral immunity. The STING-dependent antiviral signaling is amplified in neighboring cells through gap junctions. In addition, we find that STING-dependent recognition of influenza virus is essential for limiting virus replication in vivo. Our results show a mechanism by which influenza virus stimulates mtDNA release and highlight the importance of DNA sensing pathway in limiting influenza virus replication.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Helicase DDX41 is a cytosolic sensor capable of detecting double-stranded DNA in mammals. However, the function of DDX41 remains poorly understood in invertebrates. In a previous study, we identified the first DDX41 sensor in the black tiger shrimp &lt;i&gt;Penaeus monodon&lt;/i&gt; (&lt;i&gt;Pm&lt;/i&gt;DDX41) and showed that it played a role in anti-viral response. In the present study, we demonstrated that &lt;i&gt;PmDDX41&lt;/i&gt; was localized in the cytoplasm of shrimp hemocytes. However, &lt;i&gt;PmDDX41&lt;/i&gt; was localized in both the cytoplasm and nucleus of hemocytes in the presence of white spot syndrome virus (WSSV) infection or when stimulated by the nucleic acid mimics, poly(dA:dT) and poly(I:C). Similar results were observed when &lt;i&gt;Pm&lt;/i&gt;DDX41 was transfected into human embryonic kidney 293T (HEK293T) cells. Immunoprecipitation further demonstrated that &lt;i&gt;Pm&lt;/i&gt;DDX41 bound to biotin-labeled poly(dA:dT) but not poly(I:C). The overexpression of shrimp &lt;i&gt;Pm&lt;/i&gt;DDX41 and mouse stimulator of interferon gene (&lt;i&gt;Mm&lt;/i&gt;STING) in HEK293T cells synergistically promoted IFN-β and NF-κB promoter activity via the DEADc domain. Co-immunoprecipitation (Co-IP) also confirmed that there was an interaction between &lt;i&gt;Pm&lt;/i&gt;DDX41 and STING after stimulation with poly(dA:dT) but not poly(I:C). Our study is the first to demonstrate that &lt;i&gt;Pm&lt;/i&gt;DDX41 acts as a cytosolic DNA sensor that interacts with STING via its DEADc domain to trigger the IFN-β and NF-κB signaling pathways, thus activating antiviral innate immune responses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Germline DDX41 mutations are involved in familial myelodysplastic syndromes (MDSs) and acute myeloid leukemias (AMLs). We analyzed the prevalence and characteristics of DDX41-related myeloid malignancies in an unselected cohort of 1385 patients with MDS or AML. Using targeted next-generation sequencing, we identified 28 different germline DDX41 variants in 43 unrelated patients, which we classified as causal (n = 21) or unknown significance (n = 7) variants. We focused on the 33 patients having causal variants, representing 2.4% of our cohort. The median age was 69 years; most patients were men (79%). Only 9 patients (27%) had a family history of hematological malignancy, and 15 (46%) had a personal history of cytopenia years before MDS/AML diagnosis. Most patients had a normal karyotype (85%), and the most frequent somatic alteration was a second DDX41 mutation (79%). High-risk DDX41 MDS/AML patients treated with intensive chemotherapy (n = 9) or azacitidine (n = 11) had an overall response rate of 100% or 73%, respectively, with a median overall survival of 5.2 years. Our study highlights that germline DDX41 mutations are relatively common in adult MDS/AML, often without known family history, arguing for systematic screening. Salient features of DDX41-related myeloid malignancies include male preponderance, frequent preexisting cytopenia, additional somatic DDX41 mutation, and relatively good outcome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Herpes simplex virus type 1 (HSV-1), one of the human pathogens widely epidemic and transmitted among various groups of people in the world, often causes symptoms known as oral herpes or lifelong asymptomatic infection. HSV-1 employs many sophisticated strategies to escape host antiviral immune response based on its multiple coding proteins. However, the functions involved in the immune evasion of miRNAs encoded by HSV-1 during lytic (productive) infection remain poorly studied. Dual-luciferase reporter gene assay and bioinformatics revealed that Asp-Glu-Ala-Asp (DEAD)-box helicase 41 (DDX41), a cytosolic DNA sensor of the DNA-sensing pathway, was a putative direct target gene of HSV-1-encoded miR-H2-3p. The transfection of miR-H2-3p mimics inhibited the expression of DDX41 at the level of mRNA and protein, as well as the expression of interferon beta (IFN-β) and myxoma resistance protein I (MxI) induced by HSV-1 infection in THP-1 cells, and promoted the viral replication and its gene transcription. However, the transfection of miR-H2-3p inhibitor showed opposite effects. This finding indicated that HSV-1-encoded miR-H2-3p attenuated cytosolic DNA-stimulated antiviral immune response by manipulating host DNA sensor molecular DDX41 to enhance virus replication in cultured cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         In myeloid neoplasms, deletions of the long arm of chromosome 5 del(5q) and 7 (-7/del(7q) ) are common karyotypic abnormalities. The concurrence of del(5q) and -7/del(7q) accounts for poor prognosis in myelodysplastic syndromes (MDS) and acute myeloid leukemia (AML). Comprehensive analysis of copy number abnormalities and genetic mutations related to del(5q) and -7/del(7q) revealed previously cryptic pathophysiology, leading to frequent hemizygous/homozygous mutations and haploinsufficiency. In addition, detailed somatic mutations on chr5q were detected using whole-exome sequencing. CSNK1A1 and G3BP1 are located within the common deleted regions (CDRs) (5q31.1-5q33.1), and another driver gene DDX41 is present in the more telomeric region (5q35.3). All the genes mentioned above exhibited haploinsufficiency because of deletions, and low expression of G3BP1 and DDX41 correlated with poor survival. The related mutational events outside of chr5q, TP53 mutation is most frequently observed in del(5q) cases. Regarding -7/del(7q), 3 CDRs were located in 7q22, 7q34, and 7q35-36. Somatic mutations of the corresponding genes to each CDR (CUX1: 7q22, LUC7L2: 7q34, EZH2: 7q35-36) were identified, indicating that the loss of function or haploinsufficiency might result in the downstream pathological consequences. These recent findings have remarkably offered insights into genetic and clinical consequences in MDS/AML cases with del(5q) and -7/del(7q).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           The 2017 Workshop of the Society for Hematopathology/European Association for Haematopathology aimed to review clinical cases with germline predisposition to hematolymphoid neoplasms. The Workshop Panel reviewed 51 cases with germline mutations and rendered consensus diagnoses. Of these, six cases were presented at the meeting by the submitting pathologists. The cases submitted to the session covering germline predisposition included 16 cases with germline GATA2 mutations, 10 cases with germline RUNX1 mutations, two cases with germline CEBPA mutations, two germline TP53 mutations, and one case of germline DDX41 mutation. The most common diagnoses were acute myeloid leukemia (15 cases) and myelodysplastic syndrome (MDS, 14 cases). The majority of the submitted neoplasms occurring in patients with germline predisposition were myeloid neoplasms with germline mutations in GATA2 and RUNX1. The presence of a germline predisposition mutation is not sufficient for a diagnosis of a neoplasm until the appearance of standard diagnostic features of a hematolymphoid malignancy manifest: in general, the diagnostic criteria for neoplasms associated with germline predisposition disorders are the same as those for sporadic cases.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Myelodysplastic syndromes (MDS) and myelodysplastic/myeloproliferative neoplasms (MDS/MPN) are heterogeneous myeloid neoplasms that frequently evolve into secondary acute myeloid leukemia (sAML). Recent progress in next-generation sequencing technologies has allowed us to discover frequent mutations throughout the coding regions of MDS, MDS/MPN, and sAML, subsequently providing information on more than 60 driver genes in these diseases. As shown by many study groups recently, such driver mutations are acquired in a gene-specific fashion. DDX41 and SAMD9/SAMD9L mutations are observed in germline cells long before MDS presentation. In blood samples from healthy elderly individuals, somatic DNMT3A, TET2, and ASXL1 mutations are detected as age-related clonal hematopoiesis and supposed to be a risk factor for hematological neoplasms. Recent reports on MDS have shown that mutations in genes such as NRAS and FLT3, designated as Type I genes, were significantly associated with leukemic evolution. Another type (Type II) of genes, including RUNX1 and GATA2, has been shown to be related to the progression from low-risk to high-risk MDS. These driver mutations are significantly concomitant during disease progression. Overall, various types of driver mutations are sequentially acquired in MDS, accounting for the heterogeneity of these disorders.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Myelodysplastic syndromes and acute myeloid leukemia are sporadic for the majority of cases affecting the elderly population. Inherited cases, however, do occur. Genetic predispositions to myeloid malignancies can be classified into three categories: familial cancer syndromes associated with increased risk of various malignancies including myelodysplasia and acute myeloid leukemia such as Li-Fraumeni syndrome and constitutional mismatch repair deficiency (CMMRD); germline mutations conferring a specific increased risk of myelodysplastic syndrome and acute myeloid leukemia such as mutations in ANKRD26, CEBPA, DDX41, ETV6, GATA2, RUNX1, SRP72 genes; and finally primarily pediatric inherited bone marrow failure syndromes such as Fanconi anemia, dyskeratosis congenita, severe congenital neutropenia, Shwachman-Diamond syndrome and Diamond Blackfan anemia. The recognition of these germline syndromes is essential in the management and follow-up of patients. Herein, we review the conditions associated with hereditary myeloid leukemia with a special clinical focus on management and monitoring.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Myelodysplastic syndromes (MDS) are a heterogeneous group of myeloid malignancies characterized by peripheral blood cytopenia and dishematopoiesis and frequently progress to acute myeloid leukemia. Genetic defects play a major role in pathogenesis of MDS, including cytogenetic abnormalities, gene mutations, and abnormal gene expression. Chromosomal abnormalities have been detected in approximately 50-60% of MDS patients, including the deletions of chromosome 5q and 7q, trisomy 8, and complex karyotypes. Newer genomic technologies, such as single-nucleotide polymorphism array and next-generation sequencing, revealed the heterozygous deletions resulting in haploinsufficient gene expression (e.g., CSNK1A1, DDX41 on chromosome 5, CUX1, LUC7L2, EZH2 on chromosome 7) involved in the pathogenesis of MDS. In addition, recurrent somatic mutations in more than 50 genes have been identified in 80-90% of MDS. The most recurrent genetic mutations are involved in the RNA splicing (e.g., SF3B1, SRSF2, U2AF1, ZRSR2, LUC7L2, DDX41) and epigenetic modifications, such as histone modification (e.g., ASXL1, EZH2) and DNA methylation (e.g., TET2, DNMT3A, IDH1/IDH2). TP53 mutation is associated with aggressive disease and frequently coincides with deletion of chromosome 5q. This review summarizes the recent progress in molecular pathogenesis of MDS. A better understanding of the specific subgroups of MDS patients will also aid in the development of new therapeutic approach for MDS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       DEAD (Asp-Glu-Ala-Asp)-box polypeptide 41 (DDX41) is a member of the DEXDc family of helicases, that has recently been identified to be a crucial intracellular DNA sensor that triggers multiple signaling molecules to activate the type I interferon response. However, the precise function of DDX41 in fish during a viral infection remains unknown. In the present study, the DDX41 homolog from orange spotted grouper, Epinephelus coioides (EcDDX41), was cloned and its potential role in the immune response to a fish viral infection were investigated. EcDDX41 encodes a putative protein of 614 amino acid residues that contained two conserved domains: 1) DEADc domain; and 2) HELICc domain. The sequence analysis indicated that EcDDX41 shared 99%, 94%, and 86% identity with Asian seabass (Lates calcarifer), zebrafish (Danio rerio), and humans (Homo sapiens), respectively. EcDDX41 mRNA was present in all of the detected tissues, with the highest level of expression in the gills. The level of EcDDX41 expression was up-regulated following infection with Singapore grouper iridovirus (SGIV) or red-spotted grouper nervous necrosis virus (RGNNV) in grouper spleen (GS) cell cultures, suggesting that EcDDX41 may be involved in fish virus infection. Furthermore, EcDDX41 overexpression in GS cells significantly inhibited SGIV and RGNNV replication. EcDDX41 overexpression significantly increased the expression of antiviral and inflammatory cytokine genes, including interferon regulatory factor genes (e.g., IRF1, IRF2, IRF3, and IRF7), interferon induced genes (e.g., ISG15, ISG56, IFP35, Viperin, and MXI), and pro-inflammatory cytokine genes (e.g., TNFα, IL-1β, and IL-8). Moreover, EcDDX41 positively regulated the mitochondrial antiviral-signaling protein (MAVS) and TANK-binding kinase 1 (TBK1)-induced interferon immune response, but did mediate IRF3 activation (MITA) to evoke an interferon immune response in unstimulated cells. Together, our results provide novel insight into the role of fish DDX41 in the antiviral innate immune response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Myeloid neoplasms with germline DDX41 mutations have been incorporated into the 2017 WHO classification. Limited studies describing the clinicopathologic features and mutation profile are available. We searched for myeloid neoplasms with a DDX41 gene mutation tested by an 81-gene next-generation sequencing panel over a 7-month period. We identified 34 patients with myeloid neoplasms with DDX41 abnormalities; 26 (76%) men and 8 women (24%) [median age, 70 years], 20 acute myeloid leukemia (AML), 10 myelodysplastic syndrome (MDS), 1 chronic myelomonocytic leukemia (CMML) and 3 myeloproliferative neoplasms (MPN). Fifty-nine DDX41 variants were detected: 27 (46%) appeared somatic and 32 (54%) were presumably germline mutations. The majority of presumed germline mutations were upstream of the Helicase 2 domain (93%) and involved loss of the start codon (30%). The majority of somatic mutations were within the Helicase 2 domain (78%), with the missense mutation p.R525H being most common (67%). There was a significant difference in the location of germline or somatic mutations (P &amp;lt; .0001). Concomitant mutations were detected involving 19 genes, but only TP53 (n = 11, 32%), ASXL1 (n = 8, 24%), and JAK2 (n = 4, 12%) were recurrent. Twenty (59%) patients showed diploid cytogenetics. Twenty-three (68%) patients presented with AML or MDS-EB-2, suggesting an association with high-grade myeloid neoplasm. Patients with myeloid neoplasms carrying DDX41 mutations show male predominance (3:1), higher age at presentation, association with TP53 mutations, and association with high-grade myeloid neoplasms in our cohort at a referral cancer center setting. These findings support the recognition of myeloid neoplasms with DDX41 mutation as unique, need for germline confirmation, and further assessment of family members.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Acute erythroid leukemia (AEL) is a high-risk leukemia of poorly understood genetic basis, with controversy regarding diagnosis in the spectrum of myelodysplasia and myeloid leukemia. We compared genomic features of 159 childhood and adult AEL cases with non-AEL myeloid disorders and defined five age-related subgroups with distinct transcriptional profiles: adult, TP53 mutated; NPM1 mutated; KMT2A mutated/rearranged; adult, DDX41 mutated; and pediatric, NUP98 rearranged. Genomic features influenced outcome, with NPM1 mutations and HOXB9 overexpression being associated with a favorable prognosis and TP53, FLT3 or RB1 alterations associated with poor survival. Targetable signaling mutations were present in 45% of cases and included recurrent mutations of ALK and NTRK1, the latter of which drives erythroid leukemogenesis sensitive to TRK inhibition. This genomic landscape of AEL provides the framework for accurate diagnosis and risk stratification of this disease, and the rationale for testing targeted therapies in this high-risk leukemia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Germline syndromes in myeloid leukemias are being discovered increasingly in patients, and their identification is essential for proper medical management to yield positive health outcomes for patients and their families. There needs to be a greater appreciation of germline predisposition driving the development of hematologic malignancies within the field of myeloid malignancies. Characterization of the influence of germline mutations on the development of myeloid malignancies is ongoing by utilization of next generation sequencing data and prognostic panels. Here, we propose modifications to the utilization and analysis of genetic results, specifically to have a high index of clinical suspicion for germline predisposition, to use assays that are comprehensive for detection of these variants, and a few caveats to interpreting sequencing data. Presented are the benefits and shortcomings of prognostic panels and clinical examples of the utilization of the prognostic panel used within the Department of Pathology at The University of Chicago. The examples demonstrate that panels performed for prognostication on DNA derived from malignant cells are able to identify patients with germline syndromes, but they can lack coverage for genes that confer inherited susceptibility. Furthermore, the panels are often not designed to find duplication and deletion mutations, which calls for a need to improve assay design and bioinformatic approaches to interpret such variants using these data.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 101                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Cyclic dimeric guanosine monophosphate (c-di-GMP) is a universally conserved second messenger that contributes to the pathogenicity of numerous bacterial species. In recent years, growing evidence has shown that bacterial extracellular c-di-GMP can interact with the innate immune system and regulate host immune responses. This review summarizes our current understanding on the dual roles of bacterial c-di-GMP in pathogen-host interaction: activation of the antibacterial innate immune response through the cytosolic surveillance pathway and inhibition of innate immune defense for iron restriction.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 102                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 DDX41 has recently been identified as a new autosomal dominantly inherited cancer predisposition syndrome causing increased risk of adult onset acute myeloid leukemia (AML) or myelodysplastic syndrome (MDS). We report for the first time compound heterozygote germline missense DDX41 mutations located in the DEAD-box domain, identified in two siblings by exome sequencing. Both siblings have slight dysmorphic findings, psychomotor delays and intellectual disability, and one developed blastic plasmacytoid dendritic cell neoplasm (BPDCN) at age five. RNA-sequencing of bone marrow showed DDX41 expression including both mutations. However, the allele fraction of p.Pro321Leu accounted for 96% in the RNA-sequencing indicating this mutation to be the more significant variant. Exome sequencing of the leukemic blasts identified no additional known driver mutations. There is no pattern indicating autosomal dominantly inherited cancer predisposition in the family, but the father has sarcoidosis, which has been associated with heterozygous DDX41 mutation. We propose that bi-allelic mutations in DDX41 could potentially be a new cancer predisposition syndrome associated with delayed psychomotor development.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 103                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                DEAD (Asp-Glu-Ala-Asp) box polypeptide 41 (DDX41), a receptor belonging to DExD/H-box helicase family, acts as an intracellular DNA sensor and induces type I IFN production in mammals and fish. However, the function of avian DDX41 in innate immune response is still unknown. In this study, the full-length duck DDX41 (duDDX41) cDNA sequence was cloned for the first time and encoded a putative protein of 618 amino acid residues which showed the high sequence similarity with both zebra finch and chicken DDX41s. The duDDX41 mRNA was widely distributed in all tested tissues, especially the cerebrum, cerebellum, and liver. Overexpression of duDDX41 triggered the activation of transcription factors IRF1 and NF-κB, as well as IFN-β expression in DEFs. The DEADc domain of duDDX41 played an extremely vital role in duck type I IFN signaling pathway. Knockdown of duDDX41 by siRNA silencing dramatically decreased IFN-β expression stimulated by poly(dA:dT) or duck enteritis virus (DEV). In addition, the replication of DEV was significantly inhibited in duDDX41-expressed DEFs and was enhanced in DDX41 knockdown DEFs. These results suggest that DDX41 is an important cytosolic DNA sensor and plays a crucial role in duck antiviral innate immune response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 104                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       DDX41 is an important sensor for host recognition of DNA viruses and initiation of nuclear factor-κB (NF-κB) and IFN signaling pathways in mammals. However, its occurrence and functions in other vertebrates remain poorly defined. Here, a DDX41 ortholog [&lt;i&gt;Danio rerio&lt;/i&gt; DDX41 (&lt;i&gt;Dr&lt;/i&gt;DDX41)] with various conserved structural features to its mammalian counterparts was identified from a zebrafish model. This &lt;i&gt;Dr&lt;/i&gt;DDX41 was found to be a trafficking protein distributed in the nucleus of resting cells but transported into the cytoplasm under DNA stimulation. Two nuclear localization signal motifs were localized beside the coiled-coil domain, whereas one nuclear export signal motif existed in the DEADc domain. &lt;i&gt;Dr&lt;/i&gt;DDX41 acts as an initiator for the activation of NF-κB and IFN signaling pathways in a &lt;i&gt;Danio rerio&lt;/i&gt; STING (&lt;i&gt;Dr&lt;/i&gt;STING)-dependent manner through its DEADc domain, which is a typical performance of mammalian DDX41. These observations suggested the conservation of DDX41 proteins throughout the vertebrate evolution, making zebrafish an alternative model in understanding DDX41-mediated immunology. With this model system, we found that &lt;i&gt;Dr&lt;/i&gt;DDX41 contributes to &lt;i&gt;Dr&lt;/i&gt;STING-&lt;i&gt;Danio rerio&lt;/i&gt; STAT6 (&lt;i&gt;Dr&lt;/i&gt;STAT6)-mediated chemokine (&lt;i&gt;Danio rerio&lt;/i&gt; CCL20) production through its DEADc domain. To the best of our knowledge, this work is the first report showing that DDX41 is an upstream initiator in this newly identified signaling pathway. The &lt;i&gt;Dr&lt;/i&gt;DDX41-mediated signaling pathways play important roles in innate antibacterial immunity because knockdown of either &lt;i&gt;Dr&lt;/i&gt;DDX41 or &lt;i&gt;Dr&lt;/i&gt;STING/&lt;i&gt;Dr&lt;/i&gt;STAT6 significantly reduced the survival of zebrafish under &lt;i&gt;Aeromonas hydrophilia&lt;/i&gt; or &lt;i&gt;Edwardsiella tarda&lt;/i&gt; infection. Our findings would enrich the current knowledge of DDX41-mediated immunology and the evolutionary history of the DDX41 family.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 105                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Preclinical models indicate that DNA damage induces type I interferon (IFN), which is crucial for the induction of an anti-tumor immune response. In human cancers, however, the association between DNA damage and an immunogenic cell death (ICD), including the release and sensing of danger signals, the subsequent ER stress response and a functional IFN system, is less clear. &lt;b&gt;Methods:&lt;/b&gt; Neoadjuvant-treated colorectal liver metastases (CLM) patients, undergoing liver resection in with a curative intent, were retrospectively enrolled in this study (n=33). DNA damage (γH2AX), RNA and DNA sensors (RIG-I, DDX41, cGAS, STING), ER stress response (p-PKR, p-eIF2α, CALR), type I and type II IFN- induced proteins (MxA, GBP1), mature dendritic cells (CD208), and cytotoxic and memory T cells (CD3, CD8, CD45RO) were investigated by an immunohistochemistry whole-slide tissue scanning approach and further correlated with recurrence-free survival (RFS), overall survival (OS), radiographic and pathologic therapy response. &lt;b&gt;Results:&lt;/b&gt; γH2AX is a negative prognostic marker for RFS (HR 1.32, 95% CI 1.04-1.69, p=0.023) and OS (HR 1.61, 95% CI 1.23-2.11, p&amp;lt;0.001). A model comprising of DDX41, STING and p-PKR predicts radiographic therapy response (AUC=0.785, p=0.002). γH2AX predicts prognosis superior to the prognostic value of CD8. CALR positively correlates with GBP1, CD8 and cGAS. A model consisting of γH2AX, p-eIF2α, DDX41, cGAS, CD208 and CD45RO predicts pathological therapy response (AUC=0.944, p&amp;lt;0.001). &lt;b&gt;Conclusion:&lt;/b&gt; In contrast to preclinical models, DNA damage inversely correlated with ICD and its associated T cell infiltrate and potentially serves as a therapeutic target in CLM.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 106                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Host recognition of viral nucleic acids generated during infection leads to the activation of innate immune responses essential for early control of virus. Retrovirus reverse transcription creates numerous potential ligands for cytosolic host sensors that recognize foreign nucleic acids, including single-stranded RNA (ssRNA), RNA/DNA hybrids, and double-stranded DNA (dsDNA). We and others recently showed that the sensors cyclic GMP-AMP synthase (cGAS), DEAD-box helicase 41 (DDX41), and members of the Aim2-like receptor (ALR) family participate in the recognition of retroviral reverse transcripts. However, why multiple sensors might be required and their relative importance in &lt;i&gt;in vivo&lt;/i&gt; control of retroviral infection are not known. Here, we show that DDX41 primarily senses the DNA/RNA hybrid generated at the first step of reverse transcription, while cGAS recognizes dsDNA generated at the next step. We also show that both DDX41 and cGAS are needed for the antiretroviral innate immune response to murine leukemia virus (MLV) and HIV in primary mouse macrophages and dendritic cells (DCs). Using mice with cell type-specific knockout of the &lt;i&gt;Ddx41&lt;/i&gt; gene, we show that DDX41 sensing in DCs but not macrophages was critical for controlling &lt;i&gt;in vivo&lt;/i&gt; MLV infection. This suggests that DCs are essential &lt;i&gt;in vivo&lt;/i&gt; targets for infection, as well as for initiating the antiviral response. Our work demonstrates that the innate immune response to retrovirus infection depends on multiple host nucleic acid sensors that recognize different reverse transcription intermediates.&lt;b&gt;IMPORTANCE&lt;/b&gt; Viruses are detected by many different host sensors of nucleic acid, which in turn trigger innate immune responses, such as type I interferon (IFN) production, required to control infection. We show here that at least two sensors are needed to initiate a highly effective innate immune response to retroviruses-DDX41, which preferentially senses the RNA/DNA hybrid generated at the first step of retrovirus replication, and cGAS, which recognizes double-stranded DNA generated at the second step. Importantly, we demonstrate using mice lacking DDX41 or cGAS that both sensors are needed for the full antiviral response needed to control &lt;i&gt;in vivo&lt;/i&gt; MLV infection. These findings underscore the need for multiple host factors to counteract retroviral infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 107                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Next-generation sequencing (NGS)-based targeted gene capture panels are used to profile hematopoietic malignancies to guide prognostication and treatment decisions. Because these panels include genes associated with hereditary hematopoietic malignancies (HHMs), we hypothesized that these panels could identify pathogenic germ line variants in malignant cells, thereby identifying patients at risk for HHMs. In total, pathogenic or likely pathogenic variants in &lt;i&gt;ANKRD26&lt;/i&gt;, &lt;i&gt;CEBPA&lt;/i&gt;, &lt;i&gt;DDX41&lt;/i&gt;, &lt;i&gt;ETV6&lt;/i&gt;, &lt;i&gt;GATA2&lt;/i&gt;, &lt;i&gt;RUNX1&lt;/i&gt;, or &lt;i&gt;TP53&lt;/i&gt; were identified in 74 (21%) of 360 patients. Germ line tissue was available for 24 patients with 25 pathogenic or likely pathogenic variants with variant allele frequencies &amp;gt;0.4. Six (24%) of these 25 variants were of germ line origin. Three &lt;i&gt;DDX41&lt;/i&gt; variants, 2 &lt;i&gt;GATA2&lt;/i&gt; variants, and a &lt;i&gt;TP53&lt;/i&gt; variant previously implicated in Li-Fraumeni syndrome were of germ line origin. No likely pathogenic/pathogenic germ line variants possessed variant allele frequencies &amp;lt;0.4. This study demonstrates that NGS-based prognostic panels may identify individuals at risk for HHMs despite not being designed for this purpose. Furthermore, variants known to cause Li-Fraumeni syndrome as well as known pathogenic variants in genes such as &lt;i&gt;DDX41&lt;/i&gt; and &lt;i&gt;GATA2&lt;/i&gt; are especially likely to be of germ line origin. Thus, tumor-based panels may augment, but should not replace, comprehensive germ line-based testing and counseling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Senescence is a complex multiple factor proces, which is still poorly understood. The purpose of this study was to find the proteome of cultured human fetal lung diploid fibroblasts (2BS) of different population doubling (PD), as well as the altered proteome induced by salidroside (SAL) in 2BS cells. Proteins were identified by two-dimensional electrophoresis (2-DE) combining matrix-assisted laser desorption/ionization-time and flight mass spectrometry (MAL DI-TOF/MS). As a result, we found 16 proteins with two-fold variations in senescent cells or after SAL treatment, some being reduced such as reticulocalbin-1, heat shock protein beta-6, elongation factor 1-delta, F-actin-capping protein subunit alpha-1, and chloride intracellular channel 1. In contrast, 40S ribosomal protein SA, proteasome subunit alpha type-5, and zinc finger BED domain-containing protein 5 increased with cell age. Furthermore, heat shock protein beta-6, Zinc finger BED domain-containing protein 5 was increased in PD30 cells after 10 μM SAL treatment, whereas, elongation factor 1-delta, 6-phosphogluconolactonase, Nucleoside diphosphate kinase A, F-actin-capping protein subunit alpha-1, Probable ATP-dependent RNA helicase DDX41, Chloride intracellular channel 1, and Peroxiredoxin-6 were increased in PD50 cells after 10 μM SAL treatment. Some of these proteins were involved in the protein synthetic and degradative pathways, which emphasizes the metabolic disorder or functional impairment of cell senescence. Moreover, these proteins could be candidate biomarkers for evaluating the SAL anti-senescence effect.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 109                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               DEAD (Asp-Glu-Ala-Asp)-box polypeptide 41 (DDX41), a receptor belonging to the DExD family, has recently been identified as an intracellular DNA sensor in vertebrates. Here, we report on the identification and functional characterization of PmDDX41, the first cytosolic DNA sensor in shrimp. By searching a Penaeus monodon expressed sequence tag (EST) database (http://pmonodon.biotec.or.th), three cDNA fragments exhibiting similarity to DDX41 in various species were identified and assembled, resulting in a complete open reading frame of PmDDX41 that contains 1863-bp and encodes a putative protein of 620 amino acids. PmDDX41 shares 83% and 79% similarity to DDX41 homolog from the bee Apis florea and fruit fly Drosophila melanogaster, respectively and contains three conserved domains in the protein: DEADc domain, HELICc domain, and zinc finger domain. The transcript of PmDDX41 was detected in all tested tissues and was up-regulated upon infection with a DNA virus, white spot syndrome virus (WSSV). However, PmDDX41 mRNA expression was not significantly changed and down-regulated in response to a bacterium, Vibrio harveyi, or an RNA virus, yellow head virus (YHV), respectively, compared with the control phosphate-buffered saline-injected shrimp. Furthermore, the suppression of PmDDX41 by dsRNA-mediated gene silencing resulted in more rapid death of WSSV-infected shrimp and a significant decrease in the mRNA expression levels of several immune-related genes (PmIKKβ, PmIKKɛ, PmRelish, PmCactus, PmDorsal, PmPEN3, PmPEN5, and ALFPm6). These results suggest that PmDDX41 is involved in the antiviral response, probably via a DNA-sensing pathway that is triggered through the IκB kinase complex and leads to the activation of several immune-related genes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 110                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Recent progress in next-generation sequencing technologies allows us to discover frequent mutations throughout the coding regions of myelodysplastic syndromes (MDS), potentially providing us with virtually a complete spectrum of driver mutations in this disease. As shown by many study groups these days, such driver mutations are acquired in a gene-specific fashion. For instance, DDX41 mutations are observed in germline cells long before MDS presentation. In blood samples from healthy elderly individuals, somatic DNMT3A and TET2 mutations are detected as age-related clonal hematopoiesis and are believed to be a risk factor for hematological neoplasms. In MDS, mutations of genes such as NRAS and FLT3, designated as Type-1 genes, may be significantly associated with leukemic evolution. Another type (Type-2) of genes, including RUNX1 and GATA2, are related to progression from low-risk to high-risk MDS. Overall, various driver mutations are sequentially acquired in MDS, at a specific time, in either germline cells, normal hematopoietic cells, or clonal MDS cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 111                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The recognition of pathogenic DNA is important to the initiation of antiviral responses. Here, we report the identification of the first avian DEAD (Asp-Glu-Ala-Asp) box polypeptide 41 (DDX41), an important DNA sensor, in chicken cells. In our study, we confirmed that chDDX41 is not an interferon-inducible gene. Knockdown of chDDX41 expression by shRNA blocked the ability of DF-1 cells to mount an IFN-β response to DNA and associated viruses. ChDDX41 mRNAs could be upregulated by double-stranded DNA (dsDNA) analogue poly(dA:dT), but not by double-stranded RNA (dsRNA) analogue poly(I:C). In poly(dA:dT) stimulation assays, the immune molecules involved in the DDX41-mediated IFN-β pathway in human cells were universally upregulated in chicken cells. Via coimmunoprecipitation (Co-IP) experiments, we found that chDDX41 could strongly interact with chicken stimulator of IFN genes (chSTING). Therefore, our results suggest that chDDX41 is involved in the dsDNA- and dsDNA virus-mediated chDDX41-chSTING-IFN-β signaling pathway in chicken cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 112                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          While early presentation of familial leukemia syndromes is typical, long disease anticipation may mask cases of familial traits in seemingly spontaneous disease. Germline mutations in DDX41 gene have been discovered in several leukemia families, as well as in mostly adult patients with seemingly spontaneous disease but having strong family histories of myeloid neoplasia. As with other familial genes, DDX41 mutation carriers can develop neoplasia through acquisition of another somatic mutation, thereby affecting both DDX41 alleles. In other patients, somatic mutations of different driver genes can substitute for acquired missense DDX41 during progression. Conversely, non-familial cases with heterozygous somatic DDX41 mutations point towards other mutations that can substitute for the germ line founder DDX41 lesions. In either circumstance, total inactivation of DDX41 appears to be cell-lethal, explaining why frameshift germline lesions have not been found to be accompanied by deletions of the DDX41 locus on 5q. The precise function of the DDX41 protein is unknown; considerable evidence suggests its involvement in RNA splicing. Thus DDX41 can be included in the now large group of mutated spliceosomal genes affected in myeloid neoplasia. However, it appears that DDX4 is so far the only example of a germline spliceosomal mutation in leukemia. Clinically, recognition of DDX41 mutated cases may have implications for surveillance, assessment of prognosis, and, perhaps, for design of targeted therapies.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 113                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 114                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Stimulator of interferon genes (STING) is a DNA sensor and an important cytoplasmic adaptor for other DNA sensors, such as Z-DNA binding protein 1 (DAI), DEAD-box helicase 41 (DDX41), and interferon-γ-inducible protein 16 (IFI16). The activation of STING signaling leads to the production of type I interferons and some other pro-inflammatory cytokines, which are critical for host defense against viral infection. Recent accumulating evidences suggest that STING is also involved in tumor development. However, the role of STING signaling in tumorigenesis is complicated, and a comprehensive review is still lacking. In this paper, we provided an overview of the dual role of STING signaling in tumor development from clinical significance to fundamental mechanisms, as well as its pre-clinical application in cancer therapy.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 115                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Recently, DDX41 mutations have been identified both as germline and acquired somatic mutations in families with multiple cases of late-onset myelodysplastic syndrome (MDS) and/or acute myeloid leukemia. The majority of germline mutations are frameshift mutations suggesting loss of function with DDX41 acting as a tumor suppressor, and there is a common somatic missense mutation found in a majority of germline mutated tumors. Clinically, DDX41 mutations lead to development of high-risk MDS at an age similar to that observed in sporadic cohorts, presenting a unique challenge to hematologists in recognizing the familial context. Functionally, DDX41 has been shown to contribute to multiple pathways and processes including mRNA splicing, innate immunity and rRNA processing. Mutations in DDX41 result in aberrations to each of these in ways that could potentially impact on tumorigenesis-initiation, maintenance or progression. This review discusses the various molecular, clinical and biological aspects of myeloid malignancy predisposition due to DDX41 mutation and highlights how each of these suggest potential therapeutic opportunities through the use of pathway-specific inhibitors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 116                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The human DEAD (Asp-Glu-Ala-Asp) box protein DDX41, a member of the DEXDc helicase family, has nucleic acid-dependent ATPase and RNA and DNA translocase and unwinding activities. DDX41 is affected by somatic mutations in sporadic cases of myeloid neoplasms as well as in a biallelic fashion in 50% of patients with germline DDX41 mutations. The R525H mutation in DDX41 is thought to play important roles in the development of hereditary myelodysplastic syndrome and acute myelocytic leukemia. In this study, human DDX41 and its R525H mutant (R525H) were expressed in &lt;i&gt;Escherichia coli&lt;/i&gt; and purified. The ATPase activities of the recombinant DDX41 and R525H proteins were dependent on both ATP and double-stranded DNA (dsDNA), such as poly(dG-dC) and poly(dA-dT). High-throughput screening was performed with a dsDNA-dependent ATPase assay using the human R525H proteins. After hit confirmation and counterscreening, several small-molecule inhibitors were successfully identified. These compounds show DDX41-selective inhibitory activities.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 117                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Although genetic predisposition to haematological malignancies has long been known, genetic testing is not yet the part of the routine diagnostics. In the last ten years, next generation sequencing based studies identified novel germline mutations in the background of familial aggregation of certain haematologic disorders including myelodysplastic syndromes (MDS) and acute myeloid leukaemia (AML). This is supported by the fact that the myeloid neoplasms with genetic predisposition represent a new category in the revised 2016 World Health Organization classification. According to the new classification, these disorders are subdivided based on the clinical and genetic features, including myeloid neoplasms with germline predisposition alone, or with pre-existing platelet disorder, cytopaenias or other organ failures. The predisposing genetic factors include mutations in the RUNX1, CEBPA, GATA2, ANKRD26, ETV6, DDX41, TERC or TERT and SRP72 genes. The genes affected in these syndromes are important regulators of haemopoiesis and are frequently implicated in leukaemogenesis, providing deeper insight into the understanding of normal and malignant haemopoiesis. Despite the growing knowledge of germline predisposing events in the background of familial myeloid malignancies, the germline genetic component is still unknown in a subset of these pedigrees. Here, we present the first study of inherited myeloid malignancies in Hungary. We identified three families with apparent clustering of myeloid malignancies with nine affected individuals across these pedigrees. All tested individuals were negative for CEBPA, GATA2, RUNX1, ANKRD26, ETV6, DDX41, TERC or TERT and SRP72 mutations, suggesting the presence of so far unidentified predisposing mutations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 118                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         The cyclin-dependent kinase inhibitor p21 is an important player in stress pathways exhibiting both tumor-suppressive and oncogenic functions. Thus, expression of p21 has to be tightly controlled, which is achieved by numerous mechanisms at the transcriptional, translational, and posttranslational level. Performing immunoprecipitation of bromouridine-labeled p21 mRNAs that had been incubated before with cytoplasmic extracts of untreated HCT116 colon carcinoma cells, we identified the DEAD-box RNA helicase DDX41 as a novel regulator of p21 expression. DDX41 specifically precipitates with the 3'UTR, but not with the 5'UTR, of p21 mRNA. Knockdown of DDX41 increases basal and γ irradiation-induced p21 protein levels without affecting p21 mRNA expression. Conversely, overexpression of DDX41 strongly inhibits expression of a FLAG-p21 and a luciferase construct, but only in the presence of the p21 3'UTR. Together, these data suggest that this helicase regulates p21 expression at the translational level independent of the transcriptional activity of p53. However, knockdown of DDX41 completely fails to increase p21 protein levels in p53-deficient HCT116 cells. Moreover, posttranslational up-regulation of p21 achieved in both p53&lt;sup&gt;+/+&lt;/sup&gt; and p53&lt;sup&gt;-/-&lt;/sup&gt; HCT116 cells in response to pharmaceutical inhibition of the proteasome (by MG-132) or p90 ribosomal S6 kinases (by BI-D1870) is further increased by knockdown of DDX41 only in p53-proficient but not in p53-deficient cells. Although our data demonstrate that DDX41 suppresses p21 translation without disturbing the function of p53 to directly induce p21 mRNA expression, this process indirectly requires p53, perhaps in the form of another p53 target gene or as a still undefined posttranscriptional function of p53.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 119                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        &lt;i&gt;Listeria monocytogenes&lt;/i&gt; is an intracellular Gram-positive bacterium that induces expression of type I IFNs (IFN-α/IFN-β) during infection. These cytokines are detrimental to the host during infection by priming leukocytes to undergo &lt;i&gt;L. monocytogenes&lt;/i&gt;-mediated apoptosis. Our previous studies showed that C5aR1&lt;sup&gt;-/-&lt;/sup&gt; and C3aR&lt;sup&gt;-/-&lt;/sup&gt; mice are highly susceptible to &lt;i&gt;L. monocytogenes&lt;/i&gt; infection as a result of increased IFN-β-mediated apoptosis of major leukocyte cell populations, including CD4&lt;sup&gt;+&lt;/sup&gt; and CD8&lt;sup&gt;+&lt;/sup&gt; T cells. However, the mechanisms by which C3a and C5a modulate IFN-β expression during &lt;i&gt;L. monocytogenes&lt;/i&gt; infection were not examined in these initial investigations. Accordingly, we report in this article that C5a and C3a suppress IFN-β production in response to &lt;i&gt;L. monocytogenes&lt;/i&gt; via cyclic di-AMP (c-di-AMP), a secondary messenger molecule of &lt;i&gt;L. monocytogenes&lt;/i&gt;, in J774A.1 macrophage-like cells and in bone marrow-derived dendritic cells (BMDCs). Moreover, C5a and C3a suppress IFN-β production by acting through their respective receptors, because no inhibition was seen in C5aR1&lt;sup&gt;-/-&lt;/sup&gt; or C3aR&lt;sup&gt;-/-&lt;/sup&gt; BMDCs, respectively. C5a and C3a suppress IFN-β production in a manner that is dependent on Bruton's tyrosine kinase, p38 MAPK, and TANK-binding kinase 1 (TBK1), as demonstrated by the individual use of Bruton's tyrosine kinase, p38 MAPK, and TBK1 inhibitors. Pretreatment of cells with C5a and C3a reduced the expression of the IFN-β signaling molecules DDX41, STING, phosphorylated TBK1, and phosphorylated p38 MAPK in wild-type BMDCs following treatment with c-di-AMP. Collectively, these data demonstrate that C3a and C5a, via direct signaling through their specific receptors, suppress IFN-β expression by modulation of a distinct innate cytosolic surveillance pathway involving DDX41, STING, and other downstream molecular targets of &lt;i&gt;L. monocytogenes&lt;/i&gt;-generated c-di-AMP.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 120                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 121                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  The use of molecular adjuvants to improve the immunogenicity of DNA vaccines has been thoroughly studied in recent years. Glycoprotein (G)-based DNA vaccines had been proven to be effective in combating infection against Rhabdovirus (especially infectious hematopoietic necrosis virus, IHNV) in salmonids. DDX41 is a helicase known to induce antiviral and inflammatory responses by inducing a type I IFN innate immune response. To gain more information regarding G-based DNA vaccines in olive flounder (Paralicthys olivaceus), we tried to develop a more efficient G-based DNA vaccine by adding a molecular adjuvant, DDX41. We designed a DNA vaccine in which the VHSV glycoprotein (G-protein) and DDX41 were driven by the EF-1α and CMV promoters, respectively. Olive flounders were intramuscularly immunized with 1 μg of plasmids encoding the G-based DNA vaccine alone (pEF-G), the molecular adjuvant alone (pEF-D), or the vaccine-adjuvant construct (pEF-GD). At two different time points, 15 and 30 days later, the fish were intraperitoneally infected with VHSV (100 μL; 1 × 10&lt;sup&gt;6&lt;/sup&gt; TCID&lt;sub&gt;50&lt;/sub&gt;/mL). Our assays revealed that the plasmid constructs showed up-regulated expression of IFN-1 and its associated genes at day 3 post-vaccination in both kidney and spleen samples. Specifically, pEF-GD showed statistically higher expression of immune response genes than pEF-G and pEF-D treated group (p &amp;lt; 0.05/p &amp;lt; 0.001). After VHSV challenge, the fish group treated with pEF-GD showed higher survival rate than the pEF-G treated group, though difference was not statistically significant in the 15 dpv challenged group however in the 30 dpv challenged group, the difference was statistically significant (p &amp;lt; 0.05). Together, these results clearly demonstrate that DDX41 is an effective adjuvant for the G-based DNA vaccine in olive flounder. Our novel findings could facilitate the development of more effective DNA vaccines for the aquaculture industry.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 122                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Deletion of chromosome 5q (del(5q)) is the most common karyotypic abnormality in myeloid neoplasms. To define the pathogenic molecular features associated with del(5q), next-generation sequencing was applied to 133 patients with myeloid neoplasms (MDS; N = 69, MDS/MPN; N = 5, sAML; N = 29, pAML; N = 30) with del(5q) as a sole abnormally or a part of complex karyotype and results were compared to molecular features of patients diploid for chr5. A number of 5q genes with haploinsufficient expression and/or recurrent somatic mutations were identified; for these genes, CSNK1A1 and G3BP1 within the commonly deleted 5q region and DDX41 within a commonly retained region were most commonly affected by somatic mutations. These genes showed consistent haploinsufficiency in deleted cases; low expression/mutations of G3BP1 or DDX41 were associated with poor survival, likely due to decreased cellular function. The most common mutations on other chromosomes in patients with del(5q) included TP53, and mutations of FLT3 (ITD or TKD), NPM1 or TET2 and were mutually exclusive. Serial sequencing allowed for definition of clonal architecture and dynamics, in patients with exome sequencing allelic imbalance for informative SNPs facilitated simultaneous approximation of clonal size of del(5q) and clonal burden for somatic mutations. Our results illuminate the spectrum of molecular defects characteristic of del(5q), their clinical impact and succession of stepwise evolution.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 124                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 125                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Several genetic syndromes have long been associated with a predisposition to the development of leukemia, including bone marrow failure syndromes, Down syndrome, and Li Fraumeni syndrome. Recent work has better defined the leukemia risk and outcomes in these syndromes. Also, in the last several years, a number of other germ line mutations have been discovered to define new leukemia predisposition syndromes, including ANKRD26, GATA2, PAX5, ETV6, and DDX41 In addition, data suggest that a substantial proportion of patients with therapy related leukemias harbor germ line mutations in DNA damage response genes such as BRCA1/2 and TP53 Recognition of clinical associations, acquisition of a thorough family history, and high index-of-suspicion are critical in the diagnosis of these leukemia predisposition syndromes. Accurate identification of patients with germ line mutations associated with leukemia can have important clinical implications as it relates to management of the leukemia, as well as genetic counseling of family members.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        The determination of an underlying genetic predisposition is not automatically part of the diagnosis of hematological malignancies (HM) in routine practice. However, it is assumed that genetic predispositions to HM are currently underestimated due to great variations in disease phenotype, variable latency and incomplete penetrance. Most of patients do not display any biological or clinical signs besides the overt hematological disease and many of them have a lack of personal or family history of malignancies. Areas covered: Collaborative studies and important advances in molecular testing have led to the discovery of several genes recurrently deregulated in familial HM including RUNX1, CEBPA, GATA2, ANKRD26, SRP72, PAX5, DDX41, ETV6, ATG2B/GSKIP and TERT/TERC. This review summarizes biological and clinical findings encountered within these disorders. Expert commentary: Identify and manage individuals with genetic predisposition is a current challenge for hematologists. Their identification has immediate implications for hematopoietic stem cell transplantation including donor selection and conditioning regimen. Importantly, several features, including familial and personal history as well as molecular and cytogenetic findings, may help clinicians to suspect an underlying genetic predisposition.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 128                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           In the innate immune system, pattern recognition receptors (PRRs) specifically recognize ligands derived from bacteria or viruses, to trigger the responsible downstream pathways. DEAD box protein 41 (DDX41) is an intracellular PRR that triggers the downstream pathway involving the adapter STING, the kinase TBK1, and the transcription factor IRF3, to activate the type I interferon response. DDX41 is unique in that it recognizes two different ligands; i.e., double-stranded DNA (dsDNA) and cyclic dinucleotides (CDN), via its DEAD domain. However, the structural basis for the ligand recognition by the DDX41 DEAD domain has remained elusive. Here, we report two crystal structures of the DDX41 DEAD domain in apo forms, at 1.5 and 2.2 Å resolutions. A comparison of the two crystal structures revealed the flexibility in the ATP binding site, suggesting its formation upon ATP binding. Structure-guided functional analyses in vitro and in vivo demonstrated the overlapped binding surface for dsDNA and CDN, which is distinct from the ATP-binding site. We propose that the structural rearrangement of the ATP binding site is crucial for the release of ADP, enabling the fast turnover of DDX41 for the dsDNA/CDN-induced STING activation pathway.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 129                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               RNA helicases are involved in almost every aspect of RNA, from transcription to RNA decay. DExD/H-box helicases comprise the largest SF2 helicase superfamily, which are characterized by two conserved RecA-like domains. In recent years, an increasing number of unexpected functions of these proteins have been discovered. They play important roles not only in innate immune response but also in diseases like cancers and chronic hepatitis C. In this review, we summarize the recent literatures on one member of the SF2 superfamily, the DEAD-box protein DDX41. After bacterial or viral infection, DNA or cyclic-di-GMP is released to cells. After phosphorylation of Tyr414 by BTK kinase, DDX41 will act as a sensor to recognize the invaders, followed by induction of type I interferons (IFN). After the immune response, DDX41 is degraded by the E3 ligase TRIM21, using Lys9 and Lys115 of DDX41 as the ubiquitination sites. Besides the roles in innate immunity, DDX41 is also related to diseases. An increasing number of both inherited and acquired mutations in DDX41 gene are identified from myelodysplastic syndrome and/or acute myeloid leukemia (MDS/AML) patients. The review focuses on DDX41, as well as its homolog Abstrakt in Drosophila, which is important for survival at all stages throughout the life cycle of the fly.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 130                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Various genetic disorders are known to be associated with cancer predisposition. For example, children with Down syndrome are predisposed to developing acute myeloid leukemia, and those with RASopathies, such as Noonan syndrome, are predisposed to juvenile myelomonocytic leukemia. To date, more than 250 diseases or syndromes have been reported to be associated with the development of pediatric cancers. Recently, the advent of the massive parallel sequencing technique revealed several germline mutations, including RUNX1, CEBPA, GATA2, SRP72, ETV6, and DDX41, which are associated with familial myeloid malignancies. A significant number of children with myeloid malignancies may harbor pathognomonic germline variants. It is strongly recommended that precise diagnosis, genetic counseling, familial screening, and follow-up programs be provided for patients with such a predisposition to cancer. To identify genetic disorders associated with predispositions to pediatric myeloid malignancies, the development of an efficient screening system with the massive parallel sequencer for germline and somatic mutations, which would also be useful for familial genetic studies and prediction of tumor progression, is needed.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 131                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               During the past ten years, genome-wide analysis of genetic alterations in myelodysplastic syndromes (MDS) has improved our understanding of their pathogenesis. Especially, single nucleotide polymorphism array karyotyping and next-generation sequencing technologies (NGS) have unveiled frequent genetic changes in novel functional pathways, including DNA methylation, RNA splicing and cohesin complex formation, in MDS. Moreover, NGS shed light on the clonal evolution which occurs during the development and progression of MDS, pre-cancerous lesions of MDS, and the effects of germline mutations in MDS.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 132                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Myelodysplastic syndromes (MDS) are heterogeneous clonal hematopoietic disorders characterized by ineffective hematopoiesis, bone marrow dysplasia, and peripheral cytopenias. Familial forms of MDS have traditionally been considered rare, especially in adults; however, the increasing availability of somatic and germline genetic analyses has identified multiple susceptibility loci. Bone marrow failure syndromes have been well-described in the pediatric setting, e.g., Fanconi anemia (FA), dyskeratosis congenita (DC), Diamond-Blackfan anemia (DBA), and Shwachman-Diamond syndrome (SBS), hallmarked by clinically-recognizable phenotypes (e.g., radial ray anomalies in FA) and significantly increased risks for MDS and/or acute myeloid leukemia (AML) in the setting of bone marrow failure. However, additional families with multiple cases of MDS or AML have long been reported in the medical literature with little known regarding potential hereditary etiologies. Over the last decade, genomic investigation of such families has revealed multiple genes conferring inherited risks for MDS and/or AML as the primary malignancy, including RUNX1, ANKRD26, DDX41, ETV6, GATA2, and SRP72. As these syndromes are increasingly appreciated in even apparently de novo presentations of MDS, it is important for hematologists/oncologists to become familiar with these newly-described syndromes. Herein, we provide a review of familial MDS syndromes and practical aspects of management in patients with predisposition syndromes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 133                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Although multiple predispositions to hematologic malignancies exist, evaluations for hereditary cancer syndromes (HCS) are underperformed by most hematologist/oncologists. Criteria for initiating HCS evaluation are poorly defined, and results of genetic testing for hereditary hematologic malignancies have not been systematically reported. From April 2014 to August 2015, 67 patients were referred to the Hereditary Hematologic Malignancy Clinic (HHMC). Referral reasons included (1) bone marrow failure or myelodysplastic syndrome in patients ≤ 50 years, (2) evaluation for germ-line inheritance of identified RUNX1, GATA2, or CEBPA mutations on targeted next-generation sequencing panels, and (3) strong personal and/or family history of malignancy. Cultured skin fibroblasts were utilized for germ-line DNA in all patients with hematologic malignancy. Eight patients (12%) were clinically diagnosed with a HCS: 4 patients with RUNX1-related familial platelet disorder (FPD)/acute myeloid leukemia (AML), and 1 patient each with dyskeratosis congenita, Fanconi anemia, germ-line DDX41, and Li-Fraumeni syndrome (LFS). Two patients with concern for FPD/AML and LFS, respectively, had RUNX1 and TP53 variants of unknown significance. Additionally, 4 patients with prior HCS diagnosis (1 LFS, 3 FPD/AML) were referred for further evaluation and surveillance. In this HHMC-referred hematologic malignancy cohort, HCS was confirmed in 12 patients (18%). HCS identification provides insight for improved and individualized treatment, as well as screening/surveillance opportunities for family members. The HHMC has facilitated HCS diagnosis; with increased clinical awareness of hematologic malignancy predisposition syndromes, more patients who may benefit from evaluation can be identified. Mutation panels intended for prognostication may provide increased clinical suspicion for germ-line testing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 134                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           The DDX41 gene, encoding a DEAD-box type ATP-dependent RNA helicase, is rarely but reproducibly mutated in myeloid diseases. The acquired mutation in DDX41 is highly concentrated at c.G1574A (p.R525H) in the conserved motif VI located at the C-terminus of the helicase core domain where ATP interacts and is hydrolyzed. Therefore, it is likely that the p.R525H mutation perturbs ATPase activity in a dominant-negative manner. In this study, we screened for the DDX41 mutation of CD34-positive tumor cells based on mRNA sequencing and identified the p.R525H mutation in three cases among 23 patients. Intriguingly, these patients commonly exhibited acute myeloid leukemia (AML) with peripheral blood cytopenias and low blast counts, suggesting that the mutation inhibits the growth and differentiation of hematopoietic cells. Data from cord blood cells and leukemia cell lines suggest a role for DDX41 in preribosomal RNA processing, in which the expression of the p.R525H mutant causes a certain ribosomopathy phenotype in hematopoietic cells by suppressing MDM2-mediated RB degradation, thus triggering the inhibition of E2F activity. This study uncovered a pathogenic role of p.R525H DDX41 in the slow growth rate of tumor cells. Age-dependent epigenetic alterations or other somatic changes might collaborate with the mutation to cause AML.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 135                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 136                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 137                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 138                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Myelodysplastic syndrome and acute myeloid leukaemia are mainly sporadic diseases, however, rare familial cases exist. These disorders are considered rare, but are likely to be more common than currently appreciated, and are characterized by the autosomal dominant mutations of hematopoietic transcription factors. These syndromes have typical phenotypic features and are associated with an increased risk for developing overt malignancy. Currently, four recognized syndromes could be separated: familial acute myeloid leukemia with mutated CEBPA, familial myelodysplastic syndrome/acute myeloid leukemia with mutated GATA2, familial platelet disorder with propensity to myeloid malignancy with RUNX1 mutations, and telomere biology disorders due to mutations of TERC or TERT. Furthermore, there are new, emerging syndromes associated with germline mutations in novel genes including ANKRD26, ETV6, SRP72 or DDX41. This review will discuss the current understanding of the genetic basis and clinical presentation of familial leukemia and myelodysplasia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 139                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Recently our group and others have identified DDX41 mutations both as germ line and acquired somatic mutations in families with multiple cases of late onset myelodysplastic syndrome (MDS) and/or acute myeloid leukemia (AML), suggesting that DDX41 acts as a tumor suppressor. To determine whether novel DDX41 mutations could be identified in families with additional types of hematologic malignancies, our group screened two cohorts of families with a diverse range of hematologic malignancy subtypes. Among 289 families, we identified nine (3%) with DDX41 mutations. As previously observed, MDS and AML were the most common malignancies, often of the erythroblastic subtype, and 1 family displayed early-onset follicular lymphoma. Five novel mutations were identified, including missense mutations within important functional domains and start-loss and splicing mutations predicted to result in truncated proteins. We also show that most asymptomatic mutation carriers have normal blood counts until malignancy develops. This study expands both the mutation and phenotypic spectra observed in families with germ line DDX41 mutations. With an increasing number of both inherited and acquired mutations in this gene being identified, further study of how DDX41 disruption leads to hematologic malignancies is critical. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 140                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Myelodysplastic syndrome (MDS) is a clonal blood disorder characterized by ineffective hematopoiesis, cytopenias, dysplasia and an increased risk of acute myeloid leukemia (AML). With the growing availability of clinical genetic testing, there is an increasing appreciation that a number of genetic predisposition syndromes may underlie apparent de novo presentations of MDS/AML, particularly in children and young adults. Recent findings of clonal hematopoiesis in acquired aplastic anemia add another facet to our understanding of the mechanisms of MDS/AML predisposition. As more predisposition syndromes are recognized, it is becoming increasingly important for hematologists and oncologists to have familiarity with the common as well as emerging syndromes, and to have a systematic approach to diagnosis and screening of at risk patient populations. Here, we provide a practical algorithm for approaching a patient with a suspected MDS/AML predisposition, and provide an in-depth review of the established and emerging familial MDS/AML syndromes caused by mutations in the ANKRD26, CEBPA, DDX41, ETV6, GATA2, RUNX1, SRP72 genes. Finally, we discuss recent data on the role of somatic mutations in malignant transformation in acquired aplastic anemia, and review the practical aspects of MDS/AML management in patients and families with predisposition syndromes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 141                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Viral RNA represents a pattern molecule that can be recognized by RNA sensors in innate immunity. Humans and mice possess cytoplasmic DNA/RNA sensors for detecting viral replication. There are a number of DEAD (Asp-Glu-Ala-Asp; DExD/H) box-type helicases in mammals, among which retinoic acid-inducible gene 1 (RIG-I) and melanoma differentiation-associated protein 5 (MDA50) are indispensable for RNA sensing; however, they are functionally supported by a number of sensors that directly bind viral RNA or replicative RNA intermediates to convey signals to RIG-I and MDA5. Some DEAD box helicase members recognize DNA irrespective of the origin. These sensors transmit IFN-inducing signals through adaptors, including mitochondrial antiviral signaling. Viral double-stranded RNAs are reportedly sensed by the helicases DDX1, DDX21, DHX36, DHX9, DDX3, DDX41, LGP2 and DDX60, in addition to RIG-I and MDA5, and induce type I IFNs, thereby blocking viral replication. Humans and mice have all nucleic acid sensors listed here. In the RNA sensing system in chicken, it was found in the present study that most DEAD box helicases are conserved; however, DHX9 is genetically deficient in addition to reported RIG-I. Based on the current genome databases, similar DHX9 deficiency was observed in ducks and several other bird species. Because chicken, but not duck, was found to be deficient in RIG-I, the RNA-sensing system of chicken lacks RIG-I and DHX9 and is thus more fragile than that of duck or mammal. DHX9 may generally compensate for the function of RIG-I and deficiency of DHX9 possibly participates in exacerbations of viral infection such as influenza in chickens. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 142                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             In this issue of Cancer Cell, Polprasert and colleagues identified recurrent mutations in the DEAD/H-box RNA helicase gene DDX41 in familial and acquired cases of myelodsyplasia and acute myeloid leukemia. These mutations induce defects in RNA splicing and represent a new class of mutations in myeloid malignancies. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 143                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Most cases of adult myeloid neoplasms are routinely assumed to be sporadic. Here, we describe an adult familial acute myeloid leukemia (AML) syndrome caused by germline mutations in the DEAD/H-box helicase gene DDX41. DDX41 was also found to be affected by somatic mutations in sporadic cases of myeloid neoplasms as well as in a biallelic fashion in 50% of patients with germline DDX41 mutations. Moreover, corresponding deletions on 5q35.3 present in 6% of cases led to haploinsufficient DDX41 expression. DDX41 lesions caused altered pre-mRNA splicing and RNA processing. DDX41 is exemplary of other RNA helicase genes also affected by somatic mutations, suggesting that they constitute a family of tumor suppressor genes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 144                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 To detect the expression of D-E-A-D-box polypeptide 41 (DDX41) in human dental pulp tissues and cells. The mRNA and protein expressions of DDX41 in human dental pulp cells were detected by RT-PCR and immunocytochemistry, and the expression of DDX41 in human dental pulp tissues was investigated by immunohistochemistry. Strong expressions of DDX41 mRNA and protein were detected in dental pulp cells. In dental pulp tissues, DDX41 was expressed in the cytoplasm and nucleus of odontoblasts. DDX41/STING-dependent TBK1-IRF3-IFN-β signaling pathway may play a role in innate immune responses of the dental pulp to caries and pulpitis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Intrinsic restriction factors and viral nucleic acid sensors are important for the anti-viral response. Here, we show how upstream sensing of retroviral reverse transcripts integrates with the downstream effector APOBEC3, an IFN-induced cytidine deaminase that introduces lethal mutations during retroviral reverse transcription. Using a murine leukemia virus (MLV) variant with an unstable capsid that induces a strong IFNβ antiviral response, we identify three sensors, IFI203, DDX41, and cGAS, required for MLV nucleic acid recognition. These sensors then signal using the adaptor STING, leading to increased production of IFNβ and other targets downstream of the transcription factor IRF3. Using knockout and mutant mice, we show that APOBEC3 limits the levels of reverse transcripts that trigger cytosolic sensing, and that nucleic acid sensing in vivo increases expression of IFN-regulated restriction factors like APOBEC3 that in turn reduce viral load. These studies underscore the importance of the multiple layers of protection afforded by host factors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 146                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           DDX41, a receptor belonging to the DExD family, functions as a DNA sensor in the mammalian cytoplasm and mediates the antiviral response in host cells. Here, the olive flounder DDX41 was found to have 2267-bp long and encodes a putative protein of 614 amino acid residues. The olive flounder DDX41 mRNA was presented in all tested tissues, and was distinctly expressed in fish naturally infected with LCDV. High expression levels were observed in the heart, liver, kidney and stomach. Furthermore, the olive flounder DDX41 mRNA expression increased significantly in adherent (monocyte-like) cells following stimulation with a DNA virus. Reporter assays showed that the transcriptional activity of the IFN-I promoter was enhanced in DDX41-overexpressing HINAE cells treated with C-di-GMP (dinucleotides). Overexpression of DDX41 also induced the antiviral and inflammatory cytokine gene expression through cytoplasmic C-di-GMP treatment. These results suggest that DDX41 functions as a cytosolic DNA sensor that is capable of inducing antiviral activity and inflammatory responses in the olive flounder. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 147                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           The innate immune system senses cytosolic dsDNA and bacterial cyclic dinucleotides and initiates signaling via the adaptor STING to induce type 1 interferon (IFN) response. We demonstrate here that BTK-deficient cells have impaired IFN-β production and TBK1/IRF3 activation when stimulated with agonists or infected with pathogens that activate STING signaling. BTK interacts with STING and DDX41 helicase. The kinase and SH3/SH2 interaction domains of BTK bind, respectively, the DEAD-box domain of DDX41 and transmembrane region of STING. BTK phosphorylates DDX41, and its kinase activities are critical for STING-mediated IFN-β production. We show that Tyr364 and Tyr414 of DDX41 are critical for its recognition of AT-rich DNA and binding to STING, and tandem mass spectrometry identifies Tyr414 as the BTK phosphorylation site. Modeling studies further indicate that phospho-Tyr414 strengthens DDX41's interaction with STING. Hence, BTK plays a critical role in the activation of DDX41 helicase and STING signaling. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 148                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Research on innate immune signaling and regulation has recently focused on pathogen recognition receptors (PRRs) and their signaling pathways. Members of PRRs sense diverse microbial invasions or danger signals, and initiate innate immune signaling pathways, leading to proinflammatory cytokines production, which, in turn, instructs adaptive immune response development. Despite the diverse functions employed by innate immune signaling to respond to a variety of different pathogens, the innate immune response must be tightly regulated. Otherwise, aberrant, uncontrolled immune responses will lead to harmful, or even fatal, consequences. Therefore, it is essential to better discern innate immune signaling and many regulators, controlling various signaling pathways, have been identified. In this review, we focus on the recent advances in our understanding of the activation and regulation of innate immune signaling in the host response to pathogens and cancer. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 149                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Rapid and robust induction of type I IFN (IFN-I) is a critical event in host antiviral innate immune response. It has been well demonstrated that cyclic GMP-AMP (cGAMP) synthase (cGAS) plays an important role in sensing cytosolic DNA and triggering STING dependent signaling to induce IFN-I. However, it is largely unknown how cGAS itself is regulated during pathogen infection and IFN-I production. In this study, we show that pattern recognition receptor (PRR) ligands, including lipid A, LPS, poly(I:C), poly(dA:dT), and cGAMP, induce cGAS expression in an IFN-I-dependent manner in both mouse and human macrophages. Further experiments indicated that cGAS is an IFN-stimulated gene (ISG), and two adjacent IFN-sensitive response elements (ISREs) in the promoter region of cGAS mediate the induction of cGAS by IFN-I. Additionally, we show that optimal production of IFN-β triggered by poly (dA:dT) or HSV-1 requires IFNAR signaling. Knockdown of the constitutively expressed DNA sensor DDX41 attenuates poly(dA:dT)-triggered IFN-β production and cGAS induction. By analyzing the dynamic expression of poly(dA:dT)-induced IFN-β and cGAS transcripts, we have found that induction of IFN-β is earlier than cGAS. Furthermore, we have provided evidence that induction of cGAS by IFN-I meditates the subsequent positive feedback regulation of DNA-triggered IFN-I production. Thus, our study not only provides a novel mechanism of modulating cGAS expression, but also adds another layer of regulation in DNA-triggered IFN-I production by induction of cGAS. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 150                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      DEAD (Asp-Glu-Ala-Asp) box polypeptide 41 (DDX41), a member of the DEXDc helicase family, was recently identified as an intracellular DNA sensor in mouse myeloid dendritic cells. In this study, porcine DDX41 (poDDX41) was cloned and its role in the type I interferon (IFN) signaling pathway was investigated in porcine kidney (PK-15) cells. Full-length poDDX41 cDNA encodes 622 amino acid residues and contains a DEADc domain and a HELICc domain. poDDX41 mRNA is widely expressed in different tissues, especially the stomach and liver. Overexpression of poDDX41 in PK-15 cells induced IFN-β by activating transcription factors IRF3 and NF-κB. Knockdown of poDDX41 with siRNA significantly reduced IFN-β expression induced by poly(dA:dT), a double-stranded DNA (dsDNA) analogue, or pseudorabies virus, a dsDNA swine virus. Therefore, poDDX41 is involved in the dsDNA- and dsDNA-virus-mediated type I IFN signaling pathway in porcine kidney cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 151                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Cyclic dinucleotides are of importance in the field of microbiology and immunology. They function as second messengers and are thought to participate in the signal transduction of cytosolic DNA immune responses. One such dinucleotide, cyclic di-GMP (c-di-GMP), stimulates the immune system. It is thought that c-di-GMP is recognized by ATP dependent RNA helicase (DDX41) in the cytosol, forms a complex with the Stimulator of interferon genes protein (STING), triggers a signal via the tank binding kinase 1-interferon regulatory factor 3 (TBK1-IRF3) pathway and induces the production of type I interferons. Therefore c-di-GMP can be thought of as a new class of adjuvant. However, because c-di-GMP contains two phosphate groups, this prevents its use as an adjuvant because it cannot pass through the cell membrane, even though the target molecule of c-di-GMP is located in the cytoplasm. Our group has been developing a series of liposomal drug delivery systems and recently investigated YSK05 which is a synthetic, pH sensitive lipid that has a high fusogenicity. We utilized this lipid as a carrier to transport c-di-GMP into the cytosol to then use c-di-GMP as an adjuvant. Based on screening experiments, YSK05/POPE/cholesterol=40/25/35 was found to induce IFN-β in Raw264.7 cells. The induction of IFN-β from c-di-GMP liposomes was inhibited by adding BX795, a TBK1 inhibitor, indicating that the production of IFN-β caused the activation of the STING-TBK1 pathway. C-di-GMP liposomes also showed significantly higher levels of expression of CD80, CD86 and MHC class I. The c-di-GMP/YSK05 liposome facilitated antigen specific cytotoxic T cell activity and the inhibition of tumor growth in a mouse model. These findings indicate that c-di-GMP/YSK05 liposomes could be used, not only to transfer c-di-GMP to the cytosol and induce an innate immune system but also as a platform for investigating the mechanism of immune sensing with cyclic dinucleotides in vitro and in vivo. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 152                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Cytoplasmic viral RNA and DNA are recognized by RIG-I-like receptors and DNA sensors that include DAI, IFI16, DDX41, and cGAS. The RNA and DNA sensors evoke innate immune responses through the IPS-1 and STING adaptors. IPS-1 and STING activate TBK1 kinase. TBK1 is phosphorylated in its activation loop, leading to IRF3/7 activation and Type I interferon (IFN) production. IPS-1 and STING localize to the mitochondria and endoplasmic reticulum, respectively, whereas it is unclear where phosphorylated TBK1 is localized in response to cytoplasmic viral DNA. Here, we investigated phospho-TBK1 (p-TBK1) subcellular localization using a p-TBK1-specific antibody. Stimulation with vertebrate DNA by transfection increased p-TBK1 levels. Interestingly, stimulation-induced p-TBK1 exhibited mitochondrial localization in HeLa and HepG2 cells and colocalized with mitochondrial IPS-1 and MFN-1. Hepatitis B virus DNA stimulation or herpes simplex virus type-1 infection also induced p-TBK1 mitochondrial localization in HeLa cells, indicating that cytoplasmic viral DNA induces p-TBK1 mitochondrial localization in HeLa cells. In contrast, p-TBK1 did not show mitochondrial localization in RAW264.7, L929, or T-23 cells, and most of p-TBK1 colocalized with STING in response to cytoplasmic DNA in those mammalian cells, indicating cell type-specific localization of p-TBK1 in response to cytoplasmic viral DNA. A previous knockout study showed that mouse IPS-1 was dispensable for Type I IFN production in response to cytoplasmic DNA. However, we found that knockdown of IPS-1 markedly reduced p-TBK1 levels in HeLa cells. Taken together, our data elucidated the cell type-specific subcellular localization of p-TBK1 and a cell type-specific role of IPS-1 in TBK1 activation in response to cytoplasmic viral DNA. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 153                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Adenovirus (Ad) infection triggers a cell-specific antiviral response following exposure of viral DNA to the intracellular compartment. A variety of DNA sensors (DAI, AIM2, DDx41, RNA polymerase [Pol] III, and IFI16 [p204]) have been identified in recent years; however, the DNA sensor involved in detection of adenovirus has not been established. Cyclic GMP-AMP synthase (cGAS), a DNA sensor that produces a cyclic guanine-adenine dinucleotide (cGAMP) inducer of STING, has been examined to determine its role in generating an antiadenoviral response. Short hairpin RNA (shRNA) lentiviral vectors targeting TBK1, STING, and cGAS were established in murine MS1 endothelial and RAW 264.7 macrophage cell lines. Knockdown of TBK1, STING, and cGAS results in a dramatic reduction in the activation of the primary antiviral response marker phosphorylated interferon (IFN) response factor 3 (IRF3) following exposure to adenovirus. Furthermore, activation of secondary type I IFN signaling targets ((ptyr)STAT1 and (ptyr)STAT2 [(ptyr)STAT1/2]) was also compromised. Consistent with compromised activation of primary and secondary response markers, transcriptional activation of IRF3-responsive genes (beta IFN [IFN-β], ISG15, ISG54) and secondary response transcripts were diminished in cells knocked down in cGAS, STING, or TBK1. These data establish cGAS as the dominant cytosolic DNA sensor responsible for detection of internalized adenovirus leading to induction of the type I interferon antiviral cascade. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 154                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Recognizing the presence of invading pathogens is key to mounting an effective innate immune response. Mammalian cells express different classes of germline-encoded pattern recognition receptors that monitor the extracellular and intracellular compartments of host cells for signs of infection and that activate several conserved signalling pathways. An efficient immune response often requires the sequential detection of a pathogen by different receptors in different subcellular compartments, which results in a complex interplay of downstream signalling pathways. In this Review, we discuss the recent identification of previously unknown pattern recognition receptors and how they complement the repertoire of established receptors. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 155                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Traditional functions of DExD/H-box helicases are concerned with RNA metabolism; they have been shown to play a part in nearly every cellular process that involves RNA. On the other hand, it is accepted that DexD/H-box helicases also engage in activities that do not require helicase activity. A number of DExD/H-box helicases have been shown to be involved in anti-viral immunity. The RIG-like helicases, RIG-I, mda5 and lgp2, act as important cytosolic pattern recognition receptors for viral RNA. Detection of viral nucleic acids by the RIG-like helicases or other anti-viral pattern recognition receptors leads to the induction of type I interferons and pro-inflammatory cytokines. More recently, additional DExD/H-box helicases have also been implicated to act as cytosolic sensors of viral nucleic acids, including DDX3, DDX41, DHX9, DDX60, DDX1 and DHX36. However, there is evidence that at least some of these helicases might have more downstream functions in pattern recognition receptor signalling pathways, as signalling adaptors or transcriptional regulators. In an interesting twist, a lot of DExD/H-box helicases have also been identified as essential host factors for the replication of different viruses, suggesting that viruses 'hijack' their RNA helicase activities for their benefit. Interestingly, DDX3, DDX1 and DHX9 are among the helicases that are required for the replication of a diverse range of viruses. This might suggest that these helicases are highly contested targets in the ongoing 'arms race' between viruses and the host immune system. This article is part of a Special Issue entitled: The Biology of RNA helicases - Modulation for life.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 156                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          DDX41 is a sensor of intracellular double-stranded DNA (dsDNA) in myeloid dendritic cells (mDCs) that triggers a type I interferon response via the signaling adaptor STING. We identified the E3 ligase TRIM21 as a DDX41-interacting protein and found that knockdown of or deficiency in TRIM21 resulted in enhanced type I interferon responses to intracellular dsDNA and DNA viruses. Overexpression of TRIM21 resulted in more degradation of DDX41 and less production of interferon-β (IFN-β) in response to intracellular dsDNA. The SPRY-PRY domain of TRIM21 interacted with the DEADc domain of DDX41. Lys9 and Lys115 of DDX41 were the targets of TRIM21-mediated ubiquitination. TRIM21 is therefore an interferon-inducible E3 ligase that induces the Lys48 (K48)-linked ubiquitination and degradation of DDX41 and negatively regulates the innate immune response to intracellular dsDNA.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 157                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 158                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The induction of type I interferons by the bacterial secondary messengers cyclic di-GMP (c-di-GMP) or cyclic di-AMP (c-di-AMP) is dependent on a signaling axis that involves the adaptor STING, the kinase TBK1 and the transcription factor IRF3. Here we identified the heliase DDX41 as a pattern-recognition receptor (PRR) that sensed both c-di-GMP and c-di-AMP. DDX41 specifically and directly interacted with c-di-GMP. Knockdown of DDX41 via short hairpin RNA in mouse or human cells inhibited the induction of genes encoding molecules involved in the innate immune response and resulted in defective activation of STING, TBK1 and IRF3 in response to c-di-GMP or c-di-AMP. Our results suggest a mechanism whereby c-di-GMP and c-di-AMP are detected by DDX41, which forms a complex with STING to signal to TBK1-IRF3 and activate the interferon response.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 159                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               In this study, we examined the capacities of non-antigen-presenting cell types to propagate antiviral signals following infection with recombinant adenovirus or by direct nucleic acid transfection. Three murine cell lines (RAW264.7 macrophages as a positive control, FL83B hepatocytes, and MS1 endothelial cells) were assessed following exposure to adenovirus, DNA, or RNA ligands. Based on primary (interferon response factor 3 [IRF3] phosphorylation) and secondary (STAT1/2 phosphorylation) response markers, we found each cell line presented a unique response profile: RAW cells were highly responsive, MS1 cells were modified in their response, and FL83B cells were essentially nonresponsive. Comparative reverse transcription-quantitative PCR (RT-qPCR) of nucleic acid sensing components revealed major differences between the three cell types. A prominent difference was at the level of adaptor molecules; TRIF, MyD88, MAVS, and STING. TRIF was absent in MS1 and FL83B cells, whereas MyD88 levels were diminished in FL83B hepatocytes. These differences resulted in compromised TLR-mediated activation. While the cytosolic adaptor MAVS was well represented in all cell lines, the DNA adaptor STING was deficient in FL83B hepatocytes (down by nearly 3 log units). The absence of STING provides an explanation for the lack of DNA responsiveness in these cells. This hypothesis was confirmed by acquisition of IRF3 activation in Flag-STING FL83B cells following DNA transfection. To consolidate the central role of adaptors in MS1 endothelial cells, short hairpin RNA (shRNA) knockdown of STING and MAVS resulted in a ligand-specific loss of IRF3 responsiveness. In contrast to the requirement for specific adaptor proteins, a requirement for a specific DNA sensor (AIM2, DDx41, or p204) in the IRF3 activation response was not detected by shRNA knockdown in MS1 cells. The data reveal that cell-specific regulation of nucleic acid sensing cascade components influences antiviral recognition responses, that controlling levels of adaptor molecules is a recurring strategy in regulating antiviral recognition response functions, and that comparative RT-qPCR has predictive value for antiviral/innate response functions in these cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 160                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Upon LPS binding, TLR4 activates a MyD88-dependent pathway leading to the transcriptional activation of proinflammatory genes, as well as a MyD88-independent/TRIF-dependent pathway, responsible for the transcriptional induction of IFN-β. Previous findings delineated that human neutrophils are unable to induce the transcription of IFN-β in response to TLR4 stimulation. Because neutrophils do not express protein kinase C ε, a molecule recently reported as essential for initiating the MyD88-independent/TRIF-dependent pathway, we optimized an electroporation method to transfect PKCε into neutrophils with very high efficiency. By doing so, a significant IFN-β mRNA expression was induced, in the absence of LPS stimulation, not only in PKCε-overexpressing neutrophils but also in cells transfected with a series of empty DNA plasmids; however, LPS further upregulated the IFN-β transcript levels in plasmid-transfected neutrophils, regardless of PKCε overexpression. Phosphoimmunoblotting studies, as well as chromatin immunoprecipitation assays targeting the IFN-β promoter, revealed that IFN-β mRNA induction occurred through the cooperative action of IRF3, activated by transfected DNA, and NF-κB, activated by LPS. Additional immunoblotting and coimmunoprecipitation studies revealed that neutrophils constitutively express various cytosolic DNA sensors, including IFN-inducible protein 16, leucine-rich repeat (in Flightless I) interacting protein-1, and DDX41, as well as that IFN-inducible protein 16 is the intracellular receptor recognizing transfected DNA. Consistently, infection of neutrophils with intracellular pathogens, such as Bartonella henselae, Listeria monocytogenes, Legionella pneumophila, or adenovirus type 5, promoted a marked induction of IFN-β mRNA expression. Taken together, these data raise questions about the role of PKCε in driving the MyD88-independent/TRIF-dependent response and indicate that human neutrophils are able to recognize and respond to microbial cytosolic DNA.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 161                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        We have used the RAW 264.7 murine macrophage-like cell line as a platform to characterize the recognition and early signaling response to recombinant adenoviral vectors (rAdV). Infection of RAW 264.7 cells triggers an early response (2 to 6 h postinfection) that includes phosphorylation of the interferon (IFN) response factor 3 (IRF3) transcription factor, upregulation of IRF3 primary response genes (interferon-stimulated gene 56 [ISG56], beta IFN [IFN-β]), and subsequent type I IFN secondary signaling (STAT1/2 phosphorylation). Using short hairpin RNA (shRNA) lentiviral vectors, we show an essential role for Tank binding kinase 1 (TBK1) in this pathway. Data also support a role for STING (MITA) as an adaptor functioning in response to rAdV infection. Using UV/psoralen (Ps)-inactivated virus to block viral transcription, Ps-inactivated virus stimulated primary (IRF3) and secondary (STAT1/2) activation events to the same degree as untreated virus. IRF3 phosphorylation was not blocked in RAW 264.7 cells pretreated with the RNA polymerase III inhibitor ML60218. However, they were compromised in the type I IFN-dependent secondary response (phosphorylation of STAT1/STAT2). At 24 h postinfection, ML60218-treated cells were compromised in the overall antiviral response. Therefore, initial sensing of rAdV or viral DNA (vDNA) does not depend on viral template transcription, but ML60218 treatment influences cellular cascades required for an antiviral response to rAdV. Using overexpression or knockdown assays, we examined how four DNA sensors influence the antiviral response. Knockdown of DNA Activator of Interferon (DAI) and p204, the murine ortholog to IFI16, had minimal influence on IRF3 phosphorylation. However, knockdown of absent in melanoma 2 (AIM2) and the helicase DDX41 resulted in diminished levels of (pser388)IRF3 following rAdV infection. Based on these data, multiple DNA sensors contribute to an antiviral DNA recognition response, leading to TBK1-dependent IRF3 phosphorylation in RAW 264.7 cells.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 162                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Most patients with acute myeloid leukaemia (AML) die from progressive disease after relapse, which is associated with clonal evolution at the cytogenetic level. To determine the mutational spectrum associated with relapse, we sequenced the primary tumour and relapse genomes from eight AML patients, and validated hundreds of somatic mutations using deep sequencing; this allowed us to define clonality and clonal evolution patterns precisely at relapse. In addition to discovering novel, recurrently mutated genes (for example, WAC, SMC3, DIS3, DDX41 and DAXX) in AML, we also found two major clonal evolution patterns during AML relapse: (1) the founding clone in the primary tumour gained mutations and evolved into the relapse clone, or (2) a subclone of the founding clone survived initial therapy, gained additional mutations and expanded at relapse. In all cases, chemotherapy failed to eradicate the founding clone. The comparison of relapse-specific versus primary tumour mutations in all eight cases revealed an increase in transversions, probably due to DNA damage caused by cytotoxic chemotherapy. These data demonstrate that AML relapse is associated with the addition of new mutations and clonal evolution, which is shaped, in part, by the chemotherapy that the patients receive to establish and maintain remissions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 163                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The sensing of pathogen-associated DNA in the cytoplasm is an important trigger of host-defense responses that include the production of type I interferon. A new study suggests that the DExDc helicase DDX41 may function in dendritic cells as a DNA sensor to activate STING-dependent innate immune responses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The recognition of pathogenic DNA is important to the initiation of antiviral responses. Here we report the identification of DDX41, a member of the DEXDc family of helicases, as an intracellular DNA sensor in myeloid dendritic cells (mDCs). Knockdown of DDX41 expression by short hairpin RNA blocked the ability of mDCs to mount type I interferon and cytokine responses to DNA and DNA viruses. Overexpression of both DDX41 and the membrane-associated adaptor STING together had a synergistic effect in promoting Ifnb promoter activity. DDX41 bound both DNA and STING and localized together with STING in the cytosol. Knockdown of DDX41 expression blocked activation of the mitogen-activated protein kinase TBK1 and the transcription factors NF-κB and IRF3 by B-form DNA. Our results suggest that DDX41 is an additional DNA sensor that depends on STING to sense pathogenic DNA.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          DEAD-box RNA helicases play various, often critical, roles in all processes where RNAs are involved. Members of this family of proteins are linked to human disease, including cancer and viral infections. DEAD-box proteins contain two conserved domains that both contribute to RNA and ATP binding. Despite recent advances the molecular details of how these enzymes convert chemical energy into RNA remodeling is unknown. We present crystal structures of the isolated DEAD-domains of human DDX2A/eIF4A1, DDX2B/eIF4A2, DDX5, DDX10/DBP4, DDX18/myc-regulated DEAD-box protein, DDX20, DDX47, DDX52/ROK1, and DDX53/CAGE, and of the helicase domains of DDX25 and DDX41. Together with prior knowledge this enables a family-wide comparative structural analysis. We propose a general mechanism for opening of the RNA binding site. This analysis also provides insights into the diversity of DExD/H- proteins, with implications for understanding the functions of individual family members.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 166                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      RNA helicases are highly conserved enzymes that utilize the energy derived from NTP hydrolysis to modulate the structure of RNA. RNA helicases participate in all biological processes that involve RNA, including transcription, splicing and translation. Based on the sequence of the helicase domain, they are classified into families, such as DDX and DHX families of human RNA helicases. The specificity of RNA helicases to their targets is likely due to several factors, such as the sequence, interacting molecules, subcellular localization and the expression pattern of the helicases. There are several examples of the involvement of RNA helicases in differentiation. Human DDX3 has two closely related genes designated DDX3Y and DDX3X, which are localized to the Y and X chromosomes, respectively. DDX3Y protein is specifically expressed in germ cells and is essential for spermatogenesis. DDX25 is another RNA helicase which has been shown to be required for spermatogenesis. DDX4 shows specific expression in germ cells. The Drosophila ortholog of DDX4, known as vasa, is required for the formation of germ cells and oogenesis by a mechanism that involves regulating the translation of mRNAs essential for differentiation. Abstrakt is the Drosphila ortholog of DDX41, which has been shown to be involved in visual and CNS system development. DDX5 (p68) and its related DDX17 (p72) have also been implicated in organ/tissue differentiation. The ability of RNA helicases to modulate the structure and thus availability of critical RNA molecules for processing leading to protein expression is the likely mechanism by which RNA helicases contribute to differentiation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 167                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               DJ-1 was identified by us as a novel oncogene in cooperation with activated ras. Although over-expression of DJ-1 has been reported in several cancer cells, including cells in breast cancer, lung cancer and prostate cancer, the precise mechanism underlying transformation has not been clarified. In this study, we screened proteins by a yeast two-hybrid method and identified Abstrakt as a DJ-1-binding protein. Abstrakt is an RNA helicase, but it has not yet been characterized. Northern blot analysis showed that human Abstrakt was expressed ubiquitously in all tissues. Abstrakt was then found to bind to and to be colocalized in the nucleus with DJ-1 in human cells. Furthermore, Abstrakt was found to stimulate transforming activity of DJ-1 in rat 3Y1 cells transfected with DJ-1 with activated ras. These findings suggest that Abstrakt is a positive regulator for DJ-1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 168                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Protein sorting through vesicular compartments is highly regulated to maintain the integrity and signaling of intracellular organelles in eukaryotic cells. Sorting Nexin-2 (SNX2) is involved in protein sorting in the trans-Golgi network, endosome, and/or lysosome compartments, with loss of function leading to defect in protein sorting and stress on organelles. To investigate the function of SNX2, we have identified the DEAD-box helicase Abstrakt (Abs) as an SNX2-interacting protein. The N-terminal domain of Abs interacts with the phox homology (PX) domain of SNX2 suggesting that PX domains may also participate in protein-protein interaction. Interestingly, both proteins undergo nucleocytoplasmic shuttling, and this process is responsive to serum withdrawal for Abs. Finally, expression of Abs reduced the cellular expression of SNX2 without altering its steady state mRNA levels. This unexpected interaction provides a novel mechanism whereby expression of proteins involved in membrane trafficking could be regulated by an RNA helicase.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     year month day                                  jabbrv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   2022    10  17                       Exp Hematol Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   2022    10  15                        Front Mol Biosci</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   2022    10  13                                Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   2022    10  04                             Front Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   2022    09  27 Am J Physiol Gastrointest Liver Physiol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   2022    09  20                             Front Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7   2022    09  28                                Leuk Res</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   2022    09  09                               FEBS Lett</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9   2022    08  27                           Br J Haematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10  2022    08  29                             J Gen Virol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11  2022    08  24                               Blood Adv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  2022    08  26                  Bone Marrow Transplant</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  2022    08  22                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  2022    08  15                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  2022    09  28                              Urol Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  2022    08  03                Proc Natl Acad Sci U S A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17  2022    07  31                     Diagnostics (Basel)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18  2022    07  19                                  EJHaem</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19  2022    09  26                  Curr Hematol Malig Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20  2022    07  16                                iScience</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21  2022    08  22                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22  2022    05  27                                Cell Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23  2022    08  09                           Br J Haematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24  2022    04  26                                 Viruses</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25  2022    08  22                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26  2022    04  01                             Front Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27  2022    04  08                          Cell Death Dis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28  2022    04  15                       Stem Cell Reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29  2022    04  25                        BMC Med Genomics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30  2022    08  05                   Rheumatology (Oxford)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31  2022    04  14                               Virus Res</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32  2021    12  29                              Nat Commun</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33  2022    02  08                        Microbiol Spectr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34  2022    04  29                           Int J Gen Med</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35  2022    01  19                            J Clin Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36  2022    07  27                     Arch Pathol Lab Med</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37  2021    12  21                          Clin Chim Acta</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38  2022    02  11                           Genes (Basel)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39  2022    03  11                                Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40  2022    04  11                               Blood Adv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41  2021    10  22                 Hered Cancer Clin Pract</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42  2022    03  15                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43  2021    11  15                          Cell Stem Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44  2022    04  26                             Front Genet</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45  2021    11  15                            Am J Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46  2022    04  25                        Ther Adv Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47  2021    05  11                         Rinsho Ketsueki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48  2021    11  09                        Am J Clin Pathol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49  2021    12  30                                Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50  2021    10  22                           Leuk Lymphoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51  2022    04  01                           Haematologica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52  2021    11  04                               Biol Chem</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53  2021    03  12                              Oncol Lett</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54  2021    08  19                                Dev Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55  2021    07  26                  Bone Marrow Transplant</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56  2021    08  30                                Dev Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57  2021    10  22                               Oncol Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58  2022    04  01                           Haematologica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59  2022    10  06                             Front Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60  2022    03  28                               Mol Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61  2021    05  07                               Poult Sci</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62  2021    06  22                           Front Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63  2021    02  16                          Nat Rev Cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64  2021    01  25              Zhonghua Xue Ye Xue Za Zhi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65  2021    08  19                           Viral Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66  2021    10  07            Best Pract Res Clin Haematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67  2021    05  14                               Blood Adv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68  2021    02  26                           Int J Mol Sci</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69  2021    04  14                              Biosci Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70  2021    11  22                                RNA Biol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71  2021    02  23                                 Sci Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72  2021    04  27                  Fish Shellfish Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73  2020    09  14                              Cancer Sci</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74  2020    12  31                          Mayo Clin Proc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75  2021    08  02                           Br J Haematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76  2021    01  12                  Fish Shellfish Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77  2020    11  30                  Fish Shellfish Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78  2021    04  21                                Genetics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79  2020    12  04                           Front Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80  2021    02  18                   Physiology (Bethesda)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81  2020    09  30                                J Cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82  2020    04  27                              Immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83  2020    08  10                           Int J Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84  2020    02  06                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85  2020    10  15                           Microb Pathog</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86  2020    02  04                              Nat Commun</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87  2020    09  29                           Front Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88  2020    02  17                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89  2020    08  24                                 Viruses</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90  2020    05  05                            Am J Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91  2019    08  19                         Rinsho Ketsueki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92  2020    03  24                        Am J Clin Pathol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93  2019    08  19                         Rinsho Ketsueki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94  2020    01  02            Best Pract Res Clin Haematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95  2019    09  03                        Int J Clin Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96  2019    10  29                  Fish Shellfish Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97  2020    03  10                            Am J Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98  2019    04  19                               Nat Genet</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99  2019    06  20            Best Pract Res Clin Haematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100 2020    07  13                           Leuk Lymphoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 101 2020    06  01                          J Innate Immun</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 102 2019    02  25                Genes Chromosomes Cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 103 2018    12  11                        Dev Comp Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 104 2019    11  20                           Front Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 105 2019    08  27                            Theranostics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 106 2019    02  25                                    mBio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 107 2019    01  30                               Blood Adv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 2018    08  27               Environ Toxicol Pharmacol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 109 2018    11  08                        Dev Comp Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 110 2017    12  22                         Rinsho Ketsueki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 111 2018    02  14                        Dev Comp Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 112 2017    12  27                           Semin Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 113 2018    09  06                           Int J Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 114 2017    09  26                             Cancer Lett</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 115 2017    10  05                           Int J Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 116 2022    08  30                             SLAS Discov</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 117 2018    08  14                        Pathol Oncol Res</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 118 2017    06  08                             J Biol Chem</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 119 2017    09  12                               J Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 120 2018    05  28                                Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 121 2017    04  10                  Fish Shellfish Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 122 2019    11  20                            Protein Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123 2018    02  22                              Oncotarget</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 124 2018    01  08                            Protein Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 125 2017    05  15  Hematology Am Soc Hematol Educ Program</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126 2017    05  18                      Expert Rev Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127 2017    09  25                                Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 128 2018    07  02                                 Sci Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 129 2017    06  05                            Protein Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 130 2017    01  06                         Rinsho Ketsueki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 131 2017    01  06                         Rinsho Ketsueki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 132 2017    03  16                           Int J Mol Sci</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 133 2017    11  03              Clin Lymphoma Myeloma Leuk</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 134 2017    05  30                             Exp Hematol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 135 2018    03  09                                Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 136 2018    01  24                           Haematologica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 137 2016    08  09                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 138 2016    04  25                               Orv Hetil</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 139 2016    08  08                                   Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 140 2016    12  13                           Leuk Lymphoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 141 2016    05  17                       Microbiol Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 142 2015    07  21                             Cancer Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 143 2015    07  20                             Cancer Cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 144 2015    09  25           Nan Fang Yi Ke Da Xue Xue Bao</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145 2016    01  01                       Cell Host Microbe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 146 2016    01  28                  Fish Shellfish Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 147 2015    12  01                                Cell Rep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 148 2015    10  23                   Hum Vaccin Immunother</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 149 2015    04  22                               J Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 150 2015    05  29                        Dev Comp Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 151 2015    01  19                       J Control Release</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 152 2014    09  29                                PLoS One</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 153 2014    03  10                                 J Virol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 154 2013    09  30                         Nat Rev Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 155 2013    07  30                    Biochim Biophys Acta</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 156 2013    03  07                             Nat Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 157 2013    01  17                             Nat Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 158 2013    01  17                             Nat Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 159 2013    01  28                                 J Virol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 160 2012    10  15                               J Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 161 2012    06  18                                 J Virol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 162 2012    03  02                                  Nature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 163 2011    11  23                             Nat Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164 2011    11  23                             Nat Immunol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165 2011    02  18                                PLoS One</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 166 2005    08  25                            Clin Biochem</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 167 2005    06  17                             Int J Oncol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 168 2005    07  26                          J Cell Physiol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                          journal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                                                         Experimental hematology &amp;amp; oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                                                             Frontiers in molecular biosciences</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                                                                                       Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                                                                          Frontiers in oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                          American journal of physiology. Gastrointestinal and liver physiology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                                                                          Frontiers in oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7                                                                              Leukemia research</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8                                                                                   FEBS letters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9                                                                 British journal of haematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10                                                               The Journal of general virology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11                                                                                Blood advances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12                                                                   Bone marrow transplantation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15                                                                             Urologic oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16               Proceedings of the National Academy of Sciences of the United States of America</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17                                                              Diagnostics (Basel, Switzerland)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18                                                                                        EJHaem</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19                                                        Current hematologic malignancy reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20                                                                                      iScience</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22                                                                                  Cell reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23                                                                British journal of haematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24                                                                                       Viruses</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26                                                                         Frontiers in oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27                                                                      Cell death &amp;amp; disease</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28                                                                             Stem cell reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29                                                                          BMC medical genomics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30                                                                Rheumatology (Oxford, England)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31                                                                                Virus research</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32                                                                         Nature communications</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33                                                                         Microbiology spectrum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34                                                     International journal of general medicine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35  Journal of clinical oncology : official journal of the American Society of Clinical Oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36                                               Archives of pathology &amp;amp; laboratory medicine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37                             Clinica chimica acta; international journal of clinical chemistry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38                                                                                         Genes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39                                                                                      Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40                                                                                Blood advances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41                                                        Hereditary cancer in clinical practice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43                                                                                Cell stem cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44                                                                         Frontiers in genetics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45                                                                American journal of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46                                                            Therapeutic advances in hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47                                 [Rinsho ketsueki] The Japanese journal of clinical hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48                                                        American journal of clinical pathology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49                                                                                      Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50                                                                       Leukemia &amp;amp; lymphoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51                                                                                 Haematologica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52                                                                          Biological chemistry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53                                                                              Oncology letters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54                                                                            Developmental cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55                                                                   Bone marrow transplantation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56                                                                            Developmental cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57                                                                              Oncology reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58                                                                                 Haematologica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59                                                                         Frontiers in oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60                                                                            Molecular oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61                                                                               Poultry science</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62                                                                       Frontiers in immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63                                                                        Nature reviews. Cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64                                          Zhonghua xue ye xue za zhi = Zhonghua xueyexue zazhi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65                                                                              Viral immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66                                            Best practice &amp;amp; research. Clinical haematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67                                                                                Blood advances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68                                                   International journal of molecular sciences</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69                                                                            Bioscience reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70                                                                                   RNA biology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71                                                                            Scientific reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72                                                               Fish &amp;amp; shellfish immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73                                                                                Cancer science</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74                                                                       Mayo Clinic proceedings</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75                                                                British journal of haematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76                                                               Fish &amp;amp; shellfish immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77                                                               Fish &amp;amp; shellfish immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78                                                                                      Genetics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79                                                                       Frontiers in immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80                                                                    Physiology (Bethesda, Md.)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81                                                                             Journal of Cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82                                                                                    Immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83                                                           International journal of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85                                                                        Microbial pathogenesis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86                                                                         Nature communications</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87                                                                       Frontiers in immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88                                                                                         Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89                                                                                       Viruses</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90                                                                American journal of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91                                 [Rinsho ketsueki] The Japanese journal of clinical hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92                                                        American journal of clinical pathology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93                                 [Rinsho ketsueki] The Japanese journal of clinical hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94                                            Best practice &amp;amp; research. Clinical haematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95                                                    International journal of clinical oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96                                                               Fish &amp;amp; shellfish immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97                                                                American journal of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98                                                                               Nature genetics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99                                            Best practice &amp;amp; research. Clinical haematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100                                                                      Leukemia &amp;amp; lymphoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 101                                                                   Journal of innate immunity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 102                                                              Genes, chromosomes &amp;amp; cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 103                                                     Developmental and comparative immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 104                                                                      Frontiers in immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 105                                                                                 Theranostics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 106                                                                                         mBio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 107                                                                               Blood advances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108                                                    Environmental toxicology and pharmacology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 109                                                     Developmental and comparative immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 110                                [Rinsho ketsueki] The Japanese journal of clinical hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 111                                                     Developmental and comparative immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 112                                                                       Seminars in hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 113                                                          International journal of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 114                                                                               Cancer letters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 115                                                          International journal of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 116                                           SLAS discovery : advancing life sciences R &amp;amp; D</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 117                                                            Pathology oncology research : POR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 118                                                          The Journal of biological chemistry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 119                                                Journal of immunology (Baltimore, Md. : 1950)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 120                                                                                     Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 121                                                              Fish &amp;amp; shellfish immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 122                                                                           Protein &amp;amp; cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123                                                                                   Oncotarget</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 124                                                                           Protein &amp;amp; cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 125                                Hematology. American Society of Hematology. Education Program</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126                                                                  Expert review of hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127                                                                                     Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 128                                                                           Scientific reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 129                                                                           Protein &amp;amp; cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 130                                [Rinsho ketsueki] The Japanese journal of clinical hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 131                                [Rinsho ketsueki] The Japanese journal of clinical hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 132                                                  International journal of molecular sciences</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 133                                                    Clinical lymphoma, myeloma &amp;amp; leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 134                                                                      Experimental hematology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 135                                                                                     Leukemia</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 136                                                                                Haematologica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 137                                                                                        Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 138                                                                               Orvosi hetilap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 139                                                                                        Blood</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 140                                                                      Leukemia &amp;amp; lymphoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 141                                                                  Microbiology and immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 142                                                                                  Cancer cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 143                                                                                  Cancer cell</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 144                       Nan fang yi ke da xue xue bao = Journal of Southern Medical University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145                                                                      Cell host &amp;amp; microbe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 146                                                              Fish &amp;amp; shellfish immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 147                                                                                 Cell reports</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 148                                                      Human vaccines &amp;amp; immunotherapeutics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 149                                                Journal of immunology (Baltimore, Md. : 1950)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 150                                                     Developmental and comparative immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 151           Journal of controlled release : official journal of the Controlled Release Society</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 152                                                                                     PloS one</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 153                                                                          Journal of virology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 154                                                                   Nature reviews. Immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 155                                                                Biochimica et biophysica acta</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 156                                                                            Nature immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 157                                                                            Nature immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 158                                                                            Nature immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 159                                                                          Journal of virology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 160                                                Journal of immunology (Baltimore, Md. : 1950)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 161                                                                          Journal of virology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 162                                                                                       Nature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 163                                                                            Nature immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164                                                                            Nature immunology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165                                                                                     PloS one</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 166                                                                        Clinical biochemistry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 167                                                            International journal of oncology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 168                                                               Journal of cellular physiology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     keywords lastname firstname address email query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9       &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99      &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 101     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 102     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 103     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 104     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 105     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 106     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 107     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 109     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 110     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 111     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 112     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 113     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 114     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 115     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 116     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 117     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 118     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 119     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 120     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 121     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 122     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 124     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 125     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 128     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 129     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 130     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 131     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 132     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 133     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 134     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 135     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 136     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 137     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 138     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 139     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 140     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 141     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 142     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 143     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 144     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 146     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 147     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 148     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 149     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 150     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 151     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 152     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 153     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 154     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 155     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 156     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 157     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 158     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 159     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 160     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 161     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 162     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 163     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 166     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; DDX41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 167     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 168     &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rawfrequency</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ddx                          patients </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               487                               227 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         mutations                               dna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               223                               193 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          germline                             cells </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               185                               146 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             genes                           myeloid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               144                               142 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        expression                          response </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               129                               112 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          variants                               mds </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               103                               101 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               rna                            immune </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               100                                97 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        identified                              cell </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                86                                81 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          leukemia                             sting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                80                                80 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           protein                          clinical </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                75                                74 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               aml                         including </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                73                                69 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    predisposition                            innate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                69                                66 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              type                           genetic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                66                                65 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ifn                         neoplasms </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                65                                64 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             acute                      malignancies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                61                                61 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             study                           somatic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                61                                59 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              gene                              risk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                58                                58 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         syndromes                              cgas </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                58                                57 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          helicase                           sensors </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                57                                57 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         helicases                         infection </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                54                                54 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            family                        sequencing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                52                                52 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             viral                         signaling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                52                                51 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        activation                         antiviral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                49                                49 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ifnβ                           pathway </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                49                                49 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           disease                          mutation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                48                                48 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              role                          syndrome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                47                                47 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            cancer                          survival </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                45                                44 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   myelodysplastic                          involved </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                43                                42 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        interferon                          familial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                41                                40 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       recognition                            sensor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                40                                40 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       development                           deadbox </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                39                                38 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          splicing                              data </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                38                                37 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        production                         responses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                37                                37 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             virus                               ifi </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                36                                35 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         molecular                            review </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                35                                35 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         cytosolic                        pathogenic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                34                                34 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           studies                              gata </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                34                                33 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             human                         increased </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                33                                33 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     hematopoietic                               irf </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                32                                32 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          multiple                               age </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                32                                30 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         diagnosis                            domain </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                30                                30 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          function                               tbk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                30                                30 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            common                          immunity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                29                                29 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          activity                          findings </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                28                                28 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              host                            kinase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                28                                28 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          pathways                          features </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                28                                27 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     intracellular                          proteins </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                27                                27 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             based                       hematologic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                26                                26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       progression                              runx </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                26                                26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            levels                              mrna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                25                                25 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           mutated                           nucleic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                25                                25 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         potential                     transcription </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                25                                25 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          critical                               dhx </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                24                                24 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             dsdna                         functions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                24                                24 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         receptors                           sensing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                24                                24 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         treatment                           unknown </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                24                                24 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           variant                          analysis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                24                                23 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            clonal                      inflammatory </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                23                                23 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          observed                          reported </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                23                                23 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              acid                              bone </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                22                                22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             cebpa                          detected </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                22                                22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             digmp                         disorders </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                22                                22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               etv                         knockdown </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                22                                22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        malignancy                            marrow </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                22                                22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        suggesting                             tumor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                22                                22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            factor                           factors </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                21                                21 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              fish                        generation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                21                                21 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              germ                            mdsaml </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                21                                21 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        mechanisms                          revealed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                21                                21 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             roles                      significance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                21                                21 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          addition                          families </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                20                                20 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       replication                               tlr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                20                                20 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           binding                            cyclic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                19                                19 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               del                         expressed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                19                                19 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           genomic                         induction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                19                                19 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           leading                         performed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                19                                19 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          acquired                     characterized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                18                                18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          diseases                           history </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                18                                18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         inherited                             model </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                18                                18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    overexpression                         prognosis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                18                                18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               sle                           testing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                18                                18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           therapy                     understanding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                18                                18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           adaptor                             adult </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          compared                         conserved </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         essential                         formation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     hematological                           induced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           primary                        regulation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          specific                   transplantation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              trim                           viruses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                17                                17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             blood                           control </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              dead                        diagnostic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             donor                               hcc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        hereditary                      inflammation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        management                            median </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            normal                              play </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        prognostic                         regulated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          spectrum                        additional </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                16                                15 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         evolution                         mechanism </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                15                                15 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           reduced                           relapse </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                15                                15 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              stem                       therapeutic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                15                                15 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               aim                          cellular </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        chromosome                           complex </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           current                         cytokines </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            damage                             death </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    identification                        increasing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       individuals                             liver </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              loop                              mice </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           pattern                   phosphorylation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          receptor                               rig </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           samples                         secondary </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       stimulation                          analyses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                14                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            anemia                   characteristics </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          distinct                             dnmta </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            donors                    doublestranded </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        downstream                            driver </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         frequency                          frequent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            highly                            induce </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           induces                      investigated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              loss                         molecules </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      presentation                         screening </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           tissues                             ankrd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                13                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       association                         bacterial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           chronic                       cytoplasmic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   differentiation                               erv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        frequently                            genome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          identify                          included </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ipss                         monocytes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            mutant                           nuclear </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             panel                            panels </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          presence                           release </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            report                           treated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            unique                     abnormalities </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                12                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            allele                          approach </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               box                    classification </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         cytoplasm                             deadc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       demonstrate                           dexdbox </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ervs                           failure </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        functional                               hbv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     hematopoiesis                             leads </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             mouse                      pathogenesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           patient                        processing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   proinflammatory                           provide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            remain                          sporadic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             stage                            stress </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            strong                          subjects </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          targeted                           targets </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               tet                   transcriptional </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                11                                11 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          activate                         activated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        activities                          analyzed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              asxl                        biological </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         cgassting                             ciddx </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     comprehensive                         confirmed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        counseling                         cytopenia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           defects                      demonstrated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         dendritic                         detection </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          disorder                        evaluation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          evidence                             exome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         favorable                               gps </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hsct                         knowledge </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             level                      localization </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          majority                         mammalian </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          missense                          pathogen </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          polydadt                       polypeptide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               pss                        recognized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         recurrent                            tested </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           trigger                        underlying </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              vivo                          abstrakt </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                10                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             acids                            adults </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ccrcc                              ccus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            cohort                        considered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        contribute                        correlated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           cytosol                               dai </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             diamp                           domains </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ecddx                         effective </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               hcs                      heterozygous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           include                       interferons </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               isg                              lung </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              male                     mitochondrial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       neutrophils                         phenotype </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             pmddx                          positive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         processes                               raw </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        regulators                          required </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            result                              sacy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               srp                            target </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              time                             yield </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         zebrafish                      aspglualaasp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 9                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            assays                         biallelic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              brca                           cancers </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           chicken                          contrast </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              core                              ddxm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        deficiency                           defined </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           discuss                           diverse </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       established                           finally </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            gmpamp                               ild </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      increasingly                        inhibition </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             major                         malignant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           mammals                             mtdna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             onset                          outcomes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             pbmcs                         pediatric </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               pef                        peripheral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        phenotypes                        phenotypic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rate                         recognize </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        regulating                           reverse </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             short                            signal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       spliceosome                              stat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        structural                          telomere </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          triggers                       upregulated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 8                                 8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      additionally                        adenovirus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          advances                        allogeneic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       alterations                       alternative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               atp                            breast </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         candidate                            caused </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              chek                          children </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           cohorts                               crc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           crucial                              cvid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          cytokine                         dependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            detect                           develop </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        difference                        dnasensing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            drervs                              duck </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           encodes                         evaluated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              fmrp                              form </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            growth                haploinsufficiency </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           healthy                              hela </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           hnrnpab                              hspc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              icus                        implicated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      implications                         incidence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        individual                         influenza </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         inhibited                        inhibitors </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       interaction                         involving </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ips                          knockout </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         leukemias                            manner </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           nucleus                             olive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           outcome                         pathogens </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          patterns                          personal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              pgvs                              poly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        population                           precise </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          promoter                              prrs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               puf                             range </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rare                              rbps </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        recipients                            rnadna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             sbdsv                             score </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          sequence                        signalling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           species                        stimulator </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         suggested                          suggests </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          synthase                         triggered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            tumour                          vaccines </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             worse                           ability </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 7                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          adjuvant                           altered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           antigen                         apoptosis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          assessed                      associations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           biology                               btk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         carcinoma                             cgamp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      chemotherapy                        classified </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          complete                        components </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        cytopenias                               dcl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        ddxmutated                          ddxmutgl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         decreased                       deleterious </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         deletions                            dental </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         depletion                         determine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            diabzi                    diseasecausing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           elderly                           encoded </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          examined                         exhibited </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               flb                          flounder </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               flt                        frameshift </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               gbp                         generated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       hbvinfected                     heterogeneous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         highlight                            humans </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               iii                            impact </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       importantly                       independent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        indicating                      inflammasome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         inhibitor                        initiation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          insights                     interestingly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               jak                            kidney </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              lead                           lesions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         leukaemia                              life </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           ligands                         localized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           located                       macrophages </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mda                           medical </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          molecule                        monitoring </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     monocytogenes                            murine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          neoplasm                       participate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         pathology                              pdcs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             poddx                        polymerase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        predictive                           process </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           profile                          progress </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              pulp                         recipient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        regulatory                             rerio </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          screened                         selection </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            sensed                            shrimp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            single                      spliceosomal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              step                    stingdependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        structures                        suppressor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         suspected                               tbd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      technologies                        transplant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            tumors                             types </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           typical                        understood </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         unrelated                      upregulation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             usurp                           vaccine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 6                                 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          aberrant                       aberrations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      accumulation                       acquisition </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         activator                            active </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              acts                           agonist </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         algorithm                             amino </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           aspects                               atm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        autoimmune                      availability </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             basis                              beta </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        biomarkers                           blocked </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               cfi                  characterization </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             chddx                             class </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        clinically                            cloned </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            clones                              cmns </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            coding                          commonly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        comparison                      compartments </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       compromised                        conclusion </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       contributes                             cycle </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         cytotoxic                             danio </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         deficient                            define </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         developed                             dexdc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           dexhbox                         diagnoses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       differences                        diminished </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     dinucleotides                            direct </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             drddx                             duddx </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         dysplasia                         establish </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            events                       expressions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ezh                           fashion </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          fraction                          fraumeni </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             grade                           hairpin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         harboring                       hepatocytes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           homolog                               hps </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               hsv                            hybrid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           improve                          improved </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          includes                          increase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          infected                           initial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           insight                          interval </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              irgs                         karyotype </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               key                              lack </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           lacking                               led </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          leukemic                            ligase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lps                              lusc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     lymphoblastic                               lys </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        macrophage                              mass </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              mavs                        metabolism </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mlv                             mrnas </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               myd                          negative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           network                             newly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              novo                          numerous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      organization                            origin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          ortholog                              pbmc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           persons                    phosphorylated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              poor                       populations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      predisposing                        prevalence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          previous                             prior </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          profiles                           propose </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          putative                              radv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       recurrently                          referred </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            region                           regions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           remains                        retroviral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          reviewed                           routine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               sal                           setting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            severe                             shrna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           signals                         signature </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              srsf                         stability </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     statistically                        stimulated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       subcellular                          subtypes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        summarizes                      surveillance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            tbkirf                              tert </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         therapies                            tissue </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             tnfα                     transcriptome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       transcripts                       transfected </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      transfection                       translation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              umin                         uncertain </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           undergo                          unveiled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           variety                       vertebrates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             vitro                               vus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          wildtype                              xrcc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               zbp                          abnormal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 5                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               abs                           absence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            absent                       accompanied </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          adaptors                               ael </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           alleles                           allowed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       appreciated                        assessment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             atlas                            atpase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      atpdependent                               auc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              bind                             blast </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             bmdms                               car </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              carp                          carriers </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          carrying                            causal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           causing                           central </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         childhood                       chromosomes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             clone                        clustering </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       combination                       concomitant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        conditions                        confidence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       consecutive                    constitutively </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       controlling                          controls </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             count                          criteria </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             curve                               cux </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      cytogenetics                         cytometry </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              days                         defective </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       degradation                           deleted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          deletion                      demonstrates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           derived                        designated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        determined                        developing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              dexd                        discovered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       distributed                      dnadependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             drive                        drosophila </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           earlier                           effects </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         efficient                          elements </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          emerging                        endogenous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          enhanced                          enrolled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            entity                             ervkp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         erythroid                         expanding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          explored                          exposure </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             focus                           focused </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            follow                             forms </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            future                      genespecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             goose                             grass </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           grouper                           growing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        guidelines                            harbor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            health                            helicc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    hematolymphoid                         hepatitis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hhms                               idh </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ifns               immunoprecipitation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         increases                         inducible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        infections                      infiltration </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         influence                          inhibits </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          initiate                          interact </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         interacts                     intermediates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         interplay                       involvement </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          involves                             issue </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        karyotypes                           kinases </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lgp                             lines </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        literature                              lucl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          lymphoid                          lymphoma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              main                          maintain </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              mapk                           markers </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            mdsmpn                       mechanistic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           mediate                          melanoma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mir                               mns </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          modulate                       monoallelic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            months                      multivariate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           myeloma                          necrosis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         neoplasia                      neutrophilic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              nlrp                               npm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             overt                      pathological </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     physiological                               pkr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             plays                          practice </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         predicted                        prediction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   predispositions                      predominance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        progenitor                   prognostication </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          promoted                        proportion </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           purpose                      quantitative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             rates                        recognizes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       recognizing                       recombinant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          referral                        regression </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          regulate                         regulator </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          remained                         remission </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           reports                        represents </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          residues                       responsible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    responsiveness                        restricted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      retroviruses                            reveal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rnas                              samd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         samdsamdl                        senescence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               set                               sfb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  shwachmandiamond                        similarity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               snx                           sorting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      spectrometry                            splice </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        stimulates                        strategies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          strategy                           studied </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        subsequent                      subsequently </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           subtype                          suppress </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         suspicion                         synthetic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          systemic                         targeting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              terc                              toll </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             total                      typespecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               tyr                          tyrosine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               uaf                    ubiquitination </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        univariate                          upstream </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             γhax                           account </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 4                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      accumulating                           acquire </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            acting                         activates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        activating                           adapter </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            adding                        adultonset </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            advent                            affect </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             aimed                            airway </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             alpha                               als </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          american                          analogue </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          analysed                        antecedent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          aplastic                            apobec </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          apparent                       application </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        approaches                             array </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           article                             assay </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            assess                      asymptomatic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              ataxiatelangiectasia                         autosomal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         awareness                        background </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         bacterium                             basic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         belonging                           benefit </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          benefits                    bioinformatics </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         biomarker                             bmdcs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            bruton                           carried </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           cascade                           caspase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          category                              cdna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         challenge                    characteristic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        checkpoint                         cibersort </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            clinic                           closely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      collectively                       colocalized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           colonic                       comparative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           concern                            confer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         congenita                        congenital </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      consequences                      conservation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      considerable                        consistent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        constitute                      correlations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           created                           crystal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             csnka                          cultured </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       cytogenetic                            danger </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          database                         databases </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           dataset                               dcs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       ddxmediated                             ddxwt </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          decision                           defense </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              defs                            degree </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         describes                            design </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          designed                          develops </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         diagnosed                      differential </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      dinucleotide                           diploid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         discovery                         discusses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        disruption                          dominant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             dsrna                              dual </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      dyskeratosis                          elevated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        elongation                           elusive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         embryonic                            enable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          enabling                          encoding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         endosomal                       endothelial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           enzymes                        epithelial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        etiologies                             evade </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           examine                          examples </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            excess                             exist </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiments                            export </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           express                     extracellular </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         extremely                       facilitated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           fanconi                           fasting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             fetal                       fibroblasts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             field                            finger </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              flow                           focuses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          founding                           fragile </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         framework                      functionally </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          genetics                           genomes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        genomewide                             graft </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    haematological                       haemoglobin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hand                 haploinsufficient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          harbored                            hazard </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               hct                     hematologists </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    hepatocellular                            herpes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     heterogeneity                      highlighting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        highlights                               hms </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           hybrids                        hypothesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       identifying                              ifit </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     ifnstimulated                             ifnα </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        immunology                      incorporated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          indolent                         initiates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        initiating                       instability </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         integrity                       interacting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              interferonstimulated                         intrinsic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       investigate                     investigation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              isgs                            korean </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         landscape                              late </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            lethal                         leukocyte </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lfs                           limited </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          limiting                            linked </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             lipid                         liposomes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              list                             loops </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     manifestation                            marker </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            mature                              mdcs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              mddx                              mdss </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          measured                          mediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          membrane                         messenger </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       micronuclei                              mngp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            models                     modifications </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       mononuclear                               mre </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        mutational       mydindependenttrifdependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    myelodysplasia                          neuronal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       neutropenia                               ngs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          nomogram                              nras </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       nutritional                          occurred </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         oncogenic                     opportunities </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             organ                         organoids </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             pamps                        panoptosis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          parallel                patternrecognition </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               pax                               pcr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        penetrance                             pkcε </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          platelet                          platform </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            played                           porcine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           possess                       possibility </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    posttransplant                         practical </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       preclinical                          predicts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        predispose                       preexisting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     presentations                          probable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          programs                       progressive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     proliferation                         promoting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        properties                          proteome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         providing                               prr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            psting                          ptyrstat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        pyroptosis                               ras </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ratio                               rbm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       recommended                        recurrence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         regulates                          released </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         relevance                          relevant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            repair                          requires </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       restriction                   retrospectively </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        retrovirus                           revised </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             risks                            rnaseq </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            robust                              saml </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            senses                         sensitive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       sensorsting                              sets </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            shares                             signs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            snorna                       specificity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            spleen                       spontaneous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            stages                        statistics </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            status                         structure </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         subgroups                            subset </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       substantial                           subunit </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           summary                          superior </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           support                        suppressed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    susceptibility                        systematic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    systematically                              term </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            timely                       trafficking </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    transformation                     translational </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        triggering                     tumorigenesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         typically                           unclear </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        understand                      undetermined </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        unselected                         unwinding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       utilization                               utr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        validation                           vectors </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          verified                        vertebrate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              vhsv                              zinc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 3                                 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         abundance                         accounted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          accuracy                          accurate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          achieved                              acmg </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      actincapping                            action </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        actionable                           actions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          adaptive                               adv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         advantage                              aged </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          agonists                          alleleic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           allelic                        alloimmune </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        alteration                          altering </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            amlmds                         amplified </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           ancient                           animals </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     antibacterial                          antibody </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           appears                           applied </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      appreciation                             april </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      architecture                              ards </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            arrest                               asc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             asian                   aspglualaaspbox </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          assembly                         assessing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            assist                           assumed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               atr                        attenuated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        attenuates                         augmented </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      autoimmunity                  autoinflammation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           average                             avian </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          bacteria                           barrier </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               bed                        biogenesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            biopsy                          blackfan </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            blasts                             block </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             bloom                             bound </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            burden                            called </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              calr                           capable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              care                             carry </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          cascades                              casp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         causative                               cdn </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              cdro                              cdrs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      cellspecific                       cellularity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          censored                        challenged </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            change                           changed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           channel                      characterize </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           chicago                          chickens </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          chloride                       chromosomal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              chrq                               cik </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           cisting                         clarified </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          cleavage                        clinicians </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 clinicopathologic                               clm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             coimmunoprecipitation                           college </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            colony                        colorectal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          combined                         combining </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           compare                     complications </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         component                       composition </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          compound                        concurrent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      conditioning                         conducted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        conferring                      confirmation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        connection                      considerably </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        consisting                        consortium </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    constitutional                         construct </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           context                          contrary </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      contribution                        conversely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       cooperation                       cooperative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       correlation                      counterparts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            counts                          covering </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               csf                          curation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            curves                  cyclophosphamide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ddxa                         ddxddxdhx </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ddxl                              ddxx </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ddxy                            decade </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             decay                         decisions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          decrease                         dedicated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              deep                            defect </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           delayed                         delineate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        delineated                             delta </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     demonstrating                           depends </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       deregulated                         detecting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     determination                       detrimental </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               dev                              dgvs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           diamond                   diamondblackfan </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    differentially                           diffuse </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          digmpysk                       dimensional </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        disability                          discover </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           display                         displayed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       distinctive                        distinctly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              dnah                            dnarna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      dnatriggered                        dominantly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      donorderived                            double </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     downregulated                               dpv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            driven                            drives </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           driving                           dynamic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          dynamics                       effectively </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          efficacy                             eifα </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           elegans                           embryos </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            energy                       engraftment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           enhance                          entities </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               env                      epidemiology </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        epigenetic                              ercc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    erythropoiesis                          estimate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          evaluate                        evaluating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             event                         evidences </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             evoke                    evolutionarily </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      evolutionary                           evolved </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         exclusion                        exhibiting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          exhibits                           expands </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           explore                        facilitate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            fasted                          feasible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           feature                           finding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               fly                            fpdaml </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         fragments                       frequencies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               fxs                               gag </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            gained                               gap </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         gasdermin                             geese </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          genomics                               gep </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   germlinesomatic                            glands </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      glycoprotein                         grafthost </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              gvhd                             gypsy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      haematologic                    haematopoietic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      haemopoiesis                              heat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hekt                         hemocytes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     herpesviruses                   heterogeneously </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hhmc                       highlighted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               hiv                            hnrnpa </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       homeostasis                          homology </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             hspcs                        hsvencoded </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          hundreds                      hypocellular </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       hypothesize                      hypothesized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ibmfs                               icd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        idiopathic                             ifnar </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ifnb                      ifndependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      ifninducible                             ifnγ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ikzf                    immunerelevant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  immunodeficiency              immunohistochemistry </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 immunotherapeutic                           impacts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          impaired                         inclusion </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        incomplete                         indicator </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     indispensable                    individualized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          inducing                          industry </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       ineffective                        infectious </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       infiltrates                        influences </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       informative                         initiator </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         insertion                       institution </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     institutional                      intellectual </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         intensive                            intent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        interacted                      interactions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       interactome              interferencemediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               interferoninducible                      internalized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     international                      intervention </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      intriguingly                          invading </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     investigating                    investigations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         junctions                       kaplanmeier </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        karyotypic                            larger </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             lasso                           latency </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        lentiviral                        leukaemias </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    leukemogenesis                               lgl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          lifelong                            ligand </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            listed                          listeria </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          location                      lossfunction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lpd                              lsma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        lymphocyte                             lytic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       maintenance                          markedly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           massive                   mechanistically </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           meiotic                            memory </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        messengers                      metaanalysis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          metabric                            method </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           methods                       methylation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         microbial                 microenvironments </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            mimics                              mita </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  mitogenactivated                        modulation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          monocyte                           monodon </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       morphologic                             motif </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             mount                               mpl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mpn                            muscle </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           mutants                               mxa </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mxi                    myelomonocytic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                myeloproliferative                           natural </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            nature                       necroptosis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        negatively                           neurons </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          ngsbased                              nlrc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           notably                        nucleotide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         nutrition                         occurring </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         oncogenes                          oncomine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           ongoing                        organelles </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     overexpressed                       overfeeding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          overview                        parameters </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              parn                        pathologic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         pedigrees                           penaeus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           percent                         phylogeny </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        physiology                      plasmacytoid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           plasmid                          plasmids </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             pmpen                               pol </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      polymorphism                        positively </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         possessed                     postinfection </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               posttranscriptional                 posttranslational </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           predict                       predisposed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           premrna                          presumed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         prevalent                           prevent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          prevents                           priming </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             prkdc                          produced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          produces                        proficient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       progenitors                     proliferative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         prominent                           promote </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         promoters                            proper </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          proposed                       prospective </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     prospectively                        proteasome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        protecting                        protection </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     psinactivated                       psychomotor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            pursue                             pyhin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      radiographic                             raise </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           ranging                             rapid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               rat                          realtime </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           receive                          received </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     recombination                         recurring </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          regarded                          registry </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      relationship                     relationships </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          relative                          reliable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         remaining                          reporter </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         represent                      representing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       requirement                            rescue </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           respond                        responsive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          retinoic                         revealing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               rfs                             rgnnv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       rhabdovirus                 ribonucleoprotein </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         ribosomal                        rnabinding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     rnasequencing                              rocs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             route                         routinely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             rtpcr                            rtqpcr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               sac                          salivary </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             sasps                              sbds </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           science                            scores </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          searched                         seemingly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          selected                       selectively </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             sense                        separately </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      sequentially                            series </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             serve                            serves </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               sfs                              sgiv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             shock                           shorter </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          siblings                        signatures </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         silencing                           simplex </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    singlestranded                             sirna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             sites                              skin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               sod                              sole </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     sophisticated                            source </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         southeast                          spanning </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           special                           spectra </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   spermatogenesis                              spot </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          squamous                             ssdna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          standard                             steps </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          stepwise                         stimulate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         stingtype                           stomach </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   strandannealing                        strand dna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   subdistribution                          subgroup </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         submitted                        substitute </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          suitable                        summarized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       superfamily                         supported </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        suppresses                       suppressing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       surrounding                               suz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          symptoms                       synergistic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           systems                              tank </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       tankbinding                              tbds </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          template                          tendency </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          terminal                             terms </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              thai                  thrombocytopenia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        throughput                           tightly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             times                              tmem </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              tool                        transcript </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     translocation                       transmitted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             trial                              trif </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ttk                        ultimately </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            unable                          uncommon </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         undefined                        underscore </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        unexpected                      unexpectedly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        unexplored                       universally </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        university                         untreated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           unusual                             usage </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          utilized                          utilizes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         validated                          variable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        variations                            varied </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              vast                      verification </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            verify                               vhl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             vital                         waterfowl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             white                              wide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             women                          workshop </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            workup                              wssv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              zrsr                               μm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 2                                 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       abnormality                        abnormally </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          abundant                          accepted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        accounting                          accounts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        accuracies                        accurately </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     aciddependent                     acidinducible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       activations                           acutely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              adar                               add </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           address                         addressed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              adds                        adenoviral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      adenoviruses                          adherent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          adjacent                        adjustment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         adjuvants                       adolescents </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               adp                         adulthood </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          advanced                          advocate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         aeromonas                          affinity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          afforded                               afs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                agammaglobulinemia                      agedependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        agematched                             agent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            agents                              ages </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       aggregation                        aggressive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               aid                            aimasc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           aldrich                             alfpm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    algorithmbased                        alignments </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               alk                         allogenic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               alr                         alssodmut </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            alters                amlmyelodysplastic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              amls                     amplification </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         amplifier                        amplifying </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       amyotrophic                         analogous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         analysing                          analytic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           analyze                         analyzing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         aneuploid                            animal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         annealing                         anomalies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            answer                      antagonistic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    antiadenoviral                        anticancer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      anticipation                         antidsdna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       antimorphic                            antipd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    antiretroviral                    antisenescence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         antitumor                   antiviralinnate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                antiviralsignaling                      antivirotics </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            aomdss                              apcs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              apis                               apo </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        appearance                          appeared </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           applies                       approaching </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     approximation                       aquaculture </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           arguing                               arm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              arms                          articles </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              asia                            aspect </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           assayed                         assembled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        associates                 associationfrench </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            ataxia                         atgbgskip </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        atpbinding                    atpindependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      atrtriggered                         attention </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        attractive                        attributed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          atypical                           augment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            august                        australian </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           authors                            autism </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  autoinflammatory                         automated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     automatically                         autophagy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    auxininducible                          awaiting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              axis                       azacitidine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          bacillus                        bartonella </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             basal                          baseline </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              beak                              bear </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           bearing                              beat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               bee                               bel </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          believed                              bias </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             binds                       biochemical </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     bioinformatic                     biotinlabeled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              bird                             black </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           blastic                               blm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          blocking                              blot </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               bmx                            bodies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              body                 bonemarrowderived </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             bovis                           boxtype </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             bpdcn                             break </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          breakage                            breaks </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           broaden               bromouridinelabeled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  bronchoalvelolar                      btkdeficient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            buccal                            bypass </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         bystander                    caenorhabditis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        calcarifer                  calcineurinbased </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       calibration                             calls </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   calmetteguérin                    cancerspecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           cancer‑                        candidates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         canonical                        capacities </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            capsid                           capture </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            caries                           carrier </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        categories                       categorized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         causality                           caveats </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               cbl                             ccaat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ccl                             cdkna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               cdr                              cdss </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            cebpα                        celllethal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            center                           centers </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        cerebellum                          cerebrum </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          cervical             cgasddxstingsignaling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      cgasmediated                    cgasstimulator </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             chain                        challenges </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         character                  characterisation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 chddxchstingifnβ                          chemical </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         chemokine                          chimeric </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           chinese                            choice </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          choosing                            chosen </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               chr                         chromatin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           chsting                             ciirf </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      circumstance                           classes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         classical                         clearance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            clinicallyrecognizable                clinicalpathologic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         clonality                            closer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              cmml                             cmmrd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               cmn                               cmv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               cns                             codon </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         cofactors                           cohesin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        coiledcoil                         coincides </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          coioides                              coli </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           colitis                       collaborate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     collaborative                        colleagues </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        collection                        collisions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             colon                 colonystimulating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         combating                           combine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        commentary                        comparable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       compartment                        compensate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        complement                        completely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       complicated                         compounds </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   comprehensively                          comprise </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        comprising                        compromise </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         comutated                      concentrated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         concerned                       conclusions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       concordance                       concurrence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               conditioningregimen                           conduct </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         conferred                          confirms </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         connected                         consensus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       consequence                     consideration </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    considerations                         consisted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      consistently                       consolidate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        constantly                        constructs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         contained                         contested </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        continents                     contradictory </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       contributed                        controlled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     controversial                       controversy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      conventional                           convert </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            convey                       cooperating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       coordinated                       coordinates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             copia                              cord </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         correlate                        correlates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        counteract                  counterscreening </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         countries                        covariates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             cover                          coverage </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           covered                          cpgmotif </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              cpgs                               crd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           creates                        critically </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           cryptic                               ctc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  ctenopharyngodon                          cultures </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        cumulative                           curated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          curative                              cxcl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            cycles                       cyclicdigmp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   cyclindependent                          cytidine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       cytopaenias                              dadt </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              dais                            danish </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              daxx                               day </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               dba                               dca </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       ddxabstrakt                          ddxaeifa </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ddxb                          ddxbeifa </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           ddxcage                            ddxdbp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      ddxdeficient                      ddxdependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     ddxexpressing                    ddxinteracting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       ddxknockout                         ddxmutant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         ddxmutsom                   ddxmycregulated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 ddxoverexpressing                            ddxrok </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ddxs                      ddxselective </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       ddxspecific                          ddxsting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 ddxstingdependent                              ddx  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     ddx‑regulated                       deaddomains </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          dead‑box                              deah </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         deaminase                           deciles </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         deepening                            deeper </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           defence                          defences </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        definition                       degradative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          degraded                            delays </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       delineation                          delivery </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     demethylation                        department </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            depend                             depth </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      deregulation                        describing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           deserve          desorptionionizationtime </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          detailed                           details </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        detectable                       determining </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       devastating                     developmental </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            devise                          df cells </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              dgfs                     diabziinduced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       diagnostics              diamidobenzimidazole </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               die                         difficult </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           dimeric                      dimerization </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             dinucleotidesynthesis                       diphosphate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         direction                               dis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   disadvantageous                           discern </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       discoveries                       discovering </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    discrimination                  dishematopoiesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  disordersfanconi                       dispensable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         displaced                         disrupted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           distant                       distinguish </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    distinguishing                      distribution </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        disturbing                             ditof </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   diversification                         diversity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           divided                       dnamediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             dnase                     dnastimulated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      dnaunwinding                          domestic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  dominantnegative                         dominated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              dose                          doubling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          dramatic                      dramatically </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     drddxmediated                         drosphila </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    drstatmediated                      drstingdanio </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  drstingdependent                     drstingdrstat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              drug                   dsdnacdninduced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    dsdnadependent                            dsdnas </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                dsdnavirusmediated                     dsrnamediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      dsrnasensing                      dualfunction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    dualluciferase                          duckling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ducks                            ductal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    duddxexpressed                       duplication </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      duplications                          duration </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          dwarfism                             dynll </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       dysfunction                        dysmorphic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      dysregulated                     dysregulation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          economic                           editing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      edwardsiella                          effector </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         effectors                        efficiency </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              efα                              egfr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              eifa                              eifb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           elected                   electrophoresis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   electroporation                           element </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            elicit                          eligible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         elucidate                        elucidated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       elucidation                            embryo </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              emcv                          emphasis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         emphasize                        emphasizes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       emphasizing                            employ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          employed                           employs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           enabled                           enables </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              encephalomyocarditis                       encountered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      endocervical                       endoplasmic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          endosome          endothelialhematopoietic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       endothelium                      endotracheal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            engage                          enhancer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          enlarges                            enrich </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          enriched                        enrichment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            ensure                         enteritis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            entire                          entitled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         envcoding                            enzyme </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          epidemic                   epidemiological </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       epinephelus                        epithelium </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       equilibrium                               era </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         eradicate                             erccl </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    errorcorrected                         ervdanrer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     erythematosus                    erythroblastic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            escape                       escherichia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       essentially                               est </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        estimation                         etiologic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          etiology                              etnk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ets                        eukaryotic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          european                       evaluations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           evasion                            evolve </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       exacerbates                     exacerbations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             exact                              exam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          exciting                         exclusive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       exclusively                         exemplary </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             exert                           existed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          existing                         exogenous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          expanded                        experience </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    experimentally                            expert </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        explaining                       explanation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      explanations                         exploited </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               expressionmutations                           extends </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         extensive                       extensively </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            extent                          external </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        externally                          extracts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             facet                         factorκb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            failed                             fails </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          failures                          familiar </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       familiarity                        familywide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             fanci                             fancm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              fast                             fatal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            father                             fatty </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     favorablerisk                         favouring </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       feasibility                          featured </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               fed                          feedback </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         fertility                               ffs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               fga                              fgfr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             finch                      fishspecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           fitness                              flag </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           flagged                         flagsting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       flexibility                            flight </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        flightless                            florea </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         flounders                      fluorescence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               fmr                       fmrpbinding </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              fold                        follicular </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        forebrains                           foreign </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             forks                            formed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           founder                          fpdacute </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     fractionation                             frame </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            french                          frontier </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             fruit                               ftr </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       functioning                       fundamental </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      fusogenicity                              gain </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           gaining                             gamma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    gammainducible                        gatekeeper </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              gcrv                     genedeficient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  generalizability                        generating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       genetically               geneticsassociation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    genomicsdriven                         genotoxic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               geo                          germinal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     germlinebased                   germlineencoded </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             gills                              gopc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            grades                      grampositive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          granular                     granulomatous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            ground                               gse </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              gsea                              gskb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    guanineadenine                         guanosine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             guide                            guides </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  haematopathology                        hallmarked </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    haploidentical                           harbour </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           harmful                           harveyi </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        hccbearing                              head </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             heart                           heliase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             helps           hematologistoncologists </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          hematologistsoncologists                        hematology </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           hematopathologyeuropean              hemizygoushomozygous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         hemogenic                          henselae </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hepg                            hervnc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      heterozygote                               hhm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      hhmcreferred                        highjacked </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            hijack                             hinae </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        histologic                           histone </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         histories                               hit </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hla‑                           hla‑dmb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       homeostatic                              homo </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        homozygous                       hostdefense </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             hosts                               hot </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hoxb                              hpcs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              hspa                             hspaa </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             hspab                           hungary </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     hybridization                        hydrolysis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        hydrolyzed                       hydrophilia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             hyper                          hyperifn </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   hyperresponsive                   hypomethylating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ideal                           idellus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        identifies                          identity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            idhidh                              ifih </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ifil                          ifisting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    ifnardependent                              ifng </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        ifninduced                       ifninducing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      ifnregulated                      ifnsensitive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        ifnαifnβ                    ifnαproducing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     ifnβmediated                               ifp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               iga                               igg </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              iglc                               igm </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ihnv                               ilb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ilf                        illuminate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       illustrated                         imbalance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             imids                          immortal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    immunemediated                 immunestimulatory </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   immune‑response                         immunized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    immunoblotting               immunocytochemistry </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       immunogenic                    immunogenicity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               immunohistochemical                     immunological </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   immunophenotype                          impacted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        impairment                           impairs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            impede                         implicate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          improves                      inactivation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          inactive                     inappropriate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     incorporating                     incorporation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         incubated                      independence </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       indications                        indicative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        indicators                        indirectly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 individualpatient                           inducer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          inducers                         infantile </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         inference                        infiltrate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     infiltrations           inflammasomeindependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     inflammasomes                  inflammationards </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       inflammsome                        influenced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       influencing                            inform </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           informs                       inheritance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           inhibit                        inhibiting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        inhibitory                         initially </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         injection                          innatedb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        innovative                          instance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         instructs                      insufficient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            intact                          integral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        integrates                       integration </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          intended                         intensity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      interception                   interferongamma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                interferonopathies             interferonstimulating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      interferonβ             interferonγinducible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       interleukin                      intermediate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               intermediateadverse                     interneuronal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         interpret                    interpretation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       interpreted                      interpreting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     interquartile                      interstitial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     interventions                        intestines </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     intrafamiliar                   intramuscularly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 intraperitoneally                     intrinsically </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        introduced                        introduces </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      introduction                               inv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          invaders                          invasion </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         invasions                         inversely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     invertebrates                           involve </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               iqr                              irak </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     irfresponsive                              irfs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               irg                        iridovirus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              iron                       irradiation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                irradiationinduced                      irrespective </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          isoforms                          isolated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   isotypeswitched                             isres </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            issues                              iκb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      jakdependent                           january </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             janus                          juvenile </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       karyotyping                           kdmalsd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            killer                           kindred </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              kmta                           knocked </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      laboratories                        laboratory </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             lacks                           largest </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             laser                            lastly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         lateonset                           lateral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             lates                               lay </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             layer                            layers </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              lays                              lcdv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lck                        legionella </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           lengths                       leucinerich </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   leukaemogenesis                       leukemianet </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 leukemogenesisrna                      leukemogenic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        leukocytes                              lies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          lifetime                    ligandspecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             light                            limits </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     linckitscaddx                           lineage </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       linkedomics                         liposomal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          liposome                       literatures </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              load                             local </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          localize                              loci </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             locus                               log </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      loopmediated                            losses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lot                            lrrfip </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        luciferase                             lupus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       lymphocytic               lymphoproliferative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               lyn                          lysosome </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mal                             males </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            mammal                            manage </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           managed                          manifest </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    manifestations                      manipulating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            manual                               map </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            marked                     marrowderived </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           marrows                              mask </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    matrixassisted                     mavsdependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              mayo                           mddxaml </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       mdmmediated                          mdsacute </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     mdsamlderived                             mdseb </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          mediates                         mediation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          mediator                         mediators </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        medication                          medicine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         meditates                              meet </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           meeting                     megakaryocyte </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      melanogaster                         mentioned </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mer                         metabolic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        metastases                        metastasis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  metazoanspecific                     methodologies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       methodology                               mfn </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               mhc                      microbiology </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  microenvironment                              mild </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           minimal                          minimize </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           minimum                             minor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            mirnas                     misexpression </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         misfolded                          mismatch </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       missplicing                      mitochondria </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           mitosis                             mixed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              mlkl                              mnda </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        modalities                          modality </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   modeldetermined                           modeled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          modeling                  modeling studies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    moderatesevere                      modification </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          modified                         modifying </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        modulating                           monitor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            monocytogenesgenerated             monocytogenesmediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         monogenic                     monophosphate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        monosomies                          monosomy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             month                         morbidity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   morphologically                         mortality </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            motifs                          mounting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   mptpindependent                              mrps </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           mucosal                       multicenter </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        multilayer                      multitasking </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 mutatedrearranged                          mutually </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     mycobacterium                      myddependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        myelocytic                    myelodsyplasia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> myelodysplasticmyeloproliferative                     myelofibrosis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       myelogenous                            myxoma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             named                        nationally </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         naturally                            nearby </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      necessitates                       neighboring </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                neoadjuvanttreated                           nervous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           netosis                 neuroinflammation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    neuroregulator                             nexin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              nfkb                        ngstesting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          nijmegen                              nile </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         niloticus                               nol </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        noncomplex                    nonconsecutive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      nonmonocytes                     nonresponsive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          nonsense                nonsenseframeshift </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       nonspecific                     nonstructural </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            noonan                          northern </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              note                               nrh </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ntp                              ntrk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  nuclearcytoplasm                 nucleocytoplasmic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         nucleolar                        nucleoside </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   nucleotidebased                               nup </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               oas                           obscure </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      observations                          obstruct </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            occupy                             occur </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        occurrence                            occurs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            ocular                      odontoblasts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             offer                           offered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         olivaceus                           omnibus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          oncogene                       oncogenesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       oncologists                         oogenesis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          operator                       opportunity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          opposing                           optimal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         optimally                         optimized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              oral                            orally </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            orange                       oreochromis </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               orf                          organism </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       organtissue                           origins </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       orthologous                          outlines </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    overexpressing                           overfed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           overlap                        overlapped </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        paediatric                            paired </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              palb                            pamcsk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              paml                             paper </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  papillomaviruses                       paralicthys </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         partially                      participants </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      participates                          partners </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        parvovirus                              pass </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      pathogeneses                      pathogenhost </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     pathogenicity              pathogenicpathogenic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     pathognomonic                       pathologies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      pathologists              pathologyrecommended </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   pathophysiology                   pathwayspecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        patients ≤                           pausing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         penetrant                            people </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        percentile                       performance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        performing                          performs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            period                     peroxiredoxin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       persistence                        persistent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      personalized                      perspectives </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           perturb                         perturbed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          perturbs                    pharmaceutical </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             phase                         phosphate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 phosphatebuffered           phosphogluconolactonase </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             phosphoimmunoblotting                  phosphoinositide </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    phosphorylates                        phosphotbk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        phosphotyr                    phosporylation </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              phox                      phylogenetic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              pikk                           pivotal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               pkcεoverexpressing                               pkz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            plasma                        plasmacell </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                plasmidtransfected                            player </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           playing                              plcg </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           plotter                          pmcactus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          pmdorsal                            pmikkɛ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           pmikkβ                    pmonodonbiotec </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          pmrelish                       pneumophila </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            poldip                        polprasert </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             polrl                   polydadtinduced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 polydadttriggered                          polydgdc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      polyomavirus                    polyomaviruses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    polysaccharide                            pooled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              pore                             poses </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          position                     posttreatment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   postvaccination                        potentials </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           poultry                              ppih </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              ppil                         practiced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               pre                      precancerous </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          preceded                      precipitates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         precisely                         precision </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     preconceptual                        predicting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       predominant                    preferentially </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      premessenger                     preponderance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      prerequisite                      preribosomal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      preservation                         preserved </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        presumable                       presumption </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        pretreated                      pretreatment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       prevalently                         prevented </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        prevention                        preventive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          pre‑mrna                         primitive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    prioritization                        prioritize </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     probabilities                          probands </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            proces                           produce </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        productive                          products </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    prognostically                        progressed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            proleu                         prolonged </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         promising                          promotes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           prompts                        pronounced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         propagate                        propensity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       prophylaxis                          prostate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           protect                    proteinprotein </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        proteinshb                        proteomics </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            proven                          proviral </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       provisional                          proximal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         proximity                              prpf </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           pserirf                      pseudorabies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            pubmed                              pull </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          pulpitis                      purification </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          purified                          purposes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       qualitative                        quantified </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          question                         questions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              race                               rad </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            radial                           rampant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           rapidly                            rarely </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rars                       rasopathies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         rationale                               ray </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               rbp                          reaction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           reading                        rearranged </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     rearrangement                           reasons </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              reca                          receiver </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         receiving                         recommend </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   recommendations                       recruitment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               red                        redspotted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            reduce                         reduction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         reference                         referrals </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        reflecting                            regard </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           regimen                          regimens </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        registered                   relapsespecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         relapsing                           relates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          relation                         relatives </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            relics                            relies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rely                        remarkable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        remarkably                        remissions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        remodelers                        remodeling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           removal                             renal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          rendered                            repeat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           repeats                        repertoire </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           replace                        replicated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       replicative                        reportedly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    representative                       represented </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      reproducibly                     reprogramming </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           require                         resection </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        resistance                        resolution </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       resolutions                          resource </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           respect                        respective </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       respiratory                         responded </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           resting                          restrict </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          retained                           retains </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    reticulocalbin                         reticulum </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   retrovirusesddx                          revision </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       rhinocerous                              rhkt </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          ribosome                     ribosomopathy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rich                       ripkasccasp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rise                               rit </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rlhs                          rlhsmavs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               rlr                              rlrs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        rnaprotein                        rnasensing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    rna‑sequencing                               rnf </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               roc                           roughly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              rrna                              rtel </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           runxfpd                              safe </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       salidroside                           salient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    salineinjected                         salmonids </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             samdl                           sapiens </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       sarcoidosis                              sars </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           satisfy                               sbs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          scanning                 schwachmandiamond </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         sclerosis                           scoring </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            screen             screeningsurveillance </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           seabass                         searching </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          secreted                         secretion </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         secretory                           seeking </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             seeks                          segments </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         senescent                              senp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       sensitively                       sensitivity </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sensorstinginterferon                         separated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         sequenced                         sequencer </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         sequences                   sequencingbased </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        sequential                            serial </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       serological                             serum </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      seruminduced                           session </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           seventh                               sex </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             sfsrh                            shaped </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            shared                           sharing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               shb                      shortcomings </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           shuttle                         shuttling </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              sifs                            silica </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            silico                            simple </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      simultaneous                         singapore </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  singlenucleotide                  sinocyclocheilus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              situ                           sixteen </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              size                            sledai </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             slide                            slight </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              slow                            slowed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               slu                               smc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             smoke                              snps </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    snxinteracting                           society </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         soddriven                     sodmutcausing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             solid                              soma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       somatically                             space </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             spans                         specimens </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           spindle                   splicingcoupled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 splicingdependent               splicingindependent </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             spots                           spotted </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           sprypry                               src </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             srebf                            ssdnas </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             ssrna                              stag </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          staining                      standardized </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             start                         startloss </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       statistical                          statstat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            steady                          stemcell </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           sterile                          stigmata </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                stimulationinduced                          stingirf </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     stingmediated                          stingtbk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       stingtbkirf                            strain </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            strand                          stranded </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    stratification                            stream </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       strengthens                          strictly </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   strongpotential                   structureguided </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          subacute                          subclone </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        subdivided                         subjected </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        submitting                           subsets </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        succession                        sufficient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        suggestive                         summarize </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            summed                             super </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      supernatants                        supporting </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          supports                          supposed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       suppression                       suppressive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           surface                        surgically </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            survey                           survive </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          survived                       susceptible </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           suspect                        suspicious </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        sustaining                             swabs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             swine                     syndromeacute </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    syndromesacute                         syndromic </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   synergistically                synthasestimulator </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         synthesis                               tag </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            tandem                             tapes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             tarda                        targetable </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    targetspecific                          taxonomy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      tbkdependent                        tbkinduced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       tbkirfifnβ                       tbkspecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              tcga                              tcid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               tdp                         technique </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        technology                    telangiectasia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           teleost                          teleosts </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         telomeres                         telomeric </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            tended                          terminus </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          tertiary                          tertterc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   therapeutically                      therapeutics </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         thousands                               thp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            threat                         threonine </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 thrombocytopenias                              tier </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              tifs                             tiger </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             tight                           tilapia </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     timedependent                            titers </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               tkd                       tlrmediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              tlrs                            tnfaip </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              tnfr                           tnfrsfa </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       tofacitinib                        topography </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           totally                       traditional </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     traditionally                          training </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            traits         transcriptionquantitative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          transcriptionreplication                    transcriptomes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      transduction                         transfect </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          transfer                      transforming </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        transgolgi                         transient </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        transition                      transitional </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       translocase                      translocates </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     transmembrane                          transmit </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      transplanted                         transport </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       transported                     transversions </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          treating                              tree </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             trend                              trex </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            trials                        trilineage </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      trimmediated                           trisomy </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               trk                           tropism </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              true                         truncated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ttc                        tumorbased </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           tumorigenesisinitiation                 tumorinfiltrating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    tumorpromoting                     tumorspecific </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  tumorsuppressive                           tumours </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          turnover                             twist </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            ualcan                         ubiquitin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      ubiquitously                        ulcerative </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          ultimate                          unbiased </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      uncontrolled                           uncover </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         uncovered                        uncovering </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    underestimated                        undergoing </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          underlie                    underperformed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     underpinnings                      underpowered </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       underscores                      understudied </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         underwent                      unidentified </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             units                         unraveled </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      unrecognized                        unresolved </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          unstable                      unstimulated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          untested                            unwind </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            update                      upregulating </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            urgent                               usp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           utility                           utilize </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        uvpsoralen                       vaccination </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   vaccineadjuvant                   vaccinemediated </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              vacv                             valid </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          validate                            values </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       variability                         variables </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              vasa                              vdna </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        verifiable                         vesicular </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         viability                            vibrio </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           viperin                         viroporin </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         virtually                      virusinduced </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     virusmediated                            visual </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               wac                           warrant </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           weinreb                           western </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       westernblot                        widespread </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           wiskott                        withdrawal </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             worth                      wssvinfected </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               xpd                             yeast </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            yellow                               yhv </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           yielded                               ysk </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                yskpopecholesterol                       zbpripkcasp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             zebra                               zfp </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               aml                             vitro </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                              vivo                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               ×                                μg </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               μl                              αnd </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            βcxcl                             amino </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 1                                 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $TDM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;&lt;TermDocumentMatrix (terms: 3482, documents: 168)&gt;&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-/sparse entries: 11636/573340</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sparsity           : 98%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Maximal term length: 33</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Weighting          : term frequency (tf)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $df</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          word freq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ddx                       ddx  487</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> patients             patients  227</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mutations           mutations  223</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dna                       DNA  193</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> germline             germline  185</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cells                   cells  146</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> genes                   genes  144</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> myeloid               myeloid  142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> expression         expression  129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> response             response  112</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> variants             variants  103</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mds                       mds  101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rna                       RNA  100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> immune                 immune   97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> identified         identified   86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cell                     cell   81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> leukemia             leukemia   80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sting                   sting   80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> protein               protein   75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> clinical             clinical   74</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> aml                       aml   73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> including           including   69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> predisposition predisposition   69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> innate                 innate   66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> type                     type   66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> genetic               genetic   65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ifn                       ifn   65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> neoplasms           neoplasms   64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> acute                   acute   61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> malignancies     malignancies   61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $wc</span>
<span class="r-plt img"><img src="wcAbst-1.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

