<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>cpbnplot</title>
  <meta name="description" content="Plotting Bayesian network based on enrichment analysis including clusterProfiler and ReactomePA" />
  <meta name="generator" content="bookdown 0.23 and GitBook 2.6.7" />

  <meta property="og:title" content="cpbnplot" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://noriakis.github.io/software/cpbnplot/" />
  
  <meta property="og:description" content="Plotting Bayesian network based on enrichment analysis including clusterProfiler and ReactomePA" />
  <meta name="github-repo" content="noriakis/cpbnplot" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="cpbnplot" />
  
  <meta name="twitter:description" content="Plotting Bayesian network based on enrichment analysis including clusterProfiler and ReactomePA" />
  

<meta name="author" content="Noriaki Sato" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="bootstrap-based-inference.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">cpbnplot</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> cpbnplot: Bayesian network plot for enrichment analysis results</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-preprocessing-and-deg-identification-of-gse133624"><i class="fa fa-check"></i><b>1.1</b> The preprocessing and DEG identification of GSE133624</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="bootstrap-based-inference.html"><a href="bootstrap-based-inference.html"><i class="fa fa-check"></i><b>2</b> bootstrap-based inference</a>
<ul>
<li class="chapter" data-level="2.1" data-path="bootstrap-based-inference.html"><a href="bootstrap-based-inference.html#bootstrap-based-inference-1"><i class="fa fa-check"></i><b>2.1</b> bootstrap-based inference</a></li>
<li class="chapter" data-level="2.2" data-path="bootstrap-based-inference.html"><a href="bootstrap-based-inference.html#multiscale-boostrap-based-inference"><i class="fa fa-check"></i><b>2.2</b> Multiscale boostrap-based inference</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="bngeneplot.html"><a href="bngeneplot.html"><i class="fa fa-check"></i><b>3</b> <code>bngeneplot</code></a>
<ul>
<li class="chapter" data-level="3.1" data-path="bngeneplot.html"><a href="bngeneplot.html#barplot"><i class="fa fa-check"></i><b>3.1</b> <code>barplot</code></a></li>
<li class="chapter" data-level="3.2" data-path="bngeneplot.html"><a href="bngeneplot.html#the-vanilla-plot"><i class="fa fa-check"></i><b>3.2</b> The vanilla plot</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="bngeneplot.html"><a href="bngeneplot.html#show-the-confidence-of-direction"><i class="fa fa-check"></i><b>3.2.1</b> Show the confidence of direction</a></li>
<li class="chapter" data-level="3.2.2" data-path="bngeneplot.html"><a href="bngeneplot.html#show-the-hub-genes"><i class="fa fa-check"></i><b>3.2.2</b> Show the hub genes</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="bngeneplot.html"><a href="bngeneplot.html#the-plot-with-the-reference"><i class="fa fa-check"></i><b>3.3</b> The plot with the reference</a></li>
<li class="chapter" data-level="3.4" data-path="bngeneplot.html"><a href="bngeneplot.html#the-plot-with-the-depmap-information"><i class="fa fa-check"></i><b>3.4</b> The plot with the DepMap information</a></li>
<li class="chapter" data-level="3.5" data-path="bngeneplot.html"><a href="bngeneplot.html#plotting-edge-strengths"><i class="fa fa-check"></i><b>3.5</b> Plotting edge strengths</a></li>
<li class="chapter" data-level="3.6" data-path="bngeneplot.html"><a href="bngeneplot.html#plotting-genes-in-multiple-pathways"><i class="fa fa-check"></i><b>3.6</b> Plotting genes in multiple pathways</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bnpathplot.html"><a href="bnpathplot.html"><i class="fa fa-check"></i><b>4</b> <code>bnpathplot</code></a>
<ul>
<li class="chapter" data-level="4.1" data-path="bnpathplot.html"><a href="bnpathplot.html#emapplot"><i class="fa fa-check"></i><b>4.1</b> <code>emapplot</code></a></li>
<li class="chapter" data-level="4.2" data-path="bnpathplot.html"><a href="bnpathplot.html#the-vanilla-plot-1"><i class="fa fa-check"></i><b>4.2</b> The vanilla plot</a></li>
<li class="chapter" data-level="4.3" data-path="bnpathplot.html"><a href="bnpathplot.html#compare-with-the-reference"><i class="fa fa-check"></i><b>4.3</b> Compare with the reference</a></li>
<li class="chapter" data-level="4.4" data-path="bnpathplot.html"><a href="bnpathplot.html#reflect-depmap-information-to-pathways"><i class="fa fa-check"></i><b>4.4</b> Reflect DepMap information to pathways</a></li>
<li class="chapter" data-level="4.5" data-path="bnpathplot.html"><a href="bnpathplot.html#aggregating-the-pathway-databases"><i class="fa fa-check"></i><b>4.5</b> Aggregating the pathway databases</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html"><i class="fa fa-check"></i><b>5</b> Including clinical variables</a>
<ul>
<li class="chapter" data-level="5.1" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#preparation-of-data"><i class="fa fa-check"></i><b>5.1</b> Preparation of data</a></li>
<li class="chapter" data-level="5.2" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#inference-of-pathway-relationship-including-clinical-variables"><i class="fa fa-check"></i><b>5.2</b> Inference of pathway relationship including clinical variables</a></li>
<li class="chapter" data-level="5.3" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#conditional-probability-query"><i class="fa fa-check"></i><b>5.3</b> Conditional probability query</a></li>
<li class="chapter" data-level="5.4" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#gene-relationship-with-variables"><i class="fa fa-check"></i><b>5.4</b> Gene relationship with variables</a></li>
<li class="chapter" data-level="5.5" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#confirming-the-existing-knowledge"><i class="fa fa-check"></i><b>5.5</b> Confirming the existing knowledge</a></li>
<li class="chapter" data-level="5.6" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#investigating-the-network-based-on-the-clinical-question"><i class="fa fa-check"></i><b>5.6</b> Investigating the network based on the clinical question</a></li>
<li class="chapter" data-level="5.7" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#classification-using-bn"><i class="fa fa-check"></i><b>5.7</b> Classification using BN</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="the-other-options.html"><a href="the-other-options.html"><i class="fa fa-check"></i><b>6</b> The other options</a>
<ul>
<li class="chapter" data-level="6.1" data-path="the-other-options.html"><a href="the-other-options.html#bootstrap-reasoning"><i class="fa fa-check"></i><b>6.1</b> Bootstrap reasoning</a></li>
<li class="chapter" data-level="6.2" data-path="the-other-options.html"><a href="the-other-options.html#discretization"><i class="fa fa-check"></i><b>6.2</b> Discretization</a></li>
<li class="chapter" data-level="6.3" data-path="the-other-options.html"><a href="the-other-options.html#custom-visualization"><i class="fa fa-check"></i><b>6.3</b> Custom visualization</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="the-other-options.html"><a href="the-other-options.html#the-glowing-nodes-and-edges"><i class="fa fa-check"></i><b>6.3.1</b> The glowing nodes and edges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="testing-the-score-inside-the-pathway.html"><a href="testing-the-score-inside-the-pathway.html"><i class="fa fa-check"></i><b>7</b> Testing the score inside the pathway</a></li>
<li class="chapter" data-level="8" data-path="testing-r.html"><a href="testing-r.html"><i class="fa fa-check"></i><b>8</b> Testing R</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">cpbnplot</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">cpbnplot</h1>
<p class="author"><em>Noriaki Sato</em></p>
</div>
<div id="cpbnplot-bayesian-network-plot-for-enrichment-analysis-results" class="section level1" number="1">
<h1><span class="header-section-number">Chapter 1</span> cpbnplot: Bayesian network plot for enrichment analysis results</h1>
<p>The R package to plot Bayesian network inferred from expression data based on the enrichment analysis results including <a href="https://github.com/YuLab-SMU/clusterProfiler">clusterProfiler</a> or <a href="https://github.com/YuLab-SMU/ReactomePA">ReactomePA</a> results <span class="citation">(<a href="#ref-Wu2021-sk" role="doc-biblioref">Wu et al. 2021</a>; <a href="#ref-Yu2016-kh" role="doc-biblioref">Yu and He 2016</a>)</span>. It makes use of libraries including <a href="https://github.com/YuLab-SMU/clusterProfiler">clusterProfiler</a>, <a href="https://github.com/YuLab-SMU/ReactomePA">ReactomePA</a>, <a href="https://www.bnlearn.com/">bnlearn</a>, <a href="https://bioconductor.org/packages/release/bioc/html/graphite.html">graphite</a> and <a href="https://bioconductor.org/packages/release/data/experiment/html/depmap.html">depmap</a> <span class="citation">(<a href="#ref-Killian2021-ko" role="doc-biblioref">Killian and Gatto 2021</a>; <a href="#ref-Scutari2010-tj" role="doc-biblioref">Scutari 2010</a>; <a href="#ref-Sales2012-cu" role="doc-biblioref">Sales et al. 2012</a>)</span>. The description of functions and several use cases are depicted in this book using <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133624">GSE133624</a>, which contains RNA-Seq data of bladder cancer and adjacent normal bladder tissues <span class="citation">(<a href="#ref-Chen2019-ve" role="doc-biblioref">Chen et al. 2019</a>)</span>.</p>
<div id="the-preprocessing-and-deg-identification-of-gse133624" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> The preprocessing and DEG identification of GSE133624</h2>
<p>In this book, the data from GSE133624 was used for the demonstrative purpose. DESeq2 is used to identify DEGs <span class="citation">(<a href="#ref-Love2014-yt" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-2"><a href="index.html#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="index.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Load dataset and make metadata</span></span>
<span id="cb1-4"><a href="index.html#cb1-4" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;GSE133624_reads-count-all-sample.txt&quot;</span>, <span class="at">header=</span><span class="dv">1</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb1-5"><a href="index.html#cb1-5" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">colnames</span>(counts), <span class="cf">function</span> (x) <span class="fu">substring</span>(x,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb1-6"><a href="index.html#cb1-6" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> <span class="fu">data.frame</span>(meta)</span>
<span id="cb1-7"><a href="index.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(meta) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Condition&quot;</span>)</span>
<span id="cb1-8"><a href="index.html#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="index.html#cb1-9" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb1-10"><a href="index.html#cb1-10" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> meta,</span>
<span id="cb1-11"><a href="index.html#cb1-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design=</span> <span class="sc">~</span> Condition)</span>
<span id="cb1-12"><a href="index.html#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Prefiltering</span></span>
<span id="cb1-13"><a href="index.html#cb1-13" aria-hidden="true" tabindex="-1"></a>filt <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&lt;</span> <span class="dv">10</span>) <span class="sc">&gt;</span> <span class="fu">dim</span>(meta)[<span class="dv">1</span>]<span class="sc">*</span><span class="fl">0.9</span></span>
<span id="cb1-14"><a href="index.html#cb1-14" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="sc">!</span>filt,]</span>
<span id="cb1-15"><a href="index.html#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="index.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Perform DESeq2()</span></span>
<span id="cb1-17"><a href="index.html#cb1-17" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb1-18"><a href="index.html#cb1-18" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">pAdjustMethod =</span> <span class="st">&quot;bonferroni&quot;</span>)</span>
<span id="cb1-19"><a href="index.html#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="index.html#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="do">## apply variance stabilizing transformation</span></span>
<span id="cb1-21"><a href="index.html#cb1-21" aria-hidden="true" tabindex="-1"></a>v <span class="ot">=</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span>T)</span>
<span id="cb1-22"><a href="index.html#cb1-22" aria-hidden="true" tabindex="-1"></a>vsted <span class="ot">=</span> <span class="fu">assay</span>(v)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot PCA of VST values</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotPCA</span>(v, <span class="at">intgroup=</span><span class="st">&quot;Condition&quot;</span>)<span class="sc">+</span></span>
<span id="cb2-3"><a href="index.html#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="cpbnplot_doc_files/figure-html/plotVST-1.png" width="960" /></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Define the input genes, and use clusterProfiler::bitr to convert the ID.</span></span>
<span id="cb3-2"><a href="index.html#cb3-2" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">=</span> <span class="fu">subset</span>(res, padj<span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb3-3"><a href="index.html#cb3-3" aria-hidden="true" tabindex="-1"></a>cand.entrez <span class="ot">=</span> clusterProfiler<span class="sc">::</span><span class="fu">bitr</span>(<span class="fu">rownames</span>(sig), <span class="at">fromType=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">toType=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb=</span>org.Hs.eg.db)<span class="sc">$</span>ENTREZID</span>
<span id="cb3-4"><a href="index.html#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="index.html#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Perform enrichment analysis (ORA)</span></span>
<span id="cb3-6"><a href="index.html#cb3-6" aria-hidden="true" tabindex="-1"></a>pway <span class="ot">=</span> ReactomePA<span class="sc">::</span><span class="fu">enrichPathway</span>(<span class="at">gene =</span> cand.entrez)</span>
<span id="cb3-7"><a href="index.html#cb3-7" aria-hidden="true" tabindex="-1"></a>pwayGO <span class="ot">=</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichGO</span>(cand.entrez, <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb3-8"><a href="index.html#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="index.html#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Convert to SYMBOL</span></span>
<span id="cb3-10"><a href="index.html#cb3-10" aria-hidden="true" tabindex="-1"></a>pway <span class="ot">=</span> <span class="fu">setReadable</span>(pway, <span class="at">OrgDb=</span>org.Hs.eg.db)</span>
<span id="cb3-11"><a href="index.html#cb3-11" aria-hidden="true" tabindex="-1"></a>pwayGO <span class="ot">=</span> <span class="fu">setReadable</span>(pwayGO, <span class="at">OrgDb=</span>org.Hs.eg.db)</span>
<span id="cb3-12"><a href="index.html#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="index.html#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Store the similarity</span></span>
<span id="cb3-14"><a href="index.html#cb3-14" aria-hidden="true" tabindex="-1"></a>pway <span class="ot">=</span> enrichplot<span class="sc">::</span><span class="fu">pairwise_termsim</span>(pway)</span>
<span id="cb3-15"><a href="index.html#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="index.html#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Define including samples</span></span>
<span id="cb3-17"><a href="index.html#cb3-17" aria-hidden="true" tabindex="-1"></a>incSample <span class="ot">=</span> <span class="fu">rownames</span>(<span class="fu">subset</span>(meta, Condition<span class="sc">==</span><span class="st">&quot;T&quot;</span>))</span></code></pre></div>
<p>Additionally, for the gene set enrichment analysis, log2 fold change is obtained.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="index.html#cb4-1" aria-hidden="true" tabindex="-1"></a>allEntrez <span class="ot">=</span> clusterProfiler<span class="sc">::</span><span class="fu">bitr</span>(<span class="fu">rownames</span>(res), <span class="at">fromType=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">toType=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb=</span>org.Hs.eg.db)</span>
<span id="cb4-2"><a href="index.html#cb4-2" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>ENSEMBL <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)</span>
<span id="cb4-3"><a href="index.html#cb4-3" aria-hidden="true" tabindex="-1"></a>lfc <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="fu">data.frame</span>(res), allEntrez, <span class="at">by=</span><span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb4-4"><a href="index.html#cb4-4" aria-hidden="true" tabindex="-1"></a>lfc <span class="ot">&lt;-</span> lfc[<span class="fu">order</span>(lfc<span class="sc">$</span>log2FoldChange, <span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb4-5"><a href="index.html#cb4-5" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">&lt;-</span> lfc<span class="sc">$</span>log2FoldChange</span>
<span id="cb4-6"><a href="index.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneList) <span class="ot">&lt;-</span> lfc<span class="sc">$</span>ENTREZID</span>
<span id="cb4-7"><a href="index.html#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="index.html#cb4-8" aria-hidden="true" tabindex="-1"></a>pwayGSE <span class="ot">&lt;-</span> ReactomePA<span class="sc">::</span><span class="fu">gsePathway</span>(geneList)</span></code></pre></div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Chen2019-ve" class="csl-entry">
Chen, Xin, Ang Li, Bao-Fa Sun, Ying Yang, Ya-Nan Han, Xun Yuan, Ri-Xin Chen, et al. 2019. <span>“5-Methylcytosine Promotes Pathogenesis of Bladder Cancer Through Stabilizing <span class="nocase">mRNAs</span>.”</span> <em>Nat. Cell Biol.</em> 21 (8): 978–90.
</div>
<div id="ref-Killian2021-ko" class="csl-entry">
Killian, Theo, and Laurent Gatto. 2021. <span>“Exploiting the <span>DepMap</span> Cancer Dependency Data Using the Depmap <span>R</span> Package.”</span> <em>F1000Res.</em> 10 (416): 416.
</div>
<div id="ref-Love2014-yt" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for <span class="nocase">RNA-seq</span> Data with <span>DESeq2</span>.”</span> <em>Genome Biol.</em> 15 (12): 550.
</div>
<div id="ref-Sales2012-cu" class="csl-entry">
Sales, Gabriele, Enrica Calura, Duccio Cavalieri, and Chiara Romualdi. 2012. <span>“Graphite - a Bioconductor Package to Convert Pathway Topology to Gene Network.”</span> <em>BMC Bioinformatics</em> 13 (January): 20.
</div>
<div id="ref-Scutari2010-tj" class="csl-entry">
Scutari, Marco. 2010. <span>“Learning Bayesian Networks with the Bnlearn <span>R</span> Package.”</span> <em>Journal of Statistical Software, Articles</em> 35 (3): 1–22.
</div>
<div id="ref-Wu2021-sk" class="csl-entry">
Wu, Tianzhi, Erqiang Hu, Shuangbin Xu, Meijun Chen, Pingfan Guo, Zehan Dai, Tingze Feng, et al. 2021. <span>“clusterProfiler 4.0: A Universal Enrichment Tool for Interpreting Omics Data.”</span> <em>Innov. J.</em>, July, 100141.
</div>
<div id="ref-Yu2016-kh" class="csl-entry">
Yu, Guangchuang, and Qing-Yu He. 2016. <span>“<span>ReactomePA</span>: An <span>R/Bioconductor</span> Package for Reactome Pathway Analysis and Visualization.”</span> <em>Mol. Biosyst.</em> 12 (2): 477–79.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="bootstrap-based-inference.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cpbnplot_doc.pdf", "cpbnplot_doc.epub"],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
