<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Functional annotation | stana</title>
<meta name="author" content="Noriaki Sato">
<meta name="description" content="For metabolic functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for KEGG ORTHOLOGY which can be subsequently linked to KEGG PATHWAY...">
<meta name="generator" content="bookdown 0.41 with bs4_book()">
<meta property="og:title" content="Chapter 4 Functional annotation | stana">
<meta property="og:type" content="book">
<meta property="og:description" content="For metabolic functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for KEGG ORTHOLOGY which can be subsequently linked to KEGG PATHWAY...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Functional annotation | stana">
<meta name="twitter:description" content="For metabolic functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for KEGG ORTHOLOGY which can be subsequently linked to KEGG PATHWAY...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.33/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">stana</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="importing-and-filtering.html"><span class="header-section-number">2</span> Importing and filtering</a></li>
<li><a class="" href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></li>
<li><a class="active" href="function.html"><span class="header-section-number">4</span> Functional annotation</a></li>
<li><a class="" href="visualization.html"><span class="header-section-number">5</span> Visualization</a></li>
<li><a class="" href="interactive-inspection.html"><span class="header-section-number">6</span> Interactive inspection</a></li>
<li><a class="" href="analysis-example-of-gut-microbiome-of-esrd-patients.html"><span class="header-section-number">7</span> Analysis example of gut microbiome of ESRD patients</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/noriakis/stana">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="function" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Functional annotation<a class="anchor" aria-label="anchor" href="#function"><i class="fas fa-link"></i></a>
</h1>
<p>For metabolic functional profiling, several functions are prepared in <code>stana</code>. Note that these functions are primarily designed for KEGG ORTHOLOGY which can be subsequently linked to KEGG PATHWAY and the other databases in KEGG. However, the other options such as the enzyme commision numbers can be accepted.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
<div id="parsing-patric-results" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Parsing <code>PATRIC</code> results<a class="anchor" aria-label="anchor" href="#parsing-patric-results"><i class="fas fa-link"></i></a>
</h2>
<p>If the gene IDs in the gene matrix stored in the stana object is PATRIC ID, we can use <code>checkPATRIC</code> function to obtain information related to <code>PATRIC</code> functional annotation. The function accepts the input of named list of genes, and returns functional annotation results. The function uses <code>BiocFileCache()</code> to cache the obtained results from API.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"test"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"../test_genes.txt"</span><span class="op">)</span><span class="op">$</span><span class="va">V1</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; $test</span></span>
<span><span class="co">#&gt;  [1] "1280701.3.peg.1153" "1280701.3.peg.1169"</span></span>
<span><span class="co">#&gt;  [3] "1280701.3.peg.1174" "1280701.3.peg.1179"</span></span>
<span><span class="co">#&gt;  [5] "1280701.3.peg.1186" "1280701.3.peg.203" </span></span>
<span><span class="co">#&gt;  [7] "1280701.3.peg.361"  "1280701.3.peg.363" </span></span>
<span><span class="co">#&gt;  [9] "1280701.3.peg.368"  "1280701.3.peg.369" </span></span>
<span><span class="co">#&gt; [11] "1280701.3.peg.570"  "1280701.3.peg.758" </span></span>
<span><span class="co">#&gt; [13] "1280701.3.peg.762"  "1410605.3.peg.1201"</span></span>
<span><span class="co">#&gt; [15] "1410605.3.peg.1204" "1410605.3.peg.1463"</span></span>
<span><span class="co">#&gt; [17] "1410605.3.peg.1510" "1410605.3.peg.1568"</span></span>
<span><span class="co">#&gt; [19] "1410605.3.peg.450"  "1410605.3.peg.593" </span></span>
<span><span class="co">#&gt; [21] "1410605.3.peg.655"  "1410605.3.peg.833" </span></span>
<span><span class="co">#&gt; [23] "1410605.3.peg.844"  "1410605.3.peg.903" </span></span>
<span><span class="co">#&gt; [25] "1410605.3.peg.953"  "1447715.5.peg.1455"</span></span>
<span><span class="co">#&gt; [27] "1447715.5.peg.1549" "1447715.5.peg.1582"</span></span>
<span><span class="co">#&gt; [29] "1447715.5.peg.256"  "1447715.5.peg.33"  </span></span>
<span><span class="co">#&gt; [31] "1447715.5.peg.566"  "1447715.5.peg.964"</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/checkPATRIC.html">checkPATRIC</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Obtaining annotations of 3 genomes</span></span>
<span><span class="co">#&gt;   Obtaining information on 1280701.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1410605.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1447715.5</span></span>
<span><span class="co">#&gt; Checking results on cluster test</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained after removal of duplication</span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">test</span><span class="op">$</span><span class="va">DF</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scrollX<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-af325afa1344d0b3ee07" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-af325afa1344d0b3ee07">{"x":{"filter":"none","vertical":false,"data":[["254","608"],["fig|1280701.3.peg.570","fig|1280701.3.peg.1186"],["4.2.1.51","2.1.1.37"],["Prephenate dehydratase","DNA (cytosine-5-)-methyltransferase"],[400,270],["Phenylalanine, tyrosine and tryptophan biosynthesis","Cysteine and methionine metabolism"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patric_id<\/th>\n      <th>ec_number<\/th>\n      <th>ec_description<\/th>\n      <th>pathway_id<\/th>\n      <th>pathway_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"patric_id","targets":1},{"name":"ec_number","targets":2},{"name":"ec_description","targets":3},{"name":"pathway_id","targets":4},{"name":"pathway_name","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div id="draw-the-network" class="section level3" number="4.1.1">
<h3>
<span class="header-section-number">4.1.1</span> Draw the network<a class="anchor" aria-label="anchor" href="#draw-the-network"><i class="fas fa-link"></i></a>
</h3>
<p>You can draw the graph of obtained results depicting enzyme to KEGG PATHWAY relationship.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/drawPATRIC.html">drawPATRIC</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; Obtaining annotations of 3 genomes</span></span>
<span><span class="co">#&gt;   Obtaining information on 1280701.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1410605.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1447715.5</span></span>
<span><span class="co">#&gt; Checking results on cluster test</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on ec_description</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Making graph on pathway_name and ec_description</span></span>
<span><span class="co">#&gt;   subsetting to 5 label on each category</span></span>
<span><span class="co">#&gt; $test</span></span>
<span><span class="co">#&gt; $test$DF</span></span>
<span><span class="co">#&gt;                  patric_id ec_number</span></span>
<span><span class="co">#&gt; 254  fig|1280701.3.peg.570  4.2.1.51</span></span>
<span><span class="co">#&gt; 608 fig|1280701.3.peg.1186  2.1.1.37</span></span>
<span><span class="co">#&gt;                          ec_description pathway_id</span></span>
<span><span class="co">#&gt; 254              Prephenate dehydratase        400</span></span>
<span><span class="co">#&gt; 608 DNA (cytosine-5-)-methyltransferase        270</span></span>
<span><span class="co">#&gt;                                            pathway_name</span></span>
<span><span class="co">#&gt; 254 Phenylalanine, tyrosine and tryptophan biosynthesis</span></span>
<span><span class="co">#&gt; 608                  Cysteine and methionine metabolism</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$REMOVEDUP</span></span>
<span><span class="co">#&gt;                  patric_id ec_number</span></span>
<span><span class="co">#&gt; 254  fig|1280701.3.peg.570  4.2.1.51</span></span>
<span><span class="co">#&gt; 608 fig|1280701.3.peg.1186  2.1.1.37</span></span>
<span><span class="co">#&gt;                          ec_description pathway_id</span></span>
<span><span class="co">#&gt; 254              Prephenate dehydratase        400</span></span>
<span><span class="co">#&gt; 608 DNA (cytosine-5-)-methyltransferase        270</span></span>
<span><span class="co">#&gt;                                            pathway_name</span></span>
<span><span class="co">#&gt; 254 Phenylalanine, tyrosine and tryptophan biosynthesis</span></span>
<span><span class="co">#&gt; 608                  Cysteine and methionine metabolism</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$SORTED</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DNA (cytosine-5-)-methyltransferase </span></span>
<span><span class="co">#&gt;                                   1 </span></span>
<span><span class="co">#&gt;              Prephenate dehydratase </span></span>
<span><span class="co">#&gt;                                   1 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$GRAPH</span></span>
<span><span class="co">#&gt; IGRAPH d36e63c UN-- 4 2 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c)</span></span>
<span><span class="co">#&gt; + edges from d36e63c (vertex names):</span></span>
<span><span class="co">#&gt; [1] Prephenate dehydratase             --Phenylalanine, tyrosine and tryptophan biosynthesis</span></span>
<span><span class="co">#&gt; [2] DNA (cytosine-5-)-methyltransferase--Cysteine and methionine metabolism                 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$PLOT</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/drawpat-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
</div>
<div id="parsing-eggnog-mapper-v2-results" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Parsing <code>eggNOG-mapper v2</code> results<a class="anchor" aria-label="anchor" href="#parsing-eggnog-mapper-v2-results"><i class="fas fa-link"></i></a>
</h2>
<p>If the genes are not annotated, one can use annotation software and produces an assignment file. We can use <code>checkEGGNOG</code> function to read the output of eggNOG-mapper v2.
You should perform the annotation to genes using <a href="http://eggnog-mapper.embl.de/">the server</a> or the software.
Specify IDs you want to obtain to <code>ret</code>, such as “KEGG_ko” and “KEGG_Pathway”.
The annotation can be used with the other functions.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/checkEGGNOG.html">checkEGGNOG</a></span><span class="op">(</span><span class="st">"../annotations_gtdb/100224_eggnog_out.emapper.annotations"</span>,</span>
<span>    ret<span class="op">=</span><span class="st">"KEGG_ko"</span><span class="op">)</span></span>
<span><span class="va">tib</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-717340fc300d5bed6dfc" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-717340fc300d5bed6dfc">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["GCF_002846775.1_00408","GCF_002846815.1_01743","GCF_004156145.1_01406","GCF_004155565.1_00557","GCF_004155645.1_00353","GCF_000800475.2_00338"],["KEGG_ko","KEGG_ko","KEGG_ko","KEGG_ko","KEGG_ko","KEGG_ko"],["ko:K11533","ko:K11533","ko:K11533","ko:K11533","ko:K11533","ko:K11533"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>name<\/th>\n      <th>value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"ID","targets":1},{"name":"name","targets":2},{"name":"value","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div id="draw-the-network-1" class="section level3" number="4.2.1">
<h3>
<span class="header-section-number">4.2.1</span> Draw the network<a class="anchor" aria-label="anchor" href="#draw-the-network-1"><i class="fas fa-link"></i></a>
</h3>
<p>You can draw the relationships between IDs by <code>drawEGGNOG</code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/drawEGGNOG.html">drawEGGNOG</a></span><span class="op">(</span><span class="st">"../annotations_gtdb/100224_eggnog_out.emapper.annotations"</span>,</span>
<span>            candPlot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"KEGG_ko"</span>,<span class="st">"KEGG_Pathway"</span><span class="op">)</span>,</span>
<span>            geneIDs <span class="op">=</span> <span class="va">tib</span><span class="op">$</span><span class="va">ID</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $graph</span></span>
<span><span class="co">#&gt; IGRAPH d45acee UN-- 21 922 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), category (v/c), size (v/n)</span></span>
<span><span class="co">#&gt; + edges from d45acee (vertex names):</span></span>
<span><span class="co">#&gt;  [1] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt;  [3] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt;  [5] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt;  [7] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt;  [9] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt; [11] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt; [13] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt; [15] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $plot</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/drawEGGNOG-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
</div>
<div id="heatmap-of-the-gene-copy-numbers-with-functional-annotations" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Heatmap of the gene copy numbers with functional annotations<a class="anchor" aria-label="anchor" href="#heatmap-of-the-gene-copy-numbers-with-functional-annotations"><i class="fas fa-link"></i></a>
</h2>
<p>You can inspect the overview of functional differences using gene copy numbers along with <code>simplifyEnrichment</code>. The <code>plotHeatmap</code> function can be used with the stana object or preprocessed gene copy number matrix as input. In the function, <code>anno_PATRIC_keywords</code> and <code>anno_eggNOG_keywords</code> are used to plot the word clouds alongside <code>Heatmap</code> from <code>ComplexHeatmap</code>. It is easy for <code>MIDAS</code>, as the function obtains functional annotation from PATRIC API server and no annotation step is needed.</p>
<div id="midas-1" class="section level3" number="4.3.1">
<h3>
<span class="header-section-number">4.3.1</span> <code>MIDAS</code><a class="anchor" aria-label="anchor" href="#midas-1"><i class="fas fa-link"></i></a>
</h3>
<p>For <code>MIDAS</code>, the function automatically query API of <code>PATRIC</code> server using the gene names. As the gene number is large typically, one can filter the genes by options <code>filter_zero_frac</code>, <code>filter_max_frac</code> and <code>filter_max_value</code>. However, one should perform own filtering beforehand and provide the matrix to <code>mat</code>. If <code>mat</code> is specified, other filtering options will be ignored.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simplifyEnrichment">simplifyEnrichment</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS.html">loadMIDAS</a></span><span class="op">(</span><span class="st">"../merge_midas1/"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="st">"Bacteroides_uniformis_57318"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bacteroides_uniformis_57318</span></span>
<span><span class="co">#&gt;   Snps</span></span>
<span><span class="co">#&gt;     HC 13</span></span>
<span><span class="co">#&gt;     R 16</span></span>
<span><span class="co">#&gt;     Bacteroides_uniformis_57318 cleared filtering threshold in SNV</span></span>
<span><span class="co">#&gt;   Genes</span></span>
<span><span class="co">#&gt;     HC 13</span></span>
<span><span class="co">#&gt;     R 16</span></span>
<span><span class="co">#&gt;     Bacteroides_uniformis_57318 cleared filtering threshold in genes</span></span>
<span><span class="co">#&gt; Overall, 1 species met criteria in SNPs</span></span>
<span><span class="co">#&gt; Overall, 1 species met criteria in genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"Bacteroides_uniformis_57318"</span>,</span>
<span>    filter_max_value <span class="op">=</span> <span class="fl">2</span>, filter_max_frac <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">#&gt; # In resulting matrix, max: 1.99955229433902 min: 0</span></span>
<span><span class="co">#&gt; # Dimension: 4911 29</span></span>
<span><span class="co">#&gt; Obtaining gene information from PATRIC server</span></span>
<span><span class="co">#&gt; Obtaining annotations of 7 genomes</span></span>
<span><span class="co">#&gt;   Obtaining information on 1235787.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1339348.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 411479.10</span></span>
<span><span class="co">#&gt;   Obtaining information on 457393.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 585543.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 997889.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 997890.3</span></span>
<span><span class="co">#&gt; Checking results on cluster 1</span></span>
<span><span class="co">#&gt;   total of 1181 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 1145 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 2</span></span>
<span><span class="co">#&gt;   total of 702 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 679 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 3</span></span>
<span><span class="co">#&gt;   total of 486 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 473 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 4</span></span>
<span><span class="co">#&gt;   total of 175 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 174 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 5</span></span>
<span><span class="co">#&gt;   total of 233 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 227 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 6</span></span>
<span><span class="co">#&gt;   total of 997 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 960 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 7</span></span>
<span><span class="co">#&gt;   total of 472 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 456 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 8</span></span>
<span><span class="co">#&gt;   total of 939 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 907 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 9</span></span>
<span><span class="co">#&gt;   total of 374 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 364 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 10</span></span>
<span><span class="co">#&gt;   total of 552 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 525 annotation obtained after removal of duplication</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/MIDAS1_heatmap-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="the-other-types" class="section level3" number="4.3.2">
<h3>
<span class="header-section-number">4.3.2</span> The other types<a class="anchor" aria-label="anchor" href="#the-other-types"><i class="fas fa-link"></i></a>
</h3>
<p>The users should provide eggNOG annotation on <code>eggNOG</code> slot of stana object. <code>fnc</code> argument accepts <code>KEGG_Pathway</code> or <code>KEGG_Module</code> available in eggNOG annotation. The function queries <code>KEGG REST API</code> to obtain pathway and module description.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS2.html">loadMIDAS2</a></span><span class="op">(</span><span class="st">"../merge_uhgg"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">)</span>, db<span class="op">=</span><span class="st">"uhgg"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of snps: 70178</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of genes: 120158</span></span>
<span><span class="co">#&gt;     Number of samples: 31</span></span></code></pre></div>
<p>We set the eggNOG-mapper v2 annotation file to the eggNOG slot of stana object.
This can be done by <code>setAnnotation</code> function.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set the annotation file</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setAnnotation.html">setAnnotation</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">=</span><span class="st">"../annotations_uhgg/101346_eggnog_out.emapper.annotations"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>You can set <code>removeAdditional</code> argument to filter words that are to be displayed.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simplifyEnrichment">simplifyEnrichment</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span>,</span>
<span>    fnc<span class="op">=</span><span class="st">"KEGG_Module"</span>,</span>
<span>    removeAdditional<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cycle"</span>,<span class="st">"pathway"</span><span class="op">)</span>,</span>
<span>    filter_zero_frac <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    filter_max_frac <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    filter_max_value <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; # In resulting matrix, max: 4.999739 min: 0</span></span>
<span><span class="co">#&gt; # Dimension: 3501 31</span></span>
<span><span class="co">#&gt; # Looking for the annotation file by eggNOG-mapper v2</span></span>
<span><span class="co">#&gt; # Loading annotation</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/MIDAS2_heatmap-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The gene IDs can be subset by providing the IDs to the <code>geneID</code> option. If <code>variable</code> is set to other than zero, the genes with top-{<code>variable</code>} variable features were subset to be used in the heatmap visualization.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Subset to top-100 variable genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span>, variable<span class="op">=</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; # In resulting matrix, max: 49.729167 min: 0</span></span>
<span><span class="co">#&gt; # Dimension: 18808 31</span></span>
<span><span class="co">#&gt; # Looking for the annotation file by eggNOG-mapper v2</span></span>
<span><span class="co">#&gt; # Loading annotation</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/heat4-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
</div>
<div id="aggregate-the-gene-copy-numbers-based-on-the-annotations" class="section level2" number="4.4">
<h2>
<span class="header-section-number">4.4</span> Aggregate the gene copy numbers based on the annotations<a class="anchor" aria-label="anchor" href="#aggregate-the-gene-copy-numbers-based-on-the-annotations"><i class="fas fa-link"></i></a>
</h2>
<p><code>calcGF</code> function can aggregate the gene copy numbers based on the annotation (eggNOG) or the manually set annotation.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Default to use eggNOG annotation, and the user can specify which gene family to summarize</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/calcGF.html">calcGF</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span>, column<span class="op">=</span><span class="st">"EC"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"kos"</span><span class="op">)</span><span class="op">[[</span><span class="st">"101346"</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;           ERR9492489 ERR9492490 ERR9492491 ERR9492492</span></span>
<span><span class="co">#&gt; 3.6.4.12   39.173924  95.450865  52.767661  46.882936</span></span>
<span><span class="co">#&gt; 3.4.24.40   0.000000   0.841116   0.278746   0.000000</span></span>
<span><span class="co">#&gt; 3.2.1.14    1.198363   5.697443   2.435163   0.298239</span></span>
<span><span class="co">#&gt; 4.2.2.23    1.586967   8.950180   3.871217   0.326971</span></span>
<span><span class="co">#&gt; 2.7.11.1    7.415218  17.180451  17.557883  12.640404</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.000000   0.000000   0.000000   0.000000</span></span>
<span><span class="co">#&gt;           ERR9492493 ERR9492494 ERR9492495 ERR9492496</span></span>
<span><span class="co">#&gt; 3.6.4.12  109.030709  38.157274  61.418044  32.840106</span></span>
<span><span class="co">#&gt; 3.4.24.40   2.289508   0.000000   0.000000   0.000000</span></span>
<span><span class="co">#&gt; 3.2.1.14    4.716068   0.565635   3.950550   0.842613</span></span>
<span><span class="co">#&gt; 4.2.2.23    2.222589   2.685389   1.223989   0.547461</span></span>
<span><span class="co">#&gt; 2.7.11.1   13.093117  16.668716   9.321875   3.957696</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.309415   0.000000   0.000000   0.000000</span></span>
<span><span class="co">#&gt;           ERR9492497 ERR9492498 ERR9492499 ERR9492500</span></span>
<span><span class="co">#&gt; 3.6.4.12   75.977012  28.210277  61.693399  36.666227</span></span>
<span><span class="co">#&gt; 3.4.24.40   1.190823   0.117354   0.276488   0.000000</span></span>
<span><span class="co">#&gt; 3.2.1.14    3.989812   0.833608   2.955427   1.241877</span></span>
<span><span class="co">#&gt; 4.2.2.23    2.361721   0.534308   2.707128   0.000000</span></span>
<span><span class="co">#&gt; 2.7.11.1   24.343595   8.375070  15.225822   9.506420</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.399345   0.000000   0.334459   0.000000</span></span>
<span><span class="co">#&gt;           ERR9492501 ERR9492503 ERR9492504 ERR9492505</span></span>
<span><span class="co">#&gt; 3.6.4.12   48.993525  58.889647  41.316522  44.670327</span></span>
<span><span class="co">#&gt; 3.4.24.40   0.605779   0.000000   1.281429   0.431620</span></span>
<span><span class="co">#&gt; 3.2.1.14    6.470900   2.073627   1.787648   1.916355</span></span>
<span><span class="co">#&gt; 4.2.2.23    6.066784   3.795593   0.000000   1.875545</span></span>
<span><span class="co">#&gt; 2.7.11.1   21.694552  19.085136  15.225251  12.040248</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.328735   0.210771   0.000000   0.133266</span></span>
<span><span class="co">#&gt;           ERR9492507 ERR9492509 ERR9492510 ERR9492511</span></span>
<span><span class="co">#&gt; 3.6.4.12   30.414200  94.222205  67.033382  45.280463</span></span>
<span><span class="co">#&gt; 3.4.24.40   0.000000   0.000000   0.582293   0.050622</span></span>
<span><span class="co">#&gt; 3.2.1.14    0.113379   9.611853   3.004519   0.816464</span></span>
<span><span class="co">#&gt; 4.2.2.23    5.876491   9.608170   6.104285   0.227825</span></span>
<span><span class="co">#&gt; 2.7.11.1   15.459505  48.563617  16.882499   6.723153</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.000000   0.330640   0.000000   0.209655</span></span>
<span><span class="co">#&gt;           ERR9492512 ERR9492513 ERR9492514 ERR9492515</span></span>
<span><span class="co">#&gt; 3.6.4.12   37.248780  92.564239  44.501316  33.343923</span></span>
<span><span class="co">#&gt; 3.4.24.40   0.000000   0.699384   0.809852   0.000000</span></span>
<span><span class="co">#&gt; 3.2.1.14    2.928634   3.767824   1.008637   0.391071</span></span>
<span><span class="co">#&gt; 4.2.2.23    0.200991   2.143717   0.000000   2.435797</span></span>
<span><span class="co">#&gt; 2.7.11.1   13.147839  19.027770  17.290957   8.964407</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.000000   0.000000   0.000000   0.000000</span></span>
<span><span class="co">#&gt;           ERR9492519 ERR9492521 ERR9492522 ERR9492523</span></span>
<span><span class="co">#&gt; 3.6.4.12  123.352975  19.992745 113.496963  73.578048</span></span>
<span><span class="co">#&gt; 3.4.24.40   0.612824   0.044187   0.000000   0.289869</span></span>
<span><span class="co">#&gt; 3.2.1.14    2.685511   0.555955   4.377865   4.203646</span></span>
<span><span class="co">#&gt; 4.2.2.23    7.924792   2.183619   8.216322   2.000771</span></span>
<span><span class="co">#&gt; 2.7.11.1   21.618306  16.281798  27.935448   8.713224</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.745864   0.064679   0.000000   0.270559</span></span>
<span><span class="co">#&gt;           ERR9492525 ERR9492526 ERR9492528</span></span>
<span><span class="co">#&gt; 3.6.4.12   22.958707  64.120946  37.810981</span></span>
<span><span class="co">#&gt; 3.4.24.40   0.048448   0.548795   0.000000</span></span>
<span><span class="co">#&gt; 3.2.1.14    0.671990   4.579498   2.684535</span></span>
<span><span class="co">#&gt; 4.2.2.23    2.174417   9.497381   3.574399</span></span>
<span><span class="co">#&gt; 2.7.11.1    3.474492  16.346382  10.342824</span></span>
<span><span class="co">#&gt; 3.1.3.90    0.147197   0.000000   0.000000</span></span></code></pre></div>
</div>
<div id="kgeg-pathway-and-kegg-orthology" class="section level2" number="4.5">
<h2>
<span class="header-section-number">4.5</span> KGEG PATHWAY and KEGG ORTHOLOGY<a class="anchor" aria-label="anchor" href="#kgeg-pathway-and-kegg-orthology"><i class="fas fa-link"></i></a>
</h2>
<div id="visualization-of-kegg-pathway" class="section level3" number="4.5.1">
<h3>
<span class="header-section-number">4.5.1</span> Visualization of KEGG PATHWAY<a class="anchor" aria-label="anchor" href="#visualization-of-kegg-pathway"><i class="fas fa-link"></i></a>
</h3>
<p><code>KEGG PATHWAY</code> is frequently used to characterize metabolic function of microbiome. Utilizing <a href="https://github.com/noriakis/ggkegg"><code>ggkegg</code></a>, the information of intra-species diversity, particulary gene copy number differences, can be reflected onto <code>KEGG PATHWAY</code>. It needs <code>genes</code> slot filled, and some annotations files, typically <code>eggNOG-mapper v2</code>, are needed.</p>
<div id="visualizing-differences-per-species" class="section level4" number="4.5.1.1">
<h4>
<span class="header-section-number">4.5.1.1</span> Visualizing differences per species<a class="anchor" aria-label="anchor" href="#visualizing-differences-per-species"><i class="fas fa-link"></i></a>
</h4>
<p>Load the profile for multiple species.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS2.html">loadMIDAS2</a></span><span class="op">(</span><span class="st">"../merge_uhgg"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, db<span class="op">=</span><span class="st">"uhgg"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of snps: 70178</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;   102438</span></span>
<span><span class="co">#&gt;   g__Parabacteroides;s__Parabacteroides distasonis</span></span>
<span><span class="co">#&gt;     Number of snps: 18102</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of genes: 120158</span></span>
<span><span class="co">#&gt;     Number of samples: 31</span></span>
<span><span class="co">#&gt;   102438</span></span>
<span><span class="co">#&gt;   g__Parabacteroides;s__Parabacteroides distasonis</span></span>
<span><span class="co">#&gt;     Number of genes: 47046</span></span>
<span><span class="co">#&gt;     Number of samples: 29</span></span></code></pre></div>
<p>Next, we set the eggNOG-mapper v2 annotation file to the eggNOG slot of stana object.
This way, the <code>plotKEGGPathway</code> function automatically calculates the abundance by user-defined method.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set the annotation file</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setAnnotation.html">setAnnotation</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">=</span><span class="st">"../annotations_uhgg/101346_eggnog_out.emapper.annotations"</span>,</span>
<span>    <span class="st">"102438"</span><span class="op">=</span><span class="st">"../annotations_uhgg/102438_eggnog_out.emapper.annotations"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>plotKEGGPathway can be run by providing stana object, species (when multiple species, rectangular nodes will be split),
and pathway ID to visualize. Here, we visualize <code>ko00620</code>, Pyruvate metabolism for example. As for the large annotation table, the calculation takes time and you can provide pre-calculated KO table in <code>kos</code> slot of stana object, or specify <code>only_ko = TRUE</code> to first return KO table.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, only_ko<span class="op">=</span><span class="cn">TRUE</span>, multi_scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, multi_scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/onescale-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>By default, the scale is same. If you install <code>ggh4x</code>, multiple scales can be added, by specifying <code>multi_scale</code> argument.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, multi_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/multscale-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>You can provide multiple pathway IDs to pathway_id, which returns a list of plot.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span>,<span class="st">"ko00620"</span><span class="op">)</span>,</span>
<span>                      multi_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg2</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span>
<span><span class="va">gg2</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>In this way, differences in orthologies in the pathway across multiple species can be readily captured.</p>
</div>
</div>
<div id="visualizing-calculated-values-across-species" class="section level3" number="4.5.2">
<h3>
<span class="header-section-number">4.5.2</span> Visualizing calculated values across species<a class="anchor" aria-label="anchor" href="#visualizing-calculated-values-across-species"><i class="fas fa-link"></i></a>
</h3>
<p>If you want to see the sum values across species, you can set option <code>summarize=TRUE</code>. This way, the KO values across specified species are summed, and compared between groups, then plotted.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span>,<span class="st">"ko00620"</span><span class="op">)</span>,</span>
<span>                      summarize<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg2</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span>
<span><span class="va">gg2</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway_SUM-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="show-which-species-have-the-kos" class="section level3" number="4.5.3">
<h3>
<span class="header-section-number">4.5.3</span> Show which species have the KOs<a class="anchor" aria-label="anchor" href="#show-which-species-have-the-kos"><i class="fas fa-link"></i></a>
</h3>
<p>If you specify <code>point_mode=TRUE</code>, the function plot points on the KOs on the pathways
based on whether the specified species have the corresponding KOs annotated.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span><span class="op">)</span>,</span>
<span>                      sp_colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>,<span class="st">"red"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      point_mode<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="co">#&gt; Point mode enabled</span></span>
<span><span class="va">gg</span></span>
<span><span class="co">#&gt; $ko00270</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway_point-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="ranking-the-components-in-the-graph" class="section level3" number="4.5.4">
<h3>
<span class="header-section-number">4.5.4</span> Ranking the components in the graph<a class="anchor" aria-label="anchor" href="#ranking-the-components-in-the-graph"><i class="fas fa-link"></i></a>
</h3>
<p>By using calculated statistics of gene family (in this case, KO), one can rank the compounds catalyzed by these enzymes using graph information. Specify the candidate species ID and pathway ID and the function automatically calculates the values.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidygraph.data-imaginist.com">tidygraph</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/rankComponents.html">rankComponents</a></span><span class="op">(</span><span class="va">stana</span>, pid<span class="op">=</span><span class="st">"ko00270"</span>, candSp<span class="op">=</span><span class="st">"102438"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Using pre-computed KO table</span></span>
<span><span class="co">#&gt; # HC / R</span></span>
<span><span class="co">#&gt;          name        rank</span></span>
<span><span class="co">#&gt; 1  cpd:C02218  1.54370291</span></span>
<span><span class="co">#&gt; 2  cpd:C09306  1.40561913</span></span>
<span><span class="co">#&gt; 3  cpd:C00041  1.40561913</span></span>
<span><span class="co">#&gt; 4  cpd:C01077  1.10144476</span></span>
<span><span class="co">#&gt; 5  cpd:C00263  1.09552071</span></span>
<span><span class="co">#&gt; 6  cpd:C01118  0.99254895</span></span>
<span><span class="co">#&gt; 7  cpd:C03082  0.89568277</span></span>
<span><span class="co">#&gt; 8  cpd:C05823  0.87235086</span></span>
<span><span class="co">#&gt; 9  cpd:C00049  0.60348505</span></span>
<span><span class="co">#&gt; 10 cpd:C03539  0.36484440</span></span>
<span><span class="co">#&gt; 11 cpd:C05527 -0.08284236</span></span>
<span><span class="co">#&gt; 12 cpd:C03232 -0.59819630</span></span>
<span><span class="co">#&gt; 13 cpd:C02989 -1.18193737</span></span>
<span><span class="co">#&gt; 14 cpd:C00197 -1.50453597</span></span></code></pre></div>
</div>
</div>
<div id="setting-the-manual-annotation" class="section level2" number="4.6">
<h2>
<span class="header-section-number">4.6</span> Setting the manual annotation<a class="anchor" aria-label="anchor" href="#setting-the-manual-annotation"><i class="fas fa-link"></i></a>
</h2>
<p>Using <code>setMap</code> function, one can set stana object a named data frame of mapping file between gene and gene families.
In this case, the data.frame should be two column layouts, and the first column corresponds to gene ID and the second column corresponds to the gene family IDs.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setMap.html">setMap</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"geneID1"</span>,<span class="st">"geneID2"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"K00001"</span>,<span class="st">"K00002"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="pathwaywithfactor" class="section level2" number="4.7">
<h2>
<span class="header-section-number">4.7</span> <code>pathwayWithFactor</code><a class="anchor" aria-label="anchor" href="#pathwaywithfactor"><i class="fas fa-link"></i></a>
</h2>
<p>The KO abundances can be aggregated to KEGG PATHWAY abundance by <code>pathwayWithFactor</code>.
By default, the function uses the KO matrix from <code>NMF</code> slot. <code>mat</code> can be specified for the other matrices.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Row.names: KO, colnames: Samples</span></span>
<span><span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"kos"</span><span class="op">)</span><span class="op">[[</span><span class="st">"101346"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/pathwayWithFactor.html">pathwayWithFactor</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span>, summarize<span class="op">=</span><span class="va">mean</span>, mat<span class="op">=</span><span class="va">ch</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;              ERR9492489 ERR9492490 ERR9492491 ERR9492492</span></span>
<span><span class="co">#&gt; path:ko00010 0.03877215 0.13322643 0.04001532 0.06391891</span></span>
<span><span class="co">#&gt; path:ko00020 0.03425545 0.12828687 0.03985006 0.07768581</span></span>
<span><span class="co">#&gt; path:ko00030 0.04703528 0.13909857 0.06912219 0.04676925</span></span>
<span><span class="co">#&gt; path:ko00040 0.04000612 0.17272473 0.06849944 0.05096996</span></span>
<span><span class="co">#&gt; path:ko00051 0.05214005 0.07052621 0.06382708 0.03420876</span></span>
<span><span class="co">#&gt; path:ko00052 0.06397950 0.07191146 0.04914795 0.03673800</span></span>
<span><span class="co">#&gt;              ERR9492493 ERR9492494 ERR9492495 ERR9492496</span></span>
<span><span class="co">#&gt; path:ko00010 0.09384163 0.05637166 0.06068150 0.02861835</span></span>
<span><span class="co">#&gt; path:ko00020 0.08709860 0.06347143 0.08655160 0.02838112</span></span>
<span><span class="co">#&gt; path:ko00030 0.09236281 0.06725684 0.09888781 0.06070932</span></span>
<span><span class="co">#&gt; path:ko00040 0.09423838 0.09310127 0.09308659 0.04017714</span></span>
<span><span class="co">#&gt; path:ko00051 0.05898339 0.06181811 0.05094572 0.04122640</span></span>
<span><span class="co">#&gt; path:ko00052 0.06410715 0.05452204 0.14107086 0.03358680</span></span>
<span><span class="co">#&gt;              ERR9492497 ERR9492498 ERR9492499 ERR9492500</span></span>
<span><span class="co">#&gt; path:ko00010 0.08451250 0.03267193 0.09284024 0.06000493</span></span>
<span><span class="co">#&gt; path:ko00020 0.06743175 0.02951531 0.09662457 0.05163185</span></span>
<span><span class="co">#&gt; path:ko00030 0.11954092 0.04752626 0.09501196 0.12192500</span></span>
<span><span class="co">#&gt; path:ko00040 0.09575037 0.04763954 0.09862459 0.05061098</span></span>
<span><span class="co">#&gt; path:ko00051 0.08917651 0.04097972 0.09009920 0.09712665</span></span>
<span><span class="co">#&gt; path:ko00052 0.10890301 0.03198491 0.04649193 0.08376906</span></span>
<span><span class="co">#&gt;              ERR9492501 ERR9492503 ERR9492504 ERR9492505</span></span>
<span><span class="co">#&gt; path:ko00010 0.06353951 0.07530207 0.07413198 0.03851865</span></span>
<span><span class="co">#&gt; path:ko00020 0.06736760 0.05907952 0.08066299 0.03291251</span></span>
<span><span class="co">#&gt; path:ko00030 0.08062611 0.10059762 0.09074628 0.06140277</span></span>
<span><span class="co">#&gt; path:ko00040 0.05230380 0.10797610 0.05192520 0.04096908</span></span>
<span><span class="co">#&gt; path:ko00051 0.07521150 0.06162177 0.06966618 0.03856838</span></span>
<span><span class="co">#&gt; path:ko00052 0.07510973 0.05448280 0.08440857 0.05159773</span></span>
<span><span class="co">#&gt;              ERR9492507 ERR9492509 ERR9492510 ERR9492511</span></span>
<span><span class="co">#&gt; path:ko00010 0.04408184  0.1101960 0.06831861 0.04457377</span></span>
<span><span class="co">#&gt; path:ko00020 0.03055716  0.1122648 0.05686148 0.03786649</span></span>
<span><span class="co">#&gt; path:ko00030 0.05901192  0.2060442 0.07937786 0.06227169</span></span>
<span><span class="co">#&gt; path:ko00040 0.04993012  0.1811532 0.12112892 0.06357970</span></span>
<span><span class="co">#&gt; path:ko00051 0.03501170  0.1072305 0.09978821 0.04688276</span></span>
<span><span class="co">#&gt; path:ko00052 0.03050457  0.1082022 0.09676729 0.05031496</span></span>
<span><span class="co">#&gt;              ERR9492512 ERR9492513 ERR9492514 ERR9492515</span></span>
<span><span class="co">#&gt; path:ko00010 0.05663017 0.11421784 0.08596117 0.03688230</span></span>
<span><span class="co">#&gt; path:ko00020 0.04904705 0.09192427 0.07901509 0.04271306</span></span>
<span><span class="co">#&gt; path:ko00030 0.07407402 0.10263472 0.11231495 0.04835473</span></span>
<span><span class="co">#&gt; path:ko00040 0.04686250 0.10666516 0.07078609 0.05789550</span></span>
<span><span class="co">#&gt; path:ko00051 0.04646515 0.10730369 0.07973031 0.03461777</span></span>
<span><span class="co">#&gt; path:ko00052 0.05438493 0.07782405 0.09124815 0.04582225</span></span>
<span><span class="co">#&gt;              ERR9492519 ERR9492521 ERR9492522 ERR9492523</span></span>
<span><span class="co">#&gt; path:ko00010  0.2020544 0.03199985 0.11593241 0.06408050</span></span>
<span><span class="co">#&gt; path:ko00020  0.1902600 0.02793804 0.16804823 0.07081124</span></span>
<span><span class="co">#&gt; path:ko00030  0.1585779 0.05154594 0.11458300 0.06178188</span></span>
<span><span class="co">#&gt; path:ko00040  0.1028053 0.04093270 0.13291912 0.07091612</span></span>
<span><span class="co">#&gt; path:ko00051  0.1608883 0.03638188 0.25908852 0.04935026</span></span>
<span><span class="co">#&gt; path:ko00052  0.1023709 0.02991019 0.07093243 0.05402583</span></span>
<span><span class="co">#&gt;              ERR9492525 ERR9492526 ERR9492528</span></span>
<span><span class="co">#&gt; path:ko00010 0.02762329 0.07007589 0.05262720</span></span>
<span><span class="co">#&gt; path:ko00020 0.02638546 0.05562837 0.04079618</span></span>
<span><span class="co">#&gt; path:ko00030 0.04034426 0.08332154 0.09070246</span></span>
<span><span class="co">#&gt; path:ko00040 0.02860687 0.09845644 0.06750199</span></span>
<span><span class="co">#&gt; path:ko00051 0.03293917 0.08427063 0.06058627</span></span>
<span><span class="co">#&gt; path:ko00052 0.04860460 0.08358931 0.05641848</span></span></code></pre></div>
</div>
<div id="gsea" class="section level2" number="4.8">
<h2>
<span class="header-section-number">4.8</span> GSEA<a class="anchor" aria-label="anchor" href="#gsea"><i class="fas fa-link"></i></a>
</h2>
<p>The gene set enrichment analysis based on the background KEGG PATHWAY information is possible if the KO abundances are used. <code>doGSEA</code> function performs <code>GSEA</code> on the gene copy number table based on the grouping variable set in the <code>stana</code> object. The background set is obtained by KEGG REST API. The <code>gsea</code> slot is filled with the function.</p>
<div class="rmdwarning">
<p>Note that background gene set contains the human disease category and the results should be taken care of with caution.</p>
</div>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/doGSEA.html">doGSEA</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span><span class="op">)</span></span>
<span><span class="co">#&gt; HC / R</span></span></code></pre></div>
<p><code>plotGSEA</code> function can be used to draw a network representation of species ID and differential pathway at the specified threshold. List of multiple <code>stana</code> objects can be passed to the function, which helps interpret the inter-dataset (like diseases) differences of intra-species diversity. This function returns the plot by default, but <code>return_graph</code> can be set to TRUE to return only the tbl_graph object. Also, layout can be specified by <code>layout</code> argument.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidygraph.data-imaginist.com">tidygraph</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotGSEA.html">plotGSEA</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">stana</span>, <span class="va">stana</span><span class="op">)</span>, padjThreshold<span class="op">=</span><span class="fl">0.2</span>, layout<span class="op">=</span><span class="st">"fr"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/gsea2-1.png" width="100%" style="display: block; margin: auto;"></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></div>
<div class="next"><a href="visualization.html"><span class="header-section-number">5</span> Visualization</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#function"><span class="header-section-number">4</span> Functional annotation</a></li>
<li>
<a class="nav-link" href="#parsing-patric-results"><span class="header-section-number">4.1</span> Parsing PATRIC results</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#draw-the-network"><span class="header-section-number">4.1.1</span> Draw the network</a></li></ul>
</li>
<li>
<a class="nav-link" href="#parsing-eggnog-mapper-v2-results"><span class="header-section-number">4.2</span> Parsing eggNOG-mapper v2 results</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#draw-the-network-1"><span class="header-section-number">4.2.1</span> Draw the network</a></li></ul>
</li>
<li>
<a class="nav-link" href="#heatmap-of-the-gene-copy-numbers-with-functional-annotations"><span class="header-section-number">4.3</span> Heatmap of the gene copy numbers with functional annotations</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#midas-1"><span class="header-section-number">4.3.1</span> MIDAS</a></li>
<li><a class="nav-link" href="#the-other-types"><span class="header-section-number">4.3.2</span> The other types</a></li>
</ul>
</li>
<li><a class="nav-link" href="#aggregate-the-gene-copy-numbers-based-on-the-annotations"><span class="header-section-number">4.4</span> Aggregate the gene copy numbers based on the annotations</a></li>
<li>
<a class="nav-link" href="#kgeg-pathway-and-kegg-orthology"><span class="header-section-number">4.5</span> KGEG PATHWAY and KEGG ORTHOLOGY</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#visualization-of-kegg-pathway"><span class="header-section-number">4.5.1</span> Visualization of KEGG PATHWAY</a></li>
<li><a class="nav-link" href="#visualizing-calculated-values-across-species"><span class="header-section-number">4.5.2</span> Visualizing calculated values across species</a></li>
<li><a class="nav-link" href="#show-which-species-have-the-kos"><span class="header-section-number">4.5.3</span> Show which species have the KOs</a></li>
<li><a class="nav-link" href="#ranking-the-components-in-the-graph"><span class="header-section-number">4.5.4</span> Ranking the components in the graph</a></li>
</ul>
</li>
<li><a class="nav-link" href="#setting-the-manual-annotation"><span class="header-section-number">4.6</span> Setting the manual annotation</a></li>
<li><a class="nav-link" href="#pathwaywithfactor"><span class="header-section-number">4.7</span> pathwayWithFactor</a></li>
<li><a class="nav-link" href="#gsea"><span class="header-section-number">4.8</span> GSEA</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/noriakis/stana/blob/master/03-function.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/noriakis/stana/edit/master/03-function.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>stana</strong>" was written by Noriaki Sato. It was last built on 2024-10-22.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
