<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Functional annotation | stana</title>
<meta name="author" content="Noriaki Sato">
<meta name="description" content="For functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for MIDAS and MIDAS2, which profile gene abundances within species by default....">
<meta name="generator" content="bookdown 0.35 with bs4_book()">
<meta property="og:title" content="Chapter 4 Functional annotation | stana">
<meta property="og:type" content="book">
<meta property="og:description" content="For functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for MIDAS and MIDAS2, which profile gene abundances within species by default....">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Functional annotation | stana">
<meta name="twitter:description" content="For functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for MIDAS and MIDAS2, which profile gene abundances within species by default....">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.29/datatables.js"></script><link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">stana</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="importing.html"><span class="header-section-number">2</span> Importing</a></li>
<li><a class="" href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></li>
<li><a class="active" href="function.html"><span class="header-section-number">4</span> Functional annotation</a></li>
<li><a class="" href="visualization.html"><span class="header-section-number">5</span> Visualization</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/noriakis/stana">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="function" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Functional annotation<a class="anchor" aria-label="anchor" href="#function"><i class="fas fa-link"></i></a>
</h1>
<p>For functional profiling, several functions are prepared in <code>stana</code>. Note that these functions are primarily designed for <code>MIDAS</code> and <code>MIDAS2</code>, which profile gene abundances within species by default.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
<div id="parsing-patric-results" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Parsing <code>PATRIC</code> results<a class="anchor" aria-label="anchor" href="#parsing-patric-results"><i class="fas fa-link"></i></a>
</h2>
<p>Use <code>checkPATRIC</code> function to obtain information related to <code>PATRIC</code> functional annotation. The function accepts the input of named list of genes, and returns functional annotation results. The function uses <code>BiocFileCache()</code> to cache the obtained results from API.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"test"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"../test_genes.txt"</span><span class="op">)</span><span class="op">$</span><span class="va">V1</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; $test</span></span>
<span><span class="co">#&gt;  [1] "1280701.3.peg.1153" "1280701.3.peg.1169"</span></span>
<span><span class="co">#&gt;  [3] "1280701.3.peg.1174" "1280701.3.peg.1179"</span></span>
<span><span class="co">#&gt;  [5] "1280701.3.peg.1186" "1280701.3.peg.203" </span></span>
<span><span class="co">#&gt;  [7] "1280701.3.peg.361"  "1280701.3.peg.363" </span></span>
<span><span class="co">#&gt;  [9] "1280701.3.peg.368"  "1280701.3.peg.369" </span></span>
<span><span class="co">#&gt; [11] "1280701.3.peg.570"  "1280701.3.peg.758" </span></span>
<span><span class="co">#&gt; [13] "1280701.3.peg.762"  "1410605.3.peg.1201"</span></span>
<span><span class="co">#&gt; [15] "1410605.3.peg.1204" "1410605.3.peg.1463"</span></span>
<span><span class="co">#&gt; [17] "1410605.3.peg.1510" "1410605.3.peg.1568"</span></span>
<span><span class="co">#&gt; [19] "1410605.3.peg.450"  "1410605.3.peg.593" </span></span>
<span><span class="co">#&gt; [21] "1410605.3.peg.655"  "1410605.3.peg.833" </span></span>
<span><span class="co">#&gt; [23] "1410605.3.peg.844"  "1410605.3.peg.903" </span></span>
<span><span class="co">#&gt; [25] "1410605.3.peg.953"  "1447715.5.peg.1455"</span></span>
<span><span class="co">#&gt; [27] "1447715.5.peg.1549" "1447715.5.peg.1582"</span></span>
<span><span class="co">#&gt; [29] "1447715.5.peg.256"  "1447715.5.peg.33"  </span></span>
<span><span class="co">#&gt; [31] "1447715.5.peg.566"  "1447715.5.peg.964"</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/checkPATRIC.html">checkPATRIC</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Obtaining annotations of 3 genomes</span></span>
<span><span class="co">#&gt;   Obtaining information on 1280701.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1410605.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1447715.5</span></span>
<span><span class="co">#&gt; Checking results on cluster test</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained after removal of duplication</span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">test</span><span class="op">$</span><span class="va">DF</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scrollX<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-6f2900212cf6e7678209" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6f2900212cf6e7678209">{"x":{"filter":"none","vertical":false,"data":[["254","608"],["fig|1280701.3.peg.570","fig|1280701.3.peg.1186"],["4.2.1.51","2.1.1.37"],["Prephenate dehydratase","DNA (cytosine-5-)-methyltransferase"],[400,270],["Phenylalanine, tyrosine and tryptophan biosynthesis","Cysteine and methionine metabolism"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patric_id<\/th>\n      <th>ec_number<\/th>\n      <th>ec_description<\/th>\n      <th>pathway_id<\/th>\n      <th>pathway_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div id="draw-the-network" class="section level3" number="4.1.1">
<h3>
<span class="header-section-number">4.1.1</span> Draw the network<a class="anchor" aria-label="anchor" href="#draw-the-network"><i class="fas fa-link"></i></a>
</h3>
<p>You can draw the graph of obtained results depicting enzyme to KEGG PATHWAY relationship.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/drawPATRIC.html">drawPATRIC</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; Obtaining annotations of 3 genomes</span></span>
<span><span class="co">#&gt;   Obtaining information on 1280701.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1410605.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1447715.5</span></span>
<span><span class="co">#&gt; Checking results on cluster test</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on ec_description</span></span>
<span><span class="co">#&gt;   total of 2 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Making graph on pathway_name and ec_description</span></span>
<span><span class="co">#&gt;   subsetting to 5 label on each category</span></span>
<span><span class="co">#&gt; $test</span></span>
<span><span class="co">#&gt; $test$DF</span></span>
<span><span class="co">#&gt;                  patric_id ec_number</span></span>
<span><span class="co">#&gt; 254  fig|1280701.3.peg.570  4.2.1.51</span></span>
<span><span class="co">#&gt; 608 fig|1280701.3.peg.1186  2.1.1.37</span></span>
<span><span class="co">#&gt;                          ec_description pathway_id</span></span>
<span><span class="co">#&gt; 254              Prephenate dehydratase        400</span></span>
<span><span class="co">#&gt; 608 DNA (cytosine-5-)-methyltransferase        270</span></span>
<span><span class="co">#&gt;                                            pathway_name</span></span>
<span><span class="co">#&gt; 254 Phenylalanine, tyrosine and tryptophan biosynthesis</span></span>
<span><span class="co">#&gt; 608                  Cysteine and methionine metabolism</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$REMOVEDUP</span></span>
<span><span class="co">#&gt;                  patric_id ec_number</span></span>
<span><span class="co">#&gt; 254  fig|1280701.3.peg.570  4.2.1.51</span></span>
<span><span class="co">#&gt; 608 fig|1280701.3.peg.1186  2.1.1.37</span></span>
<span><span class="co">#&gt;                          ec_description pathway_id</span></span>
<span><span class="co">#&gt; 254              Prephenate dehydratase        400</span></span>
<span><span class="co">#&gt; 608 DNA (cytosine-5-)-methyltransferase        270</span></span>
<span><span class="co">#&gt;                                            pathway_name</span></span>
<span><span class="co">#&gt; 254 Phenylalanine, tyrosine and tryptophan biosynthesis</span></span>
<span><span class="co">#&gt; 608                  Cysteine and methionine metabolism</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$SORTED</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DNA (cytosine-5-)-methyltransferase </span></span>
<span><span class="co">#&gt;                                   1 </span></span>
<span><span class="co">#&gt;              Prephenate dehydratase </span></span>
<span><span class="co">#&gt;                                   1 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$GRAPH</span></span>
<span><span class="co">#&gt; IGRAPH 6a2e1c0 UN-- 4 2 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c)</span></span>
<span><span class="co">#&gt; + edges from 6a2e1c0 (vertex names):</span></span>
<span><span class="co">#&gt; [1] Prephenate dehydratase             --Phenylalanine, tyrosine and tryptophan biosynthesis</span></span>
<span><span class="co">#&gt; [2] DNA (cytosine-5-)-methyltransferase--Cysteine and methionine metabolism                 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $test$PLOT</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/drawpat-1.png" width="672"></div>
</div>
</div>
<div id="parsing-eggnog-mapper-v2-results" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Parsing <code>eggNOG-mapper v2</code> results<a class="anchor" aria-label="anchor" href="#parsing-eggnog-mapper-v2-results"><i class="fas fa-link"></i></a>
</h2>
<p>Use <code>checkEGGNOG</code> function to read the output of eggNOG-mapper v2. Specify IDs you want to obtain to <code>ret</code>, such as “KEGG_ko” and “KEGG_Pathway”.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/checkEGGNOG.html">checkEGGNOG</a></span><span class="op">(</span><span class="st">"../annotations_gtdb/100224_eggnog_out.emapper.annotations"</span>,</span>
<span>    ret<span class="op">=</span><span class="st">"KEGG_ko"</span><span class="op">)</span></span>
<span><span class="va">tib</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-614ab56a64c46055ba66" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-614ab56a64c46055ba66">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["GCF_002846775.1_00408","GCF_002846815.1_01743","GCF_004156145.1_01406","GCF_004155565.1_00557","GCF_004155645.1_00353","GCF_000800475.2_00338"],["KEGG_ko","KEGG_ko","KEGG_ko","KEGG_ko","KEGG_ko","KEGG_ko"],["ko:K11533","ko:K11533","ko:K11533","ko:K11533","ko:K11533","ko:K11533"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>name<\/th>\n      <th>value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div id="draw-the-network-1" class="section level3" number="4.2.1">
<h3>
<span class="header-section-number">4.2.1</span> Draw the network<a class="anchor" aria-label="anchor" href="#draw-the-network-1"><i class="fas fa-link"></i></a>
</h3>
<p>You can draw the relationships between IDs by <code>drawEGGNOG</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/drawEGGNOG.html">drawEGGNOG</a></span><span class="op">(</span><span class="st">"../annotations_gtdb/100224_eggnog_out.emapper.annotations"</span>,</span>
<span>            candPlot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"KEGG_ko"</span>,<span class="st">"KEGG_Pathway"</span><span class="op">)</span>,</span>
<span>            geneIDs <span class="op">=</span> <span class="va">tib</span><span class="op">$</span><span class="va">ID</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 4,287 × 3</span></span>
<span><span class="co">#&gt;    ID                    name          value                </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                 &lt;chr&gt;         &lt;chr&gt;                </span></span>
<span><span class="co">#&gt;  1 GCF_002846775.1_00408 seed_ortholog 1690.BPSG_1412       </span></span>
<span><span class="co">#&gt;  2 GCF_002846775.1_00408 eggNOG_OGs    COG0304@1|root       </span></span>
<span><span class="co">#&gt;  3 GCF_002846775.1_00408 eggNOG_OGs    COG0331@1|root       </span></span>
<span><span class="co">#&gt;  4 GCF_002846775.1_00408 eggNOG_OGs    COG2030@1|root       </span></span>
<span><span class="co">#&gt;  5 GCF_002846775.1_00408 eggNOG_OGs    COG4981@1|root       </span></span>
<span><span class="co">#&gt;  6 GCF_002846775.1_00408 eggNOG_OGs    COG0304@2|Bacteria   </span></span>
<span><span class="co">#&gt;  7 GCF_002846775.1_00408 eggNOG_OGs    COG0331@2|Bacteria   </span></span>
<span><span class="co">#&gt;  8 GCF_002846775.1_00408 eggNOG_OGs    COG2030@2|Bacteria   </span></span>
<span><span class="co">#&gt;  9 GCF_002846775.1_00408 eggNOG_OGs    COG4981@2|Bacteria   </span></span>
<span><span class="co">#&gt; 10 GCF_002846775.1_00408 eggNOG_OGs    2GIY4@201174|Actinob…</span></span>
<span><span class="co">#&gt; # ℹ 4,277 more rows</span></span>
<span><span class="co">#&gt; $graph</span></span>
<span><span class="co">#&gt; IGRAPH 6bacaf2 UN-- 21 922 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), category (v/c), size (v/n)</span></span>
<span><span class="co">#&gt; + edges from 6bacaf2 (vertex names):</span></span>
<span><span class="co">#&gt;  [1] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt;  [3] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt;  [5] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt;  [7] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt;  [9] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt; [11] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt; [13] ko:K11533--ko00061 ko:K11533--ko01100</span></span>
<span><span class="co">#&gt; [15] ko:K11533--ko01212 ko:K11533--ko04931</span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $plot</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/drawEGGNOG-1.png" width="672"></div>
</div>
</div>
<div id="heatmap-of-the-gene-abundances-with-functional-annotations" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Heatmap of the gene abundances with functional annotations<a class="anchor" aria-label="anchor" href="#heatmap-of-the-gene-abundances-with-functional-annotations"><i class="fas fa-link"></i></a>
</h2>
<p>You can inspect the overview of functional differences using gene abundances along with <code>simplifyEnrichment</code>. The <code>plotHeatmap</code> function can be used with the stana object or preprocessed gene abundance matrix as input. In the function, <code>anno_PATRIC_keywords</code> and <code>anno_eggNOG_keywords</code> are used to plot the word clouds alongside <code>Heatmap</code> from <code>ComplexHeatmap</code>. It is easy for <code>MIDAS</code>, as the function obtains functional annotation from PATRIC API server and no annotation step is needed.</p>
<div id="midas-1" class="section level3" number="4.3.1">
<h3>
<span class="header-section-number">4.3.1</span> <code>MIDAS</code><a class="anchor" aria-label="anchor" href="#midas-1"><i class="fas fa-link"></i></a>
</h3>
<p>For <code>MIDAS</code>, the function automatically query API of <code>PATRIC</code> server using the gene names. As the gene number is large typically, one can filter the genes by options <code>filter_zero_frac</code>, <code>filter_max_frac</code> and <code>filter_max_value</code>. However, one should perform own filtering beforehand and provide the matrix to <code>mat</code>. If <code>mat</code> is specified, other filtering options will be ignored.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simplifyEnrichment">simplifyEnrichment</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS.html">loadMIDAS</a></span><span class="op">(</span><span class="st">"../merge_midas1/"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="st">"Bacteroides_uniformis_57318"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bacteroides_uniformis_57318</span></span>
<span><span class="co">#&gt;   Snps</span></span>
<span><span class="co">#&gt;     HC 13</span></span>
<span><span class="co">#&gt;     R 16</span></span>
<span><span class="co">#&gt;     Bacteroides_uniformis_57318 cleared filtering threshold in SNV</span></span>
<span><span class="co">#&gt;   Genes</span></span>
<span><span class="co">#&gt;     HC 13</span></span>
<span><span class="co">#&gt;     R 16</span></span>
<span><span class="co">#&gt;     Bacteroides_uniformis_57318 cleared filtering threshold in genes</span></span>
<span><span class="co">#&gt; Overall, 1 species met criteria in SNPs</span></span>
<span><span class="co">#&gt; Overall, 1 species met criteria in genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"Bacteroides_uniformis_57318"</span>,filter_max_value <span class="op">=</span> <span class="fl">2</span>,filter_max_frac <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">#&gt; In resulting matrix, max: 1.99955229433902, min: 0</span></span>
<span><span class="co">#&gt; Dimension: 4911, 29</span></span>
<span><span class="co">#&gt; Obtaining gene information from PATRIC server</span></span>
<span><span class="co">#&gt; Obtaining annotations of 7 genomes</span></span>
<span><span class="co">#&gt;   Obtaining information on 1235787.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 1339348.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 411479.10</span></span>
<span><span class="co">#&gt;   Obtaining information on 457393.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 585543.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 997889.3</span></span>
<span><span class="co">#&gt;   Obtaining information on 997890.3</span></span>
<span><span class="co">#&gt; Checking results on cluster 1</span></span>
<span><span class="co">#&gt;   total of 273 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 261 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 2</span></span>
<span><span class="co">#&gt;   total of 334 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 329 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 3</span></span>
<span><span class="co">#&gt;   total of 693 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 662 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 4</span></span>
<span><span class="co">#&gt;   total of 175 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 174 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 5</span></span>
<span><span class="co">#&gt;   total of 310 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 303 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 6</span></span>
<span><span class="co">#&gt;   total of 898 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 868 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 7</span></span>
<span><span class="co">#&gt;   total of 521 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 503 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 8</span></span>
<span><span class="co">#&gt;   total of 701 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 675 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 9</span></span>
<span><span class="co">#&gt;   total of 1140 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 1105 annotation obtained after removal of duplication</span></span>
<span><span class="co">#&gt; Checking results on cluster 10</span></span>
<span><span class="co">#&gt;   total of 1090 annotation obtained</span></span>
<span><span class="co">#&gt;   remove duplicate based on pathway_name</span></span>
<span><span class="co">#&gt;   total of 1053 annotation obtained after removal of duplication</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/MIDAS1_heatmap-1.png" width="672"></div>
</div>
<div id="midas2-1" class="section level3" number="4.3.2">
<h3>
<span class="header-section-number">4.3.2</span> <code>MIDAS2</code><a class="anchor" aria-label="anchor" href="#midas2-1"><i class="fas fa-link"></i></a>
</h3>
<p>For <code>MIDAS2</code>, the users should provide eggNOG annotation on <code>eggNOG</code> slot of stana object. <code>fnc</code> argument accepts <code>KEGG_Pathway</code> or <code>KEGG_Module</code> available in eggNOG annotation. The function queries <code>KEGG REST API</code> to obtain pathway and module description.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">taxtbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"../metadata_uhgg.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span>,</span>
<span>                     header<span class="op">=</span><span class="fl">1</span>, row.names<span class="op">=</span><span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS2.html">loadMIDAS2</a></span><span class="op">(</span><span class="st">"../merge_uhgg"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">)</span>, taxtbl<span class="op">=</span><span class="va">taxtbl</span>, db<span class="op">=</span><span class="st">"uhgg"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SNPS</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of snps: 70178</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 12</span></span>
<span><span class="co">#&gt;       Number of samples in R: 16</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt; Genes</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of genes: 120158</span></span>
<span><span class="co">#&gt;     Number of samples: 31</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 13</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span></code></pre></div>
<p>We set the eggNOG-mapper v2 annotation file to the eggNOG slot of stana object.
This can be done by <code>setAnnotation</code> function.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set the annotation file</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setAnnotation.html">setAnnotation</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">=</span><span class="st">"../annotations_uhgg/101346_eggnog_out.emapper.annotations"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>You can set <code>removeAdditional</code> argument to filter words that are to be displayed.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simplifyEnrichment">simplifyEnrichment</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"101346"</span>,</span>
<span>    fnc<span class="op">=</span><span class="st">"KEGG_Module"</span>,</span>
<span>    removeAdditional<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cycle"</span>,<span class="st">"pathway"</span><span class="op">)</span>,</span>
<span>    filter_zero_frac <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    filter_max_frac <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    filter_max_value <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; In resulting matrix, max: 4.999739, min: 0</span></span>
<span><span class="co">#&gt; Dimension: 3501, 31</span></span>
<span><span class="co">#&gt; MIDAS2, looking for the annotation file by eggNOG-mapper v2</span></span>
<span><span class="co">#&gt; Loading annotation</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/MIDAS2_heatmap-1.png" width="672"></div>
</div>
</div>
<div id="kgeg-pathway-and-kegg-orthology" class="section level2" number="4.4">
<h2>
<span class="header-section-number">4.4</span> KGEG PATHWAY and KEGG ORTHOLOGY<a class="anchor" aria-label="anchor" href="#kgeg-pathway-and-kegg-orthology"><i class="fas fa-link"></i></a>
</h2>
<div id="visualization-of-kegg-pathway" class="section level3" number="4.4.1">
<h3>
<span class="header-section-number">4.4.1</span> Visualization of KEGG PATHWAY<a class="anchor" aria-label="anchor" href="#visualization-of-kegg-pathway"><i class="fas fa-link"></i></a>
</h3>
<p><code>KEGG PATHWAY</code> is frequently used to characterize metabolic function of microbiome. Utilizing <a href="https://github.com/noriakis/ggkegg"><code>ggkegg</code></a>, the information of intra-species diversity, particulary gene abundance differences, can be reflected onto <code>KEGG PATHWAY</code>. It needs <code>genes</code> slot filled, and some annotations files, typically <code>eggNOG-mapper v2</code>, are needed.</p>
<div id="visualizing-differences-per-species" class="section level4" number="4.4.1.1">
<h4>
<span class="header-section-number">4.4.1.1</span> Visualizing differences per species<a class="anchor" aria-label="anchor" href="#visualizing-differences-per-species"><i class="fas fa-link"></i></a>
</h4>
<p>Load the profile for multiple species.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">taxtbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"../metadata_uhgg.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span>,</span>
<span>                     header<span class="op">=</span><span class="fl">1</span>, row.names<span class="op">=</span><span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS2.html">loadMIDAS2</a></span><span class="op">(</span><span class="st">"../merge_uhgg"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, taxtbl<span class="op">=</span><span class="va">taxtbl</span>, db<span class="op">=</span><span class="st">"uhgg"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SNPS</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of snps: 70178</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 12</span></span>
<span><span class="co">#&gt;       Number of samples in R: 16</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt;   102438</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis</span></span>
<span><span class="co">#&gt;     Number of snps: 18102</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 10</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt; Genes</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of genes: 120158</span></span>
<span><span class="co">#&gt;     Number of samples: 31</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 13</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt;   102438</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis</span></span>
<span><span class="co">#&gt;     Number of genes: 47046</span></span>
<span><span class="co">#&gt;     Number of samples: 29</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 11</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span></code></pre></div>
<p>Next, we set the eggNOG-mapper v2 annotation file to the eggNOG slot of stana object.
This way, the <code>plotKEGGPathway</code> function automatically calculates the abundance by user-defined method.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set the annotation file</span></span>
<span><span class="va">stana</span><span class="op">@</span><span class="va">eggNOG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">=</span><span class="st">"../annotations_uhgg/101346_eggnog_out.emapper.annotations"</span>,</span>
<span>                     <span class="st">"102438"</span><span class="op">=</span><span class="st">"../annotations_uhgg/102438_eggnog_out.emapper.annotations"</span><span class="op">)</span></span></code></pre></div>
<p>plotKEGGPathway can be run by providing stana object, species (when multiple species, rectangular nodes will be split),
and pathway ID to visualize. Here, we visualize <code>ko00620</code>, Pyruvate metabolism for example. As for the large annotation table, the calculation takes time and you can provide pre-calculated KO table in <code>kos</code> slot of stana object, or specify <code>only_ko = TRUE</code> to first return KO table.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, only_ko<span class="op">=</span><span class="cn">TRUE</span>, multi_scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, multi_scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/onescale-1.png" width="672"></div>
<p>By default, the scale is same. If you install <code>ggh4x</code>, multiple scales can be added, by specifying <code>multi_scale</code> argument.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, multi_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/multscale-1.png" width="672"></div>
<p>You can provide multiple pathway IDs to pathway_id, which returns a list of plot.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span>,<span class="st">"ko00620"</span><span class="op">)</span>,</span>
<span>                      multi_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg2</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span>
<span><span class="va">gg2</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway-1.png" width="672"></div>
<p>In this way, differences in orthologies in the pathway across multiple species can be readily captured.</p>
</div>
</div>
<div id="visualizing-calculated-values-across-species" class="section level3" number="4.4.2">
<h3>
<span class="header-section-number">4.4.2</span> Visualizing calculated values across species<a class="anchor" aria-label="anchor" href="#visualizing-calculated-values-across-species"><i class="fas fa-link"></i></a>
</h3>
<p>If you want to see the sum values across species, you can set option <code>summarize=TRUE</code>. This way, the KO values across specified species are summed, and compared between groups, then plotted.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span>,<span class="st">"ko00620"</span><span class="op">)</span>,</span>
<span>                      summarize<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg2</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span>
<span><span class="va">gg2</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway_SUM-1.png" width="672"></div>
</div>
<div id="show-which-species-have-the-kos" class="section level3" number="4.4.3">
<h3>
<span class="header-section-number">4.4.3</span> Show which species have the KOs<a class="anchor" aria-label="anchor" href="#show-which-species-have-the-kos"><i class="fas fa-link"></i></a>
</h3>
<p>If you specify <code>point_mode=TRUE</code>, the function plot points on the KOs on the pathways
based on whether the specified species have the corresponding KOs annotated.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span><span class="op">)</span>,</span>
<span>                      sp_colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>,<span class="st">"red"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      point_mode<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="co">#&gt; Point mode enabled</span></span>
<span><span class="va">gg</span></span>
<span><span class="co">#&gt; $ko00270</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway_point-1.png" width="672"></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></div>
<div class="next"><a href="visualization.html"><span class="header-section-number">5</span> Visualization</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#function"><span class="header-section-number">4</span> Functional annotation</a></li>
<li>
<a class="nav-link" href="#parsing-patric-results"><span class="header-section-number">4.1</span> Parsing PATRIC results</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#draw-the-network"><span class="header-section-number">4.1.1</span> Draw the network</a></li></ul>
</li>
<li>
<a class="nav-link" href="#parsing-eggnog-mapper-v2-results"><span class="header-section-number">4.2</span> Parsing eggNOG-mapper v2 results</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#draw-the-network-1"><span class="header-section-number">4.2.1</span> Draw the network</a></li></ul>
</li>
<li>
<a class="nav-link" href="#heatmap-of-the-gene-abundances-with-functional-annotations"><span class="header-section-number">4.3</span> Heatmap of the gene abundances with functional annotations</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#midas-1"><span class="header-section-number">4.3.1</span> MIDAS</a></li>
<li><a class="nav-link" href="#midas2-1"><span class="header-section-number">4.3.2</span> MIDAS2</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#kgeg-pathway-and-kegg-orthology"><span class="header-section-number">4.4</span> KGEG PATHWAY and KEGG ORTHOLOGY</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#visualization-of-kegg-pathway"><span class="header-section-number">4.4.1</span> Visualization of KEGG PATHWAY</a></li>
<li><a class="nav-link" href="#visualizing-calculated-values-across-species"><span class="header-section-number">4.4.2</span> Visualizing calculated values across species</a></li>
<li><a class="nav-link" href="#show-which-species-have-the-kos"><span class="header-section-number">4.4.3</span> Show which species have the KOs</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/noriakis/stana/blob/master/03-function.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/noriakis/stana/edit/master/03-function.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>stana</strong>" was written by Noriaki Sato. It was last built on 2023-10-25.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
