<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Functional annotation | stana</title>
<meta name="author" content="Noriaki Sato">
<meta name="description" content="For functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for MIDAS and MIDAS2, which profile gene abundances within species by default....">
<meta name="generator" content="bookdown 0.33 with bs4_book()">
<meta property="og:title" content="Chapter 4 Functional annotation | stana">
<meta property="og:type" content="book">
<meta property="og:description" content="For functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for MIDAS and MIDAS2, which profile gene abundances within species by default....">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Functional annotation | stana">
<meta name="twitter:description" content="For functional profiling, several functions are prepared in stana. Note that these functions are primarily designed for MIDAS and MIDAS2, which profile gene abundances within species by default....">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">stana</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="importing.html"><span class="header-section-number">2</span> Importing</a></li>
<li><a class="" href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></li>
<li><a class="active" href="function.html"><span class="header-section-number">4</span> Functional annotation</a></li>
<li><a class="" href="visualization.html"><span class="header-section-number">5</span> Visualization</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/noriakis/stana">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="function" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Functional annotation<a class="anchor" aria-label="anchor" href="#function"><i class="fas fa-link"></i></a>
</h1>
<p>For functional profiling, several functions are prepared in <code>stana</code>. Note that these functions are primarily designed for <code>MIDAS</code> and <code>MIDAS2</code>, which profile gene abundances within species by default.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
<div id="parsing-patric-results" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Parsing <code>PATRIC</code> results<a class="anchor" aria-label="anchor" href="#parsing-patric-results"><i class="fas fa-link"></i></a>
</h2>
</div>
<div id="parsing-eggnog-mapper-v2-results" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Parsing <code>eggNOG-mapper v2</code> results<a class="anchor" aria-label="anchor" href="#parsing-eggnog-mapper-v2-results"><i class="fas fa-link"></i></a>
</h2>
<div id="visualizing-function-network" class="section level3" number="4.2.1">
<h3>
<span class="header-section-number">4.2.1</span> Visualizing function network<a class="anchor" aria-label="anchor" href="#visualizing-function-network"><i class="fas fa-link"></i></a>
</h3>
</div>
</div>
<div id="kgeg-pathway-and-kegg-orthology" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> KGEG PATHWAY and KEGG ORTHOLOGY<a class="anchor" aria-label="anchor" href="#kgeg-pathway-and-kegg-orthology"><i class="fas fa-link"></i></a>
</h2>
<div id="visualization-of-kegg-pathway" class="section level3" number="4.3.1">
<h3>
<span class="header-section-number">4.3.1</span> Visualization of KEGG PATHWAY<a class="anchor" aria-label="anchor" href="#visualization-of-kegg-pathway"><i class="fas fa-link"></i></a>
</h3>
<p><code>KEGG PATHWAY</code> is frequently used to characterize metabolic function of microbiome. Utilizing <a href="https://github.com/noriakis/ggkegg"><code>ggkegg</code></a>, the information of intra-species diversity, particulary gene abundance differences, can be reflected onto <code>KEGG PATHWAY</code>.</p>
<div id="visualizing-differences-per-species" class="section level4" number="4.3.1.1">
<h4>
<span class="header-section-number">4.3.1.1</span> Visualizing differences per species<a class="anchor" aria-label="anchor" href="#visualizing-differences-per-species"><i class="fas fa-link"></i></a>
</h4>
<p>Load the profile for multiple species.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../hd_meta.rda"</span><span class="op">)</span></span>
<span><span class="va">taxtbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"../metadata_uhgg.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span>,</span>
<span>                     header<span class="op">=</span><span class="fl">1</span>, row.names<span class="op">=</span><span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/loadMIDAS2.html">loadMIDAS2</a></span><span class="op">(</span><span class="st">"../merge_uhgg"</span>, cl<span class="op">=</span><span class="va">hd_meta</span>, candSp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, taxtbl<span class="op">=</span><span class="va">taxtbl</span>, db<span class="op">=</span><span class="st">"uhgg"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SNPS</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of snps: 70178</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 12</span></span>
<span><span class="co">#&gt;       Number of samples in R: 16</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt;   102438</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis</span></span>
<span><span class="co">#&gt;     Number of snps: 18102</span></span>
<span><span class="co">#&gt;     Number of samples: 28</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 10</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt; Genes</span></span>
<span><span class="co">#&gt;   101346</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis</span></span>
<span><span class="co">#&gt;     Number of genes: 120158</span></span>
<span><span class="co">#&gt;     Number of samples: 31</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 13</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span>
<span><span class="co">#&gt;   102438</span></span>
<span><span class="co">#&gt;   d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis</span></span>
<span><span class="co">#&gt;     Number of genes: 47046</span></span>
<span><span class="co">#&gt;     Number of samples: 29</span></span>
<span><span class="co">#&gt;       Number of samples in HC: 11</span></span>
<span><span class="co">#&gt;       Number of samples in R: 18</span></span>
<span><span class="co">#&gt;       Passed the filter</span></span></code></pre></div>
<p>Next, we set the eggNOG-mapper v2 annotation file to the eggNOG slot of stana object.
This way, the <code>plotKEGGPathway</code> function automatically calculates the abundance by user-defined method.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set the annotation file</span></span>
<span><span class="va">stana</span><span class="op">@</span><span class="va">eggNOG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"101346"</span><span class="op">=</span><span class="st">"../annotations_uhgg/101346_eggnog_out.emapper.annotations"</span>,</span>
<span>                     <span class="st">"102438"</span><span class="op">=</span><span class="st">"../annotations_uhgg/102438_eggnog_out.emapper.annotations"</span><span class="op">)</span></span></code></pre></div>
<p>plotKEGGPathway can be run by providing stana object, species (when multiple species, rectangular nodes will be split),
and pathway ID to visualize. Here, we visualize <code>ko00620</code>, Pyruvate metabolism for example. As for the large annotation table, the calculation takes time and you can provide pre-calculated KO table in <code>kos</code> slot of stana object, or specify <code>only_ko = TRUE</code> to first return KO table.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, only_ko<span class="op">=</span><span class="cn">TRUE</span>, multi_scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, multi_scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/onescale-1.svg" width="672"></div>
<p>By default, the scale is same. If you install <code>ggh4x</code>, multiple scales can be added, by specifying <code>multi_scale</code> argument.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>, pathway_id<span class="op">=</span><span class="st">"ko00620"</span>, multi_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/multscale-1.svg" width="672"></div>
<p>You can provide multiple pathway IDs to pathway_id, which returns a list of plot.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotKEGGPathway.html">plotKEGGPathway</a></span><span class="op">(</span><span class="va">stana</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"101346"</span>,<span class="st">"102438"</span><span class="op">)</span>,</span>
<span>                      pathway_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ko00270"</span>,<span class="st">"ko00620"</span><span class="op">)</span>,</span>
<span>                      multi_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; Using pre-computed KO table</span></span>
<span><span class="co">#&gt; 101346: HC / R</span></span>
<span><span class="co">#&gt; 102438: HC / R</span></span>
<span><span class="va">gg2</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span>
<span><span class="va">gg2</span></span></code></pre></div>
<div class="inline-figure"><img src="03-function_files/figure-html/pathway-1.svg" width="672"></div>
<p>In this way, differences in orthologies in the pathway across multiple species can be readily captured.</p>

</div>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></div>
<div class="next"><a href="visualization.html"><span class="header-section-number">5</span> Visualization</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#function"><span class="header-section-number">4</span> Functional annotation</a></li>
<li><a class="nav-link" href="#parsing-patric-results"><span class="header-section-number">4.1</span> Parsing PATRIC results</a></li>
<li>
<a class="nav-link" href="#parsing-eggnog-mapper-v2-results"><span class="header-section-number">4.2</span> Parsing eggNOG-mapper v2 results</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#visualizing-function-network"><span class="header-section-number">4.2.1</span> Visualizing function network</a></li></ul>
</li>
<li>
<a class="nav-link" href="#kgeg-pathway-and-kegg-orthology"><span class="header-section-number">4.3</span> KGEG PATHWAY and KEGG ORTHOLOGY</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#visualization-of-kegg-pathway"><span class="header-section-number">4.3.1</span> Visualization of KEGG PATHWAY</a></li></ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/noriakis/stana/blob/master/03-function.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/noriakis/stana/edit/master/03-function.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>stana</strong>" was written by Noriaki Sato. It was last built on 2023-07-18.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
