<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Downstream analysis toolkit for strain-level metagenomics.">
<title>Strain-level metagenomic analysis in R â€¢ stana</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Strain-level metagenomic analysis in R">
<meta property="og:description" content="Downstream analysis toolkit for strain-level metagenomics.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">stana</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/usage_of_stana.html">stana</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="stana">stana<a class="anchor" aria-label="anchor" href="#stana"></a>
</h1></div>
<p><a href="https://github.com/noriakis/stana/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/noriakis/stana/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a> <a href="https://www.codefactor.io/repository/github/noriakis/stana" class="external-link"><img src="https://www.codefactor.io/repository/github/noriakis/stana/badge" alt="CodeFactor"></a></p>
<p>Strain-level metagenomic analysis in R. Import and analyse, plot output of the software like <a href="https://github.com/snayfach/MIDAS" class="external-link">MIDAS</a>, <a href="https://github.com/czbiohub/MIDAS2" class="external-link">MIDAS2</a>, <a href="https://github.com/metasnv-tool/metaSNV" class="external-link">metaSNV v1, metaSNV v2</a> and <a href="https://github.com/MrOlm/inStrain" class="external-link">inStrain</a>. Will grow into the complete package.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"noriakis/stana"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="databases">Databases<a class="anchor" aria-label="anchor" href="#databases"></a>
</h2>
<ul>
<li>
<a href="https://github.com/snayfach/MIDAS/blob/master/docs/ref_db.md" class="external-link">midas_db_v1.2</a>: Contains 31,007 bacterial reference genomes clustered into 5,952 species groups.</li>
<li>UHGG (in MIDAS2): Contains 286,997 genomes clustered into 4,644 species (<strong>from human stool samples</strong>).</li>
<li>GTDB (in MIDAS2): Contains 258,406 genomes clustered into 45,555 bacterial and 2,339 archaeal species.</li>
<li>proGenomes2, proGenomes3</li>
</ul>
</div>
<div class="section level2">
<h2 id="example-analysis">Example analysis<a class="anchor" aria-label="anchor" href="#example-analysis"></a>
</h2>
<p>The function includes filtering of species based on clinical / environmental variables interested, calling of consensus sequencing, constructing the tree, plotting the results based on groups, and functional annotations. The below example shows the analysis of the subset of <code>PRJEB9584</code>, sequenced by HiSeq 2000. Packages including <a href="https://github.com/YuLab-SMU/ggtree" class="external-link"><code>ggtree</code></a>, <a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link"><code>ComplexHeatmap</code></a>, <a href="https://github.com/jokergoo/simplifyEnrichment" class="external-link"><code>simplyfyEnrichment</code></a>, <a href="https://github.com/KlausVigo/phangorn" class="external-link"><code>phangorn</code></a>, <a href="https://github.com/jokergoo/circlize" class="external-link"><code>circlize</code></a>, and <a href="https://github.com/thomasp85/ggraph" class="external-link"><code>ggraph</code></a> are used.</p>
<p><img src="https://github.com/noriakis/software/blob/main/images/stana_example.png?raw=true" width="800px"></p>
</div>
<div class="section level2">
<h2 id="todo">TODO<a class="anchor" aria-label="anchor" href="#todo"></a>
</h2>
<ul>
<li>Read all data in S4 class, and perform the downstream analysis. Currently the package reads the data per function.</li>
<li>Add more filters and tree inferring options in <code><a href="reference/consensusSeq.html">consensusSeq()</a></code>.</li>
<li>Add reasonable filtering options to <code>loadInStrain</code> and <code>loadmetaSNV</code>
</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing stana</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Noriaki Sato <br><small class="roles"> Maintainer, author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
