<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 Statistial analysis | stana</title>
<meta name="author" content="Noriaki Sato">
<meta name="description" content="stana provides functions to perform statistical analysis on the metagenotyping results based on loaded data. library(stana) library(phangorn) library(ggtree) ## Examine sample object...">
<meta name="generator" content="bookdown 0.35 with bs4_book()">
<meta property="og:title" content="Chapter 3 Statistial analysis | stana">
<meta property="og:type" content="book">
<meta property="og:description" content="stana provides functions to perform statistical analysis on the metagenotyping results based on loaded data. library(stana) library(phangorn) library(ggtree) ## Examine sample object...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 Statistial analysis | stana">
<meta name="twitter:description" content="stana provides functions to perform statistical analysis on the metagenotyping results based on loaded data. library(stana) library(phangorn) library(ggtree) ## Examine sample object...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.29/datatables.js"></script><link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">stana</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="importing-and-filtering.html"><span class="header-section-number">2</span> Importing and filtering</a></li>
<li><a class="active" href="statistial-analysis.html"><span class="header-section-number">3</span> Statistial analysis</a></li>
<li><a class="" href="function.html"><span class="header-section-number">4</span> Functional annotation</a></li>
<li><a class="" href="visualization.html"><span class="header-section-number">5</span> Visualization</a></li>
<li><a class="" href="interactive-inspection.html"><span class="header-section-number">6</span> Interactive inspection</a></li>
<li><a class="" href="analysis-example-of-gut-microbiome-of-esrd-patients.html"><span class="header-section-number">7</span> Analysis example of gut microbiome of ESRD patients</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/noriakis/stana">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="statistial-analysis" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Statistial analysis<a class="anchor" aria-label="anchor" href="#statistial-analysis"><i class="fas fa-link"></i></a>
</h1>
<p><code>stana</code> provides functions to perform statistical analysis on the metagenotyping results based on loaded data.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn">phangorn</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/">ggtree</a></span><span class="op">)</span></span>
<span><span class="co">## Examine sample object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"stana"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="cons" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> Consensus sequence calling<a class="anchor" aria-label="anchor" href="#cons"><i class="fas fa-link"></i></a>
</h2>
<p>The consensus sequence calling can be performed using SNV MAF matrix (or the matrix converted from the allelic counts). It can filter the confident and user-defined positions and output the multiple sequence alignment in <code>fastaList</code> slot. The <code>MIDAS</code> and <code>MIDAS2</code> output provides various statistics of SNVs that can be used for the filtering. The implementation is similar to that in <code>MIDAS</code>. Here we use pair-wise distances of sequences calculated by <code>dist.ml</code> in the default amino acid model (<code>JC69</code>), and performs neighbor-joining tree estimation by <code>NJ</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## The consensusSeq* is prepared, but consensusSeq function can automatically </span></span>
<span><span class="co">## choose which functions to use</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/consensusseq.html">consensusSeqMIDAS2</a></span><span class="op">(</span><span class="va">stana</span>, species<span class="op">=</span><span class="st">"100003"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Beginning calling for 100003</span></span>
<span><span class="co">#&gt; # Original Site number: 5019</span></span>
<span><span class="co">#&gt; #  Profiled samples: 11</span></span>
<span><span class="co">#&gt; #  Included samples: 11</span></span>
<span></span>
<span><span class="co">## Tree estimation and visualization by `phangorn` and `ggtree`</span></span>
<span><span class="va">dm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/dist.hamming.html">dist.ml</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"fastaList"</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">tre</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/NJ.html">NJ</a></span><span class="op">(</span><span class="va">dm</span><span class="op">)</span></span>
<span><span class="va">tre</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html">groupOTU</a></span><span class="op">(</span><span class="va">tre</span>, <span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"cl"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tre</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>,</span>
<span>             layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"colors"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tp</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/cons-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The sequences are stored as <code>phyDat</code> class object in <code>fastaList</code> slot, the list with the species ID as name.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="va">stana</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/stana/man/consensusseq.html">consensusSeq</a></span><span class="op">(</span>argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>site_prev<span class="op">=</span><span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Beginning calling for 100003</span></span>
<span><span class="co">#&gt; # Original Site number: 5019</span></span>
<span><span class="co">#&gt; #  Profiled samples: 11</span></span>
<span><span class="co">#&gt; #  Included samples: 11</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getFasta.html">getFasta</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; 11 sequences with 4214 character and 2782 different site patterns.</span></span>
<span><span class="co">#&gt; The states are a c g t</span></span></code></pre></div>
<p>Matrix of characters can be returned by <code>return_mat=TRUE</code>. This does not return stana object.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">stana</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/stana/man/consensusseq.html">consensusSeq</a></span><span class="op">(</span>argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>site_prev<span class="op">=</span><span class="fl">0.8</span>, return_mat<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Beginning calling for 100003</span></span>
<span><span class="co">#&gt; # Original Site number: 5019</span></span>
<span><span class="co">#&gt; #  Profiled samples: 11</span></span>
<span><span class="co">#&gt; #  Included samples: 11</span></span>
<span><span class="va">mat</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]   11 4214</span></span></code></pre></div>
<p>You can use the MSA stored in <code>fastaList</code> slot to infer the phylogenetic tree of your choices.
The <code>inferAndPlotTree</code> function can be used to internally infer and plot the tree based on grouping.
<code>dist_method</code> is set to <code>dist.ml</code> by default, and you can pass arguments to the function by <code>tree_args</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn">phangorn</a></span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/inferAndPlotTree.html">inferAndPlotTree</a></span><span class="op">(</span><span class="va">stana</span>, dist_method<span class="op">=</span><span class="st">"dist.hamming"</span>, tree_args<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>exclude<span class="op">=</span><span class="st">"all"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getTree.html">getTree</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Phylogenetic tree with 11 tips and 10 internal nodes.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Tip labels:</span></span>
<span><span class="co">#&gt;   ERR1711593, ERR1711594, ERR1711596, ERR1711598, ERR1711603, ERR1711605, ...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Rooted; includes branch lengths.</span></span></code></pre></div>
<p>Owning to the powerful functions in <code>ggtree</code> and <code>ggtreeExtra</code>, you can visualize the tree based on the metadata.
You should set <code>data.frame</code> containing covariates to stana object by <code>setMetadata</code> function. And specify the covariates to <code>meta</code> argument in <code>plotTree</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Make example metadata</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"snps"</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>    row.names<span class="op">=</span><span class="va">samples</span>,</span>
<span>    treatment<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span>, replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    marker<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Set metadata</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setMetadata.html">setMetadata</a></span><span class="op">(</span><span class="va">stana</span>, <span class="va">metadata</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Call consensus sequence</span></span>
<span><span class="co">## Infer and plot tree based on metadata</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="va">stana</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/stana/man/consensusseq.html">consensusSeq</a></span><span class="op">(</span>argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>site_prev<span class="op">=</span><span class="fl">0.95</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/stana/man/inferAndPlotTree.html">inferAndPlotTree</a></span><span class="op">(</span>meta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"treatment"</span>,<span class="st">"marker"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Beginning calling for 100003</span></span>
<span><span class="co">#&gt; # Original Site number: 5019</span></span>
<span><span class="co">#&gt; #  Profiled samples: 11</span></span>
<span><span class="co">#&gt; #  Included samples: 11</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getFasta.html">getFasta</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; 11 sequences with 896 character and 625 different site patterns.</span></span>
<span><span class="co">#&gt; The states are a c g t</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getTree.html">getTree</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Phylogenetic tree with 11 tips and 10 internal nodes.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Tip labels:</span></span>
<span><span class="co">#&gt;   ERR1711593, ERR1711594, ERR1711596, ERR1711598, ERR1711603, ERR1711605, ...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Rooted; includes branch lengths.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"treePlotList"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/tree_fruit-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The <code>site_list</code> is supported for the usecase such as calling limited to the certain gene regions. The below example calls the MSA in specific gene ID and output plot by <code>ggmsa</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## get snpsInfo slot and filter to variants in specific gene IDs</span></span>
<span><span class="va">cand_ids</span> <span class="op">&lt;-</span> <span class="fu">stana</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"snpsInfo"</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">==</span><span class="st">"UHGG000008_01913"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Call sequence</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/consensusseq.html">consensusSeqMIDAS2</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, site_list<span class="op">=</span><span class="va">cand_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Beginning calling for 100003</span></span>
<span><span class="co">#&gt; # Original Site number: 5019</span></span>
<span><span class="co">#&gt; #  Profiled samples: 11</span></span>
<span><span class="co">#&gt; #  Included samples: 11</span></span>
<span><span class="co">#&gt; # site_list specified: 22</span></span>
<span></span>
<span><span class="co">## Plot</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"ggmsa"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://doi.org/10.1093/bib/bbac222(paper)">ggmsa</a></span><span class="op">)</span></span>
<span>    <span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/read.phyDat.html">write.phyDat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getFasta.html">getFasta</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                           <span class="st">"test.fasta"</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span>
<span>    <span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggmsa/man/ggmsa.html">ggmsa</a></span><span class="op">(</span><span class="st">"test.fasta"</span>,seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> <span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggmsa/man/geom_seqlogo.html">geom_seqlogo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/sitelist-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="nmf" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> Nonnegative matrix factorization<a class="anchor" aria-label="anchor" href="#nmf"><i class="fas fa-link"></i></a>
</h2>
<p>The loaded or calculated matrix can be used for the nonnegative matrix factorization (NMF) for the unsupervised identifications of factors within species. This calculates factor x sample and sample to feature matrix, and possibly finds the pattern for the within-species diversity. The results can be summarized by the functions such as <code>plotAbundanceWithinSpecies</code>.</p>
<p>The input can be SNV, gene, or gene family (KO) abundance table. The matrix can contain <code>NA</code> or <code>-1</code> (zero depth at the position), so filtering should be performed. The <code><a href="https://rdrr.io/pkg/NMF/man/nmf.html">NMF::nmf</a></code> function or <code><a href="https://rdrr.io/pkg/NNLM/man/nnmf.html">NNLM::nnmf</a></code> function can be used for this purpose. By default, <code>estimate</code> is set to FALSE but if set to <code>TRUE</code>, it performs the estimation of rank within <code>estimate_range</code>. This assumes that multiple subspecies are in the samples and is not applicable where only one subspecies should be present. It chooses the rank based on the cophenetic correlation coefficient when the function uses the R package <code>NMF</code>, but the package implements a variety of algorithms and the rank selection method. Also, if <code>NNLM</code>, the rank selection based on cross-validation by randomly assigning the NA in the cell in the matrix, is performed.</p>
<p>The function outputs the related statistics like the proportion of NA or zero value, the relative abundances of factors after the estimation, or the features presented in each factor.</p>
<p>The method is set to <code>snmf/r</code> by default in <code>NMF</code>. For the larger matrix, the <code>NNML</code> function can be used for the faster computation by setting <code>nnlm_flag</code> to TRUE.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://renozao.github.io/NMF/">NMF</a></span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/NMF.html">NMF</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, estimate<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; # NMF started 100003, target: kos, method: snmf/r</span></span>
<span><span class="co">#&gt; # Original features: 20</span></span>
<span><span class="co">#&gt; # Original samples: 16</span></span>
<span><span class="co">#&gt; # Original matrix NA: NA</span></span>
<span><span class="co">#&gt; # Original matrix zero: 0.781</span></span>
<span><span class="co">#&gt; # Selecting KL loss</span></span>
<span><span class="co">#&gt; # Filtered features: 20</span></span>
<span><span class="co">#&gt; # Filtered samples: 16</span></span>
<span><span class="co">#&gt; # Chosen rank:3</span></span>
<span><span class="co">#&gt; # Rank 3</span></span>
<span><span class="co">#&gt; Mean relative abundances: 0.4506259 0.3696346 0.1797395 </span></span>
<span><span class="co">#&gt; Present feature per factor: 18 14 13</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"nmf"</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>The users can specify the rank. The <code>NMF</code> slot stores the list of <code>NMF</code> results (or <code>NNLM</code> results) per species.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/NMF.html">NMF</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, rank<span class="op">=</span><span class="fl">3</span>, beta<span class="op">=</span><span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="co">#&gt; # NMF started 100003, target: kos, method: snmf/r</span></span>
<span><span class="co">#&gt; # Original features: 20</span></span>
<span><span class="co">#&gt; # Original samples: 16</span></span>
<span><span class="co">#&gt; # Original matrix NA: NA</span></span>
<span><span class="co">#&gt; # Original matrix zero: 0.781</span></span>
<span><span class="co">#&gt; # Selecting KL loss</span></span>
<span><span class="co">#&gt; # Filtered features: 20</span></span>
<span><span class="co">#&gt; # Filtered samples: 16</span></span>
<span><span class="co">#&gt; # Rank 3</span></span>
<span><span class="co">#&gt; Mean relative abundances: 0.4391153 0.3652978 0.1955869 </span></span>
<span><span class="co">#&gt; Present feature per factor: 15 11 12</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"NMF"</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`100003`</span></span>
<span><span class="co">#&gt; &lt;Object of class: NMFfit&gt;</span></span>
<span><span class="co">#&gt;  # Model:</span></span>
<span><span class="co">#&gt;   &lt;Object of class:NMFstd&gt;</span></span>
<span><span class="co">#&gt;   features: 20 </span></span>
<span><span class="co">#&gt;   basis/rank: 3 </span></span>
<span><span class="co">#&gt;   samples: 16 </span></span>
<span><span class="co">#&gt;  # Details:</span></span>
<span><span class="co">#&gt;   algorithm:  snmf/r </span></span>
<span><span class="co">#&gt;   seed:  random </span></span>
<span><span class="co">#&gt;   RNG: 10403L, 624L, ..., -1119848976L [a0b56536ecb759f07f21b4b252fb5db8]</span></span>
<span><span class="co">#&gt;   distance metric:  &lt;function&gt; </span></span>
<span><span class="co">#&gt;   residuals:  5.628987 </span></span>
<span><span class="co">#&gt;   parameters: beta=0.001 </span></span>
<span><span class="co">#&gt;   Iterations: 65 </span></span>
<span><span class="co">#&gt;   Timing:</span></span>
<span><span class="co">#&gt;      user  system elapsed </span></span>
<span><span class="co">#&gt;      0.07    0.00    0.07</span></span></code></pre></div>
<p>The resulting stana object can be used with the other function. <code>plotAbundanceWithinSpecies</code> plots the (relative) abundances per sample using the grouping criteria in stana object.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotAbundanceWithinSpecies.html">plotAbundanceWithinSpecies</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, tss<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/nmf3-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The data can be obtained using <code>return_data</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotAbundanceWithinSpecies.html">plotAbundanceWithinSpecies</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, tss<span class="op">=</span><span class="cn">TRUE</span>, return_data<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     1          2         3  group</span></span>
<span><span class="co">#&gt; ERR1711593 0.98582990 0.01417010 0.0000000 Group1</span></span>
<span><span class="co">#&gt; ERR1711594 0.85247014 0.14752986 0.0000000 Group1</span></span>
<span><span class="co">#&gt; ERR1711596 0.19498362 0.04848823 0.7565282 Group1</span></span>
<span><span class="co">#&gt; ERR1711598 0.17024848 0.22117519 0.6085763 Group1</span></span>
<span><span class="co">#&gt; ERR1711599 0.00000000 1.00000000 0.0000000 Group1</span></span>
<span><span class="co">#&gt; ERR1711602 0.01749079 0.98250921 0.0000000 Group1</span></span></code></pre></div>
<p>The basis corresponds to the factor to feature matrix. This represents functional implications if the KO or gene copy number tables are used for the NMF. This information can be parsed to matrix of KEGG PATHWAY information using <code>pathwayWithFactor</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/pathwayWithFactor.html">pathwayWithFactor</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, tss<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/nmf5-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>These information can be further combined with the other functions in <code>stana</code>, like plotting factor abundances along with the tree inferred from consensus sequence alignment, linking allele frequency information and gene copy number data.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotAbundanceWithinSpecies.html">plotAbundanceWithinSpecies</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, tss<span class="op">=</span><span class="cn">TRUE</span>, return_data<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setMetadata.html">setMetadata</a></span><span class="op">(</span><span class="va">stana</span>, <span class="va">ab</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/inferAndPlotTree.html">inferAndPlotTree</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, meta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ab</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getTreePlot.html">getTreePlot</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`100003`</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/nmf6-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="permanova" class="section level2" number="3.3">
<h2>
<span class="header-section-number">3.3</span> PERMANOVA<a class="anchor" aria-label="anchor" href="#permanova"><i class="fas fa-link"></i></a>
</h2>
<p>Using <code>adonis2</code> function in <code>vegan</code>, one can compare distance matrix based on SNV frequency or gene (gene family) copy numbers, or tree-based distance between the specified group. When the <code>target="tree"</code> is specified, tree shuold be in <code>treeList</code>, with the species name as the key. The <code><a href="https://rdrr.io/pkg/ape/man/cophenetic.phylo.html">ape::cophenetic.phylo()</a></code> is used to calculate distance between tips based on branch length. Distance method can be chosen from <code>dist</code> function in <code>stats</code>, and the default is set to <code>manhattan</code>. You can specify <code>distArg</code> to pass the arguments to <code>dist</code>. Also, the distance calculated directly from sequences can be used. In this case, <code>target='fasta'</code> should be chosen, and the function to calculate distance should be provided to <code>AAfunc</code> argument.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/setTree.html">setTree</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, <span class="va">tre</span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/doAdonis.html">doAdonis</a></span><span class="op">(</span><span class="va">stana</span>, specs <span class="op">=</span> <span class="st">"100003"</span>, target<span class="op">=</span><span class="st">"tree"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Performing adonis in 100003 target is tree</span></span>
<span><span class="co">#&gt; #  F: 0.719649945825046, R2: 0.0740407267582885, Pr: 0.703</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getAdonis.html">getAdonis</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Permutation test for adonis under reduced model</span></span>
<span><span class="co">#&gt; Terms added sequentially (first to last)</span></span>
<span><span class="co">#&gt; Permutation: free</span></span>
<span><span class="co">#&gt; Number of permutations: 999</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; adonis2(formula = d ~ ., data = structure(list(group = c("Group1", "Group1", "Group1", "Group1", "Group2", "Group2", "Group2", "Group2", "Group2", "Group2", "Group2")), row.names = c("ERR1711593", "ERR1711594", "ERR1711596", "ERR1711598", "ERR1711603", "ERR1711605", "ERR1711606", "ERR1711609", "ERR1711611", "ERR1711612", "ERR1711618"), class = "data.frame"))</span></span>
<span><span class="co">#&gt;          Df SumOfSqs      R2      F Pr(&gt;F)</span></span>
<span><span class="co">#&gt; group     1  0.15557 0.07404 0.7196  0.703</span></span>
<span><span class="co">#&gt; Residual  9  1.94558 0.92596              </span></span>
<span><span class="co">#&gt; Total    10  2.10115 1.00000</span></span></code></pre></div>
<p>The corresponding principal coordinate analysis plot using distance matrix can be drawn by specifying <code>pcoa</code>. The relative eigenvalues are plotted.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/doAdonis.html">doAdonis</a></span><span class="op">(</span><span class="va">stana</span>, specs <span class="op">=</span> <span class="st">"100003"</span>,</span>
<span>    target<span class="op">=</span><span class="st">"genes"</span>, pcoa<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Performing adonis in 100003 target is genes</span></span>
<span><span class="co">#&gt; #  F: 0.950009752773493, R2: 0.0635457614064265, Pr: 0.549</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/permanova2-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>By default, the function uses grouping variable in <code>cl</code> slot. For more complex modeling, the slot <code>meta</code> can be populated by the data frame of samples by <code>setMetadata</code> like the example below. To use metadata, <code>useMeta</code> option should be specified with the formula argument.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/doAdonis.html">doAdonis</a></span><span class="op">(</span><span class="va">stana</span>, specs <span class="op">=</span> <span class="st">"100003"</span>,</span>
<span>    target<span class="op">=</span><span class="st">"genes"</span>, useMeta<span class="op">=</span><span class="cn">TRUE</span>, formula <span class="op">=</span> <span class="st">"d ~ ."</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Performing adonis in 100003 target is genes</span></span>
<span><span class="co">#&gt; # Printing raw adonis results ...</span></span>
<span><span class="co">#&gt; No residual component</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; adonis2(formula = d ~ ., data = structure(list(`1` = c(0.985829902295573, 0.852470142666385, 0.194983622925863, 0.1702484777323, 0, 0.01749079304296, 0.169551265957738, 1, 0.774981822424855, 0, 0, 1, 0.145056521442649, 0.894711680596164, 0.725364565076885, 0.0951555820116268), `2` = c(0.0141700977044264, 0.147529857333615, 0.0484882264088995, 0.221175192027691, 1, 0.98250920695704, 0.830448734042262, 0, 0, 0.911065650363313, 0, 0, 0.693043447598221, 0.0456195965920271, 0.274635434923115, 0.67607988474988</span></span>
<span><span class="co">#&gt;          Df  SumOfSqs R2 F Pr(&gt;F)</span></span>
<span><span class="co">#&gt; Model    15 501125509  1         </span></span>
<span><span class="co">#&gt; Residual  0         0  0         </span></span>
<span><span class="co">#&gt; Total    15 501125509  1</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getAdonis.html">getAdonis</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; No residual component</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; adonis2(formula = d ~ ., data = structure(list(`1` = c(0.985829902295573, 0.852470142666385, 0.194983622925863, 0.1702484777323, 0, 0.01749079304296, 0.169551265957738, 1, 0.774981822424855, 0, 0, 1, 0.145056521442649, 0.894711680596164, 0.725364565076885, 0.0951555820116268), `2` = c(0.0141700977044264, 0.147529857333615, 0.0484882264088995, 0.221175192027691, 1, 0.98250920695704, 0.830448734042262, 0, 0, 0.911065650363313, 0, 0, 0.693043447598221, 0.0456195965920271, 0.274635434923115, 0.67607988474988</span></span>
<span><span class="co">#&gt;          Df  SumOfSqs R2 F Pr(&gt;F)</span></span>
<span><span class="co">#&gt; Model    15 501125509  1         </span></span>
<span><span class="co">#&gt; Residual  0         0  0         </span></span>
<span><span class="co">#&gt; Total    15 501125509  1</span></span></code></pre></div>
</div>
<div id="comparing-gene-copy-numbers" class="section level2" number="3.4">
<h2>
<span class="header-section-number">3.4</span> Comparing gene copy numbers<a class="anchor" aria-label="anchor" href="#comparing-gene-copy-numbers"><i class="fas fa-link"></i></a>
</h2>
<p>If you have <code>genes</code> slot filled in the stana object, gene copy numbers can be compared one by one using exact Wilcoxon rank-sum test using <code>wilcox.exact</code> in <code>exactRankTests</code> computing exact conditional p-values. Note that p-values are not adjusted for multiple comparisons made.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/compareGenes.html">compareGenes</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Testing total of 21806</span></span>
<span><span class="va">res</span><span class="op">[[</span><span class="st">"UHGG000008_01733"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  Exact Wilcoxon rank sum test</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; data:  c(1.154444, 2.404241, 0, 1.421386, 1.50773, 0) and c(0.535732, 1.709442, 1.31675, 3.44086, 2.712423, 1.923076, 1.062853, c(1.154444, 2.404241, 0, 1.421386, 1.50773, 0) and 1.21147, 0, 1.509217)</span></span>
<span><span class="co">#&gt; W = 22, p-value = 0.4256</span></span>
<span><span class="co">#&gt; alternative hypothesis: true mu is not equal to 0</span></span></code></pre></div>
</div>
<div id="aggregating-gene-copy-numbers" class="section level2" number="3.5">
<h2>
<span class="header-section-number">3.5</span> Aggregating gene copy numbers<a class="anchor" aria-label="anchor" href="#aggregating-gene-copy-numbers"><i class="fas fa-link"></i></a>
</h2>
<p>The gene abundances across multiple stana object can be aggregated. This returns the new stana object where the gene abundances are combined.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This returns new stana object</span></span>
<span><span class="va">stanacomb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/combineGenes.html">combineGenes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">stana</span>, <span class="va">stana</span><span class="op">)</span>, species<span class="op">=</span><span class="st">"100003"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Common genes: 21806</span></span>
<span><span class="co">#&gt; Duplicate label found in group</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">stanacomb</span>, <span class="st">"genes"</span><span class="op">)</span><span class="op">[[</span><span class="st">"100003"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 21806    32</span></span></code></pre></div>
</div>
<div id="performing-boruta-or-feature-selection" class="section level2" number="3.6">
<h2>
<span class="header-section-number">3.6</span> Performing Boruta or feature selection<a class="anchor" aria-label="anchor" href="#performing-boruta-or-feature-selection"><i class="fas fa-link"></i></a>
</h2>
<p><code>Boruta</code> algorithm can be run on matrices to obtain important marker (SNV position or gene) for distinguishing between group by <code>doBoruta</code> function. The function performs <code>Boruta</code> algorithm on specified data and returns the <code>Boruta</code> class result. By default, the function performs fixes to tentative input. To disable this, specify <code>doFix=FALSE</code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://gitlab.com/mbq/Boruta/">Boruta</a></span><span class="op">)</span></span>
<span><span class="va">brres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/stana/man/doBoruta.html">doBoruta</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # Using grouping from the slot: Group1/Group2</span></span>
<span><span class="co">#&gt; # If needed, please provide preprocessed matrix to `mat`</span></span>
<span><span class="co">#&gt; # Feature number: 21806</span></span>
<span><span class="co">#&gt; # Performing Boruta</span></span>
<span><span class="va">brres</span></span>
<span><span class="co">#&gt; $boruta</span></span>
<span><span class="co">#&gt; Boruta performed 99 iterations in 53.41703 secs.</span></span>
<span><span class="co">#&gt; Tentatives roughfixed over the last 99 iterations.</span></span>
<span><span class="co">#&gt;  7 attributes confirmed important: UHGG000008_01798,</span></span>
<span><span class="co">#&gt; UHGG006336_00186, UHGG060667_01243, UHGG061776_01338,</span></span>
<span><span class="co">#&gt; UHGG158704_01078 and 2 more;</span></span>
<span><span class="co">#&gt;  21799 attributes confirmed unimportant:</span></span>
<span><span class="co">#&gt; UHGG000008_00008, UHGG000008_00009, UHGG000008_00010,</span></span>
<span><span class="co">#&gt; UHGG000008_00012, UHGG000008_00015 and 21794 more;</span></span></code></pre></div>
<p>Further, we visualize the copy numbers of important genes confirmed between the group.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">confirmed_genes</span> <span class="op">&lt;-</span> <span class="va">brres</span><span class="op">$</span><span class="va">boruta</span><span class="op">$</span><span class="va">finalDecision</span><span class="op">[</span><span class="va">brres</span><span class="op">$</span><span class="va">boruta</span><span class="op">$</span><span class="va">finalDecision</span><span class="op">==</span><span class="st">"Confirmed"</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/stana/man/plotGenes.html">plotGenes</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span>, <span class="va">confirmed_genes</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">geneID</span>,scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/panel_border.html">panel_border</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "#F1A340" "#998EC3"</span></span></code></pre></div>
<div class="inline-figure"><img src="02-statistcal_files/figure-html/vis-1.png" width="100%" style="display: block; margin: auto;"></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="importing-and-filtering.html"><span class="header-section-number">2</span> Importing and filtering</a></div>
<div class="next"><a href="function.html"><span class="header-section-number">4</span> Functional annotation</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#statistial-analysis"><span class="header-section-number">3</span> Statistial analysis</a></li>
<li><a class="nav-link" href="#cons"><span class="header-section-number">3.1</span> Consensus sequence calling</a></li>
<li><a class="nav-link" href="#nmf"><span class="header-section-number">3.2</span> Nonnegative matrix factorization</a></li>
<li><a class="nav-link" href="#permanova"><span class="header-section-number">3.3</span> PERMANOVA</a></li>
<li><a class="nav-link" href="#comparing-gene-copy-numbers"><span class="header-section-number">3.4</span> Comparing gene copy numbers</a></li>
<li><a class="nav-link" href="#aggregating-gene-copy-numbers"><span class="header-section-number">3.5</span> Aggregating gene copy numbers</a></li>
<li><a class="nav-link" href="#performing-boruta-or-feature-selection"><span class="header-section-number">3.6</span> Performing Boruta or feature selection</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/noriakis/stana/blob/master/02-statistcal.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/noriakis/stana/edit/master/02-statistcal.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>stana</strong>" was written by Noriaki Sato. It was last built on 2024-05-02.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
