<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Drawing KEGG global and overview map &#8212; pykegg 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=12dfc556" />
    <link rel="stylesheet" type="text/css" href="_static/nbsphinx-code-cells.css?v=2aa19091" />
    <script src="_static/documentation_options.js?v=2709fde1"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within pykegg 0.1 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Visuailzing multiple enrichment analysis results" href="vis-multiple-enrichment.html" />
    <link rel="prev" title="Example usage with PyDESeq2" href="use-with-pydeseq2.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="Drawing-KEGG-global-and-overview-map">
<h1>Drawing KEGG global and overview map<a class="headerlink" href="#Drawing-KEGG-global-and-overview-map" title="Link to this heading">¶</a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import pykegg
import requests_cache
from plotnine import (
    ggplot,
    geom_point,
    aes,
    geom_segment,
    theme_void,
    options
)
import warnings
warnings.filterwarnings(&#39;ignore&#39;)

## Cache all the downloaded files
requests_cache.install_cache(&#39;pykegg_cache&#39;)
graph = pykegg.KGML_graph(pid=&quot;ko01100&quot;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>nodes = graph.get_nodes()
coords = graph.get_coords()
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Plot without map nodes
(ggplot() +
 geom_point(aes(x=&quot;x&quot;, y=&quot;y&quot;), size=0.1,
            data=nodes[nodes.original_type!=&quot;map&quot;], color=nodes[nodes.original_type!=&quot;map&quot;].fgcolor.tolist())+
 geom_segment(aes(x=&quot;x&quot;, y=&quot;y&quot;, xend=&quot;xend&quot;, yend=&quot;yend&quot;),
              color=coords.fgcolor.tolist(), data=coords,
             size=0.1)+
 theme_void())
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/drawing-global-map_3_0.png" class="no-scaled-link" src="_images/drawing-global-map_3_0.png" style="width: 640px; height: 480px;" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Figure Size: (640 x 480)&gt;
</pre></div></div>
</div>
<section id="Colorize-edges-based-on-KO-values">
<h2>Colorize edges based on KO values<a class="headerlink" href="#Colorize-edges-based-on-KO-values" title="Link to this heading">¶</a></h2>
<p>We obtained the data from the metagenomic study of Crohn’s disease patient (<a class="reference external" href="https://doi.org/10.1093/gigascience/gix050">He et al. 2017. GigaScience</a>). Specifically, from <a class="reference external" href="http://gigadb.org/dataset/100317">GigaDB</a>, the abundance of KEGG Orthology is downloaded. The log2 fold change of mean abundance values were obtained between samples with column name starting with <code class="docutils literal notranslate"><span class="pre">CD</span></code> and <code class="docutils literal notranslate"><span class="pre">CT</span></code>.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import pandas as pd
import numpy as np
ko = pd.read_csv(&quot;../KO.profile&quot;, sep=&quot;\t&quot;, index_col=0)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>cd = [i for i in ko.columns if &quot;CD&quot; in i]
ct = [i for i in ko.columns if &quot;CT&quot; in i]
cd_mean = ko.loc[:,cd].apply(lambda x: x.mean(), axis=1)
ct_mean = ko.loc[:,ct].apply(lambda x: x.mean(), axis=1)
lfc_dict = np.log2(cd_mean / ct_mean).to_dict()
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>discard = [i for i in lfc_dict.keys() if lfc_dict[i] == np.inf] + [i for i in lfc_dict.keys() if lfc_dict[i] == -1*np.inf]
for d in discard:
    del lfc_dict[d]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>coords = pykegg.append_colors_continuous_values(coords, lfc_dict, &quot;name&quot;, center_value=0)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Plot without map nodes
(ggplot() +
 geom_segment(aes(x=&quot;x&quot;, y=&quot;y&quot;, xend=&quot;xend&quot;, yend=&quot;yend&quot;),
              color=&quot;#707070&quot;, data=coords, size=0.1)+
 geom_segment(aes(x=&quot;x&quot;, y=&quot;y&quot;, xend=&quot;xend&quot;, yend=&quot;yend&quot;),
              color=coords[coords.color.apply(lambda x: x is not None)].color.tolist(),
              data=coords[coords.color.apply(lambda x: x is not None)],
              size=0.5)+
 geom_point(aes(x=&quot;x&quot;, y=&quot;y&quot;), size=0.1,
            data=nodes[nodes.original_type!=&quot;map&quot;], color=&quot;#707070&quot;)+
 theme_void())
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/drawing-global-map_9_0.png" class="no-scaled-link" src="_images/drawing-global-map_9_0.png" style="width: 640px; height: 480px;" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;Figure Size: (640 x 480)&gt;
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">pykegg</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">API Reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Example analysis</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="example_analysis.html">Notebooks</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="pykegg-intro.html">Simple example using pykegg</a></li>
<li class="toctree-l2"><a class="reference internal" href="pykegg-parallel-edges.html">Visualization of parallel edges using plotnine</a></li>
<li class="toctree-l2"><a class="reference internal" href="single-cell-example.html">Single-cell transcriptomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="use-with-pydeseq2.html">Example usage with PyDESeq2</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Drawing KEGG global and overview map</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Colorize-edges-based-on-KO-values">Colorize edges based on KO values</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="vis-multiple-enrichment.html">Visuailzing multiple enrichment analysis results</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="example_analysis.html">Notebooks</a><ul>
      <li>Previous: <a href="use-with-pydeseq2.html" title="previous chapter">Example usage with PyDESeq2</a></li>
      <li>Next: <a href="vis-multiple-enrichment.html" title="next chapter">Visuailzing multiple enrichment analysis results</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2024, Noriaki Sato.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 7.3.7</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="_sources/drawing-global-map.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>