<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Simple example using pykegg &#8212; pykegg 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=12dfc556" />
    <link rel="stylesheet" type="text/css" href="_static/nbsphinx-code-cells.css?v=2aa19091" />
    <script src="_static/documentation_options.js?v=2709fde1"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within pykegg 0.1 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Visualization of parallel edges using plotnine" href="pykegg-parallel-edges.html" />
    <link rel="prev" title="Notebooks" href="example_analysis.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="Simple-example-using-pykegg">
<h1>Simple example using pykegg<a class="headerlink" href="#Simple-example-using-pykegg" title="Link to this heading">¶</a></h1>
<section id="Overlay-on-the-raw-KEGG-image">
<h2>Overlay on the raw KEGG image<a class="headerlink" href="#Overlay-on-the-raw-KEGG-image" title="Link to this heading">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import requests_cache
import pandas as pd
import numpy as np
from PIL import Image
import pykegg
import matplotlib as mpl
import warnings
warnings.filterwarnings(&#39;ignore&#39;)

## Make sure to cache all the downloaded files to avoid recursive connection
requests_cache.install_cache(&#39;pykegg_cache&#39;)
graph = pykegg.KGML_graph(pid=&quot;hsa03460&quot;)

## Overlay to raw image
nds = graph.get_nodes()

## Assign random color
nds = nds[nds.original_type==&quot;gene&quot;]
nds[&quot;lfc&quot;] = np.arange(-2,2,0.01)[0:nds.shape[0]]
cmap_grad = mpl.colors.LinearSegmentedColormap.from_list(&quot;cmap_grad&quot;, [&quot;yellow&quot;,&quot;green&quot;])
norm = mpl.colors.Normalize(vmin=min(nds.lfc), vmax=max(nds.lfc))
nds[&quot;color&quot;] = [
        mpl.colors.to_hex(cmap_grad(norm(x))) if x is not None else None
        for x in nds.lfc
    ]
Image.fromarray(pykegg.overlay_opencv_image(nds, pid=&quot;hsa03460&quot;))
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/pykegg-intro_2_0.png" src="_images/pykegg-intro_2_0.png" />
</div>
</div>
</section>
<section id="Convert-to-igraph-and-analyze">
<h2>Convert to <code class="docutils literal notranslate"><span class="pre">igraph</span></code> and analyze<a class="headerlink" href="#Convert-to-igraph-and-analyze" title="Link to this heading">¶</a></h2>
<p>We plot here edges and nodes, where gene and group nodes were differently colored by <code class="docutils literal notranslate"><span class="pre">plotnine</span></code>. The node and text size implicates degree. Here, we do not consider the multiple edge attributes.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Export as igraph
g = graph.get_graph(layout=&quot;kk&quot;)
print(g)

nodes = graph.get_nodes()
nodes[&quot;x&quot;] = [i[&quot;x&quot;] for i in g.vs]
nodes[&quot;y&quot;] = [i[&quot;y&quot;] for i in g.vs]
nodes[&quot;deg&quot;] = g.degree(g.vs)
nodes[&quot;graphics_name_sp&quot;] = nodes.graphics_name.apply(lambda x: x.split(&quot;,&quot;)[0])
subset_nodes = nodes[nodes.deg&gt;0]
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
IGRAPH DN-T 58 22 --
+ attr: bgcolor (v), coords (v), fgcolor (v), graphics_name (v), group (v),
  height (v), id (v), name (v), orig_id (v), original_type (v), pathway_name
  (v), type (v), width (v), x (v), xmax (v), xmin (v), y (v), ymax (v), ymin
  (v), entry1_orig_id (e), entry2_orig_id (e), reaction (e), subtypes (e),
  type (e)
+ edges (vertex names):
undefined-&gt;hsa:2177, undefined-&gt;hsa:55215, undefined-&gt;hsa:2175,
undefined-&gt;hsa:55215, undefined-&gt;hsa:57697, undefined-&gt;hsa:5889,
hsa:91442-&gt;hsa:9894, hsa:9894-&gt;hsa:545, undefined-&gt;hsa:57697,
hsa:22909-&gt;undefined, hsa:2177-&gt;hsa:675, hsa:57697-&gt;hsa:91442,
undefined-&gt;hsa:29935 hsa:6117 hsa:6118 hsa:6119, hsa:57697-&gt;hsa:100526739
hsa:201254 hsa:378708, hsa:675-&gt;hsa:5888, hsa:83990-&gt;undefined,
hsa:2177-&gt;hsa:22909, undefined-&gt;hsa:55120, hsa:83990-&gt;undefined,
undefined-&gt;hsa:2189, hsa:545-&gt;hsa:672, undefined-&gt;hsa:57697
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>edges = pykegg.return_segments(graph, subset_nodes)
edges = edges.apply(lambda x: pykegg.shorten_end(x, 0.7), axis=1)
edges[&quot;subtype&quot;] = edges.subtypes.apply(lambda x: x[0][0])
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import matplotlib.patheffects as path_effects
from plotnine import (
    ggplot,
    options,
    geom_point,
    geoms,
    aes,
    geom_segment,
    theme_void,
    geom_rect,
    geom_text,
    scale_size
)
options.figure_size = (6, 4)

(
        ggplot()
         + geom_segment(
            aes(x=&quot;x&quot;, y=&quot;y&quot;, xend=&quot;xend&quot;, yend=&quot;yend&quot;, color=&quot;subtype&quot;), data=edges,
            arrow=geoms.arrow(type=&quot;closed&quot;, length=0.05)
         )
        + geom_point(
            aes(x=&quot;x&quot;,y=&quot;y&quot;,size=&quot;deg&quot;),
            data=subset_nodes[subset_nodes.original_type == &quot;gene&quot;],
            color=&quot;black&quot;, alpha=0.9, fill=&quot;#ffffff&quot;
            )
            + geom_point(
            aes(x=&quot;x&quot;,y=&quot;y&quot;),
            data=subset_nodes[subset_nodes.original_type == &quot;group&quot;],
            color=&quot;black&quot;, alpha=0.9, fill=&quot;#ff0000&quot;
            )
           +geom_text(
             aes(x=&quot;x&quot;,y=&quot;y&quot;, label=&quot;graphics_name_sp&quot;, size=&quot;deg&quot;),
               family=&quot;sans&quot;,
            data=subset_nodes[subset_nodes.original_type == &quot;gene&quot;],
            color=&quot;black&quot;, path_effects=[path_effects.Stroke(linewidth=1, foreground=&#39;white&#39;),
                       path_effects.Normal()]
           )+
    scale_size([5,10])+
    theme_void()

)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/pykegg-intro_6_0.png" class="no-scaled-link" src="_images/pykegg-intro_6_0.png" style="width: 600px; height: 400px;" />
</div>
</div>
</section>
<section id="Plot-using-plotnine">
<h2>Plot using plotnine<a class="headerlink" href="#Plot-using-plotnine" title="Link to this heading">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Some parameters
node_x_nudge=25
node_y_nudge=25
label_size=5
split_graphics_name=&quot;,&quot;
show_label=&quot;gene&quot;
subtype_num=0
edge_color=&quot;subtype&quot;
text_label=&quot;graphics_name&quot;

node_df = graph.get_nodes(node_x_nudge=node_x_nudge, node_y_nudge=node_y_nudge)
edge_df = graph.get_edges()
if split_graphics_name:
    node_df[&quot;graphics_name&quot;] = node_df.graphics_name.apply(
        lambda x: x.split(&quot;,&quot;)[0]
    )

## Collapse subtypes
edge_df_col = []
for i in edge_df.index:
    tmp = edge_df.iloc[i, :]
    for subtype in tmp.subtypes:
        edge_df_col.append(
            [tmp.entry1, tmp.entry2, tmp.type, subtype, tmp.reaction]
        )
edge_df = pd.DataFrame(edge_df_col)
edge_df.columns = [&quot;entry1&quot;, &quot;entry2&quot;, &quot;type&quot;, &quot;subtypes&quot;, &quot;reaction&quot;]
seg_df = pd.concat(
    [
        node_df.reset_index()
        .set_index(&quot;id&quot;)
        .loc[edge_df.entry1]
        .reset_index()
        .loc[:, [&quot;x&quot;, &quot;y&quot;]],
        node_df.reset_index()
        .set_index(&quot;id&quot;)
        .loc[edge_df.entry2]
        .reset_index()
        .loc[:, [&quot;x&quot;, &quot;y&quot;]],
    ],
    axis=1,
)
seg_df.columns = [&quot;x&quot;, &quot;y&quot;, &quot;xend&quot;, &quot;yend&quot;]
seg_df = pd.concat([seg_df, edge_df], axis=1)

seg_df[&quot;subtype&quot;] = seg_df.subtypes.apply(
    lambda x: x[subtype_num] if x is not None else x
)

plot = (
    ggplot()
    + geom_segment(
        aes(x=&quot;x&quot;, y=&quot;y&quot;, xend=&quot;xend&quot;, yend=&quot;yend&quot;, color=edge_color), data=seg_df
    )
    + geom_rect(
        aes(xmin=&quot;xmin&quot;, ymin=&quot;ymin&quot;, xmax=&quot;xmax&quot;, ymax=&quot;ymax&quot;),
        data=node_df[node_df.original_type == &quot;gene&quot;],
        fill=&quot;white&quot;,
        color=&quot;grey&quot;,
    )
    + geom_rect(
        aes(xmin=&quot;xmin&quot;, ymin=&quot;ymin&quot;, xmax=&quot;xmax&quot;, ymax=&quot;ymax&quot;),
        data=node_df[node_df.original_type == &quot;compound&quot;],
        fill=&quot;white&quot;,
        color=&quot;grey&quot;,
    )
    + geom_text(
        aes(x=&quot;x&quot;, y=&quot;y&quot;, label=text_label, filter=&quot;original_type!=&#39;group&#39;&quot;),
        data=node_df[node_df.original_type == show_label],
        size=label_size,
    )
    + theme_void()
)

plot
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/pykegg-intro_8_0.png" class="no-scaled-link" src="_images/pykegg-intro_8_0.png" style="width: 600px; height: 400px;" />
</div>
</div>
</section>
<section id="Customization-of-nodes-and-edges">
<h2>Customization of nodes and edges<a class="headerlink" href="#Customization-of-nodes-and-edges" title="Link to this heading">¶</a></h2>
<p>In <code class="docutils literal notranslate"><span class="pre">plot_kegg_pathway_plotnine</span></code>, various geoms are used to plot KEGG PATHWAY information. Users would like to customize the visualization such as using arrows for directed relationships. Some functions are prepared for this purpose, outlined below.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>graph = pykegg.KGML_graph(pid=&quot;hsa04110&quot;) ## Cell cycle
options.figure_size = (12, 9) ## Change plotnine plot size
node_df = graph.get_nodes(node_x_nudge=20, node_y_nudge=10) ## Change the node size
seg_df = pykegg.return_segments(graph) ## return edge data frame for the use in `geom_segment`
seg_df2 = seg_df.apply(lambda x: pykegg.shorten_end(x, 0.8), axis=1) ## shorten end of edges for visualization of arrow
hsa = pykegg.id_to_name_dict(&quot;hsa&quot;) ## Obtain dict for converting hsa* IDs
node_df[&quot;converted&quot;] = node_df[&quot;name&quot;].apply(lambda x: pykegg.convert_id(x, hsa)) ## Convert ID
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>(
        ggplot()
         + geom_segment(
            aes(x=&quot;x&quot;, y=&quot;y&quot;, xend=&quot;xend&quot;, yend=&quot;yend&quot;), data=seg_df2,
            arrow=geoms.arrow(type=&quot;closed&quot;, length=0.05)
         )
        + geom_rect(
            aes(xmin=&quot;xmin&quot;, ymin=&quot;ymin&quot;, xmax=&quot;xmax&quot;, ymax=&quot;ymax&quot;),
            data=node_df[node_df.original_type == &quot;group&quot;],
            color=&quot;black&quot;, alpha=0, fill=&quot;#ffffff&quot;
            )
        + geom_rect(
            aes(xmin=&quot;xmin&quot;, ymin=&quot;ymin&quot;, xmax=&quot;xmax&quot;, ymax=&quot;ymax&quot;),
            data=node_df[node_df.original_type == &quot;gene&quot;],
            color=&quot;black&quot;,
            fill=node_df[node_df.original_type == &quot;gene&quot;].bgcolor)
           +geom_text(
             aes(x=&quot;x&quot;,y=&quot;y&quot;, label=&quot;converted&quot;),
               family=&quot;serif&quot;, size=8,
            data=node_df[node_df.original_type == &quot;gene&quot;],
            color=&quot;black&quot;
           )+
    theme_void()


)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/pykegg-intro_11_0.png" class="no-scaled-link" src="_images/pykegg-intro_11_0.png" style="width: 1200px; height: 900px;" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">pykegg</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">API Reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Example analysis</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="example_analysis.html">Notebooks</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">Simple example using pykegg</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Overlay-on-the-raw-KEGG-image">Overlay on the raw KEGG image</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Convert-to-igraph-and-analyze">Convert to <code class="docutils literal notranslate"><span class="pre">igraph</span></code> and analyze</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Plot-using-plotnine">Plot using plotnine</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Customization-of-nodes-and-edges">Customization of nodes and edges</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="pykegg-parallel-edges.html">Visualization of parallel edges using plotnine</a></li>
<li class="toctree-l2"><a class="reference internal" href="single-cell-example.html">Single-cell transcriptomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="use-with-pydeseq2.html">Example usage with PyDESeq2</a></li>
<li class="toctree-l2"><a class="reference internal" href="drawing-global-map.html">Drawing KEGG global and overview map</a></li>
<li class="toctree-l2"><a class="reference internal" href="vis-multiple-enrichment.html">Visuailzing multiple enrichment analysis results</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="example_analysis.html">Notebooks</a><ul>
      <li>Previous: <a href="example_analysis.html" title="previous chapter">Notebooks</a></li>
      <li>Next: <a href="pykegg-parallel-edges.html" title="next chapter">Visualization of parallel edges using plotnine</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2024, Noriaki Sato.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 7.3.7</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="_sources/pykegg-intro.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>