<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Single-cell transcriptomics &#8212; pykegg 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=12dfc556" />
    <link rel="stylesheet" type="text/css" href="_static/nbsphinx-code-cells.css?v=2aa19091" />
    <script src="_static/documentation_options.js?v=2709fde1"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within pykegg 0.1 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Example usage with PyDESeq2" href="use-with-pydeseq2.html" />
    <link rel="prev" title="Visualization of parallel edges using plotnine" href="pykegg-parallel-edges.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="Single-cell-transcriptomics">
<h1>Single-cell transcriptomics<a class="headerlink" href="#Single-cell-transcriptomics" title="Link to this heading">¶</a></h1>
<p>Provided you completed the <a class="reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html">3k PBMCs tutorial</a> of scanpy, we illustrate the depiction of marker gene information upon the KEGG pathway in this example.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import numpy as np
import pandas as pd
import scanpy as sc
import matplotlib
import warnings
warnings.filterwarnings(&#39;ignore&#39;)
matplotlib.rcParams[&#39;figure.figsize&#39;] = (6, 6)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata = sc.read_10x_mtx(
    &#39;../filtered_gene_bc_matrices/hg19&#39;,  # the directory with the `.mtx` file
    var_names=&#39;gene_symbols&#39;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata.var_names_make_unique()
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.filter_cells(adata, min_genes=200)
sc.pp.filter_genes(adata, min_cells=3)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata.var[&#39;mt&#39;] = adata.var_names.str.startswith(&#39;MT-&#39;)  # annotate the group of mitochondrial genes as &#39;mt&#39;
sc.pp.calculate_qc_metrics(adata, qc_vars=[&#39;mt&#39;], percent_top=None, log1p=False, inplace=True)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata = adata[adata.obs.n_genes_by_counts &lt; 2500, :]
adata = adata[adata.obs.pct_counts_mt &lt; 5, :]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.normalize_total(adata, target_sum=1e4)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.log1p(adata)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata = adata[:, adata.var.highly_variable]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.regress_out(adata, [&#39;total_counts&#39;, &#39;pct_counts_mt&#39;])
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.scale(adata, max_value=10)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.tl.pca(adata, svd_solver=&#39;arpack&#39;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.tl.leiden(adata)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.tl.rank_genes_groups(adata, &#39;leiden&#39;, method=&#39;t-test&#39;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Consider top-50 markers
markers = pd.DataFrame(adata.uns[&#39;rank_genes_groups&#39;][&#39;names&#39;]).head(50)
markers
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>0</th>
      <th>1</th>
      <th>2</th>
      <th>3</th>
      <th>4</th>
      <th>5</th>
      <th>6</th>
      <th>7</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>LTB</td>
      <td>CST3</td>
      <td>HLA-DPB1</td>
      <td>CCL5</td>
      <td>FCER1G</td>
      <td>NKG7</td>
      <td>HLA-DPA1</td>
      <td>PF4</td>
    </tr>
    <tr>
      <th>1</th>
      <td>IL32</td>
      <td>TYROBP</td>
      <td>CD79A</td>
      <td>NKG7</td>
      <td>AIF1</td>
      <td>CTSW</td>
      <td>HLA-DPB1</td>
      <td>PPBP</td>
    </tr>
    <tr>
      <th>2</th>
      <td>CD2</td>
      <td>S100A8</td>
      <td>HLA-DRB1</td>
      <td>IL32</td>
      <td>LST1</td>
      <td>GZMB</td>
      <td>HLA-DRB1</td>
      <td>SDPR</td>
    </tr>
    <tr>
      <th>3</th>
      <td>MAL</td>
      <td>FCN1</td>
      <td>HLA-DPA1</td>
      <td>CST7</td>
      <td>TYROBP</td>
      <td>GNLY</td>
      <td>HLA-DMA</td>
      <td>GNG11</td>
    </tr>
    <tr>
      <th>4</th>
      <td>AQP3</td>
      <td>LGALS1</td>
      <td>CD79B</td>
      <td>GZMA</td>
      <td>CST3</td>
      <td>PRF1</td>
      <td>HLA-DQA1</td>
      <td>NRGN</td>
    </tr>
    <tr>
      <th>5</th>
      <td>HINT1</td>
      <td>S100A6</td>
      <td>HLA-DQA1</td>
      <td>CTSW</td>
      <td>SAT1</td>
      <td>CST7</td>
      <td>HLA-DQB1</td>
      <td>SPARC</td>
    </tr>
    <tr>
      <th>6</th>
      <td>GIMAP7</td>
      <td>LGALS2</td>
      <td>MS4A1</td>
      <td>GZMK</td>
      <td>COTL1</td>
      <td>GZMA</td>
      <td>CST3</td>
      <td>CCL5</td>
    </tr>
    <tr>
      <th>7</th>
      <td>GIMAP5</td>
      <td>AIF1</td>
      <td>HLA-DQB1</td>
      <td>PTPRCAP</td>
      <td>IFITM3</td>
      <td>FGFBP2</td>
      <td>FCER1A</td>
      <td>RGS18</td>
    </tr>
    <tr>
      <th>8</th>
      <td>LDLRAP1</td>
      <td>GPX1</td>
      <td>CD37</td>
      <td>LYAR</td>
      <td>CTSS</td>
      <td>FCGR3A</td>
      <td>TACR2</td>
      <td>GPX1</td>
    </tr>
    <tr>
      <th>9</th>
      <td>TRABD2A</td>
      <td>LST1</td>
      <td>TCL1A</td>
      <td>KLRG1</td>
      <td>LGALS1</td>
      <td>SRGN</td>
      <td>FSTL1</td>
      <td>TPM4</td>
    </tr>
    <tr>
      <th>10</th>
      <td>ACAP1</td>
      <td>GSTP1</td>
      <td>HLA-DMA</td>
      <td>GZMH</td>
      <td>FCGR3A</td>
      <td>CD247</td>
      <td>SYTL4</td>
      <td>GP9</td>
    </tr>
    <tr>
      <th>11</th>
      <td>FYB</td>
      <td>TYMP</td>
      <td>LTB</td>
      <td>PRF1</td>
      <td>S100A11</td>
      <td>SPON2</td>
      <td>RP11-324I22.4</td>
      <td>HIST1H2AC</td>
    </tr>
    <tr>
      <th>12</th>
      <td>TRAF3IP3</td>
      <td>CTSS</td>
      <td>LINC00926</td>
      <td>CD2</td>
      <td>HLA-DPA1</td>
      <td>UBB</td>
      <td>FAXDC2</td>
      <td>CD9</td>
    </tr>
    <tr>
      <th>13</th>
      <td>ITM2A</td>
      <td>FCER1G</td>
      <td>HLA-DMB</td>
      <td>SAMD3</td>
      <td>TYMP</td>
      <td>FCER1G</td>
      <td>HLA-DMB</td>
      <td>AP001189.4</td>
    </tr>
    <tr>
      <th>14</th>
      <td>PASK</td>
      <td>COTL1</td>
      <td>HVCN1</td>
      <td>HOPX</td>
      <td>NPC2</td>
      <td>ID2</td>
      <td>HP</td>
      <td>TUBB1</td>
    </tr>
    <tr>
      <th>15</th>
      <td>SCGB3A1</td>
      <td>S100A11</td>
      <td>EAF2</td>
      <td>APOBEC3G</td>
      <td>CEBPB</td>
      <td>CCL4</td>
      <td>RP4-781K5.2</td>
      <td>ITGA2B</td>
    </tr>
    <tr>
      <th>16</th>
      <td>TNFRSF4</td>
      <td>SAT1</td>
      <td>IRF8</td>
      <td>TIGIT</td>
      <td>HLA-DRB1</td>
      <td>TYROBP</td>
      <td>PPT2-EGFL8</td>
      <td>MPP1</td>
    </tr>
    <tr>
      <th>17</th>
      <td>GIMAP4</td>
      <td>CYBA</td>
      <td>FCRLA</td>
      <td>NCR3</td>
      <td>S100A6</td>
      <td>HOPX</td>
      <td>HIST1H2AH</td>
      <td>CLU</td>
    </tr>
    <tr>
      <th>18</th>
      <td>RP11-18H21.1</td>
      <td>NPC2</td>
      <td>HLA-DOB</td>
      <td>GIMAP7</td>
      <td>CYBA</td>
      <td>GZMH</td>
      <td>GPX1</td>
      <td>TMEM40</td>
    </tr>
    <tr>
      <th>19</th>
      <td>CISH</td>
      <td>GRN</td>
      <td>PKIG</td>
      <td>CCL4</td>
      <td>SRGN</td>
      <td>CLIC3</td>
      <td>SAMD14</td>
      <td>CA2</td>
    </tr>
    <tr>
      <th>20</th>
      <td>RP11-291B21.2</td>
      <td>MS4A6A</td>
      <td>SMIM14</td>
      <td>CD247</td>
      <td>PYCARD</td>
      <td>XCL2</td>
      <td>CLEC10A</td>
      <td>ARPC1B</td>
    </tr>
    <tr>
      <th>21</th>
      <td>PTPRCAP</td>
      <td>LGALS3</td>
      <td>CD72</td>
      <td>ID2</td>
      <td>IFI30</td>
      <td>CCL5</td>
      <td>PROK2</td>
      <td>NCOA4</td>
    </tr>
    <tr>
      <th>22</th>
      <td>GPR183</td>
      <td>PYCARD</td>
      <td>P2RX5</td>
      <td>FGFBP2</td>
      <td>HLA-DPB1</td>
      <td>PTPRCAP</td>
      <td>PVALB</td>
      <td>PTCRA</td>
    </tr>
    <tr>
      <th>23</th>
      <td>SELL</td>
      <td>TALDO1</td>
      <td>SPIB</td>
      <td>ZAP70</td>
      <td>ARPC1B</td>
      <td>IGFBP7</td>
      <td>GSTP1</td>
      <td>TREML1</td>
    </tr>
    <tr>
      <th>24</th>
      <td>SRSF5</td>
      <td>FCGRT</td>
      <td>PTPRCAP</td>
      <td>ARPC5L</td>
      <td>ABI3</td>
      <td>AKR1C3</td>
      <td>HGD</td>
      <td>FERMT3</td>
    </tr>
    <tr>
      <th>25</th>
      <td>ATP6V0E2</td>
      <td>SRGN</td>
      <td>PNOC</td>
      <td>C9orf142</td>
      <td>FCN1</td>
      <td>APOBEC3G</td>
      <td>ITGA2B</td>
      <td>PGRMC1</td>
    </tr>
    <tr>
      <th>26</th>
      <td>IL23A</td>
      <td>NCF2</td>
      <td>BLNK</td>
      <td>GIMAP4</td>
      <td>APOBEC3A</td>
      <td>TTC38</td>
      <td>LINC00957</td>
      <td>SAT1</td>
    </tr>
    <tr>
      <th>27</th>
      <td>ICOS</td>
      <td>ARPC1B</td>
      <td>KIAA0125</td>
      <td>LITAF</td>
      <td>C1orf162</td>
      <td>CYBA</td>
      <td>TDRP</td>
      <td>LAMTOR1</td>
    </tr>
    <tr>
      <th>28</th>
      <td>NAP1L4</td>
      <td>HLA-DRB1</td>
      <td>SWAP70</td>
      <td>HSP90AA1</td>
      <td>CDA</td>
      <td>PRSS23</td>
      <td>LGALS2</td>
      <td>ODC1</td>
    </tr>
    <tr>
      <th>29</th>
      <td>TNFRSF25</td>
      <td>CDA</td>
      <td>CD19</td>
      <td>FCRL6</td>
      <td>C5AR1</td>
      <td>ARPC5L</td>
      <td>RP11-252A24.3</td>
      <td>ACRBP</td>
    </tr>
    <tr>
      <th>30</th>
      <td>SIRPG</td>
      <td>IFI30</td>
      <td>IGLL5</td>
      <td>KLRB1</td>
      <td>MAFB</td>
      <td>PDIA3</td>
      <td>TGFB1I1</td>
      <td>SEPT5</td>
    </tr>
    <tr>
      <th>31</th>
      <td>AKTIP</td>
      <td>C1orf162</td>
      <td>LIMD2</td>
      <td>PRR5</td>
      <td>FGL2</td>
      <td>EFHD2</td>
      <td>GRN</td>
      <td>MMD</td>
    </tr>
    <tr>
      <th>32</th>
      <td>CD96</td>
      <td>IFITM3</td>
      <td>ADAM28</td>
      <td>GYG1</td>
      <td>TNFRSF1B</td>
      <td>S1PR5</td>
      <td>RAB6B</td>
      <td>F13A1</td>
    </tr>
    <tr>
      <th>33</th>
      <td>ILF3-AS1</td>
      <td>ALDH2</td>
      <td>SNX3</td>
      <td>JAKMIP1</td>
      <td>BLVRA</td>
      <td>LITAF</td>
      <td>S100P</td>
      <td>SNCA</td>
    </tr>
    <tr>
      <th>34</th>
      <td>RIC3</td>
      <td>TNFSF13B</td>
      <td>SNX29P2</td>
      <td>STK17A</td>
      <td>BLOC1S1</td>
      <td>XCL1</td>
      <td>PBLD</td>
      <td>SRGN</td>
    </tr>
    <tr>
      <th>35</th>
      <td>PPP1R2</td>
      <td>FPR1</td>
      <td>LAT2</td>
      <td>S1PR5</td>
      <td>TNFSF10</td>
      <td>GPR56</td>
      <td>PDZK1IP1</td>
      <td>LY6G6F</td>
    </tr>
    <tr>
      <th>36</th>
      <td>GPR171</td>
      <td>CEBPB</td>
      <td>ARHGAP24</td>
      <td>SRGN</td>
      <td>LGALS3</td>
      <td>ABI3</td>
      <td>DNAJC27</td>
      <td>CMTM5</td>
    </tr>
    <tr>
      <th>37</th>
      <td>RORA</td>
      <td>ASGR1</td>
      <td>FCGR2B</td>
      <td>ARL6IP5</td>
      <td>ARRB2</td>
      <td>PLEKHF1</td>
      <td>RP11-390B4.5</td>
      <td>PTGS1</td>
    </tr>
    <tr>
      <th>38</th>
      <td>CCDC66</td>
      <td>LINC00936</td>
      <td>MZB1</td>
      <td>GZMB</td>
      <td>NCF2</td>
      <td>HAVCR2</td>
      <td>CCDC122</td>
      <td>GP1BA</td>
    </tr>
    <tr>
      <th>39</th>
      <td>RP11-589C21.6</td>
      <td>IGSF6</td>
      <td>PLD4</td>
      <td>PSME1</td>
      <td>TNFSF13B</td>
      <td>ZAP70</td>
      <td>ZNF594</td>
      <td>MYL9</td>
    </tr>
    <tr>
      <th>40</th>
      <td>DENND2D</td>
      <td>FOLR3</td>
      <td>BTK</td>
      <td>TMBIM6</td>
      <td>STX11</td>
      <td>NCR3</td>
      <td>SUOX</td>
      <td>RP11-367G6.3</td>
    </tr>
    <tr>
      <th>41</th>
      <td>SSBP1</td>
      <td>CNPY3</td>
      <td>PRKCB</td>
      <td>RORA</td>
      <td>FCGR2A</td>
      <td>PTGDR</td>
      <td>ITGB3</td>
      <td>DYNLL1</td>
    </tr>
    <tr>
      <th>42</th>
      <td>SUCLG2</td>
      <td>SERPINB1</td>
      <td>IGJ</td>
      <td>GIMAP5</td>
      <td>OAS1</td>
      <td>SAMD3</td>
      <td>SEPT4</td>
      <td>TPM1</td>
    </tr>
    <tr>
      <th>43</th>
      <td>AP3M2</td>
      <td>APOBEC3A</td>
      <td>C16orf74</td>
      <td>CD160</td>
      <td>FCGRT</td>
      <td>C5orf56</td>
      <td>CYBA</td>
      <td>SNX3</td>
    </tr>
    <tr>
      <th>44</th>
      <td>STK17A</td>
      <td>PLBD1</td>
      <td>PPP1R14A</td>
      <td>SNRPB</td>
      <td>CTD-2006K23.1</td>
      <td>PSMB8</td>
      <td>SYP</td>
      <td>TALDO1</td>
    </tr>
    <tr>
      <th>45</th>
      <td>MDS2</td>
      <td>BLVRA</td>
      <td>RPL22L1</td>
      <td>PTGDR</td>
      <td>SYTL4</td>
      <td>CCL3</td>
      <td>CD1C</td>
      <td>MARCH2</td>
    </tr>
    <tr>
      <th>46</th>
      <td>CORO1B</td>
      <td>S100A12</td>
      <td>FCRL2</td>
      <td>TERF2IP</td>
      <td>RGS19</td>
      <td>PSME1</td>
      <td>AP001189.4</td>
      <td>C2orf88</td>
    </tr>
    <tr>
      <th>47</th>
      <td>MAGEH1</td>
      <td>CCL3</td>
      <td>PARP1</td>
      <td>BUB3</td>
      <td>SMCO4</td>
      <td>FCRL6</td>
      <td>PHACTR1</td>
      <td>CLEC1B</td>
    </tr>
    <tr>
      <th>48</th>
      <td>SNHG8</td>
      <td>VAMP8</td>
      <td>GUCD1</td>
      <td>ORAI1</td>
      <td>C20orf27</td>
      <td>C9orf142</td>
      <td>DRAXIN</td>
      <td>AC147651.3</td>
    </tr>
    <tr>
      <th>49</th>
      <td>CCNG1</td>
      <td>SMCO4</td>
      <td>RP5-887A10.1</td>
      <td>CDC42EP3</td>
      <td>BTK</td>
      <td>RAMP1</td>
      <td>COLGALT2</td>
      <td>GAS2L1</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.tl.paga(adata)
sc.pl.paga(adata, plot=False)  # remove `plot=False` if you want to see the coarse-grained graph
sc.tl.umap(adata, init_pos=&#39;paga&#39;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>new_cluster_names = [
    &#39;CD4 T&#39;, &#39;CD14 Monocytes&#39;,
    &#39;B&#39;, &#39;CD8 T&#39;,
    &#39;NK&#39;, &#39;FCGR3A Monocytes&#39;,
    &#39;Dendritic&#39;, &#39;Megakaryocytes&#39;]
# adata.rename_categories(&#39;leiden&#39;, new_cluster_names)
adata.obs[&quot;leiden&quot;] = adata.obs[&quot;leiden&quot;].cat.rename_categories(new_cluster_names)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pl.umap(adata, color=&#39;leiden&#39;, legend_loc=&#39;on data&#39;,
           frameon=False, legend_fontsize=10, legend_fontoutline=2)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/single-cell-example_20_0.png" src="_images/single-cell-example_20_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pl.pca(adata, color=&#39;leiden&#39;, legend_loc=&#39;on data&#39;, title=&#39;&#39;, frameon=False, legend_fontoutline=2)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/single-cell-example_21_0.png" src="_images/single-cell-example_21_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import pykegg
from matplotlib import colors as mcolors
from PIL import Image
cols = mcolors.TABLEAU_COLORS
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import matplotlib.pyplot as plt
from matplotlib.colors import to_rgba_array
fig, ax = plt.subplots(1,1)
ax.imshow(to_rgba_array([cols[i] for i in cols.keys()]).reshape(10,1,4)[0:8,:,:])
ax.set_yticks(np.arange(8))
ax.set_yticklabels(new_cluster_names)
plt.tight_layout()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/single-cell-example_23_0.png" src="_images/single-cell-example_23_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import requests_cache

## Cache all the downloaded files
requests_cache.install_cache(&#39;pykegg_cache&#39;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>g = pykegg.KGML_graph(pid=&quot;hsa04380&quot;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>nds = g.get_nodes()
for mark in [0,1,2,3,4,5,6,7]:
    nds[&quot;in&quot;] = nds[&quot;graphics_name&quot;].apply(lambda x:
                                           len(set(markers[str(mark)]) &amp;
                                               set([i.replace(&quot; &quot;,&quot;&quot;).replace(&quot;...&quot;,&quot;&quot;) for i in x.split(&quot;,&quot;)])))
    nds[&quot;color&quot;+str(mark)] = [cols[list(cols.keys())[mark]] if i else None for i in nds[&quot;in&quot;].tolist()]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>qc = list()
for id in nds.id:
    tmp = nds[nds.id==id].loc[:,[i for i in nds.columns if i.startswith(&quot;color&quot;)]]
    tmpl = tmp.iloc[0,:].tolist()
    if len(set(tmpl))==1 and tmpl[0]==None:
        qc.append(&quot;#ffffff&quot;)
    else:
        qc.append([i for i in tmpl if i is not None])
nds[&quot;color&quot;]=qc
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>kegg_map = pykegg.overlay_opencv_image(nds, pid=&quot;hsa04380&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>Image.fromarray(kegg_map)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/single-cell-example_29_0.png" src="_images/single-cell-example_29_0.png" />
</div>
</div>
<section id="Associate-statistical-values-with-the-nodes">
<h2>Associate statistical values with the nodes<a class="headerlink" href="#Associate-statistical-values-with-the-nodes" title="Link to this heading">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[68]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Extract adjusted p-values from the scanpy object
## This time, cluster 4 identified by leiden algorithm
pval_adj = pd.concat([
    pd.DataFrame(adata.uns[&#39;rank_genes_groups&#39;][&#39;names&#39;])[&quot;4&quot;],
    pd.DataFrame(adata.uns[&#39;rank_genes_groups&#39;][&#39;pvals_adj&#39;])[&quot;4&quot;],
], axis=1)
## Make dict
pval_adj.index = pval_adj.iloc[:,0]
pval_adj_dic = pval_adj.iloc[:,1].apply(lambda x: np.log10(x)).to_dict()
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[75]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Append new color column based on dict
nds = pykegg.append_colors_continuous_values(nds,
                                       pval_adj_dic,
                                       new_color_column=&quot;pval_adj_4&quot;,
                                       two_slope=False,
                                        colors=[&quot;cyan&quot;,&quot;red&quot;])
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[78]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>img_arr = pykegg.overlay_opencv_image(nds, pid=&quot;hsa04380&quot;, fill_color=&quot;pval_adj_4&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[79]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>## Visualize
Image.fromarray(img_arr)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[79]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/single-cell-example_34_0.png" src="_images/single-cell-example_34_0.png" />
</div>
</div>
</section>
<section id="Legends">
<h2>Legends<a class="headerlink" href="#Legends" title="Link to this heading">¶</a></h2>
<p>The legend can be added through the <code class="docutils literal notranslate"><span class="pre">append_legend</span></code> function. Same specifications as when specifying colors are required for the min_value, max_value, and center_value. The position can be selected from four options: “topright,” “topleft,” “bottomright,” and “bottomleft.”</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[62]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>max([pval_adj_dic[i] for i in pval_adj_dic.keys()])
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[62]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
-0.0006253720313310442
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[80]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>Image.fromarray(pykegg.append_legend(img_arr,
                                     min_value=min([pval_adj_dic[i] for i in pval_adj_dic.keys()]),
                                    max_value=0, two_slope=False, colors=[&quot;cyan&quot;,&quot;red&quot;],
                                    width=6.5, label=&quot;log10(adjusted p-value for marker 4)&quot;))
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[80]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/single-cell-example_38_0.png" src="_images/single-cell-example_38_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">pykegg</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">API Reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Example analysis</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="example_analysis.html">Notebooks</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="pykegg-intro.html">Simple example using pykegg</a></li>
<li class="toctree-l2"><a class="reference internal" href="pykegg-parallel-edges.html">Visualization of parallel edges using plotnine</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Single-cell transcriptomics</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Associate-statistical-values-with-the-nodes">Associate statistical values with the nodes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Legends">Legends</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="use-with-pydeseq2.html">Example usage with PyDESeq2</a></li>
<li class="toctree-l2"><a class="reference internal" href="drawing-global-map.html">Drawing KEGG global and overview map</a></li>
<li class="toctree-l2"><a class="reference internal" href="vis-multiple-enrichment.html">Visuailzing multiple enrichment analysis results</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="example_analysis.html">Notebooks</a><ul>
      <li>Previous: <a href="pykegg-parallel-edges.html" title="previous chapter">Visualization of parallel edges using plotnine</a></li>
      <li>Next: <a href="use-with-pydeseq2.html" title="next chapter">Example usage with PyDESeq2</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2024, Noriaki Sato.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 7.3.7</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="_sources/single-cell-example.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>