<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Example usage with PyDESeq2 &#8212; pykegg 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=12dfc556" />
    <link rel="stylesheet" type="text/css" href="_static/nbsphinx-code-cells.css?v=2aa19091" />
    <script src="_static/documentation_options.js?v=2709fde1"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within pykegg 0.1 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Drawing KEGG global and overview map" href="drawing-global-map.html" />
    <link rel="prev" title="Single-cell transcriptomics" href="single-cell-example.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="Example-usage-with-PyDESeq2">
<h1>Example usage with PyDESeq2<a class="headerlink" href="#Example-usage-with-PyDESeq2" title="Link to this heading">¶</a></h1>
<p>The codes are derived from <a class="reference external" href="https://pydeseq2.readthedocs.io/en/latest/auto_examples/plot_step_by_step.html">Step-by-step PyDESeq2 workflow</a>. The dataset used was from the paper investigating BK polyomavirus infection in urothelial cells (<a class="reference external" href="https://www.nature.com/articles/s41388-022-02235-8">Baker et al. Oncogene. 2022</a>).</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import os
import pickle as pkl

from pydeseq2.dds import DeseqDataSet
from pydeseq2.ds import DeseqStats
from pydeseq2.utils import load_example_data
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import pandas as pd
count_df = pd.read_csv(&quot;../PRJNA728925_count.txt&quot;, sep=&quot;\t&quot;).T
count_df.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>A1BG</th>
      <th>A1BG-AS1</th>
      <th>A1CF</th>
      <th>A2M</th>
      <th>A2M-AS1</th>
      <th>A2ML1</th>
      <th>A2MP1</th>
      <th>A3GALT2</th>
      <th>A4GALT</th>
      <th>A4GNT</th>
      <th>...</th>
      <th>ZWILCH</th>
      <th>ZWINT</th>
      <th>ZXDA</th>
      <th>ZXDB</th>
      <th>ZXDC</th>
      <th>ZYG11A</th>
      <th>ZYG11B</th>
      <th>ZYX</th>
      <th>ZZEF1</th>
      <th>ZZZ3</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>SRR14509882</th>
      <td>49</td>
      <td>57</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>718</td>
      <td>1</td>
      <td>0</td>
      <td>3682</td>
      <td>0</td>
      <td>...</td>
      <td>574</td>
      <td>793</td>
      <td>223</td>
      <td>1320</td>
      <td>882</td>
      <td>1</td>
      <td>1791</td>
      <td>2787</td>
      <td>2276</td>
      <td>1576</td>
    </tr>
    <tr>
      <th>SRR14509883</th>
      <td>106</td>
      <td>82</td>
      <td>0</td>
      <td>12</td>
      <td>6</td>
      <td>83</td>
      <td>0</td>
      <td>0</td>
      <td>1715</td>
      <td>1</td>
      <td>...</td>
      <td>731</td>
      <td>1266</td>
      <td>44</td>
      <td>471</td>
      <td>389</td>
      <td>5</td>
      <td>632</td>
      <td>2127</td>
      <td>871</td>
      <td>676</td>
    </tr>
    <tr>
      <th>SRR14509884</th>
      <td>67</td>
      <td>36</td>
      <td>2</td>
      <td>25</td>
      <td>5</td>
      <td>499</td>
      <td>0</td>
      <td>0</td>
      <td>2188</td>
      <td>0</td>
      <td>...</td>
      <td>1172</td>
      <td>2256</td>
      <td>122</td>
      <td>741</td>
      <td>732</td>
      <td>14</td>
      <td>915</td>
      <td>3463</td>
      <td>1419</td>
      <td>1009</td>
    </tr>
    <tr>
      <th>SRR14509885</th>
      <td>85</td>
      <td>67</td>
      <td>0</td>
      <td>2</td>
      <td>3</td>
      <td>393</td>
      <td>0</td>
      <td>0</td>
      <td>2155</td>
      <td>2</td>
      <td>...</td>
      <td>1348</td>
      <td>3155</td>
      <td>187</td>
      <td>1566</td>
      <td>738</td>
      <td>26</td>
      <td>1198</td>
      <td>3738</td>
      <td>2044</td>
      <td>1051</td>
    </tr>
    <tr>
      <th>SRR14509886</th>
      <td>29</td>
      <td>42</td>
      <td>0</td>
      <td>2</td>
      <td>2</td>
      <td>76</td>
      <td>0</td>
      <td>0</td>
      <td>2834</td>
      <td>0</td>
      <td>...</td>
      <td>298</td>
      <td>137</td>
      <td>205</td>
      <td>743</td>
      <td>852</td>
      <td>0</td>
      <td>1548</td>
      <td>2878</td>
      <td>1952</td>
      <td>1297</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 29744 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>clinical_df = pd.read_csv(&quot;../SraRunTable_PRJNA728925.txt&quot;, sep=&quot;,&quot;)
clinical_df.index = clinical_df.Run
clinical_df.index.name = None
clinical_df.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Run</th>
      <th>Assay Type</th>
      <th>AvgSpotLen</th>
      <th>Bases</th>
      <th>BioProject</th>
      <th>BioSample</th>
      <th>Bytes</th>
      <th>Center Name</th>
      <th>Consent</th>
      <th>DATASTORE filetype</th>
      <th>...</th>
      <th>LibrarySelection</th>
      <th>LibrarySource</th>
      <th>Organism</th>
      <th>Platform</th>
      <th>ReleaseDate</th>
      <th>Sample Name</th>
      <th>source_name</th>
      <th>SRA Study</th>
      <th>Tissue</th>
      <th>viral_infection</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>SRR14509882</th>
      <td>SRR14509882</td>
      <td>RNA-Seq</td>
      <td>300</td>
      <td>11066499600</td>
      <td>PRJNA728925</td>
      <td>SAMN19107552</td>
      <td>3333844788</td>
      <td>GEO</td>
      <td>public</td>
      <td>sra,fastq</td>
      <td>...</td>
      <td>cDNA</td>
      <td>TRANSCRIPTOMIC</td>
      <td>Homo sapiens</td>
      <td>ILLUMINA</td>
      <td>2022-02-23T00:00:00Z</td>
      <td>GSM5289794</td>
      <td>Normal human urothelial cells</td>
      <td>SRP319465</td>
      <td>Ureter</td>
      <td>BKPyV (Dunlop) MOI=1</td>
    </tr>
    <tr>
      <th>SRR14509883</th>
      <td>SRR14509883</td>
      <td>RNA-Seq</td>
      <td>302</td>
      <td>8436386308</td>
      <td>PRJNA728925</td>
      <td>SAMN19107551</td>
      <td>2801216097</td>
      <td>GEO</td>
      <td>public</td>
      <td>fastq,sra</td>
      <td>...</td>
      <td>cDNA</td>
      <td>TRANSCRIPTOMIC</td>
      <td>Homo sapiens</td>
      <td>ILLUMINA</td>
      <td>2022-02-23T00:00:00Z</td>
      <td>GSM5289795</td>
      <td>Normal human urothelial cells</td>
      <td>SRP319465</td>
      <td>Ureter</td>
      <td>BKPyV (Dunlop) MOI=1</td>
    </tr>
    <tr>
      <th>SRR14509884</th>
      <td>SRR14509884</td>
      <td>RNA-Seq</td>
      <td>300</td>
      <td>9742943700</td>
      <td>PRJNA728925</td>
      <td>SAMN19107550</td>
      <td>3188119940</td>
      <td>GEO</td>
      <td>public</td>
      <td>fastq,sra</td>
      <td>...</td>
      <td>cDNA</td>
      <td>TRANSCRIPTOMIC</td>
      <td>Homo sapiens</td>
      <td>ILLUMINA</td>
      <td>2022-02-23T00:00:00Z</td>
      <td>GSM5289796</td>
      <td>Normal human urothelial cells</td>
      <td>SRP319465</td>
      <td>Ureter</td>
      <td>BKPyV (Dunlop) MOI=1</td>
    </tr>
    <tr>
      <th>SRR14509885</th>
      <td>SRR14509885</td>
      <td>RNA-Seq</td>
      <td>300</td>
      <td>11410353600</td>
      <td>PRJNA728925</td>
      <td>SAMN19107549</td>
      <td>3722953816</td>
      <td>GEO</td>
      <td>public</td>
      <td>sra,fastq</td>
      <td>...</td>
      <td>cDNA</td>
      <td>TRANSCRIPTOMIC</td>
      <td>Homo sapiens</td>
      <td>ILLUMINA</td>
      <td>2022-02-23T00:00:00Z</td>
      <td>GSM5289797</td>
      <td>Normal human urothelial cells</td>
      <td>SRP319465</td>
      <td>Ureter</td>
      <td>BKPyV (Dunlop) MOI=1</td>
    </tr>
    <tr>
      <th>SRR14509886</th>
      <td>SRR14509886</td>
      <td>RNA-Seq</td>
      <td>300</td>
      <td>9985769400</td>
      <td>PRJNA728925</td>
      <td>SAMN19107548</td>
      <td>3153799143</td>
      <td>GEO</td>
      <td>public</td>
      <td>sra,fastq</td>
      <td>...</td>
      <td>cDNA</td>
      <td>TRANSCRIPTOMIC</td>
      <td>Homo sapiens</td>
      <td>ILLUMINA</td>
      <td>2022-02-23T00:00:00Z</td>
      <td>GSM5289798</td>
      <td>Normal human urothelial cells</td>
      <td>SRP319465</td>
      <td>Ureter</td>
      <td>No infection</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 27 columns</p>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>dds = DeseqDataSet(
    counts=count_df,
    clinical=clinical_df,
    design_factors=&quot;viral_infection&quot;,  # compare samples based on the &quot;condition&quot;
    refit_cooks=True,
    n_cpus=8,
)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>dds.fit_size_factors()
dds.fit_genewise_dispersions()
dds.fit_dispersion_trend()
dds.fit_dispersion_prior()
dds.fit_MAP_dispersions()
dds.fit_LFC()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Fitting size factors...
... done in 0.05 seconds.

Fitting dispersions...
... done in 3.05 seconds.

Fitting dispersion trend curve...
... done in 6.95 seconds.

Fitting MAP dispersions...
... done in 3.70 seconds.

Fitting LFCs...
... done in 1.91 seconds.

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>dds.calculate_cooks()
if dds.refit_cooks:
    # Replace outlier counts
    dds.refit()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Refitting 357 outliers.

Fitting dispersions...
... done in 0.08 seconds.

Fitting MAP dispersions...
... done in 0.08 seconds.

Fitting LFCs...
... done in 0.08 seconds.

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>stat_res = DeseqStats(dds, alpha=0.05,contrast=[&quot;viral_infection&quot;,&quot;BKPyV (Dunlop) MOI=1&quot;,&quot;No infection&quot;])
stat_res.run_wald_test()
if stat_res.cooks_filter:
    stat_res._cooks_filtering()
stat_res.p_values
if stat_res.independent_filter:
    stat_res._independent_filtering()
else:
    stat_res._p_value_adjustment()
stat_res.summary()
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Running Wald tests...
... done in 3.29 seconds.

Log2 fold change &amp; Wald test p-value: viral_infection BKPyV (Dunlop) MOI=1 vs No infection
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>baseMean</th>
      <th>log2FoldChange</th>
      <th>lfcSE</th>
      <th>stat</th>
      <th>pvalue</th>
      <th>padj</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>A1BG</th>
      <td>65.042617</td>
      <td>-0.174495</td>
      <td>0.274393</td>
      <td>-0.635932</td>
      <td>0.524821</td>
      <td>0.998845</td>
    </tr>
    <tr>
      <th>A1BG-AS1</th>
      <td>60.814662</td>
      <td>-0.097512</td>
      <td>0.260997</td>
      <td>-0.373613</td>
      <td>0.708692</td>
      <td>0.998845</td>
    </tr>
    <tr>
      <th>A1CF</th>
      <td>0.234785</td>
      <td>-0.438579</td>
      <td>2.539843</td>
      <td>-0.172680</td>
      <td>0.862903</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>A2M</th>
      <td>3.841866</td>
      <td>0.917285</td>
      <td>0.822768</td>
      <td>1.114876</td>
      <td>0.264904</td>
      <td>0.965674</td>
    </tr>
    <tr>
      <th>A2M-AS1</th>
      <td>3.520622</td>
      <td>0.412318</td>
      <td>0.492133</td>
      <td>0.837817</td>
      <td>0.402133</td>
      <td>0.994437</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>ZYG11A</th>
      <td>4.020418</td>
      <td>0.989967</td>
      <td>0.716537</td>
      <td>1.381599</td>
      <td>0.167095</td>
      <td>0.906121</td>
    </tr>
    <tr>
      <th>ZYG11B</th>
      <td>1387.374886</td>
      <td>-0.189671</td>
      <td>0.126651</td>
      <td>-1.497585</td>
      <td>0.134241</td>
      <td>0.873862</td>
    </tr>
    <tr>
      <th>ZYX</th>
      <td>2956.789644</td>
      <td>-0.056822</td>
      <td>0.104073</td>
      <td>-0.545980</td>
      <td>0.585079</td>
      <td>0.998845</td>
    </tr>
    <tr>
      <th>ZZEF1</th>
      <td>1916.557304</td>
      <td>-0.129015</td>
      <td>0.110549</td>
      <td>-1.167041</td>
      <td>0.243194</td>
      <td>0.963318</td>
    </tr>
    <tr>
      <th>ZZZ3</th>
      <td>1102.880698</td>
      <td>-0.019320</td>
      <td>0.093984</td>
      <td>-0.205573</td>
      <td>0.837124</td>
      <td>0.998845</td>
    </tr>
  </tbody>
</table>
<p>29744 rows × 6 columns</p>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>stat_res_summary = stat_res.results_df
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sig_genes = stat_res_summary[stat_res_summary.padj&lt;0.05].index
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>lfc_key = stat_res_summary.log2FoldChange.to_dict()
</pre></div>
</div>
</div>
<section id="Visualize-the-DEG-information">
<h2>Visualize the DEG information<a class="headerlink" href="#Visualize-the-DEG-information" title="Link to this heading">¶</a></h2>
<p>Visualize the log2 fold changes of the genes in the certain pathway, and highlight DEGs.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import pykegg
import requests_cache
import numpy as np
from PIL import Image

## Cache all the downloaded
requests_cache.install_cache(&#39;pykegg_cache&#39;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>graph = pykegg.KGML_graph(pid=&quot;hsa04110&quot;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>nds = graph.get_nodes()
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>highlight_value = []
## If one of the symbols in identifiers in the nodes is in DEGs
for node in nds.graphics_name:
    in_node = [i.replace(&quot;...&quot;,&quot;&quot;) for i in node.split(&quot;,&quot;)]
    intersect = set(in_node) &amp; set(sig_genes)
    if len(intersect) &gt; 0:
        highlight_value.append(True)
    else:
        highlight_value.append(False)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>nds = pykegg.append_colors_continuous_values(nds, lfc_key)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>nds[&quot;highlight&quot;] = highlight_value
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>Image.fromarray(pykegg.overlay_opencv_image(nds,
                                            pid=&quot;hsa04110&quot;,
                                            highlight_nodes=&quot;highlight&quot;))
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/use-with-pydeseq2_18_0.png" src="_images/use-with-pydeseq2_18_0.png" />
</div>
</div>
</section>
<section id="deseq2_raw_map()-function">
<h2>deseq2_raw_map() function<a class="headerlink" href="#deseq2_raw_map()-function" title="Link to this heading">¶</a></h2>
<p>deseq2_raw_map serves as a wrapper function, consolidating the aforementioned example into a single function, with appending the colormap. It necessitates extracting PyDESeq2’s results_df as a DataFrame for input.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[61]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>pykegg.deseq2_raw_map(stat_res_summary, pid=&quot;hsa03460&quot;, legend_width=2)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[61]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/use-with-pydeseq2_20_0.png" src="_images/use-with-pydeseq2_20_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">pykegg</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">API Reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Example analysis</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="example_analysis.html">Notebooks</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="pykegg-intro.html">Simple example using pykegg</a></li>
<li class="toctree-l2"><a class="reference internal" href="pykegg-parallel-edges.html">Visualization of parallel edges using plotnine</a></li>
<li class="toctree-l2"><a class="reference internal" href="single-cell-example.html">Single-cell transcriptomics</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Example usage with PyDESeq2</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Visualize-the-DEG-information">Visualize the DEG information</a></li>
<li class="toctree-l3"><a class="reference internal" href="#deseq2_raw_map()-function">deseq2_raw_map() function</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="drawing-global-map.html">Drawing KEGG global and overview map</a></li>
<li class="toctree-l2"><a class="reference internal" href="vis-multiple-enrichment.html">Visuailzing multiple enrichment analysis results</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="example_analysis.html">Notebooks</a><ul>
      <li>Previous: <a href="single-cell-example.html" title="previous chapter">Single-cell transcriptomics</a></li>
      <li>Next: <a href="drawing-global-map.html" title="next chapter">Drawing KEGG global and overview map</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2024, Noriaki Sato.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 7.3.7</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="_sources/use-with-pydeseq2.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>