<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="wcGeneSummary">
<title>wcGeneSummary â€¢ wcGeneSummary</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="wcGeneSummary">
<meta property="og:description" content="wcGeneSummary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">wcGeneSummary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/basic_usage_of_wcGeneSummary.html">wcGeneSummary</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>wcGeneSummary</h1>
                        <h4 data-toc-skip class="author">Noriaki
Sato</h4>
            
      
      
      <div class="d-none name"><code>basic_usage_of_wcGeneSummary.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="producing-word-clouds">Producing word clouds<a class="anchor" aria-label="anchor" href="#producing-word-clouds"></a>
</h3>
<p>The basic usage of the package is producing a word cloud of gene
summaries by querying Entrez gene IDs. As many of the words are commonly
observed, you should limit word frequency by <code>excludeFreq</code>,
which is default to 5000.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Configure input genes</span>
<span class="va">inpSymbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ERCC1"</span>,<span class="st">"ERCC2"</span>,<span class="st">"ERCC3"</span>,<span class="st">"ERCC4"</span>,<span class="st">"ERCC5"</span>,<span class="st">"ERCC6"</span>,<span class="st">"ERCC8"</span><span class="op">)</span>

<span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwc</span><span class="op">$</span><span class="va">wc</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/basic-1.png" alt="A wordcloud of gene summaries." width="700"><p class="caption">
A wordcloud of gene summaries.
</p>
</div>
<p>It accepts values of the <code>wordcloud()</code> function.
<code>numWords</code> specifies how many words are to be shown on word
cloud. The function accepts the other parameters which is passed to
<code>wordcloud()</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Use palette from palettetown, palettetown::pokepal(150) (Mewtwo!)</span>
<span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, random.order<span class="op">=</span><span class="cn">FALSE</span>,
                     numWords<span class="op">=</span><span class="fl">100</span>, excludeFreq<span class="op">=</span><span class="fl">2000</span>,
                     colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>,
                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwc</span><span class="op">$</span><span class="va">wc</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/basic2-1.png" alt="A wordcloud of gene summaries, customized visualization" width="700"><p class="caption">
A wordcloud of gene summaries, customized visualization
</p>
</div>
<p>And it also returns a data frame consisting of frequency of each
term.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwc</span><span class="op">$</span><span class="va">df</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'Term frequencies.'</span>,
  row.names <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<table class="table">
<caption>Term frequencies.</caption>
<thead><tr class="header">
<th align="left">word</th>
<th align="right">freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">repair</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">DNA</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">syndrome</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">excision</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">transcription</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">cockayne</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
<p>N-gram is supported by library <code>tm</code>, specified by
<code>ngram</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, random.order<span class="op">=</span><span class="cn">FALSE</span>,
                     numWords<span class="op">=</span><span class="fl">100</span>, ngram<span class="op">=</span><span class="fl">2</span>,
                     colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>, rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span></code></pre>
<pre><code><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></code></pre>
<pre><code><span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwc2</span><span class="op">$</span><span class="va">wc</span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ngramwc-1.png" width="700"></p>
<p>Word clustering (pvclust) based on the occurrence in the text can be
visualized by specifying <code>tag=TRUE</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwcl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,
                     numWords<span class="op">=</span><span class="fl">30</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, pal <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>,
                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span></code></pre>
<pre><code><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></code></pre>
<pre><code><span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## Bootstrap (r = 0.5)... Done.</span>
<span class="co">## Bootstrap (r = 0.6)... Done.</span>
<span class="co">## Bootstrap (r = 0.7)... Done.</span>
<span class="co">## Bootstrap (r = 0.8)... Done.</span>
<span class="co">## Bootstrap (r = 0.9)... Done.</span>
<span class="co">## Bootstrap (r = 1.0)... Done.</span>
<span class="co">## Bootstrap (r = 1.1)... Done.</span>
<span class="co">## Bootstrap (r = 1.2)... Done.</span>
<span class="co">## Bootstrap (r = 1.3)... Done.</span>
<span class="co">## Bootstrap (r = 1.4)... Done.</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwcl</span><span class="op">$</span><span class="va">wc</span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagging-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwcl</span><span class="op">$</span><span class="va">pvcl</span></code></pre></div>
<pre><code><span class="co">## $clusters</span>
<span class="co">## $clusters[[1]]</span>
<span class="co">## [1] "pigmentosum" "xeroderma"  </span>
<span class="co">## </span>
<span class="co">## $clusters[[2]]</span>
<span class="co">##  [1] "activity"                  "atpdependent"             </span>
<span class="co">##  [3] "basal"                     "cerebrooculofacioskeletal"</span>
<span class="co">##  [5] "characterized"             "complementation"          </span>
<span class="co">##  [7] "damage"                    "defects"                  </span>
<span class="co">##  [9] "ercc"                      "exon"                     </span>
<span class="co">## [11] "factor"                    "forms"                    </span>
<span class="co">## [13] "functions"                 "helicase"                 </span>
<span class="co">## [15] "heterodimeric"             "incision"                 </span>
<span class="co">## [17] "including"                 "interacts"                </span>
<span class="co">## [19] "nucleotide"                "product"                  </span>
<span class="co">## [21] "transcriptioncoupled"     </span>
<span class="co">## </span>
<span class="co">## $clusters[[3]]</span>
<span class="co">## [1] "dna"    "repair"</span>
<span class="co">## </span>
<span class="co">## $clusters[[4]]</span>
<span class="co">## [1] "cockayne"      "excision"      "syndrome"      "transcription"</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## $edges</span>
<span class="co">## [1]  6 22 23 25</span></code></pre>
</div>
<div class="section level3">
<h3 id="producing-correlation-networks">Producing correlation networks<a class="anchor" aria-label="anchor" href="#producing-correlation-networks"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span>
<span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">net</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/basic3-1.png" alt="A correlation network of gene summaries" width="700"><p class="caption">
A correlation network of gene summaries
</p>
</div>
<p>The edge label can be shown by <code>edgeLabel=TRUE</code>.<br>
The threhsold of correlation can be specified by
<code>corThresh</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, edgeLabel<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">net</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/ngramnet-1.png" alt="A correlation network of gene summaries" width="700"><p class="caption">
A correlation network of gene summaries
</p>
</div>
<p>The clustering results based on <code>pvclust</code> can be shown
also on the correlation network.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>,
                     numWords<span class="op">=</span><span class="fl">20</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## Bootstrap (r = 0.5)... Done.</span>
<span class="co">## Bootstrap (r = 0.6)... Done.</span>
<span class="co">## Bootstrap (r = 0.7)... Done.</span>
<span class="co">## Bootstrap (r = 0.8)... Done.</span>
<span class="co">## Bootstrap (r = 0.9)... Done.</span>
<span class="co">## Bootstrap (r = 1.0)... Done.</span>
<span class="co">## Bootstrap (r = 1.1)... Done.</span>
<span class="co">## Bootstrap (r = 1.2)... Done.</span>
<span class="co">## Bootstrap (r = 1.3)... Done.</span>
<span class="co">## Bootstrap (r = 1.4)... Done.</span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">net</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/tagnet-1.png" alt="A correlation network of genes with cluster information" width="700"><p class="caption">
A correlation network of genes with cluster information
</p>
</div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">pvcl</span></code></pre></div>
<pre><code><span class="co">## $clusters</span>
<span class="co">## $clusters[[1]]</span>
<span class="co">## [1] "pigmentosum" "xeroderma"  </span>
<span class="co">## </span>
<span class="co">## $clusters[[2]]</span>
<span class="co">##  [1] "activity"             "atpdependent"         "complementation"     </span>
<span class="co">##  [4] "defects"              "factor"               "functions"           </span>
<span class="co">##  [7] "incision"             "interacts"            "nucleotide"          </span>
<span class="co">## [10] "product"              "transcriptioncoupled"</span>
<span class="co">## </span>
<span class="co">## $clusters[[3]]</span>
<span class="co">## [1] "dna"    "repair"</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## $edges</span>
<span class="co">## [1]  2 12 13</span></code></pre>
<p>The genes associated with the corresponding words can be shown by
specifying <code>genePlot=TRUE</code>. The edges connecting word to
corresponding gene are shown.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,
                     numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## Bootstrap (r = 0.5)... Done.</span>
<span class="co">## Bootstrap (r = 0.6)... Done.</span>
<span class="co">## Bootstrap (r = 0.7)... Done.</span>
<span class="co">## Bootstrap (r = 0.8)... Done.</span>
<span class="co">## Bootstrap (r = 0.9)... Done.</span>
<span class="co">## Bootstrap (r = 1.0)... Done.</span>
<span class="co">## Bootstrap (r = 1.1)... Done.</span>
<span class="co">## Bootstrap (r = 1.2)... Done.</span>
<span class="co">## Bootstrap (r = 1.3)... Done.</span>
<span class="co">## Bootstrap (r = 1.4)... Done.</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">net</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/gplot-1.png" alt="A correlation network of gene summaries with associated genes" width="700"><p class="caption">
A correlation network of gene summaries with associated genes
</p>
</div>
<p>The associated enriched pathways (if present) can be shown by
specifying <code>genePathPlot</code>, using <code>ggforce</code>. In
this option, the function first performs over-representation analysis on
the whole gene set, and plot enriched terms for included genes in the
plot.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, genePathPlot<span class="op">=</span><span class="st">"kegg"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,
                     tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>, genePathPlotSig<span class="op">=</span><span class="fl">0.05</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## Bootstrap (r = 0.5)... Done.</span>
<span class="co">## Bootstrap (r = 0.6)... Done.</span>
<span class="co">## Bootstrap (r = 0.7)... Done.</span>
<span class="co">## Bootstrap (r = 0.8)... Done.</span>
<span class="co">## Bootstrap (r = 0.9)... Done.</span>
<span class="co">## Bootstrap (r = 1.0)... Done.</span>
<span class="co">## Bootstrap (r = 1.1)... Done.</span>
<span class="co">## Bootstrap (r = 1.2)... Done.</span>
<span class="co">## Bootstrap (r = 1.3)... Done.</span>
<span class="co">## Bootstrap (r = 1.4)... Done.</span>
<span class="co">## Found 3 enriched term ...</span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">net</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/gpplot-1.png" alt="A correlation network of gene summaries with associated genes" width="700"><p class="caption">
A correlation network of gene summaries with associated genes
</p>
</div>
</div>
<div class="section level3">
<h3 id="text-over-represenatation-analysis-experimental">Text over represenatation analysis (experimental)<a class="anchor" aria-label="anchor" href="#text-over-represenatation-analysis-experimental"></a>
</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">keggPathways</span> <span class="op">&lt;-</span> <span class="va">org.Hs.egPATH2EG</span>
<span class="va">mappedKeys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html" class="external-link">mappedkeys</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">)</span>
<span class="va">keggList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/BimapFormatting.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">[</span><span class="va">mappedKeys</span><span class="op">]</span><span class="op">)</span>

<span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`00785`</span> <span class="co"># Lipoic acid metabolism</span>
<span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ora-1.png" width="700"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            lipoic              step              acid lipoateactivating </span>
<span class="co">##      1.701356e-14      8.782627e-08      5.708708e-07      1.507684e-04 </span>
<span class="co">##  lipoatedependent            lipoyl </span>
<span class="co">##      1.507684e-04      1.507684e-04</span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`05150`</span> <span class="co"># Staphylococcus aureus infection</span>
<span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ora-2.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##        chain   complement        class         exon      peptide         beta </span>
<span class="co">## 1.982266e-81 6.443621e-74 1.233313e-45 1.278631e-38 8.506250e-34 9.352422e-33</span></code></pre>
<p>Filter words using ORA threshold and frequency threshold by setting
<code>ora=TRUE</code>.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, ora<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## performing ORA</span>
<span class="co">## filtered 147 words (ORA) ...</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span><span class="op">$</span><span class="va">net</span></code></pre></div>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/oranet-1.png" alt="A correlation network of gene summaries (ORA)" width="700"><p class="caption">
A correlation network of gene summaries (ORA)
</p>
</div>
</div>
<div class="section level3">
<h3 id="visualization-of-pubmed-information-">Visualization of PubMed information.<a class="anchor" aria-label="anchor" href="#visualization-of-pubmed-information-"></a>
</h3>
<p>Using <code>easyPubMed</code>, one can perform the same analysis on
PubMed text.</p>
</div>
</div>
<div class="section level2">
<h2 id="custom-usage">Custom usage<a class="anchor" aria-label="anchor" href="#custom-usage"></a>
</h2>
<div class="section level3">
<h3 id="recluster-the-cluster-using-word-information">Recluster the cluster using word information<a class="anchor" aria-label="anchor" href="#recluster-the-cluster-using-word-information"></a>
</h3>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">simExample</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/returnSim.html">returnSim</a></span><span class="op">(</span><span class="fu"><a href="../reference/returnExample.html">returnExample</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">color</span>, keyType<span class="op">=</span><span class="st">"ENSEMBL"</span>, ora<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Number of clusters: 3</span>
<span class="co">## 1</span>
<span class="co">## Input genes: 12</span>
<span class="co">## Converted input genes: 12</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## performing ORA</span>
<span class="co">## filtered 109 words (ORA) ...</span>
<span class="co">## 2</span>
<span class="co">## Input genes: 13</span>
<span class="co">## Converted input genes: 13</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## performing ORA</span>
<span class="co">## filtered 237 words (ORA) ...</span>
<span class="co">## 3</span>
<span class="co">## Input genes: 7</span>
<span class="co">## Converted input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span>
<span class="co">## performing ORA</span>
<span class="co">## filtered 147 words (ORA) ...</span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html" class="external-link">heatmap</a></span><span class="op">(</span><span class="va">simExample</span><span class="op">)</span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/textclus-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="interactive-inspection-of-bayesian-network-annotated-by-words">Interactive inspection of Bayesian network annotated by words<a class="anchor" aria-label="anchor" href="#interactive-inspection-of-bayesian-network-annotated-by-words"></a>
</h3>
<p>In this example, a Bayesian network showing the module eigengenes
relationship are inferred using <code>boot.strength</code> function in
<code>bnlearn</code> from the weighted gene correlation network analysis
(WGCNA) results. The modules are annotated by word clouds produced by
<code><a href="../reference/wcGeneSummary.html">wcGeneSummary()</a></code>, and can be exported to the format of
<code>Cytoscape.js</code> or <code>vis.js</code>. In this way, module
relationship can be interactively inspected with the functional
implications.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## In this example, we simulate WGCNA results.</span>
<span class="co">## you can just use results from WGCNA.</span>
<span class="co">## Assuming WGCNA results are stored in `mod`</span>
<span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">wcGeneSummary</span><span class="fu">::</span><span class="fu"><a href="../reference/returnExample.html">returnExample</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">MEs</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">MEs</span>
<span class="va">modColors</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">colors</span>
<span class="va">ensg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">modColors</span><span class="op">)</span>

<span class="co"># library(bnlearn)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span>

<span class="co">## Replace like boot.strength(mod$MEs, R=500, algorithm = "hc")</span>
<span class="co"># dag &lt;- model2network("[ME1][ME2|ME1]") # If using bnlearn</span>
<span class="va">dag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_literal.html" class="external-link">graph_from_literal</a></span><span class="op">(</span> <span class="va">ME1</span><span class="op">-</span><span class="op">+</span><span class="va">ME2</span>, <span class="va">ME1</span><span class="op">-</span><span class="op">+</span><span class="va">ME3</span> <span class="op">)</span>

<span class="co">## Convert to igraph</span>
<span class="co"># g &lt;- as.igraph(dag)</span>
<span class="va">g</span> <span class="op">&lt;-</span> <span class="va">dag</span>

<span class="co">## Assign edge attributes</span>
<span class="co">## Skip, if you perform boot.strength, the edge attributes can be added from the result</span>
<span class="co"># el &lt;- data.frame(as_edgelist(g))</span>
<span class="co"># colnames(el) &lt;- c("from","to")</span>
<span class="co"># el &lt;- left_join(el, bs)</span>
<span class="co"># E(g)$strength &lt;- el$strength</span>
<span class="co"># E(g)$direction &lt;- el$direction</span>

<span class="co">## Node attributes</span>
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">stripName</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"ME"</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>
<span class="va">sizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">modColors</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sizes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">stripName</span><span class="op">]</span><span class="op">)</span>

<span class="co">## Directory to save images and a script</span>
<span class="va">rootDir</span> <span class="op">&lt;-</span> <span class="st">"./"</span>
<span class="va">netDir</span> <span class="op">&lt;-</span> <span class="st">"visCyjs"</span>
<span class="va">imageDir</span> <span class="op">&lt;-</span> <span class="st">"images"</span>

<span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span>, <span class="st">"/"</span>, <span class="va">imageDir</span><span class="op">)</span><span class="op">)</span>

<span class="va">images</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">plotType</span> <span class="op">&lt;-</span> <span class="st">"bar"</span>
<span class="va">numLim</span> <span class="op">&lt;-</span> <span class="fl">200</span> <span class="co"># limit for gene number</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span><span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
    <span class="va">i</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"ME"</span>,<span class="st">""</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span> <span class="co"># strip ME</span>

    <span class="va">queries</span> <span class="op">&lt;-</span> <span class="va">ensg</span><span class="op">[</span><span class="va">modColors</span><span class="op">==</span><span class="va">i</span><span class="op">]</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queries</span><span class="op">)</span><span class="op">&gt;</span><span class="va">numLim</span><span class="op">)</span> <span class="op">{</span>
        <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="st">"Sampling random genes"</span><span class="op">)</span>
        <span class="va">queries</span> <span class="op">&lt;-</span> <span class="va">queries</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queries</span><span class="op">)</span>, <span class="va">numLim</span><span class="op">)</span><span class="op">]</span> <span class="co">## Temporary restrict to randomly chosen genes, should be replaced to like kME values</span>
    <span class="op">}</span>
    
    <span class="co">## Convert to ENTREZ</span>
    <span class="va">entre</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"ENSEMBL"</span>,
        keys <span class="op">=</span> <span class="va">queries</span>, columns <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">ENTREZID</span>
    
    <span class="kw">if</span> <span class="op">(</span><span class="va">plotType</span><span class="op">==</span><span class="st">"bar"</span><span class="op">)</span><span class="op">{</span>
        <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeBar.html">makeBar</a></span><span class="op">(</span><span class="va">entre</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span> <span class="co"># get barplot</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span> <span class="co">## If wordcloud</span>
        <span class="co"># A &lt;- wcGeneSummary(entre,</span>
        <span class="co">#                    madeUpper=c("dna","rna",</span>
        <span class="co">#                                tolower(keys(org.Hs.eg.db, keytype="SYMBOL"))))</span>
        <span class="co"># palNum &lt;- sample(1:151,1) # palettetown</span>
        <span class="co"># </span>
        <span class="co"># ## This time use ggwordcloud()</span>
        <span class="co"># plt &lt;- ggwordcloud::ggwordcloud(A$df$word, A$df$freq,</span>
        <span class="co">#                      shape="circle", min.freq = 10,</span>
        <span class="co">#                      rot.per = 0.5, random.order = FALSE,</span>
        <span class="co">#                      colors = pokepal(palNum))+</span>
        <span class="co">#          scale_size_area(max_size = 40)</span>
    <span class="op">}</span>
    <span class="co">## Save images</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span>, <span class="st">"/"</span>, <span class="va">imageDir</span>, <span class="st">"/"</span>, <span class="va">i</span> ,<span class="st">".png"</span><span class="op">)</span>,
           <span class="va">plt</span>, dpi<span class="op">=</span><span class="fl">300</span>, width<span class="op">=</span><span class="fl">10</span>, height<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
    <span class="co">## Store image dir</span>
    <span class="va">images</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">images</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">imageDir</span>, <span class="st">"/"</span>, <span class="va">i</span> ,<span class="st">".png"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<pre><code><span class="co">## [1] "ME1"</span>
<span class="co">## Input genes: 12</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<pre><code><span class="co">## [1] "ME2"</span>
<span class="co">## Input genes: 13</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<pre><code><span class="co">## [1] "ME3"</span>
<span class="co">## Input genes: 7</span>
<span class="co">## filtered 62 words (frequency) ...</span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">image</span> <span class="op">&lt;-</span> <span class="va">images</span>

<span class="co">## Node shape</span>
<span class="kw">if</span> <span class="op">(</span><span class="va">plotType</span><span class="op">==</span><span class="st">"bar"</span><span class="op">)</span><span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rectangle"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> 
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"circle"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="co">## Scale the node size</span>
<span class="va">sizeMin</span> <span class="op">&lt;-</span> <span class="fl">50</span>
<span class="va">sizeMax</span> <span class="op">&lt;-</span> <span class="fl">200</span>
<span class="va">rawMin</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span>
<span class="va">rawMax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span>
<span class="va">scf</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">sizeMax</span><span class="op">-</span><span class="va">sizeMin</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="va">rawMax</span><span class="op">-</span><span class="va">rawMin</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">&lt;-</span> <span class="va">scf</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">+</span> <span class="va">sizeMin</span> <span class="op">-</span> <span class="va">scf</span> <span class="op">*</span> <span class="va">rawMin</span>

<span class="co">## Export</span>
<span class="fu"><a href="../reference/exportCyjs.html">exportCyjs</a></span><span class="op">(</span><span class="va">g</span>, <span class="va">rootDir</span>, <span class="va">netDir</span><span class="op">)</span>
<span class="co"># or, exportVisjs(g, rootDir, netDir)</span></code></pre></div>
<p>Use like <code>http-server</code> in the directory containing a
exported JavaScript, and interactively inspect the module relationship
with word information.</p>
</div>
<div class="section level3">
<h3 id="annotating-module-eigengenes-dendrogram">Annotating module eigengenes dendrogram<a class="anchor" aria-label="anchor" href="#annotating-module-eigengenes-dendrogram"></a>
</h3>
<p>As users of WGCNA typically plot dendrogramm and heatmap of module
eigengenes using <code>plotEigengeneNetworks</code>, it is useful to
combine with wcGeneSummary, which plot additional word information on a
dendrogram with one line.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># WGCNA::plotEigengeneNetworks(mod$MEs, mod$colors, plotHeatmaps = FALSE)</span>
<span class="fu"><a href="../reference/plotEigengeneNetworksWithWords.html">plotEigengeneNetworksWithWords</a></span><span class="op">(</span><span class="va">MEs</span>, <span class="va">modColors</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Bootstrap (r = 0.5)... Done.</span>
<span class="co">## Bootstrap (r = 0.6)... Done.</span>
<span class="co">## Bootstrap (r = 0.7)... Done.</span>
<span class="co">## Bootstrap (r = 0.8)... Done.</span>
<span class="co">## Bootstrap (r = 0.9)... Done.</span>
<span class="co">## Bootstrap (r = 1.0)... Done.</span>
<span class="co">## Bootstrap (r = 1.1)... Done.</span>
<span class="co">## Bootstrap (r = 1.2)... Done.</span>
<span class="co">## Bootstrap (r = 1.3)... Done.</span>
<span class="co">## Bootstrap (r = 1.4)... Done.</span></code></pre>
<div class="figure">
<img src="basic_usage_of_wcGeneSummary_files/figure-html/wgcna-1.png" alt="A ME dendrogram with words" width="700"><p class="caption">
A ME dendrogram with words
</p>
</div>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.2.0 (2022-04-22 ucrt)</span>
<span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">## Running under: Windows 10 x64 (build 19044)</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] LC_COLLATE=Japanese_Japan.utf8  LC_CTYPE=Japanese_Japan.utf8   </span>
<span class="co">## [3] LC_MONETARY=Japanese_Japan.utf8 LC_NUMERIC=C                   </span>
<span class="co">## [5] LC_TIME=Japanese_Japan.utf8    </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] igraph_1.2.11        ggraph_2.0.5         ggplot2_3.3.5       </span>
<span class="co">##  [4] org.Hs.eg.db_3.14.0  AnnotationDbi_1.57.1 IRanges_2.29.1      </span>
<span class="co">##  [7] S4Vectors_0.33.10    Biobase_2.55.0       BiocGenerics_0.41.2 </span>
<span class="co">## [10] wcGeneSummary_0.99.0 BiocStyle_2.24.0    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2             tidyselect_1.1.2       RSQLite_2.2.10        </span>
<span class="co">##   [4] htmlwidgets_1.5.4      grid_4.2.0             BiocParallel_1.29.15  </span>
<span class="co">##   [7] scatterpie_0.1.7       munsell_0.5.0          codetools_0.2-18      </span>
<span class="co">##  [10] ragg_1.2.2             withr_2.4.3            colorspace_2.0-3      </span>
<span class="co">##  [13] GOSemSim_2.21.1        NLP_0.2-1              highr_0.9             </span>
<span class="co">##  [16] knitr_1.37             rstudioapi_0.13        DOSE_3.21.2           </span>
<span class="co">##  [19] GeneSummary_0.99.3     labeling_0.4.2         slam_0.1-50           </span>
<span class="co">##  [22] GenomeInfoDbData_1.2.7 polyclip_1.10-0        bit64_4.0.5           </span>
<span class="co">##  [25] farver_2.1.0           rprojroot_2.0.2        downloader_0.4        </span>
<span class="co">##  [28] vctrs_0.4.1            treeio_1.19.1          generics_0.1.2        </span>
<span class="co">##  [31] xfun_0.30              R6_2.5.1               GenomeInfoDb_1.31.4   </span>
<span class="co">##  [34] graphlayouts_0.8.0     concaveman_1.1.0       ISOcodes_2022.01.10   </span>
<span class="co">##  [37] bitops_1.0-7           cachem_1.0.6           fgsea_1.21.0          </span>
<span class="co">##  [40] gridGraphics_0.5-1     assertthat_0.2.1       promises_1.2.0.1      </span>
<span class="co">##  [43] scales_1.1.1           enrichplot_1.16.1      gtable_0.3.0          </span>
<span class="co">##  [46] tidygraph_1.2.0        rlang_1.0.4            systemfonts_1.0.4     </span>
<span class="co">##  [49] GlobalOptions_0.1.2    splines_4.2.0          lazyeval_0.2.2        </span>
<span class="co">##  [52] stopwords_2.3          wordcloud_2.6          checkmate_2.0.0       </span>
<span class="co">##  [55] BiocManager_1.30.16    yaml_2.3.5             reshape2_1.4.4        </span>
<span class="co">##  [58] backports_1.4.1        httpuv_1.6.5           pvclust_2.2-0         </span>
<span class="co">##  [61] qvalue_2.27.0          clusterProfiler_4.3.2  tools_4.2.0           </span>
<span class="co">##  [64] bookdown_0.26          ggplotify_0.1.0        ellipsis_0.3.2        </span>
<span class="co">##  [67] jquerylib_0.1.4        RColorBrewer_1.1-2     ggdendro_0.1.23       </span>
<span class="co">##  [70] Rcpp_1.0.8             plyr_1.8.6             base64enc_0.1-3       </span>
<span class="co">##  [73] zlibbioc_1.41.0        purrr_0.3.4            RCurl_1.98-1.6        </span>
<span class="co">##  [76] GetoptLong_1.0.5       viridis_0.6.2          cowplot_1.1.1         </span>
<span class="co">##  [79] ggrepel_0.9.1          fs_1.5.2               magrittr_2.0.2        </span>
<span class="co">##  [82] data.table_1.14.2      DO.db_2.9              reactome.db_1.79.0    </span>
<span class="co">##  [85] patchwork_1.1.1        mime_0.12              evaluate_0.15         </span>
<span class="co">##  [88] xtable_1.8-4           gridExtra_2.3          compiler_4.2.0        </span>
<span class="co">##  [91] bugsigdbr_1.2.2        tibble_3.1.8           V8_4.1.0              </span>
<span class="co">##  [94] crayon_1.5.0           shadowtext_0.1.1       htmltools_0.5.2       </span>
<span class="co">##  [97] ggfun_0.0.5            later_1.3.0            tidyr_1.2.0           </span>
<span class="co">## [100] aplot_0.1.2            ReactomePA_1.40.0      DBI_1.1.2             </span>
<span class="co">## [103] tweenr_1.0.2           MASS_7.3-56            rappdirs_0.3.3        </span>
<span class="co">## [106] Matrix_1.4-1           cli_3.2.0              parallel_4.2.0        </span>
<span class="co">## [109] pkgconfig_2.0.3        pkgdown_2.0.3          xml2_1.3.3            </span>
<span class="co">## [112] ggtree_3.3.1           bslib_0.3.1            XVector_0.35.0        </span>
<span class="co">## [115] yulab.utils_0.0.4      stringr_1.4.0          digest_0.6.29         </span>
<span class="co">## [118] graph_1.73.0           Biostrings_2.63.1      tm_0.7-8              </span>
<span class="co">## [121] rmarkdown_2.14         fastmatch_1.1-3        tidytree_0.3.8        </span>
<span class="co">## [124] easyPubMed_2.13        dendextend_1.15.2      cyjShiny_1.0.34       </span>
<span class="co">## [127] curl_4.3.2             shiny_1.7.1            graphite_1.41.0       </span>
<span class="co">## [130] rjson_0.2.21           lifecycle_1.0.1        nlme_3.1-157          </span>
<span class="co">## [133] jsonlite_1.8.0         desc_1.4.0             viridisLite_0.4.0     </span>
<span class="co">## [136] fansi_1.0.2            pillar_1.7.0           lattice_0.20-45       </span>
<span class="co">## [139] KEGGREST_1.36.0        fastmap_1.1.0          httr_1.4.2            </span>
<span class="co">## [142] GO.db_3.14.0           glue_1.6.2             png_0.1-7             </span>
<span class="co">## [145] bit_4.0.4              ggforce_0.3.3          stringi_1.7.6         </span>
<span class="co">## [148] sass_0.4.0             blob_1.2.2             textshaping_0.3.6     </span>
<span class="co">## [151] memoise_2.0.1          dplyr_1.0.9            ape_5.6-1</span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
