<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="wcGeneSummary">
<title>wcGeneSummary â€¢ wcGeneSummary</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="wcGeneSummary">
<meta property="og:description" content="wcGeneSummary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">wcGeneSummary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/basic_usage_of_wcGeneSummary.html">wcGeneSummary</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>wcGeneSummary</h1>
                        <h4 data-toc-skip class="author">Noriaki
Sato</h4>
            
      
      
      <div class="d-none name"><code>basic_usage_of_wcGeneSummary.Rmd</code></div>
    </div>

    
    
<p>This package aims to visualize the word and text information
contained in the gene or the other omics identifiers such as microbiome,
and identify important words among the clusters, and compare the
clusters based on those information. It contributes to understanding the
gene clusters and aid in easy interpretation and visualization.</p>
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<p>Load the package and the database for converting identifiers. In this
example, we use mostly human-derived data, and use
<code>org.Hs.eg.db</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="producing-word-clouds">Producing word clouds<a class="anchor" aria-label="anchor" href="#producing-word-clouds"></a>
</h3>
<p>We use ERCC genes as input. The basic usage of the package is
producing a word cloud of gene summaries by querying gene IDs, default
to symbol. The other type can be set by <code>keyType</code>. As many of
the words are commonly observed, you should limit word frequency by
<code>excludeFreq</code>, which is default to 2000. TF-IDF on the all
the summary is precomputed, and <code>excludeTfIdf</code> can be
specified too.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Configure input genes</span></span>
<span><span class="va">inpSymbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ERCC1"</span>,<span class="st">"ERCC2"</span>,<span class="st">"ERCC3"</span>,<span class="st">"ERCC4"</span>,<span class="st">"ERCC5"</span>,<span class="st">"ERCC6"</span>,<span class="st">"ERCC8"</span><span class="op">)</span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic-1.png" width="768"></p>
<p>It accepts values of the <code>wordcloud()</code> function.
<code>numWords</code> specifies how many words are to be shown on word
cloud. The words are ordered by their frequency, and the subset of
1:<code>numWords</code> is used to downstream visualization.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use palette from palettetown, palettetown::pokepal(150) (Mewtwo!)</span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, random.order<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>                     excludeTfIdf <span class="op">=</span> <span class="fl">150</span>,</span>
<span>                     colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 104 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic2-1.png" width="768"></p>
<p>It also returns a data frame consisting of frequency of each
term.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwc</span><span class="op">$</span><span class="va">df</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'Term frequencies.'</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Term frequencies.</caption>
<thead><tr class="header">
<th align="left">word</th>
<th align="right">freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">repair</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">DNA</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">syndrome</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">excision</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">cockayne</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">pigmentosum</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
<p>N-gram is supported by library <code>tm</code>, specified by
<code>ngram</code>. Default is <code>1</code>, and the example
specifying <code>2</code> is shown below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, ngram<span class="op">=</span><span class="fl">2</span>, numWords<span class="op">=</span><span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc2</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ngramwc-1.png" width="768"></p>
<p>Using <code>clusterProfiler</code> functions, one can use enriched
pathway names for visualization. The <code>enrich</code> option can be
specified for <code>'kegg'</code> or <code>'reactome'</code>, this time
we specify <code>'reactome'</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, enrich<span class="op">=</span><span class="st">"reactome"</span>, tfidf<span class="op">=</span><span class="cn">TRUE</span>, numWords<span class="op">=</span><span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## performing enrichment analysis ...</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc3</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ngrampath-1.png" width="768"></p>
<p>One of the main questions is which words can be clustered together
among the words contained in the queried gene cluster. Word clustering
(<code>pvclust</code>) and identified significant clusters based on the
occurrence in the text can be visualized by specifying
<code>tag=TRUE</code>. For the significance threshold,
<code>pvclAlpha</code> can be specified. The default parameters perform
<code>pvclust</code> on the subset of dataset for words with high
frequency specified by <code>numWords</code>. If one want to perform on
whole matrix (which is a natural way), <code>tagWhole=TRUE</code> can be
specified, although it is computationally intensive.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Cluster on subset of words</span></span>
<span><span class="va">gwcl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                     pal <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwcl</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagging-1.png" width="768"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwcl</span><span class="op">$</span><span class="va">pvcl</span></span></code></pre></div>
<pre><code><span><span class="co">## $clusters</span></span>
<span><span class="co">## $clusters[[1]]</span></span>
<span><span class="co">## [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[2]]</span></span>
<span><span class="co">##  [1] "activity"                  "atpdependent"             </span></span>
<span><span class="co">##  [3] "basal"                     "cerebrooculofacioskeletal"</span></span>
<span><span class="co">##  [5] "characterized"             "complementation"          </span></span>
<span><span class="co">##  [7] "damage"                    "defects"                  </span></span>
<span><span class="co">##  [9] "exon"                      "factor"                   </span></span>
<span><span class="co">## [11] "forms"                     "functions"                </span></span>
<span><span class="co">## [13] "helicase"                  "heterodimeric"            </span></span>
<span><span class="co">## [15] "incision"                  "including"                </span></span>
<span><span class="co">## [17] "interacts"                 "light"                    </span></span>
<span><span class="co">## [19] "nucleotide"                "product"                  </span></span>
<span><span class="co">## [21] "transcriptioncoupled"     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[3]]</span></span>
<span><span class="co">## [1] "dna"    "repair"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $edges</span></span>
<span><span class="co">## [1]  6 22 23</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Cluster on whole matrix</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">4</span>, <span class="st">"PuOr"</span><span class="op">)</span> </span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="va">pal</span><span class="op">)</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gwclWhole</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, tagWhole<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                     pal <span class="op">=</span> <span class="va">pal</span>,</span>
<span>                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.69)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.19)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwclWhole</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagging-2.png" width="768"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwclWhole</span><span class="op">$</span><span class="va">pvcl</span></span></code></pre></div>
<pre><code><span><span class="co">## $clusters</span></span>
<span><span class="co">## $clusters[[1]]</span></span>
<span><span class="co">##  [1] "abnormally"        "active"            "csb"              </span></span>
<span><span class="co">##  [4] "defective"         "disease"           "hereditary"       </span></span>
<span><span class="co">##  [7] "identified"        "iih"               "p44"              </span></span>
<span><span class="co">## [10] "radiation"         "repeat"            "sensitive"        </span></span>
<span><span class="co">## [13] "transcriptionally" "ultraviolet"      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[2]]</span></span>
<span><span class="co">##  [1] "activates"       "adjacent"        "atpase"          "atpstimulated"  </span></span>
<span><span class="co">##  [5] "dnabinding"      "downstream"      "encode"          "formation"      </span></span>
<span><span class="co">##  [9] "frame"           "fusion"          "geneid"          "individual"     </span></span>
<span><span class="co">## [13] "occurs"          "polyadenylation" "promote"         "reading"        </span></span>
<span><span class="co">## [17] "sequence"        "shares"          "sites"          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[3]]</span></span>
<span><span class="co">## [1] "class" "tfiih"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[4]]</span></span>
<span><span class="co">## [1] "complementation" "defects"        </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[5]]</span></span>
<span><span class="co">## [1] "eme1"      "ercc1"     "structure" "xp6"      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[6]]</span></span>
<span><span class="co">## [1] "orf"               "piggybackderived3" "splice"           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[7]]</span></span>
<span><span class="co">## [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[8]]</span></span>
<span><span class="co">## [1] "interacts" "type"     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[9]]</span></span>
<span><span class="co">## [1] "incision"  "including" "light"    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[10]]</span></span>
<span><span class="co">## [1] "transcriptioncoupled" "upstream"            </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[11]]</span></span>
<span><span class="co">##  [1] "basic"            "bivm"             "cachexia"         "cancer"          </span></span>
<span><span class="co">##  [5] "cellular"         "characterized"    "cognitive"        "develop"         </span></span>
<span><span class="co">##  [9] "development"      "disability"       "disorder"         "exists"          </span></span>
<span><span class="co">## [13] "exposure"         "function"         "growth"           "hypersensitivity"</span></span>
<span><span class="co">## [17] "immunoglobulin"   "increased"        "motif"            "neighboring"     </span></span>
<span><span class="co">## [21] "patients"         "processes"        "read"             "referred"        </span></span>
<span><span class="co">## [25] "severe"           "singlestrand"     "skin"             "specific"        </span></span>
<span><span class="co">## [29] "susceptibility"   "uvinduced"        "variable"         "vii"             </span></span>
<span><span class="co">## [33] "xp7"             </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[12]]</span></span>
<span><span class="co">##  [1] "atpdependent"        "basal"               "belongs"            </span></span>
<span><span class="co">##  [4] "btf2tfiih"           "cancerprone"         "disorders"          </span></span>
<span><span class="co">##  [7] "factor"              "helicase"            "helicases"          </span></span>
<span><span class="co">## [10] "integral"            "mechanism"           "rad3xpd"            </span></span>
<span><span class="co">## [13] "subfamily"           "trichothiodystrophy"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[13]]</span></span>
<span><span class="co">##  [1] "alter"                     "carcinogenesis"           </span></span>
<span><span class="co">##  [3] "catalyzes"                 "cd3e"                     </span></span>
<span><span class="co">##  [5] "cerebrooculofacioskeletal" "cisplatin"                </span></span>
<span><span class="co">##  [7] "compounds"                 "crosslinks"               </span></span>
<span><span class="co">##  [9] "electrophilic"             "epsilon"                  </span></span>
<span><span class="co">## [11] "ercc4"                     "excising"                 </span></span>
<span><span class="co">## [13] "exon"                      "formed"                   </span></span>
<span><span class="co">## [15] "forms"                     "heterodimer"              </span></span>
<span><span class="co">## [17] "heterodimeric"             "induced"                  </span></span>
<span><span class="co">## [19] "interstrand"               "lesion"                   </span></span>
<span><span class="co">## [21] "lesions"                   "molecule"                 </span></span>
<span><span class="co">## [23] "overlaps"                  "pathway"                  </span></span>
<span><span class="co">## [25] "play"                      "polymorphisms"            </span></span>
<span><span class="co">## [27] "process"                   "product"                  </span></span>
<span><span class="co">## [29] "recombinational"           "required"                 </span></span>
<span><span class="co">## [31] "result"                    "strand"                   </span></span>
<span><span class="co">## [33] "xpf"                      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[14]]</span></span>
<span><span class="co">## [1] "dna"    "repair"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $edges</span></span>
<span><span class="co">##  [1]  13  31  98  99 102 105 107 108 111 118 119 121 125 131</span></span></code></pre>
<p>In this example showing ERCC genes, the term <code>DNA repair</code>
is clustered as expected.</p>
</div>
<div class="section level3">
<h3 id="producing-correlation-networks">Producing correlation networks<a class="anchor" aria-label="anchor" href="#producing-correlation-networks"></a>
</h3>
<p>The next main function is producing correlation networks between
words based on the co-occurrence of the words in the text.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic3-1.png" width="768"></p>
<p>The edge label corresponding to correlation values can be shown by
<code>edgeLabel=TRUE</code>. The number of words to be shown on plot can
be specified by <code>numWords</code>. The threshold of correlation can
be specified by <code>corThresh</code>. The text color can be changed by
<code>colorText=TRUE</code>. The type of edge can be specified by
<code>edgeLink</code>, which is by default <code>TRUE</code> (link will
be used).</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     edgeLabel<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ngramnet-1.png" width="768"></p>
<p>The clustering results based on <code>pvclust</code> can be shown
also on the correlation network, by the node color
(<code>tag=TRUE</code>).</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagnet-1.png" width="768"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">pvcl</span></span></code></pre></div>
<pre><code><span><span class="co">## $clusters</span></span>
<span><span class="co">## $clusters[[1]]</span></span>
<span><span class="co">## [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[2]]</span></span>
<span><span class="co">##  [1] "activity"             "atpdependent"         "complementation"     </span></span>
<span><span class="co">##  [4] "defects"              "factor"               "functions"           </span></span>
<span><span class="co">##  [7] "incision"             "interacts"            "nucleotide"          </span></span>
<span><span class="co">## [10] "product"              "transcriptioncoupled"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[3]]</span></span>
<span><span class="co">## [1] "dna"    "repair"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $edges</span></span>
<span><span class="co">## [1]  2 12 13</span></span></code></pre>
<p>The genes associated with the corresponding words can be shown by
specifying <code>genePlot=TRUE</code>, useful for assessing which words
is associated with interesting genes. The edges connecting words to
corresponding genes are shown. Additionally, one can specify
<code>genePlotNum</code> for limiting the genes shown, which can be
useful for identifying important genes associated with high-frequent
words.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     genePlot<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/gplot-1.png" width="768"></p>
<p>The associated enriched pathways (if present) can be shown by
specifying <code>genePathPlot</code>, using <code>ggforce</code>. In
this option, the function first performs over-representation analysis on
the whole gene set, and plot enriched terms for included genes in the
plot. Enrichment analysis here is performed by the library
<code>clusterProfiler</code> or <code>ReactomePA</code>, and one can
control which pathways to plot by <code>genePathPlotSig</code>
value.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     genePathPlot<span class="op">=</span><span class="st">"reactome"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     genePathPlotSig<span class="op">=</span><span class="fl">0.05</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">## found 47 enriched term ...</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/gpplot-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="visualization-of-pubmed-information-">Visualization of PubMed information.<a class="anchor" aria-label="anchor" href="#visualization-of-pubmed-information-"></a>
</h3>
<p>Using <code>easyPubMed</code>, one can perform the same analysis on
PubMed text like the article title and abstract. The function queries
for the input gene symbols (or microbes) and visualize. For typical use
cases, the genes identified by showing <code>genePlot</code>, or hub
genes identified in gene network analysis can be queried.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## querying pubmed for ERCC1 OR ERCC2 OR ERCC3</span></span>
<span><span class="co">## converting to a data frame ...</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc-1.png" width="768"></p>
<p>As fetching the same information is not desirable and time consuming,
the same object can be passed to <code>redo</code> option and re-perform
the analysis like tagging, or changing the visualization options.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">abtag</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                 plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, preset<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span>
<span><span class="co">## Using previous pvclust results ...</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc_2-1.png" width="768"></p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag2</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 3 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc_2-2.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="comparing-two-or-more-networks">Comparing two or more networks<a class="anchor" aria-label="anchor" href="#comparing-two-or-more-networks"></a>
</h3>
<p>One can compare two or more networks by providing list of objects
produced by wc* functions, like <code>wcGeneSummary</code> and
<code>wcAbst</code>. This can be useful for assessing the similarity and
dissimilarity of the various text sources, like PubMed, RefSeq, and
Reactome pathway names.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">8</span>,<span class="fl">9</span>,<span class="fl">10</span>,<span class="fl">11</span>,<span class="fl">12</span>,<span class="fl">13</span>,<span class="fl">14</span>,<span class="fl">16</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cxcls</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"CXCL"</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">cxcls</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 13</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 13</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>               corThresh<span class="op">=</span><span class="fl">0.2</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Not having meaningful overlaps</span></span>
<span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/compare-1.png" width="768"></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net3</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/compare-2.png" width="768"></p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span>, <span class="va">net3</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span>, <span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/compare-3.png" width="768"></p>
<p>If <code>tag</code> information is available in both gene clusters,
combined tags can be visualized by <code>tag=TRUE</code>.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keggPathways</span> <span class="op">&lt;-</span> <span class="va">org.Hs.egPATH2EG</span></span>
<span><span class="va">mappedKeys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html" class="external-link">mappedkeys</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">)</span></span>
<span><span class="va">keggList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">[</span><span class="va">mappedKeys</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`05210`</span>,</span>
<span>                      keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>                      plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>                      numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                      tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                      excludeTfIdf<span class="op">=</span><span class="fl">150</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 62</span></span>
<span><span class="co">## filtered 104 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`05211`</span>,</span>
<span>                      keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>                      plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>                      numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                      tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                      excludeTfIdf<span class="op">=</span><span class="fl">150</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 70</span></span>
<span><span class="co">## filtered 104 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagcomp-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="query-words-across-clusters">Query words across clusters<a class="anchor" aria-label="anchor" href="#query-words-across-clusters"></a>
</h3>
</div>
<div class="section level3">
<h3 id="text-over-represenatation-analysis-experimental">Text over represenatation analysis (experimental)<a class="anchor" aria-label="anchor" href="#text-over-represenatation-analysis-experimental"></a>
</h3>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`00785`</span> <span class="co"># Lipoic acid metabolism</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ora-1.png" width="700"></p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            lipoic              step              acid lipoateactivating </span></span>
<span><span class="co">##      1.615498e-14      8.089194e-08      5.359204e-07      1.488290e-04 </span></span>
<span><span class="co">##  lipoatedependent            lipoyl </span></span>
<span><span class="co">##      1.488290e-04      1.488290e-04</span></span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`05150`</span> <span class="co"># Staphylococcus aureus infection</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ora-2.png" width="700"></p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          drb        chain   complement        class         exon         beta </span></span>
<span><span class="co">## 0.000000e+00 6.765249e-82 3.613201e-74 5.708725e-46 8.532718e-39 1.663151e-34</span></span></code></pre>
<p>This thresholding can be used in RefSeq visualization. Filter words
using ORA threshold and frequency threshold by setting
<code>ora=TRUE</code>.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     ora<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 148 words (ORA) ...</span></span></code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/oranet-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="custom-usage">Custom usage<a class="anchor" aria-label="anchor" href="#custom-usage"></a>
</h2>
<div class="section level3">
<h3 id="recluster-the-cluster-using-word-information">Recluster the cluster using word information<a class="anchor" aria-label="anchor" href="#recluster-the-cluster-using-word-information"></a>
</h3>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simExample</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/returnSim.html">returnSim</a></span><span class="op">(</span><span class="fu"><a href="../reference/returnExample.html">returnExample</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">color</span>,</span>
<span>                        keyType<span class="op">=</span><span class="st">"ENSEMBL"</span>, ora<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Number of clusters: 3</span></span>
<span><span class="co">## 1</span></span>
<span><span class="co">## Input genes: 12</span></span>
<span><span class="co">## Converted input genes: 12</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 109 words (ORA) ...</span></span>
<span><span class="co">## 2</span></span>
<span><span class="co">## Input genes: 13</span></span>
<span><span class="co">## Converted input genes: 13</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 238 words (ORA) ...</span></span>
<span><span class="co">## 3</span></span>
<span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 148 words (ORA) ...</span></span></code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html" class="external-link">heatmap</a></span><span class="op">(</span><span class="va">simExample</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/textclus-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="interactive-inspection-of-bayesian-network-annotated-by-words">Interactive inspection of Bayesian network annotated by words<a class="anchor" aria-label="anchor" href="#interactive-inspection-of-bayesian-network-annotated-by-words"></a>
</h3>
<p>In this example, a Bayesian network showing the module eigengenes
relationship are inferred using <code>boot.strength</code> function in
<code>bnlearn</code> from the weighted gene correlation network analysis
(WGCNA) results. The modules are annotated by word clouds produced by
<code><a href="../reference/wcGeneSummary.html">wcGeneSummary()</a></code>, and can be exported to the format of
<code>Cytoscape.js</code> or <code>vis.js</code>. In this way, module
relationship can be interactively inspected with the functional
implications.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## In this example, we simulate WGCNA results.</span></span>
<span><span class="co">## you can just use results from WGCNA.</span></span>
<span><span class="co">## Assuming WGCNA results are stored in `mod`</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">wcGeneSummary</span><span class="fu">::</span><span class="fu"><a href="../reference/returnExample.html">returnExample</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">MEs</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">MEs</span></span>
<span><span class="va">modColors</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">colors</span></span>
<span><span class="va">ensg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">modColors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># library(bnlearn)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Replace like boot.strength(mod$MEs, R=500, algorithm = "hc")</span></span>
<span><span class="co"># dag &lt;- model2network("[ME1][ME2|ME1]") # If using bnlearn</span></span>
<span><span class="va">dag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_literal.html" class="external-link">graph_from_literal</a></span><span class="op">(</span> <span class="va">ME1</span><span class="op">-</span><span class="op">+</span><span class="va">ME2</span>, <span class="va">ME1</span><span class="op">-</span><span class="op">+</span><span class="va">ME3</span> <span class="op">)</span></span>
<span></span>
<span><span class="co">## Convert to igraph</span></span>
<span><span class="co"># g &lt;- as.igraph(dag)</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="va">dag</span></span>
<span></span>
<span><span class="co">## Assign edge attributes</span></span>
<span><span class="co">## Skip, if you perform boot.strength, the edge attributes can be added from the result</span></span>
<span><span class="co"># el &lt;- data.frame(as_edgelist(g))</span></span>
<span><span class="co"># colnames(el) &lt;- c("from","to")</span></span>
<span><span class="co"># el &lt;- left_join(el, bs)</span></span>
<span><span class="co"># E(g)$strength &lt;- el$strength</span></span>
<span><span class="co"># E(g)$direction &lt;- el$direction</span></span>
<span></span>
<span><span class="co">## Node attributes</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">stripName</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"ME"</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">sizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">modColors</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sizes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">stripName</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Directory to save images and a script</span></span>
<span><span class="va">rootDir</span> <span class="op">&lt;-</span> <span class="st">"./"</span></span>
<span><span class="va">netDir</span> <span class="op">&lt;-</span> <span class="st">"visCyjs"</span></span>
<span><span class="va">imageDir</span> <span class="op">&lt;-</span> <span class="st">"images"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span>, <span class="st">"/"</span>, <span class="va">imageDir</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">images</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plotType</span> <span class="op">&lt;-</span> <span class="st">"bar"</span></span>
<span><span class="va">numLim</span> <span class="op">&lt;-</span> <span class="fl">200</span> <span class="co"># limit for gene number</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span>    <span class="va">i</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"ME"</span>,<span class="st">""</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span> <span class="co"># strip ME</span></span>
<span></span>
<span>    <span class="va">queries</span> <span class="op">&lt;-</span> <span class="va">ensg</span><span class="op">[</span><span class="va">modColors</span><span class="op">==</span><span class="va">i</span><span class="op">]</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queries</span><span class="op">)</span><span class="op">&gt;</span><span class="va">numLim</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="st">"Sampling random genes"</span><span class="op">)</span></span>
<span>        <span class="va">queries</span> <span class="op">&lt;-</span> <span class="va">queries</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queries</span><span class="op">)</span>, <span class="va">numLim</span><span class="op">)</span><span class="op">]</span> <span class="co">## Temporary restrict to randomly chosen genes, should be replaced to like kME values</span></span>
<span>    <span class="op">}</span></span>
<span>    </span>
<span>    <span class="co">## Convert to ENTREZ</span></span>
<span>    <span class="va">entre</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"ENSEMBL"</span>,</span>
<span>        keys <span class="op">=</span> <span class="va">queries</span>, columns <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span>    </span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">plotType</span><span class="op">==</span><span class="st">"bar"</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeBar.html">makeBar</a></span><span class="op">(</span><span class="va">entre</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span> <span class="co"># get barplot</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span> <span class="co">## If wordcloud</span></span>
<span>        <span class="co"># A &lt;- wcGeneSummary(entre,</span></span>
<span>        <span class="co">#                    madeUpper=c("dna","rna",</span></span>
<span>        <span class="co">#                                tolower(keys(org.Hs.eg.db, keytype="SYMBOL"))))</span></span>
<span>        <span class="co"># palNum &lt;- sample(1:151,1) # palettetown</span></span>
<span>        <span class="co"># </span></span>
<span>        <span class="co"># ## This time use ggwordcloud()</span></span>
<span>        <span class="co"># plt &lt;- ggwordcloud::ggwordcloud(A$df$word, A$df$freq,</span></span>
<span>        <span class="co">#                      shape="circle", min.freq = 10,</span></span>
<span>        <span class="co">#                      rot.per = 0.5, random.order = FALSE,</span></span>
<span>        <span class="co">#                      colors = pokepal(palNum))+</span></span>
<span>        <span class="co">#          scale_size_area(max_size = 40)</span></span>
<span>    <span class="op">}</span></span>
<span>    <span class="co">## Save images</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span>, <span class="st">"/"</span>, <span class="va">imageDir</span>, <span class="st">"/"</span>, <span class="va">i</span> ,<span class="st">".png"</span><span class="op">)</span>,</span>
<span>           <span class="va">plt</span>, dpi<span class="op">=</span><span class="fl">300</span>, width<span class="op">=</span><span class="fl">10</span>, height<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span>    <span class="co">## Store image dir</span></span>
<span>    <span class="va">images</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">images</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">imageDir</span>, <span class="st">"/"</span>, <span class="va">i</span> ,<span class="st">".png"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ME1"</span></span>
<span><span class="co">## Input genes: 12</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<pre><code><span><span class="co">## [1] "ME2"</span></span>
<span><span class="co">## Input genes: 13</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<pre><code><span><span class="co">## [1] "ME3"</span></span>
<span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">image</span> <span class="op">&lt;-</span> <span class="va">images</span></span>
<span></span>
<span><span class="co">## Node shape</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="va">plotType</span><span class="op">==</span><span class="st">"bar"</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rectangle"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"circle"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## Scale the node size</span></span>
<span><span class="va">sizeMin</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">sizeMax</span> <span class="op">&lt;-</span> <span class="fl">200</span></span>
<span><span class="va">rawMin</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span></span>
<span><span class="va">rawMax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span></span>
<span><span class="va">scf</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">sizeMax</span><span class="op">-</span><span class="va">sizeMin</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="va">rawMax</span><span class="op">-</span><span class="va">rawMin</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">&lt;-</span> <span class="va">scf</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">+</span> <span class="va">sizeMin</span> <span class="op">-</span> <span class="va">scf</span> <span class="op">*</span> <span class="va">rawMin</span></span>
<span></span>
<span><span class="co">## Export</span></span>
<span><span class="fu"><a href="../reference/exportCyjs.html">exportCyjs</a></span><span class="op">(</span><span class="va">g</span>, <span class="va">rootDir</span>, <span class="va">netDir</span><span class="op">)</span></span>
<span><span class="co"># or, exportVisjs(g, rootDir, netDir)</span></span></code></pre></div>
<p>Use like <code>http-server</code> in the directory containing a
exported JavaScript, and interactively inspect the module relationship
with word information.</p>
<p>The example visualization is shown below (not by the code above).</p>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/wcbn.png?raw=true" alt=""><p class="caption">Example visualization of a Bayesian network</p>
</div>
</div>
<div class="section level3">
<h3 id="annotating-module-eigengenes-dendrogram">Annotating module eigengenes dendrogram<a class="anchor" aria-label="anchor" href="#annotating-module-eigengenes-dendrogram"></a>
</h3>
<p>As users of WGCNA typically plot dendrogramm and heatmap of module
eigengenes using <code>plotEigengeneNetworks</code>, it is useful to
combine with wcGeneSummary, which plot additional word information on a
dendrogram with one line.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># WGCNA::plotEigengeneNetworks(mod$MEs, mod$colors, plotHeatmaps = FALSE)</span></span>
<span><span class="fu"><a href="../reference/plotEigengeneNetworksWithWords.html">plotEigengeneNetworksWithWords</a></span><span class="op">(</span><span class="va">MEs</span>, <span class="va">modColors</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wgcna-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="assess-the-occurrence-of-the-speicific-words-across-gene-clusters">Assess the occurrence of the speicific words across gene
clusters<a class="anchor" aria-label="anchor" href="#assess-the-occurrence-of-the-speicific-words-across-gene-clusters"></a>
</h3>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'limma'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     plotMA</span></span></code></pre>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="va">query</span> <span class="op">&lt;-</span> <span class="st">"DNA repair"</span></span>
<span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/goana.html" class="external-link">getGeneKEGGLinks</a></span><span class="op">(</span>species<span class="op">=</span><span class="st">"hsa"</span><span class="op">)</span></span>
<span><span class="va">listOfGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">path</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tab</span><span class="op">$</span><span class="va">PathwayID</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">listOfGenes</span><span class="op">[[</span><span class="va">path</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">tab</span>, <span class="va">PathwayID</span><span class="op">==</span><span class="va">path</span><span class="op">)</span><span class="op">$</span><span class="va">GeneID</span></span>
<span><span class="op">}</span></span>
<span><span class="va">frq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findTerm.html">findTerm</a></span><span class="op">(</span><span class="va">query</span>, <span class="va">listOfGenes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">listOfGenes</span><span class="op">)</span>, <span class="fl">20</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                split<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                keyType<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Finding query in 20 clusters ...</span></span>
<span><span class="co">## Input genes: 44</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 87</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 197</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 47</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 131</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 99</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 4</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 112</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 85</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 75</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 26</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 27</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 295</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 36</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 148</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 27</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 32</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 364</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 531</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## Input genes: 89</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">frq</span>, check.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  check.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plt</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"path:"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">plt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">dna</span>, <span class="va">repair</span>, label <span class="op">=</span> <span class="va">plt</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span><span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>bg.color<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"dna"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"repair"</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: ggrepel: 13 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/findterm-1.png" width="480"></p>
</div>
</div>
<div class="section level2">
<h2 id="application-examples">Application examples<a class="anchor" aria-label="anchor" href="#application-examples"></a>
</h2>
<p>We demonstrate an use case, which investigates transcriptomic changes
induced by BK polyomavirus (BKPyV) infection in renal proximal tubular
epithelial cells (<a href="https://pubmed.ncbi.nlm.nih.gov/30620752/" class="external-link">An
et al.Â 2019</a>). Differentially expressed genes in 3 days
post-infection were obtained, and down-regulated genes in BKPyV infected
cells were examined.</p>
<div class="section level3">
<h3 id="load-the-necessary-packages-and-genes">Load the necessary packages and genes<a class="anchor" aria-label="anchor" href="#load-the-necessary-packages-and-genes"></a>
</h3>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">ReactomePA</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span>
<span><span class="va">degs</span> <span class="op">&lt;-</span> <span class="fu">wcGeneSummary</span><span class="fu">:::</span><span class="va">d3degDownAn2019</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">degs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 106</span></span></code></pre>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">degs</span></span></code></pre></div>
<pre><code><span><span class="co">##   [1] "PAFAH2"     "RPL7P9"     "ACP6"       "MTMR11"     "KDM5B"     </span></span>
<span><span class="co">##   [6] "RPL21P19"   "GPR137B"    "SH3YL1"     "PELI1"      "KRCC1"     </span></span>
<span><span class="co">##  [11] "RGPD5"      "GRB14"      "ITGA6"      "GPR155"     "CFLAR"     </span></span>
<span><span class="co">##  [16] "OXSM"       "FBXL2"      "RPL7AP11"   "GOLGB1"     "EEF1A1P25" </span></span>
<span><span class="co">##  [21] "DLG1"       "TRAPPC11"   "CYP4V2"     "EEF1A1P19"  "PLK2"      </span></span>
<span><span class="co">##  [26] "RAD17"      "ANKRA2"     "ARHGEF28"   "SLC22A5"    "FBXO38"    </span></span>
<span><span class="co">##  [31] "MAT2B"      "MRS2"       "HCG11"      "LMBRD1"     "SESN1"     </span></span>
<span><span class="co">##  [36] "ECHDC1"     "L3MBTL3"    "MED23"      "SGK1"       "CCZ1"      </span></span>
<span><span class="co">##  [41] "ZNF12"      "PLEKHA8"    "SEMA3C"     "RPS3AP26"   "BMT2"      </span></span>
<span><span class="co">##  [46] "OGT"        "RNF170"     "TP53INP1"   "RRM2B"      "PRNCR1"    </span></span>
<span><span class="co">##  [51] "ZNF252P"    "RPL4P5"     "KLHL9"      "SECISBP2"   "HSDL2"     </span></span>
<span><span class="co">##  [56] "CUTALP"     "TSC1"       "FBXO3"      "DDB2"       "SRSF8"     </span></span>
<span><span class="co">##  [61] "JRKL"       "TLCD5"      "ZNF248"     "ZNF33B"     "DYNC1I2P1" </span></span>
<span><span class="co">##  [66] "RPL17P50"   "FAS"        "OGA"        "WDR11"      "DHX32"     </span></span>
<span><span class="co">##  [71] "CD27-AS1"   "OVCH1-AS1"  "CAPRIN2"    "TSPAN31"    "RPSAP52"   </span></span>
<span><span class="co">##  [76] "MDM2"       "LTA4H"      "ARL1"       "APPL2"      "ZNF84"     </span></span>
<span><span class="co">##  [81] "ZNF140"     "N4BP2L2"    "TRIM13"     "MBNL2"      "RNU1-27P"  </span></span>
<span><span class="co">##  [86] "MAP4K5"     "TMEM260"    "RPL41P2"    "APH1B"      "ST20"      </span></span>
<span><span class="co">##  [91] "AKTIP"      "ZFP90"      "RPL21P120"  "TOM1L1"     "POLI"      </span></span>
<span><span class="co">##  [96] "PLCB4"      "PCMTD2"     "ZNF561"     "ZNF561-AS1" "RPL9P32"   </span></span>
<span><span class="co">## [101] "ZNF226"     "ZNF528-AS1" "ZNF419"     "ZNF154"     "BACH1"     </span></span>
<span><span class="co">## [106] "GRIK1-AS2"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="enrichment-analysis">Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h3>
<p>First, we perform enrichment analysis using ReactomePA. From the
enrichment analysis results, the cluster is related to transcriptional
regulation by TP53.</p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Convert to ENTREZID</span></span>
<span><span class="va">entre</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"SYMBOL"</span>,</span>
<span>                               keys <span class="op">=</span> <span class="va">degs</span>, columns <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span><span class="va">pway</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html" class="external-link">setReadable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html" class="external-link">enrichPathway</a></span><span class="op">(</span><span class="va">entre</span><span class="op">)</span>, <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">sigpway</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">pway</span><span class="op">@</span><span class="va">result</span>, <span class="va">p.adjust</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">sigpway</span><span class="op">$</span><span class="va">Description</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Homo sapiens\r: Transcriptional Regulation by TP53"  </span></span>
<span><span class="co">## [2] "Homo sapiens\r: RIPK1-mediated regulated necrosis"   </span></span>
<span><span class="co">## [3] "Homo sapiens\r: Regulation of necroptotic cell death"</span></span>
<span><span class="co">## [4] "Homo sapiens\r: Regulated Necrosis"                  </span></span>
<span><span class="co">## [5] "Homo sapiens\r: Regulation by c-FLIP"                </span></span>
<span><span class="co">## [6] "Homo sapiens\r: CASP8 activity is inhibited"         </span></span>
<span><span class="co">## [7] "Homo sapiens\r: Dimerization of procaspase-8"</span></span></code></pre>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html" class="external-link">cnetplot</a></span><span class="op">(</span><span class="va">pway</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ea-1.png" width="700"></p>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Genes involved in significant pathways</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">sigpway</span><span class="op">$</span><span class="va">geneID</span>, <span class="kw">function</span> <span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">x</span>,<span class="st">"/"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "PLK2"     "RAD17"    "SESN1"    "SGK1"     "TP53INP1" "RRM2B"   </span></span>
<span><span class="co">##  [7] "TSC1"     "DDB2"     "FAS"      "MDM2"     "PELI1"    "CFLAR"   </span></span>
<span><span class="co">## [13] "OGT"      "FAS"      "CFLAR"    "FAS"</span></span></code></pre>
<p>We store the name of enriched pathways in the network for the
downstream analysis.</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">netreac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">degs</span>,</span>
<span>                         enrich<span class="op">=</span><span class="st">"reactome"</span>,</span>
<span>                         plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                         numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>                         colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 106</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 106</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## performing enrichment analysis ...</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="text-mining-the-gene-summaries">Text mining the gene summaries<a class="anchor" aria-label="anchor" href="#text-mining-the-gene-summaries"></a>
</h3>
<p>Next we perform the plain function producing a correlation network,
with showing the genes related to high-frequency words in the text.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">degs</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      colorText<span class="op">=</span><span class="cn">TRUE</span>, numWords<span class="op">=</span><span class="fl">20</span>,</span>
<span>                      genePlot<span class="op">=</span><span class="cn">TRUE</span>, genePlotNum<span class="op">=</span><span class="fl">5</span>,</span>
<span>                      genePathPlot<span class="op">=</span><span class="st">"reactome"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 106</span></span>
<span><span class="co">## Converted input genes: 106</span></span>
<span><span class="co">## filtered 61 words (frequency | tfidf) ...</span></span>
<span><span class="co">## found 7 enriched term ...</span></span></code></pre>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic2app-1.png" width="700"></p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">net1</span><span class="op">$</span><span class="va">geneCount</span>, n<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">top5</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "SESN1" "GRB14" "RAD17" "DDB2"  "FBXL2"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="text-mining-the-available-literature">Text mining the available literature<a class="anchor" aria-label="anchor" href="#text-mining-the-available-literature"></a>
</h3>
<p>These genes are further queried for PubMed information.</p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span><span class="va">top5</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## querying pubmed for SESN1 OR GRB14 OR RAD17 OR DDB2 OR FBXL2</span></span>
<span><span class="co">## converting to a data frame ...</span></span></code></pre>
<p>Show the network for the article titles.</p>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">titlenet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">abst</span>, target<span class="op">=</span><span class="st">"title"</span>, tfidf<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>       plotType<span class="op">=</span><span class="st">"network"</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>       preset<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">titlenet</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/absttitleapp-1.png" width="700"></p>
<p>Show the network for the article abstract.</p>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abstnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">abst</span>, target<span class="op">=</span><span class="st">"abstract"</span>, tfidf<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>       plotType<span class="op">=</span><span class="st">"network"</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>       preset<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in weighting(x): empty document(s): character(0) character(0)</span></span>
<span><span class="co">## character(0) character(0) character(0) character(0) character(0) character(0)</span></span></code></pre>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abstnet</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/abstmainapp-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="combine-and-compare-networks">Combine and compare networks<a class="anchor" aria-label="anchor" href="#combine-and-compare-networks"></a>
</h3>
<p>From the RefSeq summary and articles related to important genes, the
cluster could have functionality of DNA damage response, which is also
upregulated by BKPyV infection. These networks can be combined to find
intersections and differences. We can see that in addition to Reactome
pathway names, plenty of information could be obtained and summarized by
querying other databases, which could aid in interpreting clusters of
genes and hypothesis generation.</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">abstnet</span>, <span class="va">titlenet</span>, <span class="va">netreac</span>, <span class="va">net1</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Abstract"</span>,<span class="st">"Title"</span>,<span class="st">"Reactome"</span>,<span class="st">"RefSeq"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/combineapp-1.png" width="960"></p>
<p>The network can be obtained by <code>returnNet=TRUE</code>, which can
be used for downstream analysis like assessment of degrees and community
detection.</p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">abstnet</span>, <span class="va">titlenet</span>, <span class="va">netreac</span>, <span class="va">net1</span><span class="op">)</span>, returnNet<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">conet</span></span></code></pre></div>
<pre><code><span><span class="co">## IGRAPH 1c7d8e6 UN-- 73 181 -- </span></span>
<span><span class="co">## + attr: Freq_1 (v/n), Freq_2 (v/n), grp (v/c), name (v/c), Freqs (v/n),</span></span>
<span><span class="co">## | col (v/c)</span></span>
<span><span class="co">## + edges from 1c7d8e6 (vertex names):</span></span>
<span><span class="co">##  [1] activation--checkpoint  activation--complex     activation--interaction</span></span>
<span><span class="co">##  [4] activation--mtorc1      activation--p53         activation--RAD17      </span></span>
<span><span class="co">##  [7] activation--response    activation--SESN1       activation--signaling  </span></span>
<span><span class="co">## [10] activation--stress      activation--pathway     activation--apoptotic  </span></span>
<span><span class="co">## [13] activation--caspase     activation--death       activation--extrinsic  </span></span>
<span><span class="co">## [16] activation--ligand      activation--presence    activation--signalling </span></span>
<span><span class="co">## [19] cancer    --DDB2        cancer    --risk        checkpoint--complex    </span></span>
<span><span class="co">## + ... omitted several edges</span></span></code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conetDeg</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/degree.html" class="external-link">degree</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span></span>
<span><span class="va">conetDeg</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">conetDeg</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  activation      damage     complex    response   signaling       SESN1 </span></span>
<span><span class="co">##          18          15          13          13          13          12 </span></span>
<span><span class="co">##        DDB2         DNA       RAD17  checkpoint         XPC     caspase </span></span>
<span><span class="co">##          11          11          11           9           8           8 </span></span>
<span><span class="co">##       GRB14 interaction    proteins </span></span>
<span><span class="co">##           7           7           7</span></span></code></pre>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wt</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html" class="external-link">walktrap.community</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span></span>
<span><span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span><span class="op">$</span><span class="va">walktrap</span> <span class="op">&lt;-</span> <span class="va">wt</span><span class="op">$</span><span class="va">membership</span></span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html" class="external-link">ggraph</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html" class="external-link">geom_node_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">walktrap</span><span class="op">)</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_link.html" class="external-link">geom_edge_link</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html" class="external-link">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span>, color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">walktrap</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                   check_overlap<span class="op">=</span><span class="cn">TRUE</span>, repel<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                   bg.color <span class="op">=</span> <span class="st">"white"</span>, segment.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>                   bg.r <span class="op">=</span> <span class="fl">.15</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_color_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Walktrap"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html" class="external-link">theme_graph</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using <span style="color: #0000BB;">"stress"</span> as default layout</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Using the `size` aesthetic in this geom was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">â„¹</span> Please use `linewidth` in the `default_aes` field and elsewhere instead.</span></span></code></pre>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/combineNet-1.png" width="960"></p>
<p>Dynamic layout can be also used to compare the networks, by
<code>graphlayouts</code>, for comparing the multiple graphs, especially
useful for time-series analysis. See the documentation of <a href="http://graphlayouts.schochastics.net/reference/layout_dynamic.html" class="external-link"><code>layout_as_dynamic</code></a>
for specifying the alpha, which is default to 0.5.</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="va">dyn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotDynamic.html">plotDynamic</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">abstnet</span>, <span class="va">titlenet</span>, <span class="va">net1</span><span class="op">)</span>,</span>
<span>                   titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Abstract"</span>,<span class="st">"Title"</span>,<span class="st">"RefSeq"</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">dyn</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/dyn-1.png" width="1920"></p>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span><span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">## Running under: Windows 10 x64 (build 19044)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] LC_COLLATE=Japanese_Japan.utf8  LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">## [3] LC_MONETARY=Japanese_Japan.utf8 LC_NUMERIC=C                   </span></span>
<span><span class="co">## [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] clusterProfiler_4.6.0 ReactomePA_1.42.0     ggrepel_0.9.2        </span></span>
<span><span class="co">##  [4] limma_3.54.0          igraph_1.3.5          RColorBrewer_1.1-3   </span></span>
<span><span class="co">##  [7] ggraph_2.1.0          ggplot2_3.4.0         org.Hs.eg.db_3.16.0  </span></span>
<span><span class="co">## [10] AnnotationDbi_1.60.0  IRanges_2.32.0        S4Vectors_0.36.0     </span></span>
<span><span class="co">## [13] Biobase_2.58.0        BiocGenerics_0.44.0   wcGeneSummary_0.99.0 </span></span>
<span><span class="co">## [16] BiocStyle_2.26.0     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] utf8_1.2.2             tidyselect_1.2.0       RSQLite_2.2.18        </span></span>
<span><span class="co">##   [4] htmlwidgets_1.5.4      grid_4.2.2             BiocParallel_1.32.1   </span></span>
<span><span class="co">##   [7] scatterpie_0.1.8       munsell_0.5.0          codetools_0.2-18      </span></span>
<span><span class="co">##  [10] ragg_1.2.4             withr_2.5.0            colorspace_2.0-3      </span></span>
<span><span class="co">##  [13] GOSemSim_2.24.0        NLP_0.2-1              highr_0.9             </span></span>
<span><span class="co">##  [16] knitr_1.40             rstudioapi_0.14        DOSE_3.24.1           </span></span>
<span><span class="co">##  [19] GeneSummary_0.99.4     labeling_0.4.2         slam_0.1-50           </span></span>
<span><span class="co">##  [22] GenomeInfoDbData_1.2.9 polyclip_1.10-4        bit64_4.0.5           </span></span>
<span><span class="co">##  [25] farver_2.1.1           rprojroot_2.0.3        downloader_0.4        </span></span>
<span><span class="co">##  [28] vctrs_0.5.0            treeio_1.22.0          generics_0.1.3        </span></span>
<span><span class="co">##  [31] gson_0.0.9             xfun_0.30              R6_2.5.1              </span></span>
<span><span class="co">##  [34] GenomeInfoDb_1.34.3    graphlayouts_0.8.3     concaveman_1.1.0      </span></span>
<span><span class="co">##  [37] ISOcodes_2022.09.29    bitops_1.0-7           cachem_1.0.6          </span></span>
<span><span class="co">##  [40] fgsea_1.24.0           gridGraphics_0.5-1     promises_1.2.0.1      </span></span>
<span><span class="co">##  [43] scales_1.2.1           enrichplot_1.18.0      gtable_0.3.1          </span></span>
<span><span class="co">##  [46] tidygraph_1.2.2        rlang_1.0.6            systemfonts_1.0.4     </span></span>
<span><span class="co">##  [49] GlobalOptions_0.1.2    splines_4.2.2          lazyeval_0.2.2        </span></span>
<span><span class="co">##  [52] stopwords_2.3          wordcloud_2.6          BiocManager_1.30.19   </span></span>
<span><span class="co">##  [55] yaml_2.3.6             reshape2_1.4.4         httpuv_1.6.6          </span></span>
<span><span class="co">##  [58] pvclust_2.2-0          qvalue_2.30.0          tools_4.2.2           </span></span>
<span><span class="co">##  [61] bookdown_0.30          ggplotify_0.1.0        ellipsis_0.3.2        </span></span>
<span><span class="co">##  [64] jquerylib_0.1.4        ggdendro_0.1.23        Rcpp_1.0.9            </span></span>
<span><span class="co">##  [67] plyr_1.8.7             base64enc_0.1-3        zlibbioc_1.44.0       </span></span>
<span><span class="co">##  [70] purrr_0.3.5            RCurl_1.98-1.9         bnlearn_4.8.1         </span></span>
<span><span class="co">##  [73] GetoptLong_1.0.5       viridis_0.6.2          cowplot_1.1.1         </span></span>
<span><span class="co">##  [76] fs_1.5.2               magrittr_2.0.3         data.table_1.14.4     </span></span>
<span><span class="co">##  [79] reactome.db_1.79.0     patchwork_1.1.2        mime_0.12             </span></span>
<span><span class="co">##  [82] evaluate_0.18          xtable_1.8-4           HDO.db_0.99.1         </span></span>
<span><span class="co">##  [85] XML_3.99-0.9           gridExtra_2.3          compiler_4.2.2        </span></span>
<span><span class="co">##  [88] bugsigdbr_1.4.0        tibble_3.1.8           V8_4.2.2              </span></span>
<span><span class="co">##  [91] crayon_1.5.2           shadowtext_0.1.2       htmltools_0.5.3       </span></span>
<span><span class="co">##  [94] ggfun_0.0.8            later_1.3.0            tidyr_1.2.1           </span></span>
<span><span class="co">##  [97] aplot_0.1.8            DBI_1.1.3              tweenr_2.0.2          </span></span>
<span><span class="co">## [100] MASS_7.3-58.1          rappdirs_0.3.3         Matrix_1.5-3          </span></span>
<span><span class="co">## [103] cli_3.4.1              parallel_4.2.2         pkgconfig_2.0.3       </span></span>
<span><span class="co">## [106] pkgdown_2.0.6          xml2_1.3.3             ggtree_3.6.2          </span></span>
<span><span class="co">## [109] bslib_0.4.1            XVector_0.38.0         yulab.utils_0.0.5     </span></span>
<span><span class="co">## [112] stringr_1.4.1          digest_0.6.30          graph_1.76.0          </span></span>
<span><span class="co">## [115] Biostrings_2.66.0      tm_0.7-8               rmarkdown_2.18        </span></span>
<span><span class="co">## [118] fastmatch_1.1-3        tidytree_0.4.1         easyPubMed_2.13       </span></span>
<span><span class="co">## [121] dendextend_1.16.0      cyjShiny_1.0.34        curl_4.3.3            </span></span>
<span><span class="co">## [124] shiny_1.7.3            graphite_1.44.0        rjson_0.2.21          </span></span>
<span><span class="co">## [127] lifecycle_1.0.3        nlme_3.1-160           jsonlite_1.8.3        </span></span>
<span><span class="co">## [130] desc_1.4.2             viridisLite_0.4.1      fansi_1.0.3           </span></span>
<span><span class="co">## [133] pillar_1.8.1           lattice_0.20-45        KEGGREST_1.38.0       </span></span>
<span><span class="co">## [136] fastmap_1.1.0          httr_1.4.4             GO.db_3.16.0          </span></span>
<span><span class="co">## [139] glue_1.6.2             png_0.1-7              bit_4.0.4             </span></span>
<span><span class="co">## [142] ggforce_0.4.1          stringi_1.7.8          sass_0.4.2            </span></span>
<span><span class="co">## [145] blob_1.2.3             textshaping_0.3.6      memoise_2.0.1         </span></span>
<span><span class="co">## [148] dplyr_1.0.99.9000      ape_5.6-2</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
