<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="wcGeneSummary">
<title>wcGeneSummary â€¢ wcGeneSummary</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="wcGeneSummary">
<meta property="og:description" content="wcGeneSummary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">wcGeneSummary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/basic_usage_of_wcGeneSummary.html">wcGeneSummary</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>wcGeneSummary</h1>
                        <h4 data-toc-skip class="author">Noriaki
Sato</h4>
            
      
      
      <div class="d-none name"><code>basic_usage_of_wcGeneSummary.Rmd</code></div>
    </div>

    
    
<p>This package aims to visualize the word and text information
contained in the gene or the other omics identifiers such as microbiome,
and compare the clusters based on those information.</p>
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<p>Load the package and the database for converting identifiers.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="producing-word-clouds">Producing word clouds<a class="anchor" aria-label="anchor" href="#producing-word-clouds"></a>
</h3>
<p>The basic usage of the package is producing a word cloud of gene
summaries by querying gene IDs, default to symbol. As many of the words
are commonly observed, you should limit word frequency by
<code>excludeFreq</code>, which is default to 2000. TF-IDF on the all
the summary is precomputed, and <code>excludeTfIdf</code> can be
specified too.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Configure input genes</span></span>
<span><span class="va">inpSymbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ERCC1"</span>,<span class="st">"ERCC2"</span>,<span class="st">"ERCC3"</span>,<span class="st">"ERCC4"</span>,<span class="st">"ERCC5"</span>,<span class="st">"ERCC6"</span>,<span class="st">"ERCC8"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic-1.png" width="700"></p>
<p>It accepts values of the <code>wordcloud()</code> function.
<code>numWords</code> specifies how many words are to be shown on word
cloud.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use palette from palettetown, palettetown::pokepal(150) (Mewtwo!)</span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, random.order<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>                     excludeTfIdf <span class="op">=</span> <span class="fl">150</span>,</span>
<span>                     colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 107 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic2-1.png" width="700"></p>
<p>It also returns a data frame consisting of frequency of each
term.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwc</span><span class="op">$</span><span class="va">df</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'Term frequencies.'</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Term frequencies.</caption>
<thead><tr class="header">
<th align="left">word</th>
<th align="right">freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">repair</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">DNA</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">syndrome</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">excision</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">cockayne</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">pigmentosum</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
<p>N-gram is supported by library <code>tm</code>, specified by
<code>ngram</code>. Default is <code>1</code>, and the example
specifying <code>2</code> is shown below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, ngram<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc2</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ngramwc-1.png" width="576"></p>
<p>One of the main questions is which words can be clustered together
among the words contained in the cluster. Word clustering
(<code>pvclust</code>) based on the occurrence in the text can be
visualized by specifying <code>tag=TRUE</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwcl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">30</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, pal <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwcl</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagging-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwcl</span><span class="op">$</span><span class="va">pvcl</span></span></code></pre></div>
<pre><code><span><span class="co">## $clusters</span></span>
<span><span class="co">## $clusters[[1]]</span></span>
<span><span class="co">## [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[2]]</span></span>
<span><span class="co">##  [1] "activity"                  "atpdependent"             </span></span>
<span><span class="co">##  [3] "basal"                     "cerebrooculofacioskeletal"</span></span>
<span><span class="co">##  [5] "characterized"             "complementation"          </span></span>
<span><span class="co">##  [7] "damage"                    "defects"                  </span></span>
<span><span class="co">##  [9] "ercc"                      "exon"                     </span></span>
<span><span class="co">## [11] "factor"                    "forms"                    </span></span>
<span><span class="co">## [13] "functions"                 "helicase"                 </span></span>
<span><span class="co">## [15] "heterodimeric"             "incision"                 </span></span>
<span><span class="co">## [17] "including"                 "interacts"                </span></span>
<span><span class="co">## [19] "nucleotide"                "product"                  </span></span>
<span><span class="co">## [21] "transcriptioncoupled"     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[3]]</span></span>
<span><span class="co">## [1] "dna"    "repair"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $edges</span></span>
<span><span class="co">## [1]  6 22 23</span></span></code></pre>
<p>In this example showing ERCC genes, the term <code>DNA repair</code>
is clustered as expected.</p>
</div>
<div class="section level3">
<h3 id="producing-correlation-networks">Producing correlation networks<a class="anchor" aria-label="anchor" href="#producing-correlation-networks"></a>
</h3>
<p>The next main function is producing correlation networks between
words based on the co-occurrence of the words in the text.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/basic3-1.png" width="700"></p>
<p>The edge label can be shown by <code>edgeLabel=TRUE</code>.<br>
The threhsold of correlation can be specified by <code>corThresh</code>.
The text color can be changed by <code>colorText=TRUE</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, edgeLabel<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ngramnet-1.png" width="700"></p>
<p>The clustering results based on <code>pvclust</code> can be shown
also on the correlation network, by the node color.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagnet-1.png" width="700"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">pvcl</span></span></code></pre></div>
<pre><code><span><span class="co">## $clusters</span></span>
<span><span class="co">## $clusters[[1]]</span></span>
<span><span class="co">## [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[2]]</span></span>
<span><span class="co">##  [1] "activity"             "atpdependent"         "complementation"     </span></span>
<span><span class="co">##  [4] "defects"              "factor"               "functions"           </span></span>
<span><span class="co">##  [7] "incision"             "interacts"            "nucleotide"          </span></span>
<span><span class="co">## [10] "product"              "transcriptioncoupled"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $clusters[[3]]</span></span>
<span><span class="co">## [1] "dna"    "repair"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $edges</span></span>
<span><span class="co">## [1]  2 12 13</span></span></code></pre>
<p>The genes associated with the corresponding words can be shown by
specifying <code>genePlot=TRUE</code>, useful for assessing which words
is associated with interesting genes. The edges connecting words to
corresponding genes are shown.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/gplot-1.png" width="700"></p>
<p>The associated enriched pathways (if present) can be shown by
specifying <code>genePathPlot</code>, using <code>ggforce</code>. In
this option, the function first performs over-representation analysis on
the whole gene set, and plot enriched terms for included genes in the
plot. Enrichment analysis here is performed by the library
<code>clusterProfiler</code> or <code>ReactomePA</code>, and one can
control which pathways to plot by <code>geePathPlotSig</code> value.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, genePathPlot<span class="op">=</span><span class="st">"kegg"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>, genePathPlotSig<span class="op">=</span><span class="fl">0.05</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">## Found 3 enriched term ...</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/gpplot-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualization-of-pubmed-information-">Visualization of PubMed information.<a class="anchor" aria-label="anchor" href="#visualization-of-pubmed-information-"></a>
</h3>
<p>Using <code>easyPubMed</code>, one can perform the same analysis on
PubMed text like the article title and abstract. The function queries
for the input gene symbols (or microbes) and visualize.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## querying pubmed for ERCC1 OR ERCC2 OR ERCC3</span></span>
<span><span class="co">## converting to a data frame ...</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc-1.png" width="700"></p>
<p>As fetching the same information is not desiable and time consuming,
the same object can be passed to <code>redo</code> option and reperform
the analysis like tagging, or changing the visualization options.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">abtag</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span>
<span><span class="co">## Using previous pvclust results ...</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag</span><span class="op">$</span><span class="va">wc</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc_2-1.png" width="700"></p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag2</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 3 rows containing missing values (geom_point).</span></span></code></pre>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc_2-2.png" width="700"></p>
<div class="section level4">
<h4 id="comparing-two-networks">Comparing two networks<a class="anchor" aria-label="anchor" href="#comparing-two-networks"></a>
</h4>
<p>One can compare two networks by providing two objects produced by wc*
functions, like <code>wcGeneSummary</code> and <code>wcAbst</code>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">8</span>,<span class="fl">9</span>,<span class="fl">10</span>,<span class="fl">11</span>,<span class="fl">12</span>,<span class="fl">13</span>,<span class="fl">14</span>,<span class="fl">16</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cxcls</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"CXCL"</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">cxcls</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 13</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Converted input genes: 13</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Resuming from the previous results ...</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Not having meaningful overlaps</span></span>
<span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span>, titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/compare-1.png" width="700"></p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net3</span>, titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/compare-2.png" width="700"></p>
<p>If <code>tag</code> information is available in both gene clusters,
combined tags can be visualized by <code>tag=TRUE</code>.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keggPathways</span> <span class="op">&lt;-</span> <span class="va">org.Hs.egPATH2EG</span></span>
<span><span class="va">mappedKeys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html" class="external-link">mappedkeys</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">)</span></span>
<span><span class="va">keggList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">[</span><span class="va">mappedKeys</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`05210`</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.3</span>, numWords<span class="op">=</span><span class="fl">30</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, excludeTfIdf<span class="op">=</span><span class="fl">150</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 62</span></span>
<span><span class="co">## filtered 107 words (frequency) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`05211`</span>,</span>
<span>                      keyType<span class="op">=</span><span class="st">"ENTREZID"</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.3</span>, numWords<span class="op">=</span><span class="fl">30</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, excludeTfIdf<span class="op">=</span><span class="fl">150</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 70</span></span>
<span><span class="co">## filtered 107 words (frequency) ...</span></span>
<span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/tagcomp-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="text-over-represenatation-analysis-experimental">Text over represenatation analysis (experimental)<a class="anchor" aria-label="anchor" href="#text-over-represenatation-analysis-experimental"></a>
</h3>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`00785`</span> <span class="co"># Lipoic acid metabolism</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ora-1.png" width="700"></p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            lipoic              step              acid lipoateactivating </span></span>
<span><span class="co">##      1.701356e-14      8.782627e-08      5.708708e-07      1.507684e-04 </span></span>
<span><span class="co">##  lipoatedependent            lipoyl </span></span>
<span><span class="co">##      1.507684e-04      1.507684e-04</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`05150`</span> <span class="co"># Staphylococcus aureus infection</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/ora-2.png" width="700"></p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        chain   complement        class         exon      peptide         beta </span></span>
<span><span class="co">## 1.982266e-81 6.443621e-74 1.233313e-45 1.278631e-38 8.506250e-34 9.352422e-33</span></span></code></pre>
<p>Filter words using ORA threshold and frequency threshold by setting
<code>ora=TRUE</code>.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, ora<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 147 words (ORA) ...</span></span></code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">$</span><span class="va">net</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/oranet-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="custom-usage">Custom usage<a class="anchor" aria-label="anchor" href="#custom-usage"></a>
</h2>
<div class="section level3">
<h3 id="recluster-the-cluster-using-word-information">Recluster the cluster using word information<a class="anchor" aria-label="anchor" href="#recluster-the-cluster-using-word-information"></a>
</h3>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simExample</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/returnSim.html">returnSim</a></span><span class="op">(</span><span class="fu"><a href="../reference/returnExample.html">returnExample</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">color</span>, keyType<span class="op">=</span><span class="st">"ENSEMBL"</span>, ora<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Number of clusters: 3</span></span>
<span><span class="co">## 1</span></span>
<span><span class="co">## Input genes: 12</span></span>
<span><span class="co">## Converted input genes: 12</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 109 words (ORA) ...</span></span>
<span><span class="co">## 2</span></span>
<span><span class="co">## Input genes: 13</span></span>
<span><span class="co">## Converted input genes: 13</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 237 words (ORA) ...</span></span>
<span><span class="co">## 3</span></span>
<span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## Converted input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span>
<span><span class="co">## performing ORA</span></span>
<span><span class="co">## filtered 147 words (ORA) ...</span></span></code></pre>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html" class="external-link">heatmap</a></span><span class="op">(</span><span class="va">simExample</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/textclus-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="interactive-inspection-of-bayesian-network-annotated-by-words">Interactive inspection of Bayesian network annotated by words<a class="anchor" aria-label="anchor" href="#interactive-inspection-of-bayesian-network-annotated-by-words"></a>
</h3>
<p>In this example, a Bayesian network showing the module eigengenes
relationship are inferred using <code>boot.strength</code> function in
<code>bnlearn</code> from the weighted gene correlation network analysis
(WGCNA) results. The modules are annotated by word clouds produced by
<code><a href="../reference/wcGeneSummary.html">wcGeneSummary()</a></code>, and can be exported to the format of
<code>Cytoscape.js</code> or <code>vis.js</code>. In this way, module
relationship can be interactively inspected with the functional
implications.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## In this example, we simulate WGCNA results.</span></span>
<span><span class="co">## you can just use results from WGCNA.</span></span>
<span><span class="co">## Assuming WGCNA results are stored in `mod`</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">wcGeneSummary</span><span class="fu">::</span><span class="fu"><a href="../reference/returnExample.html">returnExample</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">MEs</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">MEs</span></span>
<span><span class="va">modColors</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">colors</span></span>
<span><span class="va">ensg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">modColors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># library(bnlearn)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Replace like boot.strength(mod$MEs, R=500, algorithm = "hc")</span></span>
<span><span class="co"># dag &lt;- model2network("[ME1][ME2|ME1]") # If using bnlearn</span></span>
<span><span class="va">dag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_literal.html" class="external-link">graph_from_literal</a></span><span class="op">(</span> <span class="va">ME1</span><span class="op">-</span><span class="op">+</span><span class="va">ME2</span>, <span class="va">ME1</span><span class="op">-</span><span class="op">+</span><span class="va">ME3</span> <span class="op">)</span></span>
<span></span>
<span><span class="co">## Convert to igraph</span></span>
<span><span class="co"># g &lt;- as.igraph(dag)</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="va">dag</span></span>
<span></span>
<span><span class="co">## Assign edge attributes</span></span>
<span><span class="co">## Skip, if you perform boot.strength, the edge attributes can be added from the result</span></span>
<span><span class="co"># el &lt;- data.frame(as_edgelist(g))</span></span>
<span><span class="co"># colnames(el) &lt;- c("from","to")</span></span>
<span><span class="co"># el &lt;- left_join(el, bs)</span></span>
<span><span class="co"># E(g)$strength &lt;- el$strength</span></span>
<span><span class="co"># E(g)$direction &lt;- el$direction</span></span>
<span></span>
<span><span class="co">## Node attributes</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">stripName</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"ME"</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">sizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">modColors</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sizes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">stripName</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Directory to save images and a script</span></span>
<span><span class="va">rootDir</span> <span class="op">&lt;-</span> <span class="st">"./"</span></span>
<span><span class="va">netDir</span> <span class="op">&lt;-</span> <span class="st">"visCyjs"</span></span>
<span><span class="va">imageDir</span> <span class="op">&lt;-</span> <span class="st">"images"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span>, <span class="st">"/"</span>, <span class="va">imageDir</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">images</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plotType</span> <span class="op">&lt;-</span> <span class="st">"bar"</span></span>
<span><span class="va">numLim</span> <span class="op">&lt;-</span> <span class="fl">200</span> <span class="co"># limit for gene number</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span>    <span class="va">i</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"ME"</span>,<span class="st">""</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span> <span class="co"># strip ME</span></span>
<span></span>
<span>    <span class="va">queries</span> <span class="op">&lt;-</span> <span class="va">ensg</span><span class="op">[</span><span class="va">modColors</span><span class="op">==</span><span class="va">i</span><span class="op">]</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queries</span><span class="op">)</span><span class="op">&gt;</span><span class="va">numLim</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="st">"Sampling random genes"</span><span class="op">)</span></span>
<span>        <span class="va">queries</span> <span class="op">&lt;-</span> <span class="va">queries</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queries</span><span class="op">)</span>, <span class="va">numLim</span><span class="op">)</span><span class="op">]</span> <span class="co">## Temporary restrict to randomly chosen genes, should be replaced to like kME values</span></span>
<span>    <span class="op">}</span></span>
<span>    </span>
<span>    <span class="co">## Convert to ENTREZ</span></span>
<span>    <span class="va">entre</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"ENSEMBL"</span>,</span>
<span>        keys <span class="op">=</span> <span class="va">queries</span>, columns <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span>    </span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">plotType</span><span class="op">==</span><span class="st">"bar"</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeBar.html">makeBar</a></span><span class="op">(</span><span class="va">entre</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span> <span class="co"># get barplot</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span> <span class="co">## If wordcloud</span></span>
<span>        <span class="co"># A &lt;- wcGeneSummary(entre,</span></span>
<span>        <span class="co">#                    madeUpper=c("dna","rna",</span></span>
<span>        <span class="co">#                                tolower(keys(org.Hs.eg.db, keytype="SYMBOL"))))</span></span>
<span>        <span class="co"># palNum &lt;- sample(1:151,1) # palettetown</span></span>
<span>        <span class="co"># </span></span>
<span>        <span class="co"># ## This time use ggwordcloud()</span></span>
<span>        <span class="co"># plt &lt;- ggwordcloud::ggwordcloud(A$df$word, A$df$freq,</span></span>
<span>        <span class="co">#                      shape="circle", min.freq = 10,</span></span>
<span>        <span class="co">#                      rot.per = 0.5, random.order = FALSE,</span></span>
<span>        <span class="co">#                      colors = pokepal(palNum))+</span></span>
<span>        <span class="co">#          scale_size_area(max_size = 40)</span></span>
<span>    <span class="op">}</span></span>
<span>    <span class="co">## Save images</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">rootDir</span>, <span class="va">netDir</span>, <span class="st">"/"</span>, <span class="va">imageDir</span>, <span class="st">"/"</span>, <span class="va">i</span> ,<span class="st">".png"</span><span class="op">)</span>,</span>
<span>           <span class="va">plt</span>, dpi<span class="op">=</span><span class="fl">300</span>, width<span class="op">=</span><span class="fl">10</span>, height<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span>    <span class="co">## Store image dir</span></span>
<span>    <span class="va">images</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">images</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">imageDir</span>, <span class="st">"/"</span>, <span class="va">i</span> ,<span class="st">".png"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ME1"</span></span>
<span><span class="co">## Input genes: 12</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<pre><code><span><span class="co">## [1] "ME2"</span></span>
<span><span class="co">## Input genes: 13</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<pre><code><span><span class="co">## [1] "ME3"</span></span>
<span><span class="co">## Input genes: 7</span></span>
<span><span class="co">## filtered 62 words (frequency) ...</span></span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">image</span> <span class="op">&lt;-</span> <span class="va">images</span></span>
<span></span>
<span><span class="co">## Node shape</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="va">plotType</span><span class="op">==</span><span class="st">"bar"</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rectangle"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"circle"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## Scale the node size</span></span>
<span><span class="va">sizeMin</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">sizeMax</span> <span class="op">&lt;-</span> <span class="fl">200</span></span>
<span><span class="va">rawMin</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span></span>
<span><span class="va">rawMax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span></span>
<span><span class="va">scf</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">sizeMax</span><span class="op">-</span><span class="va">sizeMin</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="va">rawMax</span><span class="op">-</span><span class="va">rawMin</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">&lt;-</span> <span class="va">scf</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">size</span> <span class="op">+</span> <span class="va">sizeMin</span> <span class="op">-</span> <span class="va">scf</span> <span class="op">*</span> <span class="va">rawMin</span></span>
<span></span>
<span><span class="co">## Export</span></span>
<span><span class="fu"><a href="../reference/exportCyjs.html">exportCyjs</a></span><span class="op">(</span><span class="va">g</span>, <span class="va">rootDir</span>, <span class="va">netDir</span><span class="op">)</span></span>
<span><span class="co"># or, exportVisjs(g, rootDir, netDir)</span></span></code></pre></div>
<p>Use like <code>http-server</code> in the directory containing a
exported JavaScript, and interactively inspect the module relationship
with word information.</p>
<p>The example visualization is shown below (not by the code above).</p>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/wcbn.png?raw=true" alt=""><p class="caption">Example visualization of a Bayesian network</p>
</div>
</div>
<div class="section level3">
<h3 id="annotating-module-eigengenes-dendrogram">Annotating module eigengenes dendrogram<a class="anchor" aria-label="anchor" href="#annotating-module-eigengenes-dendrogram"></a>
</h3>
<p>As users of WGCNA typically plot dendrogramm and heatmap of module
eigengenes using <code>plotEigengeneNetworks</code>, it is useful to
combine with wcGeneSummary, which plot additional word information on a
dendrogram with one line.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># WGCNA::plotEigengeneNetworks(mod$MEs, mod$colors, plotHeatmaps = FALSE)</span></span>
<span><span class="fu"><a href="../reference/plotEigengeneNetworksWithWords.html">plotEigengeneNetworksWithWords</a></span><span class="op">(</span><span class="va">MEs</span>, <span class="va">modColors</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">## Bootstrap (r = 1.4)... Done.</span></span></code></pre>
<p><img src="basic_usage_of_wcGeneSummary_files/figure-html/wgcna-1.png" width="700"></p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23 ucrt)</span></span>
<span><span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">## Running under: Windows 10 x64 (build 22000)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] LC_COLLATE=Japanese_Japan.utf8  LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">## [3] LC_MONETARY=Japanese_Japan.utf8 LC_NUMERIC=C                   </span></span>
<span><span class="co">## [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] igraph_1.3.4         ggraph_2.0.6         ggplot2_3.3.6       </span></span>
<span><span class="co">##  [4] org.Hs.eg.db_3.15.0  AnnotationDbi_1.58.0 IRanges_2.30.0      </span></span>
<span><span class="co">##  [7] S4Vectors_0.34.0     Biobase_2.56.0       BiocGenerics_0.42.0 </span></span>
<span><span class="co">## [10] wcGeneSummary_0.99.0 BiocStyle_2.24.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] shadowtext_0.1.2       fastmatch_1.1-3        systemfonts_1.0.4     </span></span>
<span><span class="co">##   [4] plyr_1.8.7             lazyeval_0.2.2         splines_4.2.1         </span></span>
<span><span class="co">##   [7] BiocParallel_1.30.3    GenomeInfoDb_1.32.2    digest_0.6.29         </span></span>
<span><span class="co">##  [10] yulab.utils_0.0.5      htmltools_0.5.3        bugsigdbr_1.2.2       </span></span>
<span><span class="co">##  [13] GOSemSim_2.22.0        viridis_0.6.2          GO.db_3.15.0          </span></span>
<span><span class="co">##  [16] fansi_1.0.3            GeneSummary_0.99.3     magrittr_2.0.3        </span></span>
<span><span class="co">##  [19] memoise_2.0.1          tm_0.7-8               Biostrings_2.64.0     </span></span>
<span><span class="co">##  [22] graphlayouts_0.8.0     pvclust_2.2-0          wordcloud_2.6         </span></span>
<span><span class="co">##  [25] pkgdown_2.0.6          enrichplot_1.16.1      colorspace_2.0-3      </span></span>
<span><span class="co">##  [28] rappdirs_0.3.3         blob_1.2.3             ggrepel_0.9.1         </span></span>
<span><span class="co">##  [31] textshaping_0.3.6      xfun_0.31              dplyr_1.0.9           </span></span>
<span><span class="co">##  [34] crayon_1.5.1           RCurl_1.98-1.7         jsonlite_1.8.0        </span></span>
<span><span class="co">##  [37] graph_1.74.0           scatterpie_0.1.7       ape_5.6-2             </span></span>
<span><span class="co">##  [40] glue_1.6.2             polyclip_1.10-0        stopwords_2.3         </span></span>
<span><span class="co">##  [43] gtable_0.3.0           zlibbioc_1.42.0        XVector_0.36.0        </span></span>
<span><span class="co">##  [46] GetoptLong_1.0.5       graphite_1.42.0        scales_1.2.0          </span></span>
<span><span class="co">##  [49] DOSE_3.22.0            DBI_1.1.3              Rcpp_1.0.9            </span></span>
<span><span class="co">##  [52] xtable_1.8-4           viridisLite_0.4.0      tidytree_0.3.9        </span></span>
<span><span class="co">##  [55] gridGraphics_0.5-1     reactome.db_1.81.0     bit_4.0.4             </span></span>
<span><span class="co">##  [58] htmlwidgets_1.5.4      httr_1.4.3             fgsea_1.22.0          </span></span>
<span><span class="co">##  [61] RColorBrewer_1.1-3     ellipsis_0.3.2         pkgconfig_2.0.3       </span></span>
<span><span class="co">##  [64] XML_3.99-0.10          farver_2.1.1           sass_0.4.2            </span></span>
<span><span class="co">##  [67] utf8_1.2.2             labeling_0.4.2         later_1.3.0           </span></span>
<span><span class="co">##  [70] ggplotify_0.1.0        tidyselect_1.1.2       rlang_1.0.4           </span></span>
<span><span class="co">##  [73] reshape2_1.4.4         munsell_0.5.0          tools_4.2.1           </span></span>
<span><span class="co">##  [76] cachem_1.0.6           downloader_0.4         cli_3.3.0             </span></span>
<span><span class="co">##  [79] generics_0.1.3         RSQLite_2.2.15         ggdendro_0.1.23       </span></span>
<span><span class="co">##  [82] evaluate_0.15          stringr_1.4.0          fastmap_1.1.0         </span></span>
<span><span class="co">##  [85] yaml_2.3.5             ragg_1.2.2             ggtree_3.4.1          </span></span>
<span><span class="co">##  [88] knitr_1.39             bit64_4.0.5            fs_1.5.2              </span></span>
<span><span class="co">##  [91] tidygraph_1.2.1        purrr_0.3.4            dendextend_1.16.0     </span></span>
<span><span class="co">##  [94] KEGGREST_1.36.3        ReactomePA_1.40.0      nlme_3.1-157          </span></span>
<span><span class="co">##  [97] mime_0.12              slam_0.1-50            aplot_0.1.6           </span></span>
<span><span class="co">## [100] DO.db_2.9              xml2_1.3.3             compiler_4.2.1        </span></span>
<span><span class="co">## [103] rstudioapi_0.13        png_0.1-7              treeio_1.20.1         </span></span>
<span><span class="co">## [106] tibble_3.1.7           tweenr_1.0.2           cyjShiny_1.0.34       </span></span>
<span><span class="co">## [109] bslib_0.4.0            stringi_1.7.8          highr_0.9             </span></span>
<span><span class="co">## [112] desc_1.4.1             easyPubMed_2.13        lattice_0.20-45       </span></span>
<span><span class="co">## [115] Matrix_1.4-1           vctrs_0.4.1            pillar_1.8.0          </span></span>
<span><span class="co">## [118] lifecycle_1.0.1        BiocManager_1.30.18    jquerylib_0.1.4       </span></span>
<span><span class="co">## [121] GlobalOptions_0.1.2    cowplot_1.1.1          data.table_1.14.2     </span></span>
<span><span class="co">## [124] bitops_1.0-7           httpuv_1.6.5           patchwork_1.1.1       </span></span>
<span><span class="co">## [127] qvalue_2.28.0          R6_2.5.1               promises_1.2.0.1      </span></span>
<span><span class="co">## [130] bookdown_0.29          gridExtra_2.3          codetools_0.2-18      </span></span>
<span><span class="co">## [133] MASS_7.3-57            assertthat_0.2.1       rprojroot_2.0.3       </span></span>
<span><span class="co">## [136] rjson_0.2.21           withr_2.5.0            GenomeInfoDbData_1.2.8</span></span>
<span><span class="co">## [139] parallel_4.2.1         ISOcodes_2022.01.10    clusterProfiler_4.4.4 </span></span>
<span><span class="co">## [142] grid_4.2.1             ggfun_0.0.6            tidyr_1.2.0           </span></span>
<span><span class="co">## [145] rmarkdown_2.14         ggforce_0.3.3          base64enc_0.1-3       </span></span>
<span><span class="co">## [148] shiny_1.7.2            NLP_0.2-1</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
