<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="wcGeneSummary">
<title>Mining microbiome-related literature • wcGeneSummary</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Mining microbiome-related literature">
<meta property="og:description" content="wcGeneSummary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">wcGeneSummary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/application_example_BKPyV.html">Application example</a>
    <a class="dropdown-item" href="../articles/basic_usage_of_wcGeneSummary.html">wcGeneSummary</a>
    <a class="dropdown-item" href="../articles/microbiome_usage.html">Mining microbiome-related literature</a>
    <a class="dropdown-item" href="../articles/pathview_text.html">Pathview text function</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Mining microbiome-related literature</h1>
                        <h4 data-toc-skip class="author">Noriaki
Sato</h4>
            
      
      
      <div class="d-none name"><code>microbiome_usage.Rmd</code></div>
    </div>

    
    
<p>The general usage for mining the information related to
microbiome-related signatures.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span></code></pre></div>
<p>We use BugSigDB, and its R port bugsigdbr to obtain the curated
dataset of the relationship with bacterial taxonomy and human diseases
(<a href="https://bugsigdb.org/Main_Page" class="external-link">Geistlinger et al. 2022</a>).
Users can query microbiome names, which will be searched for MetaPhlAn
taxonomic annotation. If <code>target="title"</code>, the title of the
corresponding articles will be summarized.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span><span class="op">)</span>,tag<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html" class="external-link">makeCluster</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>    pal<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">10</span>, <span class="st">"Set2"</span><span class="op">)</span>,numWords<span class="op">=</span><span class="fl">80</span>,min.freq<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input microbes: 2</span></span>
<span><span class="co">##   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">##   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">## Including 28 entries ...</span></span>
<span><span class="co">## filter based on BugSigDB</span></span>
<span><span class="co">## Filtering 0 words (frequency and/or tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic</span><span class="op">@</span><span class="va">freqDf</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    word freq</span></span>
<span><span class="co">## gut                 Gut    8</span></span>
<span><span class="co">## oral               Oral    5</span></span>
<span><span class="co">## patients       patients    5</span></span>
<span><span class="co">## study             study    3</span></span>
<span><span class="co">## arthritis     arthritis    2</span></span>
<span><span class="co">## association association    2</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<p><img src="microbiome_usage_files/figure-html/bsdb_basic-1.png" width="768">
If <code>target="abstract"</code>, the corresponding abstract will be
fetched and be summarized.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span><span class="op">)</span>,tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"abstract"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html" class="external-link">makeCluster</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>    pal<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">10</span>, <span class="st">"Set2"</span><span class="op">)</span>,numWords<span class="op">=</span><span class="fl">80</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input microbes: 2</span></span>
<span><span class="co">##   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">##   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">## Including 28 entries ...</span></span>
<span><span class="co">## Target is abstract ...</span></span>
<span><span class="co">##   Querying PubMed for 17 pmids ...</span></span>
<span><span class="co">##   Querying without API key ...</span></span>
<span><span class="co">## filter based on BugSigDB</span></span>
<span><span class="co">## Filtering 0 words (frequency and/or tfidf) ...</span></span>
<span><span class="co">## Multiscale bootstrap... Done.</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic2</span><span class="op">@</span><span class="va">freqDf</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                word freq</span></span>
<span><span class="co">## patients   patients   37</span></span>
<span><span class="co">## gut             Gut   30</span></span>
<span><span class="co">## oral           Oral   27</span></span>
<span><span class="co">## bacterial Bacterial   25</span></span>
<span><span class="co">## species     species   24</span></span>
<span><span class="co">## microbial Microbial   23</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic2</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<p><img src="microbiome_usage_files/figure-html/bsdb_basic2-1.png" width="768">
For successful visualization, pre-caculated TF-IDF and frequency data
frame is available and one can use them to filter the highly occurring
words, or the other prefiltering option used in
<code>wcGeneSummary</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rmwords</span> <span class="op">&lt;-</span> <span class="fu">wcGeneSummary</span><span class="fu">:::</span><span class="va">allTfIdfBSDB</span></span>
<span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="va">rmwords</span><span class="op">[</span><span class="va">rmwords</span><span class="op">$</span><span class="va">tfidf</span><span class="op">&gt;</span><span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">rmwords</span><span class="op">$</span><span class="va">tfidf</span>, <span class="fl">0.95</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">filter</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##               tfidf       word</span></span>
<span><span class="co">## microbiome 36.97275 microbiome</span></span>
<span><span class="co">## gut        34.20444        gut</span></span>
<span><span class="co">## patients   32.38373   patients</span></span>
<span><span class="co">## microbiota 31.96498 microbiota</span></span>
<span><span class="co">## cancer     24.63359     cancer</span></span>
<span><span class="co">## oral       22.65245       oral</span></span></code></pre>
<p>The network visualization is possible by enabling
<code>plotType="network"</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span><span class="op">)</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input microbes: 2</span></span>
<span><span class="co">##   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">##   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">## Including 28 entries ...</span></span>
<span><span class="co">## filter based on BugSigDB</span></span>
<span><span class="co">## Filtering 0 words (frequency and/or tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<p><img src="microbiome_usage_files/figure-html/bsdb_basic_network-1.png" width="768"></p>
<p>The words-to-species relationship can be plotted by
<code>mbPlot=TRUE</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span>,</span>
<span>    <span class="st">"Veillonella parvula"</span>,<span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>, mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input microbes: 4</span></span>
<span><span class="co">##   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">##   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">##   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">##   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">## Including 90 entries ...</span></span>
<span><span class="co">## filter based on BugSigDB</span></span>
<span><span class="co">## Filtering 0 words (frequency and/or tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<p><img src="microbiome_usage_files/figure-html/bsdb_basic_network_mb-1.png" width="768"></p>
<p>As the BugSigDB contains the relationship between bacterial taxonomy
and disease, disease name can also be plotted. When
<code>disPlot=TRUE</code>, the <code>mbPlot</code> will be set to
<code>TRUE</code> by default.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span>,</span>
<span>    <span class="st">"Veillonella parvula"</span>,<span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>, mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>, disPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input microbes: 4</span></span>
<span><span class="co">##   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">##   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">##   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">##   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">## Including 90 entries ...</span></span>
<span><span class="co">## filter based on BugSigDB</span></span>
<span><span class="co">## Filtering 0 words (frequency and/or tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net3</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<p><img src="microbiome_usage_files/figure-html/bsdb_basic_network_dis-1.png" width="960"></p>
<p>Other than curated databases, the PubMed query can also be performed
with setting <code>curate=FALSE</code>. This way, the text information
of the latest literature for the microbes and diseases can be plotted.
The options for use in function obtaining PubMed information can be
specified to <code>abstArg</code> in list format.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span>,</span>
<span>    <span class="st">"Veillonella parvula"</span>,<span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>, mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">FALSE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>, disPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    abstArg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>retMax<span class="op">=</span><span class="fl">80</span>, sortOrder<span class="op">=</span><span class="st">"pubdate"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Input microbes: 4</span></span>
<span><span class="co">##   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">##   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">##   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">##   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">## Including 90 entries ...</span></span>
<span><span class="co">## Proceeding without API key</span></span>
<span><span class="co">## filter based on BugSigDB</span></span>
<span><span class="co">## Filtering 0 words (frequency and/or tfidf) ...</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net4</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<p><img src="microbiome_usage_files/figure-html/no_curate-1.png" width="960"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span><span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">## Running under: Windows 10 x64 (build 22621)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] LC_COLLATE=Japanese_Japan.utf8  LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">## [3] LC_MONETARY=Japanese_Japan.utf8 LC_NUMERIC=C                   </span></span>
<span><span class="co">## [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] RColorBrewer_1.1-3   ggraph_2.1.0         ggplot2_3.4.0       </span></span>
<span><span class="co">## [4] wcGeneSummary_0.99.0 BiocStyle_2.26.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] GeneSummary_0.99.4     colorspace_2.0-3       rjson_0.2.21          </span></span>
<span><span class="co">##   [4] ISOcodes_2022.09.29    ellipsis_0.3.2         rprojroot_2.0.3       </span></span>
<span><span class="co">##   [7] XVector_0.38.0         ggdendro_0.1.23        base64enc_0.1-3       </span></span>
<span><span class="co">##  [10] GlobalOptions_0.1.2    fs_1.5.2               rstudioapi_0.14       </span></span>
<span><span class="co">##  [13] farver_2.1.1           graphlayouts_0.8.4     ggrepel_0.9.2         </span></span>
<span><span class="co">##  [16] bit64_4.0.5            AnnotationDbi_1.60.0   fansi_1.0.3           </span></span>
<span><span class="co">##  [19] xml2_1.3.3             codetools_0.2-18       cachem_1.0.6          </span></span>
<span><span class="co">##  [22] knitr_1.41             polyclip_1.10-4        jsonlite_1.8.4        </span></span>
<span><span class="co">##  [25] dbplyr_2.3.0           png_0.1-8              graph_1.76.0          </span></span>
<span><span class="co">##  [28] ggforce_0.4.1          shiny_1.7.4            BiocManager_1.30.19   </span></span>
<span><span class="co">##  [31] bugsigdbr_1.5.3        rentrez_1.2.3          compiler_4.2.2        </span></span>
<span><span class="co">##  [34] httr_1.4.4             assertthat_0.2.1       fastmap_1.1.0         </span></span>
<span><span class="co">##  [37] cli_3.6.0              later_1.3.0            tweenr_2.0.2          </span></span>
<span><span class="co">##  [40] htmltools_0.5.4        tools_4.2.2            igraph_1.3.5          </span></span>
<span><span class="co">##  [43] NLP_0.2-1              gtable_0.3.1           glue_1.6.2            </span></span>
<span><span class="co">##  [46] GenomeInfoDbData_1.2.9 dplyr_1.0.10           rappdirs_0.3.3        </span></span>
<span><span class="co">##  [49] Rcpp_1.0.9             slam_0.1-50            Biobase_2.58.0        </span></span>
<span><span class="co">##  [52] jquerylib_0.1.4        pkgdown_2.0.7          vctrs_0.5.1           </span></span>
<span><span class="co">##  [55] Biostrings_2.66.0      xfun_0.36              stringr_1.5.0         </span></span>
<span><span class="co">##  [58] stopwords_2.3          mime_0.12              lifecycle_1.0.3       </span></span>
<span><span class="co">##  [61] pvclust_2.2-0          dendextend_1.16.0      XML_3.99-0.13         </span></span>
<span><span class="co">##  [64] org.Hs.eg.db_3.16.0    zlibbioc_1.44.0        MASS_7.3-58.1         </span></span>
<span><span class="co">##  [67] scales_1.2.1           tidygraph_1.2.2        ragg_1.2.5            </span></span>
<span><span class="co">##  [70] promises_1.2.0.1       parallel_4.2.2         cyjShiny_1.0.34       </span></span>
<span><span class="co">##  [73] curl_5.0.0             yaml_2.3.6             memoise_2.0.1         </span></span>
<span><span class="co">##  [76] gridExtra_2.3          yulab.utils_0.0.6      sass_0.4.4            </span></span>
<span><span class="co">##  [79] stringi_1.7.12         RSQLite_2.2.20         highr_0.10            </span></span>
<span><span class="co">##  [82] S4Vectors_0.36.1       desc_1.4.2             filelock_1.0.2        </span></span>
<span><span class="co">##  [85] BiocGenerics_0.44.0    GenomeInfoDb_1.34.6    rlang_1.0.6           </span></span>
<span><span class="co">##  [88] pkgconfig_2.0.3        systemfonts_1.0.4      bitops_1.0-7          </span></span>
<span><span class="co">##  [91] evaluate_0.19          purrr_1.0.1            labeling_0.4.2        </span></span>
<span><span class="co">##  [94] htmlwidgets_1.6.1      patchwork_1.1.2        cowplot_1.1.1         </span></span>
<span><span class="co">##  [97] bit_4.0.5              tidyselect_1.2.0       magrittr_2.0.3        </span></span>
<span><span class="co">## [100] bookdown_0.31          R6_2.5.1               IRanges_2.32.0        </span></span>
<span><span class="co">## [103] generics_0.1.3         DBI_1.1.3              pillar_1.8.1          </span></span>
<span><span class="co">## [106] withr_2.5.0            KEGGREST_1.38.0        RCurl_1.98-1.9        </span></span>
<span><span class="co">## [109] tibble_3.1.8           crayon_1.5.2           wordcloud_2.6         </span></span>
<span><span class="co">## [112] utf8_1.2.2             BiocFileCache_2.6.0    rmarkdown_2.19        </span></span>
<span><span class="co">## [115] viridis_0.6.2          GetoptLong_1.0.5       grid_4.2.2            </span></span>
<span><span class="co">## [118] blob_1.2.3             digest_0.6.31          xtable_1.8-4          </span></span>
<span><span class="co">## [121] tm_0.7-10              tidyr_1.2.1            httpuv_1.6.8          </span></span>
<span><span class="co">## [124] gridGraphics_0.5-1     textshaping_0.3.6      stats4_4.2.2          </span></span>
<span><span class="co">## [127] munsell_0.5.0          ggplotify_0.1.0        viridisLite_0.4.1     </span></span>
<span><span class="co">## [130] bslib_0.4.2</span></span></code></pre>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
