<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="wcGeneSummary">
<title>Pathview text function • wcGeneSummary</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pathview text function">
<meta property="og:description" content="wcGeneSummary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">wcGeneSummary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/application_example_BKPyV.html">Application example</a>
    <a class="dropdown-item" href="../articles/basic_usage_of_wcGeneSummary.html">wcGeneSummary</a>
    <a class="dropdown-item" href="../articles/microbiome_usage.html">Mining microbiome-related literature</a>
    <a class="dropdown-item" href="../articles/pathview_text.html">Pathview text function</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Pathview text function</h1>
                        <h4 data-toc-skip class="author">Noriaki
Sato</h4>
            
      
      
      <div class="d-none name"><code>pathview_text.Rmd</code></div>
    </div>

    
    
<p>KEGG mapping plots produced by pathview can be populated with text
information (<a href="https://academic.oup.com/bioinformatics/article/29/14/1830/232698%5D" class="external-link">Luo
and Brouwer. 2013</a>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/datapplab/pathview" class="external-link">pathview</a></span><span class="op">)</span></span></code></pre></div>
<p>Suppose an example for KEGG Orthology obtained from some
microbiome-related experiments. By using <a href="https://github.com/YuLab-SMU/MicrobiomeProfiler" class="external-link">MicrobiomeProfiler</a>,
enriched pathway using over-representation analysis can be obtained as
follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"K00640"</span>,<span class="st">"K23304"</span>,<span class="st">"K01760"</span>,<span class="st">"K01697"</span>,<span class="st">"K00549"</span>,<span class="st">"K00899"</span><span class="op">)</span></span>
<span><span class="va">enriched</span> <span class="op">&lt;-</span> <span class="fu">MicrobiomeProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MicrobiomeProfiler/man/enrichKO.html" class="external-link">enrichKO</a></span><span class="op">(</span><span class="va">kos</span><span class="op">)</span></span>
<span><span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html" class="external-link">cnetplot</a></span><span class="op">(</span><span class="va">enriched</span><span class="op">)</span></span></code></pre></div>
<p><img src="pathview_text_files/figure-html/micro-1.png" width="700">
Now plot these KOs in the pathview.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## As the output is PNG, this script is omitted.</span></span>
<span><span class="va">pvInput</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">kos</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pvInput</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">kos</span></span>
<span><span class="co"># pathview(kos, pathway.id = "00270", species = "ko")</span></span></code></pre></div>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/ko00270.pathview.png?raw=true" alt=""><p class="caption">raw pathview plot</p>
</div>
<p>To use the <code><a href="../reference/pathviewText.html">pathviewText()</a></code> one should specify which
database to retrieve the text, <code>"refseq"</code> or
<code>"abstract"</code>. The default is “abstract”, which fetches
information from RefSeq database. In this case, KO number is not
included in RefSeq, thus the abstract is chosen. However, querying in KO
number itself is not useful. We conver the KO number to EC number using
KEGG REST API, and fetches the description using <code><a href="../reference/wcEC.html">wcEC()</a></code>
function, and query the enzyme name in PubMed. In this way, one must
provide <code>searchTerm</code>, and <code>termMap</code>, which map the
KO and EC. Also, <code>node.types="ortholog"</code> must be passed to
<code>pathview</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Again, the codes are commented out.</span></span>
<span></span>
<span><span class="co"># ecko &lt;- data.table::fread("https://rest.kegg.jp/link/ko/ec",</span></span>
<span><span class="co">#                             header=FALSE)</span></span>
<span><span class="co"># ecko$EC &lt;- sapply(strsplit(ecko$V1, ":"),"[",2)</span></span>
<span><span class="co"># ecko$KO &lt;- sapply(strsplit(ecko$V2, ":"), "[", 2)</span></span>
<span><span class="co"># filt &lt;- ecko[ ecko$KO %in% kos, ]</span></span>
<span><span class="co"># ecnum &lt;- filt$EC</span></span>
<span><span class="co"># termMap &lt;- filt[,c("KO","EC")] |&gt; `colnames&lt;-`(c("query","number"))</span></span>
<span><span class="co"># ecQuery &lt;- wcEC("enzyme.dat", ecnum = ecnum, onlyTerm =TRUE)</span></span>
<span><span class="co"># ecMap &lt;- wcEC("enzyme.dat", ecnum = ecnum, onlyDf =TRUE)</span></span>
<span><span class="co"># termMap &lt;- merge(termMap, ecMap[,c("number","desc")], by="number") |&gt;</span></span>
<span><span class="co">#   `colnames&lt;-`(c("number","query","description"))</span></span>
<span></span>
<span><span class="co">## Main function</span></span>
<span></span>
<span><span class="co"># pathviewText(kos, keyType = "KO",</span></span>
<span><span class="co">#      target="abstract",</span></span>
<span><span class="co">#      pid = "00270",</span></span>
<span><span class="co">#      org = "ko", </span></span>
<span><span class="co">#      searchTerms = ecQuery,</span></span>
<span><span class="co">#      termMap = termMap,</span></span>
<span><span class="co">#      node.types="ortholog")</span></span>
<span></span>
<span><span class="co">## We can transpose the barplot by trans=TRUE</span></span>
<span></span>
<span><span class="co"># pathviewText(kos, keyType = "KO",</span></span>
<span><span class="co">#      target="abstract",</span></span>
<span><span class="co">#      pid = "00270",</span></span>
<span><span class="co">#      org = "ko", </span></span>
<span><span class="co">#      searchTerms = ecQuery,</span></span>
<span><span class="co">#      termMap = termMap,</span></span>
<span><span class="co">#      node.types="ortholog")</span></span></code></pre></div>
<p>The below picture shows transposed results.</p>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/pathviewText.png?raw=true" alt=""><p class="caption">annotated pathview plot - KO</p>
</div>
<p>For gene queries, we can obtain the same plot using relatively simple
query.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TP53"</span>,<span class="st">"CDC45"</span>,<span class="st">"CDC6"</span><span class="op">)</span></span></code></pre></div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
