<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 Pathview text | wcGeneSummary (OSplot)</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.33 with bs4_book()">
<meta property="og:title" content="Chapter 9 Pathview text | wcGeneSummary (OSplot)">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/wcGeneSummary/pathview-text.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 Pathview text | wcGeneSummary (OSplot)">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="9.1 KO queries KEGG mapping plots produced by pathview can be populated with text information (Luo and Brouwer. 2013). library(wcGeneSummary) library(dplyr) library(pathview);library(png) Suppose...">
<meta property="og:description" content="9.1 KO queries KEGG mapping plots produced by pathview can be populated with text information (Luo and Brouwer. 2013). library(wcGeneSummary) library(dplyr) library(pathview);library(png) Suppose...">
<meta name="twitter:description" content="9.1 KO queries KEGG mapping plots produced by pathview can be populated with text information (Luo and Brouwer. 2013). library(wcGeneSummary) library(dplyr) library(pathview);library(png) Suppose...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">wcGeneSummary (OSplot)</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="" href="the-basic-usage-for-microbiome-related-data.html"><span class="header-section-number">3</span> The basic usage for microbiome-related data</a></li>
<li><a class="" href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">4</span> Interactive inspection of gene cluster network annotated by words</a></li>
<li><a class="" href="annotating-gene-cluster-dendrogram.html"><span class="header-section-number">5</span> Annotating gene cluster dendrogram</a></li>
<li><a class="" href="assess-the-occurrence-of-the-speicific-words-across-gene-clusters.html"><span class="header-section-number">6</span> Assess the occurrence of the speicific words across gene clusters</a></li>
<li><a class="" href="recluster-the-cluster-using-word-information.html"><span class="header-section-number">7</span> Recluster the cluster using word information</a></li>
<li><a class="" href="including-the-other-databases-an-example-in-metacyc.html"><span class="header-section-number">8</span> Including the other databases (an example in MetaCyc)</a></li>
<li><a class="active" href="pathview-text.html"><span class="header-section-number">9</span> Pathview text</a></li>
<li><a class="" href="application-examples.html"><span class="header-section-number">10</span> Application examples</a></li>
<li><a class="" href="single-cell-transcriptomic-data.html"><span class="header-section-number">11</span> Single-cell transcriptomic data</a></li>
<li><a class="" href="parsing.html"><span class="header-section-number">12</span> Parsing</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="pathview-text" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> Pathview text<a class="anchor" aria-label="anchor" href="#pathview-text"><i class="fas fa-link"></i></a>
</h1>
<div id="ko-queries" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> KO queries<a class="anchor" aria-label="anchor" href="#ko-queries"><i class="fas fa-link"></i></a>
</h2>
<p>KEGG mapping plots produced by pathview can be populated with text information (<a href="https://academic.oup.com/bioinformatics/article/29/14/1830/232698%5D">Luo and Brouwer. 2013</a>).</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/datapplab/pathview">pathview</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.rforge.net/png/">png</a></span><span class="op">)</span></span></code></pre></div>
<p>Suppose an example for KEGG Orthology obtained from some microbiome-related experiments.
By using <a href="https://github.com/YuLab-SMU/MicrobiomeProfiler">MicrobiomeProfiler</a>, enriched pathway using over-representation analysis can be obtained as follows.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"K00640"</span>,<span class="st">"K23304"</span>,<span class="st">"K01760"</span>,<span class="st">"K01697"</span>,<span class="st">"K00549"</span>,<span class="st">"K00899"</span><span class="op">)</span></span>
<span><span class="va">enriched</span> <span class="op">&lt;-</span> <span class="fu">MicrobiomeProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MicrobiomeProfiler/man/enrichKO.html">enrichKO</a></span><span class="op">(</span><span class="va">kos</span><span class="op">)</span></span>
<span><span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">enriched</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure">
<img src="05-pathview_text_files/figure-html/micro-1.png" width="672">
Now plot these KOs in the pathview.</div>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## As the output is PNG, this script is omitted.</span></span>
<span><span class="va">pvInput</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">kos</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pvInput</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">kos</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pathview/man/pathview.html">pathview</a></span><span class="op">(</span><span class="va">kos</span>, pathway.id <span class="op">=</span> <span class="st">"00270"</span>, species <span class="op">=</span> <span class="st">"ko"</span><span class="op">)</span></span>
<span><span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.raster.html">rasterGrob</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/png/man/readPNG.html">readPNG</a></span><span class="op">(</span><span class="st">"ko00270.pathview.png"</span><span class="op">)</span>, interpolate<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; rastergrob[GRID.rastergrob.93]</span></span></code></pre></div>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/ko00270.pathview.png?raw=true" width="500" alt=""><p class="caption">raw pathview plot</p>
</div>
<p>To use the <code><a href="https://rdrr.io/pkg/wcGeneSummary/man/pathviewText.html">pathviewText()</a></code> one should specify which database to retrieve the text, <code>"refseq"</code> or <code>"abstract"</code>.
The default is “refseq”, which fetches information from RefSeq database. In this case, KO number is not included in RefSeq, thus the abstract is chosen. However, querying in KO number itself is not useful. We conver the KO number to EC number using KEGG REST API, and fetches the description using <code><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcEC.html">wcEC()</a></code> function, and query the enzyme name in PubMed. In this way, one must provide <code>searchTerm</code> obtained by <code>onlyTerm=TRUE</code>, and <code>termMap</code>, which map the KO and EC. Note that <code>termMap</code> must have <code>query</code> and <code>description</code> columns. Also, <code>node.types="ortholog"</code> must be passed to <code>pathview</code>.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">ecko</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/ko/ec"</span>,</span>
<span>                            header<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ecko</span><span class="op">$</span><span class="va">EC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">ecko</span><span class="op">$</span><span class="va">V1</span>, <span class="st">":"</span><span class="op">)</span>,<span class="st">"["</span>,<span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">ecko</span><span class="op">$</span><span class="va">KO</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">ecko</span><span class="op">$</span><span class="va">V2</span>, <span class="st">":"</span><span class="op">)</span>, <span class="st">"["</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">filt</span> <span class="op">&lt;-</span> <span class="va">ecko</span><span class="op">[</span> <span class="va">ecko</span><span class="op">$</span><span class="va">KO</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">kos</span>, <span class="op">]</span></span>
<span><span class="va">ecnum</span> <span class="op">&lt;-</span> <span class="va">filt</span><span class="op">$</span><span class="va">EC</span></span>
<span><span class="va">termMap</span> <span class="op">&lt;-</span> <span class="va">filt</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"KO"</span>,<span class="st">"EC"</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu">`colnames&lt;-`</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"query"</span>,<span class="st">"number"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ecQuery</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcEC.html">wcEC</a></span><span class="op">(</span><span class="st">"../enzyme.dat"</span>, ecnum <span class="op">=</span> <span class="va">ecnum</span>, onlyTerm <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Processing EC file</span></span>
<span><span class="va">ecMap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcEC.html">wcEC</a></span><span class="op">(</span><span class="st">"../enzyme.dat"</span>, ecnum <span class="op">=</span> <span class="va">ecnum</span>, onlyDf <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Processing EC file</span></span>
<span><span class="va">termMap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">termMap</span>, <span class="va">ecMap</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"number"</span>,<span class="st">"desc"</span><span class="op">)</span><span class="op">]</span>, by<span class="op">=</span><span class="st">"number"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">`colnames&lt;-`</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"number"</span>,<span class="st">"query"</span>,<span class="st">"description"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Main function</span></span>
<span></span>
<span><span class="va">hor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/pathviewText.html">pathviewText</a></span><span class="op">(</span><span class="va">kos</span>, keyType <span class="op">=</span> <span class="st">"KO"</span>,</span>
<span>     target<span class="op">=</span><span class="st">"abstract"</span>,</span>
<span>     pid <span class="op">=</span> <span class="st">"00270"</span>,</span>
<span>     org <span class="op">=</span> <span class="st">"ko"</span>, </span>
<span>     searchTerms <span class="op">=</span> <span class="va">ecQuery</span>,</span>
<span>     termMap <span class="op">=</span> <span class="va">termMap</span>,</span>
<span>     node.types<span class="op">=</span><span class="st">"ortholog"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Proceeding without API key</span></span>
<span></span>
<span><span class="va">hor</span><span class="op">$</span><span class="va">concat</span></span></code></pre></div>
<div class="inline-figure"><img src="05-pathview_text_files/figure-html/pathview2-1.png" width="672"></div>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## We can transpose the barplot by trans=TRUE</span></span>
<span></span>
<span><span class="va">ver</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/pathviewText.html">pathviewText</a></span><span class="op">(</span><span class="va">kos</span>, keyType <span class="op">=</span> <span class="st">"KO"</span>,</span>
<span>     target<span class="op">=</span><span class="st">"abstract"</span>,</span>
<span>     pid <span class="op">=</span> <span class="st">"00270"</span>,</span>
<span>     org <span class="op">=</span> <span class="st">"ko"</span>, </span>
<span>       trans <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>     searchTerms <span class="op">=</span> <span class="va">ecQuery</span>,</span>
<span>     termMap <span class="op">=</span> <span class="va">termMap</span>,</span>
<span>     node.types<span class="op">=</span><span class="st">"ortholog"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Proceeding without API key</span></span>
<span></span>
<span><span class="va">ver</span><span class="op">$</span><span class="va">concat</span></span></code></pre></div>
<div class="inline-figure"><img src="05-pathview_text_files/figure-html/pathview2-2.png" width="672"></div>
</div>
<div id="gene-queries" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> Gene queries<a class="anchor" aria-label="anchor" href="#gene-queries"><i class="fas fa-link"></i></a>
</h2>
<p>For gene queries, we can obtain the same plot using relatively simple query.</p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TP53"</span>,<span class="st">"CDC45"</span>,<span class="st">"CDC6"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ver</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/pathviewText.html">pathviewText</a></span><span class="op">(</span><span class="va">query</span>,</span>
<span>     keyType <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>     target<span class="op">=</span><span class="st">"abstract"</span>,</span>
<span>     pid <span class="op">=</span> <span class="st">"04110"</span>,</span>
<span>     org <span class="op">=</span> <span class="st">"hsa"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Converting to ENTREZID</span></span>
<span><span class="co">#&gt;   Converted input genes: 3</span></span>
<span><span class="co">#&gt; Proceeding without API key</span></span>
<span></span>
<span><span class="va">ver</span><span class="op">$</span><span class="va">concat</span></span></code></pre></div>
<div class="inline-figure"><img src="05-pathview_text_files/figure-html/gene-1.png" width="672"></div>
<p>The returned <code>osplot</code> object can be accessed as the name <code>text</code> in the list. Note that only the nodes (words) within the final network are to be plotted. If all the words are to be included, path the arguments to <code>argList</code> to include all.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ver</span><span class="op">$</span><span class="va">text</span></span>
<span><span class="co">#&gt; Type: pubmed_abstract</span></span>
<span><span class="co">#&gt; Number of words: 20</span></span>
<span><span class="co">#&gt; TP53 OR CDC45 OR CDC6</span></span>
<span><span class="co">#&gt; 380.3 Kb</span></span>
<span></span>
<span><span class="va">ver</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/pathviewText.html">pathviewText</a></span><span class="op">(</span><span class="va">query</span>,</span>
<span>     keyType <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>     target<span class="op">=</span><span class="st">"abstract"</span>,</span>
<span>     pid <span class="op">=</span> <span class="st">"04110"</span>,</span>
<span>     org <span class="op">=</span> <span class="st">"hsa"</span>, argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>numWords<span class="op">=</span><span class="cn">Inf</span>, corThresh<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Converting to ENTREZID</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 3</span></span>
<span><span class="co">#&gt; Info: Downloading xml files for hsa04110, 1/1 pathways..</span></span>
<span><span class="co">#&gt; Info: Downloading png files for hsa04110, 1/1 pathways..</span></span>
<span><span class="co">#&gt; Info: Working in directory C:/Users/nsato/Dropbox/build_wgcs/book</span></span>
<span><span class="co">#&gt; Info: Writing image file hsa04110.custom.cols.png</span></span>
<span><span class="co">#&gt; Proceeding without API key</span></span>
<span></span>
<span><span class="va">ver</span><span class="op">$</span><span class="va">concat</span></span></code></pre></div>
<div class="inline-figure"><img src="05-pathview_text_files/figure-html/assess-1.png" width="672"></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="including-the-other-databases-an-example-in-metacyc.html"><span class="header-section-number">8</span> Including the other databases (an example in MetaCyc)</a></div>
<div class="next"><a href="application-examples.html"><span class="header-section-number">10</span> Application examples</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#pathview-text"><span class="header-section-number">9</span> Pathview text</a></li>
<li><a class="nav-link" href="#ko-queries"><span class="header-section-number">9.1</span> KO queries</a></li>
<li><a class="nav-link" href="#gene-queries"><span class="header-section-number">9.2</span> Gene queries</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>wcGeneSummary (OSplot)</strong>" was written by Noriaki Sato. It was last built on 2023-03-30.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
