<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="stana">
<title>stana â€¢ stana</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="stana">
<meta property="og:description" content="stana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">stana</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/usage_of_stana.html">stana</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>stana</h1>
                        <h4 data-toc-skip class="author">Noriaki
Sato</h4>
            
      
      
      <div class="d-none name"><code>usage_of_stana.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="stana">stana<a class="anchor" aria-label="anchor" href="#stana"></a>
</h2>
<p><code>stana</code> is a library for strain-level analysis of
metagenomic dataset using single nucleotide variants profile and gene
contents. Multiple softwares are described, such as <code>MIDAS2</code>,
<code>metaSNV</code>, and <code>InStrain</code>. The library load the
profile to <code>stana</code> class object and perform various analysis
like marker gene detection and visualization of trees based on metadata.
Here, we included the sample stana class object profiled from 26
metagenome samples processed with <code>MIDAS2</code> against UHGG, and
random grouping from <code>PRJEB9584</code>. The data was loaded with
<code>loadMIDAS2</code> function. One species profile, identifier
<code>100003</code> (<code>s__Alistipes shahii</code>) is loaded.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Specify the output of `merge` command with grouping</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadMIDAS2.html">loadMIDAS2</a></span><span class="op">(</span><span class="st">"merge_midas2_sample"</span>, cl<span class="op">=</span><span class="va">randomGroup</span>, candSp<span class="op">=</span><span class="st">"100003"</span>,</span>
<span>                    loadSummary<span class="op">=</span><span class="cn">TRUE</span>,loadInfo<span class="op">=</span><span class="cn">TRUE</span>,loadDepth<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                    taxtbl<span class="op">=</span><span class="va">taxtbl</span>, db<span class="op">=</span><span class="st">"uhgg"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span>
<span><span class="co">## Examine sample object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"stana"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">stana</span></span></code></pre></div>
<pre><code><span><span class="co">## Type: MIDAS2</span></span>
<span><span class="co">## Directory: midas2_sample_merge_uhgg</span></span>
<span><span class="co">## Species: 1</span></span>
<span><span class="co">## Filter type: group, number: 2, proportion: 0.8</span></span>
<span><span class="co">## Loaded SNV table: 1</span></span>
<span><span class="co">##   Species cleared SNV filter: 1</span></span>
<span><span class="co">## Loaded gene table (copynum): 1</span></span>
<span><span class="co">##   Species cleared gene filter: 1</span></span>
<span><span class="co">## 7.2 Mb</span></span></code></pre>
<div class="section level3">
<h3 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h3>
<p>The species that satisfies the filtering criteria specified can be
accessed by the slot <code>clearSnps</code> storing species ID. If
<code>taxtbl</code> was provided on <code>load*</code> functions,
<code>clearSnpsSpecies</code> has species name.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span><span class="op">@</span><span class="va">clearSnps</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "100003"</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stana</span><span class="op">@</span><span class="va">clearSnpsSpecies</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__Alistipes shahii"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="calling-consensus-sequence">Calling consensus sequence<a class="anchor" aria-label="anchor" href="#calling-consensus-sequence"></a>
</h3>
<p>You can call the consensus sequence based on SNV frequency table. The
implementation and filtering options are based on the original
<code>MIDAS</code> script (<code>call_consensus.py</code>). The
resulting FASTA will be written to the current directory, and is loaded
into <code>fastaList</code> slot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn" class="external-link">phangorn</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/" class="external-link">ggtree</a></span><span class="op">)</span></span>
<span><span class="va">stana</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/consensusSeqMIDAS2.html">consensusSeqMIDAS2</a></span><span class="op">(</span><span class="va">stana</span>, species<span class="op">=</span><span class="st">"100003"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Beginning calling for 100003</span></span>
<span><span class="co">##   Site number: 5019</span></span>
<span><span class="co">##   Outputting consensus sequence to 100003_consensus.fasta</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Tree estimation and visualization by `phangorn` and `ggtree`</span></span>
<span><span class="va">dm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/dist.hamming.html" class="external-link">dist.ml</a></span><span class="op">(</span><span class="va">stana</span><span class="op">@</span><span class="va">fastaList</span><span class="op">$</span><span class="va">`100003`</span><span class="op">)</span></span>
<span><span class="va">tre</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/NJ.html" class="external-link">NJ</a></span><span class="op">(</span><span class="va">dm</span><span class="op">)</span></span>
<span><span class="va">tre</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html" class="external-link">groupOTU</a></span><span class="op">(</span><span class="va">tre</span>, <span class="va">stana</span><span class="op">@</span><span class="va">cl</span><span class="op">)</span></span>
<span><span class="va">tp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html" class="external-link">ggtree</a></span><span class="op">(</span><span class="va">tre</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>,</span>
<span>             layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html" class="external-link">geom_tippoint</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">tp</span></span></code></pre></div>
<p><img src="usage_of_stana_files/figure-html/cons-1.png" width="480"></p>
<div class="section level4">
<h4 id="perform-permanova-based-on-the-groups">Perform PERMANOVA based on the groups<a class="anchor" aria-label="anchor" href="#perform-permanova-based-on-the-groups"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/doAdonis.html">doAdonis</a></span><span class="op">(</span><span class="va">stana</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="identify-key-elements-for-differentiating-groups">Identify key elements for differentiating groups<a class="anchor" aria-label="anchor" href="#identify-key-elements-for-differentiating-groups"></a>
</h4>
<p>Using the profiled matrices, the <code>Boruta</code> algorithm can be
run to identify the key elements that distinguish the groups. The
<code>doBoruta</code> function performs the <code>Boruta</code>
algorithm between the clusters specified in <code>cl</code> (if
<code>NULL</code>, the <code>cl</code> slot is used). If pre-processed
matrices are to be used, please pass them to <code>mat</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://gitlab.com/mbq/Boruta/" class="external-link">Boruta</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## You can provide preprocessed matrix to `mat`</span></span>
<span><span class="fu"><a href="../reference/doBoruta.html">doBoruta</a></span><span class="op">(</span><span class="va">stana</span>, target <span class="op">=</span> <span class="st">"gene"</span>, sp<span class="op">=</span><span class="st">"100003"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using grouping from the slot</span></span>
<span><span class="co">## If needed, please provide preprocessed matrix of genes to `mat`</span></span>
<span><span class="co">## Feature number: 21806</span></span>
<span><span class="co">## Performing Boruta</span></span></code></pre>
<pre><code><span><span class="co">## $boruta</span></span>
<span><span class="co">## Boruta performed 99 iterations in 28.17062 secs.</span></span>
<span><span class="co">## Tentatives roughfixed over the last 99 iterations.</span></span>
<span><span class="co">##  12 attributes confirmed important: UHGG000008_01641, UHGG025024_01181,</span></span>
<span><span class="co">## UHGG060667_01243, UHGG061776_01339, UHGG158704_01078 and 7 more;</span></span>
<span><span class="co">##  21794 attributes confirmed unimportant: UHGG000008_00008,</span></span>
<span><span class="co">## UHGG000008_00009, UHGG000008_00010, UHGG000008_00012, UHGG000008_00015</span></span>
<span><span class="co">## and 21789 more;</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="parsing-functional-annotations">Parsing functional annotations<a class="anchor" aria-label="anchor" href="#parsing-functional-annotations"></a>
</h4>
<p>Functional annotation of genes can be obtained through
<code>PATRIC</code> server (used in <code>MIDAS</code>), and annotation
results such as ortholog assignment using <code>eggNOG-mapper v2</code>.
<code>checkPATRIC</code> and <code>checkEGGNOG</code> function can be
used to parse the obtained data. The below example returns KEGG
ORTHOLOGY from <code>eggNOG-mapper v2</code> annotation file.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotationFile</span> <span class="op">&lt;-</span> <span class="st">"100003_uhgg_eggnog_out.emapper.annotations"</span></span>
<span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkEGGNOG.html">checkEGGNOG</a></span><span class="op">(</span><span class="va">annotationFile</span>, ret <span class="op">=</span> <span class="st">"KEGG_ko"</span><span class="op">)</span></span>
<span><span class="va">annot</span></span></code></pre></div>
<p>Using the parsed annotation tibble, gene copy number data frame can
be summarised by user-defined function, such as <code>mean</code>. Note
that annotation tibble must have <code>ID</code> and <code>value</code>
column specifying gene ID in <code>ID</code> and functional annotation
such as KEGG ORTHOLOGY ID in <code>value</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/summariseAbundance.html">summariseAbundance</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100224"</span>, <span class="va">annot</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h4>
<p>Several visualization functions are implemented.</p>
<div class="section level5">
<h5 id="plotting-mean-coverage-across-samples">Plotting mean coverage across samples<a class="anchor" aria-label="anchor" href="#plotting-mean-coverage-across-samples"></a>
</h5>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCoverage.html">plotCoverage</a></span><span class="op">(</span><span class="va">stana</span>, <span class="st">"100003"</span><span class="op">)</span><span class="op">+</span> <span class="co">## grouping is already set</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">ids</span><span class="op">)</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_of_stana_files/figure-html/cov-1.png" width="480"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23 ucrt)</span></span>
<span><span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">## Running under: Windows 10 x64 (build 22621)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] LC_COLLATE=Japanese_Japan.utf8  LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">## [3] LC_MONETARY=Japanese_Japan.utf8 LC_NUMERIC=C                   </span></span>
<span><span class="co">## [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] Boruta_8.0.0     ggtree_3.7.1.002 phangorn_2.11.1  ape_5.7-1       </span></span>
<span><span class="co">## [5] stana_0.99.0     BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] colorspace_2.1-0    rjson_0.2.21        rprojroot_2.0.3    </span></span>
<span><span class="co">##   [4] circlize_0.4.15     GlobalOptions_0.1.2 fs_1.6.1           </span></span>
<span><span class="co">##   [7] aplot_0.1.10        rstudioapi_0.14     farver_2.1.1       </span></span>
<span><span class="co">##  [10] graphlayouts_0.8.4  ggrepel_0.9.3       bit64_4.0.5        </span></span>
<span><span class="co">##  [13] fansi_1.0.4         ranger_0.15.1       codetools_0.2-19   </span></span>
<span><span class="co">##  [16] splines_4.2.1       cachem_1.0.7        knitr_1.42         </span></span>
<span><span class="co">##  [19] polyclip_1.10-4     jsonlite_1.8.0      cluster_2.1.4      </span></span>
<span><span class="co">##  [22] dbplyr_2.3.2        ggforce_0.4.1       BiocManager_1.30.20</span></span>
<span><span class="co">##  [25] compiler_4.2.1      httr_1.4.5          Matrix_1.5-4       </span></span>
<span><span class="co">##  [28] fastmap_1.1.1       lazyeval_0.2.2      cli_3.6.1          </span></span>
<span><span class="co">##  [31] tweenr_2.0.2        htmltools_0.5.5     tools_4.2.1        </span></span>
<span><span class="co">##  [34] igraph_1.4.2        gtable_0.3.3        glue_1.6.2         </span></span>
<span><span class="co">##  [37] dplyr_1.1.1         rappdirs_0.3.3      fastmatch_1.1-3    </span></span>
<span><span class="co">##  [40] Rcpp_1.0.10         jquerylib_0.1.4     pkgdown_2.0.7      </span></span>
<span><span class="co">##  [43] vctrs_0.6.2         nlme_3.1-162        ggraph_2.1.0       </span></span>
<span><span class="co">##  [46] xfun_0.38           stringr_1.5.0       lifecycle_1.0.3    </span></span>
<span><span class="co">##  [49] MASS_7.3-58.3       scales_1.2.1        tidygraph_1.2.3    </span></span>
<span><span class="co">##  [52] ragg_1.2.5          parallel_4.2.1      RColorBrewer_1.1-3 </span></span>
<span><span class="co">##  [55] yaml_2.3.7          curl_5.0.0          memoise_2.0.1      </span></span>
<span><span class="co">##  [58] gridExtra_2.3       ggplot2_3.4.2       ggfun_0.0.9        </span></span>
<span><span class="co">##  [61] yulab.utils_0.0.6   sass_0.4.5          stringi_1.7.12     </span></span>
<span><span class="co">##  [64] RSQLite_2.3.1       highr_0.10          desc_1.4.2         </span></span>
<span><span class="co">##  [67] tidytree_0.4.2      permute_0.9-7       filelock_1.0.2     </span></span>
<span><span class="co">##  [70] shape_1.4.6         rlang_1.1.0         pkgconfig_2.0.3    </span></span>
<span><span class="co">##  [73] systemfonts_1.0.4   bitops_1.0-7        evaluate_0.20      </span></span>
<span><span class="co">##  [76] lattice_0.21-8      purrr_1.0.1         treeio_1.22.0      </span></span>
<span><span class="co">##  [79] patchwork_1.1.2     labeling_0.4.2      bit_4.0.5          </span></span>
<span><span class="co">##  [82] tidyselect_1.2.0    magrittr_2.0.3      bookdown_0.33      </span></span>
<span><span class="co">##  [85] R6_2.5.1            generics_0.1.3      DBI_1.1.3          </span></span>
<span><span class="co">##  [88] pillar_1.9.0        withr_2.5.0         mgcv_1.8-42        </span></span>
<span><span class="co">##  [91] RCurl_1.98-1.12     tibble_3.2.1        crayon_1.5.2       </span></span>
<span><span class="co">##  [94] utf8_1.2.3          BiocFileCache_2.6.1 rmarkdown_2.21     </span></span>
<span><span class="co">##  [97] viridis_0.6.2       GetoptLong_1.0.5    grid_4.2.1         </span></span>
<span><span class="co">## [100] data.table_1.14.8   blob_1.2.4          vegan_2.6-4        </span></span>
<span><span class="co">## [103] digest_0.6.29       tidyr_1.3.0         gridGraphics_0.5-1 </span></span>
<span><span class="co">## [106] textshaping_0.3.6   munsell_0.5.0       viridisLite_0.4.1  </span></span>
<span><span class="co">## [109] ggplotify_0.1.0     bslib_0.4.2         quadprog_1.5-8</span></span></code></pre>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noriaki Sato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
