<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 2 Basic usage | wcGeneSummary (OSplot)</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.31 with bs4_book()">
<meta property="og:title" content="Chapter 2 Basic usage | wcGeneSummary (OSplot)">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/wcGeneSummary/basic-usage.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 2 Basic usage | wcGeneSummary (OSplot)">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="Load the package and the database for converting identifiers. In this example, we use mostly human-derived data, and use org.Hs.eg.db. library(wcGeneSummary) library(org.Hs.eg.db) library(ggplot2)...">
<meta property="og:description" content="Load the package and the database for converting identifiers. In this example, we use mostly human-derived data, and use org.Hs.eg.db. library(wcGeneSummary) library(org.Hs.eg.db) library(ggplot2)...">
<meta name="twitter:description" content="Load the package and the database for converting identifiers. In this example, we use mostly human-derived data, and use org.Hs.eg.db. library(wcGeneSummary) library(org.Hs.eg.db) library(ggplot2)...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">wcGeneSummary (OSplot)</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="active" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="" href="the-basic-usage-for-microbiome-related-data.html"><span class="header-section-number">3</span> The basic usage for microbiome-related data</a></li>
<li><a class="" href="custom-usage---interactive-inspection-of-bayesian-network-annotated-by-words.html"><span class="header-section-number">4</span> Custom usage - Interactive inspection of Bayesian network annotated by words</a></li>
<li><a class="" href="custom-usage---annotating-module-eigengenes-dendrogram.html"><span class="header-section-number">5</span> Custom usage - Annotating module eigengenes dendrogram</a></li>
<li><a class="" href="custom-usage---assess-the-occurrence-of-the-speicific-words-across-gene-clusters.html"><span class="header-section-number">6</span> Custom usage - Assess the occurrence of the speicific words across gene clusters</a></li>
<li><a class="" href="custom-usage---recluster-the-cluster-using-word-information.html"><span class="header-section-number">7</span> Custom usage - Recluster the cluster using word information</a></li>
<li><a class="" href="custom-usage---pathview-text.html"><span class="header-section-number">8</span> Custom usage - pathview text</a></li>
<li><a class="" href="application-examples.html"><span class="header-section-number">9</span> Application examples</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="basic-usage" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"><i class="fas fa-link"></i></a>
</h1>
<p>Load the package and the database for converting identifiers.
In this example, we use mostly human-derived data, and use <code>org.Hs.eg.db</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://talgalili.github.io/dendextend/">dendextend</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div id="producing-word-clouds" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Producing word clouds<a class="anchor" aria-label="anchor" href="#producing-word-clouds"><i class="fas fa-link"></i></a>
</h2>
<p>We use ERCC genes as input. The basic usage of the package is producing a word cloud of gene summaries by querying gene IDs, default to symbol. The other type can be set by <code>keyType</code>. As many of the words are commonly observed, you should limit word frequency by <code>excludeFreq</code>, which is default to 2000. TF-IDF on the all the summary is precomputed, and <code>exclude="tfidf"</code> can be specified too.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Configure input genes</span></span>
<span><span class="va">inpSymbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ERCC1"</span>,<span class="st">"ERCC2"</span>,<span class="st">"ERCC3"</span>,<span class="st">"ERCC4"</span>,<span class="st">"ERCC5"</span>,<span class="st">"ERCC6"</span>,<span class="st">"ERCC8"</span><span class="op">)</span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="va">gwc</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/basic-1.png" width="768"></div>
<p>It accepts values of the <code>wordcloud()</code> function. <code>numWords</code> specifies how many words are to be shown on word cloud. The words are ordered by their frequency, and the subset of 1:<code>numWords</code> is used to downstream visualization.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use palette from palettetown, palettetown::pokepal(150) (Mewtwo!)</span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>                     excludeFreq<span class="op">=</span><span class="fl">5000</span>,</span>
<span>                     argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>                      <span class="st">"random.order"</span><span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     <span class="st">"colors"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html">palette</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     <span class="st">"rot.per"</span><span class="op">=</span><span class="fl">0.4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 14 words (frequency and/or tfidf)</span></span>
<span><span class="va">gwc</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/basic2-1.png" width="768"></div>
<p>By default, <code>preserve=TRUE</code>, which indicates the funciton tries to preserve the original cases of characters.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, </span>
<span>                     numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>                     excludeFreq<span class="op">=</span><span class="fl">5000</span>,</span>
<span>                     preserve<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>                       rot.per<span class="op">=</span><span class="fl">0.4</span>,</span>
<span>                       colors<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span>,</span>
<span>                       random.order<span class="op">=</span><span class="cn">FALSE</span></span>
<span>                     <span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 14 words (frequency and/or tfidf)</span></span>
<span><span class="va">gwc_p</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/basic_pres_false-1.png" width="768"></div>
<p>It also returns a data frame consisting of frequency of each term.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span></span>
<span><span class="co">#&gt; type: refseq</span></span>
<span><span class="co">#&gt; Number of words: 100</span></span>
<span><span class="co">#&gt; ERCC1/ERCC2/ERCC3/ERCC4/ERCC5/ERCC6/ERCC8</span></span>
<span><span class="co">#&gt; 133.5 Kb</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gwc</span><span class="op">@</span><span class="va">freqDf</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'Term frequencies.'</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:table">Table 2.1: </span>Term frequencies.</caption>
<thead><tr class="header">
<th align="left">word</th>
<th align="right">freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">repair</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">DNA</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">syndrome</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">excision</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">transcription</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">Cockayne</td>
<td align="right">6</td>
</tr>
</tbody>
</table></div>
<p>N-gram is supported by library <code>tm</code>, specified by <code>ngram</code>.
Default is <code>1</code>, and the example specifying <code>2</code> is shown below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                      ngram<span class="op">=</span><span class="fl">2</span>,</span>
<span>                      numWords<span class="op">=</span><span class="fl">50</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="va">gwc2</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/ngramwc-1.png" width="768"></div>
<p>Using <code>clusterProfiler</code> functions, one can use enriched pathway names for visualization.
The <code>enrich</code> option can be specified for <code>'kegg'</code> or <code>'reactome'</code>, this time we specify <code>'reactome'</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                      enrich<span class="op">=</span><span class="st">"reactome"</span>, tfidf<span class="op">=</span><span class="cn">TRUE</span>, numWords<span class="op">=</span><span class="fl">50</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Performing enrichment analysis</span></span>
<span><span class="va">gwc3</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/ngrampath-1.png" width="768"></div>
<p>One of the main questions is which words can be clustered together among the words contained in the queried gene cluster. Word clustering (<code>pvclust</code>) and identified significant clusters based on the occurrence in the text can be visualized by specifying <code>tag=TRUE</code>. For the significance threshold, <code>pvclAlpha</code> can be specified. The default parameters perform <code>pvclust</code> on the subset of dataset for words with high frequency specified by <code>numWords</code>. If one want to perform on whole matrix (which is a natural way), <code>tagWhole=TRUE</code> can be specified, although it is computationally intensive. One can pass clusters to perform parallel computing owning to <code>pvclust</code> function, by specifying <code>cl</code> as below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">4</span>, <span class="st">"PuOr"</span><span class="op">)</span> </span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="va">pal</span><span class="op">)</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Cluster on subset of words</span></span>
<span><span class="va">gwcl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                     pal <span class="op">=</span> <span class="va">pal</span>,</span>
<span>                     argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="va">gwcl</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/tagging-1.png" width="768"></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwcl</span><span class="op">@</span><span class="va">pvpick</span></span>
<span><span class="co">#&gt; $clusters</span></span>
<span><span class="co">#&gt; $clusters[[1]]</span></span>
<span><span class="co">#&gt; [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[2]]</span></span>
<span><span class="co">#&gt;  [1] "activity"                  "atpdependent"             </span></span>
<span><span class="co">#&gt;  [3] "basal"                     "cerebrooculofacioskeletal"</span></span>
<span><span class="co">#&gt;  [5] "characterized"             "complementation"          </span></span>
<span><span class="co">#&gt;  [7] "damage"                    "defects"                  </span></span>
<span><span class="co">#&gt;  [9] "exon"                      "factor"                   </span></span>
<span><span class="co">#&gt; [11] "forms"                     "functions"                </span></span>
<span><span class="co">#&gt; [13] "helicase"                  "heterodimeric"            </span></span>
<span><span class="co">#&gt; [15] "incision"                  "including"                </span></span>
<span><span class="co">#&gt; [17] "interacts"                 "light"                    </span></span>
<span><span class="co">#&gt; [19] "nucleotide"                "product"                  </span></span>
<span><span class="co">#&gt; [21] "transcriptioncoupled"     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[3]]</span></span>
<span><span class="co">#&gt; [1] "dna"    "repair"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[4]]</span></span>
<span><span class="co">#&gt; [1] "cockayne"      "excision"      "syndrome"     </span></span>
<span><span class="co">#&gt; [4] "transcription"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $edges</span></span>
<span><span class="co">#&gt; [1]  6 22 23 25</span></span>
<span></span>
<span><span class="co">## Cluster on whole matrix</span></span>
<span></span>
<span><span class="va">gwclWhole</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, tagWhole<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                     pal <span class="op">=</span> <span class="va">pal</span>,</span>
<span>                     cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>                     argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="va">gwclWhole</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/tagging-2.png" width="768"></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwclWhole</span><span class="op">@</span><span class="va">pvpick</span></span>
<span><span class="co">#&gt; $clusters</span></span>
<span><span class="co">#&gt; $clusters[[1]]</span></span>
<span><span class="co">#&gt;  [1] "abnormally"        "active"           </span></span>
<span><span class="co">#&gt;  [3] "csb"               "defective"        </span></span>
<span><span class="co">#&gt;  [5] "disease"           "hereditary"       </span></span>
<span><span class="co">#&gt;  [7] "identified"        "iih"              </span></span>
<span><span class="co">#&gt;  [9] "p44"               "radiation"        </span></span>
<span><span class="co">#&gt; [11] "repeat"            "sensitive"        </span></span>
<span><span class="co">#&gt; [13] "transcriptionally" "ultraviolet"      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[2]]</span></span>
<span><span class="co">#&gt;  [1] "activates"       "adjacent"        "atpase"         </span></span>
<span><span class="co">#&gt;  [4] "atpstimulated"   "dnabinding"      "downstream"     </span></span>
<span><span class="co">#&gt;  [7] "encode"          "formation"       "frame"          </span></span>
<span><span class="co">#&gt; [10] "fusion"          "individual"      "occurs"         </span></span>
<span><span class="co">#&gt; [13] "polyadenylation" "promote"         "reading"        </span></span>
<span><span class="co">#&gt; [16] "sequence"        "shares"          "sites"          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[3]]</span></span>
<span><span class="co">#&gt; [1] "class" "tfiih"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[4]]</span></span>
<span><span class="co">#&gt; [1] "complementation" "defects"        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[5]]</span></span>
<span><span class="co">#&gt; [1] "eme1"      "ercc1"     "structure" "xp6"      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[6]]</span></span>
<span><span class="co">#&gt; [1] "orf"               "piggybackderived3"</span></span>
<span><span class="co">#&gt; [3] "splice"           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[7]]</span></span>
<span><span class="co">#&gt; [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[8]]</span></span>
<span><span class="co">#&gt; [1] "interacts" "type"     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[9]]</span></span>
<span><span class="co">#&gt; [1] "incision"  "including" "light"    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[10]]</span></span>
<span><span class="co">#&gt; [1] "transcriptioncoupled" "upstream"            </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[11]]</span></span>
<span><span class="co">#&gt;  [1] "basic"            "bivm"            </span></span>
<span><span class="co">#&gt;  [3] "cachexia"         "cancer"          </span></span>
<span><span class="co">#&gt;  [5] "cellular"         "characterized"   </span></span>
<span><span class="co">#&gt;  [7] "cognitive"        "develop"         </span></span>
<span><span class="co">#&gt;  [9] "development"      "disability"      </span></span>
<span><span class="co">#&gt; [11] "disorder"         "exists"          </span></span>
<span><span class="co">#&gt; [13] "exposure"         "function"        </span></span>
<span><span class="co">#&gt; [15] "growth"           "hypersensitivity"</span></span>
<span><span class="co">#&gt; [17] "immunoglobulin"   "increased"       </span></span>
<span><span class="co">#&gt; [19] "motif"            "neighboring"     </span></span>
<span><span class="co">#&gt; [21] "patients"         "processes"       </span></span>
<span><span class="co">#&gt; [23] "read"             "referred"        </span></span>
<span><span class="co">#&gt; [25] "severe"           "singlestrand"    </span></span>
<span><span class="co">#&gt; [27] "skin"             "specific"        </span></span>
<span><span class="co">#&gt; [29] "susceptibility"   "uvinduced"       </span></span>
<span><span class="co">#&gt; [31] "variable"         "vii"             </span></span>
<span><span class="co">#&gt; [33] "xp7"             </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[12]]</span></span>
<span><span class="co">#&gt;  [1] "atpdependent"        "basal"              </span></span>
<span><span class="co">#&gt;  [3] "belongs"             "btf2tfiih"          </span></span>
<span><span class="co">#&gt;  [5] "cancerprone"         "disorders"          </span></span>
<span><span class="co">#&gt;  [7] "factor"              "helicase"           </span></span>
<span><span class="co">#&gt;  [9] "helicases"           "integral"           </span></span>
<span><span class="co">#&gt; [11] "mechanism"           "rad3xpd"            </span></span>
<span><span class="co">#&gt; [13] "subfamily"           "trichothiodystrophy"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[13]]</span></span>
<span><span class="co">#&gt;  [1] "alter"                     "carcinogenesis"           </span></span>
<span><span class="co">#&gt;  [3] "catalyzes"                 "cd3e"                     </span></span>
<span><span class="co">#&gt;  [5] "cerebrooculofacioskeletal" "cisplatin"                </span></span>
<span><span class="co">#&gt;  [7] "compounds"                 "crosslinks"               </span></span>
<span><span class="co">#&gt;  [9] "electrophilic"             "epsilon"                  </span></span>
<span><span class="co">#&gt; [11] "ercc4"                     "excising"                 </span></span>
<span><span class="co">#&gt; [13] "exon"                      "formed"                   </span></span>
<span><span class="co">#&gt; [15] "forms"                     "heterodimer"              </span></span>
<span><span class="co">#&gt; [17] "heterodimeric"             "induced"                  </span></span>
<span><span class="co">#&gt; [19] "interstrand"               "lesion"                   </span></span>
<span><span class="co">#&gt; [21] "lesions"                   "molecule"                 </span></span>
<span><span class="co">#&gt; [23] "overlaps"                  "pathway"                  </span></span>
<span><span class="co">#&gt; [25] "play"                      "polymorphisms"            </span></span>
<span><span class="co">#&gt; [27] "process"                   "product"                  </span></span>
<span><span class="co">#&gt; [29] "recombinational"           "required"                 </span></span>
<span><span class="co">#&gt; [31] "result"                    "strand"                   </span></span>
<span><span class="co">#&gt; [33] "xpf"                      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[14]]</span></span>
<span><span class="co">#&gt; [1] "dna"    "repair"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $edges</span></span>
<span><span class="co">#&gt;  [1]  13  30  97  98 101 104 106 107 110 117 118 119 124 130</span></span></code></pre></div>
<p>In this example showing ERCC genes, the term <code>DNA repair</code> is clustered as expected.</p>
</div>
<div id="producing-correlation-networks" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Producing correlation networks<a class="anchor" aria-label="anchor" href="#producing-correlation-networks"><i class="fas fa-link"></i></a>
</h2>
<p>The next main function is producing correlation networks between words based on the co-occurrence of the words in the text.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/basic3-1.png" width="768"></div>
<p>The edge label corresponding to correlation values can be shown by <code>edgeLabel=TRUE</code>. The number of words to be shown on plot can be specified by <code>numWords</code>. The threshold of correlation can be specified by <code>corThresh</code>. The text color can be changed by <code>colorText=TRUE</code>. The type of edge can be specified by <code>edgeLink</code>, which is by default <code>TRUE</code> (link will be used).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     edgeLabel<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/ngramnet-1.png" width="768"></div>
<p>The clustering results based on <code>pvclust</code> can be shown also on the correlation network, by the node color (<code>tag=TRUE</code>). We can plot <code>igraph</code> object as well, using basic <code>plot</code> function implemented.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/tagnet-1.png" width="768"></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span><span class="op">@</span><span class="va">pvpick</span></span>
<span><span class="co">#&gt; $clusters</span></span>
<span><span class="co">#&gt; $clusters[[1]]</span></span>
<span><span class="co">#&gt; [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[2]]</span></span>
<span><span class="co">#&gt;  [1] "activity"             "atpdependent"        </span></span>
<span><span class="co">#&gt;  [3] "complementation"      "defects"             </span></span>
<span><span class="co">#&gt;  [5] "factor"               "functions"           </span></span>
<span><span class="co">#&gt;  [7] "incision"             "interacts"           </span></span>
<span><span class="co">#&gt;  [9] "nucleotide"           "product"             </span></span>
<span><span class="co">#&gt; [11] "transcriptioncoupled"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[3]]</span></span>
<span><span class="co">#&gt; [1] "dna"    "repair"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $edges</span></span>
<span><span class="co">#&gt; [1]  2 12 13</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/tagnet-2.png" width="768"></div>
<p>The genes associated with the corresponding words can be shown by specifying <code>genePlot=TRUE</code>, useful for assessing which words is associated with interesting genes. The edges connecting words to corresponding genes are shown.
Additionally, one can specify <code>genePlotNum</code> for limiting the genes shown, which can be useful for identifying important genes associated with high-frequent words.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     genePlot<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/gplot-1.png" width="768"></div>
<p>The associated enriched pathways (if present) can be shown by specifying <code>genePathPlot</code>, using <code>ggforce</code>.
In this option, the function first performs over-representation analysis on the whole gene set, and plot enriched terms for included genes in the plot. Enrichment analysis here is performed by the library <code>clusterProfiler</code> or <code>ReactomePA</code>, and one can control which pathways to plot by <code>genePathPlotSig</code> value.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://joelgombin.github.io/concaveman/">concaveman</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggforce.data-imaginist.com">ggforce</a></span><span class="op">)</span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     genePathPlot<span class="op">=</span><span class="st">"reactome"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                     tag<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                     genePathPlotSig<span class="op">=</span><span class="fl">0.05</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Found 47 enriched term</span></span>
<span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/gpplot-1.png" width="768"></div>
</div>
<div id="visualization-of-pubmed-information." class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Visualization of PubMed information.<a class="anchor" aria-label="anchor" href="#visualization-of-pubmed-information."><i class="fas fa-link"></i></a>
</h2>
<p>Using <code>rentrez</code>, one can perform the same analysis on PubMed text like the article title and abstract. The function queries for the input gene symbols (or microbes) and visualize. For typical use cases, the genes identified by showing <code>genePlot</code>, or hub genes identified in gene network analysis can be queried. The basic parameters for searching PubMed, like max number of articles retrieved and how to sort the articles can be specified by <code>retMax</code> and <code>sortOrder</code>. Be sure to obtain <a href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/">an api key</a> when querying heavily, and specify in <code>apiKey</code> argument.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcAbst.html">wcAbst</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, retMax<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Proceeding without API key</span></span>
<span><span class="va">ab</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc-1.png" width="768"></div>
<p>As fetching the same information is not desirable and time consuming, the same object can be passed to <code>redo</code> option and re-perform the analysis like tagging, or changing the visualization options.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">5</span>, <span class="st">"PuOr"</span><span class="op">)</span> </span>
<span><span class="va">abtag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>, pal<span class="op">=</span><span class="va">pal</span><span class="op">)</span></span>
<span><span class="co">#&gt; Resuming from the previous results</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="va">abtag2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">abtag</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                 plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, preset<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Resuming from the previous results</span></span>
<span><span class="co">#&gt; Using previous pvclust results</span></span>
<span><span class="va">abtag</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc_2-1.png" width="768"></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag2</span><span class="op">@</span><span class="va">net</span></span>
<span><span class="co">#&gt; Warning: Removed 3 rows containing missing values</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/wcabst_wc_2-2.png" width="768"></div>
</div>
<div id="comparing-two-or-more-networks" class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Comparing two or more networks<a class="anchor" aria-label="anchor" href="#comparing-two-or-more-networks"><i class="fas fa-link"></i></a>
</h2>
<p>One can compare two or more networks by providing list of objects produced by wc* functions, like <code>wcGeneSummary</code> and <code>wcAbst</code>. This can be useful for assessing the similarity and dissimilarity of the various text sources, like PubMed, RefSeq, and Reactome pathway names.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">8</span>,<span class="fl">9</span>,<span class="fl">10</span>,<span class="fl">11</span>,<span class="fl">12</span>,<span class="fl">13</span>,<span class="fl">14</span>,<span class="fl">16</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">cxcls</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"CXCL"</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">cxcls</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 13</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 13</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcAbst.html">wcAbst</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>               corThresh<span class="op">=</span><span class="fl">0.2</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Resuming from the previous results</span></span>
<span></span>
<span><span class="co">## Not having meaningful overlaps</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_stringMetric,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label)): font family not found in Windows</span></span>
<span><span class="co">#&gt; font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_stringMetric,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label)): font family not found in Windows</span></span>
<span><span class="co">#&gt; font database</span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/compare-1.png" width="1152"></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net3</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/compare-2.png" width="1152"></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span>, <span class="va">net3</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span>, <span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/compare-3.png" width="1152"></div>
<p>If <code>tag</code> information is available in both gene clusters, combined tags can be visualized by <code>tag=TRUE</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keggPathways</span> <span class="op">&lt;-</span> <span class="va">org.Hs.egPATH2EG</span></span>
<span><span class="va">mappedKeys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html">mappedkeys</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">)</span></span>
<span><span class="va">keggList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">[</span><span class="va">mappedKeys</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04110`</span>,</span>
<span>                      keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>                      plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>                      numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                      tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                      tfidf<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 124</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04210`</span>,</span>
<span>                      keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>                      plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                      corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>                      numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                      tfidf<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                      tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 87</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>, tag<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span>
<span></span>
<span><span class="co">#&gt; Warning in grid.Call(C_textBounds,</span></span>
<span><span class="co">#&gt; as.graphicsAnnot(x$label), x$x, x$y, : font family not found</span></span>
<span><span class="co">#&gt; in Windows font database</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/tagcomp-1.png" width="1440"></div>
</div>
<div id="text-over-represenatation-analysis-experimental" class="section level2" number="2.5">
<h2>
<span class="header-section-number">2.5</span> Text over represenatation analysis (experimental)<a class="anchor" aria-label="anchor" href="#text-over-represenatation-analysis-experimental"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`00785`</span> <span class="co"># Lipoic acid metabolism</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/ora-1.png" width="480"></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;            lipoic              step              acid </span></span>
<span><span class="co">#&gt;      1.600963e-14      8.016934e-08      5.300425e-07 </span></span>
<span><span class="co">#&gt; lipoateactivating  lipoatedependent            lipoyl </span></span>
<span><span class="co">#&gt;      1.484931e-04      1.484931e-04      1.484931e-04</span></span>
<span></span>
<span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`05150`</span> <span class="co"># Staphylococcus aureus infection</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/ora-2.png" width="480"></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;          drb        chain   complement        class </span></span>
<span><span class="co">#&gt; 0.000000e+00 3.669590e-82 3.266180e-74 5.496543e-46 </span></span>
<span><span class="co">#&gt;         exon         beta </span></span>
<span><span class="co">#&gt; 6.851722e-39 1.318324e-34</span></span></code></pre></div>
<p>This thresholding can be used in RefSeq visualization.
Filter words using ORA threshold and frequency threshold by setting <code>ora=TRUE</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcGeneSummary.html">wcGeneSummary</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     ora<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 65 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Performing ORA</span></span>
<span><span class="co">#&gt; Filtered 148 words (ORA)</span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_wcGeneSummary_files/figure-html/oranet-1.png" width="672"></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19044)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Japanese_Japan.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Japanese_Japan.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                   </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils    </span></span>
<span><span class="co">#&gt; [6] datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] dplyr_1.0.99.9000     dendextend_1.16.0    </span></span>
<span><span class="co">#&gt;  [3] clusterProfiler_4.6.0 ReactomePA_1.42.0    </span></span>
<span><span class="co">#&gt;  [5] RColorBrewer_1.1-3    ggraph_2.1.0         </span></span>
<span><span class="co">#&gt;  [7] ggplot2_3.4.0         org.Hs.eg.db_3.16.0  </span></span>
<span><span class="co">#&gt;  [9] AnnotationDbi_1.60.0  IRanges_2.32.0       </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.36.1      Biobase_2.58.0       </span></span>
<span><span class="co">#&gt; [13] BiocGenerics_0.44.0   wcGeneSummary_0.99.0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] shadowtext_0.1.2       fastmatch_1.1-3       </span></span>
<span><span class="co">#&gt;   [3] plyr_1.8.8             igraph_1.3.5          </span></span>
<span><span class="co">#&gt;   [5] lazyeval_0.2.2         splines_4.2.2         </span></span>
<span><span class="co">#&gt;   [7] BiocParallel_1.32.5    GenomeInfoDb_1.34.6   </span></span>
<span><span class="co">#&gt;   [9] digest_0.6.31          yulab.utils_0.0.6     </span></span>
<span><span class="co">#&gt;  [11] htmltools_0.5.4        bugsigdbr_1.4.1       </span></span>
<span><span class="co">#&gt;  [13] GOSemSim_2.24.0        viridis_0.6.2         </span></span>
<span><span class="co">#&gt;  [15] GO.db_3.16.0           fansi_1.0.3           </span></span>
<span><span class="co">#&gt;  [17] GeneSummary_0.99.4     magrittr_2.0.3        </span></span>
<span><span class="co">#&gt;  [19] memoise_2.0.1          tm_0.7-10             </span></span>
<span><span class="co">#&gt;  [21] Biostrings_2.66.0      graphlayouts_0.8.4    </span></span>
<span><span class="co">#&gt;  [23] pvclust_2.2-0          wordcloud_2.6         </span></span>
<span><span class="co">#&gt;  [25] enrichplot_1.18.3      colorspace_2.0-3      </span></span>
<span><span class="co">#&gt;  [27] rappdirs_0.3.3         blob_1.2.3            </span></span>
<span><span class="co">#&gt;  [29] ggrepel_0.9.2          xfun_0.36             </span></span>
<span><span class="co">#&gt;  [31] crayon_1.5.2           RCurl_1.98-1.9        </span></span>
<span><span class="co">#&gt;  [33] jsonlite_1.8.4         scatterpie_0.1.8      </span></span>
<span><span class="co">#&gt;  [35] graph_1.76.0           ape_5.6-2             </span></span>
<span><span class="co">#&gt;  [37] glue_1.6.2             polyclip_1.10-4       </span></span>
<span><span class="co">#&gt;  [39] stopwords_2.3          gtable_0.3.1          </span></span>
<span><span class="co">#&gt;  [41] zlibbioc_1.44.0        XVector_0.38.0        </span></span>
<span><span class="co">#&gt;  [43] GetoptLong_1.0.5       graphite_1.44.0       </span></span>
<span><span class="co">#&gt;  [45] rentrez_1.2.3          scales_1.2.1          </span></span>
<span><span class="co">#&gt;  [47] DOSE_3.24.2            DBI_1.1.3             </span></span>
<span><span class="co">#&gt;  [49] Rcpp_1.0.9             viridisLite_0.4.1     </span></span>
<span><span class="co">#&gt;  [51] xtable_1.8-4           tidytree_0.4.2        </span></span>
<span><span class="co">#&gt;  [53] gridGraphics_0.5-1     reactome.db_1.82.0    </span></span>
<span><span class="co">#&gt;  [55] bit_4.0.5              htmlwidgets_1.6.1     </span></span>
<span><span class="co">#&gt;  [57] httr_1.4.4             fgsea_1.24.0          </span></span>
<span><span class="co">#&gt;  [59] ellipsis_0.3.2         pkgconfig_2.0.3       </span></span>
<span><span class="co">#&gt;  [61] XML_3.99-0.13          farver_2.1.1          </span></span>
<span><span class="co">#&gt;  [63] sass_0.4.4             utf8_1.2.2            </span></span>
<span><span class="co">#&gt;  [65] labeling_0.4.2         ggplotify_0.1.0       </span></span>
<span><span class="co">#&gt;  [67] tidyselect_1.2.0       rlang_1.0.6           </span></span>
<span><span class="co">#&gt;  [69] reshape2_1.4.4         later_1.3.0           </span></span>
<span><span class="co">#&gt;  [71] munsell_0.5.0          tools_4.2.2           </span></span>
<span><span class="co">#&gt;  [73] cachem_1.0.6           downloader_0.4        </span></span>
<span><span class="co">#&gt;  [75] cli_3.6.0              generics_0.1.3        </span></span>
<span><span class="co">#&gt;  [77] RSQLite_2.2.20         gson_0.0.9            </span></span>
<span><span class="co">#&gt;  [79] evaluate_0.19          stringr_1.5.0         </span></span>
<span><span class="co">#&gt;  [81] fastmap_1.1.0          ggdendro_0.1.23       </span></span>
<span><span class="co">#&gt;  [83] yaml_2.3.6             ggtree_3.6.2          </span></span>
<span><span class="co">#&gt;  [85] knitr_1.41             bit64_4.0.5           </span></span>
<span><span class="co">#&gt;  [87] fs_1.5.2               tidygraph_1.2.2       </span></span>
<span><span class="co">#&gt;  [89] purrr_1.0.1            KEGGREST_1.38.0       </span></span>
<span><span class="co">#&gt;  [91] nlme_3.1-161           mime_0.12             </span></span>
<span><span class="co">#&gt;  [93] slam_0.1-50            aplot_0.1.9           </span></span>
<span><span class="co">#&gt;  [95] xml2_1.3.3             compiler_4.2.2        </span></span>
<span><span class="co">#&gt;  [97] rstudioapi_0.14        curl_5.0.0            </span></span>
<span><span class="co">#&gt;  [99] png_0.1-8              treeio_1.22.0         </span></span>
<span><span class="co">#&gt; [101] tibble_3.1.8           tweenr_2.0.2          </span></span>
<span><span class="co">#&gt; [103] bslib_0.4.2            stringi_1.7.12        </span></span>
<span><span class="co">#&gt; [105] cyjShiny_1.0.34        highr_0.10            </span></span>
<span><span class="co">#&gt; [107] lattice_0.20-45        Matrix_1.5-3          </span></span>
<span><span class="co">#&gt; [109] vctrs_0.5.1            pillar_1.8.1          </span></span>
<span><span class="co">#&gt; [111] lifecycle_1.0.3        jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt; [113] GlobalOptions_0.1.2    data.table_1.14.6     </span></span>
<span><span class="co">#&gt; [115] cowplot_1.1.1          bitops_1.0-7          </span></span>
<span><span class="co">#&gt; [117] httpuv_1.6.8           patchwork_1.1.2       </span></span>
<span><span class="co">#&gt; [119] qvalue_2.30.0          R6_2.5.1              </span></span>
<span><span class="co">#&gt; [121] bookdown_0.31          promises_1.2.0.1      </span></span>
<span><span class="co">#&gt; [123] gridExtra_2.3          codetools_0.2-18      </span></span>
<span><span class="co">#&gt; [125] MASS_7.3-58.1          rjson_0.2.21          </span></span>
<span><span class="co">#&gt; [127] withr_2.5.0            GenomeInfoDbData_1.2.9</span></span>
<span><span class="co">#&gt; [129] parallel_4.2.2         ISOcodes_2022.09.29   </span></span>
<span><span class="co">#&gt; [131] grid_4.2.2             ggfun_0.0.9           </span></span>
<span><span class="co">#&gt; [133] tidyr_1.2.1            HDO.db_0.99.1         </span></span>
<span><span class="co">#&gt; [135] rmarkdown_2.19         downlit_0.4.2         </span></span>
<span><span class="co">#&gt; [137] ggforce_0.4.1          NLP_0.2-1             </span></span>
<span><span class="co">#&gt; [139] shiny_1.7.4            base64enc_0.1-3</span></span></code></pre></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html"><span class="header-section-number">1</span> About</a></div>
<div class="next"><a href="the-basic-usage-for-microbiome-related-data.html"><span class="header-section-number">3</span> The basic usage for microbiome-related data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#basic-usage"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="nav-link" href="#producing-word-clouds"><span class="header-section-number">2.1</span> Producing word clouds</a></li>
<li><a class="nav-link" href="#producing-correlation-networks"><span class="header-section-number">2.2</span> Producing correlation networks</a></li>
<li><a class="nav-link" href="#visualization-of-pubmed-information."><span class="header-section-number">2.3</span> Visualization of PubMed information.</a></li>
<li><a class="nav-link" href="#comparing-two-or-more-networks"><span class="header-section-number">2.4</span> Comparing two or more networks</a></li>
<li><a class="nav-link" href="#text-over-represenatation-analysis-experimental"><span class="header-section-number">2.5</span> Text over represenatation analysis (experimental)</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>wcGeneSummary (OSplot)</strong>" was written by Noriaki Sato. It was last built on 2023-01-26.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
