<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 8 Custom usage - pathview text | wcGeneSummary (OSplot)</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.31 with bs4_book()">
<meta property="og:title" content="Chapter 8 Custom usage - pathview text | wcGeneSummary (OSplot)">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/wcGeneSummary/custom-usage---pathview-text.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 8 Custom usage - pathview text | wcGeneSummary (OSplot)">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="KEGG mapping plots produced by pathview can be populated with text information (Luo and Brouwer. 2013). library(wcGeneSummary) library(pathview) Suppose an example for KEGG Orthology obtained from...">
<meta property="og:description" content="KEGG mapping plots produced by pathview can be populated with text information (Luo and Brouwer. 2013). library(wcGeneSummary) library(pathview) Suppose an example for KEGG Orthology obtained from...">
<meta name="twitter:description" content="KEGG mapping plots produced by pathview can be populated with text information (Luo and Brouwer. 2013). library(wcGeneSummary) library(pathview) Suppose an example for KEGG Orthology obtained from...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">wcGeneSummary (OSplot)</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="" href="the-basic-usage-for-microbiome-related-data.html"><span class="header-section-number">3</span> The basic usage for microbiome-related data</a></li>
<li><a class="" href="custom-usage---interactive-inspection-of-bayesian-network-annotated-by-words.html"><span class="header-section-number">4</span> Custom usage - Interactive inspection of Bayesian network annotated by words</a></li>
<li><a class="" href="custom-usage---annotating-module-eigengenes-dendrogram.html"><span class="header-section-number">5</span> Custom usage - Annotating module eigengenes dendrogram</a></li>
<li><a class="" href="custom-usage---assess-the-occurrence-of-the-speicific-words-across-gene-clusters.html"><span class="header-section-number">6</span> Custom usage - Assess the occurrence of the speicific words across gene clusters</a></li>
<li><a class="" href="custom-usage---recluster-the-cluster-using-word-information.html"><span class="header-section-number">7</span> Custom usage - Recluster the cluster using word information</a></li>
<li><a class="active" href="custom-usage---pathview-text.html"><span class="header-section-number">8</span> Custom usage - pathview text</a></li>
<li><a class="" href="application-examples.html"><span class="header-section-number">9</span> Application examples</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/noriakis/wcGeneSummary">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="custom-usage---pathview-text" class="section level1" number="8">
<h1>
<span class="header-section-number">8</span> Custom usage - pathview text<a class="anchor" aria-label="anchor" href="#custom-usage---pathview-text"><i class="fas fa-link"></i></a>
</h1>
<p>KEGG mapping plots produced by pathview can be populated with text information (<a href="https://academic.oup.com/bioinformatics/article/29/14/1830/232698%5D">Luo and Brouwer. 2013</a>).</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/datapplab/pathview">pathview</a></span><span class="op">)</span></span></code></pre></div>
<p>Suppose an example for KEGG Orthology obtained from some microbiome-related experiments.
By using <a href="https://github.com/YuLab-SMU/MicrobiomeProfiler">MicrobiomeProfiler</a>, enriched pathway using over-representation analysis can be obtained as follows.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"K00640"</span>,<span class="st">"K23304"</span>,<span class="st">"K01760"</span>,<span class="st">"K01697"</span>,<span class="st">"K00549"</span>,<span class="st">"K00899"</span><span class="op">)</span></span>
<span><span class="va">enriched</span> <span class="op">&lt;-</span> <span class="fu">MicrobiomeProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MicrobiomeProfiler/man/enrichKO.html">enrichKO</a></span><span class="op">(</span><span class="va">kos</span><span class="op">)</span></span>
<span><span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">enriched</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure">
<img src="04-pathview_text_files/figure-html/micro-1.png" width="672">
Now plot these KOs in the pathview.</div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## As the output is PNG, this script is omitted.</span></span>
<span><span class="va">pvInput</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">kos</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pvInput</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">kos</span></span>
<span><span class="co"># pathview(kos, pathway.id = "00270", species = "ko")</span></span></code></pre></div>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/ko00270.pathview.png?raw=true" width="500" alt=""><p class="caption">raw pathview plot</p>
</div>
<p>To use the <code><a href="https://rdrr.io/pkg/wcGeneSummary/man/pathviewText.html">pathviewText()</a></code> one should specify which database to retrieve the text, <code>"refseq"</code> or <code>"abstract"</code>.
The default is “abstract”, which fetches information from RefSeq database. In this case, KO number is not included in RefSeq, thus the abstract is chosen. However, querying in KO number itself is not useful. We conver the KO number to EC number using KEGG REST API, and fetches the description using <code><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcEC.html">wcEC()</a></code> function, and query the enzyme name in PubMed. In this way, one must provide <code>searchTerm</code>, and <code>termMap</code>, which map the KO and EC. Also, <code>node.types="ortholog"</code> must be passed to <code>pathview</code>.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Again, the codes are commented out.</span></span>
<span></span>
<span><span class="co"># ecko &lt;- data.table::fread("https://rest.kegg.jp/link/ko/ec",</span></span>
<span><span class="co">#                             header=FALSE)</span></span>
<span><span class="co"># ecko$EC &lt;- sapply(strsplit(ecko$V1, ":"),"[",2)</span></span>
<span><span class="co"># ecko$KO &lt;- sapply(strsplit(ecko$V2, ":"), "[", 2)</span></span>
<span><span class="co"># filt &lt;- ecko[ ecko$KO %in% kos, ]</span></span>
<span><span class="co"># ecnum &lt;- filt$EC</span></span>
<span><span class="co"># termMap &lt;- filt[,c("KO","EC")] |&gt; `colnames&lt;-`(c("query","number"))</span></span>
<span><span class="co"># ecQuery &lt;- wcEC("enzyme.dat", ecnum = ecnum, onlyTerm =TRUE)</span></span>
<span><span class="co"># ecMap &lt;- wcEC("enzyme.dat", ecnum = ecnum, onlyDf =TRUE)</span></span>
<span><span class="co"># termMap &lt;- merge(termMap, ecMap[,c("number","desc")], by="number") |&gt;</span></span>
<span><span class="co">#   `colnames&lt;-`(c("number","query","description"))</span></span>
<span></span>
<span><span class="co">## Main function</span></span>
<span></span>
<span><span class="co"># pathviewText(kos, keyType = "KO",</span></span>
<span><span class="co">#      target="abstract",</span></span>
<span><span class="co">#      pid = "00270",</span></span>
<span><span class="co">#      org = "ko", </span></span>
<span><span class="co">#      searchTerms = ecQuery,</span></span>
<span><span class="co">#      termMap = termMap,</span></span>
<span><span class="co">#      node.types="ortholog")</span></span>
<span></span>
<span><span class="co">## We can transpose the barplot by trans=TRUE</span></span>
<span></span>
<span><span class="co"># pathviewText(kos, keyType = "KO",</span></span>
<span><span class="co">#      target="abstract",</span></span>
<span><span class="co">#      pid = "00270",</span></span>
<span><span class="co">#      org = "ko", </span></span>
<span><span class="co">#      trans = TRUE,</span></span>
<span><span class="co">#      searchTerms = ecQuery,</span></span>
<span><span class="co">#      termMap = termMap,</span></span>
<span><span class="co">#      node.types="ortholog")</span></span></code></pre></div>
<p>The below picture shows transposed results.</p>
<div class="figure">
<img src="https://github.com/noriakis/software/blob/main/images/pathviewText.png?raw=true" width="500" alt=""><p class="caption">annotated pathview plot - KO</p>
</div>
<p>For gene queries, we can obtain the same plot using relatively simple query.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TP53"</span>,<span class="st">"CDC45"</span>,<span class="st">"CDC6"</span><span class="op">)</span></span></code></pre></div>

</div>
  <div class="chapter-nav">
<div class="prev"><a href="custom-usage---recluster-the-cluster-using-word-information.html"><span class="header-section-number">7</span> Custom usage - Recluster the cluster using word information</a></div>
<div class="next"><a href="application-examples.html"><span class="header-section-number">9</span> Application examples</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#custom-usage---pathview-text"><span class="header-section-number">8</span> Custom usage - pathview text</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/noriakis/wcGeneSummary/blob/master/04-pathview_text.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/noriakis/wcGeneSummary/edit/master/04-pathview_text.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>wcGeneSummary (OSplot)</strong>" was written by Noriaki Sato. It was last built on 2023-01-26.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
