<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 The basic usage for microbiome-related data | wcGeneSummary (OSplot)</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.31 with bs4_book()">
<meta property="og:title" content="Chapter 3 The basic usage for microbiome-related data | wcGeneSummary (OSplot)">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/wcGeneSummary/the-basic-usage-for-microbiome-related-data.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 The basic usage for microbiome-related data | wcGeneSummary (OSplot)">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content='The general usage for mining the information related to microbiome-related signatures. library(wcGeneSummary) library(ggplot2) library(ggraph) library(RColorBrewer) load(system.file("extdata",...'>
<meta property="og:description" content='The general usage for mining the information related to microbiome-related signatures. library(wcGeneSummary) library(ggplot2) library(ggraph) library(RColorBrewer) load(system.file("extdata",...'>
<meta name="twitter:description" content='The general usage for mining the information related to microbiome-related signatures. library(wcGeneSummary) library(ggplot2) library(ggraph) library(RColorBrewer) load(system.file("extdata",...'>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">wcGeneSummary (OSplot)</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="active" href="the-basic-usage-for-microbiome-related-data.html"><span class="header-section-number">3</span> The basic usage for microbiome-related data</a></li>
<li><a class="" href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">4</span> Interactive inspection of gene cluster network annotated by words</a></li>
<li><a class="" href="annotating-gene-cluster-dendrogram.html"><span class="header-section-number">5</span> Annotating gene cluster dendrogram</a></li>
<li><a class="" href="assess-the-occurrence-of-the-speicific-words-across-gene-clusters.html"><span class="header-section-number">6</span> Assess the occurrence of the speicific words across gene clusters</a></li>
<li><a class="" href="recluster-the-cluster-using-word-information.html"><span class="header-section-number">7</span> Recluster the cluster using word information</a></li>
<li><a class="" href="including-the-other-databases-an-example-in-metacyc.html"><span class="header-section-number">8</span> Including the other databases (an example in MetaCyc)</a></li>
<li><a class="" href="pathview-text.html"><span class="header-section-number">9</span> Pathview text</a></li>
<li><a class="" href="application-examples.html"><span class="header-section-number">10</span> Application examples</a></li>
<li><a class="" href="parsing.html"><span class="header-section-number">11</span> Parsing</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="the-basic-usage-for-microbiome-related-data" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> The basic usage for microbiome-related data<a class="anchor" aria-label="anchor" href="#the-basic-usage-for-microbiome-related-data"><i class="fas fa-link"></i></a>
</h1>
<p>The general usage for mining the information related to microbiome-related signatures.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">wcGeneSummary</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"wcGeneSummary"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We use BugSigDB, and its R port bugsigdbr to obtain the curated dataset of the relationship with bacterial taxonomy and human diseases (<a href="https://bugsigdb.org/Main_Page">Geistlinger et al. 2022</a>). Users can query microbiome names, which will be searched for MetaPhlAn taxonomic annotation. If <code>target="title"</code>, the title of the corresponding articles will be summarized.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span><span class="op">)</span>,tag<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>    pal<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">10</span>, <span class="st">"Set2"</span><span class="op">)</span>,numWords<span class="op">=</span><span class="fl">80</span>,argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>min.freq<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 2</span></span>
<span><span class="co">#&gt;   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">#&gt;   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">#&gt; Including 28 entries</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="va">basic</span><span class="op">@</span><span class="va">freqDf</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    word freq</span></span>
<span><span class="co">#&gt; gut                 Gut    8</span></span>
<span><span class="co">#&gt; oral               Oral    5</span></span>
<span><span class="co">#&gt; patients       patients    5</span></span>
<span><span class="co">#&gt; study             study    3</span></span>
<span><span class="co">#&gt; arthritis     arthritis    2</span></span>
<span><span class="co">#&gt; association association    2</span></span>
<span><span class="co">#&gt; bacterial     bacterial    2</span></span>
<span><span class="co">#&gt; covid19         COVID19    2</span></span>
<span><span class="co">#&gt; diabetes       diabetes    2</span></span>
<span><span class="co">#&gt; dysbiosis     Dysbiosis    2</span></span>
<span><span class="co">#&gt; infant           infant    2</span></span>
<span><span class="co">#&gt; infection     Infection    2</span></span>
<span><span class="co">#&gt; obese             obese    2</span></span>
<span><span class="co">#&gt; respiratory respiratory    2</span></span>
<span><span class="co">#&gt; sequencing   sequencing    2</span></span>
<span><span class="co">#&gt; surgery         surgery    2</span></span>
<span><span class="co">#&gt; tract             tract    2</span></span>
<span><span class="co">#&gt;  features      features    1</span></span>
<span><span class="co">#&gt; 16s                 16S    1</span></span>
<span><span class="co">#&gt; aerosol         Aerosol    1</span></span>
<span><span class="va">basic</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/bsdb_basic-1.png" width="768"></div>
<p>If <code>target="abstract"</code>, the corresponding abstract will be fetched and be summarized.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span><span class="op">)</span>,tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"abstract"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>    pal<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">10</span>, <span class="st">"Set2"</span><span class="op">)</span>,numWords<span class="op">=</span><span class="fl">80</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 2</span></span>
<span><span class="co">#&gt;   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">#&gt;   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">#&gt; Including 28 entries</span></span>
<span><span class="co">#&gt; Target is abstract</span></span>
<span><span class="co">#&gt;   Querying PubMed for 17 pmids</span></span>
<span><span class="co">#&gt;   Querying without API key</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="va">basic2</span><span class="op">@</span><span class="va">freqDf</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                word freq</span></span>
<span><span class="co">#&gt; patients   patients   37</span></span>
<span><span class="co">#&gt; gut             Gut   30</span></span>
<span><span class="co">#&gt; oral           Oral   27</span></span>
<span><span class="co">#&gt; bacterial Bacterial   25</span></span>
<span><span class="co">#&gt; species     species   24</span></span>
<span><span class="co">#&gt; microbial Microbial   23</span></span>
<span><span class="va">basic2</span><span class="op">@</span><span class="va">wc</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/bsdb_basic2-1.png" width="768"></div>
<p>For successful visualization, pre-caculated TF-IDF and frequency data frame is available and one can use them to filter the highly occurring words, or the other prefiltering option used in <code>wcGeneSummary</code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rmwords</span> <span class="op">&lt;-</span> <span class="va">allFreqBSDB</span></span>
<span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="va">rmwords</span><span class="op">[</span><span class="va">rmwords</span><span class="op">$</span><span class="va">freq</span><span class="op">&gt;</span><span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">rmwords</span><span class="op">$</span><span class="va">freq</span>, <span class="fl">0.95</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">filter</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt;             freq        word</span></span>
<span><span class="co">#&gt; microbiota   275  microbiota</span></span>
<span><span class="co">#&gt; gut          242         gut</span></span>
<span><span class="co">#&gt; microbiome   175  microbiome</span></span>
<span><span class="co">#&gt; patients     146    patients</span></span>
<span><span class="co">#&gt; study         69       study</span></span>
<span><span class="co">#&gt; cancer        64      cancer</span></span>
<span><span class="co">#&gt; composition   62 composition</span></span>
<span><span class="co">#&gt; oral          55        oral</span></span>
<span><span class="co">#&gt; human         49       human</span></span>
<span><span class="co">#&gt; intestinal    48  intestinal</span></span>
<span><span class="co">#&gt; children      45    children</span></span>
<span><span class="co">#&gt; microbial     45   microbial</span></span>
<span><span class="co">#&gt; disease       43     disease</span></span>
<span><span class="co">#&gt; fecal         41       fecal</span></span>
<span><span class="co">#&gt; alterations   36 alterations</span></span>
<span><span class="co">#&gt; analysis      34    analysis</span></span>
<span><span class="co">#&gt; association   34 association</span></span>
<span><span class="co">#&gt; dysbiosis     34   dysbiosis</span></span>
<span><span class="co">#&gt; infection     34   infection</span></span>
<span><span class="co">#&gt; risk          33        risk</span></span></code></pre></div>
<p>The network visualization is possible by enabling <code>plotType="network"</code>.
The same parameters that can be passed to <code>wcGeneSummary</code> can be used.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Neisseria"</span>,<span class="st">"Veillonella"</span><span class="op">)</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">60</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 2</span></span>
<span><span class="co">#&gt;   Found 76 entries for Neisseria</span></span>
<span><span class="co">#&gt;   Found 221 entries for Veillonella</span></span>
<span><span class="co">#&gt; Including 502 entries</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="va">net</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/bsdb_basic_network-1.png" width="768"></div>
<p>The words-to-species relationship can be plotted by <code>mbPlot=TRUE</code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span>,</span>
<span>    <span class="st">"Veillonella parvula"</span>,<span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>, mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 4</span></span>
<span><span class="co">#&gt;   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">#&gt;   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">#&gt;   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">#&gt;   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">#&gt; Including 90 entries</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="va">net2</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/bsdb_basic_network_mb-1.png" width="768"></div>
<p>As the BugSigDB contains the relationship between bacterial taxonomy and disease, disease name can also be plotted. When <code>disPlot=TRUE</code>, the <code>mbPlot</code>
will be set to <code>TRUE</code> by default.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span>,</span>
<span>    <span class="st">"Veillonella parvula"</span>,<span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>, mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">TRUE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>, disPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 4</span></span>
<span><span class="co">#&gt;   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">#&gt;   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">#&gt;   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">#&gt;   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">#&gt; Including 90 entries</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="va">net3</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/bsdb_basic_network_dis-1.png" width="960"></div>
<p>Other than curated databases, the PubMed query can also be performed with setting <code>curate=FALSE</code>. This way, the text information of the latest literature for the microbes and diseases can be plotted. The options for use in function obtaining PubMed information can be specified to <code>abstArg</code> in list format, like <code>sortOrder="pubdate"</code>.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Veillonella dispar"</span>,<span class="st">"Neisseria flava"</span>,</span>
<span>    <span class="st">"Veillonella parvula"</span>,<span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>, mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    curate<span class="op">=</span><span class="cn">FALSE</span>,target<span class="op">=</span><span class="st">"title"</span>,pre<span class="op">=</span><span class="cn">TRUE</span>,plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    additionalRemove<span class="op">=</span><span class="va">filter</span><span class="op">$</span><span class="va">word</span>, disPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">40</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    abstArg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>retMax<span class="op">=</span><span class="fl">80</span>, sortOrder<span class="op">=</span><span class="st">"pubdate"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 4</span></span>
<span><span class="co">#&gt;   Found 17 entries for Veillonella dispar</span></span>
<span><span class="co">#&gt;   Found 1 entries for Neisseria flava</span></span>
<span><span class="co">#&gt;   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">#&gt;   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">#&gt; Including 90 entries</span></span>
<span><span class="co">#&gt; Proceeding without API key</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="va">net4</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/no_curate-1.png" width="960"></div>
<p>For microbiome analysis, it is often the case that investigating coded enzymes is important. Using <code>wcEC</code> function and <code>getUPtax</code> function, the queried species or genus can be linked to possible interaction with enzymes using following databases. The downloaded file path should be specified to the function like below to link the queried taxonomy and enzymes. Specifically, enzymes listed in <code>enzyme.dat</code> are searched, and corresponding UniProt identifiers are obtained, followed by mapping using <code>speclist.txt</code>. This way, the links to microbe - textual information - enzyme can be plotted.</p>
<ul>
<li><a href="https://enzyme.expasy.org/">enzyme.dat - from Expasy</a></li>
<li><a href="https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/docs/speclist.txt">speclist.txt - UniProt Controlled vocabulary of species</a></li>
</ul>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Not run</span></span>
<span><span class="va">vp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Veillonella parvula"</span><span class="op">)</span>,</span>
<span>                 plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                curate<span class="op">=</span><span class="cn">TRUE</span>, target<span class="op">=</span><span class="st">"title"</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                mbPlot <span class="op">=</span> <span class="cn">TRUE</span>, ecPlot<span class="op">=</span><span class="cn">TRUE</span>, disPlot<span class="op">=</span><span class="cn">TRUE</span>, tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,colorText<span class="op">=</span><span class="cn">TRUE</span>, pre<span class="op">=</span><span class="cn">TRUE</span>, numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                nodePal<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">10</span>, <span class="st">"Set2"</span><span class="op">)</span>,</span>
<span>                ecFile<span class="op">=</span><span class="st">"../enzyme.dat"</span>,</span>
<span>                upTaxFile <span class="op">=</span> <span class="st">"../speclist.txt"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 1</span></span>
<span><span class="co">#&gt;   Found 20 entries for Veillonella parvula</span></span>
<span><span class="co">#&gt; Including 31 entries</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Processing EC file</span></span>
<span><span class="co">#&gt;   Linking taxonomy to EC</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="va">vp</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/enzyme-1.png" width="960"></div>
<p>Further, the relationship between metabolites and microbiome is of interest. Recent studies have revealed various associations in gut microbiome composition and human plasma metabolites, as well as in the other environments.</p>
<ul>
<li>Wishart DS, Oler E, Peters H, et al. MiMeDB: the Human Microbial Metabolome Database. Nucleic Acids Res. 2023;51(D1):D611-D620. <a href="doi:10.1093/nar/gkac868" class="uri">doi:10.1093/nar/gkac868</a>
</li>
<li>Muller E, Algavi YM, Borenstein E. The gut microbiome-metabolome dataset collection: a curated resource for integrative meta-analysis. npj Biofilms and Microbiomes. 2022;8(1):1-7. <a href="doi:10.1038/s41522-022-00345-5" class="uri">doi:10.1038/s41522-022-00345-5</a>
</li>
<li>Dekkers KF, Sayols-Baixeras S, Baldanzi G, et al. An online atlas of human plasma metabolite signatures of gut microbiome composition. Nat Commun. 2022;13(1):5370. <a href="doi:10.1038/s41467-022-33050-0" class="uri">doi:10.1038/s41467-022-33050-0</a>
</li>
</ul>
<p>We now use data obtained in <a href="https://www.nature.com/articles/s41467-022-33050-0">Dekkers et al.</a>.
First, read the downloaded file using <code>readxl</code>.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metab</span> <span class="op">&lt;-</span> <span class="fu">readxl</span><span class="fu">::</span><span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html">read_excel</a></span><span class="op">(</span></span>
<span>  <span class="st">"../41467_2022_33050_MOESM8_ESM.xlsx"</span>,skip <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<p>Pass this tibble, as well as the columns to represent <code>taxonomy</code>, <code>metabolites</code>, and <code>quantitative values to threshold</code> to <code>metCol</code>. In this case, <code>metCol &lt;- c("Metagenomic species", "Metabolite", "Spearman's ρ")</code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">metabEx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/wcBSDB.html">wcBSDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Akkermansia muciniphila"</span><span class="op">)</span>,</span>
<span>                edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                curate<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>                nodePal<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">10</span>, <span class="st">"Dark2"</span><span class="op">)</span>,</span>
<span>                pre<span class="op">=</span><span class="cn">TRUE</span>, tag<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                additionalRemove <span class="op">=</span> <span class="va">filter</span><span class="op">$</span><span class="va">word</span>,</span>
<span>                target<span class="op">=</span><span class="st">"abstract"</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                plotType<span class="op">=</span><span class="st">"network"</span>, numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>                mbPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                metCol<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Metagenomic species"</span>, <span class="st">"Metabolite"</span>, <span class="st">"Spearman's ρ"</span><span class="op">)</span>,</span>
<span>                metab <span class="op">=</span><span class="va">metab</span>, metThresh<span class="op">=</span><span class="fl">0.15</span>,</span>
<span>                preserve <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,</span>
<span>                abstArg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>retMax<span class="op">=</span><span class="fl">80</span>,</span>
<span>                               sortOrder<span class="op">=</span><span class="st">"relevance"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input microbes: 1</span></span>
<span><span class="co">#&gt;   Found 21 entries for Akkermansia muciniphila</span></span>
<span><span class="co">#&gt; Including 31 entries</span></span>
<span><span class="co">#&gt; Target is abstract</span></span>
<span><span class="co">#&gt;   Querying PubMed for 20 pmids</span></span>
<span><span class="co">#&gt;   Querying without API key</span></span>
<span><span class="co">#&gt; Filter based on BugSigDB</span></span>
<span><span class="co">#&gt; Filtering 0 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Checking metabolites</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="va">metabEx</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/metabex-1.png" width="960"></div>
<p>In this way, we can plot links between microbes - metabolites - textual information. For all the information combined, one can plot textual information - metabolites - coded enzymes - diseases - microbes link in one query.</p>
<p>For the complex network, the resulting image might be unreadable.
<code>exportCyjsWithoutImage</code> function can be used to export the graph to readily interactive interface using <code>Cytoscape.js</code>. The below chunk shows the output produced by the function, hosted by GitHub Pages.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## exportCyjsWithoutImage(metabEx@igraph, rootDir=".", netDir="",nodeColorDiscretePal = "Pastel1")</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/include_url.html">include_url</a></span><span class="op">(</span><span class="st">"https://noriakis.github.io/cyjs_test/complex"</span><span class="op">)</span></span></code></pre></div>
<iframe src="https://noriakis.github.io/cyjs_test/complex" width="100%" height="400px" data-external="1" style="border: none;">
</iframe>
<div id="annotating-the-dendrogram-of-taxonomy" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> Annotating the dendrogram of taxonomy<a class="anchor" aria-label="anchor" href="#annotating-the-dendrogram-of-taxonomy"><i class="fas fa-link"></i></a>
</h2>
<p>Annotating the dendrogram (or cladogram) of taxonomy is possible by the <code>plotEigengeneNetworksWithWords</code> function. The used need to provide customized data.frame depicting query (taxonomy) and its corresponding text, like MetaCyc.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Read pathway description</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">"../../metacyc/24.5/data/pathways.dat"</span></span>
<span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/parseMetaCycPathway.html">parseMetaCycPathway</a></span><span class="op">(</span><span class="va">file</span>,</span>
<span>                              candSp<span class="op">=</span><span class="st">"all"</span>,</span>
<span>                              withTax<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                              clear<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                              noComma <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Convert the species name by taxonomizr</span></span>
<span><span class="va">input</span><span class="op">$</span><span class="va">spConverted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/convertMetaCyc.html">convertMetaCyc</a></span><span class="op">(</span><span class="va">input</span><span class="op">$</span><span class="va">species</span><span class="op">)</span></span>
<span><span class="va">input</span><span class="op">$</span><span class="va">spConverted</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Bacteria;Cyanobacteria;NA;Synechococcales;Prochloraceae;Prochloron;Prochloron didemni"                                 </span></span>
<span><span class="co">#&gt; [2] "Bacteria;Pseudomonadota;Gammaproteobacteria;Enterobacterales;Enterobacteriaceae;Escherichia;Escherichia coli"          </span></span>
<span><span class="co">#&gt; [3] "Bacteria;Pseudomonadota;Gammaproteobacteria;Enterobacterales;Enterobacteriaceae;Salmonella;Salmonella enterica"        </span></span>
<span><span class="co">#&gt; [4] "Bacteria;Pseudomonadota;Gammaproteobacteria;Enterobacterales;Enterobacteriaceae;Escherichia;Escherichia coli"          </span></span>
<span><span class="co">#&gt; [5] "Bacteria;Pseudomonadota;Betaproteobacteria;Burkholderiales;Burkholderiaceae;Paraburkholderia;Paraburkholderia phymatum"</span></span>
<span><span class="co">#&gt; [6] "Bacteria;Pseudomonadota;Betaproteobacteria;Burkholderiales;Burkholderiaceae;Cupriavidus;Cupriavidus basilensis"</span></span></code></pre></div>
<p>Now we fetch species under <code>Bacillota</code>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bc</span> <span class="op">&lt;-</span> <span class="va">input</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Bacillota"</span>, <span class="va">input</span><span class="op">$</span><span class="va">spConverted</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">bc</span> <span class="op">&lt;-</span> <span class="va">bc</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">bc</span><span class="op">$</span><span class="va">pathwayID</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">bc</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;      pathwayID</span></span>
<span><span class="co">#&gt; 12  PCEDEG-PWY</span></span>
<span><span class="co">#&gt; 173   PWY-7872</span></span>
<span><span class="co">#&gt; 349   PWY-7013</span></span>
<span><span class="co">#&gt; 748   PWY-7886</span></span>
<span><span class="co">#&gt; 749   PWY-8146</span></span>
<span><span class="co">#&gt; 799   PWY-7942</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               text</span></span>
<span><span class="co">#&gt; 12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Another group of organisms, called the halorespirators, has been shown to metabolize tetrachloroethene in a process known as dehalorespiration, in which halogenated organic compounds are used as electron acceptors in an anaerobic respiratory process . Halorespiration can result in the complete dechlorination of PCE and TRICHLOROETHENE (TCE) to the benign end products ETHYLENE-CMPD and CPD-9312. In most cases different organisms specialize in different steps of this pathway. Thus far, only TAX-243164 has been shown to completely dechlorinate TETRACHLOROETHENE to ETHYLENE-CMPD through anaerobic reductive dechlorination (dehalorespiration) .  TAX-243164 achieves this utilizing only two enzymes: EC-1.21.99.5 (PCE-RDase), which degrades PCE to TCE, and EC-1.21.99.M1 (TCE-RDase), which performs all the other steps of the pathway. However, TCE-RDase has a very low activity using DICHLOROETHENE and VINYL-CHLORIDE (vinyl chloride, or VC) as substrates, and is apparently catalyzing these reductions as a slow, cometabolic reaction (the reaction rate in &amp;mu;molminmg is 5.2 for TCE but only 0.036 for VC) . This phenomenon poses a problem for the treatment of contaminated sites, as it results in the accumulation of VINYL-CHLORIDE, which is the most toxic compound of all the chloroethenes. The problem can be alleviated by other organisms, such as TAX-311424, which have a dedicated EC-1.21.99.M2, which reduces VC and all dichloroethene (DCE) isomers at a high rate .  All of these enzymes are inhibited by the addition of iodoalkanes, and this inhibition is reversed by exposure to light, indicating the presence of corrinoid cofactors. In addition, most (but not all) of these enzymes have been shown to contain at least one  cluster. </span></span>
<span><span class="co">#&gt; 173                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The pathway for the degradation of CPD-19877 was found by a bioinformatic analysis of genes clustered with a D-threonate-binding protein (SBP), which is a part of a TRAP (tripartite ATP-independent permease) transporter for the compound .  Following import into the cell, CPD-19877 is phosphorylated to form ERYTHRONATE-4P by EC-2.7.1.220. The enzyme, encoded by the G-55076 gene, is one of the first members of a large family of enzymes, known as DUF (domain of unknown function) 1537, to which a function has been assigned .  Following phosphorylation, the compound is dehygrogenated at position 3 by EC-1.1.1.409. This dehydrogenase, with has some similarity to EC-1.1.1.262, catalyzes a dehydrogenation, generating an unstable product that undergoes a decarboxylation to yield the central metabolite DIHYDROXY-ACETONE-PHOSPHATE . </span></span>
<span><span class="co">#&gt; 349  PROPANE-1-2-DIOL (propylene glycol) is produced from LACTALD during the bacterial degradation of L-rhamnose and L-Fucopyranoses (see PWY0-1315 and the associated pathway links) (in ). TAX-28901 can utilize PROPANE-1-2-DIOL as a carbon source and its metabolism may be a virulence factor . Under fermentative conditions, CPD-665 (propionaldehyde) is formed and converted to PROPANOL and PROPIONATE (propionate), ATP is produced, and NAD is regenerated (in ). The reactions forming CPD-665 and PROPIONYL-COA occur within a specialized bacterial microcompartment known as a metabolosome (see the About This Pathway section below).   The first enzyme in this pathway is CPLX1R65-171, a diol dehydratase composed of polypeptides PduCDE. This enzyme is dependent upon ADENOSYLCOBALAMIN (vitamin B12), which requires anaerobic conditions for its de novo biosynthesis (see pathway PWY-5507). Initially it was thought that PROPANE-1-2-DIOL utilization required oxygen, leading to an apparent ADENOSYLCOBALAMIN paradox. This paradox was resolved by the discovery that TAX-28901 can use CPD-14 as an alternative electron acceptor during anaerobic degradation of both PROPANE-1-2-DIOL and ETHANOL-AMINE .  The adenosyl group of the ADENOSYLCOBALAMIN cofactor of CPLX1R65-171 is unstable in vivo and can be converted to CPD0-1256 in a reaction that inactivates the enzyme . Active enzyme can be regenerated within the microcompartment by the combined action of a reactivase PduGH  (biochemically characterized in TAX-571 ), a dedicated STM2050-MONOMER , and a cobalamin reductase PduS, that together regenerate ADENOSYLCOBALAMIN. Thus, the microcompartment contains a complete ADENOSYLCOBALAMIN recycling system  (and in ).  Under aerobic conditions, exogenously added ADENOSYLCOBALAMIN (or its precursor COBINAMIDE) can be imported into the cell to allow use of PROPANE-1-2-DIOL as a carbon and energy source via the PWY0-42 as shown in the pathway link  and in .  There is also evidence for the production of PROPANOL and PROPIONATE via this pathway in TAX-1637 species, although the PWY0-42 may not be present in these organisms .  In addition to the organisms shown here, genes for ADENOSYLCOBALAMIN-dependent propanediol utilization have also been identified in species of FRAME:TAX-620, FRAME:TAX-629, FRAME:TAX-1578 and FRAME:TAX-1357, as well as FRAME:TAX-562 E24377A (reviewed in ).  About This Pathway  The first two reactions of this pathway occur within the lumen of a bacterial microcompartment (BMC), a capsid-like shell composed of multiple proteins that is analogous to the carboxysome of cyanobacteria. The ADENOSYLCOBALAMIN-dependent PROPANE-1-2-DIOL utilization (Pdu) microcompartment consists of major polypeptides PduABB'CDEGHJKOPTU which includes enzymes PduCDE, PduGH, PduO and PduP, as well as BMC-domain containing proteins PduABB'JKNTU . A structural proein (PduM) has also been described . The proposed function of this microcompartment is to sequester toxic CPD-665 (propionaldehyde).  The remaining reactions occur in the cytosol (reviewed in  and ).  The conversion of CPD-665 to PROPANOL is catalyzed within the compartment by an alcohol dehydrogenasealdehyde reductase encoded by the STM2052 gene, which provides ample supply of NAD+ to EC-1.2.1.87 (STM2051 PduP) .  The PROPIONATE formed can be activated to PROPIONYL-COA by the reversible reactions catalyzed by PduW and PduL. PROPIONYL-COA Propanoyl-CoA can also be formed from PROPIONATE by the Acs and PrpE enzymes in pathway PWY0-42 (see EC-6.2.1.17). PROPIONYL-COA Propanoyl-CoA is thus an intermediate in the catabolism of both PROPANE-1-2-DIOL (this pathway) and PROPIONATE (in pathway PWY0-42) and is necessary for expression of the prpBCDE genes of the PWY0-42 during growth on PROPANE-1-2-DIOL .  The genes of this pathway are clustered and are located adjacent to the cbicob gene cluster needed for ADENOSYLCOBALAMIN biosynthesis. Both clusters are regulated by the pocR gene ( and in ).  The 21 gene regulon encoding the pdu organelle and propanediol degradation enzymes from TAX-546 was cloned and expressed successfully in TAX-562, allowing PROPANE-1-2-DIOL utilization . </span></span>
<span><span class="co">#&gt; 748                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Bacteria do not possess the PWY3O-450 eukaryote-like CDP-choline-dependent pathway for phosphatidylcholine synthesis, but it is now well-documented that a wide range of Gram-positive and Gram-negative species that colonize mucosal tissue utilize a pathway very similar to the eukaryotic pathway as a means to decorate cell wall glycoconjugates with phosphocholine. While the addition of phosphocholine to these glycoconjugates is not as common as acetylation or methylation, it has far more potent biological consequences due to the multiple interactions between phosphocholine and host proteins, which are important at all stages of the infection process by pathogens .   These phosphocholine glycoconjugate modifications are found on various surface structures depending on the species. Examples include Lipopolysaccharides lipopolysaccharides (TAX-727) , pili (TAX-487 and TAX-485) , Lipoteichoic-Acids lipoteichoic acid (TAX-1313) , a 43-kDa surface protein (TAX-287) , and surface components of diverse commensal oral species .  The first two enzyme activities in the pathway, EC-2.7.1.32, and EC-2.7.7.15, are essentially identical to those of the eukaryotic pathway. These enzymes are encoded by homologues of the eukaryotic genes (G-44724 and G-44725, respectively) that have been detected in many species that colonize mucosal tissue.  The third and last enzyme, diphosphonucleoside choline phosphotransferase, catalyzes the transfer of the phosphoclholine moiety to a cell surface glycoconjugate. The gene encoding this enzyme has not been definitively identified in many of these organisms , but is proposed to be G-43708 in TAX-727  and TAX-1313 . </span></span>
<span><span class="co">#&gt; 749                                                                                                                           C-type cytochromes (cytC) are electron-transfer proteins that have one or more HEME_C groups. They are characterized by the covalent attachment of the heme to the polypeptide chain via two (or rarely one) thioether bonds formed between thiol groups of cysteine residues and vinyl groups of a PROTOHEME molecule. The two cysteine residues almost always occur in the amino-acid sequence CXXCH .  C-type cytochromes possess a wide range of properties and function in a large number of different redox processes. In general, the differences in c-type cytochromes among bacteria are much larger than those among animal species. The use of the term term c-type cytochromes was introduced to distinguish this family of diverse proteins from the well-characterized mitochondrial protein, which is often referred to as cytochrome c.  Bacterial c-type cytochromes function in the electron transport chains of bacteria with many different types of energy metabolism, including phototrophs, methylotrophs, denitrifiers, sulfate reducers and the nitrogen-fixers.   The biogenesis of c-type cytochromes involves covalent attachment of PROTOHEME to two cysteines at a conserved CXXCH sequence in the apocytochrome. The histidine residue acts as an axial ligand to the heme iron. Three unique cytochrome c assembly pathways, known as systems I, II, and III, have ben described. This pathway describes system III, which is found in the mitochondria of fungi, vertebrates, invertebrates, and sme protozoa.  About This Pathway  The first indication for the existence of the system II pathway was the discovery that the ccsA gene (named for cytochrome c synthesis) in chloroplasts contains a WWD domain (a tryptophan-rich domain that is known to maintain a heme molecule in the reduced state by two external histidines) that is different from that found in the bacterial CcmF genes of system I . The gene was later shown to be required for cytochrome f biosynthesis (cytochrome f, a part of the cytochrome b6f complex, is analogous to cytochrome c1 of the bc1 complex and contains the CXXCH motif), and led to the discovery of the rest of the involved genes . Analogous genes were discovered in the Gram-positive bacterium TAX-1423  and the &amp;beta;-proteobacterium TAX-520 .  The system is now known to operate in a wide range of organisms, including chloroplasts, Gram-positive bacteria, cyanobacteria, most &amp;beta;-proteobacteria, some &amp;delta;-proteobacteria, and all &amp;epsilon;-proteobacteria .  In organisms that utilize system II the apocytochrome polypeptide, upon translocation from the cytosol, is exposed to the activity of EC-1.8.4.15, which forms a disulfide bridge between the two cysteines in the CXXCH motif. This bridge must be reduced before the heme molecule can be attached to the polypeptide. System II uses reducing equivalents from inside the cell (likely thioredoxin) to reduce the cysteine residues. Dedicated thioredoxin-related proteins called ResA (in TAX-1423) and CcsX (in TAX-520) can reduce the cysteines directly . These proteins are kept in their reduced active form by the DsbD protein, or by the similar, smaller protein CcdA .  The other components of system II are the CcsA and CcsB proteins, which form a tight complex with the activity of EC-4.4.1.17. In a few bacterial species (such as TAX-209, TAX-816, and TAX-843) the two proteins are fused into one large ORF called CcsBA. The two proteins are membrane-bound. CcsA, which has either eight or six transmembrane domains , contains the WWD domain that binds the heme molecule, as well as conserved histidine residues that are believed to help keep the heme iron in its reduced form. The reduced heme molecule moves from the cytosol through CcsA until it reaches the WWD domain, where it is liganded in place. CcsB has four transmembrane domains, with the third and fourth separated by a large periplasmic domain that includes the apocytochrome c CXXCH recognition function. CcsB thus recruites the apocytochrome polypeptide and presents it to the heme bound to CcsA . </span></span>
<span><span class="co">#&gt; 799                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     The cyclization is not limited to free amino acids. N-terminal glutaminyl and glutamyl residues of polypeptides can also spontaneously cyclize to 5-OXOPROLINE residues. The former transformation is also enzymically catalyzed by EC-2.3.2.5 . Once formed, the abberant residue is removed from the peptide by EC-3.4.19.3, yielding free 5-OXOPROLINE .  5-OXOPROLINE is also formed by additional routes. One such route is from GLUTATHIONE, in a reaction catalyzed by EC-4.3.2.7 . In eukaryotic organisms it is also formed by EC-4.3.2.9 as part of the PWY-4041 .   Accumulation of 5-OXOPROLINE is deleterious to the organism. Its reported effects include growth inhibition in prokaryotes and plants  and interference with energy production, lipid synthesis, and antioxidant defenses in mammalian brain . Human inborn errors of glutathione metabolism that lead to 5-OXOPROLINE buildup result in metabolic acidosis, hemolytic anemia, neurological problems, and massive urinary excretion of 5-OXOPROLINE .  An ATP-dependent enzyme that hydrolyzes 5-OXOPROLINE to GLT, EC-3.5.2.9, has been known for a while, as it functions in the PWY-4041. Until recently, the enzyme, which is encoded by the HS11319 gene, was thought to be a eukaryotic enzyme exclusively . However, in 2017 a widespread prokaryotic enzyme that catalyzes the same reaction was discovered . The trimeric enzyme is encoded by the G6382, G6380, and G6381 genes. Inactivation of any of these genes in TAX-1423 slowed growth and caused 5-OXOPROLINE accumulation in the cells. </span></span>
<span><span class="co">#&gt;                                                         commonName</span></span>
<span><span class="co">#&gt; 12                                   tetrachloroethene degradation</span></span>
<span><span class="co">#&gt; 173                                    D-erythronate degradation I</span></span>
<span><span class="co">#&gt; 349                        (&lt;i&gt;S&lt;/i&gt;)-propane-1,2-diol degradation</span></span>
<span><span class="co">#&gt; 748 cell-surface glycoconjugate-linked phosphocholine biosynthesis</span></span>
<span><span class="co">#&gt; 749                cytochrome &lt;i&gt;c&lt;/i&gt; biogenesis (system II type)</span></span>
<span><span class="co">#&gt; 799                                     5-oxo-L-proline metabolism</span></span>
<span><span class="co">#&gt;        species taxonomicRange</span></span>
<span><span class="co">#&gt; 12   TAX-55583          TAX-2</span></span>
<span><span class="co">#&gt; 173 TAX-498761          TAX-2</span></span>
<span><span class="co">#&gt; 349   TAX-1642          TAX-2</span></span>
<span><span class="co">#&gt; 748   TAX-1313          TAX-2</span></span>
<span><span class="co">#&gt; 749 TAX-224308      TAX-28216</span></span>
<span><span class="co">#&gt; 799   TAX-1423       TAX-2759</span></span>
<span><span class="co">#&gt;                                                                                                   spConverted</span></span>
<span><span class="co">#&gt; 12     Bacteria;Bacillota;Clostridia;Eubacteriales;Desulfitobacteriaceae;Dehalobacter;Dehalobacter restrictus</span></span>
<span><span class="co">#&gt; 173 Bacteria;Bacillota;Clostridia;Eubacteriales;Heliobacteriaceae;Heliomicrobium;Heliomicrobium modesticaldum</span></span>
<span><span class="co">#&gt; 349                              Bacteria;Bacillota;Bacilli;Bacillales;Listeriaceae;Listeria;Listeria innocua</span></span>
<span><span class="co">#&gt; 748        Bacteria;Bacillota;Bacilli;Lactobacillales;Streptococcaceae;Streptococcus;Streptococcus pneumoniae</span></span>
<span><span class="co">#&gt; 749                              Bacteria;Bacillota;Bacilli;Bacillales;Bacillaceae;Bacillus;Bacillus subtilis</span></span>
<span><span class="co">#&gt; 799                              Bacteria;Bacillota;Bacilli;Bacillales;Bacillaceae;Bacillus;Bacillus subtilis</span></span></code></pre></div>
<p>This time we use a random dendrogram and rename the rows with taxonomic name included in MetaCyc.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">uniq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html">sample</a></span><span class="op">(</span><span class="va">bc</span><span class="op">$</span><span class="va">species</span>, <span class="fl">10</span><span class="op">)</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">uniq</span><span class="op">)</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">uniq</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># if (sum(is.na(sapply(strsplit(bc[bc$species %in% uniq,]$spConverted, ";"), "[", 7)))==0) {</span></span>
<span>      <span class="kw">break</span></span>
<span>    <span class="co"># }</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">100</span><span class="op">)</span>,<span class="fl">100</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">uniq</span></span>
<span></span>
<span><span class="co">## Dendrogram</span></span>
<span><span class="va">dhc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html">as.dendrogram</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">dhc</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/example-1.png" width="672"></div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Maket it cladogram</span></span>
<span><span class="va">dhc</span> <span class="op">&lt;-</span> <span class="fu">phylogram</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/phylogram/reference/as.cladogram.html">as.cladogram</a></span><span class="op">(</span><span class="va">dhc</span><span class="op">)</span></span></code></pre></div>
<p>Using the dendrogram (<code>dhc</code> argument) and input data.frame with the column name <code>query</code>, we can plot the dendrogram with pathway information. Note we need to provide the function named vector of nodes (corresponding to gene clusters when the gene as input). This time, wordclouds are to be plotted and additional arguments can by specified in <code>argList</code>.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Some filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../allFreqMetaCyc.rda"</span><span class="op">)</span></span>
<span><span class="va">deleter</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">allFreqMetaCyc</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">freq</span><span class="op">&gt;</span><span class="fl">5000</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">word</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">word</span></span>
<span><span class="co"># Named vector</span></span>
<span><span class="va">sampled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sampled</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sampled</span></span>
<span><span class="co"># Query column (including node names) is specified as query</span></span>
<span><span class="va">input</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="va">input</span><span class="op">$</span><span class="va">species</span></span>
<span></span>
<span><span class="va">micro</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/wcGeneSummary/man/plotEigengeneNetworksWithWords.html">plotEigengeneNetworksWithWords</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="va">sampled</span>,</span>
<span>                               useWC <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Use wordcloud</span></span>
<span>                               useFunc <span class="op">=</span> <span class="st">"wcMan"</span>, <span class="co"># Use manual function (as the input is custom data.frame)</span></span>
<span>                               useDf<span class="op">=</span><span class="va">input</span>,dendPlot<span class="op">=</span><span class="st">"ggplot"</span>,dhc<span class="op">=</span><span class="va">dhc</span>,</span>
<span>                               argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>additionalRemove<span class="op">=</span><span class="va">deleter</span>,</span>
<span>                                ngram<span class="op">=</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>                               useWGCNA<span class="op">=</span><span class="cn">FALSE</span>, spacer<span class="op">=</span><span class="fl">0.05</span>,</span>
<span>                               horiz<span class="op">=</span><span class="cn">FALSE</span>, wcScale <span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">scaled</span> <span class="op">&lt;-</span> <span class="va">micro</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Non-scaled and scaled</span></span>
<span><span class="va">micro</span> <span class="op">+</span> <span class="va">scaled</span></span></code></pre></div>
<div class="inline-figure"><img src="02-microbiome_usage_files/figure-html/plotDendroWord-1.png" width="1920"></div>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19044)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Japanese_Japan.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Japanese_Japan.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                   </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [6] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] RColorBrewer_1.1-3   ggraph_2.1.0        </span></span>
<span><span class="co">#&gt; [3] ggplot2_3.4.0        wcGeneSummary_0.99.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] GeneSummary_0.99.4     colorspace_2.0-3      </span></span>
<span><span class="co">#&gt;   [3] rjson_0.2.21           ellipsis_0.3.2        </span></span>
<span><span class="co">#&gt;   [5] XVector_0.38.0         GlobalOptions_0.1.2   </span></span>
<span><span class="co">#&gt;   [7] base64enc_0.1-3        ggdendro_0.1.23       </span></span>
<span><span class="co">#&gt;   [9] fs_1.5.2               rstudioapi_0.14       </span></span>
<span><span class="co">#&gt;  [11] farver_2.1.1           graphlayouts_0.8.4    </span></span>
<span><span class="co">#&gt;  [13] ggrepel_0.9.2          bit64_4.0.5           </span></span>
<span><span class="co">#&gt;  [15] AnnotationDbi_1.60.0   fansi_1.0.3           </span></span>
<span><span class="co">#&gt;  [17] xml2_1.3.3             downlit_0.4.2         </span></span>
<span><span class="co">#&gt;  [19] phylogram_2.1.0        cachem_1.0.6          </span></span>
<span><span class="co">#&gt;  [21] knitr_1.41             polyclip_1.10-4       </span></span>
<span><span class="co">#&gt;  [23] jsonlite_1.8.4         png_0.1-8             </span></span>
<span><span class="co">#&gt;  [25] graph_1.76.0           ggforce_0.4.1         </span></span>
<span><span class="co">#&gt;  [27] shiny_1.7.4            bugsigdbr_1.4.1       </span></span>
<span><span class="co">#&gt;  [29] rentrez_1.2.3          compiler_4.2.2        </span></span>
<span><span class="co">#&gt;  [31] httr_1.4.4             fastmap_1.1.0         </span></span>
<span><span class="co">#&gt;  [33] cli_3.6.0              later_1.3.0           </span></span>
<span><span class="co">#&gt;  [35] tweenr_2.0.2           htmltools_0.5.4       </span></span>
<span><span class="co">#&gt;  [37] tools_4.2.2            igraph_1.3.5          </span></span>
<span><span class="co">#&gt;  [39] NLP_0.2-1              gtable_0.3.1          </span></span>
<span><span class="co">#&gt;  [41] glue_1.6.2             GenomeInfoDbData_1.2.9</span></span>
<span><span class="co">#&gt;  [43] dplyr_1.0.99.9000      Rcpp_1.0.9            </span></span>
<span><span class="co">#&gt;  [45] slam_0.1-50            Biobase_2.58.0        </span></span>
<span><span class="co">#&gt;  [47] jquerylib_0.1.4        vctrs_0.5.1           </span></span>
<span><span class="co">#&gt;  [49] Biostrings_2.66.0      nlme_3.1-161          </span></span>
<span><span class="co">#&gt;  [51] ape_5.6-2              xfun_0.36             </span></span>
<span><span class="co">#&gt;  [53] stringr_1.5.0          mime_0.12             </span></span>
<span><span class="co">#&gt;  [55] lifecycle_1.0.3        pvclust_2.2-0         </span></span>
<span><span class="co">#&gt;  [57] XML_3.99-0.13          dendextend_1.16.0     </span></span>
<span><span class="co">#&gt;  [59] org.Hs.eg.db_3.16.0    zlibbioc_1.44.0       </span></span>
<span><span class="co">#&gt;  [61] MASS_7.3-58.1          scales_1.2.1          </span></span>
<span><span class="co">#&gt;  [63] tidygraph_1.2.2        promises_1.2.0.1      </span></span>
<span><span class="co">#&gt;  [65] parallel_4.2.2         cyjShiny_1.0.34       </span></span>
<span><span class="co">#&gt;  [67] yaml_2.3.6             memoise_2.0.1         </span></span>
<span><span class="co">#&gt;  [69] gridExtra_2.3          yulab.utils_0.0.6     </span></span>
<span><span class="co">#&gt;  [71] sass_0.4.4             stringi_1.7.12        </span></span>
<span><span class="co">#&gt;  [73] RSQLite_2.2.20         highr_0.10            </span></span>
<span><span class="co">#&gt;  [75] S4Vectors_0.36.1       BiocGenerics_0.44.0   </span></span>
<span><span class="co">#&gt;  [77] GenomeInfoDb_1.34.6    rlang_1.0.6           </span></span>
<span><span class="co">#&gt;  [79] pkgconfig_2.0.3        bitops_1.0-7          </span></span>
<span><span class="co">#&gt;  [81] lattice_0.20-45        evaluate_0.19         </span></span>
<span><span class="co">#&gt;  [83] purrr_1.0.1            patchwork_1.1.2       </span></span>
<span><span class="co">#&gt;  [85] htmlwidgets_1.6.1      cowplot_1.1.1         </span></span>
<span><span class="co">#&gt;  [87] bit_4.0.5              tidyselect_1.2.0      </span></span>
<span><span class="co">#&gt;  [89] magrittr_2.0.3         bookdown_0.31         </span></span>
<span><span class="co">#&gt;  [91] R6_2.5.1               IRanges_2.32.0        </span></span>
<span><span class="co">#&gt;  [93] generics_0.1.3         DBI_1.1.3             </span></span>
<span><span class="co">#&gt;  [95] pillar_1.8.1           withr_2.5.0           </span></span>
<span><span class="co">#&gt;  [97] KEGGREST_1.38.0        RCurl_1.98-1.9        </span></span>
<span><span class="co">#&gt;  [99] tibble_3.1.8           crayon_1.5.2          </span></span>
<span><span class="co">#&gt; [101] wordcloud_2.6          utf8_1.2.2            </span></span>
<span><span class="co">#&gt; [103] rmarkdown_2.19         viridis_0.6.2         </span></span>
<span><span class="co">#&gt; [105] GetoptLong_1.0.5       grid_4.2.2            </span></span>
<span><span class="co">#&gt; [107] blob_1.2.3             digest_0.6.31         </span></span>
<span><span class="co">#&gt; [109] xtable_1.8-4           tm_0.7-10             </span></span>
<span><span class="co">#&gt; [111] tidyr_1.2.1            httpuv_1.6.8          </span></span>
<span><span class="co">#&gt; [113] gridGraphics_0.5-1     stats4_4.2.2          </span></span>
<span><span class="co">#&gt; [115] munsell_0.5.0          viridisLite_0.4.1     </span></span>
<span><span class="co">#&gt; [117] ggplotify_0.1.0        bslib_0.4.2</span></span></code></pre></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></div>
<div class="next"><a href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">4</span> Interactive inspection of gene cluster network annotated by words</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#the-basic-usage-for-microbiome-related-data"><span class="header-section-number">3</span> The basic usage for microbiome-related data</a></li>
<li><a class="nav-link" href="#annotating-the-dendrogram-of-taxonomy"><span class="header-section-number">3.1</span> Annotating the dendrogram of taxonomy</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>wcGeneSummary (OSplot)</strong>" was written by Noriaki Sato. It was last built on 2023-03-02.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
