<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 Application examples | biotextgraph</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.33 with bs4_book()">
<meta property="og:title" content="Chapter 9 Application examples | biotextgraph">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/biotextgraph/app.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 Application examples | biotextgraph">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-9FXY6YW4YZ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-9FXY6YW4YZ');
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="We demonstrate an use case of the package, which investigates transcriptomic changes induced by BK polyomavirus (BKPyV) infection in renal proximal tubular epithelial cells (Assetta et al. 2016)....">
<meta property="og:description" content="We demonstrate an use case of the package, which investigates transcriptomic changes induced by BK polyomavirus (BKPyV) infection in renal proximal tubular epithelial cells (Assetta et al. 2016)....">
<meta name="twitter:description" content="We demonstrate an use case of the package, which investigates transcriptomic changes induced by BK polyomavirus (BKPyV) infection in renal proximal tubular epithelial cells (Assetta et al. 2016)....">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">biotextgraph</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="" href="metacyc.html"><span class="header-section-number">3</span> Microbial functional analysis</a></li>
<li><a class="" href="the-other-usage-for-microbiome-related-data.html"><span class="header-section-number">4</span> The other usage for microbiome-related data</a></li>
<li><a class="" href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">5</span> Interactive inspection of gene cluster network annotated by words</a></li>
<li><a class="" href="annotating-gene-cluster-dendrogram.html"><span class="header-section-number">6</span> Annotating gene cluster dendrogram</a></li>
<li><a class="" href="quantitative-analysis.html"><span class="header-section-number">7</span> Quantitative analysis</a></li>
<li><a class="" href="pathview-text.html"><span class="header-section-number">8</span> Pathview text</a></li>
<li><a class="active" href="app.html"><span class="header-section-number">9</span> Application examples</a></li>
<li><a class="" href="single-cell-transcriptomic-data.html"><span class="header-section-number">10</span> Single-cell transcriptomic data</a></li>
<li><a class="" href="parsing.html"><span class="header-section-number">11</span> Parsing</a></li>
<li><a class="" href="tidy.html"><span class="header-section-number">12</span> Perform each process separately</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="app" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> Application examples<a class="anchor" aria-label="anchor" href="#app"><i class="fas fa-link"></i></a>
</h1>
<p>We demonstrate an use case of the package, which investigates transcriptomic changes induced by BK polyomavirus (BKPyV) infection in renal proximal tubular epithelial cells (<a href="https://pubmed.ncbi.nlm.nih.gov/27381292/">Assetta et al. 2016</a>). Differentially expressed mRNAs in 3 days post-infection were obtained, and down-regulated mRNAs in BKPyV infected cells were examined.</p>
<div id="load-the-necessary-packages-and-genes" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> Load the necessary packages and genes<a class="anchor" aria-label="anchor" href="#load-the-necessary-packages-and-genes"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biotextgraph</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggforce.data-imaginist.com">ggforce</a></span><span class="op">)</span> <span class="co">## For genePathPlot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"biotextgraph"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">degs</span> <span class="op">&lt;-</span> <span class="va">d3degDownAssetta2016</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">degs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 191</span></span>
<span><span class="va">degs</span></span>
<span><span class="co">#&gt;   [1] "ABCB4"           "ABCB7"           "AKTIP"          </span></span>
<span><span class="co">#&gt;   [4] "ALS2"            "ANKRA2"          "ANTXR1"         </span></span>
<span><span class="co">#&gt;   [7] "APH1B"           "ARHGEF28"        "ARNTL"          </span></span>
<span><span class="co">#&gt;  [10] "ATMIN"           "BDH2"            "BIRC2"          </span></span>
<span><span class="co">#&gt;  [13] "BTG1"            "BTG2"            "C4A"            </span></span>
<span><span class="co">#&gt;  [16] "C7orf60"         "C8orf4"          "CALCOCO1"       </span></span>
<span><span class="co">#&gt;  [19] "CAPRIN2"         "CARS"            "CBLB"           </span></span>
<span><span class="co">#&gt;  [22] "CCNDBP1"         "CCNG2"           "CCZ1B"          </span></span>
<span><span class="co">#&gt;  [25] "CDC42EP3"        "CDH6"            "CFLAR"          </span></span>
<span><span class="co">#&gt;  [28] "CHMP1B"          "CHMP4C"          "CLUAP1"         </span></span>
<span><span class="co">#&gt;  [31] "COG2"            "COG3"            "CPQ"            </span></span>
<span><span class="co">#&gt;  [34] "CROT"            "CTTNBP2NL"       "CYP4V2"         </span></span>
<span><span class="co">#&gt;  [37] "DAB2"            "DDB2"            "DDX17"          </span></span>
<span><span class="co">#&gt;  [40] "DDX5"            "DGKA"            "DHX32"          </span></span>
<span><span class="co">#&gt;  [43] "DLG1"            "DYNC2LI1"        "DYRK2"          </span></span>
<span><span class="co">#&gt;  [46] "EFHC1"           "EIF4A2"          "ERMAP"          </span></span>
<span><span class="co">#&gt;  [49] "ERMARD"          "EXOC1"           "FAM134B"        </span></span>
<span><span class="co">#&gt;  [52] "FAM160B1"        "FAM21C"          "FAM84B"         </span></span>
<span><span class="co">#&gt;  [55] "FANK1"           "FAS"             "FBXO38"         </span></span>
<span><span class="co">#&gt;  [58] "FCHO2"           "FGD6"            "FLJ22447"       </span></span>
<span><span class="co">#&gt;  [61] "FMNL2"           "GADD45A"         "GJA1"           </span></span>
<span><span class="co">#&gt;  [64] "GLIDR"           "GLT8D1"          "GOPC"           </span></span>
<span><span class="co">#&gt;  [67] "GPBP1L1"         "GPR155"          "GPR75-ASB3"     </span></span>
<span><span class="co">#&gt;  [70] "GRAMD3"          "HADHB"           "HCG11"          </span></span>
<span><span class="co">#&gt;  [73] "HDAC9"           "HDHD2"           "HERPUD1"        </span></span>
<span><span class="co">#&gt;  [76] "HSDL2"           "ICA1"            "ICK"            </span></span>
<span><span class="co">#&gt;  [79] "IFNGR1"          "IFT46"           "IFT74"          </span></span>
<span><span class="co">#&gt;  [82] "IRF6"            "ITGA2"           "ITGA6"          </span></span>
<span><span class="co">#&gt;  [85] "ITGAV"           "ITGB6"           "KDM5B"          </span></span>
<span><span class="co">#&gt;  [88] "KIF3A"           "KIF5B"           "KLHL20"         </span></span>
<span><span class="co">#&gt;  [91] "KLHL24"          "KLHL9"           "KPNA5"          </span></span>
<span><span class="co">#&gt;  [94] "KRCC1"           "L3MBTL3"         "LINC00657"      </span></span>
<span><span class="co">#&gt;  [97] "LOC100131564"    "LZTFL1"          "MAMDC2"         </span></span>
<span><span class="co">#&gt; [100] "MAP4K5"          "MAT2B"           "MBNL2"          </span></span>
<span><span class="co">#&gt; [103] "MDM2"            "MECOM"           "MFSD1"          </span></span>
<span><span class="co">#&gt; [106] "MGEA5"           "MICU3"           "MSANTD4"        </span></span>
<span><span class="co">#&gt; [109] "NBPF11"          "NCBP2"           "NEAT1"          </span></span>
<span><span class="co">#&gt; [112] "NTM"             "OGT"             "PAFAH1B2"       </span></span>
<span><span class="co">#&gt; [115] "PAFAH2"          "PCMTD2"          "PDE4D"          </span></span>
<span><span class="co">#&gt; [118] "PDP1"            "PELI1"           "PEX1"           </span></span>
<span><span class="co">#&gt; [121] "PHF14"           "PHOSPHO2-KLHL23" "PIK3IP1"        </span></span>
<span><span class="co">#&gt; [124] "PLA2R1"          "PLCB4"           "PLK2"           </span></span>
<span><span class="co">#&gt; [127] "POLI"            "POSTN"           "PPAN-P2RY11"    </span></span>
<span><span class="co">#&gt; [130] "PPFIBP1"         "PPP2CB"          "PRICKLE1"       </span></span>
<span><span class="co">#&gt; [133] "PROS1"           "PSMD5-AS1"       "RAD17"          </span></span>
<span><span class="co">#&gt; [136] "RHOQ"            "RIMKLB"          "RNA18S5"        </span></span>
<span><span class="co">#&gt; [139] "RNF170"          "RNF20"           "RNU1-28P"       </span></span>
<span><span class="co">#&gt; [142] "RPL23AP53"       "RRM2B"           "RRN3"           </span></span>
<span><span class="co">#&gt; [145] "RRN3P1"          "SEMA3C"          "SERINC1"        </span></span>
<span><span class="co">#&gt; [148] "SESN1"           "SESN3"           "SGK1"           </span></span>
<span><span class="co">#&gt; [151] "SLC22A5"         "SLC37A3"         "SVIL"           </span></span>
<span><span class="co">#&gt; [154] "SYT11"           "TARSL2"          "TBC1D19"        </span></span>
<span><span class="co">#&gt; [157] "TBCK"            "TBRG1"           "TGFA"           </span></span>
<span><span class="co">#&gt; [160] "TGFB2"           "TIPARP"          "TMEM136"        </span></span>
<span><span class="co">#&gt; [163] "TNFRSF10B"       "TNFRSF10D"       "TOM1L1"         </span></span>
<span><span class="co">#&gt; [166] "TP53INP1"        "TRIM13"          "TRIM32"         </span></span>
<span><span class="co">#&gt; [169] "TRIM4"           "TSC1"            "TSPYL5"         </span></span>
<span><span class="co">#&gt; [172] "UGT2B7"          "UNC13B"          "UPRT"           </span></span>
<span><span class="co">#&gt; [175] "VPS41"           "VPS8"            "WDR11"          </span></span>
<span><span class="co">#&gt; [178] "WDR19"           "XPC"             "YPEL2"          </span></span>
<span><span class="co">#&gt; [181] "ZC2HC1A"         "ZFAND5"          "ZFP90"          </span></span>
<span><span class="co">#&gt; [184] "ZFR"             "ZMAT3"           "ZNF12"          </span></span>
<span><span class="co">#&gt; [187] "ZNF248"          "ZNF322"          "ZNF561"         </span></span>
<span><span class="co">#&gt; [190] "ZNF626"          "ZSCAN30"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="enrichment-analysis" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>First, we perform enrichment analysis using ReactomePA.
From the enrichment analysis results, the cluster is related to transcriptional regulation by TP53.</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Convert to ENTREZID</span></span>
<span><span class="va">entre</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"SYMBOL"</span>,</span>
<span>                               keys <span class="op">=</span> <span class="va">degs</span>, columns <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span><span class="va">pway</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html">enrichPathway</a></span><span class="op">(</span><span class="va">entre</span><span class="op">)</span>, <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">sigpway</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">pway</span><span class="op">@</span><span class="va">result</span>, <span class="va">p.adjust</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">sigpway</span><span class="op">$</span><span class="va">Description</span></span>
<span><span class="co">#&gt;  [1] "Transcriptional Regulation by TP53"                              </span></span>
<span><span class="co">#&gt;  [2] "RIPK1-mediated regulated necrosis"                               </span></span>
<span><span class="co">#&gt;  [3] "Regulation of necroptotic cell death"                            </span></span>
<span><span class="co">#&gt;  [4] "Regulated Necrosis"                                              </span></span>
<span><span class="co">#&gt;  [5] "Intraflagellar transport"                                        </span></span>
<span><span class="co">#&gt;  [6] "Regulation by c-FLIP"                                            </span></span>
<span><span class="co">#&gt;  [7] "CASP8 activity is inhibited"                                     </span></span>
<span><span class="co">#&gt;  [8] "Dimerization of procaspase-8"                                    </span></span>
<span><span class="co">#&gt;  [9] "TP53 Regulates Transcription of Death Receptors and Ligands"     </span></span>
<span><span class="co">#&gt; [10] "Caspase activation via Death Receptors in the presence of ligand"</span></span>
<span><span class="co">#&gt; [11] "FOXO-mediated transcription of cell cycle genes"                 </span></span>
<span><span class="co">#&gt; [12] "TP53 Regulates Transcription of Cell Death Genes"                </span></span>
<span><span class="co">#&gt; [13] "Cilium Assembly"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">pway</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/ea-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Genes involved in significant pathways</span></span>
<span><span class="va">excheck</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">sigpway</span><span class="op">$</span><span class="va">geneID</span>,</span>
<span>                                <span class="kw">function</span> <span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">x</span>,<span class="st">"/"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">excheck</span></span>
<span><span class="co">#&gt;  [1] "BTG2"      "DDB2"      "DYRK2"     "FAS"      </span></span>
<span><span class="co">#&gt;  [5] "GADD45A"   "MDM2"      "PLK2"      "PPP2CB"   </span></span>
<span><span class="co">#&gt;  [9] "RAD17"     "RRM2B"     "SESN1"     "SESN3"    </span></span>
<span><span class="co">#&gt; [13] "SGK1"      "TNFRSF10B" "TNFRSF10D" "TP53INP1" </span></span>
<span><span class="co">#&gt; [17] "TSC1"      "BIRC2"     "CFLAR"     "FAS"      </span></span>
<span><span class="co">#&gt; [21] "OGT"       "PELI1"     "TNFRSF10B" "BIRC2"    </span></span>
<span><span class="co">#&gt; [25] "CFLAR"     "CHMP4C"    "FAS"       "OGT"      </span></span>
<span><span class="co">#&gt; [29] "PELI1"     "TNFRSF10B" "CLUAP1"    "DYNC2LI1" </span></span>
<span><span class="co">#&gt; [33] "IFT46"     "IFT74"     "KIF3A"     "WDR19"    </span></span>
<span><span class="co">#&gt; [37] "CFLAR"     "FAS"       "TNFRSF10B" "FAS"      </span></span>
<span><span class="co">#&gt; [41] "TNFRSF10B" "TNFRSF10D" "BTG1"      "CCNG2"    </span></span>
<span><span class="co">#&gt; [45] "GADD45A"   "FAS"       "TNFRSF10B" "TNFRSF10D"</span></span>
<span><span class="co">#&gt; [49] "TP53INP1"  "CLUAP1"    "DYNC2LI1"  "EXOC1"    </span></span>
<span><span class="co">#&gt; [53] "IFT46"     "IFT74"     "KIF3A"     "LZTFL1"   </span></span>
<span><span class="co">#&gt; [57] "WDR19"</span></span></code></pre></div>
<p>We store the name of enriched pathways in the network for the downstream analysis.</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">netreac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/refseq.html">refseq</a></span><span class="op">(</span><span class="va">degs</span>,</span>
<span>            enrich<span class="op">=</span><span class="st">"reactome"</span>,</span>
<span>            plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>            numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>            colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 191</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 174</span></span>
<span><span class="co">#&gt; Performing enrichment analysis</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span></code></pre></div>
</div>
<div id="text-mining-the-gene-summaries" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> Text mining the gene summaries<a class="anchor" aria-label="anchor" href="#text-mining-the-gene-summaries"><i class="fas fa-link"></i></a>
</h2>
<p>Next we perform the plain function producing a correlation network, with showing the top-genes related to high-frequency words in the text in RefSeq summary. We obtained the list of these genes from geneCount slot.</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/refseq.html">refseq</a></span><span class="op">(</span><span class="va">excheck</span>,</span>
<span>                plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                colorText<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>                corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>                genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                genePlotNum<span class="op">=</span><span class="fl">5</span>,</span>
<span>                edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                genePathPlot<span class="op">=</span><span class="st">"reactome"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 57</span></span>
<span><span class="co">#&gt;   Converted input genes: 32</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Found 28 enriched term</span></span>
<span><span class="va">net1</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/basic2app-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span><span class="op">@</span><span class="va">geneCount</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     RAD17     SESN1      TSC1      DDB2 TNFRSF10B      BTG1 </span></span>
<span><span class="co">#&gt;         9         9         6         5         5         4 </span></span>
<span><span class="co">#&gt;     DYRK2     BIRC2     CCNG2       FAS   GADD45A    LZTFL1 </span></span>
<span><span class="co">#&gt;         4         3         3         3         3         3 </span></span>
<span><span class="co">#&gt;      MDM2     SESN3      SGK1     CFLAR    CHMP4C       OGT </span></span>
<span><span class="co">#&gt;         3         3         3         2         2         2 </span></span>
<span><span class="co">#&gt; TNFRSF10D     WDR19      BTG2  DYNC2LI1     EXOC1     IFT74 </span></span>
<span><span class="co">#&gt;         2         2         1         1         1         1 </span></span>
<span><span class="co">#&gt;    PPP2CB     RRM2B </span></span>
<span><span class="co">#&gt;         1         1</span></span>
<span><span class="va">top</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">net1</span><span class="op">@</span><span class="va">geneCount</span><span class="op">[</span><span class="va">net1</span><span class="op">@</span><span class="va">geneCount</span><span class="op">&gt;=</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">top</span></span>
<span><span class="co">#&gt; [1] "RAD17"     "SESN1"     "TSC1"      "DDB2"     </span></span>
<span><span class="co">#&gt; [5] "TNFRSF10B"</span></span></code></pre></div>
</div>
<div id="text-mining-the-available-literature" class="section level2" number="9.4">
<h2>
<span class="header-section-number">9.4</span> Text mining the available literature<a class="anchor" aria-label="anchor" href="#text-mining-the-available-literature"><i class="fas fa-link"></i></a>
</h2>
<p>These genes are further queried for PubMed information.
The results were obtained in April, 2023.
First, show the network for the article titles.</p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># titlenet &lt;- pubmed(top,</span></span>
<span><span class="co">#                    sortOrder="relevance",</span></span>
<span><span class="co">#                    target="title",</span></span>
<span><span class="co">#                    plotType="network",</span></span>
<span><span class="co">#                    colorText=TRUE,</span></span>
<span><span class="co">#                    madeUpperGenes=FALSE,</span></span>
<span><span class="co">#                    corThresh=0.2,</span></span>
<span><span class="co">#                    pre=TRUE,</span></span>
<span><span class="co">#                    retMax=40,</span></span>
<span><span class="co">#                    numWords=40,</span></span>
<span><span class="co">#                    edgeLink=FALSE)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"titlenet_20230420.rda"</span><span class="op">)</span></span>
<span><span class="va">titlenet</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/absttitleapp-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>Obtain and show the network for the article abstract.</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># abstnet &lt;- pubmed(top,</span></span>
<span><span class="co">#                    target="abstract",</span></span>
<span><span class="co">#                    plotType="network",</span></span>
<span><span class="co">#                    colorText=TRUE,</span></span>
<span><span class="co">#                    madeUpperGenes=FALSE,</span></span>
<span><span class="co">#                    corThresh=0.2,</span></span>
<span><span class="co">#                    pre=TRUE,</span></span>
<span><span class="co">#                    retMax=40,</span></span>
<span><span class="co">#                    numWords=40,</span></span>
<span><span class="co">#                    edgeLink=FALSE)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"abstnet_20230420.rda"</span><span class="op">)</span></span>
<span><span class="va">abstnet</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/abstmainapp-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="combine-and-inspect-merged-network" class="section level2" number="9.5">
<h2>
<span class="header-section-number">9.5</span> Combine and inspect merged network<a class="anchor" aria-label="anchor" href="#combine-and-inspect-merged-network"><i class="fas fa-link"></i></a>
</h2>
<p>From the RefSeq summary and articles related to important genes, the cluster could have functionality of DNA damage response, which is also upregulated by BKPyV infection. These networks can be combined to find intersections and differences. We can see that in addition to Reactome pathway names, plenty of information could be obtained and summarized by querying other databases, which could aid in interpreting clusters of genes and hypothesis generation.</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">abstnet</span>, <span class="va">titlenet</span>, <span class="va">netreac</span>, <span class="va">net1</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Abstract"</span>,<span class="st">"Title"</span>,<span class="st">"Reactome"</span>,<span class="st">"RefSeq"</span><span class="op">)</span><span class="op">)</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/combineapp-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>From the network, DNA damage repair pathway, especially nucleotide excision repair related to DDB2, METTL14,and RAD17 might be related to BKPyV infection, which cannot be prioritize based on log2FoldChange or enrichment analysis.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># btg &lt;- compareWordNet(list(net1,titlenet,abstnet,netreac))</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"btg_20230420.rda"</span><span class="op">)</span></span>
<span><span class="va">conet</span> <span class="op">&lt;-</span> <span class="va">btg</span><span class="op">@</span><span class="va">igraphRaw</span></span>
<span></span>
<span><span class="va">ddrNms</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">nm</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/V.html">V</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">nm</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dna"</span>,<span class="st">"damage"</span>,<span class="st">"repair"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">ddrNms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">ddrNms</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/neighbors.html">neighbors</a></span><span class="op">(</span><span class="va">conet</span>, <span class="va">nm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="va">ddrNms</span></span>
<span><span class="co">#&gt;  [1] "checkpoint"      "cycle"           "DNA"            </span></span>
<span><span class="co">#&gt;  [4] "phosphorylation" "required"        "response"       </span></span>
<span><span class="co">#&gt;  [7] "SESN1"           "DDB2"            "RAD17"          </span></span>
<span><span class="co">#&gt; [10] "checkpoint"      "cycle"           "damage"         </span></span>
<span><span class="co">#&gt; [13] "phosphorylation" "required"        "response"       </span></span>
<span><span class="co">#&gt; [16] "SESN1"           "DDB2"            "RAD17"          </span></span>
<span><span class="co">#&gt; [19] "Cancer"          "degradation"     "protein"        </span></span>
<span><span class="co">#&gt; [22] "regulates"       "Repair"          "CDT2"           </span></span>
<span><span class="co">#&gt; [25] "chromatin"       "complex"         "Damage"         </span></span>
<span><span class="co">#&gt; [28] "function"        "induced"         "DNA"            </span></span>
<span><span class="co">#&gt; [31] "DDB2"            "Excision"        "Human"          </span></span>
<span><span class="co">#&gt; [34] "mice"            "Nucleotide"      "protein"        </span></span>
<span><span class="co">#&gt; [37] "regulates"       "level"           "METTL14"        </span></span>
<span><span class="co">#&gt; [40] "checkpoint"      "DNA"             "response"       </span></span>
<span><span class="co">#&gt; [43] "RAD17"           "Cancer"          "chromatin"      </span></span>
<span><span class="co">#&gt; [46] "complex"         "function"</span></span>
<span></span>
<span><span class="va">ddrRelated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/subgraph.html">induced.subgraph</a></span><span class="op">(</span><span class="va">conet</span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/V.html">V</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">ddrNms</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/V.html">V</a></span><span class="op">(</span><span class="va">ddrRelated</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/V.html">V</a></span><span class="op">(</span><span class="va">ddrRelated</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">keys</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,<span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html">ggraph</a></span><span class="op">(</span><span class="va">ddrRelated</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_diagonal.html">geom_edge_diagonal2</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"grey80"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html">geom_node_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">SYMBOL</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span>, color<span class="op">=</span><span class="va">SYMBOL</span><span class="op">)</span>,check_overlap<span class="op">=</span><span class="cn">TRUE</span>, repel<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                   bg.color <span class="op">=</span> <span class="st">"white"</span>, segment.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>                   bg.r <span class="op">=</span> <span class="fl">.15</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"steelblue"</span>,<span class="st">"tomato"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html">theme_graph</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/ddr-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The network can be obtained by <code>returnNet=TRUE</code>, or is stored in returned <code>biotext</code> class object in the slot <code>igraphRaw</code>, which can be used for downstream analysis like assessment of degrees and graph-based clustering.</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">conetDeg</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/degree.html">degree</a></span><span class="op">(</span><span class="va">conet</span><span class="op">)</span></span>
<span><span class="va">conetDeg</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">conetDeg</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span></span>
<span><span class="co">#&gt;      SESN1       DDB2        DNA      RAD17      Human </span></span>
<span><span class="co">#&gt;         26         21         20         18         15 </span></span>
<span><span class="co">#&gt;     Cancer checkpoint   response       TP53  regulates </span></span>
<span><span class="co">#&gt;         14         12         12         12         11 </span></span>
<span><span class="co">#&gt;    induced activation    protein     Repair     damage </span></span>
<span><span class="co">#&gt;         11         10         10         10          9</span></span>
<span></span>
<span><span class="co">## Graph-based clustering by igraph::cluster_fast_greedy()</span></span>
<span><span class="co">## Filling the slot `communities` and assign `igraphRaw` slot `community` attribute.</span></span>
<span><span class="va">btg</span> <span class="op">&lt;-</span> <span class="va">btg</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/graph_cluster.html">graph_cluster</a></span><span class="op">(</span>func<span class="op">=</span><span class="fu">igraph</span><span class="fu">::</span><span class="va"><a href="https://r.igraph.org/reference/cluster_fast_greedy.html">cluster_fast_greedy</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check the membership for merged network</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">btg</span><span class="op">@</span><span class="va">communities</span><span class="op">$</span><span class="va">membership</span>,</span>
<span>           <span class="va">btg</span><span class="op">@</span><span class="va">communities</span><span class="op">$</span><span class="va">names</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">`colnames&lt;-`</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"membership"</span>,<span class="st">"word"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">membership</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>words<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">word</span>,collapse<span class="op">=</span><span class="st">","</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<thead><tr>
<th style="text-align:right;">
membership
</th>
<th style="text-align:left;">
words
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
apoptosis,caspase,death,factor,TNFRSF10B,Apoptosis,carcinogenesis,Lung,promotes,Regulation,Roles
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
cellular,checkpoint,cycle,damage,DNA,phosphorylation,required,response,DDB2,RAD17,Cancer,degradation,Excision,Muscle,Nucleotide,protein,regulates,Repair,CDT2,chromatin,complex,Damage,function,Knockdown,level,METTL14,role
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
functions,growth,interacts,kinase,p53,suppressor,tumor,SESN1,TSC1,Human,Cisplatin,effects,HNSCC,increased,induced,levels,Macrophages,regulation,revealed,Study,Transcription,treatment
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
TP53,neuroblastoma,paediatric,Caspase,Death,ligand,Ligands,necroptotic,presence,Regulates,Transcriptional
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Carcinoma,epithelialmesenchymal,head,miR3773p,Neck,Squamous,Transition
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
DNAbinding,irradiation,mice,protects
</td>
</tr>
<tr>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
activation,density,inflammation,lipoproteininduced,macrophages,oxidized,Role,Sestrin1
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Necrosis,Regulated,RIPK1mediated
</td>
</tr>
<tr>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
activity,CASP8,inhibited
</td>
</tr>
<tr>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Intraflagellar,transport
</td>
</tr>
<tr>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
Dimerization,procaspase8
</td>
</tr>
<tr>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
acid,cyclin
</td>
</tr>
</tbody>
</table></div>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html">ggraph</a></span><span class="op">(</span><span class="va">btg</span><span class="op">@</span><span class="va">igraphRaw</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_link.html">geom_edge_link</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"grey80"</span>, width<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html">geom_node_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">community</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu">graphhighlight</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphhighlight/man/highlight_node.html">highlight_node</a></span><span class="op">(</span>glow<span class="op">=</span><span class="cn">TRUE</span>, filter<span class="op">=</span><span class="va">community</span><span class="op">==</span><span class="fl">2</span>,glow_base_size <span class="op">=</span> <span class="cn">TRUE</span>, glow_size<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span>, color<span class="op">=</span><span class="va">community</span><span class="op">)</span>,</span>
<span>                   check_overlap<span class="op">=</span><span class="cn">TRUE</span>, repel<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                   bg.color <span class="op">=</span> <span class="st">"white"</span>, segment.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>                   bg.r <span class="op">=</span> <span class="fl">.15</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html">theme_graph</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/combineNet-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>Dynamic layout can be also used to compare the networks, by <code>graphlayouts</code>, for comparing the multiple graphs, especially useful for time-series analysis. See the documentation of <a href="http://graphlayouts.schochastics.net/reference/layout_dynamic.html"><code>layout_as_dynamic</code></a> for specifying the alpha, which is default to 0.5.</p>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span></span>
<span><span class="va">dyn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotDynamic.html">plotDynamic</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">abstnet</span>, <span class="va">titlenet</span><span class="op">)</span>, concat<span class="op">=</span><span class="st">"union"</span>,</span>
<span>                   titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Abstract"</span>,<span class="st">"Title"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="va">dyn</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_BKPyV_files/figure-html/dyn-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 22621)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Japanese_Japan.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Japanese_Japan.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                   </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils    </span></span>
<span><span class="co">#&gt; [6] datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggforce_0.4.1             igraph_1.5.0             </span></span>
<span><span class="co">#&gt;  [3] ggraph_2.1.0.9000         clusterProfiler_4.7.1.003</span></span>
<span><span class="co">#&gt;  [5] ReactomePA_1.40.0         org.Hs.eg.db_3.15.0      </span></span>
<span><span class="co">#&gt;  [7] AnnotationDbi_1.60.2      IRanges_2.32.0           </span></span>
<span><span class="co">#&gt;  [9] S4Vectors_0.36.2          Biobase_2.58.0           </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.44.0       biotextgraph_0.99.0      </span></span>
<span><span class="co">#&gt; [13] ggplot2_3.4.2            </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] shadowtext_0.1.2       fastmatch_1.1-3       </span></span>
<span><span class="co">#&gt;   [3] plyr_1.8.8             lazyeval_0.2.2        </span></span>
<span><span class="co">#&gt;   [5] splines_4.2.1          BiocParallel_1.32.6   </span></span>
<span><span class="co">#&gt;   [7] GenomeInfoDb_1.34.9    digest_0.6.31         </span></span>
<span><span class="co">#&gt;   [9] yulab.utils_0.0.6      htmltools_0.5.4       </span></span>
<span><span class="co">#&gt;  [11] bugsigdbr_1.4.3        GOSemSim_2.24.0       </span></span>
<span><span class="co">#&gt;  [13] viridis_0.6.2          GO.db_3.16.0          </span></span>
<span><span class="co">#&gt;  [15] fansi_1.0.4            GeneSummary_0.99.4    </span></span>
<span><span class="co">#&gt;  [17] magrittr_2.0.3         memoise_2.0.1         </span></span>
<span><span class="co">#&gt;  [19] tm_0.7-11              Biostrings_2.66.0     </span></span>
<span><span class="co">#&gt;  [21] graphlayouts_0.8.4     pvclust_2.2-0         </span></span>
<span><span class="co">#&gt;  [23] wordcloud_2.6          enrichplot_1.18.3     </span></span>
<span><span class="co">#&gt;  [25] colorspace_2.1-0       rappdirs_0.3.3        </span></span>
<span><span class="co">#&gt;  [27] blob_1.2.4             ggrepel_0.9.3         </span></span>
<span><span class="co">#&gt;  [29] xfun_0.37              dplyr_1.1.1           </span></span>
<span><span class="co">#&gt;  [31] crayon_1.5.2           RCurl_1.98-1.10       </span></span>
<span><span class="co">#&gt;  [33] jsonlite_1.8.4         scatterpie_0.1.8      </span></span>
<span><span class="co">#&gt;  [35] graph_1.76.0           ape_5.7-1             </span></span>
<span><span class="co">#&gt;  [37] glue_1.6.2             polyclip_1.10-4       </span></span>
<span><span class="co">#&gt;  [39] gtable_0.3.3           zlibbioc_1.44.0       </span></span>
<span><span class="co">#&gt;  [41] XVector_0.38.0         GetoptLong_1.0.5      </span></span>
<span><span class="co">#&gt;  [43] graphite_1.42.0        rentrez_1.2.3         </span></span>
<span><span class="co">#&gt;  [45] scales_1.2.1           DOSE_3.24.2           </span></span>
<span><span class="co">#&gt;  [47] DBI_1.1.3              Rcpp_1.0.10           </span></span>
<span><span class="co">#&gt;  [49] viridisLite_0.4.1      xtable_1.8-4          </span></span>
<span><span class="co">#&gt;  [51] tidytree_0.4.2         gridGraphics_0.5-1    </span></span>
<span><span class="co">#&gt;  [53] reactome.db_1.81.0     bit_4.0.5             </span></span>
<span><span class="co">#&gt;  [55] htmlwidgets_1.6.2      httr_1.4.5            </span></span>
<span><span class="co">#&gt;  [57] fgsea_1.24.0           RColorBrewer_1.1-3    </span></span>
<span><span class="co">#&gt;  [59] ellipsis_0.3.2         pkgconfig_2.0.3       </span></span>
<span><span class="co">#&gt;  [61] XML_3.99-0.14          farver_2.1.1          </span></span>
<span><span class="co">#&gt;  [63] sass_0.4.5             utf8_1.2.3            </span></span>
<span><span class="co">#&gt;  [65] ggplotify_0.1.0        tidyselect_1.2.0      </span></span>
<span><span class="co">#&gt;  [67] rlang_1.1.0            reshape2_1.4.4        </span></span>
<span><span class="co">#&gt;  [69] later_1.3.0            munsell_0.5.0         </span></span>
<span><span class="co">#&gt;  [71] tools_4.2.1            cachem_1.0.7          </span></span>
<span><span class="co">#&gt;  [73] downloader_0.4         cli_3.6.1             </span></span>
<span><span class="co">#&gt;  [75] generics_0.1.3         RSQLite_2.3.0         </span></span>
<span><span class="co">#&gt;  [77] gson_0.1.0             evaluate_0.20         </span></span>
<span><span class="co">#&gt;  [79] stringr_1.5.0          fastmap_1.1.1         </span></span>
<span><span class="co">#&gt;  [81] ggdendro_0.1.23        yaml_2.3.7            </span></span>
<span><span class="co">#&gt;  [83] ggtree_3.6.2           knitr_1.42            </span></span>
<span><span class="co">#&gt;  [85] bit64_4.0.5            fs_1.6.1              </span></span>
<span><span class="co">#&gt;  [87] tidygraph_1.2.3        purrr_1.0.1           </span></span>
<span><span class="co">#&gt;  [89] KEGGREST_1.38.0        dendextend_1.16.0     </span></span>
<span><span class="co">#&gt;  [91] nlme_3.1-157           mime_0.12             </span></span>
<span><span class="co">#&gt;  [93] slam_0.1-50            aplot_0.1.10          </span></span>
<span><span class="co">#&gt;  [95] xml2_1.3.3             compiler_4.2.1        </span></span>
<span><span class="co">#&gt;  [97] rstudioapi_0.14        png_0.1-8             </span></span>
<span><span class="co">#&gt;  [99] treeio_1.22.0          tibble_3.2.1          </span></span>
<span><span class="co">#&gt; [101] tweenr_2.0.2           bslib_0.4.2           </span></span>
<span><span class="co">#&gt; [103] stringi_1.7.12         cyjShiny_1.0.34       </span></span>
<span><span class="co">#&gt; [105] lattice_0.20-45        Matrix_1.5-4          </span></span>
<span><span class="co">#&gt; [107] vctrs_0.6.1            pillar_1.9.0          </span></span>
<span><span class="co">#&gt; [109] lifecycle_1.0.3        jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt; [111] GlobalOptions_0.1.2    data.table_1.14.8     </span></span>
<span><span class="co">#&gt; [113] cowplot_1.1.1          bitops_1.0-7          </span></span>
<span><span class="co">#&gt; [115] httpuv_1.6.9           patchwork_1.1.2       </span></span>
<span><span class="co">#&gt; [117] qvalue_2.30.0          R6_2.5.1              </span></span>
<span><span class="co">#&gt; [119] bookdown_0.33          promises_1.2.0.1      </span></span>
<span><span class="co">#&gt; [121] gridExtra_2.3          codetools_0.2-18      </span></span>
<span><span class="co">#&gt; [123] MASS_7.3-57            rjson_0.2.21          </span></span>
<span><span class="co">#&gt; [125] withr_2.5.0            GenomeInfoDbData_1.2.9</span></span>
<span><span class="co">#&gt; [127] parallel_4.2.1         ggfun_0.0.9           </span></span>
<span><span class="co">#&gt; [129] grid_4.2.1             tidyr_1.3.0           </span></span>
<span><span class="co">#&gt; [131] HDO.db_0.99.1          rmarkdown_2.20        </span></span>
<span><span class="co">#&gt; [133] downlit_0.4.2          NLP_0.2-1             </span></span>
<span><span class="co">#&gt; [135] shiny_1.7.4            base64enc_0.1-3</span></span></code></pre></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="pathview-text.html"><span class="header-section-number">8</span> Pathview text</a></div>
<div class="next"><a href="single-cell-transcriptomic-data.html"><span class="header-section-number">10</span> Single-cell transcriptomic data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#app"><span class="header-section-number">9</span> Application examples</a></li>
<li><a class="nav-link" href="#load-the-necessary-packages-and-genes"><span class="header-section-number">9.1</span> Load the necessary packages and genes</a></li>
<li><a class="nav-link" href="#enrichment-analysis"><span class="header-section-number">9.2</span> Enrichment analysis</a></li>
<li><a class="nav-link" href="#text-mining-the-gene-summaries"><span class="header-section-number">9.3</span> Text mining the gene summaries</a></li>
<li><a class="nav-link" href="#text-mining-the-available-literature"><span class="header-section-number">9.4</span> Text mining the available literature</a></li>
<li><a class="nav-link" href="#combine-and-inspect-merged-network"><span class="header-section-number">9.5</span> Combine and inspect merged network</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>biotextgraph</strong>" was written by Noriaki Sato. It was last built on 2023-08-03.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
