<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 Application examples | biotextgraph</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.36 with bs4_book()">
<meta property="og:title" content="Chapter 9 Application examples | biotextgraph">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/biotextgraph/app.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 Application examples | biotextgraph">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-9FXY6YW4YZ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-9FXY6YW4YZ');
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="9.1 Actual application of the package  9.1.1 The selection of genes The input of genes can be selected in various ways. In most cases, the list obtained by differential expression analysis can be...">
<meta property="og:description" content="9.1 Actual application of the package  9.1.1 The selection of genes The input of genes can be selected in various ways. In most cases, the list obtained by differential expression analysis can be...">
<meta name="twitter:description" content="9.1 Actual application of the package  9.1.1 The selection of genes The input of genes can be selected in various ways. In most cases, the list obtained by differential expression analysis can be...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">biotextgraph</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="" href="metacyc.html"><span class="header-section-number">3</span> Microbial functional analysis</a></li>
<li><a class="" href="the-other-usage-for-microbiome-related-data.html"><span class="header-section-number">4</span> The other usage for microbiome-related data</a></li>
<li><a class="" href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">5</span> Interactive inspection of gene cluster network annotated by words</a></li>
<li><a class="" href="annotating-gene-cluster-dendrogram.html"><span class="header-section-number">6</span> Annotating gene cluster dendrogram</a></li>
<li><a class="" href="quantitative-analysis.html"><span class="header-section-number">7</span> Quantitative analysis</a></li>
<li><a class="" href="single-cell-transcriptomic-data.html"><span class="header-section-number">8</span> Single-cell transcriptomic data</a></li>
<li><a class="active" href="app.html"><span class="header-section-number">9</span> Application examples</a></li>
<li><a class="" href="parsing.html"><span class="header-section-number">10</span> Parsing</a></li>
<li><a class="" href="tidy.html"><span class="header-section-number">11</span> Perform each process separately</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="app" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> Application examples<a class="anchor" aria-label="anchor" href="#app"><i class="fas fa-link"></i></a>
</h1>
<div id="actual-application-of-the-package" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> Actual application of the package<a class="anchor" aria-label="anchor" href="#actual-application-of-the-package"><i class="fas fa-link"></i></a>
</h2>
<div id="the-selection-of-genes" class="section level3" number="9.1.1">
<h3>
<span class="header-section-number">9.1.1</span> The selection of genes<a class="anchor" aria-label="anchor" href="#the-selection-of-genes"><i class="fas fa-link"></i></a>
</h3>
<p>The input of genes can be selected in various ways. In most cases, the list obtained by differential expression analysis can be used as an input to the function. Additionally, the cluster of genes identified in gene clustering analysis such as weighted gene co-expression network analysis can be used as input.</p>
</div>
</div>
<div id="gene-cluster" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> Gene cluster<a class="anchor" aria-label="anchor" href="#gene-cluster"><i class="fas fa-link"></i></a>
</h2>
<div id="load-the-necessary-packages-and-genes" class="section level3" number="9.2.1">
<h3>
<span class="header-section-number">9.2.1</span> Load the necessary packages and genes<a class="anchor" aria-label="anchor" href="#load-the-necessary-packages-and-genes"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biotextgraph</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"biotextgraph"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>For the omics analysis involving transcript or genes, we would not obtain a list containing the single gene type such as ERCC genes shown in the basic usage example. In almost all the cases, the gene comes from various biological pathways like obtained in the analysis mentioned in the above section. Here, we introduce an example using the example cluster of WGCNA analysis obtained from the transcriptomic dataset investigating bladder cancer (<a href="https://doi.org/10.1038/s41556-019-0361-y">Chen et al. 2019</a>). The gene set contains 621 genes.</p>
<p>First, we perform over-representation analysis on the gene set (KEGG) to grasp the biological functions of these genes.</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">converted</span> <span class="op">&lt;-</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/bitr.html">bitr</a></span><span class="op">(</span><span class="va">exampleWGCNAcluster</span>,</span>
<span>    fromType<span class="op">=</span><span class="st">"ENSEMBL"</span>, toType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>    OrgDb<span class="op">=</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">ora</span> <span class="op">&lt;-</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichKEGG.html">enrichKEGG</a></span><span class="op">(</span><span class="va">converted</span><span class="op">)</span></span>
<span><span class="va">ora</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">p.adjust</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="st">"Description"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "p53 signaling pathway"                </span></span>
<span><span class="co">#&gt; [2] "Pathogenic Escherichia coli infection"</span></span>
<span><span class="co">#&gt; [3] "PI3K-Akt signaling pathway"           </span></span>
<span><span class="co">#&gt; [4] "ECM-receptor interaction"             </span></span>
<span><span class="co">#&gt; [5] "Small cell lung cancer"               </span></span>
<span><span class="co">#&gt; [6] "Lipid and atherosclerosis"            </span></span>
<span><span class="co">#&gt; [7] "Hepatocellular carcinoma"             </span></span>
<span><span class="co">#&gt; [8] "Influenza A"</span></span>
<span><span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">ora</span>, showCategory<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_files/figure-html/aa_kegg-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>We use the function in <code>biotextgraph</code> to make a summarized visualization of textual information, along with associated genes. As for the input reaching to a hundred of genes, there is an option <code>filterByGO</code> term, which filters the text mining results to those used in GO terms. This is useful for limiting the visualization to biologically-relevant terms.</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">check</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">converted</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                filterByGO<span class="op">=</span><span class="cn">TRUE</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span>, autoThresh<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 168</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; `filterByGO` option enabled. Filtering by GO terms ...</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">check</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_files/figure-html/aa_btg-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>Some genes are not related to the significantly enriched pathway, and one would like to inspect the biological function of these genes.</p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extraction of non-related genes</span></span>
<span><span class="va">enr_genes</span> <span class="op">&lt;-</span> <span class="va">ora</span><span class="op">@</span><span class="va">result</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">p.adjust</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">geneID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="st">"/"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">no_enr</span> <span class="op">&lt;-</span> <span class="va">converted</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">converted</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">enr_genes</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">no_enr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 135</span></span></code></pre></div>
<p>This time, we enable the <code>filterByGO</code> option along with <code>ngram=2</code>, which produces 2-gram visualization.</p>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">check_noenr_WGO</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">no_enr</span>,</span>
<span>    layout<span class="op">=</span><span class="st">"nicely"</span>,</span>
<span>    ngram<span class="op">=</span><span class="fl">2</span>,</span>
<span>    keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>    filterByGO<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    autoThresh<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    corThresh<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>    docsum<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 135</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; `filterByGO` option enabled. Filtering by GO terms ...</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">check_noenr_WGO</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_files/figure-html/aa_noenr-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>This way, we can filter the unnecessary words from the many terms included in the textual information of identifiers, and focus on biologically relevant terms contained in the identifiers.</p>
</div>
</div>
<div id="degs" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> DEGs<a class="anchor" aria-label="anchor" href="#degs"><i class="fas fa-link"></i></a>
</h2>
<p>We demonstrate an use case of the package, which investigates transcriptomic changes induced by BK polyomavirus (BKPyV) infection in renal proximal tubular epithelial cells (<a href="https://pubmed.ncbi.nlm.nih.gov/27381292/">Assetta et al. 2016</a>). Differentially expressed mRNAs in 3 days post-infection were obtained, and down-regulated mRNAs in BKPyV infected cells were examined.</p>
<div id="load-the-necessary-packages-and-genes-1" class="section level3" number="9.3.1">
<h3>
<span class="header-section-number">9.3.1</span> Load the necessary packages and genes<a class="anchor" aria-label="anchor" href="#load-the-necessary-packages-and-genes-1"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biotextgraph</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span>;<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggforce.data-imaginist.com">ggforce</a></span><span class="op">)</span> <span class="co">## For genePathPlot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"biotextgraph"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">degs</span> <span class="op">&lt;-</span> <span class="va">d3degDownAssetta2016</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">degs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 191</span></span>
<span><span class="va">degs</span></span>
<span><span class="co">#&gt;   [1] "ABCB4"           "ABCB7"           "AKTIP"          </span></span>
<span><span class="co">#&gt;   [4] "ALS2"            "ANKRA2"          "ANTXR1"         </span></span>
<span><span class="co">#&gt;   [7] "APH1B"           "ARHGEF28"        "ARNTL"          </span></span>
<span><span class="co">#&gt;  [10] "ATMIN"           "BDH2"            "BIRC2"          </span></span>
<span><span class="co">#&gt;  [13] "BTG1"            "BTG2"            "C4A"            </span></span>
<span><span class="co">#&gt;  [16] "C7orf60"         "C8orf4"          "CALCOCO1"       </span></span>
<span><span class="co">#&gt;  [19] "CAPRIN2"         "CARS"            "CBLB"           </span></span>
<span><span class="co">#&gt;  [22] "CCNDBP1"         "CCNG2"           "CCZ1B"          </span></span>
<span><span class="co">#&gt;  [25] "CDC42EP3"        "CDH6"            "CFLAR"          </span></span>
<span><span class="co">#&gt;  [28] "CHMP1B"          "CHMP4C"          "CLUAP1"         </span></span>
<span><span class="co">#&gt;  [31] "COG2"            "COG3"            "CPQ"            </span></span>
<span><span class="co">#&gt;  [34] "CROT"            "CTTNBP2NL"       "CYP4V2"         </span></span>
<span><span class="co">#&gt;  [37] "DAB2"            "DDB2"            "DDX17"          </span></span>
<span><span class="co">#&gt;  [40] "DDX5"            "DGKA"            "DHX32"          </span></span>
<span><span class="co">#&gt;  [43] "DLG1"            "DYNC2LI1"        "DYRK2"          </span></span>
<span><span class="co">#&gt;  [46] "EFHC1"           "EIF4A2"          "ERMAP"          </span></span>
<span><span class="co">#&gt;  [49] "ERMARD"          "EXOC1"           "FAM134B"        </span></span>
<span><span class="co">#&gt;  [52] "FAM160B1"        "FAM21C"          "FAM84B"         </span></span>
<span><span class="co">#&gt;  [55] "FANK1"           "FAS"             "FBXO38"         </span></span>
<span><span class="co">#&gt;  [58] "FCHO2"           "FGD6"            "FLJ22447"       </span></span>
<span><span class="co">#&gt;  [61] "FMNL2"           "GADD45A"         "GJA1"           </span></span>
<span><span class="co">#&gt;  [64] "GLIDR"           "GLT8D1"          "GOPC"           </span></span>
<span><span class="co">#&gt;  [67] "GPBP1L1"         "GPR155"          "GPR75-ASB3"     </span></span>
<span><span class="co">#&gt;  [70] "GRAMD3"          "HADHB"           "HCG11"          </span></span>
<span><span class="co">#&gt;  [73] "HDAC9"           "HDHD2"           "HERPUD1"        </span></span>
<span><span class="co">#&gt;  [76] "HSDL2"           "ICA1"            "ICK"            </span></span>
<span><span class="co">#&gt;  [79] "IFNGR1"          "IFT46"           "IFT74"          </span></span>
<span><span class="co">#&gt;  [82] "IRF6"            "ITGA2"           "ITGA6"          </span></span>
<span><span class="co">#&gt;  [85] "ITGAV"           "ITGB6"           "KDM5B"          </span></span>
<span><span class="co">#&gt;  [88] "KIF3A"           "KIF5B"           "KLHL20"         </span></span>
<span><span class="co">#&gt;  [91] "KLHL24"          "KLHL9"           "KPNA5"          </span></span>
<span><span class="co">#&gt;  [94] "KRCC1"           "L3MBTL3"         "LINC00657"      </span></span>
<span><span class="co">#&gt;  [97] "LOC100131564"    "LZTFL1"          "MAMDC2"         </span></span>
<span><span class="co">#&gt; [100] "MAP4K5"          "MAT2B"           "MBNL2"          </span></span>
<span><span class="co">#&gt; [103] "MDM2"            "MECOM"           "MFSD1"          </span></span>
<span><span class="co">#&gt; [106] "MGEA5"           "MICU3"           "MSANTD4"        </span></span>
<span><span class="co">#&gt; [109] "NBPF11"          "NCBP2"           "NEAT1"          </span></span>
<span><span class="co">#&gt; [112] "NTM"             "OGT"             "PAFAH1B2"       </span></span>
<span><span class="co">#&gt; [115] "PAFAH2"          "PCMTD2"          "PDE4D"          </span></span>
<span><span class="co">#&gt; [118] "PDP1"            "PELI1"           "PEX1"           </span></span>
<span><span class="co">#&gt; [121] "PHF14"           "PHOSPHO2-KLHL23" "PIK3IP1"        </span></span>
<span><span class="co">#&gt; [124] "PLA2R1"          "PLCB4"           "PLK2"           </span></span>
<span><span class="co">#&gt; [127] "POLI"            "POSTN"           "PPAN-P2RY11"    </span></span>
<span><span class="co">#&gt; [130] "PPFIBP1"         "PPP2CB"          "PRICKLE1"       </span></span>
<span><span class="co">#&gt; [133] "PROS1"           "PSMD5-AS1"       "RAD17"          </span></span>
<span><span class="co">#&gt; [136] "RHOQ"            "RIMKLB"          "RNA18S5"        </span></span>
<span><span class="co">#&gt; [139] "RNF170"          "RNF20"           "RNU1-28P"       </span></span>
<span><span class="co">#&gt; [142] "RPL23AP53"       "RRM2B"           "RRN3"           </span></span>
<span><span class="co">#&gt; [145] "RRN3P1"          "SEMA3C"          "SERINC1"        </span></span>
<span><span class="co">#&gt; [148] "SESN1"           "SESN3"           "SGK1"           </span></span>
<span><span class="co">#&gt; [151] "SLC22A5"         "SLC37A3"         "SVIL"           </span></span>
<span><span class="co">#&gt; [154] "SYT11"           "TARSL2"          "TBC1D19"        </span></span>
<span><span class="co">#&gt; [157] "TBCK"            "TBRG1"           "TGFA"           </span></span>
<span><span class="co">#&gt; [160] "TGFB2"           "TIPARP"          "TMEM136"        </span></span>
<span><span class="co">#&gt; [163] "TNFRSF10B"       "TNFRSF10D"       "TOM1L1"         </span></span>
<span><span class="co">#&gt; [166] "TP53INP1"        "TRIM13"          "TRIM32"         </span></span>
<span><span class="co">#&gt; [169] "TRIM4"           "TSC1"            "TSPYL5"         </span></span>
<span><span class="co">#&gt; [172] "UGT2B7"          "UNC13B"          "UPRT"           </span></span>
<span><span class="co">#&gt; [175] "VPS41"           "VPS8"            "WDR11"          </span></span>
<span><span class="co">#&gt; [178] "WDR19"           "XPC"             "YPEL2"          </span></span>
<span><span class="co">#&gt; [181] "ZC2HC1A"         "ZFAND5"          "ZFP90"          </span></span>
<span><span class="co">#&gt; [184] "ZFR"             "ZMAT3"           "ZNF12"          </span></span>
<span><span class="co">#&gt; [187] "ZNF248"          "ZNF322"          "ZNF561"         </span></span>
<span><span class="co">#&gt; [190] "ZNF626"          "ZSCAN30"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="enrichment-analysis" class="section level3" number="9.3.2">
<h3>
<span class="header-section-number">9.3.2</span> Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"><i class="fas fa-link"></i></a>
</h3>
<p>First, we perform enrichment analysis using ReactomePA.
From the enrichment analysis results, the cluster is related to transcriptional regulation by TP53.</p>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Convert to ENTREZID</span></span>
<span><span class="va">entre</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"SYMBOL"</span>,</span>
<span>                               keys <span class="op">=</span> <span class="va">degs</span>, columns <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span><span class="va">pway</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html">enrichPathway</a></span><span class="op">(</span><span class="va">entre</span><span class="op">)</span>, <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">sigpway</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">pway</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">p.adjust</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">sigpway</span><span class="op">$</span><span class="va">Description</span></span>
<span><span class="co">#&gt;  [1] "Transcriptional Regulation by TP53"                              </span></span>
<span><span class="co">#&gt;  [2] "RIPK1-mediated regulated necrosis"                               </span></span>
<span><span class="co">#&gt;  [3] "Regulation of necroptotic cell death"                            </span></span>
<span><span class="co">#&gt;  [4] "Regulated Necrosis"                                              </span></span>
<span><span class="co">#&gt;  [5] "Intraflagellar transport"                                        </span></span>
<span><span class="co">#&gt;  [6] "Regulation by c-FLIP"                                            </span></span>
<span><span class="co">#&gt;  [7] "CASP8 activity is inhibited"                                     </span></span>
<span><span class="co">#&gt;  [8] "Dimerization of procaspase-8"                                    </span></span>
<span><span class="co">#&gt;  [9] "TP53 Regulates Transcription of Death Receptors and Ligands"     </span></span>
<span><span class="co">#&gt; [10] "Caspase activation via Death Receptors in the presence of ligand"</span></span>
<span><span class="co">#&gt; [11] "FOXO-mediated transcription of cell cycle genes"                 </span></span>
<span><span class="co">#&gt; [12] "Cilium Assembly"                                                 </span></span>
<span><span class="co">#&gt; [13] "TP53 Regulates Transcription of Cell Death Genes"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">pway</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_files/figure-html/ea-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Genes involved in significant pathways</span></span>
<span><span class="va">excheck</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">sigpway</span><span class="op">$</span><span class="va">geneID</span>,</span>
<span>                                <span class="kw">function</span> <span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">x</span>,<span class="st">"/"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excheck</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 57</span></span>
<span></span>
<span><span class="co">## Not involved in significant pathways</span></span>
<span><span class="va">no_enr</span> <span class="op">&lt;-</span> <span class="va">degs</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">degs</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">excheck</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">no_enr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 159</span></span></code></pre></div>
</div>
<div id="text-mining-the-gene-summaries" class="section level3" number="9.3.3">
<h3>
<span class="header-section-number">9.3.3</span> Text mining the gene summaries<a class="anchor" aria-label="anchor" href="#text-mining-the-gene-summaries"><i class="fas fa-link"></i></a>
</h3>
<p>Next we perform the plain function producing a correlation network, with showing the top-genes related to high-frequency words in the text in RefSeq summary. We make two networks, those related to significantly enriched pathways and those not related to the significant pathways.</p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">excheck</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>, numWords<span class="op">=</span><span class="fl">50</span>, docsum<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 57</span></span>
<span><span class="co">#&gt;   Converted input genes: 32</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.3</span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">no_enr</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>, numWords<span class="op">=</span><span class="fl">50</span>, docsum<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 159</span></span>
<span><span class="co">#&gt;   Converted input genes: 142</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.1</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net2</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_files/figure-html/basic2app-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="combine-and-inspect-merged-network" class="section level3" number="9.3.4">
<h3>
<span class="header-section-number">9.3.4</span> Combine and inspect merged network<a class="anchor" aria-label="anchor" href="#combine-and-inspect-merged-network"><i class="fas fa-link"></i></a>
</h3>
<p>These networks can be combined to find intersections and differences, from textual perspective. We can see from the results that the genes related to non-enriched pathways could be associated with the membrane transport and adhesion, while genes in the enriched pathways have textual information resembling to those in curated biological databases (such as DNA damage response and tumor suppression).</p>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>,</span>
<span>               titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Enrich"</span>,<span class="st">"No_Enrich"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="06-application_example_files/figure-html/combineapp-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/local/package/r/4.3.0/lib64/R/lib/libRblas.so </span></span>
<span><span class="co">#&gt; LAPACK: /usr/local/package/r/4.3.0/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=ja_JP.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=ja_JP.UTF-8        LC_COLLATE=ja_JP.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=ja_JP.UTF-8    LC_MESSAGES=ja_JP.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=ja_JP.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=ja_JP.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Tokyo</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils    </span></span>
<span><span class="co">#&gt; [6] datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggforce_0.4.1         igraph_1.5.1         </span></span>
<span><span class="co">#&gt;  [3] ggraph_2.1.0          clusterProfiler_4.8.3</span></span>
<span><span class="co">#&gt;  [5] ReactomePA_1.44.0     org.Hs.eg.db_3.18.0  </span></span>
<span><span class="co">#&gt;  [7] AnnotationDbi_1.64.1  IRanges_2.34.1       </span></span>
<span><span class="co">#&gt;  [9] S4Vectors_0.38.1      Biobase_2.60.0       </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.46.0   biotextgraph_0.99.0  </span></span>
<span><span class="co">#&gt; [13] ggplot2_3.4.2        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3      ggdendro_0.1.23        </span></span>
<span><span class="co">#&gt;   [3] rstudioapi_0.15.0       jsonlite_1.8.7         </span></span>
<span><span class="co">#&gt;   [5] pvclust_2.2-0           magrittr_2.0.3         </span></span>
<span><span class="co">#&gt;   [7] farver_2.1.1            rmarkdown_2.23         </span></span>
<span><span class="co">#&gt;   [9] GlobalOptions_0.1.2     fs_1.6.3               </span></span>
<span><span class="co">#&gt;  [11] zlibbioc_1.46.0         vctrs_0.6.3            </span></span>
<span><span class="co">#&gt;  [13] memoise_2.0.1           cyjShiny_1.0.42        </span></span>
<span><span class="co">#&gt;  [15] RCurl_1.98-1.13         ggtree_3.8.2           </span></span>
<span><span class="co">#&gt;  [17] base64enc_0.1-3         htmltools_0.5.6        </span></span>
<span><span class="co">#&gt;  [19] gridGraphics_0.5-1      sass_0.4.7             </span></span>
<span><span class="co">#&gt;  [21] GeneSummary_0.99.6      bslib_0.5.1            </span></span>
<span><span class="co">#&gt;  [23] htmlwidgets_1.6.2       plyr_1.8.9             </span></span>
<span><span class="co">#&gt;  [25] cachem_1.0.8            mime_0.12              </span></span>
<span><span class="co">#&gt;  [27] lifecycle_1.0.3         pkgconfig_2.0.3        </span></span>
<span><span class="co">#&gt;  [29] gson_0.1.0              Matrix_1.6-3           </span></span>
<span><span class="co">#&gt;  [31] R6_2.5.1                fastmap_1.1.1          </span></span>
<span><span class="co">#&gt;  [33] GenomeInfoDbData_1.2.10 shiny_1.7.4.1          </span></span>
<span><span class="co">#&gt;  [35] aplot_0.2.2             enrichplot_1.20.3      </span></span>
<span><span class="co">#&gt;  [37] digest_0.6.33           colorspace_2.1-0       </span></span>
<span><span class="co">#&gt;  [39] patchwork_1.1.3         RSQLite_2.3.3          </span></span>
<span><span class="co">#&gt;  [41] labeling_0.4.2          fansi_1.0.4            </span></span>
<span><span class="co">#&gt;  [43] httr_1.4.6              polyclip_1.10-6        </span></span>
<span><span class="co">#&gt;  [45] compiler_4.3.0          downloader_0.4         </span></span>
<span><span class="co">#&gt;  [47] bit64_4.0.5             withr_2.5.0            </span></span>
<span><span class="co">#&gt;  [49] graphite_1.46.0         BiocParallel_1.34.2    </span></span>
<span><span class="co">#&gt;  [51] viridis_0.6.4           DBI_1.1.3              </span></span>
<span><span class="co">#&gt;  [53] highr_0.10              dendextend_1.17.1      </span></span>
<span><span class="co">#&gt;  [55] MASS_7.3-58.4           ISOcodes_2022.09.29    </span></span>
<span><span class="co">#&gt;  [57] rappdirs_0.3.3          rjson_0.2.21           </span></span>
<span><span class="co">#&gt;  [59] HDO.db_0.99.1           tools_4.3.0            </span></span>
<span><span class="co">#&gt;  [61] stopwords_2.3           scatterpie_0.2.1       </span></span>
<span><span class="co">#&gt;  [63] ape_5.7-1               rentrez_1.2.3          </span></span>
<span><span class="co">#&gt;  [65] httpuv_1.6.11           glue_1.6.2             </span></span>
<span><span class="co">#&gt;  [67] nlme_3.1-162            GOSemSim_2.26.1        </span></span>
<span><span class="co">#&gt;  [69] promises_1.2.1          grid_4.3.0             </span></span>
<span><span class="co">#&gt;  [71] shadowtext_0.1.2        reshape2_1.4.4         </span></span>
<span><span class="co">#&gt;  [73] fgsea_1.26.0            generics_0.1.3         </span></span>
<span><span class="co">#&gt;  [75] gtable_0.3.3            tidyr_1.3.0            </span></span>
<span><span class="co">#&gt;  [77] bugsigdbr_1.8.1         data.table_1.14.8      </span></span>
<span><span class="co">#&gt;  [79] tidygraph_1.2.3         xml2_1.3.5             </span></span>
<span><span class="co">#&gt;  [81] utf8_1.2.3              XVector_0.40.0         </span></span>
<span><span class="co">#&gt;  [83] stringr_1.5.0           ggrepel_0.9.4          </span></span>
<span><span class="co">#&gt;  [85] pillar_1.9.0            yulab.utils_0.1.0      </span></span>
<span><span class="co">#&gt;  [87] later_1.3.1             splines_4.3.0          </span></span>
<span><span class="co">#&gt;  [89] dplyr_1.1.2             tweenr_2.0.2           </span></span>
<span><span class="co">#&gt;  [91] treeio_1.24.3           lattice_0.21-8         </span></span>
<span><span class="co">#&gt;  [93] bit_4.0.5               tidyselect_1.2.0       </span></span>
<span><span class="co">#&gt;  [95] GO.db_3.17.0            tm_0.7-11              </span></span>
<span><span class="co">#&gt;  [97] Biostrings_2.68.1       reactome.db_1.84.0     </span></span>
<span><span class="co">#&gt;  [99] downlit_0.4.3           knitr_1.43             </span></span>
<span><span class="co">#&gt; [101] gridExtra_2.3           NLP_0.2-1              </span></span>
<span><span class="co">#&gt; [103] bookdown_0.36           xfun_0.40              </span></span>
<span><span class="co">#&gt; [105] graphlayouts_1.0.2      stringi_1.7.12         </span></span>
<span><span class="co">#&gt; [107] lazyeval_0.2.2          ggfun_0.1.3            </span></span>
<span><span class="co">#&gt; [109] yaml_2.3.7              evaluate_0.21          </span></span>
<span><span class="co">#&gt; [111] codetools_0.2-19        wordcloud_2.6          </span></span>
<span><span class="co">#&gt; [113] tibble_3.2.1            qvalue_2.32.0          </span></span>
<span><span class="co">#&gt; [115] graph_1.80.0            ggplotify_0.1.2        </span></span>
<span><span class="co">#&gt; [117] cli_3.6.1               xtable_1.8-4           </span></span>
<span><span class="co">#&gt; [119] munsell_0.5.0           jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [121] Rcpp_1.0.11             GenomeInfoDb_1.36.4    </span></span>
<span><span class="co">#&gt; [123] png_0.1-8               XML_3.99-0.15          </span></span>
<span><span class="co">#&gt; [125] parallel_4.3.0          ellipsis_0.3.2         </span></span>
<span><span class="co">#&gt; [127] blob_1.2.4              DOSE_3.26.2            </span></span>
<span><span class="co">#&gt; [129] bitops_1.0-7            tidytree_0.4.5         </span></span>
<span><span class="co">#&gt; [131] viridisLite_0.4.2       slam_0.1-50            </span></span>
<span><span class="co">#&gt; [133] scales_1.2.1            purrr_1.0.2            </span></span>
<span><span class="co">#&gt; [135] crayon_1.5.2            GetoptLong_1.0.5       </span></span>
<span><span class="co">#&gt; [137] rlang_1.1.1             cowplot_1.1.1          </span></span>
<span><span class="co">#&gt; [139] fastmatch_1.1-4         KEGGREST_1.40.1</span></span></code></pre></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="single-cell-transcriptomic-data.html"><span class="header-section-number">8</span> Single-cell transcriptomic data</a></div>
<div class="next"><a href="parsing.html"><span class="header-section-number">10</span> Parsing</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#app"><span class="header-section-number">9</span> Application examples</a></li>
<li>
<a class="nav-link" href="#actual-application-of-the-package"><span class="header-section-number">9.1</span> Actual application of the package</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#the-selection-of-genes"><span class="header-section-number">9.1.1</span> The selection of genes</a></li></ul>
</li>
<li>
<a class="nav-link" href="#gene-cluster"><span class="header-section-number">9.2</span> Gene cluster</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#load-the-necessary-packages-and-genes"><span class="header-section-number">9.2.1</span> Load the necessary packages and genes</a></li></ul>
</li>
<li>
<a class="nav-link" href="#degs"><span class="header-section-number">9.3</span> DEGs</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#load-the-necessary-packages-and-genes-1"><span class="header-section-number">9.3.1</span> Load the necessary packages and genes</a></li>
<li><a class="nav-link" href="#enrichment-analysis"><span class="header-section-number">9.3.2</span> Enrichment analysis</a></li>
<li><a class="nav-link" href="#text-mining-the-gene-summaries"><span class="header-section-number">9.3.3</span> Text mining the gene summaries</a></li>
<li><a class="nav-link" href="#combine-and-inspect-merged-network"><span class="header-section-number">9.3.4</span> Combine and inspect merged network</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>biotextgraph</strong>" was written by Noriaki Sato. It was last built on 2024-01-20.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
