<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 2 Basic usage | biotextgraph</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.35 with bs4_book()">
<meta property="og:title" content="Chapter 2 Basic usage | biotextgraph">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/biotextgraph/basic-usage.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 2 Basic usage | biotextgraph">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-9FXY6YW4YZ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-9FXY6YW4YZ');
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="Load the package and the database for converting identifiers. In this example, we use mostly human-derived data, and use org.Hs.eg.db. library(biotextgraph) library(org.Hs.eg.db) library(ggplot2)...">
<meta property="og:description" content="Load the package and the database for converting identifiers. In this example, we use mostly human-derived data, and use org.Hs.eg.db. library(biotextgraph) library(org.Hs.eg.db) library(ggplot2)...">
<meta name="twitter:description" content="Load the package and the database for converting identifiers. In this example, we use mostly human-derived data, and use org.Hs.eg.db. library(biotextgraph) library(org.Hs.eg.db) library(ggplot2)...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">biotextgraph</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="active" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="" href="metacyc.html"><span class="header-section-number">3</span> Microbial functional analysis</a></li>
<li><a class="" href="the-other-usage-for-microbiome-related-data.html"><span class="header-section-number">4</span> The other usage for microbiome-related data</a></li>
<li><a class="" href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">5</span> Interactive inspection of gene cluster network annotated by words</a></li>
<li><a class="" href="annotating-gene-cluster-dendrogram.html"><span class="header-section-number">6</span> Annotating gene cluster dendrogram</a></li>
<li><a class="" href="quantitative-analysis.html"><span class="header-section-number">7</span> Quantitative analysis</a></li>
<li><a class="" href="single-cell-transcriptomic-data.html"><span class="header-section-number">8</span> Single-cell transcriptomic data</a></li>
<li><a class="" href="app.html"><span class="header-section-number">9</span> Application examples</a></li>
<li><a class="" href="parsing.html"><span class="header-section-number">10</span> Parsing</a></li>
<li><a class="" href="tidy.html"><span class="header-section-number">11</span> Perform each process separately</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="basic-usage" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"><i class="fas fa-link"></i></a>
</h1>
<p>Load the package and the database for converting identifiers.
In this example, we use mostly human-derived data, and use <code>org.Hs.eg.db</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biotextgraph</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://talgalili.github.io/dendextend/">dendextend</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"biotextgraph"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="producing-networks" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Producing networks<a class="anchor" aria-label="anchor" href="#producing-networks"><i class="fas fa-link"></i></a>
</h2>
<p>The main function is producing networks between words based on the co-occurrence or correlation of the words in the text, given the list of gene identifiers. <code>refseq</code> is a function that imports gene text from RefSeq and summarizes the text information. Here, we use seven ERCC genes as input. This function returns a <code>biotext</code> class object, which contains various types of information. The <code>net</code> slot stores the <code>ggraph</code>, which represents the visualization result of the network.</p>
<p>The <code>plotNet</code> method plots the network. This can modify the options for the visualization. such as the layout and the coloring of the nodes and edges without querying the database again. The same method for the wordcloud is prepared as <code>plotWC</code>. The <code>plotNet</code> and <code>plotWC</code> method by default override preset visualization option, and this behaviour can be turned off by specifying <code>asis=TRUE</code>. We can obtain the attribute in the slot by the <code>getSlot</code> method.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Configure input genes</span></span>
<span><span class="va">inpSymbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ERCC1"</span>,<span class="st">"ERCC2"</span>,<span class="st">"ERCC3"</span>,<span class="st">"ERCC4"</span>,<span class="st">"ERCC5"</span>,<span class="st">"ERCC6"</span>,<span class="st">"ERCC8"</span><span class="op">)</span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="01-basic_usage_of_biotextgraph_files/figure-html/basic3-1.png" width="100%" style="display: block; margin: auto;">
By default, the gene ID type is set to <code>SYMBOL</code>. The other type can be set by <code>keyType</code>. As many of the words are commonly observed, filtering based on pre-computed word frequency on whole RefSeq data is provided. You should limit word frequency by <code>excludeFreq</code>, which is default to 2000. TF-IDF on the all the summary is also precomputed, and <code>exclude="tfidf"</code> can be specified too.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, excludeFreq<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 141 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.519</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/basic32-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>For visualization, The edge label corresponding to correlation or cooccurrence values can be shown by <code>edgeLabel=TRUE</code>. The number of words to be shown on plot can be specified by <code>numWords</code>. The threshold of correlation can be specified by <code>corThresh</code>. The visualized network layout can be specified by passing <code>layout</code> argument. The text color can be changed by <code>colorText=TRUE</code>. The type of edge can be specified by <code>edgeLink</code>, which is by default <code>TRUE</code> (link will be used).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    edgeLabel<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.4</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">20</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>, layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="01-basic_usage_of_biotextgraph_files/figure-html/ngramnet-1.png" width="100%" style="display: block; margin: auto;">
One of the main questions is which words can be clustered together among the words contained in the queried gene cluster. Word clustering (<code>pvclust</code>) and identified significant clusters based on the occurrence in the text can be visualized by specifying <code>tag="cor"</code>. For the significance threshold, <code>pvclAlpha</code> can be specified. The default parameters perform <code>pvclust</code> on the subset of dataset for words with high frequency specified by <code>numWords</code>. If one want to perform on whole matrix of TDM (which is a natural way), <code>tagWhole=TRUE</code> can be specified, although it is computationally intensive. One can pass clusters to perform parallel computing owning to <code>pvclust</code> function, by specifying <code>cl</code> as below. For tag coloring, <code>tagPalette</code> can be used. The <code>igraph</code> contained in the object can also be plotted by passing to <code>plot</code> method.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">20</span>, tag<span class="op">=</span><span class="st">"correlation"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/tagnet-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">net</span>, <span class="st">"pvpick"</span><span class="op">)</span></span>
<span><span class="co">#&gt; $clusters</span></span>
<span><span class="co">#&gt; $clusters[[1]]</span></span>
<span><span class="co">#&gt; [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[2]]</span></span>
<span><span class="co">#&gt;  [1] "activity"             "atpdependent"        </span></span>
<span><span class="co">#&gt;  [3] "complementation"      "defects"             </span></span>
<span><span class="co">#&gt;  [5] "factor"               "functions"           </span></span>
<span><span class="co">#&gt;  [7] "incision"             "interacts"           </span></span>
<span><span class="co">#&gt;  [9] "nucleotide"           "product"             </span></span>
<span><span class="co">#&gt; [11] "transcriptioncoupled"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[3]]</span></span>
<span><span class="co">#&gt; [1] "dna"    "repair"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $edges</span></span>
<span><span class="co">#&gt; [1]  2 12 13</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/tagnet-2.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="identifying-important-genes-in-the-network" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Identifying important genes in the network<a class="anchor" aria-label="anchor" href="#identifying-important-genes-in-the-network"><i class="fas fa-link"></i></a>
</h2>
<p>The other important aim is finding important genes (queries) in the network. The genes associated with the words can be shown by specifying <code>genePlot=TRUE</code>, useful for assessing which words is associated with interesting genes. The edges connecting words to corresponding genes are shown.
One can specify <code>genePlotNum</code> for limiting the genes shown by ranking of how often a gene is associated with the high-frequency words. This can be useful for identifying important genes among the network.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    genePlot<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"correlation"</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/gplot-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The associated enriched pathways (if present) can be shown by specifying <code>genePathPlot</code>, using <code>ggforce</code>. In this option, the function first performs over-representation analysis on the whole gene set, and plot enriched terms for included genes in the plot. Enrichment analysis here is performed by the library <code>clusterProfiler</code> or <code>ReactomePA</code>, and one can control which pathways to plot by <code>genePathPlotSig</code> value.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://joelgombin.github.io/concaveman/">concaveman</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggforce.data-imaginist.com">ggforce</a></span><span class="op">)</span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    genePathPlot<span class="op">=</span><span class="st">"reactome"</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"correlation"</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    genePathPlotSig<span class="op">=</span><span class="fl">0.05</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Found 48 enriched term</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/gpplot-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>By default, the associated genes are plotted without colorization (grey) and without nodes. If preferred, set <code>colorize=TRUE</code> to colorize the associated genes by <code>geneColor</code> and showing the nodes by adding pseudo-frequency corresponding to the minimum frequency of words in the network. In this way, color of nodes corresponding to words are shown by the gradient of frequency, and the queried genes are shown by <code>geneColor</code>. <code>addFreqToGene</code> also adds pseudo-frequency to gene nodes and show the node, but are colorized according to the specified minimum frequency.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    genePlot<span class="op">=</span><span class="cn">TRUE</span>, corThresh<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>    colorize<span class="op">=</span><span class="cn">TRUE</span>, geneColor<span class="op">=</span><span class="st">"pink"</span>,</span>
<span>    colorText<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"cor"</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/gplot2-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="producing-word-clouds" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Producing word clouds<a class="anchor" aria-label="anchor" href="#producing-word-clouds"><i class="fas fa-link"></i></a>
</h2>
<p>The other basic usage of the package is producing a word cloud of summaries of identifiers. Specify <code>plotType="wc"</code> or use <code><a href="https://rdrr.io/pkg/biotextgraph/man/plot_wordcloud.html">plot_wordcloud()</a></code> function for this purpose.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/obtain_refseq.html">obtain_refseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DDX41"</span>,<span class="st">"PNKP"</span>,<span class="st">"IRF3"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/make_corpus.html">make_corpus</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/make_TDM.html">make_TDM</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plot_wordcloud.html">plot_wordcloud</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 3</span></span>
<span><span class="co">#&gt;   Converted input genes: 3</span></span>
<span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"wc"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">gwc</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/basic-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>It accepts values of the <code>wordcloud()</code> function. <code>numWords</code> specifies how many words are to be shown on word cloud. The words are ordered by their frequency, and the subset of 1:<code>numWords</code> is used to downstream visualization. The arguments for wordcloud visualization must be passed to <code>argList</code> arguments as a list. <code>scaleFreq</code> can be specified to scale the frequency when the observation count is low.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>    scaleFreq<span class="op">=</span><span class="fl">2</span>,</span>
<span>    excludeFreq<span class="op">=</span><span class="fl">5000</span>,</span>
<span>    argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>        <span class="st">"random.order"</span><span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>        colors<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span>,</span>
<span>        <span class="st">"rot.per"</span><span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 30 words (frequency and/or tfidf)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">gwc</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/basic2-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>By default, <code>preserve=TRUE</code>, which indicates the funciton tries to preserve the original cases of characters. Note that if both the lower case words and capitalized words are present, all words are converted to capitalized words, like <code>damage</code> and <code>Damage</code> would be shown as <code>Damage</code> if <code>preserve=TRUE</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>    excludeFreq<span class="op">=</span><span class="fl">5000</span>,</span>
<span>    preserve<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>        rot.per<span class="op">=</span><span class="fl">0.4</span>,</span>
<span>        colors<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span>,</span>
<span>        random.order<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 30 words (frequency and/or tfidf)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">gwc_p</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/basic_pres_false-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>It also returns a data frame consisting of frequency of each term in the slot name <code>freqDf</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc</span></span>
<span><span class="co">#&gt; Type: refseq</span></span>
<span><span class="co">#&gt; Number of words: 100</span></span>
<span><span class="co">#&gt; Query: ERCC1/ERCC2/ERCC3/ERCC4/ERCC5/ERCC6/ERCC8</span></span>
<span><span class="co">#&gt; 182.1 Kb</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">gwc</span>, <span class="st">"freqDf"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    caption <span class="op">=</span> <span class="st">'Term frequencies.'</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:table">Table 2.1: </span>Term frequencies.</caption>
<thead><tr class="header">
<th align="left">word</th>
<th align="right">freq</th>
<th align="left">wcColor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">repair</td>
<td align="right">32</td>
<td align="left">black</td>
</tr>
<tr class="even">
<td align="left">DNA</td>
<td align="right">22</td>
<td align="left">black</td>
</tr>
<tr class="odd">
<td align="left">syndrome</td>
<td align="right">18</td>
<td align="left">black</td>
</tr>
<tr class="even">
<td align="left">excision</td>
<td align="right">16</td>
<td align="left">black</td>
</tr>
<tr class="odd">
<td align="left">transcription</td>
<td align="right">14</td>
<td align="left">black</td>
</tr>
<tr class="even">
<td align="left">cockayne</td>
<td align="right">12</td>
<td align="left">black</td>
</tr>
</tbody>
</table></div>
<p>N-gram is supported by library <code>tm</code>, specified by <code>ngram</code>.
Default is <code>1</code>, and the example specifying <code>2</code> is shown below.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>    ngram<span class="op">=</span><span class="fl">2</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>    argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>        rot.per<span class="op">=</span><span class="fl">0.4</span>,</span>
<span>        colors<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span>,</span>
<span>        random.order<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">gwc2</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/ngramwc-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>Using <code>clusterProfiler</code> functions, one can use enriched pathway names for visualization.
The <code>enrich</code> option can be specified for <code>'kegg'</code> or <code>'reactome'</code>, this time we specify <code>'reactome'</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwc3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>    enrich<span class="op">=</span><span class="st">"reactome"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Performing enrichment analysis</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">gwc3</span>, argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"min.freq"</span><span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/ngrampath-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>In the word cloud, it is also possible to visualize tag information with colors. In the example below, clustering was performed for all matrices, and the results were visualized based on the colors of <code>tagPalette</code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Prepare the palette for tag coloring</span></span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span> </span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="va">pal</span><span class="op">)</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">## Cluster on whole matrix</span></span>
<span><span class="va">gwclWhole</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"cor"</span>,</span>
<span>    tagPalette <span class="op">=</span> <span class="va">pal</span>,</span>
<span>    scaleFreq<span class="op">=</span><span class="fl">5</span>,</span>
<span>    cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>    argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">gwclWhole</span>, <span class="st">"pvpick"</span><span class="op">)</span></span>
<span><span class="co">#&gt; $clusters</span></span>
<span><span class="co">#&gt; $clusters[[1]]</span></span>
<span><span class="co">#&gt; [1] "pigmentosum" "xeroderma"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[2]]</span></span>
<span><span class="co">#&gt; [1] "dna"    "repair"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $clusters[[3]]</span></span>
<span><span class="co">#&gt;  [1] "abnormally"                "activates"                </span></span>
<span><span class="co">#&gt;  [3] "active"                    "activity"                 </span></span>
<span><span class="co">#&gt;  [5] "adjacent"                  "alter"                    </span></span>
<span><span class="co">#&gt;  [7] "atpase"                    "atpdependent"             </span></span>
<span><span class="co">#&gt;  [9] "atpstimulated"             "basal"                    </span></span>
<span><span class="co">#&gt; [11] "basic"                     "belongs"                  </span></span>
<span><span class="co">#&gt; [13] "bivm"                      "cerebrooculofacioskeletal"</span></span>
<span><span class="co">#&gt; [15] "characterized"             "complementation"          </span></span>
<span><span class="co">#&gt; [17] "damage"                    "defects"                  </span></span>
<span><span class="co">#&gt; [19] "exon"                      "factor"                   </span></span>
<span><span class="co">#&gt; [21] "forms"                     "functions"                </span></span>
<span><span class="co">#&gt; [23] "helicase"                  "heterodimeric"            </span></span>
<span><span class="co">#&gt; [25] "incision"                  "including"                </span></span>
<span><span class="co">#&gt; [27] "interacts"                 "light"                    </span></span>
<span><span class="co">#&gt; [29] "nucleotide"                "orf"                      </span></span>
<span><span class="co">#&gt; [31] "pathway"                   "piggybackderived3"        </span></span>
<span><span class="co">#&gt; [33] "product"                   "result"                   </span></span>
<span><span class="co">#&gt; [35] "skin"                      "specific"                 </span></span>
<span><span class="co">#&gt; [37] "splice"                    "transcriptioncoupled"     </span></span>
<span><span class="co">#&gt; [39] "trichothiodystrophy"       "type"                     </span></span>
<span><span class="co">#&gt; [41] "upstream"                 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $edges</span></span>
<span><span class="co">#&gt; [1] 16 41 43</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">gwclWhole</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/tagging-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>In this example querying ERCC genes, the term <code>DNA repair</code> is clustered as expected.</p>
<div id="plotting-associated-genes-in-wc" class="section level3" number="2.3.1">
<h3>
<span class="header-section-number">2.3.1</span> Plotting associated genes in WC<a class="anchor" aria-label="anchor" href="#plotting-associated-genes-in-wc"><i class="fas fa-link"></i></a>
</h3>
<p>By using the feature <code>label_content</code> in <code>ggwordcloud</code>, the user can choose to plot associated gene names along with the words, by <code>genePlot=TRUE</code>, as same as the network.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneplotWC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"cor"</span>,</span>
<span>    genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    tagPalette <span class="op">=</span> <span class="va">pal</span>,</span>
<span>    scaleFreq<span class="op">=</span><span class="fl">5</span>,</span>
<span>    cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>    argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>rot.per<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and</span></span>
<span><span class="co">#&gt; columns</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">geneplotWC</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in wordcloud_boxes(data_points =</span></span>
<span><span class="co">#&gt; points_valid_first, boxes = boxes, : One word could not fit</span></span>
<span><span class="co">#&gt; on page. It has been removed.</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/assoc_wc-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
</div>
<div id="visualization-of-pubmed-information." class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Visualization of PubMed information.<a class="anchor" aria-label="anchor" href="#visualization-of-pubmed-information."><i class="fas fa-link"></i></a>
</h2>
<p>Using <code>rentrez</code>, one can perform the same analysis on PubMed text like the article title and abstract. The function queries for the input gene symbols (or the other queries) and visualize. For typical use cases, the genes identified by showing <code>genePlot</code>, or hub genes identified in gene network analysis can be queried. The basic parameters for searching PubMed, like max number of articles retrieved and how to sort the articles can be specified by <code>retMax</code> and <code>sortOrder</code>. Be sure to obtain <a href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/">an api key</a> when querying heavily, and specify in <code>apiKey</code> argument. These functions including <code>pubmed</code> and <code>refseq</code> serve as wrappers for several other functions, but a detailed explanation can be found in <a href="tidy.html#tidy">11</a>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">pubmed</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, retMax<span class="op">=</span><span class="fl">20</span>, apiKey<span class="op">=</span><span class="va">apiKey</span>, plotType<span class="op">=</span><span class="st">"wc"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the</span></span>
<span><span class="co">#&gt; existing scale.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotWC</a></span><span class="op">(</span><span class="va">ab</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/pubmed_wc-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>The returned PubMed IDs are stored in <code>pmids</code> slot.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pmids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">ab</span>, <span class="st">"pmids"</span><span class="op">)</span></span>
<span><span class="va">pmids</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 20</span></span></code></pre></div>
<p>As fetching the same information is not desirable and time consuming, the same object can be passed to <code>redo</code> option and re-perform the analysis like tagging, or changing the visualization options. Also, using <code>obtain_pubmed</code> function, only the text can be obtained and be processed by the downstream functions.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abtag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">pubmed</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, tag<span class="op">=</span><span class="st">"cor"</span>, cl<span class="op">=</span><span class="fu">snow</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snow/man/snow-startstop.html">makeCluster</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>, apiKey<span class="op">=</span><span class="va">apiKey</span><span class="op">)</span></span>
<span><span class="co">#&gt; Resuming from the previous results</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.3</span></span>
<span><span class="va">abtag2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">pubmed</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">abtag</span>, tag<span class="op">=</span><span class="st">"cor"</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"network"</span>, corThresh<span class="op">=</span><span class="fl">0.2</span>, pre<span class="op">=</span><span class="cn">TRUE</span>, apiKey<span class="op">=</span><span class="va">apiKey</span><span class="op">)</span></span>
<span><span class="co">#&gt; Resuming from the previous results</span></span>
<span><span class="co">#&gt; Using previous pvclust resultsIgnoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.201</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">abtag2</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/pubmed_wc_2-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>If only the gene symbols are to be plotted, specify <code>onlyGene=TRUE</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">pubmed</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    onlyGene<span class="op">=</span><span class="cn">TRUE</span>, apiKey<span class="op">=</span><span class="va">apiKey</span>, numWords<span class="op">=</span><span class="fl">150</span><span class="op">)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.6</span></span>
<span><span class="co">#&gt; Subsetting to the gene symbol in orgDb</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/onlygene-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>It is sometimes the case that the results of the search is dominated by one well-investigated or well-characterized genes (such as TP53).
To overcome this, the <code>perQuery</code> option can be specified to search the database one by one. It is recommended to use API key for querying a large number of genes.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">netPerQuery</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">pubmed</a></span><span class="op">(</span><span class="va">inpSymbol</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, perQuery<span class="op">=</span><span class="cn">TRUE</span>, onlyGene<span class="op">=</span><span class="cn">TRUE</span>, apiKey<span class="op">=</span><span class="va">apiKey</span>, numWords<span class="op">=</span><span class="fl">150</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Querying ERCC1</span></span>
<span><span class="co">#&gt;   Querying ERCC2</span></span>
<span><span class="co">#&gt;   Querying ERCC3</span></span>
<span><span class="co">#&gt;   Querying ERCC4</span></span>
<span><span class="co">#&gt;   Querying ERCC5</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.3</span></span>
<span><span class="co">#&gt; Subsetting to the gene symbol in orgDb</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">netPerQuery</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/perquerytest-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="comparing-two-or-more-networks" class="section level2" number="2.5">
<h2>
<span class="header-section-number">2.5</span> Comparing two or more networks<a class="anchor" aria-label="anchor" href="#comparing-two-or-more-networks"><i class="fas fa-link"></i></a>
</h2>
<p>One can compare two or more networks by providing list of <code>biotext</code> objects produced by text mining various databases, like <code>refseq</code>, <code>pubmed</code>, <code>obtain_refseq</code>, etc. This can be useful for assessing the similarity and dissimilarity of the various text sources, like PubMed, RefSeq, and Reactome pathway names. Additionally, performing graph-based clustering on merged networks can potentially identify groups of related terms within the overall network.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">8</span>,<span class="fl">9</span>,<span class="fl">10</span>,<span class="fl">11</span>,<span class="fl">12</span>,<span class="fl">13</span>,<span class="fl">14</span>,<span class="fl">16</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">cxcls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">cxcls</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"CXCL"</span>,<span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">cxcls</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    corThresh<span class="op">=</span><span class="fl">0.5</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 13</span></span>
<span><span class="co">#&gt;   Converted input genes: 13</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.4</span></span>
<span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">pubmed</a></span><span class="op">(</span>redo<span class="op">=</span><span class="va">ab</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    corThresh<span class="op">=</span><span class="fl">0.2</span>, numWords<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Resuming from the previous results</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.3</span></span>
<span></span>
<span><span class="co">## Not having meaningful overlaps</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>,</span>
<span>    titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/compare-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net3</span><span class="op">)</span>,</span>
<span>    titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/compare-2.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span>, <span class="va">net3</span><span class="op">)</span>,</span>
<span>    titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ercc"</span>,<span class="st">"cxcl"</span>, <span class="st">"ercc-PubMed"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/compare-3.png" width="100%" style="display: block; margin: auto;"></div>
<p>If <code>tag</code> information is available in both gene clusters, combined tags can be visualized by <code>tag=TRUE</code>. If using <code>geom_mark_hull</code> to show the tagging information, specify <code>hull=TRUE</code> in <code>compareWordNet</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keggPathways</span> <span class="op">&lt;-</span> <span class="va">org.Hs.egPATH2EG</span></span>
<span><span class="va">mappedKeys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html">mappedkeys</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">)</span></span>
<span><span class="va">keggList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">keggPathways</span><span class="op">[</span><span class="va">mappedKeys</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04110`</span>,</span>
<span>    keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>    corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"cor"</span>,</span>
<span>    tfidf<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 124</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.1</span></span>
<span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04210`</span>,</span>
<span>    keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>    corThresh<span class="op">=</span><span class="fl">0.3</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">30</span>,</span>
<span>    tfidf<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    tag<span class="op">=</span><span class="st">"cor"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 87</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.1</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>, tag<span class="op">=</span><span class="cn">TRUE</span>, hull<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/tagcomp-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>An application example of using text mining for transcriptome analysis of BK polyomavirus infection by combining the functions described here <a href="app.html#app">9</a>.</p>
<div id="connecting-the-networks-and-identify-shared-genes" class="section level3" number="2.5.1">
<h3>
<span class="header-section-number">2.5.1</span> Connecting the networks and identify shared genes<a class="anchor" aria-label="anchor" href="#connecting-the-networks-and-identify-shared-genes"><i class="fas fa-link"></i></a>
</h3>
<p>If you specify <code>genePlot</code> (or equivalent) in calculating network in both networks, you can connect two networks based on specific words with showing relevant genes by using <code><a href="https://rdrr.io/pkg/biotextgraph/man/connectGenes.html">connectGenes()</a></code>. As it states genes in the title, it can be used with the other networks besides genes. By default, only the queried word is retrieved from the original network, and by specifying <code>neighbors</code>, it fecthes the neighbor words.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org">scales</a></span><span class="op">)</span></span>
<span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04110`</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span>, corThresh<span class="op">=</span><span class="fl">0.3</span>, numWords<span class="op">=</span><span class="fl">30</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 124</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.2</span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04114`</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span>, corThresh<span class="op">=</span><span class="fl">0.3</span>, numWords<span class="op">=</span><span class="fl">30</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 112</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.2</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/connectGenes.html">connectGenes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"4110"</span><span class="op">=</span><span class="va">net1</span>, <span class="st">"4114"</span><span class="op">=</span><span class="va">net2</span><span class="op">)</span>, <span class="st">"cycle"</span>, neighbor<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tidygraph</span><span class="fu">::</span><span class="fu"><a href="https://tidygraph.data-imaginist.com/reference/tbl_graph.html">as_tbl_graph</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>SYMBOL<span class="op">=</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">keys</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="st">"SYMBOL"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="co">## Highlighy symbol</span></span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html">ggraph</a></span><span class="op">(</span>layout<span class="op">=</span><span class="st">"nicely"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_diagonal.html">geom_edge_diagonal</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"grey"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html">geom_node_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>filter<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2</span>, color<span class="op">=</span><span class="fu"><a href="https://scales.r-lib.org/reference/muted.html">muted</a></span><span class="op">(</span><span class="st">"red"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu">graphhighlight</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphhighlight/man/highlight_node.html">highlight_node</a></span><span class="op">(</span>glow<span class="op">=</span><span class="cn">TRUE</span>, filter<span class="op">=</span><span class="va">SYMBOL</span>,</span>
<span>        glow_base_size <span class="op">=</span> <span class="cn">TRUE</span>, glow_size<span class="op">=</span><span class="fl">0.5</span>,highlight_color <span class="op">=</span> <span class="fu"><a href="https://scales.r-lib.org/reference/muted.html">muted</a></span><span class="op">(</span><span class="st">"red"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html">geom_node_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>filter<span class="op">=</span><span class="op">!</span><span class="va">.data</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span>, filter<span class="op">=</span><span class="va">name</span><span class="op">!=</span><span class="st">"cycle"</span><span class="op">)</span>, repel<span class="op">=</span><span class="cn">TRUE</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span>, filter<span class="op">=</span><span class="va">name</span><span class="op">==</span><span class="st">"cycle"</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">6</span>, repel<span class="op">=</span><span class="cn">TRUE</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html">theme_graph</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/conn-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
</div>
<div id="text-over-represenatation-analysis-experimental" class="section level2" number="2.6">
<h2>
<span class="header-section-number">2.6</span> Text over represenatation analysis (experimental)<a class="anchor" aria-label="anchor" href="#text-over-represenatation-analysis-experimental"><i class="fas fa-link"></i></a>
</h2>
<p>For RefSeq, overrepresentation-based filtering and prioritization of words can be performed using pre-computed background.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`00785`</span> <span class="co"># Lipoic acid metabolism</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/ora-1.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;            lipoic              step              acid </span></span>
<span><span class="co">#&gt;      1.600963e-14      8.016934e-08      5.300425e-07 </span></span>
<span><span class="co">#&gt; lipoateactivating  lipoatedependent            lipoyl </span></span>
<span><span class="co">#&gt;      1.484931e-04      1.484931e-04      1.484931e-04</span></span>
<span></span>
<span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">keggList</span><span class="op">$</span><span class="va">`05150`</span> <span class="co"># Staphylococcus aureus infection</span></span>
<span><span class="va">pvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/textORA.html">textORA</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/ora-2.png" width="100%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pvs</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;          drb        chain   complement        class </span></span>
<span><span class="co">#&gt; 0.000000e+00 3.669590e-82 3.266180e-74 5.496543e-46 </span></span>
<span><span class="co">#&gt;         exon         beta </span></span>
<span><span class="co">#&gt; 6.851722e-39 1.318324e-34</span></span></code></pre></div>
<p>This thresholding can be used in RefSeq visualization in the above functions.
Filter words using ORA threshold and frequency threshold by setting <code>ora=TRUE</code>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>                     ora<span class="op">=</span><span class="cn">TRUE</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt;   Converted input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Performing ORA</span></span>
<span><span class="co">#&gt; Filtered 148 words (ORA)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/oranet-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>After obtaining the ORA results, one can plot volcano-plot like plot for the results using <code>plotORA</code> function.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggrepel.slowkow.com/">ggrepel</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotORA.html">plotORA</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/oraplot-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="dependency-analyis-using-udpipe" class="section level2" number="2.7">
<h2>
<span class="header-section-number">2.7</span> Dependency analyis using udpipe<a class="anchor" aria-label="anchor" href="#dependency-analyis-using-udpipe"><i class="fas fa-link"></i></a>
</h2>
<p>Using <a href="https://github.com/bnosac/udpipe"><code>udpipe</code></a> package (<a href="https://aclanthology.org/K17-3009/">Straka and Straková. 2017</a>), one can performe dependency analysis of texts in various databases. Set <code>useUdpipe</code> to <code>TRUE</code>, and specify downloaded model to be used in <code>udpipeModel</code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">biotextgraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DDX41"</span>,<span class="st">"PNKP"</span>,<span class="st">"ERCC2"</span><span class="op">)</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"network"</span>, useUdpipe<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    udpipeModel<span class="op">=</span><span class="st">"english-ewt-ud-2.5-191206.udpipe"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using udpipe mode</span></span>
<span><span class="co">#&gt; Input genes: 3</span></span>
<span><span class="co">#&gt;   Converted input genes: 3</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">p</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/udpipe-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="changing-the-font" class="section level2" number="2.8">
<h2>
<span class="header-section-number">2.8</span> Changing the font<a class="anchor" aria-label="anchor" href="#changing-the-font"><i class="fas fa-link"></i></a>
</h2>
<p>Font can be specified by <code>fontFamily</code>, which affects all the labels including edges and nodes.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sysdata.rda"</span>, package <span class="op">=</span> <span class="st">"biotextgraph"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">degs</span> <span class="op">&lt;-</span> <span class="va">d3degUpAssetta2016</span></span>
<span><span class="co">## Use alien encounter fonts (http://www.hipsthetic.com/alien-encounters-free-80s-font-family/)</span></span>
<span><span class="fu">sysfonts</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sysfonts/man/font_add.html">font_add</a></span><span class="op">(</span>family<span class="op">=</span><span class="st">"alien"</span>,regular<span class="op">=</span><span class="st">"SFAlienEncounters.ttf"</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">biotextgraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">degs</span>,</span>
<span>    plotType<span class="op">=</span><span class="st">"network"</span>,</span>
<span>    numWords<span class="op">=</span><span class="fl">50</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>    fontFamily<span class="op">=</span><span class="st">"alien"</span>, autoThresh<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>    colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 636</span></span>
<span><span class="co">#&gt;   Converted input genes: 552</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span><span class="va">p</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/font-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>For complex networks, changing the layout is possible by the small function <code>changeLayout</code>.
Specify which layout algorithm to choose in <code>igraph</code>. This can be also accomplished by specifying the layout in <code>plotNet</code> method.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/changeLayout.html">changeLayout</a></span><span class="op">(</span><span class="va">p</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="va"><a href="https://r.igraph.org/reference/layout_with_graphopt.html">layout.graphopt</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span>asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/clay-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="biofabric-layout" class="section level2" number="2.9">
<h2>
<span class="header-section-number">2.9</span> Biofabric layout<a class="anchor" aria-label="anchor" href="#biofabric-layout"><i class="fas fa-link"></i></a>
</h2>
<p>It is possible to use the <code>BioFabric</code> layout from the extracted <code>igraph</code>. In this case, one wants to add text position in x-axis using <code><a href="https://rdrr.io/pkg/biotextgraph/man/obtainTextPosition.html">biotextgraph::obtainTextPosition</a></code>. For the beautiful <code>BioFabric</code> layout, please refer to their original homepage <a href="https://biofabric.systemsbiology.net/">here</a>, and their publication available <a href="https://doi.org/10.1186/1471-2105-13-275">here</a> (Longabaugh, W.J. 2012).</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DDX41"</span>,<span class="st">"PNKP"</span><span class="op">)</span>, autoThresh<span class="op">=</span><span class="cn">FALSE</span>, genePlot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt;   Converted input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="va">ex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">ex</span>, <span class="st">"igraph"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/obtainTextPosition.html">obtainTextPosition</a></span><span class="op">(</span><span class="op">)</span> <span class="co">## Obtain text position for the X-axis</span></span>
<span><span class="va">ex</span> <span class="op">|&gt;</span> </span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html">ggraph</a></span><span class="op">(</span><span class="st">"fabric"</span>,</span>
<span>  sort.by <span class="op">=</span> <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/layout_tbl_graph_fabric.html">node_rank_fabric</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_range.html">geom_node_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_span.html">geom_edge_span</a></span><span class="op">(</span>end_shape <span class="op">=</span> <span class="st">'circle'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">xmin</span><span class="op">-</span><span class="fl">4</span>, label<span class="op">=</span><span class="va">name</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2.5</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu">shadowtext</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shadowtext/man/geom_shadowtext.html">geom_shadowtext</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">center</span>,</span>
<span>            color<span class="op">=</span><span class="va">nodeCat</span>, y<span class="op">=</span><span class="va">y</span><span class="op">+</span><span class="fl">1</span>, label<span class="op">=</span><span class="va">name</span><span class="op">)</span>,</span>
<span>        size<span class="op">=</span><span class="fl">2.5</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey20"</span>, <span class="st">"grey80"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html">theme_graph</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/biof-1.png" width="100%" style="display: block; margin: auto;"></div>
<div id="highlighting-genes-in-the-layout" class="section level3" number="2.9.1">
<h3>
<span class="header-section-number">2.9.1</span> Highlighting genes in the layout<a class="anchor" aria-label="anchor" href="#highlighting-genes-in-the-layout"><i class="fas fa-link"></i></a>
</h3>
<p>For more practical use, we highlight gene names in the layout using <code>ggfx</code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggfx.data-imaginist.com">ggfx</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use sample DEGs</span></span>
<span><span class="va">ex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">d3degUpAssetta2016</span>,</span>
<span>                              genePlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                              numWords<span class="op">=</span><span class="fl">50</span>,</span>
<span>                              preserve<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                              autoThresh<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 636</span></span>
<span><span class="co">#&gt;   Converted input genes: 552</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span></span>
<span><span class="co"># Obtain text positions, and plot using various geoms.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">ex</span>, <span class="st">"igraph"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/obtainTextPosition.html">obtainTextPosition</a></span><span class="op">(</span>verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>  <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html">ggraph</a></span><span class="op">(</span><span class="st">"fabric"</span>, sort.by <span class="op">=</span> <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/layout_tbl_graph_fabric.html">node_rank_fabric</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_range.html">geom_node_range</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">nodeCat</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_span.html">geom_edge_span</a></span><span class="op">(</span>end_shape <span class="op">=</span> <span class="st">'circle'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggfx.data-imaginist.com/reference/with_outer_glow.html">with_outer_glow</a></span><span class="op">(</span><span class="fu">ggkegg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggkegg/man/geom_node_shadowtext.html">geom_node_shadowtext</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">xmin</span><span class="op">-</span><span class="fl">4</span>, label<span class="op">=</span><span class="va">name</span>, filter<span class="op">=</span><span class="va">type</span><span class="op">==</span><span class="st">"Genes"</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2.5</span>,</span>
<span>                                                 color<span class="op">=</span><span class="st">"tomato"</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span>, <span class="st">"gold"</span>,expand<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">+</span> <span class="co">## Show genes (shadowtext + ggfx)</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_text.html">geom_node_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">xmin</span><span class="op">-</span><span class="fl">4</span>, label<span class="op">=</span><span class="va">name</span>, filter<span class="op">=</span><span class="va">type</span><span class="op">!=</span><span class="st">"Genes"</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu">ggkegg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggkegg/man/geom_node_shadowtext.html">geom_node_shadowtext</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">center</span>, size<span class="op">=</span><span class="va">xmin</span>, filter<span class="op">=</span><span class="va">type</span><span class="op">!=</span><span class="st">"Genes"</span>, color<span class="op">=</span><span class="va">nodeCat</span>, y<span class="op">=</span><span class="va">y</span><span class="op">+</span><span class="fl">1</span>, label<span class="op">=</span><span class="va">name</span><span class="op">)</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggfx.data-imaginist.com/reference/with_outer_glow.html">with_outer_glow</a></span><span class="op">(</span><span class="fu">ggkegg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggkegg/man/geom_node_shadowtext.html">geom_node_shadowtext</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">center</span>, filter<span class="op">=</span><span class="va">type</span><span class="op">==</span><span class="st">"Genes"</span>, color<span class="op">=</span><span class="va">nodeCat</span>, y<span class="op">=</span><span class="va">y</span><span class="op">+</span><span class="fl">1</span>, label<span class="op">=</span><span class="va">name</span><span class="op">)</span>,</span>
<span>                                                 size<span class="op">=</span><span class="fl">2.5</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span>,<span class="st">"gold"</span>,expand<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tomato"</span>, <span class="st">"grey20"</span><span class="op">)</span>,name<span class="op">=</span><span class="st">"Category"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html">theme_graph</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="st">'reverse'</span>, range<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1.5</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>size<span class="op">=</span><span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/biof2-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="biofabric-layouts-for-combined-networks" class="section level3" number="2.9.2">
<h3>
<span class="header-section-number">2.9.2</span> Biofabric layouts for combined networks<a class="anchor" aria-label="anchor" href="#biofabric-layouts-for-combined-networks"><i class="fas fa-link"></i></a>
</h3>
<p>Can be used in cojuction with <code><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet()</a></code>. The BioFabric layout is especially useful for visualizing complex networks. For interactively inspecting complex networks, please refer to the section <a href="the-other-usage-for-microbiome-related-data.html#interactive">4.3.1</a>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04110`</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>               corThresh<span class="op">=</span><span class="fl">0.3</span>, numWords<span class="op">=</span><span class="fl">30</span>, autoThresh<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 124</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">keggList</span><span class="op">$</span><span class="va">`04114`</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span>,</span>
<span>               corThresh<span class="op">=</span><span class="fl">0.3</span>, numWords<span class="op">=</span><span class="fl">30</span>, autoThresh<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 112</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/compareWordNet.html">compareWordNet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>,titles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"4110"</span>,<span class="st">"4114"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Grouping is stored in `col` variable of nodes</span></span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html">ggraph</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/getSlot.html">getSlot</a></span><span class="op">(</span><span class="va">comp</span>, <span class="st">"igraphRaw"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/obtainTextPosition.html">obtainTextPosition</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"fabric"</span>,sort.by<span class="op">=</span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/layout_tbl_graph_fabric.html">node_rank_fabric</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_range.html">geom_node_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_span.html">geom_edge_span</a></span><span class="op">(</span>end_shape <span class="op">=</span> <span class="st">'circle'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/geom_node_shadowtext.html">geom_node_shadowtext</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">xmin</span><span class="op">-</span><span class="fl">4</span>,</span>
<span>        label<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"grey20"</span>,size<span class="op">=</span><span class="fl">2</span>, bg.colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/geom_node_shadowtext.html">geom_node_shadowtext</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">center</span>,</span>
<span>        y<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">y</span><span class="op">+</span><span class="fl">1</span>, label<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">name</span>, color<span class="op">=</span><span class="va">col</span><span class="op">)</span>, bg.colour<span class="op">=</span><span class="st">"white"</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html">theme_graph</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/fabriccomb-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="wrapper-for-biofabric-layout" class="section level3" number="2.9.3">
<h3>
<span class="header-section-number">2.9.3</span> Wrapper for <code>BioFabric</code> layout<a class="anchor" aria-label="anchor" href="#wrapper-for-biofabric-layout"><i class="fas fa-link"></i></a>
</h3>
<p>The funciton <code>plot_biofabric</code> is prepared, which accepts <code>biotext</code> class object.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plot_biofabric.html">plot_biofabric</a></span><span class="op">(</span>end_shape<span class="op">=</span><span class="st">"square"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/pbfbrc-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
</div>
<div id="dynamic-layout" class="section level2" number="2.10">
<h2>
<span class="header-section-number">2.10</span> Dynamic layout<a class="anchor" aria-label="anchor" href="#dynamic-layout"><i class="fas fa-link"></i></a>
</h2>
<p>Dynamic layout can be also used to compare the networks, by <code>graphlayouts</code>, for comparing the multiple graphs, especially useful for time-series analysis. See the documentation of <a href="http://graphlayouts.schochastics.net/reference/layout_dynamic.html"><code>layout_as_dynamic</code></a> for specifying the alpha, which is default to 0.5.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span></span>
<span><span class="va">dyn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotDynamic.html">plotDynamic</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span><span class="op">)</span>, concat<span class="op">=</span><span class="st">"union"</span><span class="op">)</span></span>
<span><span class="va">dyn</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/dyn-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="input-type" class="section level2" number="2.11">
<h2>
<span class="header-section-number">2.11</span> Input type<a class="anchor" aria-label="anchor" href="#input-type"><i class="fas fa-link"></i></a>
</h2>
<p>Some input types can be used to conveniently perform text mining. <code>refseqWGCNA</code> accepts <code>blockwiseModules()</code> results from <code>WGCNA</code>, and recursively returns the biotext class object for each of the cluster.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"./blockwiseModule.rda"</span><span class="op">)</span></span>
<span><span class="va">test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/refseqWGCNA.html">refseqWGCNA</a></span><span class="op">(</span><span class="va">bwmod</span><span class="op">)</span></span>
<span><span class="co">#&gt; Processing a total of 17 clusters</span></span>
<span><span class="co">#&gt; Input genes: 1066</span></span>
<span><span class="co">#&gt;   Converted input genes: 885</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.001</span></span>
<span><span class="co">#&gt; Input genes: 2396</span></span>
<span><span class="co">#&gt;   Converted input genes: 2199</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0</span></span>
<span><span class="co">#&gt; Input genes: 12526</span></span>
<span><span class="co">#&gt;   Converted input genes: 9512</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0</span></span>
<span><span class="co">#&gt; Input genes: 5586</span></span>
<span><span class="co">#&gt;   Converted input genes: 4595</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0</span></span>
<span><span class="co">#&gt; Input genes: 181</span></span>
<span><span class="co">#&gt;   Converted input genes: 168</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.101</span></span>
<span><span class="co">#&gt; Input genes: 379</span></span>
<span><span class="co">#&gt;   Converted input genes: 338</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.1</span></span>
<span><span class="co">#&gt; Input genes: 774</span></span>
<span><span class="co">#&gt;   Converted input genes: 626</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.104</span></span>
<span><span class="co">#&gt; Input genes: 147</span></span>
<span><span class="co">#&gt;   Converted input genes: 126</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.101</span></span>
<span><span class="co">#&gt; Input genes: 87</span></span>
<span><span class="co">#&gt;   Converted input genes: 85</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.201</span></span>
<span><span class="co">#&gt; Input genes: 94</span></span>
<span><span class="co">#&gt;   Converted input genes: 92</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.102</span></span>
<span><span class="co">#&gt; Input genes: 113</span></span>
<span><span class="co">#&gt;   Converted input genes: 104</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.101</span></span>
<span><span class="co">#&gt; Input genes: 83</span></span>
<span><span class="co">#&gt;   Converted input genes: 77</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.101</span></span>
<span><span class="co">#&gt; Input genes: 233</span></span>
<span><span class="co">#&gt;   Converted input genes: 141</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.3</span></span>
<span><span class="co">#&gt; Input genes: 68</span></span>
<span><span class="co">#&gt;   Converted input genes: 67</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.2</span></span>
<span><span class="co">#&gt; Input genes: 45</span></span>
<span><span class="co">#&gt;   Converted input genes: 45</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0</span></span>
<span><span class="co">#&gt; Input genes: 86</span></span>
<span><span class="co">#&gt;   Converted input genes: 47</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.2</span></span>
<span><span class="co">#&gt; Input genes: 36</span></span>
<span><span class="co">#&gt;   Converted input genes: 9</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Error in cor(x = c(`2024` = 1, `70nt` = 1, antisense = 1, base = 1, capped = 1,  : </span></span>
<span><span class="co">#&gt;   supply both 'x' and 'y' or a matrix-like 'x'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">test</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 17</span></span>
<span><span class="va">test</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">net</span> <span class="op">+</span> <span class="va">test</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">net</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/refseqwgcna-1.png" width="100%" style="display: block; margin: auto;"></div>
<p><code>refseqDESeq2</code> accepts the resulting object from <code><a href="https://rdrr.io/pkg/DESeq2/man/results.html">DESeq2::results()</a></code> function, with specified criteria such as LFC and p-values and returns the biotext class object like as follows.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2">DESeq2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/makeExampleDESeqDataSet.html">makeExampleDESeqDataSet</a></span><span class="op">(</span>m<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds</span>, contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">"B"</span>,<span class="st">"A"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">keys</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">res</span></span>
<span><span class="co">#&gt; log2 fold change (MLE): condition B vs A </span></span>
<span><span class="co">#&gt; Wald test p-value: condition B vs A </span></span>
<span><span class="co">#&gt; DataFrame with 1000 rows and 6 columns</span></span>
<span><span class="co">#&gt;                  baseMean log2FoldChange     lfcSE</span></span>
<span><span class="co">#&gt;                 &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000173801 30.616784      0.9499853  0.794602</span></span>
<span><span class="co">#&gt; ENSG00000265789 21.514495      0.1038858  1.024781</span></span>
<span><span class="co">#&gt; ENSG00000249531  0.750854      3.0643242  4.855112</span></span>
<span><span class="co">#&gt; ENSG00000202358 51.533085      0.0681512  0.666996</span></span>
<span><span class="co">#&gt; ENSG00000119912  6.392006     -0.6407534  1.733944</span></span>
<span><span class="co">#&gt; ...                   ...            ...       ...</span></span>
<span><span class="co">#&gt; ENSG00000274518  88.69205     -0.0515992  0.581795</span></span>
<span><span class="co">#&gt; ENSG00000104154  11.28361      0.3631596  1.243822</span></span>
<span><span class="co">#&gt; ENSG00000148841  13.49360     -0.2693230  1.104549</span></span>
<span><span class="co">#&gt; ENSG00000202283   5.38738     -0.2669877  1.608832</span></span>
<span><span class="co">#&gt; ENSG00000106261  41.40248      0.2672459  0.689372</span></span>
<span><span class="co">#&gt;                       stat    pvalue      padj</span></span>
<span><span class="co">#&gt;                  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000173801   1.195548  0.231873  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000265789   0.101374  0.919254  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000249531   0.631154  0.527940  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000202358   0.102176  0.918617  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000119912  -0.369535  0.711729  0.999624</span></span>
<span><span class="co">#&gt; ...                    ...       ...       ...</span></span>
<span><span class="co">#&gt; ENSG00000274518 -0.0886896  0.929329  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000104154  0.2919706  0.770309  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000148841 -0.2438306  0.807362  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000202283 -0.1659513  0.868195  0.999624</span></span>
<span><span class="co">#&gt; ENSG00000106261  0.3876656  0.698264  0.999624</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/refseqDESeq2.html">refseqDESeq2</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">log2FoldChange</span><span class="op">&gt;</span><span class="fl">2</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">plotNet</a></span><span class="op">(</span>asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Input genes: 58</span></span>
<span><span class="co">#&gt;   Converted input genes: 59</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.303</span></span></code></pre></div>
<div class="inline-figure"><img src="01-basic_usage_of_biotextgraph_files/figure-html/refseqdeseq2-1.png" width="100%" style="display: block; margin: auto;"></div>
</div>
<div id="split-by-ea" class="section level2" number="2.12">
<h2>
<span class="header-section-number">2.12</span> Split by EA<a class="anchor" aria-label="anchor" href="#split-by-ea"><i class="fas fa-link"></i></a>
</h2>
<p>If <code>splitByEA</code> option is enabled, the function first performs enrichment analysis on the queried input, and output the list of biotext object for each significant pathway, as well as for those not related to significant pathways. This is useful for confirming the prior knowledge in terms of textual information perspective.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eanets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/generalf.html">refseq</a></span><span class="op">(</span><span class="va">inpSymbol</span>, splitByEA<span class="op">=</span><span class="st">"reactome"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Total of 48 pathways, including non-enrichment terms</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="co">#&gt; Input genes: 5</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.7</span></span>
<span><span class="co">#&gt; Input genes: 5</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.7</span></span>
<span><span class="co">#&gt; Input genes: 7</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.5</span></span>
<span><span class="co">#&gt; Input genes: 5</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.7</span></span>
<span><span class="co">#&gt; Input genes: 4</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.8</span></span>
<span><span class="co">#&gt; Input genes: 4</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.8</span></span>
<span><span class="co">#&gt; Input genes: 3</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 3</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 3</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 3</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span>
<span><span class="co">#&gt; Input genes: 2</span></span>
<span><span class="co">#&gt; Filter based on GeneSummary</span></span>
<span><span class="co">#&gt; Filtered 77 words (frequency and/or tfidf)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 1</span></span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 22621)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Japanese_Japan.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Japanese_Japan.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Japanese_Japan.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                   </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Japanese_Japan.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Tokyo</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils    </span></span>
<span><span class="co">#&gt; [6] datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggkegg_1.0.7          testthat_3.1.10      </span></span>
<span><span class="co">#&gt;  [3] XML_3.99-0.14         tidygraph_1.2.3      </span></span>
<span><span class="co">#&gt;  [5] ggfx_1.0.1            igraph_1.5.1         </span></span>
<span><span class="co">#&gt;  [7] GetoptLong_1.0.5      ggrepel_0.9.5        </span></span>
<span><span class="co">#&gt;  [9] dplyr_1.1.4           dendextend_1.17.1    </span></span>
<span><span class="co">#&gt; [11] clusterProfiler_4.9.5 ReactomePA_1.46.0    </span></span>
<span><span class="co">#&gt; [13] RColorBrewer_1.1-3    ggraph_2.1.0.9000    </span></span>
<span><span class="co">#&gt; [15] org.Hs.eg.db_3.17.0   AnnotationDbi_1.63.2 </span></span>
<span><span class="co">#&gt; [17] IRanges_2.35.2        S4Vectors_0.38.1     </span></span>
<span><span class="co">#&gt; [19] Biobase_2.61.0        BiocGenerics_0.47.0  </span></span>
<span><span class="co">#&gt; [21] biotextgraph_0.99.0   ggplot2_3.4.4        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.3.1                </span></span>
<span><span class="co">#&gt;   [2] later_1.3.1                  </span></span>
<span><span class="co">#&gt;   [3] bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;   [4] ggplotify_0.1.2              </span></span>
<span><span class="co">#&gt;   [5] filelock_1.0.2               </span></span>
<span><span class="co">#&gt;   [6] tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;   [7] polyclip_1.10-4              </span></span>
<span><span class="co">#&gt;   [8] graph_1.79.1                 </span></span>
<span><span class="co">#&gt;   [9] lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt;  [10] rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [11] bugsigdbr_1.8.1              </span></span>
<span><span class="co">#&gt;  [12] processx_3.8.2               </span></span>
<span><span class="co">#&gt;  [13] NLP_0.2-1                    </span></span>
<span><span class="co">#&gt;  [14] lattice_0.21-8               </span></span>
<span><span class="co">#&gt;  [15] MASS_7.3-60                  </span></span>
<span><span class="co">#&gt;  [16] magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;  [17] sass_0.4.7                   </span></span>
<span><span class="co">#&gt;  [18] rmarkdown_2.25               </span></span>
<span><span class="co">#&gt;  [19] remotes_2.4.2.1              </span></span>
<span><span class="co">#&gt;  [20] jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt;  [21] yaml_2.3.7                   </span></span>
<span><span class="co">#&gt;  [22] httpuv_1.6.11                </span></span>
<span><span class="co">#&gt;  [23] sessioninfo_1.2.2            </span></span>
<span><span class="co">#&gt;  [24] pkgbuild_1.4.2               </span></span>
<span><span class="co">#&gt;  [25] cowplot_1.1.1                </span></span>
<span><span class="co">#&gt;  [26] DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [27] pkgload_1.3.2.1              </span></span>
<span><span class="co">#&gt;  [28] zlibbioc_1.47.0              </span></span>
<span><span class="co">#&gt;  [29] purrr_1.0.2                  </span></span>
<span><span class="co">#&gt;  [30] downlit_0.4.3                </span></span>
<span><span class="co">#&gt;  [31] RCurl_1.98-1.12              </span></span>
<span><span class="co">#&gt;  [32] yulab.utils_0.1.0            </span></span>
<span><span class="co">#&gt;  [33] tweenr_2.0.2                 </span></span>
<span><span class="co">#&gt;  [34] rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt;  [35] pvclust_2.2-0                </span></span>
<span><span class="co">#&gt;  [36] GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">#&gt;  [37] ISOcodes_2022.09.29          </span></span>
<span><span class="co">#&gt;  [38] enrichplot_1.21.3            </span></span>
<span><span class="co">#&gt;  [39] cyjShiny_1.0.42              </span></span>
<span><span class="co">#&gt;  [40] tm_0.7-11                    </span></span>
<span><span class="co">#&gt;  [41] tidytree_0.4.5               </span></span>
<span><span class="co">#&gt;  [42] rentrez_1.2.3                </span></span>
<span><span class="co">#&gt;  [43] reactome.db_1.86.0           </span></span>
<span><span class="co">#&gt;  [44] commonmark_1.9.0             </span></span>
<span><span class="co">#&gt;  [45] ggwordcloud_0.6.0            </span></span>
<span><span class="co">#&gt;  [46] codetools_0.2-19             </span></span>
<span><span class="co">#&gt;  [47] DOSE_3.27.2                  </span></span>
<span><span class="co">#&gt;  [48] xml2_1.3.5                   </span></span>
<span><span class="co">#&gt;  [49] ggforce_0.4.1                </span></span>
<span><span class="co">#&gt;  [50] tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [51] aplot_0.2.1                  </span></span>
<span><span class="co">#&gt;  [52] farver_2.1.1                 </span></span>
<span><span class="co">#&gt;  [53] viridis_0.6.4                </span></span>
<span><span class="co">#&gt;  [54] GeneSummary_0.99.6           </span></span>
<span><span class="co">#&gt;  [55] BiocFileCache_2.9.1          </span></span>
<span><span class="co">#&gt;  [56] base64enc_0.1-3              </span></span>
<span><span class="co">#&gt;  [57] showtext_0.9-6               </span></span>
<span><span class="co">#&gt;  [58] jsonlite_1.8.7               </span></span>
<span><span class="co">#&gt;  [59] ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [60] systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt;  [61] tools_4.3.1                  </span></span>
<span><span class="co">#&gt;  [62] ragg_1.2.5                   </span></span>
<span><span class="co">#&gt;  [63] treeio_1.25.4                </span></span>
<span><span class="co">#&gt;  [64] snow_0.4-4                   </span></span>
<span><span class="co">#&gt;  [65] HPO.db_0.99.2                </span></span>
<span><span class="co">#&gt;  [66] Rcpp_1.0.11                  </span></span>
<span><span class="co">#&gt;  [67] glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [68] gridExtra_2.3                </span></span>
<span><span class="co">#&gt;  [69] xfun_0.40                    </span></span>
<span><span class="co">#&gt;  [70] usethis_2.2.2                </span></span>
<span><span class="co">#&gt;  [71] qvalue_2.33.0                </span></span>
<span><span class="co">#&gt;  [72] GenomeInfoDb_1.37.4          </span></span>
<span><span class="co">#&gt;  [73] withr_2.5.0                  </span></span>
<span><span class="co">#&gt;  [74] BiocManager_1.30.22          </span></span>
<span><span class="co">#&gt;  [75] fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [76] fansi_1.0.4                  </span></span>
<span><span class="co">#&gt;  [77] callr_3.7.3                  </span></span>
<span><span class="co">#&gt;  [78] digest_0.6.33                </span></span>
<span><span class="co">#&gt;  [79] R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [80] mime_0.12                    </span></span>
<span><span class="co">#&gt;  [81] gridGraphics_0.5-1           </span></span>
<span><span class="co">#&gt;  [82] textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [83] colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [84] Cairo_1.6-1                  </span></span>
<span><span class="co">#&gt;  [85] GO.db_3.17.0                 </span></span>
<span><span class="co">#&gt;  [86] markdown_1.8                 </span></span>
<span><span class="co">#&gt;  [87] RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;  [88] utf8_1.2.3                   </span></span>
<span><span class="co">#&gt;  [89] tidyr_1.3.0                  </span></span>
<span><span class="co">#&gt;  [90] generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [91] data.table_1.14.8            </span></span>
<span><span class="co">#&gt;  [92] prettyunits_1.1.1            </span></span>
<span><span class="co">#&gt;  [93] graphlayouts_1.0.0           </span></span>
<span><span class="co">#&gt;  [94] stopwords_2.3                </span></span>
<span><span class="co">#&gt;  [95] httr_1.4.7                   </span></span>
<span><span class="co">#&gt;  [96] htmlwidgets_1.6.2            </span></span>
<span><span class="co">#&gt;  [97] scatterpie_0.2.1             </span></span>
<span><span class="co">#&gt;  [98] graphite_1.48.0              </span></span>
<span><span class="co">#&gt;  [99] pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt; [100] gtable_0.3.4                 </span></span>
<span><span class="co">#&gt; [101] blob_1.2.4                   </span></span>
<span><span class="co">#&gt; [102] XVector_0.41.1               </span></span>
<span><span class="co">#&gt; [103] brio_1.1.3                   </span></span>
<span><span class="co">#&gt; [104] shadowtext_0.1.2             </span></span>
<span><span class="co">#&gt; [105] htmltools_0.5.6              </span></span>
<span><span class="co">#&gt; [106] profvis_0.3.8                </span></span>
<span><span class="co">#&gt; [107] sysfonts_0.8.8               </span></span>
<span><span class="co">#&gt; [108] bookdown_0.35                </span></span>
<span><span class="co">#&gt; [109] fgsea_1.27.1                 </span></span>
<span><span class="co">#&gt; [110] scales_1.3.0                 </span></span>
<span><span class="co">#&gt; [111] png_0.1-8                    </span></span>
<span><span class="co">#&gt; [112] wordcloud_2.6                </span></span>
<span><span class="co">#&gt; [113] ggfun_0.1.3                  </span></span>
<span><span class="co">#&gt; [114] ggdendro_0.1.23              </span></span>
<span><span class="co">#&gt; [115] knitr_1.44                   </span></span>
<span><span class="co">#&gt; [116] rstudioapi_0.15.0            </span></span>
<span><span class="co">#&gt; [117] reshape2_1.4.4               </span></span>
<span><span class="co">#&gt; [118] rjson_0.2.21                 </span></span>
<span><span class="co">#&gt; [119] nlme_3.1-163                 </span></span>
<span><span class="co">#&gt; [120] curl_5.0.2                   </span></span>
<span><span class="co">#&gt; [121] showtextdb_3.0               </span></span>
<span><span class="co">#&gt; [122] cachem_1.0.8                 </span></span>
<span><span class="co">#&gt; [123] GlobalOptions_0.1.2          </span></span>
<span><span class="co">#&gt; [124] stringr_1.5.0                </span></span>
<span><span class="co">#&gt; [125] BiocVersion_3.18.0           </span></span>
<span><span class="co">#&gt; [126] miniUI_0.1.1.1               </span></span>
<span><span class="co">#&gt; [127] parallel_4.3.1               </span></span>
<span><span class="co">#&gt; [128] HDO.db_0.99.1                </span></span>
<span><span class="co">#&gt; [129] desc_1.4.2                   </span></span>
<span><span class="co">#&gt; [130] pillar_1.9.0                 </span></span>
<span><span class="co">#&gt; [131] grid_4.3.1                   </span></span>
<span><span class="co">#&gt; [132] vctrs_0.6.5                  </span></span>
<span><span class="co">#&gt; [133] urlchecker_1.0.1             </span></span>
<span><span class="co">#&gt; [134] slam_0.1-50                  </span></span>
<span><span class="co">#&gt; [135] promises_1.2.1               </span></span>
<span><span class="co">#&gt; [136] dbplyr_2.3.3                 </span></span>
<span><span class="co">#&gt; [137] xtable_1.8-4                 </span></span>
<span><span class="co">#&gt; [138] evaluate_0.21                </span></span>
<span><span class="co">#&gt; [139] magick_2.7.5                 </span></span>
<span><span class="co">#&gt; [140] cli_3.6.1                    </span></span>
<span><span class="co">#&gt; [141] compiler_4.3.1               </span></span>
<span><span class="co">#&gt; [142] rlang_1.1.1                  </span></span>
<span><span class="co">#&gt; [143] crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [144] labeling_0.4.3               </span></span>
<span><span class="co">#&gt; [145] ps_1.7.5                     </span></span>
<span><span class="co">#&gt; [146] plyr_1.8.8                   </span></span>
<span><span class="co">#&gt; [147] fs_1.6.3                     </span></span>
<span><span class="co">#&gt; [148] stringi_1.7.12               </span></span>
<span><span class="co">#&gt; [149] viridisLite_0.4.2            </span></span>
<span><span class="co">#&gt; [150] BiocParallel_1.35.4          </span></span>
<span><span class="co">#&gt; [151] MPO.db_0.99.7                </span></span>
<span><span class="co">#&gt; [152] munsell_0.5.0                </span></span>
<span><span class="co">#&gt; [153] Biostrings_2.69.2            </span></span>
<span><span class="co">#&gt; [154] lazyeval_0.2.2               </span></span>
<span><span class="co">#&gt; [155] devtools_2.4.5               </span></span>
<span><span class="co">#&gt; [156] GOSemSim_2.27.3              </span></span>
<span><span class="co">#&gt; [157] Matrix_1.6-5                 </span></span>
<span><span class="co">#&gt; [158] patchwork_1.2.0              </span></span>
<span><span class="co">#&gt; [159] bit64_4.0.5                  </span></span>
<span><span class="co">#&gt; [160] KEGGREST_1.41.0              </span></span>
<span><span class="co">#&gt; [161] shiny_1.7.5                  </span></span>
<span><span class="co">#&gt; [162] interactiveDisplayBase_1.39.0</span></span>
<span><span class="co">#&gt; [163] AnnotationHub_3.9.2          </span></span>
<span><span class="co">#&gt; [164] gridtext_0.1.5               </span></span>
<span><span class="co">#&gt; [165] memoise_2.0.1                </span></span>
<span><span class="co">#&gt; [166] bslib_0.5.1                  </span></span>
<span><span class="co">#&gt; [167] ggtree_3.9.1                 </span></span>
<span><span class="co">#&gt; [168] fastmatch_1.1-4              </span></span>
<span><span class="co">#&gt; [169] bit_4.0.5                    </span></span>
<span><span class="co">#&gt; [170] ape_5.7-1                    </span></span>
<span><span class="co">#&gt; [171] gson_0.1.0</span></span></code></pre></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html"><span class="header-section-number">1</span> About</a></div>
<div class="next"><a href="metacyc.html"><span class="header-section-number">3</span> Microbial functional analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#basic-usage"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="nav-link" href="#producing-networks"><span class="header-section-number">2.1</span> Producing networks</a></li>
<li><a class="nav-link" href="#identifying-important-genes-in-the-network"><span class="header-section-number">2.2</span> Identifying important genes in the network</a></li>
<li>
<a class="nav-link" href="#producing-word-clouds"><span class="header-section-number">2.3</span> Producing word clouds</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#plotting-associated-genes-in-wc"><span class="header-section-number">2.3.1</span> Plotting associated genes in WC</a></li></ul>
</li>
<li><a class="nav-link" href="#visualization-of-pubmed-information."><span class="header-section-number">2.4</span> Visualization of PubMed information.</a></li>
<li>
<a class="nav-link" href="#comparing-two-or-more-networks"><span class="header-section-number">2.5</span> Comparing two or more networks</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#connecting-the-networks-and-identify-shared-genes"><span class="header-section-number">2.5.1</span> Connecting the networks and identify shared genes</a></li></ul>
</li>
<li><a class="nav-link" href="#text-over-represenatation-analysis-experimental"><span class="header-section-number">2.6</span> Text over represenatation analysis (experimental)</a></li>
<li><a class="nav-link" href="#dependency-analyis-using-udpipe"><span class="header-section-number">2.7</span> Dependency analyis using udpipe</a></li>
<li><a class="nav-link" href="#changing-the-font"><span class="header-section-number">2.8</span> Changing the font</a></li>
<li>
<a class="nav-link" href="#biofabric-layout"><span class="header-section-number">2.9</span> Biofabric layout</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#highlighting-genes-in-the-layout"><span class="header-section-number">2.9.1</span> Highlighting genes in the layout</a></li>
<li><a class="nav-link" href="#biofabric-layouts-for-combined-networks"><span class="header-section-number">2.9.2</span> Biofabric layouts for combined networks</a></li>
<li><a class="nav-link" href="#wrapper-for-biofabric-layout"><span class="header-section-number">2.9.3</span> Wrapper for BioFabric layout</a></li>
</ul>
</li>
<li><a class="nav-link" href="#dynamic-layout"><span class="header-section-number">2.10</span> Dynamic layout</a></li>
<li><a class="nav-link" href="#input-type"><span class="header-section-number">2.11</span> Input type</a></li>
<li><a class="nav-link" href="#split-by-ea"><span class="header-section-number">2.12</span> Split by EA</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>biotextgraph</strong>" was written by Noriaki Sato. It was last built on 2024-02-01.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
