<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 Microbial functional analysis | biotextgraph</title>
<meta name="author" content="Noriaki Sato">
<meta name="generator" content="bookdown 0.36 with bs4_book()">
<meta property="og:title" content="Chapter 3 Microbial functional analysis | biotextgraph">
<meta property="og:type" content="book">
<meta property="og:url" content="https://noriakis.github.io/software/biotextgraph/metacyc.html">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 Microbial functional analysis | biotextgraph">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-9FXY6YW4YZ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-9FXY6YW4YZ');
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="In this section, we introduce making word networks showing microbial function using taxonomic idenfiers as input. For mining the pathway information in curated database MetaCyc (Caspi et...">
<meta property="og:description" content="In this section, we introduce making word networks showing microbial function using taxonomic idenfiers as input. For mining the pathway information in curated database MetaCyc (Caspi et...">
<meta name="twitter:description" content="In this section, we introduce making word networks showing microbial function using taxonomic idenfiers as input. For mining the pathway information in curated database MetaCyc (Caspi et...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">biotextgraph</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></li>
<li><a class="active" href="metacyc.html"><span class="header-section-number">3</span> Microbial functional analysis</a></li>
<li><a class="" href="the-other-usage-for-microbiome-related-data.html"><span class="header-section-number">4</span> The other usage for microbiome-related data</a></li>
<li><a class="" href="interactive-inspection-of-gene-cluster-network-annotated-by-words.html"><span class="header-section-number">5</span> Interactive inspection of gene cluster network annotated by words</a></li>
<li><a class="" href="annotating-gene-cluster-dendrogram.html"><span class="header-section-number">6</span> Annotating gene cluster dendrogram</a></li>
<li><a class="" href="quantitative-analysis.html"><span class="header-section-number">7</span> Quantitative analysis</a></li>
<li><a class="" href="single-cell-transcriptomic-data.html"><span class="header-section-number">8</span> Single-cell transcriptomic data</a></li>
<li><a class="" href="app.html"><span class="header-section-number">9</span> Application examples</a></li>
<li><a class="" href="parsing.html"><span class="header-section-number">10</span> Parsing</a></li>
<li><a class="" href="tidy.html"><span class="header-section-number">11</span> Perform each process separately</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="metacyc" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Microbial functional analysis<a class="anchor" aria-label="anchor" href="#metacyc"><i class="fas fa-link"></i></a>
</h1>
<p>In this section, we introduce making word networks showing microbial function using taxonomic idenfiers as input. For mining the pathway information in curated database MetaCyc (<a href="https://academic.oup.com/nar/article/48/D1/D445/5581728">Caspi et al. 2020</a>), users can prepare <code>pathways.dat</code> from MetaCyc flat files. Note that you must have a valid license of using MetaCyc. See <a href="https://www.metacyc.org/MetaCycUserGuide.shtml">User guides</a> of MetaCyc, and <a href="www.biocyc.org">BioCyc</a>. The other options will be KEGG Pathway textual information.</p>
<p>Suppose we would like to know pathway and related information of “<em>Bifidobacterium longum</em>” and “<em>Escherichia coli</em>”. The function <code>parseMetaCycPathway</code> can be used to parse the summarized comment of pathways using these queries. Note that the function search for <code>TAXONOMIC-RANGE</code> or <code>SPECIES</code> information in <code>pathways.dat</code> if <code>withTax=TRUE</code>. The resulting data.frame looks like below.</p>
<p>For selecting species investigated, you can use your own dataset or the database such as <code>BugSigDB</code>, described in the section <a href="the-other-usage-for-microbiome-related-data.html#disease">4.1</a>.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biotextgraph</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Two species selected</span></span>
<span><span class="va">candidateSpecies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Staphylococcus aureus"</span>,<span class="st">"Escherichia coli"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Downloaded pathway description file</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">"metacyc/24.5/data/pathways.dat"</span></span>
<span></span>
<span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/parseMetaCycPathway.html">parseMetaCycPathway</a></span><span class="op">(</span><span class="va">file</span>, <span class="va">candidateSpecies</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">input</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="op">!</span><span class="va">text</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<thead><tr>
<th style="text-align:left;">
pathwayID
</th>
<th style="text-align:left;">
commonName
</th>
<th style="text-align:left;">
query
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
PWY-7622
</td>
<td style="text-align:left;">
UDP-&amp;alpha;-D-galactofuranose biosynthesis
</td>
<td style="text-align:left;">
Escherichia coli
</td>
</tr></tbody>
</table></div>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">input2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/parseMetaCycPathway.html">parseMetaCycPathway</a></span><span class="op">(</span><span class="va">file</span>, <span class="va">candidateSpecies</span>, withTax<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">input2</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="op">!</span><span class="va">text</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<thead><tr>
<th style="text-align:left;">
pathwayID
</th>
<th style="text-align:left;">
commonName
</th>
<th style="text-align:left;">
species
</th>
<th style="text-align:left;">
taxonomicRange
</th>
<th style="text-align:left;">
query
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
PWY-7622
</td>
<td style="text-align:left;">
UDP-&amp;alpha;-D-galactofuranose biosynthesis
</td>
<td style="text-align:left;">
TAX-746128,TAX-330879,TAX-162425
</td>
<td style="text-align:left;">
TAX-6231,TAX-3052,TAX-147538,TAX-5654,TAX-2
</td>
<td style="text-align:left;">
Escherichia coli
</td>
</tr></tbody>
</table></div>
<p>You can strip some tags and symbols in the text by <code>clear=TRUE</code>.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/parseMetaCycPathway.html">parseMetaCycPathway</a></span><span class="op">(</span><span class="va">file</span>, <span class="va">candidateSpecies</span>, withTax<span class="op">=</span><span class="cn">TRUE</span>, clear<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## Not shown because of the long text</span></span></code></pre></div>
<p>These data frames can be passed to <code>manual</code> function, which performs the same analysis as RefSeq or PubMed information. The input data frame has to have <code>"text"</code> column to make word cloud or a network.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Set remove words</span></span>
<span><span class="va">remove_words</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cits"</span>,<span class="st">"frame"</span>,<span class="st">"pathway"</span>,<span class="st">"gene"</span>,<span class="st">"genes"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metawc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/manual.html">manual</a></span><span class="op">(</span><span class="va">input3</span>, plotType<span class="op">=</span><span class="st">"wc"</span>,</span>
<span>                additionalRemove<span class="op">=</span><span class="va">remove_words</span>,</span>
<span>                numWords<span class="op">=</span><span class="fl">100</span>,</span>
<span>                 argList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>                   rot.per<span class="op">=</span><span class="fl">0.4</span>,</span>
<span>                   colors<span class="op">=</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span>,</span>
<span>                   random.order<span class="op">=</span><span class="cn">FALSE</span></span>
<span>                 <span class="op">)</span><span class="op">)</span></span>
<span><span class="va">metawc</span></span>
<span><span class="co">#&gt; Type: manual</span></span>
<span><span class="co">#&gt; Number of words: 100</span></span>
<span><span class="co">#&gt; Query: </span></span>
<span><span class="co">#&gt; 526 Kb</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotWC.html">plotWC</a></span><span class="op">(</span><span class="va">metawc</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-metacyc_manual_files/figure-html/wcmeta-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>For plotting the network, query column must be specified if plotting the query with the word information.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="va">metanet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/manual.html">manual</a></span><span class="op">(</span><span class="va">input</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"query"</span>,<span class="st">"text"</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                  additionalRemove<span class="op">=</span><span class="va">remove_words</span>,</span>
<span>                  numWords<span class="op">=</span><span class="fl">40</span>, colorText<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.4</span></span>
<span><span class="va">metanet</span></span>
<span><span class="co">#&gt; Type: manual</span></span>
<span><span class="co">#&gt; Number of words: 40</span></span>
<span><span class="co">#&gt; Query: </span></span>
<span><span class="co">#&gt; Graph: V(40), E(95)</span></span>
<span><span class="co">#&gt; Degree: acids(8)/cluster(8)/enzyme(8)/sialic(8)/addition(7)</span></span>
<span><span class="co">#&gt; 560.8 Kb</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotNet.html">plotNet</a></span><span class="op">(</span><span class="va">metanet</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-metacyc_manual_files/figure-html/wcmeta2-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>For column other than the query and text, in this example <code>commonName</code> and <code>pathwayID</code>, the relationship between query and these columns are to be plotted. In this example, the connection between query and text is not shown because <code>queryPlot</code> is FALSE.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">metanet2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/manual.html">manual</a></span><span class="op">(</span><span class="va">input</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"query"</span>,<span class="st">"text"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                   layout<span class="op">=</span><span class="st">"kk"</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                   additionalRemove<span class="op">=</span><span class="va">remove_words</span>,</span>
<span>                   numWords<span class="op">=</span><span class="fl">20</span>, colorText<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                   queryPlot<span class="op">=</span><span class="cn">TRUE</span>, ngram<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.2</span></span>
<span><span class="va">metanet2</span></span>
<span><span class="co">#&gt; Type: manual</span></span>
<span><span class="co">#&gt; Number of words: 20</span></span>
<span><span class="co">#&gt; Query: </span></span>
<span><span class="co">#&gt; Graph: V(22), E(110)</span></span>
<span><span class="co">#&gt; Degree: Escherichia coli(41)/escherichia coli(22)/biosynthetic cluster(15)/coli purified(13)/staphylococcus aureus(13)</span></span>
<span><span class="co">#&gt; 830.2 Kb</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotNet.html">plotNet</a></span><span class="op">(</span><span class="va">metanet2</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-metacyc_manual_files/figure-html/wcmeta3-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>When taxonomy parsing is available, query by the NCBI Taxonomy ID.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set candSp to all and noComma to TRUE</span></span>
<span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/parseMetaCycPathway.html">parseMetaCycPathway</a></span><span class="op">(</span><span class="va">file</span>, candSp<span class="op">=</span><span class="st">"all"</span>, withTax <span class="op">=</span> <span class="cn">TRUE</span>, noComma<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">input2</span> <span class="op">&lt;-</span> <span class="va">input</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"TAX-2157"</span>,<span class="va">input</span><span class="op">$</span><span class="va">taxonomicRange</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">input2</span> <span class="op">&lt;-</span> <span class="va">input2</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">input2</span><span class="op">$</span><span class="va">pathwayID</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">onlyText</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">input2</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"text"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">`colnames&lt;-`</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"text"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">input2Net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/manual.html">manual</a></span><span class="op">(</span><span class="va">onlyText</span>, additionalRemove<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cits"</span>,<span class="st">"frame"</span>,</span>
<span>                       <span class="st">"gene"</span>,<span class="st">"genes"</span>,<span class="st">"proteins"</span>,</span>
<span>                       <span class="st">"pathway"</span>,<span class="st">"pathways"</span>,<span class="st">"enzyme"</span>,<span class="st">"enzymes"</span>,</span>
<span>                       <span class="st">"bacteria"</span>,<span class="st">"reaction"</span>,<span class="st">"protein"</span>,<span class="st">"biosynthesis"</span>,</span>
<span>                       <span class="st">"organism"</span>,<span class="st">"organisms"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.1</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotNet.html">plotNet</a></span><span class="op">(</span><span class="va">input2Net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-metacyc_manual_files/figure-html/query-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>Also, if you want to search for the NCBI taxonomy identifiers and want to use species names as queries, First you should convert the IDs using <code>convertMetaCyc</code> function using <code>taxonomizr</code>.
Next you search for converted names for the interested species, and input this data frame to <code>manual</code>.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span><span class="op">$</span><span class="va">converted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/convertMetaCyc.html">convertMetaCyc</a></span><span class="op">(</span><span class="va">input</span><span class="op">$</span><span class="va">species</span><span class="op">)</span></span>
<span><span class="va">input3</span> <span class="op">&lt;-</span> <span class="va">input</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Bifidobacterium"</span>,<span class="va">input</span><span class="op">$</span><span class="va">converted</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">input3</span> <span class="op">&lt;-</span> <span class="va">input3</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">input3</span><span class="op">$</span><span class="va">pathwayID</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">input3</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Bifidobacterium"</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">input3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">input3</span> <span class="op">&lt;-</span> <span class="va">input3</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"text"</span>,<span class="st">"pathwayID"</span>,<span class="st">"query"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">input3Net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/manual.html">manual</a></span><span class="op">(</span><span class="va">input3</span>, plotType<span class="op">=</span><span class="st">"network"</span>, queryPlot<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                    layout<span class="op">=</span><span class="st">"lgl"</span>, edgeLink<span class="op">=</span><span class="cn">FALSE</span>, tag<span class="op">=</span><span class="st">"cor"</span>,</span>
<span>                    additionalRemove<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cits"</span>,<span class="st">"frame"</span>,</span>
<span>                                       <span class="st">"gene"</span>,<span class="st">"genes"</span>,<span class="st">"proteins"</span>,</span>
<span>                                       <span class="st">"pathway"</span>,<span class="st">"pathways"</span>,<span class="st">"enzyme"</span>,<span class="st">"enzymes"</span>,</span>
<span>                                       <span class="st">"bacteria"</span>,<span class="st">"reaction"</span>,<span class="st">"protein"</span>,<span class="st">"biosynthesis"</span>,</span>
<span>                                       <span class="st">"organism"</span>,<span class="st">"organisms"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.5)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.6)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.7)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.8)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 0.9)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.0)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.1)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.2)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.3)... Done.</span></span>
<span><span class="co">#&gt; Bootstrap (r = 1.4)... Done.</span></span>
<span><span class="co">#&gt; Ignoring corThresh, automatically determine the value</span></span>
<span><span class="co">#&gt; threshold = 0.8</span></span>
<span><span class="co">#&gt; Including columns pathwayID to link with query</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biotextgraph/man/plotNet.html">plotNet</a></span><span class="op">(</span><span class="va">input3Net</span>, asis<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02-metacyc_manual_files/figure-html/query2-1.png" width="100%" style="display: block; margin: auto;"></div>
<p>Includes BioCyc (TM) pathway/genome databases under license from SRI International.<br><img src="https://biocyc.org/graphics2021/BioCyc-logo-color-genome.svg" width="100px"></p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/local/package/r/4.3.0/lib64/R/lib/libRblas.so </span></span>
<span><span class="co">#&gt; LAPACK: /usr/local/package/r/4.3.0/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=ja_JP.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=ja_JP.UTF-8        LC_COLLATE=ja_JP.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=ja_JP.UTF-8    LC_MESSAGES=ja_JP.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=ja_JP.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=ja_JP.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Tokyo</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [6] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] digest_0.6.33     R6_2.5.1          bookdown_0.36    </span></span>
<span><span class="co">#&gt;  [4] fastmap_1.1.1     xfun_0.40         cachem_1.0.8     </span></span>
<span><span class="co">#&gt;  [7] knitr_1.43        memoise_2.0.1     htmltools_0.5.6  </span></span>
<span><span class="co">#&gt; [10] rmarkdown_2.23    xml2_1.3.5        cli_3.6.1        </span></span>
<span><span class="co">#&gt; [13] downlit_0.4.3     sass_0.4.7        withr_2.5.0      </span></span>
<span><span class="co">#&gt; [16] jquerylib_0.1.4   compiler_4.3.0    rstudioapi_0.15.0</span></span>
<span><span class="co">#&gt; [19] tools_4.3.0       evaluate_0.21     bslib_0.5.1      </span></span>
<span><span class="co">#&gt; [22] yaml_2.3.7        fs_1.6.3          jsonlite_1.8.7   </span></span>
<span><span class="co">#&gt; [25] rlang_1.1.1</span></span></code></pre></div>

</div>
  <div class="chapter-nav">
<div class="prev"><a href="basic-usage.html"><span class="header-section-number">2</span> Basic usage</a></div>
<div class="next"><a href="the-other-usage-for-microbiome-related-data.html"><span class="header-section-number">4</span> The other usage for microbiome-related data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#metacyc"><span class="header-section-number">3</span> Microbial functional analysis</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>biotextgraph</strong>" was written by Noriaki Sato. It was last built on 2023-12-20.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
