<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Including clinical variables | CBNplot</title>
  <meta name="description" content="Plotting Bayesian network based on enrichment analysis including clusterProfiler and ReactomePA" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Including clinical variables | CBNplot" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Plotting Bayesian network based on enrichment analysis including clusterProfiler and ReactomePA" />
  <meta name="github-repo" content="noriakis/CBNplot" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Including clinical variables | CBNplot" />
  
  <meta name="twitter:description" content="Plotting Bayesian network based on enrichment analysis including clusterProfiler and ReactomePA" />
  

<meta name="author" content="Noriaki Sato" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bnpathplot.html"/>
<link rel="next" href="the-other-options.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CBNplot</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> CBNplot: Bayesian network plot for enrichment analysis results</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-preprocessing-and-deg-identification-of-gse133624"><i class="fa fa-check"></i><b>1.1</b> The preprocessing and DEG identification of GSE133624</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="bootstrap-based-inference.html"><a href="bootstrap-based-inference.html"><i class="fa fa-check"></i><b>2</b> bootstrap-based inference</a>
<ul>
<li class="chapter" data-level="2.1" data-path="bootstrap-based-inference.html"><a href="bootstrap-based-inference.html#bootstrap-based-inference-1"><i class="fa fa-check"></i><b>2.1</b> bootstrap-based inference</a></li>
<li class="chapter" data-level="2.2" data-path="bootstrap-based-inference.html"><a href="bootstrap-based-inference.html#multiscale-boostrap-based-inference"><i class="fa fa-check"></i><b>2.2</b> Multiscale boostrap-based inference</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="bngeneplot.html"><a href="bngeneplot.html"><i class="fa fa-check"></i><b>3</b> <code>bngeneplot</code></a>
<ul>
<li class="chapter" data-level="3.1" data-path="bngeneplot.html"><a href="bngeneplot.html#barplot"><i class="fa fa-check"></i><b>3.1</b> <code>barplot</code></a></li>
<li class="chapter" data-level="3.2" data-path="bngeneplot.html"><a href="bngeneplot.html#the-vanilla-plot"><i class="fa fa-check"></i><b>3.2</b> The vanilla plot</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="bngeneplot.html"><a href="bngeneplot.html#show-the-confidence-of-direction"><i class="fa fa-check"></i><b>3.2.1</b> Show the confidence of direction</a></li>
<li class="chapter" data-level="3.2.2" data-path="bngeneplot.html"><a href="bngeneplot.html#show-the-hub-genes"><i class="fa fa-check"></i><b>3.2.2</b> Show the hub genes</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="bngeneplot.html"><a href="bngeneplot.html#the-plot-with-the-reference"><i class="fa fa-check"></i><b>3.3</b> The plot with the reference</a></li>
<li class="chapter" data-level="3.4" data-path="bngeneplot.html"><a href="bngeneplot.html#the-plot-with-the-depmap-information"><i class="fa fa-check"></i><b>3.4</b> The plot with the DepMap information</a></li>
<li class="chapter" data-level="3.5" data-path="bngeneplot.html"><a href="bngeneplot.html#plotting-edge-strengths"><i class="fa fa-check"></i><b>3.5</b> Plotting edge strengths</a></li>
<li class="chapter" data-level="3.6" data-path="bngeneplot.html"><a href="bngeneplot.html#plotting-genes-in-multiple-pathways"><i class="fa fa-check"></i><b>3.6</b> Plotting genes in multiple pathways</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bnpathplot.html"><a href="bnpathplot.html"><i class="fa fa-check"></i><b>4</b> <code>bnpathplot</code></a>
<ul>
<li class="chapter" data-level="4.1" data-path="bnpathplot.html"><a href="bnpathplot.html#emapplot"><i class="fa fa-check"></i><b>4.1</b> <code>emapplot</code></a></li>
<li class="chapter" data-level="4.2" data-path="bnpathplot.html"><a href="bnpathplot.html#the-vanilla-plot-1"><i class="fa fa-check"></i><b>4.2</b> The vanilla plot</a></li>
<li class="chapter" data-level="4.3" data-path="bnpathplot.html"><a href="bnpathplot.html#compare-with-the-reference"><i class="fa fa-check"></i><b>4.3</b> Compare with the reference</a></li>
<li class="chapter" data-level="4.4" data-path="bnpathplot.html"><a href="bnpathplot.html#reflect-depmap-information-to-pathways"><i class="fa fa-check"></i><b>4.4</b> Reflect DepMap information to pathways</a></li>
<li class="chapter" data-level="4.5" data-path="bnpathplot.html"><a href="bnpathplot.html#aggregating-the-pathway-databases"><i class="fa fa-check"></i><b>4.5</b> Aggregating the pathway databases</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html"><i class="fa fa-check"></i><b>5</b> Including clinical variables</a>
<ul>
<li class="chapter" data-level="5.1" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#preparation-of-data"><i class="fa fa-check"></i><b>5.1</b> Preparation of data</a></li>
<li class="chapter" data-level="5.2" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#inference-of-pathway-relationship-including-clinical-variables"><i class="fa fa-check"></i><b>5.2</b> Inference of pathway relationship including clinical variables</a></li>
<li class="chapter" data-level="5.3" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#conditional-probability-query"><i class="fa fa-check"></i><b>5.3</b> Conditional probability query</a></li>
<li class="chapter" data-level="5.4" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#gene-relationship-with-variables"><i class="fa fa-check"></i><b>5.4</b> Gene relationship with variables</a></li>
<li class="chapter" data-level="5.5" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#confirming-the-existing-knowledge"><i class="fa fa-check"></i><b>5.5</b> Confirming the existing knowledge</a></li>
<li class="chapter" data-level="5.6" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#investigating-the-network-based-on-the-clinical-question"><i class="fa fa-check"></i><b>5.6</b> Investigating the network based on the clinical question</a></li>
<li class="chapter" data-level="5.7" data-path="including-clinical-variables.html"><a href="including-clinical-variables.html#classification-using-bn"><i class="fa fa-check"></i><b>5.7</b> Classification using BN</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="the-other-options.html"><a href="the-other-options.html"><i class="fa fa-check"></i><b>6</b> The other options</a>
<ul>
<li class="chapter" data-level="6.1" data-path="the-other-options.html"><a href="the-other-options.html#using-microbiome-related-data"><i class="fa fa-check"></i><b>6.1</b> Using microbiome-related data</a></li>
<li class="chapter" data-level="6.2" data-path="the-other-options.html"><a href="the-other-options.html#discretization"><i class="fa fa-check"></i><b>6.2</b> Discretization</a></li>
<li class="chapter" data-level="6.3" data-path="the-other-options.html"><a href="the-other-options.html#custom-visualization"><i class="fa fa-check"></i><b>6.3</b> Custom visualization</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="the-other-options.html"><a href="the-other-options.html#the-glowing-nodes-and-edges"><i class="fa fa-check"></i><b>6.3.1</b> The glowing nodes and edges</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="the-other-options.html"><a href="the-other-options.html#comparing-multi-scale-and-standard-bootstrapping"><i class="fa fa-check"></i><b>6.4</b> Comparing multi scale and standard bootstrapping</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="comparing-networks.html"><a href="comparing-networks.html"><i class="fa fa-check"></i><b>7</b> Comparing networks</a>
<ul>
<li class="chapter" data-level="7.1" data-path="comparing-networks.html"><a href="comparing-networks.html#comparing-multiple-networks"><i class="fa fa-check"></i><b>7.1</b> Comparing multiple networks</a></li>
<li class="chapter" data-level="7.2" data-path="comparing-networks.html"><a href="comparing-networks.html#testing-r"><i class="fa fa-check"></i><b>7.2</b> Testing R</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CBNplot</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="including-clinical-variables" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Including clinical variables<a href="including-clinical-variables.html#including-clinical-variables" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="preparation-of-data" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Preparation of data<a href="including-clinical-variables.html#preparation-of-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One of the important reasons of the Bayesian network analysis is assessing the relationship between gene expressions and clinical variables. <code>CBNplot</code> offers incorporating metadata into inference. As a demonstrative purpose, the enrichment analysis results from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133624">GSE133624</a> is applied on data from The Cancer Genome Atlas Program (TCGA) <span class="citation">(<a href="#ref-Cancer_Genome_Atlas_Research_Network2013-cd" role="doc-biblioref">Cancer Genome Atlas Research Network et al. 2013</a>)</span>. Specifically, TCGA-BLCA data is downloaded by the useful library <code>TCGAbiolinks</code> <span class="citation">(<a href="#ref-Colaprico2016-xq" role="doc-biblioref">Colaprico et al. 2016</a>)</span>.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="including-clinical-variables.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TCGAbiolinks)</span>
<span id="cb32-2"><a href="including-clinical-variables.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Load dataset</span></span>
<span id="cb32-3"><a href="including-clinical-variables.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file=</span><span class="st">&quot;tcgablcaData.rda&quot;</span>)</span></code></pre></div>
<p>We filtered the metadata based on the variables to be included in the inference. In this analysis, <code>age_at_diagnosis</code> and <code>paper_Combined T and LN category</code>, which is a sum of Tumor stage T 1/2 (0) vs. 3/4 (1) and LN negative (0) vs positive (1), are included. We again applied VST on the data.</p>
</div>
<div id="inference-of-pathway-relationship-including-clinical-variables" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Inference of pathway relationship including clinical variables<a href="including-clinical-variables.html#inference-of-pathway-relationship-including-clinical-variables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We assess the relationship between the curated biological pathway information and clinical variables mentioned above. Variables other than expression data can be specified with <code>otherVar</code>, as well as <code>otherVarName</code> for their name. The order of <code>otherVar</code> must be same as column order of expression data. We use all the significant pathways of corrected p-values below 0.05.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="including-clinical-variables.html#cb33-1" aria-hidden="true" tabindex="-1"></a>bnCov <span class="ot">&lt;-</span> <span class="fu">bnpathplot</span>(pway,</span>
<span id="cb33-2"><a href="including-clinical-variables.html#cb33-2" aria-hidden="true" tabindex="-1"></a>                    vstedTCGA,</span>
<span id="cb33-3"><a href="including-clinical-variables.html#cb33-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">nCategory =</span> <span class="dv">1000</span>,</span>
<span id="cb33-4"><a href="including-clinical-variables.html#cb33-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">adjpCutOff =</span> <span class="fl">0.01</span>,</span>
<span id="cb33-5"><a href="including-clinical-variables.html#cb33-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">expSample=</span><span class="fu">rownames</span>(metadata),</span>
<span id="cb33-6"><a href="including-clinical-variables.html#cb33-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">algo=</span><span class="st">&quot;hc&quot;</span>, <span class="at">strType=</span><span class="st">&quot;normal&quot;</span>,</span>
<span id="cb33-7"><a href="including-clinical-variables.html#cb33-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">otherVar=</span>metadata,</span>
<span id="cb33-8"><a href="including-clinical-variables.html#cb33-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>, <span class="st">&quot;Category&quot;</span>),</span>
<span id="cb33-9"><a href="including-clinical-variables.html#cb33-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">R=</span><span class="dv">50</span>, <span class="at">cl=</span>parallel<span class="sc">::</span><span class="fu">makeCluster</span>(<span class="dv">10</span>),</span>
<span id="cb33-10"><a href="including-clinical-variables.html#cb33-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">returnNet=</span>T,</span>
<span id="cb33-11"><a href="including-clinical-variables.html#cb33-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">shadowText=</span>T)</span>
<span id="cb33-12"><a href="including-clinical-variables.html#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Check DAG</span></span>
<span id="cb33-13"><a href="including-clinical-variables.html#cb33-13" aria-hidden="true" tabindex="-1"></a>igraph<span class="sc">::</span><span class="fu">is.dag</span>(<span class="fu">as.igraph</span>(bnCov<span class="sc">$</span>av))</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="including-clinical-variables.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit the parameter to network based on the data</span></span>
<span id="cb35-2"><a href="including-clinical-variables.html#cb35-2" aria-hidden="true" tabindex="-1"></a>bnFit <span class="ot">&lt;-</span> <span class="fu">bn.fit</span>(bnCov<span class="sc">$</span>av, bnCov<span class="sc">$</span>df)</span></code></pre></div>
<p>Plot the resulting network.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="including-clinical-variables.html#cb36-1" aria-hidden="true" tabindex="-1"></a>bnCov<span class="sc">$</span>plot</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/plotNet-1.png" width="960" /></p>
</div>
<div id="conditional-probability-query" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Conditional probability query<a href="including-clinical-variables.html#conditional-probability-query" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next we perform conditional probability queries by the bnlearn function <code>cpdist</code> to elucidate how the clinical variables affect pathway regulation. First we fit the inferred network to the original data. These are stored in the named list. <em>Logic sampling</em> is performed unless otherwise stated.</p>
<p>Perform <code>cpdist</code>, and visualize the distribution of “Molecules associated with elastic fibres” conditional on the tumor category using <code>ggdist</code>.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="including-clinical-variables.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bnlearn)</span>
<span id="cb37-2"><a href="including-clinical-variables.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)</span>
<span id="cb37-3"><a href="including-clinical-variables.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb37-4"><a href="including-clinical-variables.html#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="including-clinical-variables.html#cb37-5" aria-hidden="true" tabindex="-1"></a>candPath <span class="ot">&lt;-</span> <span class="st">&quot;Molecules associated with elastic fibres&quot;</span></span>
<span id="cb37-6"><a href="including-clinical-variables.html#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="including-clinical-variables.html#cb37-7" aria-hidden="true" tabindex="-1"></a>efz <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFit, <span class="at">nodes=</span><span class="fu">c</span>(candPath), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">0</span>))</span>
<span id="cb37-8"><a href="including-clinical-variables.html#cb37-8" aria-hidden="true" tabindex="-1"></a>efo <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFit, <span class="at">nodes=</span><span class="fu">c</span>(candPath), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">1</span>))</span>
<span id="cb37-9"><a href="including-clinical-variables.html#cb37-9" aria-hidden="true" tabindex="-1"></a>eft <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFit, <span class="at">nodes=</span><span class="fu">c</span>(candPath), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">2</span>))</span>
<span id="cb37-10"><a href="including-clinical-variables.html#cb37-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-11"><a href="including-clinical-variables.html#cb37-11" aria-hidden="true" tabindex="-1"></a>effect <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb37-12"><a href="including-clinical-variables.html#cb37-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">val =</span> <span class="fu">c</span>(efz[,<span class="dv">1</span>], efo[,<span class="dv">1</span>], eft[,<span class="dv">1</span>]),</span>
<span id="cb37-13"><a href="including-clinical-variables.html#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">stage =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;0&quot;</span>,<span class="fu">nrow</span>(efz)), <span class="fu">rep</span>(<span class="st">&quot;1&quot;</span>, <span class="fu">nrow</span>(efo)), <span class="fu">rep</span>(<span class="st">&quot;2&quot;</span>, <span class="fu">nrow</span>(eft)))</span>
<span id="cb37-14"><a href="including-clinical-variables.html#cb37-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb37-15"><a href="including-clinical-variables.html#cb37-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-16"><a href="including-clinical-variables.html#cb37-16" aria-hidden="true" tabindex="-1"></a>disMean <span class="ot">&lt;-</span> effect <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(val))</span>
<span id="cb37-17"><a href="including-clinical-variables.html#cb37-17" aria-hidden="true" tabindex="-1"></a>stageWMean <span class="ot">&lt;-</span> <span class="fu">paste0</span>(disMean<span class="sc">$</span>stage, <span class="st">&quot; (mean=&quot;</span>, <span class="fu">round</span>(disMean<span class="sc">$</span>mean,<span class="dv">3</span>), <span class="st">&quot;)&quot;</span>)</span>
<span id="cb37-18"><a href="including-clinical-variables.html#cb37-18" aria-hidden="true" tabindex="-1"></a>effect<span class="sc">$</span>stageLabel <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(stageWMean[<span class="dv">1</span>],<span class="fu">nrow</span>(efz)), <span class="fu">rep</span>(stageWMean[<span class="dv">2</span>], <span class="fu">nrow</span>(efo)), <span class="fu">rep</span>(stageWMean[<span class="dv">3</span>], <span class="fu">nrow</span>(eft)))</span>
<span id="cb37-19"><a href="including-clinical-variables.html#cb37-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-20"><a href="including-clinical-variables.html#cb37-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(effect, <span class="fu">aes</span>(<span class="at">x=</span>val, <span class="at">y=</span>stage, <span class="at">color=</span>stageLabel, <span class="at">fill=</span>stageLabel)) <span class="sc">+</span></span>
<span id="cb37-21"><a href="including-clinical-variables.html#cb37-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb37-22"><a href="including-clinical-variables.html#cb37-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb37-23"><a href="including-clinical-variables.html#cb37-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_dotsinterval</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(candPath)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/ef-1.png" width="960" /></p>
<p>How the down-regulation in “Cell Cycle Checkpoints” affects the other pathways? This time using the importance sampling method, <em>likelihood weighting</em>.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="including-clinical-variables.html#cb38-1" aria-hidden="true" tabindex="-1"></a>predNodes <span class="ot">&lt;-</span> <span class="fu">names</span>(bnFit)</span>
<span id="cb38-2"><a href="including-clinical-variables.html#cb38-2" aria-hidden="true" tabindex="-1"></a>predNodes <span class="ot">&lt;-</span> predNodes[predNodes <span class="sc">!=</span> <span class="st">&quot;Cell Cycle Checkpoints&quot;</span>]</span>
<span id="cb38-3"><a href="including-clinical-variables.html#cb38-3" aria-hidden="true" tabindex="-1"></a>maxVal <span class="ot">&lt;-</span> <span class="fu">max</span>(bnCov<span class="sc">$</span>df[candPath])</span>
<span id="cb38-4"><a href="including-clinical-variables.html#cb38-4" aria-hidden="true" tabindex="-1"></a>minVal <span class="ot">&lt;-</span> <span class="fu">min</span>(bnCov<span class="sc">$</span>df[candPath])</span>
<span id="cb38-5"><a href="including-clinical-variables.html#cb38-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-6"><a href="including-clinical-variables.html#cb38-6" aria-hidden="true" tabindex="-1"></a>lowCCC <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFit, <span class="at">nodes=</span>predNodes, <span class="at">evidence=</span><span class="fu">list</span>(<span class="st">&quot;Cell Cycle Checkpoints&quot;</span><span class="ot">=</span>minVal), <span class="at">method=</span><span class="st">&quot;lw&quot;</span>)</span>
<span id="cb38-7"><a href="including-clinical-variables.html#cb38-7" aria-hidden="true" tabindex="-1"></a>lowW <span class="ot">&lt;-</span> <span class="fu">attributes</span>(lowCCC)<span class="sc">$</span>weights</span>
<span id="cb38-8"><a href="including-clinical-variables.html#cb38-8" aria-hidden="true" tabindex="-1"></a>highCCC <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFit, <span class="at">nodes=</span>predNodes, <span class="at">evidence=</span><span class="fu">list</span>(<span class="st">&quot;Cell Cycle Checkpoints&quot;</span><span class="ot">=</span>maxVal), <span class="at">method=</span><span class="st">&quot;lw&quot;</span>)</span>
<span id="cb38-9"><a href="including-clinical-variables.html#cb38-9" aria-hidden="true" tabindex="-1"></a>highW <span class="ot">&lt;-</span> <span class="fu">attributes</span>(highCCC)<span class="sc">$</span>weights</span>
<span id="cb38-10"><a href="including-clinical-variables.html#cb38-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-11"><a href="including-clinical-variables.html#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Remove the factor</span></span>
<span id="cb38-12"><a href="including-clinical-variables.html#cb38-12" aria-hidden="true" tabindex="-1"></a>highCCC<span class="sc">$</span>Category <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb38-13"><a href="including-clinical-variables.html#cb38-13" aria-hidden="true" tabindex="-1"></a>lowCCC<span class="sc">$</span>Category <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb38-14"><a href="including-clinical-variables.html#cb38-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-15"><a href="including-clinical-variables.html#cb38-15" aria-hidden="true" tabindex="-1"></a>difMeanCCC <span class="ot">&lt;-</span> <span class="fu">apply</span>(highCCC, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">weighted.mean</span>(x, highW)) <span class="sc">-</span> <span class="fu">apply</span>(lowCCC, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">weighted.mean</span>(x, lowW))</span>
<span id="cb38-16"><a href="including-clinical-variables.html#cb38-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-17"><a href="including-clinical-variables.html#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Top absolute value</span></span>
<span id="cb38-18"><a href="including-clinical-variables.html#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(difMeanCCC[<span class="fu">order</span>(<span class="fu">abs</span>(difMeanCCC), <span class="at">decreasing=</span><span class="cn">TRUE</span>)]), <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">&quot;difference&quot;</span>))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
difference
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
M Phase
</td>
<td style="text-align:right;">
23.15520
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitotic Prometaphase
</td>
<td style="text-align:right;">
20.93115
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitotic Metaphase and Anaphase
</td>
<td style="text-align:right;">
20.70000
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitotic Anaphase
</td>
<td style="text-align:right;">
20.56985
</td>
</tr>
<tr>
<td style="text-align:left;">
Separation of Sister Chromatids
</td>
<td style="text-align:right;">
19.35587
</td>
</tr>
<tr>
<td style="text-align:left;">
Resolution of Sister Chromatid Cohesion
</td>
<td style="text-align:right;">
19.16356
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="including-clinical-variables.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Reflect the difference in the plot modifying ggplot2 object</span></span>
<span id="cb39-2"><a href="including-clinical-variables.html#cb39-2" aria-hidden="true" tabindex="-1"></a>changeCol <span class="ot">&lt;-</span> bnCov<span class="sc">$</span>plot<span class="sc">$</span>data</span>
<span id="cb39-3"><a href="including-clinical-variables.html#cb39-3" aria-hidden="true" tabindex="-1"></a>difMeanCCC <span class="ot">&lt;-</span> difMeanCCC[changeCol<span class="sc">$</span>name]</span>
<span id="cb39-4"><a href="including-clinical-variables.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(difMeanCCC) <span class="ot">&lt;-</span> changeCol<span class="sc">$</span>name</span>
<span id="cb39-5"><a href="including-clinical-variables.html#cb39-5" aria-hidden="true" tabindex="-1"></a>changeCol<span class="sc">$</span>color <span class="ot">&lt;-</span> difMeanCCC</span>
<span id="cb39-6"><a href="including-clinical-variables.html#cb39-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-7"><a href="including-clinical-variables.html#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Replace the color, change the legend</span></span>
<span id="cb39-8"><a href="including-clinical-variables.html#cb39-8" aria-hidden="true" tabindex="-1"></a>bnCov<span class="sc">$</span>plot<span class="sc">$</span>data <span class="ot">&lt;-</span> changeCol</span>
<span id="cb39-9"><a href="including-clinical-variables.html#cb39-9" aria-hidden="true" tabindex="-1"></a>bnCov<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">scale_color_continuous</span>(<span class="at">low=</span><span class="st">&quot;blue&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>,<span class="at">name=</span><span class="st">&quot;difference&quot;</span>)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/ccc-1.png" width="1440" /></p>
</div>
<div id="gene-relationship-with-variables" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Gene relationship with variables<a href="including-clinical-variables.html#gene-relationship-with-variables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the genes in interesting pathway, clinical variables can be incorporated too. We investigated the genes involved in the reactome “Molecules associated with elastic fibres”.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="including-clinical-variables.html#cb40-1" aria-hidden="true" tabindex="-1"></a>num <span class="ot">&lt;-</span> <span class="fu">which</span>(pway<span class="sc">@</span>result<span class="sc">$</span>Description<span class="sc">==</span><span class="st">&quot;Molecules associated with elastic fibres&quot;</span>)</span>
<span id="cb40-2"><a href="including-clinical-variables.html#cb40-2" aria-hidden="true" tabindex="-1"></a>bnGeneCov <span class="ot">&lt;-</span> <span class="fu">bngeneplot</span>(pway,</span>
<span id="cb40-3"><a href="including-clinical-variables.html#cb40-3" aria-hidden="true" tabindex="-1"></a>                vstedTCGA, <span class="at">pathNum=</span>num,</span>
<span id="cb40-4"><a href="including-clinical-variables.html#cb40-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">expSample=</span><span class="fu">rownames</span>(metadata),</span>
<span id="cb40-5"><a href="including-clinical-variables.html#cb40-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVar=</span>metadata,</span>
<span id="cb40-6"><a href="including-clinical-variables.html#cb40-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">hub=</span><span class="dv">5</span>, <span class="at">R=</span><span class="dv">100</span>,</span>
<span id="cb40-7"><a href="including-clinical-variables.html#cb40-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Category&quot;</span>),</span>
<span id="cb40-8"><a href="including-clinical-variables.html#cb40-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">cl=</span>parallel<span class="sc">::</span><span class="fu">makeCluster</span>(<span class="dv">10</span>),</span>
<span id="cb40-9"><a href="including-clinical-variables.html#cb40-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">returnNet=</span>T)</span></code></pre></div>
<p>Plot the resulting network of genes.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="including-clinical-variables.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot</span></span>
<span id="cb41-2"><a href="including-clinical-variables.html#cb41-2" aria-hidden="true" tabindex="-1"></a>bnGeneCov<span class="sc">$</span>plot</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/fit2-1.png" width="960" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="including-clinical-variables.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Check DAG</span></span>
<span id="cb42-2"><a href="including-clinical-variables.html#cb42-2" aria-hidden="true" tabindex="-1"></a>igraph<span class="sc">::</span><span class="fu">is.dag</span>(<span class="fu">as.igraph</span>(bnGeneCov<span class="sc">$</span>av))</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="including-clinical-variables.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit the parameter to network based on the data</span></span>
<span id="cb44-2"><a href="including-clinical-variables.html#cb44-2" aria-hidden="true" tabindex="-1"></a>bnFitGene <span class="ot">&lt;-</span> <span class="fu">bn.fit</span>(bnGeneCov<span class="sc">$</span>av, bnGeneCov<span class="sc">$</span>df)</span></code></pre></div>
<p>Perform <code>cpdist</code>, and examine the mean and distribution using <code>ggdist</code>. We can see that the expression of the gene EFEMP1, which is reported to be a candidate for a biomarker of aggressive bladder cancer or therapeutic targets <span class="citation">(<a href="#ref-Han2017-bi" role="doc-biblioref">Han et al. 2017</a>)</span>, is going up with each stage.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="including-clinical-variables.html#cb45-1" aria-hidden="true" tabindex="-1"></a>candGene <span class="ot">&lt;-</span> <span class="st">&quot;EFEMP1&quot;</span></span>
<span id="cb45-2"><a href="including-clinical-variables.html#cb45-2" aria-hidden="true" tabindex="-1"></a>efz <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFitGene, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">0</span>))</span>
<span id="cb45-3"><a href="including-clinical-variables.html#cb45-3" aria-hidden="true" tabindex="-1"></a>efo <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFitGene, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">1</span>))</span>
<span id="cb45-4"><a href="including-clinical-variables.html#cb45-4" aria-hidden="true" tabindex="-1"></a>eft <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFitGene, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">2</span>))</span>
<span id="cb45-5"><a href="including-clinical-variables.html#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="including-clinical-variables.html#cb45-6" aria-hidden="true" tabindex="-1"></a>effect <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb45-7"><a href="including-clinical-variables.html#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">val =</span> <span class="fu">c</span>(efz[,<span class="dv">1</span>], efo[,<span class="dv">1</span>], eft[,<span class="dv">1</span>]),</span>
<span id="cb45-8"><a href="including-clinical-variables.html#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">stage =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;0&quot;</span>,<span class="fu">nrow</span>(efz)), <span class="fu">rep</span>(<span class="st">&quot;1&quot;</span>, <span class="fu">nrow</span>(efo)), <span class="fu">rep</span>(<span class="st">&quot;2&quot;</span>, <span class="fu">nrow</span>(eft)))</span>
<span id="cb45-9"><a href="including-clinical-variables.html#cb45-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-10"><a href="including-clinical-variables.html#cb45-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-11"><a href="including-clinical-variables.html#cb45-11" aria-hidden="true" tabindex="-1"></a>disMean <span class="ot">&lt;-</span> effect <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(val))</span>
<span id="cb45-12"><a href="including-clinical-variables.html#cb45-12" aria-hidden="true" tabindex="-1"></a>stageWMean <span class="ot">&lt;-</span> <span class="fu">paste0</span>(disMean<span class="sc">$</span>stage, <span class="st">&quot; (mean=&quot;</span>, <span class="fu">round</span>(disMean<span class="sc">$</span>mean,<span class="dv">3</span>), <span class="st">&quot;)&quot;</span>)</span>
<span id="cb45-13"><a href="including-clinical-variables.html#cb45-13" aria-hidden="true" tabindex="-1"></a>effect<span class="sc">$</span>stageLabel <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(stageWMean[<span class="dv">1</span>],<span class="fu">nrow</span>(efz)), <span class="fu">rep</span>(stageWMean[<span class="dv">2</span>], <span class="fu">nrow</span>(efo)), <span class="fu">rep</span>(stageWMean[<span class="dv">3</span>], <span class="fu">nrow</span>(eft)))</span>
<span id="cb45-14"><a href="including-clinical-variables.html#cb45-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-15"><a href="including-clinical-variables.html#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(effect, <span class="fu">aes</span>(<span class="at">x=</span>val, <span class="at">y=</span>stage, <span class="at">color=</span>stageLabel, <span class="at">fill=</span>stageLabel)) <span class="sc">+</span></span>
<span id="cb45-16"><a href="including-clinical-variables.html#cb45-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb45-17"><a href="including-clinical-variables.html#cb45-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb45-18"><a href="including-clinical-variables.html#cb45-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_dotsinterval</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(candGene)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/efemp1-1.png" width="960" /></p>
<p>We can reflect the difference to the plot. In the previous EFEMP1 plot:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="including-clinical-variables.html#cb46-1" aria-hidden="true" tabindex="-1"></a>candGene <span class="ot">&lt;-</span> <span class="fu">names</span>(bnFitGene)</span>
<span id="cb46-2"><a href="including-clinical-variables.html#cb46-2" aria-hidden="true" tabindex="-1"></a>candGene <span class="ot">&lt;-</span> candGene[candGene <span class="sc">!=</span> <span class="st">&quot;Category&quot;</span>]</span>
<span id="cb46-3"><a href="including-clinical-variables.html#cb46-3" aria-hidden="true" tabindex="-1"></a>efz2 <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFitGene, <span class="at">nodes=</span>candGene, <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">0</span>))</span>
<span id="cb46-4"><a href="including-clinical-variables.html#cb46-4" aria-hidden="true" tabindex="-1"></a>eft2 <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnFitGene, <span class="at">nodes=</span>candGene, <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">2</span>))</span>
<span id="cb46-5"><a href="including-clinical-variables.html#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="including-clinical-variables.html#cb46-6" aria-hidden="true" tabindex="-1"></a>difMean <span class="ot">&lt;-</span> <span class="fu">apply</span>(eft2, <span class="dv">2</span>, mean) <span class="sc">-</span> <span class="fu">apply</span>(efz2, <span class="dv">2</span>, mean)</span>
<span id="cb46-7"><a href="including-clinical-variables.html#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="including-clinical-variables.html#cb46-8" aria-hidden="true" tabindex="-1"></a>changeCol <span class="ot">&lt;-</span> bnGeneCov<span class="sc">$</span>plot<span class="sc">$</span>data</span>
<span id="cb46-9"><a href="including-clinical-variables.html#cb46-9" aria-hidden="true" tabindex="-1"></a>difMean <span class="ot">&lt;-</span> difMean[changeCol<span class="sc">$</span>name]</span>
<span id="cb46-10"><a href="including-clinical-variables.html#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(difMean) <span class="ot">&lt;-</span> changeCol<span class="sc">$</span>name</span>
<span id="cb46-11"><a href="including-clinical-variables.html#cb46-11" aria-hidden="true" tabindex="-1"></a>changeCol<span class="sc">$</span>color <span class="ot">&lt;-</span> difMean</span>
<span id="cb46-12"><a href="including-clinical-variables.html#cb46-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-13"><a href="including-clinical-variables.html#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Replace shape and color</span></span>
<span id="cb46-14"><a href="including-clinical-variables.html#cb46-14" aria-hidden="true" tabindex="-1"></a>changeCol<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">19</span>, <span class="fu">dim</span>(bnGeneCov<span class="sc">$</span>plot<span class="sc">$</span>data)[<span class="dv">1</span>])</span>
<span id="cb46-15"><a href="including-clinical-variables.html#cb46-15" aria-hidden="true" tabindex="-1"></a>bnGeneCov<span class="sc">$</span>plot<span class="sc">$</span>data <span class="ot">&lt;-</span> changeCol</span>
<span id="cb46-16"><a href="including-clinical-variables.html#cb46-16" aria-hidden="true" tabindex="-1"></a>bnGeneCov<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">scale_color_continuous</span>(<span class="at">low=</span><span class="st">&quot;blue&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>,<span class="at">name=</span><span class="st">&quot;difference&quot;</span>)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/diffplot-1.png" width="960" /></p>
</div>
<div id="confirming-the-existing-knowledge" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Confirming the existing knowledge<a href="including-clinical-variables.html#confirming-the-existing-knowledge" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To confirm the validity of the inferred Bayesian network, we can focus on some genes that is already validated to be related to clinical information or is incorporated into the daily clinical practice. To obtain the pathways that include the specific gene, one can use <code>obtainPath</code> function. This time we focus on the gene MMP2, as the gene has been reported to be related to clinical variables in bladder cancer <span class="citation">(<a href="#ref-Vasala2003-jo" role="doc-biblioref">Vasala, Pääkkö, and Turpeenniemi-Hujanen 2003</a>; <a href="#ref-Fouad2019-bl" role="doc-biblioref">Fouad et al. 2019</a>; <a href="#ref-Winerdal2018-pp" role="doc-biblioref">Winerdal et al. 2018</a>)</span>.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="including-clinical-variables.html#cb47-1" aria-hidden="true" tabindex="-1"></a>pathSub <span class="ot">&lt;-</span> <span class="fu">obtainPath</span>(pway, <span class="st">&quot;MMP2&quot;</span>)</span></code></pre></div>
<p>Using the top pathway involving MMP2, construct the network and plot.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="including-clinical-variables.html#cb48-1" aria-hidden="true" tabindex="-1"></a>bnGeneCov2 <span class="ot">&lt;-</span> <span class="fu">bngeneplot</span>(pathSub,</span>
<span id="cb48-2"><a href="including-clinical-variables.html#cb48-2" aria-hidden="true" tabindex="-1"></a>                vstedTCGA, <span class="at">pathNum=</span><span class="dv">1</span>,</span>
<span id="cb48-3"><a href="including-clinical-variables.html#cb48-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">expSample=</span><span class="fu">rownames</span>(metadata),</span>
<span id="cb48-4"><a href="including-clinical-variables.html#cb48-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVar=</span>metadata,</span>
<span id="cb48-5"><a href="including-clinical-variables.html#cb48-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">hub=</span><span class="dv">5</span>, <span class="at">R=</span><span class="dv">100</span>, <span class="at">algo=</span><span class="st">&quot;hc&quot;</span>,</span>
<span id="cb48-6"><a href="including-clinical-variables.html#cb48-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Category&quot;</span>),</span>
<span id="cb48-7"><a href="including-clinical-variables.html#cb48-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">cl=</span>parallel<span class="sc">::</span><span class="fu">makeCluster</span>(<span class="dv">10</span>),</span>
<span id="cb48-8"><a href="including-clinical-variables.html#cb48-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">returnNet=</span>T)</span>
<span id="cb48-9"><a href="including-clinical-variables.html#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="including-clinical-variables.html#cb48-10" aria-hidden="true" tabindex="-1"></a>bnGeneCov2<span class="sc">$</span>plot</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/geneNet2-1.png" width="960" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="including-clinical-variables.html#cb49-1" aria-hidden="true" tabindex="-1"></a>bnGeneCov2Fit <span class="ot">&lt;-</span> <span class="fu">bn.fit</span>(bnGeneCov2<span class="sc">$</span>av, bnGeneCov2<span class="sc">$</span>df)</span></code></pre></div>
<p>Predict the distribution.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="including-clinical-variables.html#cb50-1" aria-hidden="true" tabindex="-1"></a>candGene <span class="ot">&lt;-</span> <span class="st">&quot;MMP2&quot;</span></span>
<span id="cb50-2"><a href="including-clinical-variables.html#cb50-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-3"><a href="including-clinical-variables.html#cb50-3" aria-hidden="true" tabindex="-1"></a>mz <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnGeneCov2Fit, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">0</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb50-4"><a href="including-clinical-variables.html#cb50-4" aria-hidden="true" tabindex="-1"></a>mo <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnGeneCov2Fit, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">1</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb50-5"><a href="including-clinical-variables.html#cb50-5" aria-hidden="true" tabindex="-1"></a>mt <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnGeneCov2Fit, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">2</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb50-6"><a href="including-clinical-variables.html#cb50-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-7"><a href="including-clinical-variables.html#cb50-7" aria-hidden="true" tabindex="-1"></a>effect <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb50-8"><a href="including-clinical-variables.html#cb50-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">val =</span> <span class="fu">c</span>(mz[,<span class="dv">1</span>], mo[,<span class="dv">1</span>], mt[,<span class="dv">1</span>]),</span>
<span id="cb50-9"><a href="including-clinical-variables.html#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">stage =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;0&quot;</span>,<span class="fu">nrow</span>(mz)), <span class="fu">rep</span>(<span class="st">&quot;1&quot;</span>, <span class="fu">nrow</span>(mo)), <span class="fu">rep</span>(<span class="st">&quot;2&quot;</span>, <span class="fu">nrow</span>(mt)))</span>
<span id="cb50-10"><a href="including-clinical-variables.html#cb50-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-11"><a href="including-clinical-variables.html#cb50-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-12"><a href="including-clinical-variables.html#cb50-12" aria-hidden="true" tabindex="-1"></a>disMean <span class="ot">&lt;-</span> effect <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(val))</span>
<span id="cb50-13"><a href="including-clinical-variables.html#cb50-13" aria-hidden="true" tabindex="-1"></a>stageWMean <span class="ot">&lt;-</span> <span class="fu">paste0</span>(disMean<span class="sc">$</span>stage, <span class="st">&quot; (mean=&quot;</span>, <span class="fu">round</span>(disMean<span class="sc">$</span>mean,<span class="dv">3</span>), <span class="st">&quot;)&quot;</span>)</span>
<span id="cb50-14"><a href="including-clinical-variables.html#cb50-14" aria-hidden="true" tabindex="-1"></a>effect<span class="sc">$</span>stageLabel <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(stageWMean[<span class="dv">1</span>],<span class="fu">nrow</span>(mz)), <span class="fu">rep</span>(stageWMean[<span class="dv">2</span>], <span class="fu">nrow</span>(mo)), <span class="fu">rep</span>(stageWMean[<span class="dv">3</span>], <span class="fu">nrow</span>(mt)))</span>
<span id="cb50-15"><a href="including-clinical-variables.html#cb50-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-16"><a href="including-clinical-variables.html#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(effect, <span class="fu">aes</span>(<span class="at">x=</span>val, <span class="at">y=</span>stage, <span class="at">color=</span>stageLabel, <span class="at">fill=</span>stageLabel)) <span class="sc">+</span></span>
<span id="cb50-17"><a href="including-clinical-variables.html#cb50-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb50-18"><a href="including-clinical-variables.html#cb50-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb50-19"><a href="including-clinical-variables.html#cb50-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_dotsinterval</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(candGene)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/mmp2-1.png" width="960" /></p>
<p>It is interesting to investigate whether the result is similar in the other database, like Gene Ontology doing the same analysis.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="including-clinical-variables.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Perform the same analysis on GO enrichment result</span></span>
<span id="cb51-2"><a href="including-clinical-variables.html#cb51-2" aria-hidden="true" tabindex="-1"></a>pathSubGO <span class="ot">&lt;-</span> <span class="fu">obtainPath</span>(pwayGO, <span class="st">&quot;MMP2&quot;</span>)</span>
<span id="cb51-3"><a href="including-clinical-variables.html#cb51-3" aria-hidden="true" tabindex="-1"></a>pathSubGO<span class="sc">@</span>result[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">c</span>(<span class="st">&quot;Description&quot;</span>,<span class="st">&quot;geneID&quot;</span>,<span class="st">&quot;Count&quot;</span>)]</span></code></pre></div>
<pre><code>FALSE                                   Description
FALSE GO:0001968                fibronectin binding
FALSE GO:0004222      metalloendopeptidase activity
FALSE GO:0008237          metallopeptidase activity
FALSE GO:0004252 serine-type endopeptidase activity
FALSE GO:0008236     serine-type peptidase activity
FALSE                                                                                                                                                                                                                             geneID
FALSE GO:0001968                                                                                                                                           MYOC/FSTL3/FBLN1/MMP2/CCDC80/VEGFA/CCN2/THBS1/SFRP2/EPHA1/IGFBP6/LRRC15/SSC5D
FALSE GO:0004222                                                         NRDC/MMP2/MMP11/ADAM2/COPS5/NLN/MMP19/MMP24/ADAMTS8/MMP27/MMP13/ADAM10/ECE2/ADAM33/ADAM17/ADAMTS1/ADAMTS5/ADAMTS4/MMP10/ADAMTS20/PAPPA/LMLN/MMP23B/MMP1/TRABD2B
FALSE GO:0008237 NAALAD2/NRDC/MMP2/MMP11/CPQ/ADAM2/PSMD14/COPS5/CPXM2/XPNPEP2/NLN/MMP19/STAMBP/MMP24/CPA4/ADAMTS8/MMP27/MMP13/ADAM10/ECE2/ADAM33/ADAM17/ADAMTS1/ADAMTS5/ADAMTS4/MYSM1/CPA3/MMP10/ADAMTS20/PAPPA/LMLN/MMP23B/MMP1/TRABD2B
FALSE GO:0004252                                                                             LTF/MMP2/CMA1/TPSD1/MMP11/ACR/CTSG/GZMH/GZMK/TPSG1/MMP19/PCSK2/F10/MASP1/F12/CTSV/MMP13/C1R/MMP10/KLK6/TPSAB1/C1S/MMP1/TPSB2/GZMM/ELANE/CFD
FALSE GO:0008236                                                                LTF/NAALAD2/MMP2/CMA1/TPSD1/MMP11/ACR/CTSG/GZMH/CPVL/GZMK/TPSG1/MMP19/PCSK2/F10/MASP1/F12/CTSV/MMP13/C1R/MMP10/KLK6/TPSAB1/C1S/MMP1/TPSB2/GZMM/ELANE/CFD
FALSE            Count
FALSE GO:0001968    13
FALSE GO:0004222    25
FALSE GO:0008237    34
FALSE GO:0004252    27
FALSE GO:0008236    29</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="including-clinical-variables.html#cb53-1" aria-hidden="true" tabindex="-1"></a>num <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span></span>
<span id="cb53-2"><a href="including-clinical-variables.html#cb53-2" aria-hidden="true" tabindex="-1"></a>bnCovGO <span class="ot">&lt;-</span> <span class="fu">bngeneplot</span>(pathSubGO,</span>
<span id="cb53-3"><a href="including-clinical-variables.html#cb53-3" aria-hidden="true" tabindex="-1"></a>                vstedTCGA,</span>
<span id="cb53-4"><a href="including-clinical-variables.html#cb53-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">pathNum=</span>num,</span>
<span id="cb53-5"><a href="including-clinical-variables.html#cb53-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">algo=</span><span class="st">&quot;hc&quot;</span>,</span>
<span id="cb53-6"><a href="including-clinical-variables.html#cb53-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">expSample=</span><span class="fu">rownames</span>(metadata),</span>
<span id="cb53-7"><a href="including-clinical-variables.html#cb53-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVar=</span>metadata,</span>
<span id="cb53-8"><a href="including-clinical-variables.html#cb53-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">R=</span><span class="dv">300</span>, <span class="at">layout=</span><span class="st">&quot;sugiyama&quot;</span>,</span>
<span id="cb53-9"><a href="including-clinical-variables.html#cb53-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Category&quot;</span>),</span>
<span id="cb53-10"><a href="including-clinical-variables.html#cb53-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">cl=</span>parallel<span class="sc">::</span><span class="fu">makeCluster</span>(<span class="dv">10</span>),</span>
<span id="cb53-11"><a href="including-clinical-variables.html#cb53-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">returnNet=</span><span class="cn">TRUE</span>)</span>
<span id="cb53-12"><a href="including-clinical-variables.html#cb53-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-13"><a href="including-clinical-variables.html#cb53-13" aria-hidden="true" tabindex="-1"></a>bnCovGO<span class="sc">$</span>av<span class="sc">$</span>nodes<span class="sc">$</span>Category</span></code></pre></div>
<pre><code>FALSE $mb
FALSE [1] &quot;CCDC80&quot;
FALSE 
FALSE $nbr
FALSE [1] &quot;CCDC80&quot;
FALSE 
FALSE $parents
FALSE character(0)
FALSE 
FALSE $children
FALSE [1] &quot;CCDC80&quot;</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="including-clinical-variables.html#cb55-1" aria-hidden="true" tabindex="-1"></a>bnGeneCovGOFit <span class="ot">&lt;-</span> <span class="fu">bn.fit</span>(bnCovGO<span class="sc">$</span>av, bnCovGO<span class="sc">$</span>df)</span>
<span id="cb55-2"><a href="including-clinical-variables.html#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="including-clinical-variables.html#cb55-3" aria-hidden="true" tabindex="-1"></a>mz <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnGeneCovGOFit, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">0</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb55-4"><a href="including-clinical-variables.html#cb55-4" aria-hidden="true" tabindex="-1"></a>mo <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnGeneCovGOFit, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">1</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb55-5"><a href="including-clinical-variables.html#cb55-5" aria-hidden="true" tabindex="-1"></a>mt <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnGeneCovGOFit, <span class="at">nodes=</span><span class="fu">c</span>(candGene), <span class="at">evidence=</span>(Category<span class="sc">==</span><span class="dv">2</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb55-6"><a href="including-clinical-variables.html#cb55-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-7"><a href="including-clinical-variables.html#cb55-7" aria-hidden="true" tabindex="-1"></a>effect <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb55-8"><a href="including-clinical-variables.html#cb55-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">val =</span> <span class="fu">c</span>(mz[,<span class="dv">1</span>], mo[,<span class="dv">1</span>], mt[,<span class="dv">1</span>]),</span>
<span id="cb55-9"><a href="including-clinical-variables.html#cb55-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">stage =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;0&quot;</span>,<span class="fu">nrow</span>(mz)), <span class="fu">rep</span>(<span class="st">&quot;1&quot;</span>, <span class="fu">nrow</span>(mo)), <span class="fu">rep</span>(<span class="st">&quot;2&quot;</span>, <span class="fu">nrow</span>(mt)))</span>
<span id="cb55-10"><a href="including-clinical-variables.html#cb55-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-11"><a href="including-clinical-variables.html#cb55-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-12"><a href="including-clinical-variables.html#cb55-12" aria-hidden="true" tabindex="-1"></a>disMean <span class="ot">&lt;-</span> effect <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(val))</span>
<span id="cb55-13"><a href="including-clinical-variables.html#cb55-13" aria-hidden="true" tabindex="-1"></a>stageWMean <span class="ot">&lt;-</span> <span class="fu">paste0</span>(disMean<span class="sc">$</span>stage, <span class="st">&quot; (mean=&quot;</span>, <span class="fu">round</span>(disMean<span class="sc">$</span>mean,<span class="dv">3</span>), <span class="st">&quot;)&quot;</span>)</span>
<span id="cb55-14"><a href="including-clinical-variables.html#cb55-14" aria-hidden="true" tabindex="-1"></a>effect<span class="sc">$</span>stageLabel <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(stageWMean[<span class="dv">1</span>],<span class="fu">nrow</span>(mz)), <span class="fu">rep</span>(stageWMean[<span class="dv">2</span>], <span class="fu">nrow</span>(mo)), <span class="fu">rep</span>(stageWMean[<span class="dv">3</span>], <span class="fu">nrow</span>(mt)))</span>
<span id="cb55-15"><a href="including-clinical-variables.html#cb55-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-16"><a href="including-clinical-variables.html#cb55-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(effect, <span class="fu">aes</span>(<span class="at">x=</span>val, <span class="at">y=</span>stage, <span class="at">color=</span>stageLabel, <span class="at">fill=</span>stageLabel)) <span class="sc">+</span></span>
<span id="cb55-17"><a href="including-clinical-variables.html#cb55-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb55-18"><a href="including-clinical-variables.html#cb55-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;tomato&quot;</span>)) <span class="sc">+</span></span>
<span id="cb55-19"><a href="including-clinical-variables.html#cb55-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_dotsinterval</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(candGene) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">caption=</span><span class="fu">paste</span>(pathSubGO<span class="sc">@</span>result<span class="sc">$</span>Description[num], <span class="at">collapse=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/mmp2_go-1.png" width="960" /></p>
</div>
<div id="investigating-the-network-based-on-the-clinical-question" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Investigating the network based on the clinical question<a href="including-clinical-variables.html#investigating-the-network-based-on-the-clinical-question" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After confirming the knowledge, it is interesting to test how difference in clinical variables affect gene expression. <code>bnlearn</code> can naturally handle this again using <code>cpdist</code>. We now include two more variables, <code>age_at_diagnosis</code>, <code>gender</code>, <code>paper_Noninvasive.bladder.cancer.therapy</code>, <code>paper_Combined.T.and.LN.category</code>. Inference based on these information, we can ask:</p>
<blockquote>
<p><em>Which genes have the biggest expression differences between those treated with BCG and with none, considering the networks of genes and clinical variables of age, gender, tumor category in TCGA-BLCA dataset using genes significantly differed in GSE133624 and involved in curated biological pathways related to MMP2?</em></p>
</blockquote>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="including-clinical-variables.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Subset to significant pathways, and those related to MMP2</span></span>
<span id="cb56-2"><a href="including-clinical-variables.html#cb56-2" aria-hidden="true" tabindex="-1"></a>pway<span class="sc">@</span>result <span class="ot">&lt;-</span> <span class="fu">subset</span>(pway<span class="sc">@</span>result, p.adjust<span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb56-3"><a href="including-clinical-variables.html#cb56-3" aria-hidden="true" tabindex="-1"></a>pathSub <span class="ot">&lt;-</span> <span class="fu">obtainPath</span>(pway, <span class="st">&quot;MMP2&quot;</span>)</span>
<span id="cb56-4"><a href="including-clinical-variables.html#cb56-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-5"><a href="including-clinical-variables.html#cb56-5" aria-hidden="true" tabindex="-1"></a>bnCovGene3 <span class="ot">&lt;-</span> <span class="fu">bngeneplot</span>(pathSub,</span>
<span id="cb56-6"><a href="including-clinical-variables.html#cb56-6" aria-hidden="true" tabindex="-1"></a>                vstedBCG,</span>
<span id="cb56-7"><a href="including-clinical-variables.html#cb56-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">pathNum =</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(pathSub)),</span>
<span id="cb56-8"><a href="including-clinical-variables.html#cb56-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">expSample=</span><span class="fu">rownames</span>(metadata),</span>
<span id="cb56-9"><a href="including-clinical-variables.html#cb56-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVar=</span>metadata,</span>
<span id="cb56-10"><a href="including-clinical-variables.html#cb56-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">hub=</span><span class="dv">5</span>, <span class="at">R=</span><span class="dv">50</span>,</span>
<span id="cb56-11"><a href="including-clinical-variables.html#cb56-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Gender&quot;</span>,<span class="st">&quot;Therapy&quot;</span>,<span class="st">&quot;Category&quot;</span>),</span>
<span id="cb56-12"><a href="including-clinical-variables.html#cb56-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">cl=</span>parallel<span class="sc">::</span><span class="fu">makeCluster</span>(<span class="dv">10</span>),</span>
<span id="cb56-13"><a href="including-clinical-variables.html#cb56-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">returnNet=</span>T)</span>
<span id="cb56-14"><a href="including-clinical-variables.html#cb56-14" aria-hidden="true" tabindex="-1"></a>bnCovGene3Fit <span class="ot">&lt;-</span> <span class="fu">bn.fit</span>(bnCovGene3<span class="sc">$</span>av, bnCovGene3<span class="sc">$</span>df)</span>
<span id="cb56-15"><a href="including-clinical-variables.html#cb56-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-16"><a href="including-clinical-variables.html#cb56-16" aria-hidden="true" tabindex="-1"></a>allGenes <span class="ot">&lt;-</span> <span class="fu">names</span>(bnCovGene3<span class="sc">$</span>av<span class="sc">$</span>nodes)</span>
<span id="cb56-17"><a href="including-clinical-variables.html#cb56-17" aria-hidden="true" tabindex="-1"></a>allGenes <span class="ot">&lt;-</span> allGenes[<span class="sc">!</span>allGenes <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Gender&quot;</span>,<span class="st">&quot;Therapy&quot;</span>,<span class="st">&quot;Category&quot;</span>)]</span>
<span id="cb56-18"><a href="including-clinical-variables.html#cb56-18" aria-hidden="true" tabindex="-1"></a>no <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnCovGene3Fit, <span class="at">nodes=</span>allGenes, <span class="at">evidence=</span>(Therapy<span class="sc">==</span><span class="st">&quot;none&quot;</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb56-19"><a href="including-clinical-variables.html#cb56-19" aria-hidden="true" tabindex="-1"></a>bcg <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(bnCovGene3Fit, <span class="at">nodes=</span>allGenes, <span class="at">evidence=</span>(Therapy<span class="sc">==</span><span class="st">&quot;Bacillus Calmette.Guerin (BCG)&quot;</span>), <span class="at">method=</span><span class="st">&quot;ls&quot;</span>)</span>
<span id="cb56-20"><a href="including-clinical-variables.html#cb56-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-21"><a href="including-clinical-variables.html#cb56-21" aria-hidden="true" tabindex="-1"></a>difMean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(bcg, <span class="dv">2</span>, mean)<span class="sc">-</span><span class="fu">apply</span>(no, <span class="dv">2</span>, mean))</span>
<span id="cb56-22"><a href="including-clinical-variables.html#cb56-22" aria-hidden="true" tabindex="-1"></a>difMean<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(difMean)</span>
<span id="cb56-23"><a href="including-clinical-variables.html#cb56-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(difMean) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;difference&quot;</span>,<span class="st">&quot;name&quot;</span>)</span>
<span id="cb56-24"><a href="including-clinical-variables.html#cb56-24" aria-hidden="true" tabindex="-1"></a>difMean <span class="ot">&lt;-</span> difMean[<span class="fu">order</span>(<span class="fu">abs</span>(difMean<span class="sc">$</span>difference), <span class="at">decreasing=</span>T),]</span>
<span id="cb56-25"><a href="including-clinical-variables.html#cb56-25" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(difMean, <span class="at">n=</span><span class="dv">5</span>))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
difference
</th>
<th style="text-align:left;">
name
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
COL25A1
</td>
<td style="text-align:right;">
0.2344216
</td>
<td style="text-align:left;">
COL25A1
</td>
</tr>
<tr>
<td style="text-align:left;">
MFAP5
</td>
<td style="text-align:right;">
0.1834830
</td>
<td style="text-align:left;">
MFAP5
</td>
</tr>
<tr>
<td style="text-align:left;">
COL11A1
</td>
<td style="text-align:right;">
0.1494830
</td>
<td style="text-align:left;">
COL11A1
</td>
</tr>
<tr>
<td style="text-align:left;">
COL10A1
</td>
<td style="text-align:right;">
0.1381034
</td>
<td style="text-align:left;">
COL10A1
</td>
</tr>
<tr>
<td style="text-align:left;">
LAMC2
</td>
<td style="text-align:right;">
-0.1340511
</td>
<td style="text-align:left;">
LAMC2
</td>
</tr>
</tbody>
</table>
</div>
<div id="classification-using-bn" class="section level2 hasAnchor" number="5.7">
<h2><span class="header-section-number">5.7</span> Classification using BN<a href="including-clinical-variables.html#classification-using-bn" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Inferred BN can be used as a classifier of conditions. In this analysis, we perform the classification of whether the cancer samples are harboring TP53 mutation or not (column <code>paper_mutation in TP53</code>). First, we make a metadata table as same as the above examples.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="including-clinical-variables.html#cb57-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tcgaData<span class="sc">@</span>colData) <span class="sc">%&gt;%</span></span>
<span id="cb57-2"><a href="including-clinical-variables.html#cb57-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(age_at_diagnosis, gender, paper_mutation.in.TP53, paper_Combined.T.and.LN.category) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-3"><a href="including-clinical-variables.html#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(paper_mutation.in.TP53<span class="sc">!=</span><span class="st">&quot;ND&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-4"><a href="including-clinical-variables.html#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(paper_Combined.T.and.LN.category<span class="sc">!=</span><span class="st">&quot;ND&quot;</span>)</span>
<span id="cb57-5"><a href="including-clinical-variables.html#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(metadata<span class="sc">$</span>paper_mutation.in.TP53)</span></code></pre></div>
<pre><code>FALSE 
FALSE  no yes 
FALSE 184 163</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="including-clinical-variables.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Set TP53 status to numeric of 0/1.</span></span>
<span id="cb59-2"><a href="including-clinical-variables.html#cb59-2" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>paper_mutation.in.TP53 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(metadata<span class="sc">$</span>paper_mutation.in.TP53))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb59-3"><a href="including-clinical-variables.html#cb59-3" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>age_at_diagnosis <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">scale</span>(metadata<span class="sc">$</span>age_at_diagnosis))</span>
<span id="cb59-4"><a href="including-clinical-variables.html#cb59-4" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>paper_Combined.T.and.LN.category <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(metadata<span class="sc">$</span>paper_Combined.T.and.LN.category)</span>
<span id="cb59-5"><a href="including-clinical-variables.html#cb59-5" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>gender <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(metadata<span class="sc">$</span>gender)</span></code></pre></div>
<p>Split the data to train/test according to TP53 mutation status using <code>caret</code> <span class="citation">(<a href="#ref-Kuhn2008-cc" role="doc-biblioref">Kuhn 2008</a>)</span>. In this analysis, the five-fold cross validation (stratified) is performed. Fit the model using the expression of genes in the pathway. This time the classification performance of significant pathways (corrected p &lt; <code>1e-5</code>) are to be compared. <code>onlyDf</code> option can be enabled to return only the data.frame containing data for prediction, useful for testing purpose, and using the resulting data for the other <a href="https://sign.hgc.jp/">softwares</a>.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="including-clinical-variables.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">53</span>) <span class="co"># Seed for split</span></span>
<span id="cb60-2"><a href="including-clinical-variables.html#cb60-2" aria-hidden="true" tabindex="-1"></a>trainIndex <span class="ot">&lt;-</span> caret<span class="sc">::</span><span class="fu">createFolds</span>(<span class="fu">factor</span>(metadata<span class="sc">$</span>paper_mutation.in.TP53), <span class="at">k =</span> <span class="dv">5</span>, <span class="at">list =</span> <span class="cn">TRUE</span>, <span class="at">returnTrain=</span><span class="cn">TRUE</span>)</span>
<span id="cb60-3"><a href="including-clinical-variables.html#cb60-3" aria-hidden="true" tabindex="-1"></a>allnets <span class="ot">&lt;-</span> <span class="fu">list</span>() <span class="do">## Store network in the list</span></span>
<span id="cb60-4"><a href="including-clinical-variables.html#cb60-4" aria-hidden="true" tabindex="-1"></a>allClassRes <span class="ot">&lt;-</span> <span class="fu">list</span>() <span class="do">## Store prediction in the list</span></span>
<span id="cb60-5"><a href="including-clinical-variables.html#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="including-clinical-variables.html#cb60-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Already VSTed DF</span></span>
<span id="cb60-7"><a href="including-clinical-variables.html#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;trainDf.rda&quot;</span>)</span>
<span id="cb60-8"><a href="including-clinical-variables.html#cb60-8" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;testDf.rda&quot;</span>)</span>
<span id="cb60-9"><a href="including-clinical-variables.html#cb60-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-10"><a href="including-clinical-variables.html#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (f <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="dv">5</span>)) {</span>
<span id="cb60-11"><a href="including-clinical-variables.html#cb60-11" aria-hidden="true" tabindex="-1"></a>  nets <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb60-12"><a href="including-clinical-variables.html#cb60-12" aria-hidden="true" tabindex="-1"></a>  classRes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb60-13"><a href="including-clinical-variables.html#cb60-13" aria-hidden="true" tabindex="-1"></a>  foldTrainIndex <span class="ot">&lt;-</span> trainIndex[[f]]</span>
<span id="cb60-14"><a href="including-clinical-variables.html#cb60-14" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Recursively fit and test for significant pathways</span></span>
<span id="cb60-15"><a href="including-clinical-variables.html#cb60-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pnum <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">dim</span>(<span class="fu">subset</span>(pway<span class="sc">@</span>result, p.adjust<span class="sc">&lt;</span><span class="fl">1e-5</span>))[<span class="dv">1</span>], <span class="dv">1</span>)) {</span>
<span id="cb60-16"><a href="including-clinical-variables.html#cb60-16" aria-hidden="true" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(<span class="dv">12</span>)</span>
<span id="cb60-17"><a href="including-clinical-variables.html#cb60-17" aria-hidden="true" tabindex="-1"></a>      bnCovTrain <span class="ot">&lt;-</span> <span class="fu">bngeneplot</span>(pway, <span class="fu">assay</span>(trainDf[[f]]), <span class="at">pathNum=</span>pnum, <span class="at">layout=</span><span class="st">&quot;sugiyama&quot;</span>,</span>
<span id="cb60-18"><a href="including-clinical-variables.html#cb60-18" aria-hidden="true" tabindex="-1"></a>                               <span class="at">expSample=</span><span class="fu">rownames</span>(metadata[foldTrainIndex,]), <span class="at">algo=</span><span class="st">&quot;hc&quot;</span>, <span class="at">strType=</span><span class="st">&quot;normal&quot;</span>,</span>
<span id="cb60-19"><a href="including-clinical-variables.html#cb60-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">otherVar=</span>metadata[foldTrainIndex,], <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Gender&quot;</span>,<span class="st">&quot;TP53&quot;</span>,<span class="st">&quot;Category&quot;</span>),</span>
<span id="cb60-20"><a href="including-clinical-variables.html#cb60-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">R=</span><span class="dv">50</span>, <span class="at">cl=</span>cl, <span class="at">returnNet=</span>T)</span>
<span id="cb60-21"><a href="including-clinical-variables.html#cb60-21" aria-hidden="true" tabindex="-1"></a>      <span class="do">## Return only DF for testing</span></span>
<span id="cb60-22"><a href="including-clinical-variables.html#cb60-22" aria-hidden="true" tabindex="-1"></a>      bnCovTest <span class="ot">&lt;-</span> <span class="fu">bngeneplot</span>(pway, <span class="fu">assay</span>(testDf[[f]]), <span class="at">pathNum=</span>pnum,</span>
<span id="cb60-23"><a href="including-clinical-variables.html#cb60-23" aria-hidden="true" tabindex="-1"></a>                               <span class="at">expSample=</span><span class="fu">rownames</span>(metadata[<span class="sc">-</span>foldTrainIndex,]),</span>
<span id="cb60-24"><a href="including-clinical-variables.html#cb60-24" aria-hidden="true" tabindex="-1"></a>                               <span class="at">otherVar=</span>metadata[<span class="sc">-</span>foldTrainIndex,],  <span class="at">otherVarName=</span><span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Gender&quot;</span>,<span class="st">&quot;TP53&quot;</span>,<span class="st">&quot;Category&quot;</span>),</span>
<span id="cb60-25"><a href="including-clinical-variables.html#cb60-25" aria-hidden="true" tabindex="-1"></a>                               <span class="at">onlyDf=</span>T)</span>
<span id="cb60-26"><a href="including-clinical-variables.html#cb60-26" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb60-27"><a href="including-clinical-variables.html#cb60-27" aria-hidden="true" tabindex="-1"></a>      <span class="do">## If DAG and TP53 have parents</span></span>
<span id="cb60-28"><a href="including-clinical-variables.html#cb60-28" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> ( igraph<span class="sc">::</span><span class="fu">is.dag</span>(bnlearn<span class="sc">::</span><span class="fu">as.igraph</span>(bnCovTrain<span class="sc">$</span>av)) <span class="sc">&amp;&amp;</span> <span class="fu">length</span>(bnCovTrain<span class="sc">$</span>av<span class="sc">$</span>nodes<span class="sc">$</span>TP53<span class="sc">$</span>parents) <span class="sc">&gt;=</span> <span class="dv">1</span> ){</span>
<span id="cb60-29"><a href="including-clinical-variables.html#cb60-29" aria-hidden="true" tabindex="-1"></a>        bnCovLargeFit <span class="ot">&lt;-</span> bnlearn<span class="sc">::</span><span class="fu">bn.fit</span>(bnCovTrain<span class="sc">$</span>av, bnCovTrain<span class="sc">$</span>df)</span>
<span id="cb60-30"><a href="including-clinical-variables.html#cb60-30" aria-hidden="true" tabindex="-1"></a>        pred <span class="ot">&lt;-</span> sigmoid<span class="sc">::</span><span class="fu">sigmoid</span>(<span class="fu">predict</span>(bnCovLargeFit, <span class="at">node=</span><span class="st">&quot;TP53&quot;</span>, <span class="at">data=</span>bnCovTest, <span class="at">method =</span> <span class="st">&quot;bayes-lw&quot;</span>)) <span class="co"># Use sigmoid function</span></span>
<span id="cb60-31"><a href="including-clinical-variables.html#cb60-31" aria-hidden="true" tabindex="-1"></a>        classRes[[pway<span class="sc">@</span>result<span class="sc">$</span>Description[pnum]]] <span class="ot">&lt;-</span> pred</span>
<span id="cb60-32"><a href="including-clinical-variables.html#cb60-32" aria-hidden="true" tabindex="-1"></a>        nets[[pway<span class="sc">@</span>result<span class="sc">$</span>Description[pnum]]] <span class="ot">&lt;-</span> bnCovTrain</span>
<span id="cb60-33"><a href="including-clinical-variables.html#cb60-33" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb60-34"><a href="including-clinical-variables.html#cb60-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">message</span>(<span class="fu">paste0</span>(<span class="st">&quot;Among pathway &quot;</span>, pway<span class="sc">@</span>result<span class="sc">$</span>Description[pnum], <span class="st">&quot;, no parent node of TP53 is found, or inferred network is not dag.&quot;</span>))</span>
<span id="cb60-35"><a href="including-clinical-variables.html#cb60-35" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb60-36"><a href="including-clinical-variables.html#cb60-36" aria-hidden="true" tabindex="-1"></a>      parallel<span class="sc">::</span><span class="fu">stopCluster</span>(cl)    </span>
<span id="cb60-37"><a href="including-clinical-variables.html#cb60-37" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb60-38"><a href="including-clinical-variables.html#cb60-38" aria-hidden="true" tabindex="-1"></a>  allnets[[f]] <span class="ot">&lt;-</span> nets</span>
<span id="cb60-39"><a href="including-clinical-variables.html#cb60-39" aria-hidden="true" tabindex="-1"></a>  allClassRes[[f]] <span class="ot">&lt;-</span> classRes</span>
<span id="cb60-40"><a href="including-clinical-variables.html#cb60-40" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Using the library <code>pROC</code>, calculate the area under ROC (auROC) <span class="citation">(<a href="#ref-Robin2011-mk" role="doc-biblioref">Robin et al. 2011</a>)</span>.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="including-clinical-variables.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pROC)</span>
<span id="cb61-2"><a href="including-clinical-variables.html#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplotify)</span>
<span id="cb61-3"><a href="including-clinical-variables.html#cb61-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-4"><a href="including-clinical-variables.html#cb61-4" aria-hidden="true" tabindex="-1"></a>rocDf <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb61-5"><a href="including-clinical-variables.html#cb61-5" aria-hidden="true" tabindex="-1"></a>allRocList <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb61-6"><a href="including-clinical-variables.html#cb61-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (f <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="dv">5</span>)) {</span>
<span id="cb61-7"><a href="including-clinical-variables.html#cb61-7" aria-hidden="true" tabindex="-1"></a>  correct <span class="ot">&lt;-</span> metadata[<span class="sc">-</span>trainIndex[[f]],]<span class="sc">$</span>paper_mutation.in.TP53</span>
<span id="cb61-8"><a href="including-clinical-variables.html#cb61-8" aria-hidden="true" tabindex="-1"></a>  predDf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(allClassRes[[f]])</span>
<span id="cb61-9"><a href="including-clinical-variables.html#cb61-9" aria-hidden="true" tabindex="-1"></a>  predDf<span class="sc">$</span>label <span class="ot">&lt;-</span> correct</span>
<span id="cb61-10"><a href="including-clinical-variables.html#cb61-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-11"><a href="including-clinical-variables.html#cb61-11" aria-hidden="true" tabindex="-1"></a>  rocList <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb61-12"><a href="including-clinical-variables.html#cb61-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">dim</span>(predDf)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span>)){</span>
<span id="cb61-13"><a href="including-clinical-variables.html#cb61-13" aria-hidden="true" tabindex="-1"></a>    rocList[[<span class="fu">names</span>(predDf)[i]]] <span class="ot">&lt;-</span> <span class="fu">roc</span>(predDf<span class="sc">$</span>label, predDf[,i], <span class="at">ci=</span><span class="cn">TRUE</span>, <span class="at">direction=</span><span class="st">&quot;&lt;&quot;</span>) <span class="co"># check direction</span></span>
<span id="cb61-14"><a href="including-clinical-variables.html#cb61-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb61-15"><a href="including-clinical-variables.html#cb61-15" aria-hidden="true" tabindex="-1"></a>  tmpRocDf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">data.frame</span>(purrr<span class="sc">::</span><span class="fu">map</span>(rocList, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(x<span class="sc">$</span>auc)))))</span>
<span id="cb61-16"><a href="including-clinical-variables.html#cb61-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmpRocDf) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;auc&quot;</span>,f))</span>
<span id="cb61-17"><a href="including-clinical-variables.html#cb61-17" aria-hidden="true" tabindex="-1"></a>  tmpRocDf<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tmpRocDf)</span>
<span id="cb61-18"><a href="including-clinical-variables.html#cb61-18" aria-hidden="true" tabindex="-1"></a>  allRocList[[f]] <span class="ot">&lt;-</span> tmpRocDf</span>
<span id="cb61-19"><a href="including-clinical-variables.html#cb61-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-20"><a href="including-clinical-variables.html#cb61-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-21"><a href="including-clinical-variables.html#cb61-21" aria-hidden="true" tabindex="-1"></a>allRocListDf <span class="ot">&lt;-</span> allRocList <span class="sc">%&gt;%</span></span>
<span id="cb61-22"><a href="including-clinical-variables.html#cb61-22" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">reduce</span>(left_join, <span class="at">by =</span> <span class="st">&quot;name&quot;</span>)</span>
<span id="cb61-23"><a href="including-clinical-variables.html#cb61-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-24"><a href="including-clinical-variables.html#cb61-24" aria-hidden="true" tabindex="-1"></a>rocMean <span class="ot">&lt;-</span> allRocListDf <span class="sc">%&gt;%</span></span>
<span id="cb61-25"><a href="including-clinical-variables.html#cb61-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-26"><a href="including-clinical-variables.html#cb61-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Min =</span> <span class="fu">min</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="st">&quot;auc&quot;</span>)), <span class="at">na.rm=</span>T),</span>
<span id="cb61-27"><a href="including-clinical-variables.html#cb61-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Max =</span> <span class="fu">max</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="st">&quot;auc&quot;</span>)), <span class="at">na.rm=</span>T),</span>
<span id="cb61-28"><a href="including-clinical-variables.html#cb61-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Mean =</span> <span class="fu">mean</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="st">&quot;auc&quot;</span>)), <span class="at">na.rm=</span>T),</span>
<span id="cb61-29"><a href="including-clinical-variables.html#cb61-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sd =</span> <span class="fu">sd</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="st">&quot;auc&quot;</span>)), <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb61-30"><a href="including-clinical-variables.html#cb61-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name ,Mean, Sd) <span class="sc">%&gt;%</span></span>
<span id="cb61-31"><a href="including-clinical-variables.html#cb61-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Mean))</span>
<span id="cb61-32"><a href="including-clinical-variables.html#cb61-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-33"><a href="including-clinical-variables.html#cb61-33" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(rocMean, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">booktab=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-34"><a href="including-clinical-variables.html#cb61-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">font_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<table class="table" style="font-size: 10px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
name
</th>
<th style="text-align:right;">
Mean
</th>
<th style="text-align:right;">
Sd
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNA.strand.elongation
</td>
<td style="text-align:right;">
0.7586171
</td>
<td style="text-align:right;">
0.0476902
</td>
</tr>
<tr>
<td style="text-align:left;">
S.Phase
</td>
<td style="text-align:right;">
0.7442158
</td>
<td style="text-align:right;">
0.0417892
</td>
</tr>
<tr>
<td style="text-align:left;">
RHO.GTPases.Activate.Formins
</td>
<td style="text-align:right;">
0.7361936
</td>
<td style="text-align:right;">
0.0372955
</td>
</tr>
<tr>
<td style="text-align:left;">
Activation.of.ATR.in.response.to.replication.stress
</td>
<td style="text-align:right;">
0.7313293
</td>
<td style="text-align:right;">
0.0309443
</td>
</tr>
<tr>
<td style="text-align:left;">
Amplification..of.signal.from.unattached..kinetochores.via.a.MAD2..inhibitory.signal
</td>
<td style="text-align:right;">
0.7298620
</td>
<td style="text-align:right;">
0.0356467
</td>
</tr>
<tr>
<td style="text-align:left;">
Resolution.of.Sister.Chromatid.Cohesion
</td>
<td style="text-align:right;">
0.7245308
</td>
<td style="text-align:right;">
0.0627438
</td>
</tr>
<tr>
<td style="text-align:left;">
Homologous.DNA.Pairing.and.Strand.Exchange
</td>
<td style="text-align:right;">
0.7236192
</td>
<td style="text-align:right;">
0.0361203
</td>
</tr>
<tr>
<td style="text-align:left;">
HDR.through.Homologous.Recombination..HRR.
</td>
<td style="text-align:right;">
0.7108159
</td>
<td style="text-align:right;">
0.0408204
</td>
</tr>
<tr>
<td style="text-align:left;">
EML4.and.NUDC.in.mitotic.spindle.formation
</td>
<td style="text-align:right;">
0.6847067
</td>
<td style="text-align:right;">
0.0969711
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitotic.Spindle.Checkpoint
</td>
<td style="text-align:right;">
0.6727978
</td>
<td style="text-align:right;">
0.0843631
</td>
</tr>
<tr>
<td style="text-align:left;">
Separation.of.Sister.Chromatids
</td>
<td style="text-align:right;">
0.5731690
</td>
<td style="text-align:right;">
0.1567807
</td>
</tr>
</tbody>
</table>
<p>Show the resulting network with the top auROC, and the ROC plot using <code>pROC</code> <span class="citation">(<a href="#ref-Robin2011-mk" role="doc-biblioref">Robin et al. 2011</a>)</span>.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="including-clinical-variables.html#cb62-1" aria-hidden="true" tabindex="-1"></a>topPath <span class="ot">&lt;-</span> rocMean[<span class="dv">1</span>,<span class="st">&quot;name&quot;</span>]</span>
<span id="cb62-2"><a href="including-clinical-variables.html#cb62-2" aria-hidden="true" tabindex="-1"></a>candFold <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(allRocListDf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name <span class="sc">==</span> <span class="fu">as.character</span>(topPath)) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">which.max</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="st">&quot;auc&quot;</span>)))))</span>
<span id="cb62-3"><a href="including-clinical-variables.html#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="including-clinical-variables.html#cb62-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-5"><a href="including-clinical-variables.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="do">## With the CI</span></span>
<span id="cb62-6"><a href="including-clinical-variables.html#cb62-6" aria-hidden="true" tabindex="-1"></a>candRoc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(metadata[<span class="sc">-</span>trainIndex[[candFold]],]<span class="sc">$</span>paper_mutation.in.TP53)</span>
<span id="cb62-7"><a href="including-clinical-variables.html#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(candRoc) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;label&quot;</span>)</span>
<span id="cb62-8"><a href="including-clinical-variables.html#cb62-8" aria-hidden="true" tabindex="-1"></a>candRoc<span class="sc">$</span>pred <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">unlist</span>(allClassRes[[candFold]][<span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="fu">as.character</span>(topPath))]))</span>
<span id="cb62-9"><a href="including-clinical-variables.html#cb62-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-10"><a href="including-clinical-variables.html#cb62-10" aria-hidden="true" tabindex="-1"></a>rocobj1 <span class="ot">&lt;-</span> <span class="fu">plot.roc</span>(candRoc<span class="sc">$</span>label, candRoc<span class="sc">$</span>pred, <span class="at">print.auc =</span> <span class="cn">TRUE</span>, <span class="at">ci=</span><span class="cn">TRUE</span>, <span class="at">col=</span><span class="st">&quot;black&quot;</span>, <span class="at">direction=</span><span class="st">&quot;&lt;&quot;</span>,</span>
<span id="cb62-11"><a href="including-clinical-variables.html#cb62-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">main =</span> <span class="st">&quot;Mutation in TP53&quot;</span>, <span class="at">percent=</span><span class="cn">TRUE</span>)</span>
<span id="cb62-12"><a href="including-clinical-variables.html#cb62-12" aria-hidden="true" tabindex="-1"></a>ciobj <span class="ot">&lt;-</span> <span class="fu">ci.se</span>(rocobj1, <span class="at">specificities =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="dv">5</span>), <span class="at">progress=</span><span class="st">&quot;none&quot;</span>) </span>
<span id="cb62-13"><a href="including-clinical-variables.html#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ciobj, <span class="at">type =</span> <span class="st">&quot;shape&quot;</span>, <span class="at">col =</span> <span class="st">&quot;steelblue&quot;</span>)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/cl_plot-1.png" width="1920" /></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="including-clinical-variables.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Along with the network</span></span>
<span id="cb63-2"><a href="including-clinical-variables.html#cb63-2" aria-hidden="true" tabindex="-1"></a>topNet <span class="ot">&lt;-</span> allnets[[candFold]][[<span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot; &quot;</span>, topPath)]]</span>
<span id="cb63-3"><a href="including-clinical-variables.html#cb63-3" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/cl_plot-2.png" width="1920" /></p>
<p>Using <code>bnlearn::cpdist</code>, check the difference in the distribution mean when the value of the node TP53 is above and below 0.5.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="including-clinical-variables.html#cb64-1" aria-hidden="true" tabindex="-1"></a>topFit <span class="ot">&lt;-</span> bnlearn<span class="sc">::</span><span class="fu">bn.fit</span>(topNet<span class="sc">$</span>av, topNet<span class="sc">$</span>df)</span>
<span id="cb64-2"><a href="including-clinical-variables.html#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="including-clinical-variables.html#cb64-3" aria-hidden="true" tabindex="-1"></a>candNodes <span class="ot">&lt;-</span> <span class="fu">names</span>(topNet<span class="sc">$</span>av<span class="sc">$</span>nodes)</span>
<span id="cb64-4"><a href="including-clinical-variables.html#cb64-4" aria-hidden="true" tabindex="-1"></a>candNodes <span class="ot">&lt;-</span> candNodes[<span class="sc">!</span>candNodes <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>,<span class="st">&quot;Category&quot;</span>,<span class="st">&quot;Gender&quot;</span>)]</span>
<span id="cb64-5"><a href="including-clinical-variables.html#cb64-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-6"><a href="including-clinical-variables.html#cb64-6" aria-hidden="true" tabindex="-1"></a>tp53low <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(topFit, <span class="at">nodes=</span>candNodes, <span class="at">evidence=</span>(TP53 <span class="sc">&lt;</span> <span class="fl">0.5</span>))</span>
<span id="cb64-7"><a href="including-clinical-variables.html#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tp53low)</span></code></pre></div>
<pre><code>FALSE [1] 5225   21</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="including-clinical-variables.html#cb66-1" aria-hidden="true" tabindex="-1"></a>tp53high <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(topFit, <span class="at">nodes=</span>candNodes, <span class="at">evidence=</span>(TP53 <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb66-2"><a href="including-clinical-variables.html#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tp53high)</span></code></pre></div>
<pre><code>FALSE [1] 4717   21</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="including-clinical-variables.html#cb68-1" aria-hidden="true" tabindex="-1"></a>difMeanTp53 <span class="ot">&lt;-</span> <span class="fu">apply</span>(tp53high, <span class="dv">2</span>, mean) <span class="sc">-</span> <span class="fu">apply</span>(tp53low, <span class="dv">2</span>, mean)</span>
<span id="cb68-2"><a href="including-clinical-variables.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(difMeanTp53[<span class="fu">order</span>(<span class="fu">abs</span>(difMeanTp53), <span class="at">decreasing=</span>T)]), <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">&quot;difference&quot;</span>))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
difference
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MCM2
</td>
<td style="text-align:right;">
0.3978904
</td>
</tr>
<tr>
<td style="text-align:left;">
RFC4
</td>
<td style="text-align:right;">
0.3534718
</td>
</tr>
<tr>
<td style="text-align:left;">
MCM8
</td>
<td style="text-align:right;">
0.3250150
</td>
</tr>
<tr>
<td style="text-align:left;">
PRIM1
</td>
<td style="text-align:right;">
0.3243060
</td>
</tr>
<tr>
<td style="text-align:left;">
PRIM2
</td>
<td style="text-align:right;">
0.3031842
</td>
</tr>
<tr>
<td style="text-align:left;">
GINS4
</td>
<td style="text-align:right;">
0.2868035
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="including-clinical-variables.html#cb69-1" aria-hidden="true" tabindex="-1"></a>changeCol <span class="ot">&lt;-</span> topNet<span class="sc">$</span>plot<span class="sc">$</span>data</span>
<span id="cb69-2"><a href="including-clinical-variables.html#cb69-2" aria-hidden="true" tabindex="-1"></a>difMeanTp53 <span class="ot">&lt;-</span> difMeanTp53[changeCol<span class="sc">$</span>name]</span>
<span id="cb69-3"><a href="including-clinical-variables.html#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(difMeanTp53) <span class="ot">&lt;-</span> changeCol<span class="sc">$</span>name</span>
<span id="cb69-4"><a href="including-clinical-variables.html#cb69-4" aria-hidden="true" tabindex="-1"></a>changeCol<span class="sc">$</span>color <span class="ot">&lt;-</span> difMeanTp53</span>
<span id="cb69-5"><a href="including-clinical-variables.html#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="including-clinical-variables.html#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Replace shape and color</span></span>
<span id="cb69-7"><a href="including-clinical-variables.html#cb69-7" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot<span class="sc">$</span>data <span class="ot">&lt;-</span> changeCol</span>
<span id="cb69-8"><a href="including-clinical-variables.html#cb69-8" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">scale_color_continuous</span>(<span class="at">low=</span><span class="st">&quot;blue&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>,<span class="at">name=</span><span class="st">&quot;difference&quot;</span>)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/tp53difplot-1.png" width="1920" /></p>
<p>When the TP53 takes the extreme values on <em>logic sampling</em>.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="including-clinical-variables.html#cb70-1" aria-hidden="true" tabindex="-1"></a>topFit <span class="ot">&lt;-</span> bnlearn<span class="sc">::</span><span class="fu">bn.fit</span>(topNet<span class="sc">$</span>av, topNet<span class="sc">$</span>df)</span>
<span id="cb70-2"><a href="including-clinical-variables.html#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="including-clinical-variables.html#cb70-3" aria-hidden="true" tabindex="-1"></a>candNodes <span class="ot">&lt;-</span> <span class="fu">names</span>(topNet<span class="sc">$</span>av<span class="sc">$</span>nodes)</span>
<span id="cb70-4"><a href="including-clinical-variables.html#cb70-4" aria-hidden="true" tabindex="-1"></a>candNodes <span class="ot">&lt;-</span> candNodes[<span class="sc">!</span>candNodes <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>,<span class="st">&quot;Category&quot;</span>,<span class="st">&quot;Gender&quot;</span>)]</span>
<span id="cb70-5"><a href="including-clinical-variables.html#cb70-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-6"><a href="including-clinical-variables.html#cb70-6" aria-hidden="true" tabindex="-1"></a>tp53low <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(topFit, <span class="at">nodes=</span>candNodes, <span class="at">evidence=</span>(TP53 <span class="sc">&lt;</span> <span class="fl">0.01</span>))</span>
<span id="cb70-7"><a href="including-clinical-variables.html#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tp53low)</span></code></pre></div>
<pre><code>FALSE [1] 1760   21</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="including-clinical-variables.html#cb72-1" aria-hidden="true" tabindex="-1"></a>tp53high <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(topFit, <span class="at">nodes=</span>candNodes, <span class="at">evidence=</span>(TP53 <span class="sc">&gt;</span> <span class="fl">0.99</span>))</span>
<span id="cb72-2"><a href="including-clinical-variables.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tp53high)</span></code></pre></div>
<pre><code>FALSE [1] 1552   21</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="including-clinical-variables.html#cb74-1" aria-hidden="true" tabindex="-1"></a>difMeanTp53 <span class="ot">&lt;-</span> <span class="fu">apply</span>(tp53high, <span class="dv">2</span>, mean) <span class="sc">-</span> <span class="fu">apply</span>(tp53low, <span class="dv">2</span>, mean)</span>
<span id="cb74-2"><a href="including-clinical-variables.html#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(difMeanTp53[<span class="fu">order</span>(<span class="fu">abs</span>(difMeanTp53), <span class="at">decreasing=</span>T)]), <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">&quot;difference&quot;</span>))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
difference
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MCM2
</td>
<td style="text-align:right;">
0.7375946
</td>
</tr>
<tr>
<td style="text-align:left;">
RFC4
</td>
<td style="text-align:right;">
0.6955627
</td>
</tr>
<tr>
<td style="text-align:left;">
MCM8
</td>
<td style="text-align:right;">
0.5989058
</td>
</tr>
<tr>
<td style="text-align:left;">
PRIM1
</td>
<td style="text-align:right;">
0.5895650
</td>
</tr>
<tr>
<td style="text-align:left;">
PRIM2
</td>
<td style="text-align:right;">
0.5783348
</td>
</tr>
<tr>
<td style="text-align:left;">
GINS1
</td>
<td style="text-align:right;">
0.4874811
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="including-clinical-variables.html#cb75-1" aria-hidden="true" tabindex="-1"></a>changeCol <span class="ot">&lt;-</span> topNet<span class="sc">$</span>plot<span class="sc">$</span>data</span>
<span id="cb75-2"><a href="including-clinical-variables.html#cb75-2" aria-hidden="true" tabindex="-1"></a>difMeanTp53 <span class="ot">&lt;-</span> difMeanTp53[changeCol<span class="sc">$</span>name]</span>
<span id="cb75-3"><a href="including-clinical-variables.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(difMeanTp53) <span class="ot">&lt;-</span> changeCol<span class="sc">$</span>name</span>
<span id="cb75-4"><a href="including-clinical-variables.html#cb75-4" aria-hidden="true" tabindex="-1"></a>changeCol<span class="sc">$</span>color <span class="ot">&lt;-</span> difMeanTp53</span>
<span id="cb75-5"><a href="including-clinical-variables.html#cb75-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-6"><a href="including-clinical-variables.html#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Replace shape and color</span></span>
<span id="cb75-7"><a href="including-clinical-variables.html#cb75-7" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot<span class="sc">$</span>data <span class="ot">&lt;-</span> changeCol</span>
<span id="cb75-8"><a href="including-clinical-variables.html#cb75-8" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">scale_color_continuous</span>(<span class="at">low=</span><span class="st">&quot;blue&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>,<span class="at">name=</span><span class="st">&quot;difference&quot;</span>)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/tp53difplotextreme-1.png" width="1920" /></p>
<p>Perform <em>likelihood weighting</em>.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="including-clinical-variables.html#cb76-1" aria-hidden="true" tabindex="-1"></a>tp53low <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(topFit, <span class="at">nodes=</span>candNodes, <span class="at">evidence=</span><span class="fu">list</span>(<span class="at">TP53 =</span> <span class="dv">0</span>), <span class="at">method=</span><span class="st">&quot;lw&quot;</span>)</span>
<span id="cb76-2"><a href="including-clinical-variables.html#cb76-2" aria-hidden="true" tabindex="-1"></a>tp53high <span class="ot">&lt;-</span> <span class="fu">cpdist</span>(topFit, <span class="at">nodes=</span>candNodes, <span class="at">evidence=</span><span class="fu">list</span>(<span class="at">TP53 =</span> <span class="dv">1</span>), <span class="at">method=</span><span class="st">&quot;lw&quot;</span>)</span>
<span id="cb76-3"><a href="including-clinical-variables.html#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="including-clinical-variables.html#cb76-4" aria-hidden="true" tabindex="-1"></a>difMeanTp53 <span class="ot">&lt;-</span> <span class="fu">apply</span>(tp53high, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">weighted.mean</span>(x, <span class="fu">attributes</span>(tp53high)<span class="sc">$</span>weights)) <span class="sc">-</span> <span class="fu">apply</span>(tp53low, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">weighted.mean</span>(x, <span class="fu">attributes</span>(tp53low)<span class="sc">$</span>weights))</span>
<span id="cb76-5"><a href="including-clinical-variables.html#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="including-clinical-variables.html#cb76-6" aria-hidden="true" tabindex="-1"></a>changeCol <span class="ot">&lt;-</span> topNet<span class="sc">$</span>plot<span class="sc">$</span>data</span>
<span id="cb76-7"><a href="including-clinical-variables.html#cb76-7" aria-hidden="true" tabindex="-1"></a>difMeanTp53 <span class="ot">&lt;-</span> difMeanTp53[changeCol<span class="sc">$</span>name]</span>
<span id="cb76-8"><a href="including-clinical-variables.html#cb76-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-9"><a href="including-clinical-variables.html#cb76-9" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(difMeanTp53[<span class="fu">order</span>(<span class="fu">abs</span>(difMeanTp53), <span class="at">decreasing=</span>T)]), <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">&quot;difference&quot;</span>))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
difference
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MCM2
</td>
<td style="text-align:right;">
0.5113805
</td>
</tr>
<tr>
<td style="text-align:left;">
RFC4
</td>
<td style="text-align:right;">
0.4569469
</td>
</tr>
<tr>
<td style="text-align:left;">
MCM8
</td>
<td style="text-align:right;">
0.4353333
</td>
</tr>
<tr>
<td style="text-align:left;">
PRIM1
</td>
<td style="text-align:right;">
0.4136929
</td>
</tr>
<tr>
<td style="text-align:left;">
PRIM2
</td>
<td style="text-align:right;">
0.3935092
</td>
</tr>
<tr>
<td style="text-align:left;">
GINS1
</td>
<td style="text-align:right;">
0.3680252
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="including-clinical-variables.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(difMeanTp53) <span class="ot">&lt;-</span> changeCol<span class="sc">$</span>name</span>
<span id="cb77-2"><a href="including-clinical-variables.html#cb77-2" aria-hidden="true" tabindex="-1"></a>changeCol<span class="sc">$</span>color <span class="ot">&lt;-</span> difMeanTp53</span>
<span id="cb77-3"><a href="including-clinical-variables.html#cb77-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-4"><a href="including-clinical-variables.html#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Replace shape and color</span></span>
<span id="cb77-5"><a href="including-clinical-variables.html#cb77-5" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot<span class="sc">$</span>data <span class="ot">&lt;-</span> changeCol</span>
<span id="cb77-6"><a href="including-clinical-variables.html#cb77-6" aria-hidden="true" tabindex="-1"></a>topNet<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">scale_color_continuous</span>(<span class="at">low=</span><span class="st">&quot;blue&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>,<span class="at">name=</span><span class="st">&quot;difference&quot;</span>)</span></code></pre></div>
<p><img src="CBNplot_doc_files/figure-html/tp53difplotlw-1.png" width="1920" /></p>
<p><code>queryCpDistLw</code> function performs sampling by <em>likelihood weighting</em> using cpdist, and returns the data.frame with weights. It just performs <code>cpdist</code> on the queried level and produce a plot. <code>queryCpDistLs</code> performs <em>logic sampling</em>.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="including-clinical-variables.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Only likelihood weighting is supported in queryCpDist.</span></span>
<span id="cb78-2"><a href="including-clinical-variables.html#cb78-2" aria-hidden="true" tabindex="-1"></a>q1 <span class="ot">&lt;-</span> <span class="fu">queryCpDistLw</span>(topFit, <span class="fu">names</span>(difMeanTp53)[<span class="dv">1</span>], <span class="at">evidence=</span><span class="st">&quot;TP53&quot;</span>, <span class="at">level=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.5</span>,<span class="dv">1</span>))</span>
<span id="cb78-3"><a href="including-clinical-variables.html#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(q1<span class="sc">$</span>df[,<span class="fu">c</span>(<span class="fu">names</span>(difMeanTp53)[<span class="dv">1</span>],<span class="st">&quot;weights&quot;</span>)]))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:right;">
RFC2
</th>
<th style="text-align:right;">
weights
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
11.29786
</td>
<td style="text-align:right;">
0.8774882
</td>
</tr>
<tr>
<td style="text-align:right;">
10.16672
</td>
<td style="text-align:right;">
0.3570390
</td>
</tr>
<tr>
<td style="text-align:right;">
10.30900
</td>
<td style="text-align:right;">
0.2065809
</td>
</tr>
<tr>
<td style="text-align:right;">
10.82745
</td>
<td style="text-align:right;">
0.8837212
</td>
</tr>
<tr>
<td style="text-align:right;">
10.36271
</td>
<td style="text-align:right;">
0.8340634
</td>
</tr>
<tr>
<td style="text-align:right;">
10.50688
</td>
<td style="text-align:right;">
0.7633695
</td>
</tr>
</tbody>
</table>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Cancer_Genome_Atlas_Research_Network2013-cd" class="csl-entry">
Cancer Genome Atlas Research Network, John N Weinstein, Eric A Collisson, Gordon B Mills, Kenna R Mills Shaw, Brad A Ozenberger, Kyle Ellrott, Ilya Shmulevich, Chris Sander, and Joshua M Stuart. 2013. <span>“The Cancer Genome Atlas <span>Pan-Cancer</span> Analysis Project.”</span> <em>Nat. Genet.</em> 45 (10): 1113–20.
</div>
<div id="ref-Colaprico2016-xq" class="csl-entry">
Colaprico, Antonio, Tiago C Silva, Catharina Olsen, Luciano Garofano, Claudia Cava, Davide Garolini, Thais S Sabedot, et al. 2016. <span>“<span>TCGAbiolinks</span>: An <span>R/Bioconductor</span> Package for Integrative Analysis of <span>TCGA</span> Data.”</span> <em>Nucleic Acids Res.</em> 44 (8): e71.
</div>
<div id="ref-Fouad2019-bl" class="csl-entry">
Fouad, Hanan, Hosni Salem, Doha El-Sayed Ellakwa, and Manal Abdel-Hamid. 2019. <span>“<span>MMP-2</span> and <span>MMP-9</span> as Prognostic Markers for the Early Detection of Urinary Bladder Cancer.”</span> <em>J. Biochem. Mol. Toxicol.</em> 33 (4): e22275.
</div>
<div id="ref-Han2017-bi" class="csl-entry">
Han, A L, B A Veeneman, L El-Sawy, K C Day, M L Day, S A Tomlins, and E T Keller. 2017. <span>“Fibulin-3 Promotes Muscle-Invasive Bladder Cancer.”</span> <em>Oncogene</em> 36 (37): 5243–51.
</div>
<div id="ref-Kuhn2008-cc" class="csl-entry">
Kuhn, Max. 2008. <span>“Building Predictive Models in <span>R</span> Using the Caret Package.”</span> <em>Journal of Statistical Software, Articles</em> 28 (5): 1–26.
</div>
<div id="ref-Robin2011-mk" class="csl-entry">
Robin, Xavier, Natacha Turck, Alexandre Hainard, Natalia Tiberti, Frédérique Lisacek, Jean-Charles Sanchez, and Markus Müller. 2011. <span>“<span class="nocase">pROC</span>: An Open-Source Package for <span>R</span> and s+ to Analyze and Compare <span>ROC</span> Curves.”</span> <em>BMC Bioinformatics</em> 12 (March): 77.
</div>
<div id="ref-Vasala2003-jo" class="csl-entry">
Vasala, Kaija, Paavo Pääkkö, and Taina Turpeenniemi-Hujanen. 2003. <span>“Matrix Metalloproteinase-2 Immunoreactive Protein as a Prognostic Marker in Bladder Cancer.”</span> <em>Urology</em> 62 (5): 952–57.
</div>
<div id="ref-Winerdal2018-pp" class="csl-entry">
Winerdal, Malin E, David Krantz, Ciputra A Hartana, Ali A Zirakzadeh, Ludvig Linton, Emma A Bergman, Robert Rosenblatt, et al. 2018. <span>“Urinary Bladder Cancer Tregs Suppress <span>MMP2</span> and Potentially Regulate Invasiveness.”</span> <em>Cancer Immunol Res</em> 6 (5): 528–38.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bnpathplot.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="the-other-options.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CBNplot_doc.pdf", "CBNplot_doc.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
